<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Bio::Das::Lite: Perl extension for the das (http+xml) protocol (http://biodas.org/)</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Perl extension for the das (http+xml) protocol (http://biodas.org/)">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="Bio::Das::Lite (3pm) manual">
  <meta name="twitter:description" content="Perl extension for the das (http+xml) protocol (http://biodas.org/)">
  <meta name="twitter:image" content="https://www.carta.tech/images/libbio-das-lite-perl-Bio::Das::Lite-3pm.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man3pm/Bio::Das::Lite.3pm.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="Bio::Das::Lite (3pm) manual" />
  <meta property="og:description" content="Perl extension for the das (http+xml) protocol (http://biodas.org/)" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libbio-das-lite-perl-Bio::Das::Lite-3pm.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">Bio::Das::Lite<small> (3pm)</small></h1>
        <p class="lead">Perl extension for the das (http+xml) protocol (http://biodas.org/)</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/Bio::Das::Lite.3pm.html">
      <span itemprop="name">Bio::Das::Lite: Perl extension for the das (http+xml) protocol (http://biodas.org/)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libbio-das-lite-perl/">
      <span itemprop="name">libbio-das-lite-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/Bio::Das::Lite.3pm.html">
      <span itemprop="name">Bio::Das::Lite: Perl extension for the das (http+xml) protocol (http://biodas.org/)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">VERSION</h2>
        <div class="sectioncontent">

<pre>
  See $Bio::Das::Lite::VERSION
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>  use Bio::Das::Lite;   my $bdl     = Bio::Das::Lite-&gt;new_from_registry({&apos;category&apos; =&gt; &apos;GRCh_37,Chromosome,Homo sapiens&apos;});   my $results = $bdl-&gt;features(&apos;22&apos;);</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SUBROUTINES/METHODS</h2>
        <div class="sectioncontent">
<h3>new : Constructor</h3>
<p>  my $das = Bio::Das::Lite-&gt;new(&apos;http://das.ensembl.org/das/ensembl1834&apos;);</p><p>  my $das = Bio::Das::Lite-&gt;new({                                &apos;timeout&apos;    =&gt; 60,                                &apos;dsn&apos;        =&gt; &apos;http://user:pass@das.ensembl.org/das/ensembl1834&apos;,                                &apos;http_proxy&apos; =&gt; &apos;http://user:pass@webcache.local.com:3128/&apos;,                               });</p><p> Options can be: dsn        (optional scalar or array ref, URLs of DAS services)                  timeout    (optional int,      HTTP fetch timeout in seconds)                  http_proxy (optional scalar,   web cache or proxy if not set in %ENV)                  no_proxy   (optional list/ref, non-proxiable domains if not set in %ENV)                  caching    (optional bool,     primitive caching on/off)                  callback   (optional code ref, callback for processed XML blocks)                  registry   (optional array ref containing DAS registry service URLs                              defaults to &apos;http://das.sanger.ac.uk/registry/services/das&apos;)                  proxy_user (optional scalar,   username for authenticating forward-proxy)                  proxy_pass (optional scalar,   password for authenticating forward-proxy)                  user_agent (optional scalar,   User-Agent HTTP request header value)</p>
<h3>new_from_registry : Constructor</h3>
<p>  Similar to &apos;new&apos; above but supports &apos;capability&apos; and &apos;category&apos;   in the given hashref, using them to query the DAS registry and   configuring the DSNs accordingly.</p><p>  my $das = Bio::Das::Lite-&gt;new_from_registry({                                              &apos;capability&apos; =&gt; [&apos;features&apos;],                                              &apos;category&apos;   =&gt; [&apos;Protein Sequence&apos;],                                             });</p><p> Options are as above, plus                  capability OR capabilities   (optional arrayref of capabilities)                  category                     (optional arrayref of categories)</p><p>  For a complete list of capabilities and categories, see:</p><p>    http://das.sanger.ac.uk/registry/</p><p>  The category can optionally be a full coordinate system name,   allowing further restriction by authority, version and species.   For example:       &apos;Protein Sequence&apos; OR       &apos;UniProt,Protein Sequence&apos; OR       &apos;GRCh_37,Chromosome,Homo sapiens&apos;</p>
<h3>http_proxy : Get/Set http_proxy</h3>
<p>    $das-&gt;http_proxy(&apos;http://user:pass@squid.myco.com:3128/&apos;);</p>
<h3>proxy_user : Get/Set proxy username for authenticating forward-proxies</h3>
<p>  This is only required if the username wasn&apos;t specified when setting http_proxy</p><p>    $das-&gt;proxy_user(&apos;myusername&apos;);</p>
<h3>proxy_pass : Get/Set proxy password for authenticating forward-proxies</h3>
<p>  This is only required if the password wasn&apos;t specified when setting http_proxy</p><p>    $das-&gt;proxy_pass(&apos;secretpassword&apos;);</p>
<h3>no_proxy : Get/Set domains to not use proxy for</h3>
<p>    $das-&gt;no_proxy(&apos;ebi.ac.uk&apos;, &apos;localhost&apos;);     OR     $das-&gt;no_proxy( [&apos;ebi.ac.uk&apos;, &apos;localhost&apos;] );</p><p>    Always returns an arrayref</p>
<h3>user_agent : Get/Set user-agent for request headers</h3>
<p>    $das-&gt;user_agent(&apos;GroovyDAS/1.0&apos;);</p>
<h3>timeout : Get/Set timeout</h3>
<p>    $das-&gt;timeout(30);</p>
<h3>caching : Get/Set caching</h3>
<p>    $das-&gt;<strong>caching</strong>(1);</p>
<h3>callback : Get/Set callback code ref</h3>
<p>    $das-&gt;callback(sub { });</p>
<h3>basename : Get base \s-1URL\s0(s) of service</h3>
<p>    $das-&gt;basename(optional $dsn);</p>
<h3>dsn : Get/Set \s-1DSN\s0</h3>
<p>  $das-&gt;dsn(&apos;http://das.ensembl.org/das/ensembl1834/&apos;); # give dsn (scalar or arrayref) here if not specified in new()</p><p>  Or, if you want to add to the existing dsn list and you&apos;re feeling sneaky...</p><p>  push @{$das-&gt;dsn}, &apos;http://my.server/das/additionalsource&apos;;</p>
<h3>dsns : Retrieve information about other sources served from this server.</h3>
<p> Note this call is &apos;dsns&apos;, as differentiated from &apos;dsn&apos; which is the current configured source</p><p>  my $src_data = $das-&gt;dsns();</p>
<h3>entry_points : Retrieve the list of entry_points for this source</h3>
<p>  e.g. chromosomes and associated information (e.g. sequence length and version)</p><p>  my $entry_points  = $das-&gt;entry_points();</p>
<h3>Types of argument for 'types', 'features', 'sequence' calls:</h3>
<p>  Segment Id:   &apos;1&apos;</p><p>  Segment Id with range:   &apos;1:1,1000&apos;</p><p>  Segment Id with range and type:   {     &apos;segment&apos; =&gt; &apos;1:1,1000&apos;,     &apos;type&apos;    =&gt; &apos;exon&apos;,   }</p><p>  Multiple Ids with ranges and types:   [     {       &apos;segment&apos; =&gt; &apos;1:1,1000&apos;,       &apos;type&apos;    =&gt; &apos;exon&apos;,     },     {       &apos;segment&apos; =&gt; &apos;2:1,1000&apos;,       &apos;type&apos;    =&gt; &apos;exon&apos;,     },   ]</p><p>  See DAS specifications for other parameters</p>
<h3>types : Find out about different data types available from this source</h3>
<p>  my $types         = $das-&gt;types(); # takes optional args - see DAS specs</p><p> Retrieve the types of data available for this source  e.g. 32k_cloneset, karyotype, swissprot</p>
<h3>features : Retrieve features from a segment</h3>
<p>   e.g. clones on a chromosome</p><p>  #########   # Different ways to fetch features -   #   my $feature_data1 = $das-&gt;features(&apos;1:1,100000&apos;);   my $feature_data2 = $das-&gt;features([&apos;1:1,100000&apos;, &apos;2:20435000,21435000&apos;]);   my $feature_data3 = $das-&gt;features({                                       &apos;segment&apos; =&gt; &apos;1:1,1000&apos;,                                       &apos;type&apos;    =&gt; &apos;karyotype&apos;,                                       # optional args - see DAS Spec                                      });   my $feature_data4 = $das-&gt;features([                                       {&apos;segment&apos;  =&gt; &apos;1:1,1000000&apos;,&apos;type&apos; =&gt; &apos;karyotype&apos;,},                                       {&apos;segment&apos;  =&gt; &apos;2:1,1000000&apos;,},                                       {&apos;group_id&apos; =&gt; &apos;OTTHUMG00000036084&apos;,},                                      ]);</p><p>  #########   # Feature fetch with callback   #   my $callback = sub {                       my $struct = shift;                       print {*STDERR} Dumper($struct);                      };   # then:   $das-&gt;callback($callback);   $das-&gt;features(&apos;1:1,1000000&apos;);</p><p>  # or:   $das-&gt;features(&apos;1:1,1000000&apos;, $callback);</p><p>  # or:   $das-&gt;features([&apos;1:1,1000000&apos;, &apos;2:1,1000000&apos;, &apos;3:1,1000000&apos;], $callback);</p><p>  # or:   $das-&gt;features([{&apos;group_id&apos; =&gt; &apos;OTTHUMG00000036084&apos;}, &apos;2:1,1000000&apos;, &apos;3:1,1000000&apos;], $callback);</p>
<h3>alignment : Retrieve protein alignment data for a query.  This can be a multiple sequence alignment or pairwise alignment.  Note - this has not been tested for structural alignments as there is currently no Das source avialable.</h3>
<p>  my $alignment = $das-&gt;alignment({query =&gt; &apos;Q01234&apos;});</p>
<h3>structure : Retrieve known structure (i.e. \s-1PDB\s0) for a query</h3>
<p>  my $structure = $das-&gt;structure({ query =&gt; &apos;pdb_id&apos;});</p>
<h3>sources : Retrieves the list of sources form the \s-1DAS\s0 registry, via a \s-1DAS\s0 call.</h3>
<p>  my $sources = $das-&gt;source;</p>
<h3>sequence : Retrieve sequence data for a segment (probably dna or protein)</h3>
<p>  my $sequence      = $das-&gt;sequence(&apos;2:1,1000&apos;); # segment:start,stop (e.g. chromosome 2, bases 1 to 1000)</p>
<h3>stylesheet : Retrieve stylesheet data</h3>
<p>  my $style_data    = $das-&gt;stylesheet();   my $style_data2   = $das-&gt;stylesheet($callback);</p>
<h3>statuscodes : Retrieve \s-1HTTP\s0 status codes for request URLs</h3>
<p>  my $code         = $das-&gt;statuscodes($url);   my $code_hashref = $das-&gt;statuscodes();</p>
<h3>max_hosts set number of running concurrent host connections</h3>
<p>  THIS METHOD IS NOW DEPRECATED AND HAS NO EFFECT</p><p>  $das-&gt;<strong>max_hosts</strong>(7);   print $das-&gt;max_hosts();</p>
<h3>max_req set number of running concurrent requests per host</h3>
<p>  THIS METHOD IS NOW DEPRECATED AND HAS NO EFFECT</p><p>  $das-&gt;<strong>max_req</strong>(5);   print $das-&gt;max_req();</p>
<h3>registry : Get/Set accessor for DAS-Registry service URLs</h3>
<p>  $biodaslite-&gt;registry(&apos;http://www.dasregistry.org/das&apos;);</p><p>  my $registry_arrayref = $biodaslite-&gt;registry();</p>
<h3>registry_sources : Arrayref of dassource objects from the configured registry services</h3>
<p>  my $sources_ref = $biodaslite-&gt;registry_sources();</p><p>  my $sources_ref = $biodaslite-&gt;registry_sources({     &apos;capability&apos; =&gt; [&apos;features&apos;,&apos;stylesheet&apos;],   });</p><p>  my $sources_ref = $biodaslite-&gt;registry_sources({     &apos;category&apos; =&gt; [&apos;Protein Sequence&apos;],   });</p>
<h3>build_queries</h3>
<p>Constructs an arrayref of \s-1DAS\s0 requests including parameters for each call</p>
<h3>build_requests</h3>
<p>Constructs the WWW::Curl callbacks</p>
<h3>postprocess</h3>
<p>Applies processing to the result set, e.g. removal of whitespace from sequence responses.</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This module is an implementation of a client for the \s-1DAS\s0 protocol (\s-1XML\s0 over \s-1HTTP\s0 primarily for biological-data).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DEPENDENCIES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    strict
  </dt>
  <dd>
    
  </dd>
  <dt>
    warnings
  </dt>
  <dd>
    
  </dd>
  <dt>
    WWW::Curl
  </dt>
  <dd>
    
  </dd>
  <dt>
    HTTP::Response
  </dt>
  <dd>
    
  </dd>
  <dt>
    Carp
  </dt>
  <dd>
    
  </dd>
  <dt>
    English
  </dt>
  <dd>
    
  </dd>
  <dt>
    Readonly
  </dt>
  <dd>
    
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DIAGNOSTICS</h2>
        <div class="sectioncontent">
<p>  Set $Bio::Das::Lite::DEBUG = 1;</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">CONFIGURATION AND ENVIRONMENT</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">INCOMPATIBILITIES</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS AND LIMITATIONS</h2>
        <div class="sectioncontent">
<p>  The max_req and max_hosts methods are now deprecated and have no effect.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO Bio::Das::Lite&hellip;</h2>
        <div class="sectioncontent">
<p>\s-1DAS\s0 Specifications at: http://biodas.org/documents/spec.html</p><p>ProServer (A \s-1DAS\s0 Server implementation also by the author) at:</p>
<pre>
   http://www.sanger.ac.uk/proserver/
</pre>
<p>The venerable Bio::Das suite (\s-1CPAN\s0 and http://www.biodas.org/download/Bio::Das/).</p><p>The \s-1DAS\s0 Registry at:</p>
<pre>
   http://das.sanger.ac.uk/registry/
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Roger Pettett, &lt;rpettett@cpan.org&gt;</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">LICENSE AND COPYRIGHT</h2>
        <div class="sectioncontent">
<p>Copyright (C) 2007 \s-1GRL\s0, by Roger Pettett</p><p>This library is free software; you can redistribute it and/or modify it under the same terms as Perl itself, either Perl version 5.8.4 or, at your option, any later version of Perl 5 you may have available.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="Bio::Chado::Schema::Util.3pm.html"><span aria-hidden="true">&larr;</span> Bio::Chado::Schema::Util.3pm: Utility functions shared by bio::chado::schema objects</a></li>
   <li class="next"><a href="Bit::Vector::Minimal.3pm.html">Bit::Vector::Minimal.3pm: Object-oriented wrapper around vec() <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
