<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Boulder::Genbank: Fetch genbank data records as parsed boulder stones</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Fetch genbank data records as parsed boulder stones">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="Boulder::Genbank (3pm) manual">
  <meta name="twitter:description" content="Fetch genbank data records as parsed boulder stones">
  <meta name="twitter:image" content="https://www.carta.tech/images/libboulder-perl-Boulder::Genbank-3pm.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man3pm/Boulder::Genbank.3pm.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="Boulder::Genbank (3pm) manual" />
  <meta property="og:description" content="Fetch genbank data records as parsed boulder stones" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libboulder-perl-Boulder::Genbank-3pm.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">Boulder::Genbank<small> (3pm)</small></h1>
        <p class="lead">Fetch genbank data records as parsed boulder stones</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/Boulder::Genbank.3pm.html">
      <span itemprop="name">Boulder::Genbank: Fetch genbank data records as parsed boulder stones</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libboulder-perl/">
      <span itemprop="name">libboulder-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/Boulder::Genbank.3pm.html">
      <span itemprop="name">Boulder::Genbank: Fetch genbank data records as parsed boulder stones</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">

<pre>
  use Boulder::Genbank

  # network access via Entrez
   $gb = Boulder::Genbank-&gt;newFh( qw(M57939 M28274 L36028) );

   while ($data = &lt;$gb&gt;) {
       print $data-&gt;Accession;

       @introns = $data-&gt;features-&gt;Intron;
       print "There are ",scalar(@introns)," introns.&#92;n";
       $dna = $data-&gt;Sequence;
       print "The dna is ",length($dna)," bp long.&#92;n";

       my @features = $data-&gt;features(-type=&gt;[ qw(Exon Source Satellite) ],
                                      -pos=&gt;[90,310] );
       foreach (@features) {
          print $_-&gt;Type,"&#92;n";
          print $_-&gt;Position,"&#92;n";
          print $_-&gt;Gene,"&#92;n";
      }
    }

  # another syntax
  $gb = new Boulder::Genbank(-accessor=&gt;&apos;Entrez&apos;,
                             -fetch =&gt; [qw/M57939 M28274 L36028/]);

  # local access via Yank
  $gb = new Boulder::Genbank(-accessor=&gt;&apos;Yank&apos;,
                             -fetch=&gt;[qw/M57939 M28274 L36028/]);
  while (my $s = $gb-&gt;get) {
     # etc.
  }

  # parse a file of Genbank records
  $gb = new Boulder::Genbank(-accessor=&gt;&apos;File&apos;,
                             -fetch =&gt; &apos;/usr/local/db/gbpri3.seq&apos;);
  while (my $s = $gb-&gt;get) {
     # etc.
  }

  # parse flatfile records yourself
  open (GB,"/usr/local/db/gbpri3.seq");
  local $/ = "//&#92;n";
  while (&lt;GB&gt;) {
     my $s = Boulder::Genbank-&gt;parse($_);
     # etc.
  }
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>Boulder::Genbank provides retrieval and parsing services for \s-1NCBI\s0 Genbank-format records.  It returns Genbank entries in Stone format, allowing easy access to the various fields and values. Boulder::Genbank is a descendent of Boulder::Stream, and provides a stream-like interface to a series of Stone objects.</p><p>&gt;&gt; \s-1IMPORTANT\s0 \s-1NOTE\s0 &lt;&lt;</p><p>As of January 2002, \s-1NCBI\s0 has changed their Batch Entrez interface.  I have modified Boulder::Genbank so as to use a \*(L"demo\*(R" interface, which fixes things, but this isn't guaranteed in the long run.</p><p>I have written to \s-1NCBI\s0, and they may fix this \*(-- or they may not.</p><p>&gt;&gt; \s-1IMPORTANT\s0 \s-1NOTE\s0 &lt;&lt;</p><p>Access to Genbank is provided by three different <em>accessors</em>, which together give access to remote and local Genbank databases.  When you create a new Boulder::Genbank stream, you provide one of the three accessors, along with accessor-specific parameters that control what entries to fetch.  The three accessors are:</p>
<dl class='dl-vertical'>
  <dt>
    Entrez
  </dt>
  <dd>
    <p>This provides access to NetEntrez, accessing the most recent Genbank information directly from \s-1NCBI\s0's Web site.  The parameters passed to this accessor are either a series of Genbank accession numbers, or an Entrez query (see http://www.ncbi.nlm.nih.gov/Entrez/linking.html). If you provide a list of accession numbers, the stream will return a series of stones corresponding to the numbers.  Otherwise, if you provided an Entrez query, the entries returned will be in the order returned by Entez.</p>
  </dd>
  <dt>
    File
  </dt>
  <dd>
    <p>This provides access to local Genbank entries by reading from a flat file (typically one of the .seq files downloadable from \s-1NCBI\s0's Web site). The stream will return a Stone corresponding to each of the entries in the file, starting from the top of the file and working downward.  The parameter in this case is the path to the local file.</p>
  </dd>
  <dt>
    Yank
  </dt>
  <dd>
    <p>This provides access to local Genbank entries using Will Fitzhugh's Yank program.  Yank provides fast indexed access to a Genbank flat file using the accession number as the key.  The parameter passed to the Yank accessor is a list of accession numbers.  Stones will be returned in the requested order.  By default the yank binary lives in /usr/local/bin/yank.  To support other locations, you may define the environment variable \s-1YANK\s0 to contain the full path.</p>
  </dd>

</dl>
<p>It is also possible to parse a single Genbank entry from a text string stored in a scalar variable, returning a Stone object.</p><h3>Boulder::Genbank methods</h3>
<p>This section lists the public methods that the <em>Boulder::Genbank</em> class makes available.</p>
<dl class='dl-vertical'>
  <dt>
    <em>new()</em>
  </dt>
  <dd>
    <p>   # Network fetch via Entrez, with accession numbers    $gb=new Boulder::Genbank(-accessor  =&gt;  &apos;Entrez&apos;,                             -fetch     =&gt;  [qw/M57939 M28274 L36028/]);</p><p>   # Same, but shorter and uses -&gt; operator    $gb = Boulder::Genbank-&gt;new qw(M57939 M28274 L36028);</p><p>   # Network fetch via Entrez, with a query</p><p>   # Network fetch via Entrez, with a query    $query = &apos;Homo sapiens[Organism] AND EST[Keyword]&apos;;    $gb=new Boulder::Genbank(-accessor  =&gt;  &apos;Entrez&apos;,                             -fetch     =&gt;  $query);</p><p>   # Local fetch via Yank, with accession numbers    $gb=new Boulder::Genbank(-accessor  =&gt;  &apos;Yank&apos;,                             -fetch     =&gt;  [qw/M57939 M28274 L36028/]);</p><p>   # Local fetch via File    $gb=new Boulder::Genbank(-accessor  =&gt;  &apos;File&apos;,                             -fetch     =&gt;  &apos;/usr/local/genbank/gbpri3.seq&apos;); The <em>new()</em> method creates a new <em>Boulder::Genbank</em> stream on the accessor provided.  The three possible accessors are <strong>Entrez</strong>, <strong>Yank</strong> and <strong>File</strong>.  If successful, the method returns the stream object.  Otherwise it returns undef. <em>new()</em> takes the following arguments:         -accessor       Name of the accessor to use         -fetch          Parameters to pass to the accessor         -proxy          Path to an HTTP proxy, used when using                          the Entrez accessor over a firewall. Specify the accessor to use with the <strong>-accessor</strong> argument.  If not specified, it defaults to <strong>Entrez</strong>. <strong>-fetch</strong> is an accessor-specific argument.  The possibilities are: For <strong>Entrez</strong>, the <strong>-fetch</strong> argument may point to a scalar, in which case it is interpreted as an Entrez query string.  See http://www.ncbi.nlm.nih.gov/Entrez/linking.html for a description of the query syntax.  Alternatively, <strong>-fetch</strong> may point to an array reference, in which case it is interpreted as a list of accession numbers to retrieve.  If <strong>-fetch</strong> points to a hash, it is interpreted as extended information.  See \*(L"Extended Entrez Parameters\*(R" below. For <strong>Yank</strong>, the <strong>-fetch</strong> argument must point to an array reference containing the accession numbers to retrieve. For <strong>File</strong>, the <strong>-fetch</strong> argument must point to a string-valued scalar, which will be interpreted as the path to the file to read Genbank entries from. For Entrez (and Entrez only) Boulder::Genbank allows you to use a shortcut syntax in which you provde <em>new()</em> with a list of accession numbers:   $gb = new Boulder::Genbank(&apos;M57939&apos;,&apos;M28274&apos;,&apos;L36028&apos;);</p>
  </dd>
  <dt>
    <em>newFh()</em>
  </dt>
  <dd>
    <p>This works like <em>new()</em>, but returns a filehandle.  To recover each GenBank record read from the filehandle with the &lt;&gt; operator:   $fh = Boulder::GenBank-&gt;newFh(&apos;M57939&apos;,&apos;M28274&apos;,&apos;L36028&apos;);   while ($record = &lt;$fh&gt;) {      print $record-&gt;asString;   }</p>
  </dd>
  <dt>
    <em>get()</em>
  </dt>
  <dd>
    <p>The <em>get()</em> method is inherited from <em>Boulder::Stream</em>, and simply returns the next parsed Genbank Stone, or undef if there is nothing more to fetch.  It has the same semantics as the parent class, including the ability to restrict access to certain top-level tags. The object returned is a Stone::GB_Sequence object, which is a descendent of Stone.</p>
  </dd>
  <dt>
    <em>put()</em>
  </dt>
  <dd>
    <p>The <em>put()</em> method is inherited from the parent Boulder::Stream class, and will write the passed Stone to standard output in Boulder format. This means that it is currently not possible to write a Boulder::Genbank object back into Genbank flatfile form.</p>
  </dd>

</dl>

<h3>Extended Entrez Parameters</h3>
<p>The Entrez accessor recognizes extended parameters that allow you the ability to customize the search.  Instead of passing a query string scalar or a list of accession numbers as the <strong>-fetch</strong> argument, pass a hash reference.  The hashref should contain one or more of the following keys:</p>
<dl class='dl-vertical'>
  <dt>
    <strong>-query</strong>
  </dt>
  <dd>
    <p>The Entrez query to process.</p>
  </dd>
  <dt>
    <strong>-accession</strong>
  </dt>
  <dd>
    <p>The list of accession numbers to fetch, as an array ref.</p>
  </dd>
  <dt>
    <strong>-db</strong>
  </dt>
  <dd>
    <p>The database to search.  This is a single-letter database code selected from the following list:   m  MEDLINE   p  Protein   n  Nucleotide   s  Popset</p>
  </dd>
  <dt>
    <strong>-proxy</strong>
  </dt>
  <dd>
    <p>An \s-1HTTP\s0 proxy to use.  For example:    -proxy =&gt; http://www.firewall.com:9000 If you think you need this, get the correct \s-1URL\s0 from your system administrator.</p>
  </dd>

</dl>
<p>As an example, here's how to search for ESTs from Oryza sativa that have been entered or modified since 1999.</p><p>  my $gb = new Boulder::Genbank( -accessor=&gt;Entrez,                                  -query=&gt;&apos;Oryza sativa[Organism] AND EST[Keyword] AND 1999[MDAT]&apos;,                                  -db   =&gt; &apos;n&apos;                                 });</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">METHODS DEFINED BY THE GENBANK STONE OBJECT</h2>
        <div class="sectioncontent">
<p>Each record returned from the Boulder::Genbank stream defines a set of methods that correspond to features and other fields in the Genbank flat file record.  Stone::GB_Sequence gives the full details, but they are listed for reference here: Get the length of the sequence. Get the start position of the sequence, currently always \*(L"1\*(R". Get the end position of the sequence, currently always the same as the length. <em>features()</em> will search the entry feature list for those features that meet certain criteria.  The criteria are specified using the <strong>-pos</strong> and/or <strong>-type</strong> argument names, as shown below.</p>
<dl class='dl-vertical'>
  <dt>
    -pos
  </dt>
  <dd>
    <p>Provide a position or range of positions which the feature must <strong>overlap</strong>.  A single position is specified in this way:    -pos =&gt; 1500;         # feature must overlap postion 1500 or a range of positions in this way:    -pos =&gt; [1000,1500];  # 1000 to 1500 inclusive If no criteria are provided, then <em>features()</em> returns all the features, and is equivalent to calling the <em>Features()</em> accessor.</p>
  </dd>
  <dt>
    -type, -types
  </dt>
  <dd>
    <p>Filter the list of features by type or a set of types.  Matches are case-insensitive, so \*(L"exon\*(R", \*(L"Exon\*(R" and \*(L"\s-1EXON\s0\*(R" are all equivalent. You may call with a single type as in:    -type =&gt; &apos;Exon&apos; or with a list of types, as in    -types =&gt; [&apos;Exon&apos;,&apos;CDS&apos;] The names \*(L"-type\*(R" and \*(L"-types\*(R" can be used interchangeably. Returns a Bio::Seq object from the Bioperl project.  Dies with an error message unless the Bio::Seq module is installed.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OUTPUT TAGS</h2>
        <div class="sectioncontent">
<p>The tags returned by the parsing operation are taken from the \s-1NCBI\s0 \s-1ASN\s0.1 schema.  For consistency, they are normalized so that the initial letter is capitalized, and all subsequent letters are lowercase.  This section contains an abbreviated list of the most useful/common tags.  See \*(L"The \s-1NCBI\s0 Data Model\*(R", by James Ostell and Jonathan Kans in \*(L"Bioinformatics: A Practical Guide to the Analysis of Genes and Proteins\*(R" (Eds. A. Baxevanis and F. Ouellette), pp 121-144 for the full listing.</p><h3>Top-Level Tags</h3>
<p>These are tags that appear at the top level of the parsed Genbank entry.</p>
<dl class='dl-vertical'>
  <dt>
    Accession
  </dt>
  <dd>
    <p>The accession number of this entry.  Because of the vagaries of the Genbank data model, an entry may have multiple accession numbers (e.g. after a merging operation).  Accession may therefore be a multi-valued tag. Example:       my $accessionNo = $s-&gt;Accession;</p>
  </dd>
  <dt>
    Authors
  </dt>
  <dd>
    <p>The list of authors, as they appear on the \s-1AUTHORS\s0 line of the Genbank record.  No attempt is made to parse them into individual authors.</p>
  </dd>
  <dt>
    Basecount
  </dt>
  <dd>
    <p>The nucleotide basecount for the entry.  It is presented as a Boulder Stone with keys \*(L"a\*(R", \*(L"c\*(R", \*(L"t\*(R" and \*(L"g\*(R".  Example:      my $A = $s-&gt;Basecount-&gt;A;      my $C = $s-&gt;Basecount-&gt;C;      my $G = $s-&gt;Basecount-&gt;G;      my $T = $s-&gt;Basecount-&gt;T;      print "GC content is ",($G+$C)/($A+$C+$G+$T),"&#92;n";</p>
  </dd>
  <dt>
    Blob
  </dt>
  <dd>
    <p>The entire flatfile record as an unparsed chunk of text (a \*(L"blob\*(R"). This is a handy way of reassembling the record for human inspection.</p>
  </dd>
  <dt>
    Comment
  </dt>
  <dd>
    <p>The \s-1COMMENT\s0 line from the Genbank record.</p>
  </dd>
  <dt>
    Definition
  </dt>
  <dd>
    <p>The \s-1DEFINITION\s0 line from the Genbank record, unmodified.</p>
  </dd>
  <dt>
    Features
  </dt>
  <dd>
    <p>The \s-1FEATURES\s0 table.  This is a complex stone object with multiple subtags.  See the \*(L"The Features Tag\*(R" for details.</p>
  </dd>
  <dt>
    Journal
  </dt>
  <dd>
    <p>The \s-1JOURNAL\s0 line from the Genbank record, unmodified.</p>
  </dd>
  <dt>
    Keywords
  </dt>
  <dd>
    <p>The \s-1KEYWORDS\s0 line from the Genbank record, unmodified.  No attempt is made to parse the keywords into separate values. Example:     my $keywords = $s-&gt;Keywords</p>
  </dd>
  <dt>
    Locus
  </dt>
  <dd>
    <p>The \s-1LOCUS\s0 line from the Genbank record.  It is not further parsed.</p>
  </dd>
  <dt>
    Medline, Nid
  </dt>
  <dd>
    <p>References to other database accession numbers.</p>
  </dd>
  <dt>
    Organism
  </dt>
  <dd>
    <p>The taxonomic name of the organism from which this entry was derived. This line is taken from the Genbank entry unmodified.  See the \s-1NCBI\s0 data model documentation for an explanation of their taxonomic syntax.</p>
  </dd>
  <dt>
    Reference
  </dt>
  <dd>
    <p>The \s-1REFERENCE\s0 line from the Genbank entry.  There are often multiple Reference lines.  Example:   my @references = $s-&gt;Reference;</p>
  </dd>
  <dt>
    Sequence
  </dt>
  <dd>
    <p>The \s-1DNA\s0 or \s-1RNA\s0 sequence of the entry.  This is presented as a single lower-case string, with all base numbers and formatting characters removed.</p>
  </dd>
  <dt>
    Source
  </dt>
  <dd>
    <p>The entry's \s-1SOURCE\s0 field; often giving clues on how the sequencing was performed.</p>
  </dd>
  <dt>
    Title
  </dt>
  <dd>
    <p>The \s-1TITLE\s0 field from the paper describing this entry, if any.</p>
  </dd>

</dl>

<h3>The Features Tag</h3>
<p>The Features tag points to a Stone record that contains multiple subtags.  Each subtag is the name of a feature which points, in turn, to a Stone that describes the feature's location and other attributes. The full list of feature is beyond this document, but the following are the features that are most often seen:</p><p>        Cds             a CDS         Intron          an intron         Exon            an exon         Gene            a gene         Mrna            an mRNA         Polya_site      a putative polyadenylation signal         Repeat_unit     a repetitive region         Source          More information about the organism and cell                         type the sequence was derived from         Satellite       a microsatellite (dinucleotide repeat)</p><p>Each feature will contain one or more of the following subtags:</p>
<dl class='dl-vertical'>
  <dt>
    DB_xref
  </dt>
  <dd>
    <p>A cross-reference to another database in the form DB_NAME:accession_number.  See the \s-1NCBI\s0 Web site for a description of these cross references.</p>
  </dd>
  <dt>
    Evidence
  </dt>
  <dd>
    <p>The evidence for this feature, either \*(L"experimental\*(R" or \*(L"predicted\*(R".</p>
  </dd>
  <dt>
    Gene
  </dt>
  <dd>
    <p>If the feature involves a gene, this will be the gene's name (or one of its names).  This subtag is often seen in \*(L"Gene\*(R" and Cds features. Example:         foreach ($s-&gt;Features-&gt;Cds) {            my $gene = $_-&gt;Gene;            my $position = $_-&gt;Position;            Print "Gene $gene ($position)&#92;n";         }</p>
  </dd>
  <dt>
    Map
  </dt>
  <dd>
    <p>If the feature is mapped, this provides a map position, usually as a cytogenetic band.</p>
  </dd>
  <dt>
    Note
  </dt>
  <dd>
    <p>A grab-back for various text notes.</p>
  </dd>
  <dt>
    Number
  </dt>
  <dd>
    <p>When multiple features of this type occur, this field is used to number them.  Ordinarily this field is not needed because Boulder::Genbank preserves the order of features.</p>
  </dd>
  <dt>
    Organism
  </dt>
  <dd>
    <p>If the feature is Source, this provides the source organism.</p>
  </dd>
  <dt>
    Position
  </dt>
  <dd>
    <p>The position of this feature, usually expresed as a range (1970..1975).</p>
  </dd>
  <dt>
    Product
  </dt>
  <dd>
    <p>The protein product of the feature, if applicable, as a text string.</p>
  </dd>
  <dt>
    Translation
  </dt>
  <dd>
    <p>The protein translation of the feature, if applicable.</p>
  </dd>

</dl>


        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO Boulder::Genbank&hellip;</h2>
        <div class="sectioncontent">
<p>Boulder, Boulder::Blast</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Lincoln Stein &lt;lstein@cshl.org&gt;.</p><p>Copyright (c) 1997-2000 Lincoln D. Stein</p><p>This library is free software; you can redistribute it and/or modify it under the same terms as Perl itself.  See \s-1DISCLAIMER\s0.txt for disclaimers of warranty.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLE GENBANK OBJECT</h2>
        <div class="sectioncontent">
<p>The following is an excerpt from a moderately complex Genbank Stone. The Sequence line and several other long lines have been truncated for readability.</p><p> Authors=Spritz,R.A., Strunk,K., Surowy,C.S.O., Hoch,S., Barton,D.E. and Francke,U.  Authors=Spritz,R.A., Strunk,K., Surowy,C.S. and Mohrenweiser,H.W.  Locus=HUMRNP7011   2155 bp    DNA             PRI       03-JUL-1991  Accession=M57939  Accession=J04772  Accession=M57733  Keywords=ribonucleoprotein antigen.  Sequence=aagcttttccaggcagtgcgagatagaggagcgcttgagaaggcaggttttgcagcagacggcagtgacagcccag...  Definition=Human small nuclear ribonucleoprotein (U1-70K) gene, exon 10 and 11.  Journal=Nucleic Acids Res. 15, 10373-10391 (1987)  Journal=Genomics 8, 371-379 (1990)  Nid=g337441  Medline=88096573  Medline=91065657  Features={    Polya_site={      Evidence=experimental      Position=1989      Gene=U1-70K    }    Polya_site={      Position=1990      Gene=U1-70K    }    Polya_site={      Evidence=experimental      Position=1992      Gene=U1-70K    }    Polya_site={      Evidence=experimental      Position=1998      Gene=U1-70K    }    Source={      Organism=Homo sapiens      Db_xref=taxon:9606      Position=1..2155      Map=19q13.3    }    Cds={      Codon_start=1      Product=ribonucleoprotein antigen      Db_xref=PID:g337445      Position=join(M57929:329..475,M57930:183..245,M57930:358..412, ...      Gene=U1-70K      Translation=MTQFLPPNLLALFAPRDPIPYLPPLEKLPHEKHHNQPYCGIAPYIREFEDPRDAPPPTR...    }    Cds={      Codon_start=1      Product=ribonucleoprotein antigen      Db_xref=PID:g337444      Evidence=experimental      Position=join(M57929:329..475,M57930:183..245,M57930:358..412, ...      Gene=U1-70K      Translation=MTQFLPPNLLALFAPRDPIPYLPPLEKLPHEKHHNQPYCGIAPYIREFEDPR...    }    Polya_signal={      Position=1970..1975      Note=putative      Gene=U1-70K    }    Intron={      Evidence=experimental      Position=1100..1208      Gene=U1-70K    }    Intron={      Number=10      Evidence=experimental      Position=1100..1181      Gene=U1-70K    }    Intron={      Number=9      Evidence=experimental      Position=order(M57937:702..921,1..1011)      Note=2.1 kb gap      Gene=U1-70K    }    Intron={      Position=order(M57935:272..406,M57936:1..284,M57937:1..599, &lt;1..&gt;1208)      Gene=U1-70K    }    Intron={      Evidence=experimental      Position=order(M57935:284..406,M57936:1..284,M57937:1..599, &lt;1..&gt;1208)      Note=first gap-0.14 kb, second gap-0.62 kb      Gene=U1-70K    }    Intron={      Number=8      Evidence=experimental      Position=order(M57935:272..406,M57936:1..284,M57937:1..599, &lt;1..&gt;1181)      Note=first gap-0.14 kb, second gap-0.62 kb      Gene=U1-70K    }    Exon={      Number=10      Evidence=experimental      Position=1012..1099      Gene=U1-70K    }    Exon={      Number=11      Evidence=experimental      Position=1182..(1989.1998)      Gene=U1-70K    }    Exon={      Evidence=experimental      Position=1209..(1989.1998)      Gene=U1-70K    }    Mrna={      Product=ribonucleoprotein antigen      Position=join(M57928:358..668,M57929:319..475,M57930:183..245, ...      Gene=U1-70K    }    Mrna={      Product=ribonucleoprotein antigen      Citation=[2]      Evidence=experimental      Position=join(M57928:358..668,M57929:319..475,M57930:183..245, ...      Gene=U1-70K    }    Gene={      Position=join(M57928:207..719,M57929:1..562,M57930:1..577, ...      Gene=U1-70K    }  }  Reference=1  (sites)  Reference=2  (bases 1 to 2155)  =</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="Boulder::Blast::WU.3pm.html"><span aria-hidden="true">&larr;</span> Boulder::Blast::WU.3pm: Parse and read wu-blast files</a></li>
   <li class="next"><a href="Boulder::LocusLink.3pm.html">Boulder::LocusLink.3pm: Fetch locuslink data records as parsed boulder stones <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
