<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Genome::Model::Tools::Music::PathScan::PathScan: The path-scan significance test for mutations in groups of putative cancer genes</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="The path-scan significance test for mutations in groups of putative cancer genes">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="Genome::Model::Tools::Music::PathScan::PathScan (3pm) manual">
  <meta name="twitter:description" content="The path-scan significance test for mutations in groups of putative cancer genes">
  <meta name="twitter:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-Genome::Model::Tools::Music::PathScan::PathScan-3pm.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man3pm/Genome::Model::Tools::Music::PathScan::PathScan.3pm.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="Genome::Model::Tools::Music::PathScan::PathScan (3pm) manual" />
  <meta property="og:description" content="The path-scan significance test for mutations in groups of putative cancer genes" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-Genome::Model::Tools::Music::PathScan::PathScan-3pm.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">Genome::Model::Tools::Music::PathScan::PathScan<small> (3pm)</small></h1>
        <p class="lead">The path-scan significance test for mutations in groups of putative cancer genes</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/Genome::Model::Tools::Music::PathScan::PathScan.3pm.html">
      <span itemprop="name">Genome::Model::Tools::Music::PathScan::PathScan: The path-scan significance test for mutations in groups of putative cancer genes</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libgenome-model-tools-music-perl/">
      <span itemprop="name">libgenome-model-tools-music-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/Genome::Model::Tools::Music::PathScan::PathScan.3pm.html">
      <span itemprop="name">Genome::Model::Tools::Music::PathScan::PathScan: The path-scan significance test for mutations in groups of putative cancer genes</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">

<pre>
        use PathScan;

        my $pmobj = PathScan-&gt;new ($list_of_gene_lengths);
        my $pval = $pmobj-&gt;path_scan ($actual_hits, $background_mutation_rate);
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This package calculates the so-called path-scan statistic P-value for sets of putative cancer genes under the null hypothesis that somatic mutations found in data are the result of a random process characterized by the background mutation rate. This test is applied to, for example, a biologically-relevant group of genes, say all the genes in a particular pathway, for which somatic mutation data are available. A low p-value would imply that the null hypothesis should be rejected. In other words, the result suggests that the mutation configuration in this pathway is probably not the result of a strictly random process.</p><h3>Nature of the Path-Scan Test</h3>
<p>This statistic considers individual genes in a \*(L"binary\*(R" fashion, i.e. a gene is either mutated (has one or more mutations) or it is not mutated. <em>The number of mutations in a mutated gene is not</em> considered. This is the \*(L"path-scan\*(R" aspect of the test.</p><p>Why is such information discarded? The somatic background mutation rate is typically very small compared to the size of the average gene. Consequently, the expected number of mutations in any given gene is very low, much less than one, in fact. Under the null hypothesis, most genes will have no mutations. Genes with one (or just a few) may be interesting, but when many genes in a biologically-relevant group (say a pathway) have one (or just a few) mutations, that could be a sign of some underlying <em>non-random</em> process. In other words, this test is useful in cases where many genes in a group might each contribute a small component (i.e. a small fitness advantage) in the context of the disease process. What this test is not concerned with (and will <em>not</em> detect) is the case where a single, specific gene has a non-random association and it reflects this fact via a large number of mutations. Other single-gene tests should presumably flag such cases. The path-scan test should, therefore, be thought of as just one tool within a larger statistical \*(L"toolbox\*(R".</p>
<h3>Assumptions in the Test</h3>
<p>The main assumption is that a single background mutation rate applies to the set of genes of interest. That is, the rate does not vary among genes, among chromosomes (if more than one hosts genes of interest), etc.</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS AND OPPORTUNITIES FOR EXTENSION</h2>
        <div class="sectioncontent">
<p>Coefficients are recalculated for every individual test, but it would be good for these to persist between tests, adding more as necessary (i.e. if a subsequent test involves more genes than the current one).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Michael C. Wendl</p><p>mwendl@wustl.edu</p><p>Copyright (C) 2007, 2008 Washington University</p><p>This program is free software; you can redistribute it and/or modify it under the terms of the \s-1GNU\s0 General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.</p><p>This program is distributed in the hope that it will be useful, but \s-1WITHOUT\s0 \s-1ANY\s0 \s-1WARRANTY\s0; without even the implied warranty of \s-1MERCHANTABILITY\s0 or \s-1FITNESS\s0 \s-1FOR\s0 A \s-1PARTICULAR\s0 \s-1PURPOSE\s0.  See the \s-1GNU\s0 General Public License for more details.</p><p>You should have received a copy of the \s-1GNU\s0 General Public License along with this program; if not, write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, \s-1MA\s0 02111-1307, \s-1USA\s0.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">METHODS</h2>
        <div class="sectioncontent">
<p>The available methods are as follows.</p><h3>new</h3>
<p>This is the usual object constructor, which optionally takes all the gene-length data as input. If you want to use the exact probability solution or the asymptotic approximate solution, pass all lengths in a single list reference</p><p>        my $pmobj = PathScan-&gt;new ([3434, 54565, 6445, ...]);</p><p>but if you want to use the convolution approximation method, divide the list of gene sizes into the desired number of bins and pass each of these as a reference</p><p>        my $pmobj = PathScan-&gt;new ([3434, 54565], [6445, ...]);</p><p>In other words, the way you pass these arguments at partially determines the context in which you will obtain your P-value for this set of genes. The latter choice is typically betetr, as it gives good accuracy and good computational efficiency. Conversely, the exact solution is identically correct, but can be difficult to compute. The asymptotic approximation is always computationally efficient, but not necessarily accurate for small test sets.</p>
<h3>path_scan</h3>
<p>This function calculates the path-scan statistic in one of the appropriate contexts (exact or convolution approximation, as described above). It takes the actual number of \*(L"hits\*(R" you've observed in the data, i.e. the number of genes that have a mutated status.</p><p>        my $pval = $pmobj-&gt;path_scan (7);</p><p>If you have not yet done the pre-processing with respect to the background mutation rate (see below), then pre-processing can be executed implicitly by passing the rate as the second argument.</p><p>        my $pval = $pmobj-&gt;path_scan (7, 0.000001);</p>
<h3>cdf_truncated</h3>
<p>This function returns the cummulative distribution in the context of the convolution approximation <em>truncated</em> such that it contains only enough information to process the given number of hits.</p><p>        my $pvals_list = $pmobj-&gt;cdf_truncated ($hits);</p><p>The list is ordered from most extreme to least extreme probability tail values, i.e. the last value in the list is always unity. However, tailed p-values more extreme than that associated with the argument are not, in fact, calculated, but rather are replaced with the flag -1. This saves processing time and also reduces the chances of numerical overflow for large pathways, as the full \s-1CDF\s0 must ultimately raise an \*(L"mval\*(R" (&gt;1) to a power equal to the number of genes in the pathway. The method assumes you have already done the pre-processing with respect to the background mutation rate.</p>
<h3>cdf</h3>
<p>This function returns the cummulative distribution in one of the appropriate contexts (exact or convolution approximation, as described above). There are no arguments,</p><p>        my $pvals_list = $pmobj-&gt;cdf;</p><p>unless you have not yet done the pre-processing with respect to the background mutation rate (see below), in which case the pre-processing can be executed implicitly by passing the rate as the sole argument.</p><p>        my $pvals_list = $pmobj-&gt;cdf (0.000001);</p><p>The list is ordered from most extreme to least extreme probability tail values, i.e. the last value in the list is always unity.</p>
<h3>cdf_asymptot</h3>
<p>This function returns the cummulative distribution based on asymptotic analysis. There are no arguments, i.e.</p><p>        my $pvals_list = $pmobj-&gt;cdf_asymptot;</p><p>unless you have not yet done the pre-processing with respect to the background mutation rate (see below), in which case the pre-processing can be executed implicitly by passing the rate as the sole argument.</p><p>        my $pvals_list = $pmobj-&gt;cdf_asymptot (0.000001);</p><p>The list is ordered from most extreme to least extreme probability tail values, i.e. the last value in the list is always unity.</p><p>Note that asymptotic analysis gives a function (the Poisson) having infinite support. The infinite tail probability for all values past the most extreme physical case are all bundled into that most extreme p-value.</p>
<h3>path_scan_asymptot</h3>
<p>This function calculates the path-scan statistic in the asymptotic (Poisson) context. It takes the actual number of \*(L"hits\*(R" you've observed in the data, i.e. the number of genes that have a mutated status.</p><p>        my $pval = $pmobj-&gt;path_scan_asymptot (7);</p><p>If you have not yet done the pre-processing with respect to the background mutation rate (see below), then pre-processing can be executed implicitly by passing the rate as the second argument.</p><p>        my $pval = $pmobj-&gt;path_scan_asymptot (7, 0.000001);</p><p>You must set up the object, somewhat paradoxically, <em>as if you will be doing the calculation in the exact</em> context. (This is a consequence of how data are stored internally within the object.)</p>
<h3>additional methods</h3>
<p>The basic functionality of this package is encompassed in the methods described above. However, some lower-level functions can also sometimes be useful.</p><p><em>p_value_exact</em></p><p>This function returns the exact value of the probability <em>mass</em> for a specific number of hits.</p><p>        $pval_exact = $pmobj-&gt;p_value_exact (7);</p><p>You must make sure to call this only if you've configured the object in the exact context (see above).</p><p><em>p_value_binomial_approx</em></p><p>This function returns the convolution approximated value (i.e. using the binomial binning approximation) of the probability <em>mass</em> for a specific number of hits.</p><p>        $pval_exact = $pmobj-&gt;p_value_binomial_approx (7);</p><p>You must make sure to call this only if you've configured the object in the approximate binomial context (see above). Also, you must explicitly calculate the necessary binomial coefficients beforehand (see \*(C`binom_coeffs\*(C').</p><p><em>p_value_asymptot_approx</em></p><p>This function returns the asymptotic approximated value (i.e. using the Poisson limit approximation) of the probability <em>mass</em> for a specific number of hits.</p><p>        $pval_exact = $pmobj-&gt;p_value_asymptot_approx (7);</p><p>Somewhat paradoxically, you must make sure to call this only if you've configured the object in the exact context (see above).</p><p><em>store_genes</em></p><p>Stores the raw gene length data. Use this if you did not pass these data to \*(C`new\*(C' before you call any calculation methods. Works in the same way as \*(C`new\*(C', described above. Specifically, the context is partially determined by whether you pass a single list (exact context or asymptotic approximation)</p><p>        $pmobj-&gt;store_genes ([3434, 54565, 6445, ...]);</p><p>or more than one list (convolution approximate context)</p><p>        $pmobj-&gt;store_genes ([3434, 54565], [6445, ...]);</p><p><em>binom_coeffs</em></p><p>Calculates the binomial coefficients needed in the binomial (convolution) approximate solution.</p><p>        $pmobj-&gt;binom_coeffs;</p><p>The internal data structure is essentially the symmetric half of the appropriately-sized Pascal triangle. Considerable memory is saved by not storing the full triangle.</p><p><em>preprocess</em></p><p>Calculates the Bernoulli kernel probabilities for the individual genes or gene bins</p><p>        $pmobj_binom-&gt;preprocess ($background_mutation_rate);</p><p>The data structure can be re-configured to run the test with different background mutation rates by just re-calling this routine with a different value</p><p>        $pmobj_binom-&gt;preprocess ($new_background_mutation_rate);</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">
<p>The following examples may be helpful in using this package. In each case, assume we have first executed some required preliminary code.</p><p>        #_\|_USE THE PACKAGE            use PathScan;</p><p>        #_\|_SOME DATA FOR AN "EXACT CONTEXT" CALCULATION            my $genes_exact = [               4000, 4000, 4000, 4000, 4000,               15000, 15000, 15000, 15000, 15000,               35000, 35000, 35000, 35000, 35000            ];</p><p>        #_\|_SOME DATA FOR AN "APPROXIMATE CONTEXT" CALCULATION            my @genes_binned = (               [4000, 4000, 4000, 4000, 4000],               [15000, 15000, 15000, 15000, 15000],               [35000, 35000, 35000, 35000, 35000]            );</p><h3>simple path-scan test</h3>
<p>Here, we compare the values returned by both the exact and approximate algorithms over the whole domain of possible hits for a case where the answers should be identical.</p><p>        #_\|_SET BACKGROUND MUTATION RATE            my $rho = 0.00002;</p><p>        #_\|_CONFIGURE OBJECTS IN "EXACT" AND "APPROXIMATE" CONTEXTS            my $pmobj_exact = PathScan-&gt;new ($genes_exact);            $pmobj_exact-&gt;preprocess ($rho);</p><p>           my $pmobj_binom = PathScan-&gt;new (@genes_binned);            $pmobj_binom-&gt;preprocess ($rho);</p><p>        #_\|_CALCULATE AND TALLY THE MAXIMUM DIFFERENCE            my $maxdiff = 0;            for (my $i = 0; $i &lt;= scalar @{$genes_exact}; $i++) {               my $pm_pval_exact = $pmobj_exact-&gt;path_scan($i);               my $pm_pval_binom = $pmobj_binom-&gt;path_scan($i);               my $diff = abs ($pm_pval_exact - $pm_pval_binom);               $maxdiff = $diff if $diff &gt; $maxdiff;               print "$i hits: $pm_pval_exact   $pm_pval_binom   $diff&#92;n";            }            print "MAXIMUM DIFFERENCE IS $maxdiff&#92;n";</p>
<h3>testing at different background rates</h3>
<p>This example shows how to run the test for a fixed number of hits, say 7 in this case, for various different background mutation rates.</p><p>        #_\|_CONFIGURE OBJECT            my $pmobj_binom = PathScan-&gt;new (@genes_binned);</p><p>        #_\|_CALCULATE            for (my $rho = 0.00001; $rho &lt;= 0.0001; $rho += 0.00001) {               my $pm_pval_binom = $pmobj_binom-&gt;path_scan(7, $rho);               print "7 hits at background $rho : P = $pm_pval_binom&#92;n";            }</p><p>Note that we did not run \*(C`preprocess\*(C' explicitly, but rather let the \*(C`path_scan\*(C' method call it implicitly for each new value of the background mutation rate.</p>
<h3>computing asymptotic approximate solution</h3>
<p>The asymptotic (Poisson) approximate probabiltiy value is straightforward to compute.</p><p>        #_\|_SET BACKGROUND MUTATION RATE            my $rho = 0.00002;</p><p>        #_\|_CONFIGURE OBJECT            my $pmobj_poisson = PathScan-&gt;new ($genes_exact);            $pmobj_poisson-&gt;preprocess ($rho);</p><p>        #_\|_P-VALUE FOR 7 OBSERVED MUTATED GENES            my $pm_pval_poisson = $pmobj_exact-&gt;path_scan_asymptot (7);</p>
<h3>accessing individual probability masses</h3>
<p>The probability masses for specific numbers of hits can also be calculated.</p><p>        #_\|_SET BACKGROUND MUTATION RATE            my $rho = 0.00002;</p><p>        #_\|_CONFIGURE OBJECT            my $pmobj_exact = PathScan-&gt;new ($genes_exact);            $pmobj_exact-&gt;preprocess ($rho);</p><p>        #_\|_CALCULATE MASSES            my $total_prob = 0;            for (my $i = 0; $i &lt;= scalar @{$genes_exact}; $i++) {               my $pval_exact = $pmobj_exact-&gt;p_value_exact ($i);               $total_prob += $pval_exact;               print "$i hits : probability mass =  $pval_exact&#92;n";            }            print "total probability = $total_prob&#92;n";</p>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="Genome::Model::Tools::Music::PathScan::CombinePvals.3pm.html"><span aria-hidden="true">&larr;</span> Genome::Model::Tools::Music::PathScan::CombinePvals.3pm: Combining probabilities from independent tests of significance into a single aggregate figure</a></li>
   <li class="next"><a href="Genome::Model::Tools::Music::PathScan::PopulationPathScan.3pm.html">Genome::Model::Tools::Music::PathScan::PopulationPathScan.3pm: Apply pathscan test to populations rather than just single individuals <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
