<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>NHGRI::Blastall: Perl extension for running and parsing ncbi's blast 2.x</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Perl extension for running and parsing ncbi's blast 2.x">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="NHGRI::Blastall (3pm) manual">
  <meta name="twitter:description" content="Perl extension for running and parsing ncbi's blast 2.x">
  <meta name="twitter:image" content="https://www.carta.tech/images/libnhgri-blastall-perl-NHGRI::Blastall-3pm.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man3pm/NHGRI::Blastall.3pm.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="NHGRI::Blastall (3pm) manual" />
  <meta property="og:description" content="Perl extension for running and parsing ncbi's blast 2.x" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libnhgri-blastall-perl-NHGRI::Blastall-3pm.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">NHGRI::Blastall<small> (3pm)</small></h1>
        <p class="lead">Perl extension for running and parsing ncbi's blast 2.x</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/NHGRI::Blastall.3pm.html">
      <span itemprop="name">NHGRI::Blastall: Perl extension for running and parsing ncbi's blast 2.x</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libnhgri-blastall-perl/">
      <span itemprop="name">libnhgri-blastall-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man3pm/NHGRI::Blastall.3pm.html">
      <span itemprop="name">NHGRI::Blastall: Perl extension for running and parsing ncbi's blast 2.x</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>If you have \s-1NCBI\s0's \s-1BLAST2\s0 or WU-BLAST installed locally and your environment is already setup you can use Perl's object-oriented capabilities to run your BLASTs.  Also if you have a blastcl3 binary from the toolkit (or binaries from our \s-1FTP\s0 site) you can run \s-1BLAST\s0 over the network.  There are also methods to blast single sequences against each other using the bl2seq binaries (also in the toolkit and binaries). You can blast one sequence against a library of sequences using the blast_one_to_many method.  You can format databases with formatdb method. You can also have NHGRI::Blastall read existing \s-1BLAST\s0 reports.  If you have a database of repetitive \s-1DNA\s0 or other \s-1DNA\s0 you would like to mask out, you can use the mask method to mask the data against these databases. You can then use either the filter or result methods to parse the report and access the various elements of the data.</p>
<dl class='dl-vertical'>
  <dt>
    \s-1RUNNING\s0 \s-1NEW\s0 \s-1BLASTS\s0
  </dt>
  <dd>
    
<pre>
  use NHGRI::Blastall;
  my $b = new NHGRI::Blastall;
  # If you are running NCBI&apos;s Local BLAST
  $b-&gt;blastall( p =&gt; &apos;blastn&apos;,
                d =&gt; &apos;nr&apos;,
                i =&gt; &apos;infile&apos;,
                o =&gt; &apos;outfile&apos;
              );
  # If you are running NCBI&apos;s blastcl3 network client
  $b-&gt;blastcl3( p =&gt; &apos;blastn&apos;,
                d =&gt; &apos;nr&apos;,
                i =&gt; &apos;infile&apos;,
                o =&gt; &apos;outfile&apos;
              );
  # If you are running WU-BLAST locally
  $b-&gt;wu_blastall( p      =&gt; &apos;blastn&apos;,
                   d      =&gt; &apos;nr&apos;,
                   nogap  =&gt; &apos;!&apos;,     #use ! for arguments w/o parameter
                   i      =&gt; &apos;infile&apos;,
                   o      =&gt; &apos;outfile&apos;
                 );

  See BLASTALL for more info
</pre>

  </dd>
  <dt>
    \s-1BLASTING\s0 2 \s-1SEQUENCES\s0
  </dt>
  <dd>
    <p>  use NHGRI::Blastall;   my $b = new NHGRI::Blastall;   $b-&gt;bl2seq(i =&gt; &apos;seq1&apos;,              j =&gt; &apos;seq2&apos;,              p =&gt; &apos;tblastx&apos;             );</p><p>  See BL2SEQ for more info</p>
  </dd>
  <dt>
    \s-1BLASTING\s0 1 \s-1SEQUENCE\s0 \s-1AGAINST\s0 A \s-1FASTA\s0 \s-1LIBRARY\s0 \s-1OF\s0 \s-1SEQUENCES\s0
  </dt>
  <dd>
    <p>  # a library is a FASTA file with multiple FASTA formatted sequences   use NHGRI::Blastall;   my $b = new NHGRI::Blastall;   $b-&gt;blast_one_to_many(i =&gt; &apos;seq1&apos;,                         d =&gt; &apos;seq2.lib&apos;,                         p =&gt; &apos;tblastx&apos;,                        );</p><p>  See BLAST_ONE_TO_MANY for more info</p>
  </dd>
  <dt>
    \s-1INITIALIZING\s0 \s-1EXISTING\s0 \s-1BLAST\s0 \s-1REPORTS\s0
  </dt>
  <dd>
    <p>  use NHGRI::Blastall;   my $b = new NHGRI::Blastall;   $b-&gt;read_report(&apos;/path/to/report&apos;);</p>
  </dd>
  <dt>
    \s-1MASKING\s0 \s-1SEQUENCES\s0
  </dt>
  <dd>
    <p>  use NHGRI::Blastall;   my $b = new NHGRI::Blastall;   $masked_seq = $b-&gt;mask( type =&gt; &apos;wu_blastall&apos;,                           p    =&gt; &apos;blastn&apos;,                           d    =&gt; &apos;alu&apos;,                           i    =&gt; &apos;infile&apos;                         );</p><p>  See MASKING for more info</p>
  </dd>
  <dt>
    \s-1CREATING\s0 \s-1BLAST\s0 \s-1INDEXES\s0
  </dt>
  <dd>
    <p>  use NHGRI::Blastall;   my $b = new NHGRI::Blastall;   $b-&gt;formatdb( i =&gt; &apos;est&apos;,                 p =&gt; &apos;F&apos;,                 o =&gt; &apos;T&apos;,               );</p><p>  See FORMATDB for more info</p>
  </dd>
  <dt>
    \s-1PRINTING\s0 \s-1REPORTS\s0
  </dt>
  <dd>
    <p>  $b-&gt;print_report();   # this method only opens the report and prints.  It does not print   # summary reports</p>
  </dd>
  <dt>
    \s-1FILTERING\s0 \s-1BLAST\s0 \s-1RESULTS\s0
  </dt>
  <dd>
    <p>  @hits = $b-&gt;filter( scores     =&gt; &apos;38.2&apos;,                       identities =&gt; &apos;.98&apos;                     );</p><p>  # returns an array of hash references.   See HASHREF for more info on manipulating the results.   See FILTERING for more info on using the filter method</p>
  </dd>
  <dt>
    \s-1GETTING\s0 \s-1AT\s0 \s-1ELEMENTS\s0
  </dt>
  <dd>
    <p>  @ids = $b-&gt;result(&apos;id&apos;);   @scores = $b-&gt;result(&apos;scores&apos;,$ids[0]);  # second param must be an id</p><p>  See RESULT for more info on using the result method   See ELEMENTS for element names</p>
  </dd>
  <dt>
    \s-1GETTING\s0 \s-1AT\s0 \s-1ALL\s0 \s-1THE\s0 \s-1DATA\s0
  </dt>
  <dd>
    <p>  @results = $b-&gt;result();  # returns an array of hashes</p><p>  See HASHREF for information on the array of hashes that is returned.   See DUMP RESULTS to see how to work with the array of hashes</p>
  </dd>
  <dt>
    \s-1ADJUSTING\s0 \s-1THE\s0 \s-1DEFLINE\s0 \s-1REGEX\s0
  </dt>
  <dd>
    <p>  $b = new NHGRI::Blastall (-DB_ID_REGEX =&gt; &apos;[^ ]+&apos;);</p><p>  See DB_ID_REGEX for more info</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BLASTALL</h2>
        <div class="sectioncontent">
<p>This method provides a simple object oriented frontend to \s-1BLAST\s0. This module works with either \s-1NCBI\s0's blastall binary distributed with \s-1BLAST\s0 2.x, WU-BLAST or over the web through \s-1NCBI\s0's Web Site. The blastall function accepts as a parameter an anonymous hash with keys that are the command line options (See \s-1BLASTALL\s0 \s-1OPTIONS\s0) and values which are the corresponding values to those options.  You may want to set the \s-1BLASTALL\s0 variable in Blastall.pm to the full path of your `blastall' binary, especially if you will be running scripts as cron jobs or if blastall is not in the system path.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BLASTALL OPTIONS</h2>
        <div class="sectioncontent">
<p>For wu_blastall you need to use \s-1NCBI\s0 type switches for the following</p>
<pre>
   [C<strong>-i</strong>] for infile
   [C<strong>-o</strong>] for outfile
   [C<strong>-p</strong>] for program
   [C<strong>-d</strong>] for database
</pre>
<p>the rest of the parameters \s-1MUST\s0 be the parameters available through WU-BLAST (e.g. -sump, -nogap -compat1.4, etc.) use a `!' to specify that an argument has no parameters.  See the example at the top of the manpage.</p><p>These are the options that \s-1NCBI\s0's blastall and binary accepts and these are the same options that are accepted by the blastall and blastcl3 methods. \s-1NOTE:\s0 You must set the proper environmental variables for the blastall method to work (\s-1BLASTDB\s0,BLASTMAT).</p><ul>
<li><p><strong>p</strong> =&gt; Program Name</p></li><li><p><strong>d</strong> =&gt; Database                                           default=nr</p></li><li><p><strong>i</strong> =&gt; QueryFile</p></li><li><p><strong>e</strong> =&gt; Expectation vaule (E)                              default=10.0</p></li><li><p><strong>m</strong> =&gt; alignment view                                     default=0</p>
<pre>
    0 = pairwise,
    1 = master-slave showing identities,
    2 = master-slave no identities,
    3 = flat master-slave, show identities,
    4 = flat master-slave, no identities,
    5 = master-slave no identities and blunt ends,
    6 = flat master-slave, no identities and blunt ends
</pre>
</li><li><p><strong>o</strong> =&gt; \s-1BLAST\s0 report Output File                           default=stdout</p></li><li><p><strong>F</strong> =&gt; Filter query sequence                              default=T</p>
<pre>
    (\s-1DUST\s0 with blastn, \s-1SEG\s0 with others)
</pre>
</li><li><p><strong>G</strong> =&gt; Cost to open a gap                                 default=0</p>
<pre>
    (zero invokes default behavior)
</pre>
</li><li><p><strong>E</strong> =&gt; Cost to extend a gap                               default=0</p>
<pre>
    (zero invokes default behavior)
</pre>
</li><li><p><strong>X</strong> =&gt; X dropoff value for gapped alignment (in bits)     default=0</p>
<pre>
    (zero invokes default behavior)
</pre>
</li><li><p><strong>I</strong> =&gt; Show \s-1GI\s0's in deflines                              default=F</p></li><li><p><strong>q</strong> =&gt; Penalty for a nucleotide mismatch (blastn only)    default=-3</p></li><li><p><strong>r</strong> =&gt; Reward for a nucleotide match (blastn only)        default=1</p></li><li><p><strong>v</strong> =&gt; Number of one-line descriptions (V)                default=500</p></li><li><p><strong>b</strong> =&gt; Number of alignments to show (B)                   default=250</p></li><li><p><strong>f</strong> =&gt; Threshold for extending hits, default if zero      default=0</p></li><li><p><strong>g</strong> =&gt; Perfom gapped alignment (\s-1NA\s0 with tblastx)          default=T</p></li><li><p><strong>Q</strong> =&gt; Query Genetic code to use                          default=1</p></li><li><p><strong>D</strong> =&gt; \s-1DB\s0 Genetic code (for tblast[nx] only)              default=1</p></li><li><p><strong>a</strong> =&gt; Number of processors to use                        default=1</p></li><li><p><strong>O</strong> =&gt; SeqAlign file                                      Optional</p></li><li><p><strong>J</strong> =&gt; Believe the query defline                          default=F</p></li><li><p><strong>M</strong> =&gt; Matrix                                             default=BLOSUM62</p></li><li><p><strong>W</strong> =&gt; Word size, default if zero                         default=0</p></li><li><p><strong>z</strong> =&gt; Effective length of the database                   default=0</p>
<pre>
     (use zero for the real size)
</pre>
</li><li><p><strong>K</strong> =&gt; Number of best hits from a region to keep          default=100</p></li><li><p><strong>L</strong> =&gt; Length of region used to judge hits                default=20</p></li><li><p><strong>Y</strong> =&gt; Effective length of the search space               default=0</p>
<pre>
     (use zero for the real size)
</pre>
</li><li><p><strong>S</strong> =&gt; Query strands to search against database           default=3</p>
<pre>
     (for blast[nx], and tblastx).
     3 is both, 1 is top, 2 is bottom
</pre>
</li><li><p><strong>T</strong> =&gt; Produce \s-1HTML\s0 output [T/F]                          default=F</p></li><li><p><strong>l</strong> =&gt; Restrict search of database to list of \s-1GI\s0's [String]</p></li>
</ul><p>\s-1NOTE:\s0 If you do not supply an `o' option (outfile),</p>
<pre>
      the following environment variables are checked in order:
      `\s-1TMPDIR\s0', `\s-1TEMP\s0', and `\s-1TMP\s0'.
      If one of them is set, outfiles are created relative to the
      directory it specifies.  If none of them are set, the first
      possible one of the following directories is used:
      /var/tmp , /usr/tmp , /temp , /tmp ,
      This file is deleted after the NHGRI::Blastall object is destroyed.
      It is recommended that you create a tmp directory in your home
      directory and set one of the above environmental vars to point
      to this directory and then set the permissions on this directory
      to 0700.  Writing to a \*(L"public\*(R" tmp directory can have
      security ramifications.
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BL2SEQ</h2>
        <div class="sectioncontent">
<p>This method uses the bl2seq binary (distributed with \s-1BLAST\s0 executables and source) to \s-1BLAST\s0 one sequence against another sequence. Like the blastall method the bl2seq method accepts the same options that the bl2seq binary accepts.  Run bl2seq without options from the command line to get a full list of options.  An important note about the options, when running blastx 1st sequence should be nucleotide; when running tblastn 2nd sequence should be nucleotide.</p><p>  use NHGRI::Blastall;   my $b = new NHGRI::Blastall;   $b-&gt;bl2seq(i =&gt; &apos;seq1.nt&apos;,              j =&gt; &apos;seq2.aa&apos;,              p =&gt; &apos;blastx&apos;             );</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BLAST_ONE_TO_MANY</h2>
        <div class="sectioncontent">
<p>This method allows for blasting one sequence against a \s-1FASTA\s0 library of sequences.  Behind the scenes, \s-1BLAST\s0 indexes are created (in the same directory as the \s-1FASTA\s0 library) using the provided \s-1FASTA\s0 library and the one sequence is used to search against this database. If the program completes successfully, the databases are removed. To compare two sequences, use the bl2seq method which is faster and less messy (no tmp indexes).   This method accepts the same options as the blastall binary with the d option corresponding to the \s-1FASTA\s0 library.</p><p>  use NHGRI::Blastall;   my $b = new NHGRI::Blastall;   $b-&gt;blast_one_to_many(i =&gt; &apos;seq.aa&apos;,                         d =&gt; &apos;seq.nt.lib&apos;,                         e =&gt; &apos;0.001&apos;,                         p =&gt; &apos;tblastn&apos;,                        );</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">MASKING</h2>
        <div class="sectioncontent">
<p>Screens \s-1DNA\s0 sequences in fasta format against the database specified in the blastall 'd' option.  The mask method accepts the same parameters as the blastall method.  Any matches to the masking database will be substituted with \*(L"N\*(R"s.  The mask method returns the masked sequence. Performs similar function as xblast, an old \s-1NCBI\s0 program written in C.</p><p>Set the type parameter to wu_blastall, blastcl3 or blastall depending on your configuration.</p><p>  $masked_seq = $b-&gt;mask( type =&gt; &apos;blastcl3&apos;, # defaults to blastall                           p    =&gt; &apos;blastn&apos;,                           d    =&gt; &apos;alu&apos;,                           i    =&gt; &apos;infile&apos;                         );</p><p>To get the mask coordinates back call the mask method in an array context.</p><p>    @mask = $b-&gt;mask(p    =&gt; &apos;blastn&apos;,                      d    =&gt; &apos;alu&apos;,                      i    =&gt; &apos;infile&apos;                     );     $masked_seq = $mask[0];        # same as above masked seq     $ra_masked_coords = $mask[1];  # reference to array of mask coordinates</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FORMATDB</h2>
        <div class="sectioncontent">
<p>This method creates \s-1BLAST\s0 indexes using the formatdb binary which is distributed with \s-1BLAST\s0.  It accepts the same parameters as formatdb. The remove_formatdb_indexes method will remove databases created using the formatdb method (if called by the same object). formatdb leaves a file called formatdb.log by default in the current working directory (if it has permission).  To change this behavior, use the l option to direct the sequence to /dev/null or somewhere else.</p><p>  use NHGRI::Blastall;   my $b = new NHGRI::Blastall;   $b-&gt;formatdb( i =&gt; &apos;swissprot&apos;,                 p =&gt; &apos;T&apos;,                 l =&gt; &apos;/dev/null&apos;,                 o =&gt; &apos;T&apos;,               );</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DB_ID_REGEX</h2>
        <div class="sectioncontent">
<p>By default Blastall.pm expects \s-1FASTA\s0 deflines of \s-1BLAST\s0 databases to be formatted like Genbank database (gi|GINUMBER|DATABASE|ACCESSION|SUMMARY). The default regular expression is [^&#92;|]+(?:&#92;|[^&#92;|,&#92;s]*){1,10} When using non-genbankformatted deflines, it may become necessary to adjust the regular expression that identifies the unique identifier in a defline. This can be done with the -DB_ID_REGEX parameter to the new method.  For example</p><p>    $b = new NHGRI::Blastall( -DB_ID_REGEX =&gt; &apos;[^ ]+&apos; );</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FILTERING</h2>
        <div class="sectioncontent">
<p>The filter method accepts an anonymous hash in which the keys are elements of the blast report and the values are limits that are put on the result set.</p><p>The following are the Filter elements and their default operations.</p><p>    id                  =&gt; regular expression match     defline             =&gt; regular expression match     subject_length      =&gt; greater than     scores              =&gt; greater than     expects             =&gt; less than     identities          =&gt; greater than     match_length        =&gt; greater than     subject_strand      =&gt; equals     query_frames        =&gt; equals     subject_frames      =&gt; equals</p><p>so if you would like to limit your results to entries that have scores greater than 38.2 and identities greater than 98% you would say...</p><p>    @hits = $b-&gt;filter( scores      =&gt; &apos;38.2&apos;,                         identities  =&gt; &apos;.98&apos;                       );</p><p>you can also override the defaults.  if you would like only scores that are less than 38.2 you could say...</p><p>    @hits = $b-&gt;filter( scores =&gt; &apos;&lt;38.2&apos; );</p><p>or if you wanted only identities that were equal to 1 and you didn't care about the hits array you could say...</p><p>    $b-&gt;filter( identities =&gt; &apos;=1&apos; );</p><p>Regular expression matches are case insensitive.  If you wanted only records with the word \*(L"human\*(R" in the defline you could say...</p><p>    @hits = $b-&gt;filter( defline =&gt; &apos;HuMaN&apos; );</p><p>After you run the filter method on an object the object only contains those results which passed the filter. This will effect additional calls to the filter method as well as calls to other methods (e.g. result).  To reset the NHGRI::Blastall object you can use the unfilter method.</p><p>    $b-&gt;unfilter();</p><p>See \s-1DUMP\s0 \s-1RESULTS\s0 for info on how to manipulate the array of hash refs.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RESULT</h2>
        <div class="sectioncontent">
<p>  The result method has 3 possible invocations.  The first invocation   is when it is called without parameters.</p><p>  @results = $b-&gt;result();</p><p>  This invocation returns an array of hash references.   See HASHREF for further explanation of this structure.</p><p>  To get a list of all the ids do...</p><p>  @ids = $b-&gt;result(&apos;id&apos;);</p><p>  These ids can be used to get at specific elements.  If 2 parameters   are present and the first one is an element (See ELEMENTS for a list   of ELEMENTS) and the second one is an id then the routine will   return a list of elements corresponding to the id.</p><p>  @scores = $b-&gt;result(&apos;scores&apos;,$ids[0]);  # second param must be an id</p><p>  If more than 2 elements are passed the function will return undef.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ACCESSOR METHODS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <strong>get_report</strong>
  </dt>
  <dd>
    <p>    returns the filename of the BLAST report.</p>
  </dd>
  <dt>
    <strong>get_database_description</strong>
  </dt>
  <dd>
    <p>    returns description given to the database during formatting of db.     e.g. All non-redundant GenBank CDStranslations+PDB+SwissProt+PIR</p>
  </dd>
  <dt>
    <strong>get_database_sequence_count</strong>
  </dt>
  <dd>
    <p>    returns the number of sequences in the database.</p>
  </dd>
  <dt>
    <strong>get_database_letter_count</strong>
  </dt>
  <dd>
    <p>    returns the number of total letters in the database.</p>
  </dd>
  <dt>
    <strong>get_blast_program</strong>
  </dt>
  <dd>
    <p>    returns the BLAST program name that appears at the top of the report.     either BLASTN, BLASTP, BLASTX, TBLASTN or TBLASTX</p>
  </dd>
  <dt>
    <strong>get_blast_version</strong>
  </dt>
  <dd>
    <p>    returns the version of the BLAST program that was used.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ELEMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <strong>id</strong>
  </dt>
  <dd>
    <p>     an example of an id is &#96;&gt;gb|U19386|MMU19386&apos; the initial &#96;&gt;&apos;      is just a flag.  The next characters up until the first pipe      is the database the subject was taken from.  The next characters      up to the next pipe is the Genbank accession number.  The last      characters are the locus.  This element is used as a unique      identifier by the NHGRI::Blastall module.      (SCALAR)</p>
  </dd>
  <dt>
    <strong>defline</strong>
  </dt>
  <dd>
    <p>    The definition line taken from the subject      (SCALAR)</p>
  </dd>
  <dt>
    <strong>subject_length</strong>
  </dt>
  <dd>
    <p>    This is the length of the full subject, not the      length of the match.      (SCALAR)</p>
  </dd>
  <dt>
    <strong>scores</strong>
  </dt>
  <dd>
    <p>    This is score (in bits) of the match.      (ARRAY)</p>
  </dd>
  <dt>
    <strong>expects</strong>
  </dt>
  <dd>
    <p>    This is the statistical significance (&#96;E value&apos;) for the match.      (ARRAY)</p>
  </dd>
  <dt>
    <strong>identities</strong>
  </dt>
  <dd>
    <p>    This is the number of identities divided by the match     length in decimal format. (Listed as a fraction and a percentage     in a BLAST report.)     (ARRAY)</p>
  </dd>
  <dt>
    <strong>match_lengths</strong>
  </dt>
  <dd>
    <p>    this is the number of base pairs that match up.     (ARRAY)</p>
  </dd>
  <dt>
    <strong>query_starts</strong>
  </dt>
  <dd>
    <p>    This is the number of the first base which matched     with the subject.     (ARRAY)</p>
  </dd>
  <dt>
    <strong>query_ends</strong>
  </dt>
  <dd>
    <p>    This is the number of the last base which matched     with the subject.     (ARRAY)</p>
  </dd>
  <dt>
    <strong>subject_starts</strong>
  </dt>
  <dd>
    <p>    This is the number of the first base which matched     with the query.     (ARRAY)</p>
  </dd>
  <dt>
    <strong>subject_ends</strong>
  </dt>
  <dd>
    <p>    This is the number of the last base which matched     with the query.     (ARRAY)</p>
  </dd>
  <dt>
    <strong>subject_strands</strong>
  </dt>
  <dd>
    <p>    This is either plus or minus depending on the orientation     of the subject sequence in the match.     (ARRAY)</p>
  </dd>
  <dt>
    <strong>query_strands</strong>
  </dt>
  <dd>
    <p>    This is either plus or minus depending on the orientation     of the query sequence in the match.     (ARRAY)</p>
  </dd>
  <dt>
    <strong>query_frames</strong>
  </dt>
  <dd>
    <p>    If you are running a blastx or tblastx search in which the     query_sequence is translated this is the frame the query     sequence matched.     (ARRAY)</p>
  </dd>
  <dt>
    <strong>subject_frames</strong>
  </dt>
  <dd>
    <p>    If you are running a tblastn or tblastx search in which the     subject sequence is translated, this is the frame where the     subject sequence matched.     (ARRAY)</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">HASHREF</h2>
        <div class="sectioncontent">
<p>  Each hash ref contains an id, defline and subject Length.  Because   there can be multiple scores, expect values, Identities, match_lengths,   query_starts, query_strands and subject_starts, these are stored   as array references.  The following is an array containing two hash   refs.</p><p>  @hits = (       {&apos;id&apos;                 =&gt; &apos;&gt;gb|U79716|HSU79716&apos;,        &apos;defline&apos;            =&gt; &apos;Human reelin (RELN) mRNA, complete cds&apos;,        &apos;subject_length&apos;     =&gt; &apos;11580&apos;,        &apos;scores&apos;             =&gt; [ 684, 123               ],        &apos;expects&apos;            =&gt; [ 0.0, 3e-26             ],        &apos;identities&apos;         =&gt; [ .99430199, .89256198   ],        &apos;match_lengths&apos;      =&gt; [ 351, 121               ],        &apos;query_starts&apos;       =&gt; [ 3, 404                 ],        &apos;query_ends&apos;         =&gt; [ 303, 704                 ],        &apos;subject_starts&apos;     =&gt; [ 5858, 6259             ],        &apos;subject_ends&apos;       =&gt; [ 6158, 6559             ],        &apos;subject_strands&apos;    =&gt; [ &apos;plus&apos;, &apos;minus&apos;        ],        &apos;query_strands&apos;      =&gt; [ &apos;plus&apos;, &apos;plus&apos;         ],        &apos;query_frames&apos;       =&gt; [ &apos;+1&apos;, &apos;-3&apos;             ],        &apos;subject_frames&apos;     =&gt; [ &apos;+2&apos;, &apos;-1&apos;             ],       },       {&apos;id&apos;                 =&gt; &apos;&gt;gb|U24703|MMU24703&apos;,        &apos;defline&apos;            =&gt; &apos;Mus musculus reelin mRNA, complete cds&apos;,        &apos;subject_length&apos;     =&gt; &apos;11673&apos;,        &apos;scores&apos;             =&gt; [ 319, 38.2              ],        &apos;expects&apos;            =&gt; [ 2e-85, 1.2             ],        &apos;identities&apos;         =&gt; [ .86455331, 1           ],        &apos;match_lengths&apos;      =&gt; [ 347, 19                ],        &apos;query_starts&apos;       =&gt; [ 3, 493                 ],        &apos;query_ends&apos;         =&gt; [ 303, 793                 ],        &apos;subject_starts&apos;     =&gt; [ 5968, 6457             ]        &apos;subject_ends&apos;       =&gt; [ 6268, 6757             ],        &apos;subject_strands&apos;    =&gt; [ &apos;plus&apos;, &apos;minus&apos;        ],        &apos;query_strands&apos;      =&gt; [ &apos;plus&apos;, &apos;plus&apos;         ],        &apos;query_frames&apos;       =&gt; [ &apos;+3&apos;, &apos;-3&apos;             ],        &apos;subject_frames&apos;     =&gt; [ &apos;+1&apos;, &apos;-2&apos;             ],       }   );</p><p>  See ELEMENTS for explanation of each element.   See DUMP RESULTS and/or the <strong>perlref</strong>(1) manpage for clues on working       with this structure.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DUMP RESULTS</h2>
        <div class="sectioncontent">
<p>  When calling the result function or with no parameters, or calling the   filter function, an array of references to hashes is returned.   Each elment of the array is a reference to a hash containing 1 record.   See HASHREF for details on this structure.  The following   routine will go through each element of the array of hashes and   then print out the element and it&apos;s corresponding value or values.   See <strong>perlref</strong>(1) for more info on references.</p><p>  sub dump_results {       foreach $rh_r (@results) {           while (($key,$value) = each %$rh_r) {               if (ref($value) eq "ARRAY") {                   print "$key: ";                   foreach $v (@$value) {                       print "$v ";                   }                   print "&#92;n";               } else {                   print "$key: $value&#92;n";               }           }       }   }</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<ul>
<li><p>Joseph Ryan (jfryan@nhgri.nih.gov)</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">CONTACT ADDRESS</h2>
        <div class="sectioncontent">
<p>If you have problems, questions, comments send to webblaster@nhgri.nih.gov</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT INFORMATION</h2>
        <div class="sectioncontent">
<p>This software/database is \*(L"United States Government Work\*(R" under the terms of the United States Copyright Act. It was written as part of the authors' official duties for the United States Government and thus cannot be copyrighted. This software/database is freely available to the public for use without a copyright notice. Restrictions cannot be placed on its present or future use.</p><p>Although all reasonable efforts have been taken to ensure the accuracy and reliability of the software and data, the National Human Genome Research Institute (\s-1NHGRI\s0) and the U.S.  Government does not and cannot warrant the performance or results that may be obtained by using this software or data. \s-1NHGRI\s0 and the U.S. Government disclaims all warranties as to performance, merchantability or fitness for any particular purpose.</p><p>In any work or product derived from this material, proper attribution of the authors as the source of the software or data should be made, using http://genome.nhgri.nih.gov/blastall as the citation.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ENVIRONMENT VARIABLES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <strong>\s-1BLASTDB\s0</strong>
  </dt>
  <dd>
    <p>location of \s-1BLAST\s0 formated databases</p>
  </dd>
  <dt>
    <strong>\s-1BLASTMAT\s0</strong>
  </dt>
  <dd>
    <p>location of \s-1BLAST\s0 matrices</p>
  </dd>
  <dt>
    <strong>\s-1TMPDIR\s0</strong> <strong>\s-1TEMP\s0</strong> <strong>\s-1TMP\s0</strong>
  </dt>
  <dd>
    <p>If the `o' option is not passed to the blastall method than NHGRI::Blastall looks for one of these vars (in order) to store the \s-1BLAST\s0 report.  This report is destroyed after the NHGRI::Blastall.pm object is destroyed.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO NHGRI::Blastall&hellip;</h2>
        <div class="sectioncontent">
<p><em>perl</em>\|(1) <em>perlref</em>\|(1)</p><p>http://www.ncbi.nlm.nih.gov/BLAST/newblast.html</p><p>ftp://ncbi.nlm.nih.gov/blast/db/README</p><p>http://www.ncbi.nlm.nih.gov/BLAST/tutorial/Altschul-1.html</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="MySQL::Diff::Utils.3pm.html"><span aria-hidden="true">&larr;</span> MySQL::Diff::Utils.3pm: Supporting functions for mysql:diff</a></li>
   <li class="next"><a href="NOTEDB::binary.3pm.html">NOTEDB::binary.3pm: Module lib for accessing a notedb from perl <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
