<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Man Pages in emboss</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Man Pages in emboss">
</head>
<body>
  <div class="container final">
          <div class="page-header">
      <h1 class="font-effect-destruction"><small>Man Pages in</small> emboss</h1>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/emboss/">
      <span itemprop="name">emboss</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
</ol>
    <ul class="list-group">
<li class="list-group-item">
<span class="badge">318</span>
<h4>Section 1e: <a href="https://www.carta.tech/man-pages/man1/">Executable programs or shell commands</a></h4>

<dl class="dl-vertical">
 <dt><a href="/man-pages/man1/aaindexextract.1e.html"><strong>aaindexextract.1e</strong></a></dt>

 <dd>Extract amino acid property data from aaindex</dd>

 <dt><a href="/man-pages/man1/abiview.1e.html"><strong>abiview.1e</strong></a></dt>

 <dd>Display the trace in an abi sequencer file</dd>

 <dt><a href="/man-pages/man1/acdc.1e.html"><strong>acdc.1e</strong></a></dt>

 <dd>Test an application acd file</dd>

 <dt><a href="/man-pages/man1/acdlog.1e.html"><strong>acdlog.1e</strong></a></dt>

 <dd>Test application acd file processing and trace data structures</dd>

 <dt><a href="/man-pages/man1/acdpretty.1e.html"><strong>acdpretty.1e</strong></a></dt>

 <dd>Correctly reformat an application acd file</dd>

 <dt><a href="/man-pages/man1/acdrelations.1e.html"><strong>acdrelations.1e</strong></a></dt>

 <dd>Add relations: attribute to acd files</dd>

 <dt><a href="/man-pages/man1/acdtable.1e.html"><strong>acdtable.1e</strong></a></dt>

 <dd>Generate an html table of parameters from an application acd file</dd>

 <dt><a href="/man-pages/man1/acdtrace.1e.html"><strong>acdtrace.1e</strong></a></dt>

 <dd>Trace processing of an application acd file (for testing)</dd>

 <dt><a href="/man-pages/man1/acdvalid.1e.html"><strong>acdvalid.1e</strong></a></dt>

 <dd>Validate an application acd file</dd>

 <dt><a href="/man-pages/man1/ajbad.1e.html"><strong>ajbad.1e</strong></a></dt>

 <dd>Test file for acd parsing</dd>

 <dt><a href="/man-pages/man1/ajfeatest.1e.html"><strong>ajfeatest.1e</strong></a></dt>

 <dd>Reads and writes (returns) a sequence and its features</dd>

 <dt><a href="/man-pages/man1/ajtest.1e.html"><strong>ajtest.1e</strong></a></dt>

 <dd>Test file for acd parsing</dd>

 <dt><a href="/man-pages/man1/aligncopy.1e.html"><strong>aligncopy.1e</strong></a></dt>

 <dd>Reads and writes alignments</dd>

 <dt><a href="/man-pages/man1/aligncopypair.1e.html"><strong>aligncopypair.1e</strong></a></dt>

 <dd>Reads and writes pairs from alignments</dd>

 <dt><a href="/man-pages/man1/antigenic.1e.html"><strong>antigenic.1e</strong></a></dt>

 <dd>Finds antigenic sites in proteins</dd>

 <dt><a href="/man-pages/man1/backtranambig.1e.html"><strong>backtranambig.1e</strong></a></dt>

 <dd>Back-translate a protein sequence to ambiguous nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/backtranseq.1e.html"><strong>backtranseq.1e</strong></a></dt>

 <dd>Back-translate a protein sequence to a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/banana.1e.html"><strong>banana.1e</strong></a></dt>

 <dd>Plot bending and curvature data for b-dna</dd>

 <dt><a href="/man-pages/man1/biosed.1e.html"><strong>biosed.1e</strong></a></dt>

 <dd>Replace or delete sequence sections</dd>

 <dt><a href="/man-pages/man1/btwisted.1e.html"><strong>btwisted.1e</strong></a></dt>

 <dd>Calculate the twisting in a b-dna sequence</dd>

 <dt><a href="/man-pages/man1/cachedas.1e.html"><strong>cachedas.1e</strong></a></dt>

 <dd>Generates server cache file for das servers or for the das registry</dd>

 <dt><a href="/man-pages/man1/cachedbfetch.1e.html"><strong>cachedbfetch.1e</strong></a></dt>

 <dd>Generates server cache file for dbfetch/wsdbfetch data sources</dd>

 <dt><a href="/man-pages/man1/cacheebeyesearch.1e.html"><strong>cacheebeyesearch.1e</strong></a></dt>

 <dd>Generates server cache file for eb-eye search domains</dd>

 <dt><a href="/man-pages/man1/cacheensembl.1e.html"><strong>cacheensembl.1e</strong></a></dt>

 <dd>Generates server cache file for an ensembl server</dd>

 <dt><a href="/man-pages/man1/cai.1e.html"><strong>cai.1e</strong></a></dt>

 <dd>Calculate codon adaptation index</dd>

 <dt><a href="/man-pages/man1/chaos.1e.html"><strong>chaos.1e</strong></a></dt>

 <dd>Draw a chaos game representation plot for a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/charge.1e.html"><strong>charge.1e</strong></a></dt>

 <dd>Draw a protein charge plot</dd>

 <dt><a href="/man-pages/man1/checktrans.1e.html"><strong>checktrans.1e</strong></a></dt>

 <dd>Reports stop codons and orf statistics of a protein</dd>

 <dt><a href="/man-pages/man1/chips.1e.html"><strong>chips.1e</strong></a></dt>

 <dd>Calculates nc codon usage statistic</dd>

 <dt><a href="/man-pages/man1/cirdna.1e.html"><strong>cirdna.1e</strong></a></dt>

 <dd>Draws circular maps of dna constructs</dd>

 <dt><a href="/man-pages/man1/codcmp.1e.html"><strong>codcmp.1e</strong></a></dt>

 <dd>Codon usage table comparison</dd>

 <dt><a href="/man-pages/man1/codcopy.1e.html"><strong>codcopy.1e</strong></a></dt>

 <dd>Copy and reformat a codon usage table</dd>

 <dt><a href="/man-pages/man1/coderet.1e.html"><strong>coderet.1e</strong></a></dt>

 <dd>Extract cds, mrna and translations from feature tables</dd>

 <dt><a href="/man-pages/man1/complex.1e.html"><strong>complex.1e</strong></a></dt>

 <dd>Find the linguistic complexity in nucleotide sequences</dd>

 <dt><a href="/man-pages/man1/compseq.1e.html"><strong>compseq.1e</strong></a></dt>

 <dd>Calculate the composition of unique words in sequences</dd>

 <dt><a href="/man-pages/man1/cons.1e.html"><strong>cons.1e</strong></a></dt>

 <dd>Create a consensus sequence from a multiple alignment</dd>

 <dt><a href="/man-pages/man1/consambig.1e.html"><strong>consambig.1e</strong></a></dt>

 <dd>Create an ambiguous consensus sequence from a multiple alignment</dd>

 <dt><a href="/man-pages/man1/corbatest.1e.html"><strong>corbatest.1e</strong></a></dt>

 <dd>Test of embl corba retrieval</dd>

 <dt><a href="/man-pages/man1/cpgplot.1e.html"><strong>cpgplot.1e</strong></a></dt>

 <dd>Identify and plot cpg islands in nucleotide sequence(s)</dd>

 <dt><a href="/man-pages/man1/cpgreport.1e.html"><strong>cpgreport.1e</strong></a></dt>

 <dd>Identify and report cpg-rich regions in nucleotide sequence(s)</dd>

 <dt><a href="/man-pages/man1/cusp.1e.html"><strong>cusp.1e</strong></a></dt>

 <dd>Create a codon usage table from nucleotide sequence(s)</dd>

 <dt><a href="/man-pages/man1/cutgextract.1e.html"><strong>cutgextract.1e</strong></a></dt>

 <dd>Extract codon usage tables from cutg database</dd>

 <dt><a href="/man-pages/man1/cutseq.1e.html"><strong>cutseq.1e</strong></a></dt>

 <dd>Removes a section from a sequence</dd>

 <dt><a href="/man-pages/man1/dan.1e.html"><strong>dan.1e</strong></a></dt>

 <dd>Calculates nucleic acid melting temperature</dd>

 <dt><a href="/man-pages/man1/dastest.1e.html"><strong>dastest.1e</strong></a></dt>

 <dd>Return das sources from a das server or from the das registry</dd>

 <dt><a href="/man-pages/man1/dbfetchtest.1e.html"><strong>dbfetchtest.1e</strong></a></dt>

 <dd>Generates server cache file for wsdbfetch data sources</dd>

 <dt><a href="/man-pages/man1/dbget.1e.html"><strong>dbget.1e</strong></a></dt>

 <dd>Retrieve database entries for a named database, data type and data identifier</dd>

 <dt><a href="/man-pages/man1/dbgeturl.1e.html"><strong>dbgeturl.1e</strong></a></dt>

 <dd>Retrieve links to database entries for a named database, data type and data identifier</dd>

 <dt><a href="/man-pages/man1/dbiblast.1e.html"><strong>dbiblast.1e</strong></a></dt>

 <dd>Index a blast database</dd>

 <dt><a href="/man-pages/man1/dbifasta.1e.html"><strong>dbifasta.1e</strong></a></dt>

 <dd>Index a fasta file database</dd>

 <dt><a href="/man-pages/man1/dbiflat.1e.html"><strong>dbiflat.1e</strong></a></dt>

 <dd>Index a flat file database</dd>

 <dt><a href="/man-pages/man1/dbigcg.1e.html"><strong>dbigcg.1e</strong></a></dt>

 <dd>Index a gcg formatted database</dd>

 <dt><a href="/man-pages/man1/dbshowdat.1e.html"><strong>dbshowdat.1e</strong></a></dt>

 <dd>Show possible database queries that return a given data type</dd>

 <dt><a href="/man-pages/man1/dbshowquery.1e.html"><strong>dbshowquery.1e</strong></a></dt>

 <dd>Show possible database queries that use a given data identifier</dd>

 <dt><a href="/man-pages/man1/dbtell.1e.html"><strong>dbtell.1e</strong></a></dt>

 <dd>Display information about a public database</dd>

 <dt><a href="/man-pages/man1/dbtellquery.1e.html"><strong>dbtellquery.1e</strong></a></dt>

 <dd>Show possible queries of a database</dd>

 <dt><a href="/man-pages/man1/dbxcompress.1e.html"><strong>dbxcompress.1e</strong></a></dt>

 <dd>Compress an uncompressed dbx index</dd>

 <dt><a href="/man-pages/man1/dbxedam.1e.html"><strong>dbxedam.1e</strong></a></dt>

 <dd>Index the edam ontology using b+tree indices</dd>

 <dt><a href="/man-pages/man1/dbxfasta.1e.html"><strong>dbxfasta.1e</strong></a></dt>

 <dd>Index a fasta file database using b+tree indices</dd>

 <dt><a href="/man-pages/man1/dbxflat.1e.html"><strong>dbxflat.1e</strong></a></dt>

 <dd>Index a flat file database using b+tree indices</dd>

 <dt><a href="/man-pages/man1/dbxgcg.1e.html"><strong>dbxgcg.1e</strong></a></dt>

 <dd>Index a gcg formatted database using b+tree indices</dd>

 <dt><a href="/man-pages/man1/dbxobo.1e.html"><strong>dbxobo.1e</strong></a></dt>

 <dd>Index an obo ontology using b+tree indices</dd>

 <dt><a href="/man-pages/man1/dbxreport.1e.html"><strong>dbxreport.1e</strong></a></dt>

 <dd>Validate index and report internals for dbx databases</dd>

 <dt><a href="/man-pages/man1/dbxresource.1e.html"><strong>dbxresource.1e</strong></a></dt>

 <dd>Index a data resource catalogue using b+tree indices</dd>

 <dt><a href="/man-pages/man1/dbxstat.1e.html"><strong>dbxstat.1e</strong></a></dt>

 <dd>Dump statistics for dbx databases</dd>

 <dt><a href="/man-pages/man1/dbxtax.1e.html"><strong>dbxtax.1e</strong></a></dt>

 <dd>Index ncbi taxonomy using b+tree indices</dd>

 <dt><a href="/man-pages/man1/dbxuncompress.1e.html"><strong>dbxuncompress.1e</strong></a></dt>

 <dd>Uncompress a compressed dbx index</dd>

 <dt><a href="/man-pages/man1/degapseq.1e.html"><strong>degapseq.1e</strong></a></dt>

 <dd>Removes non-alphabetic (e.g. gap) characters from sequences</dd>

 <dt><a href="/man-pages/man1/density.1e.html"><strong>density.1e</strong></a></dt>

 <dd>Draw a nucleic acid density plot</dd>

 <dt><a href="/man-pages/man1/descseq.1e.html"><strong>descseq.1e</strong></a></dt>

 <dd>Alter the name or description of a sequence.</dd>

 <dt><a href="/man-pages/man1/diffseq.1e.html"><strong>diffseq.1e</strong></a></dt>

 <dd>Compare and report features of two similar sequences</dd>

 <dt><a href="/man-pages/man1/digest.1e.html"><strong>digest.1e</strong></a></dt>

 <dd>Reports on protein proteolytic enzyme or reagent cleavage sites</dd>

 <dt><a href="/man-pages/man1/distmat.1e.html"><strong>distmat.1e</strong></a></dt>

 <dd>Create a distance matrix from a multiple sequence alignment</dd>

 <dt><a href="/man-pages/man1/docxmlvalid.1e.html"><strong>docxmlvalid.1e</strong></a></dt>

 <dd>Check an xml document is well formed and conforms to a schema</dd>

 <dt><a href="/man-pages/man1/docxmlwell.1e.html"><strong>docxmlwell.1e</strong></a></dt>

 <dd>Check an xml document is well formed</dd>

 <dt><a href="/man-pages/man1/domtesta.1e.html"><strong>domtesta.1e</strong></a></dt>

 <dd>Reads an xml file into the dom and writes it back out</dd>

 <dt><a href="/man-pages/man1/domtestb.1e.html"><strong>domtestb.1e</strong></a></dt>

 <dd>Create, manipulate and write out xml</dd>

 <dt><a href="/man-pages/man1/domtestc.1e.html"><strong>domtestc.1e</strong></a></dt>

 <dd>Create and write out some typical xml</dd>

 <dt><a href="/man-pages/man1/domtestd.1e.html"><strong>domtestd.1e</strong></a></dt>

 <dd>Create some xml and search for an element</dd>

 <dt><a href="/man-pages/man1/dotmatcher.1e.html"><strong>dotmatcher.1e</strong></a></dt>

 <dd>Draw a threshold dotplot of two sequences</dd>

 <dt><a href="/man-pages/man1/dotpath.1e.html"><strong>dotpath.1e</strong></a></dt>

 <dd>Draw a non-overlapping wordmatch dotplot of two sequences</dd>

 <dt><a href="/man-pages/man1/dottup.1e.html"><strong>dottup.1e</strong></a></dt>

 <dd>Displays a wordmatch dotplot of two sequences</dd>

 <dt><a href="/man-pages/man1/dreg.1e.html"><strong>dreg.1e</strong></a></dt>

 <dd>Regular expression search of nucleotide sequence(s)</dd>

 <dt><a href="/man-pages/man1/drfinddata.1e.html"><strong>drfinddata.1e</strong></a></dt>

 <dd>Find public databases by data type</dd>

 <dt><a href="/man-pages/man1/drfindformat.1e.html"><strong>drfindformat.1e</strong></a></dt>

 <dd>Find public databases by format</dd>

 <dt><a href="/man-pages/man1/drfindid.1e.html"><strong>drfindid.1e</strong></a></dt>

 <dd>Find public databases by identifier</dd>

 <dt><a href="/man-pages/man1/drfindresource.1e.html"><strong>drfindresource.1e</strong></a></dt>

 <dd>Find public databases by resource</dd>

 <dt><a href="/man-pages/man1/drget.1e.html"><strong>drget.1e</strong></a></dt>

 <dd>Get data resource entries</dd>

 <dt><a href="/man-pages/man1/drtext.1e.html"><strong>drtext.1e</strong></a></dt>

 <dd>Get data resource entries complete text</dd>

 <dt><a href="/man-pages/man1/edamclean.1e.html"><strong>edamclean.1e</strong></a></dt>

 <dd>Validate and fix edam obo ontology</dd>

 <dt><a href="/man-pages/man1/edamdef.1e.html"><strong>edamdef.1e</strong></a></dt>

 <dd>Find edam ontology terms by definition</dd>

 <dt><a href="/man-pages/man1/edamhasinput.1e.html"><strong>edamhasinput.1e</strong></a></dt>

 <dd>Find edam ontology terms by has_input relation</dd>

 <dt><a href="/man-pages/man1/edamhasoutput.1e.html"><strong>edamhasoutput.1e</strong></a></dt>

 <dd>Find edam ontology terms by has_output relation</dd>

 <dt><a href="/man-pages/man1/edamisformat.1e.html"><strong>edamisformat.1e</strong></a></dt>

 <dd>Find edam ontology terms by is_format_of relation</dd>

 <dt><a href="/man-pages/man1/edamisid.1e.html"><strong>edamisid.1e</strong></a></dt>

 <dd>Find edam ontology terms by is_identifier_of relation</dd>

 <dt><a href="/man-pages/man1/edamname.1e.html"><strong>edamname.1e</strong></a></dt>

 <dd>Find edam ontology terms by name</dd>

 <dt><a href="/man-pages/man1/edialign.1e.html"><strong>edialign.1e</strong></a></dt>

 <dd>Local multiple alignment of sequences</dd>

 <dt><a href="/man-pages/man1/einverted.1e.html"><strong>einverted.1e</strong></a></dt>

 <dd>Finds inverted repeats in nucleotide sequences</dd>

 <dt><a href="/man-pages/man1/em_cons.1e.html"><strong>em_cons.1e</strong></a></dt>

 <dd>Create a consensus sequence from a multiple alignment</dd>

 <dt><a href="/man-pages/man1/em_pscan.1e.html"><strong>em_pscan.1e</strong></a></dt>

 <dd>Scans protein sequence(s) with fingerprints from the prints database</dd>

 <dt><a href="/man-pages/man1/embossdata.1e.html"><strong>embossdata.1e</strong></a></dt>

 <dd>Find and retrieve emboss data files</dd>

 <dt><a href="/man-pages/man1/embossversion.1e.html"><strong>embossversion.1e</strong></a></dt>

 <dd>Reports the current emboss version number</dd>

 <dt><a href="/man-pages/man1/emma.1e.html"><strong>emma.1e</strong></a></dt>

 <dd>Multiple sequence alignment (clustalw wrapper)</dd>

 <dt><a href="/man-pages/man1/emowse.1e.html"><strong>emowse.1e</strong></a></dt>

 <dd>Search protein sequences by digest fragment molecular weight</dd>

 <dt><a href="/man-pages/man1/ensembltest.1e.html"><strong>ensembltest.1e</strong></a></dt>

 <dd>Demonstration of the ensembl api to be.</dd>

 <dt><a href="/man-pages/man1/entrails.1e.html"><strong>entrails.1e</strong></a></dt>

 <dd>Reports the internal data from the emboss code</dd>

 <dt><a href="/man-pages/man1/entrailsbook.1e.html"><strong>entrailsbook.1e</strong></a></dt>

 <dd>Reports the internal structures in emboss book format</dd>

 <dt><a href="/man-pages/man1/entrailshtml.1e.html"><strong>entrailshtml.1e</strong></a></dt>

 <dd>Reports the internal structures in html</dd>

 <dt><a href="/man-pages/man1/entrailswiki.1e.html"><strong>entrailswiki.1e</strong></a></dt>

 <dd>Reports the internal structures in wikitext</dd>

 <dt><a href="/man-pages/man1/entret.1e.html"><strong>entret.1e</strong></a></dt>

 <dd>Retrieves sequence entries from flatfile databases and files</dd>

 <dt><a href="/man-pages/man1/entrygo.1e.html"><strong>entrygo.1e</strong></a></dt>

 <dd>Identify and retrieve sequence entries containing go terms</dd>

 <dt><a href="/man-pages/man1/entrytax.1e.html"><strong>entrytax.1e</strong></a></dt>

 <dd>Identify and retrieve sequence entries containing taxonomy ids</dd>

 <dt><a href="/man-pages/man1/epestfind.1e.html"><strong>epestfind.1e</strong></a></dt>

 <dd>Finds pest motifs as potential proteolytic cleavage sites</dd>

 <dt><a href="/man-pages/man1/eprimer3.1e.html"><strong>eprimer3.1e</strong></a></dt>

 <dd>Picks pcr primers and hybridization oligos</dd>

 <dt><a href="/man-pages/man1/eprimer32.1e.html"><strong>eprimer32.1e</strong></a></dt>

 <dd>Picks pcr primers and hybridization oligos</dd>

 <dt><a href="/man-pages/man1/equicktandem.1e.html"><strong>equicktandem.1e</strong></a></dt>

 <dd>Finds tandem repeats in nucleotide sequences</dd>

 <dt><a href="/man-pages/man1/est2genome.1e.html"><strong>est2genome.1e</strong></a></dt>

 <dd>Align est sequences to genomic dna sequence</dd>

 <dt><a href="/man-pages/man1/etandem.1e.html"><strong>etandem.1e</strong></a></dt>

 <dd>Finds tandem repeats in a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/extractalign.1e.html"><strong>extractalign.1e</strong></a></dt>

 <dd>Extract regions from a sequence alignment</dd>

 <dt><a href="/man-pages/man1/extractfeat.1e.html"><strong>extractfeat.1e</strong></a></dt>

 <dd>Extract features from sequence(s)</dd>

 <dt><a href="/man-pages/man1/extractseq.1e.html"><strong>extractseq.1e</strong></a></dt>

 <dd>Extract regions from a sequence</dd>

 <dt><a href="/man-pages/man1/featcopy.1e.html"><strong>featcopy.1e</strong></a></dt>

 <dd>Reads and writes a feature table</dd>

 <dt><a href="/man-pages/man1/featreport.1e.html"><strong>featreport.1e</strong></a></dt>

 <dd>Reads and writes a feature table</dd>

 <dt><a href="/man-pages/man1/feattext.1e.html"><strong>feattext.1e</strong></a></dt>

 <dd>Return a feature table original text</dd>

 <dt><a href="/man-pages/man1/findkm.1e.html"><strong>findkm.1e</strong></a></dt>

 <dd>Calculate and plot enzyme reaction data</dd>

 <dt><a href="/man-pages/man1/freak.1e.html"><strong>freak.1e</strong></a></dt>

 <dd>Generate residue/base frequency table or plot</dd>

 <dt><a href="/man-pages/man1/fuzznuc.1e.html"><strong>fuzznuc.1e</strong></a></dt>

 <dd>Search for patterns in nucleotide sequences</dd>

 <dt><a href="/man-pages/man1/fuzzpro.1e.html"><strong>fuzzpro.1e</strong></a></dt>

 <dd>Search for patterns in protein sequences</dd>

 <dt><a href="/man-pages/man1/fuzztran.1e.html"><strong>fuzztran.1e</strong></a></dt>

 <dd>Search for patterns in protein sequences (translated)</dd>

 <dt><a href="/man-pages/man1/garnier.1e.html"><strong>garnier.1e</strong></a></dt>

 <dd>Predicts protein secondary structure using gor method</dd>

 <dt><a href="/man-pages/man1/geecee.1e.html"><strong>geecee.1e</strong></a></dt>

 <dd>Calculate fractional gc content of nucleic acid sequences</dd>

 <dt><a href="/man-pages/man1/getorf.1e.html"><strong>getorf.1e</strong></a></dt>

 <dd>Finds and extracts open reading frames (orfs)</dd>

 <dt><a href="/man-pages/man1/godef.1e.html"><strong>godef.1e</strong></a></dt>

 <dd>Find go ontology terms by definition</dd>

 <dt><a href="/man-pages/man1/goname.1e.html"><strong>goname.1e</strong></a></dt>

 <dd>Find go ontology terms by name</dd>

 <dt><a href="/man-pages/man1/goseq.1e.html"><strong>goseq.1e</strong></a></dt>

 <dd>Identify sequence entries containing go terms</dd>

 <dt><a href="/man-pages/man1/helixturnhelix.1e.html"><strong>helixturnhelix.1e</strong></a></dt>

 <dd>Identify nucleic acid-binding motifs in protein sequences</dd>

 <dt><a href="/man-pages/man1/histogramtest.1e.html"><strong>histogramtest.1e</strong></a></dt>

 <dd>Test of graphics</dd>

 <dt><a href="/man-pages/man1/hmoment.1e.html"><strong>hmoment.1e</strong></a></dt>

 <dd>Calculate and plot hydrophobic moment for protein sequence(s)</dd>

 <dt><a href="/man-pages/man1/idtell.1e.html"><strong>idtell.1e</strong></a></dt>

 <dd>Identify the type of a data identifier or query term</dd>

 <dt><a href="/man-pages/man1/iep.1e.html"><strong>iep.1e</strong></a></dt>

 <dd>Calculate the isoelectric point of proteins</dd>

 <dt><a href="/man-pages/man1/infoalign.1e.html"><strong>infoalign.1e</strong></a></dt>

 <dd>Display basic information about a multiple sequence alignment</dd>

 <dt><a href="/man-pages/man1/infobase.1e.html"><strong>infobase.1e</strong></a></dt>

 <dd>Return information on a given nucleotide base</dd>

 <dt><a href="/man-pages/man1/inforesidue.1e.html"><strong>inforesidue.1e</strong></a></dt>

 <dd>Return information on a given amino acid residue</dd>

 <dt><a href="/man-pages/man1/infoseq.1e.html"><strong>infoseq.1e</strong></a></dt>

 <dd>Display basic information about sequences</dd>

 <dt><a href="/man-pages/man1/intconv.1e.html"><strong>intconv.1e</strong></a></dt>

 <dd>Convert ints to ajints and longs to ajlongs</dd>

 <dt><a href="/man-pages/man1/isdbdata.1e.html"><strong>isdbdata.1e</strong></a></dt>

 <dd>Check whether a supplied string is a known data type/identifier name</dd>

 <dt><a href="/man-pages/man1/isdbname.1e.html"><strong>isdbname.1e</strong></a></dt>

 <dd>Check whether a supplied string is a known database name</dd>

 <dt><a href="/man-pages/man1/isochore.1e.html"><strong>isochore.1e</strong></a></dt>

 <dd>Plots isochores in dna sequences</dd>

 <dt><a href="/man-pages/man1/jaspextract.1e.html"><strong>jaspextract.1e</strong></a></dt>

 <dd>Extract data from jaspar</dd>

 <dt><a href="/man-pages/man1/jaspscan.1e.html"><strong>jaspscan.1e</strong></a></dt>

 <dd>Scans dna sequences for transcription factors</dd>

 <dt><a href="/man-pages/man1/lindna.1e.html"><strong>lindna.1e</strong></a></dt>

 <dd>Draws linear maps of dna constructs</dd>

 <dt><a href="/man-pages/man1/listor.1e.html"><strong>listor.1e</strong></a></dt>

 <dd>Write a list file of the logical or of two sets of sequences</dd>

 <dt><a href="/man-pages/man1/makenucseq.1e.html"><strong>makenucseq.1e</strong></a></dt>

 <dd>Create random nucleotide sequences</dd>

 <dt><a href="/man-pages/man1/makeprotseq.1e.html"><strong>makeprotseq.1e</strong></a></dt>

 <dd>Create random protein sequences</dd>

 <dt><a href="/man-pages/man1/marscan.1e.html"><strong>marscan.1e</strong></a></dt>

 <dd>Finds matrix/scaffold recognition (mrs) signatures in dna sequences</dd>

 <dt><a href="/man-pages/man1/martattributes.1e.html"><strong>martattributes.1e</strong></a></dt>

 <dd>Return attributes from a mart dataset from a mart host</dd>

 <dt><a href="/man-pages/man1/martdatasets.1e.html"><strong>martdatasets.1e</strong></a></dt>

 <dd>Return datasets from a mart from a registry</dd>

 <dt><a href="/man-pages/man1/martfilters.1e.html"><strong>martfilters.1e</strong></a></dt>

 <dd>Return filters from a mart dataset from a mart host</dd>

 <dt><a href="/man-pages/man1/martquery.1e.html"><strong>martquery.1e</strong></a></dt>

 <dd>Perform a biomart query</dd>

 <dt><a href="/man-pages/man1/martregistry.1e.html"><strong>martregistry.1e</strong></a></dt>

 <dd>Show biomart registries listed on a host</dd>

 <dt><a href="/man-pages/man1/martseqs.1e.html"><strong>martseqs.1e</strong></a></dt>

 <dd>Show biomart datasets that can return sequences</dd>

 <dt><a href="/man-pages/man1/maskambignuc.1e.html"><strong>maskambignuc.1e</strong></a></dt>

 <dd>Masks all ambiguity characters in nucleotide sequences with n</dd>

 <dt><a href="/man-pages/man1/maskambigprot.1e.html"><strong>maskambigprot.1e</strong></a></dt>

 <dd>Masks all ambiguity characters in protein sequences with x</dd>

 <dt><a href="/man-pages/man1/maskfeat.1e.html"><strong>maskfeat.1e</strong></a></dt>

 <dd>Write a sequence with masked features</dd>

 <dt><a href="/man-pages/man1/maskseq.1e.html"><strong>maskseq.1e</strong></a></dt>

 <dd>Write a sequence with masked regions</dd>

 <dt><a href="/man-pages/man1/matcher.1e.html"><strong>matcher.1e</strong></a></dt>

 <dd>Waterman-eggert local alignment of two sequences</dd>

 <dt><a href="/man-pages/man1/megamerger.1e.html"><strong>megamerger.1e</strong></a></dt>

 <dd>Merge two large overlapping dna sequences</dd>

 <dt><a href="/man-pages/man1/merger.1e.html"><strong>merger.1e</strong></a></dt>

 <dd>Merge two overlapping sequences</dd>

 <dt><a href="/man-pages/man1/msbar.1e.html"><strong>msbar.1e</strong></a></dt>

 <dd>Mutate a sequence</dd>

 <dt><a href="/man-pages/man1/mwcontam.1e.html"><strong>mwcontam.1e</strong></a></dt>

 <dd>Find weights common to multiple molecular weights files</dd>

 <dt><a href="/man-pages/man1/mwfilter.1e.html"><strong>mwfilter.1e</strong></a></dt>

 <dd>Filter noisy data from molecular weights file</dd>

 <dt><a href="/man-pages/man1/needle.1e.html"><strong>needle.1e</strong></a></dt>

 <dd>Needleman-wunsch global alignment of two sequences</dd>

 <dt><a href="/man-pages/man1/needleall.1e.html"><strong>needleall.1e</strong></a></dt>

 <dd>Many-to-many pairwise alignments of two sequence sets</dd>

 <dt><a href="/man-pages/man1/newcoils.1e.html"><strong>newcoils.1e</strong></a></dt>

 <dd>Predicts coils protein secondary structure</dd>

 <dt><a href="/man-pages/man1/newcpgreport.1e.html"><strong>newcpgreport.1e</strong></a></dt>

 <dd>Identify cpg islands in nucleotide sequence(s)</dd>

 <dt><a href="/man-pages/man1/newcpgseek.1e.html"><strong>newcpgseek.1e</strong></a></dt>

 <dd>Identify and report cpg-rich regions in nucleotide sequence(s)</dd>

 <dt><a href="/man-pages/man1/newseq.1e.html"><strong>newseq.1e</strong></a></dt>

 <dd>Create a sequence file from a typed-in sequence</dd>

 <dt><a href="/man-pages/man1/nohtml.1e.html"><strong>nohtml.1e</strong></a></dt>

 <dd>Remove mark-up (e.g. html tags) from an ascii text file</dd>

 <dt><a href="/man-pages/man1/noreturn.1e.html"><strong>noreturn.1e</strong></a></dt>

 <dd>Remove carriage return from ascii files</dd>

 <dt><a href="/man-pages/man1/nospace.1e.html"><strong>nospace.1e</strong></a></dt>

 <dd>Remove whitespace from an ascii text file</dd>

 <dt><a href="/man-pages/man1/notab.1e.html"><strong>notab.1e</strong></a></dt>

 <dd>Replace tabs with spaces in an ascii text file</dd>

 <dt><a href="/man-pages/man1/notseq.1e.html"><strong>notseq.1e</strong></a></dt>

 <dd>Write to file a subset of an input stream of sequences</dd>

 <dt><a href="/man-pages/man1/nthseq.1e.html"><strong>nthseq.1e</strong></a></dt>

 <dd>Write to file a single sequence from an input stream of sequences</dd>

 <dt><a href="/man-pages/man1/nthseqset.1e.html"><strong>nthseqset.1e</strong></a></dt>

 <dd>Reads and writes (returns) one set of sequences from many</dd>

 <dt><a href="/man-pages/man1/octanol.1e.html"><strong>octanol.1e</strong></a></dt>

 <dd>Draw a white-wimley protein hydropathy plot</dd>

 <dt><a href="/man-pages/man1/oddcomp.1e.html"><strong>oddcomp.1e</strong></a></dt>

 <dd>Identify proteins with specified sequence word composition</dd>

 <dt><a href="/man-pages/man1/ontocount.1e.html"><strong>ontocount.1e</strong></a></dt>

 <dd>Count ontology term(s)</dd>

 <dt><a href="/man-pages/man1/ontoget.1e.html"><strong>ontoget.1e</strong></a></dt>

 <dd>Get ontology term(s)</dd>

 <dt><a href="/man-pages/man1/ontogetcommon.1e.html"><strong>ontogetcommon.1e</strong></a></dt>

 <dd>Get common ancestor for terms</dd>

 <dt><a href="/man-pages/man1/ontogetdown.1e.html"><strong>ontogetdown.1e</strong></a></dt>

 <dd>Get ontology term(s) by parent id</dd>

 <dt><a href="/man-pages/man1/ontogetobsolete.1e.html"><strong>ontogetobsolete.1e</strong></a></dt>

 <dd>Get ontology ontology terms</dd>

 <dt><a href="/man-pages/man1/ontogetroot.1e.html"><strong>ontogetroot.1e</strong></a></dt>

 <dd>Get ontology root terms by child identifier</dd>

 <dt><a href="/man-pages/man1/ontogetsibs.1e.html"><strong>ontogetsibs.1e</strong></a></dt>

 <dd>Get ontology term(s) by id with common parent</dd>

 <dt><a href="/man-pages/man1/ontogetup.1e.html"><strong>ontogetup.1e</strong></a></dt>

 <dd>Get ontology term(s) by id of child</dd>

 <dt><a href="/man-pages/man1/ontoisobsolete.1e.html"><strong>ontoisobsolete.1e</strong></a></dt>

 <dd>Report whether an ontology term id is obsolete</dd>

 <dt><a href="/man-pages/man1/ontotext.1e.html"><strong>ontotext.1e</strong></a></dt>

 <dd>Get ontology term(s) original full text</dd>

 <dt><a href="/man-pages/man1/origsplitter.1e.html"><strong>origsplitter.1e</strong></a></dt>

 <dd>Split a sequence into (overlapping) smaller sequences</dd>

 <dt><a href="/man-pages/man1/origunion.1e.html"><strong>origunion.1e</strong></a></dt>

 <dd>Reads sequence fragments and builds one sequence</dd>

 <dt><a href="/man-pages/man1/palindrome.1e.html"><strong>palindrome.1e</strong></a></dt>

 <dd>Finds inverted repeats in nucleotide sequence(s)</dd>

 <dt><a href="/man-pages/man1/pasteseq.1e.html"><strong>pasteseq.1e</strong></a></dt>

 <dd>Insert one sequence into another</dd>

 <dt><a href="/man-pages/man1/patmatdb.1e.html"><strong>patmatdb.1e</strong></a></dt>

 <dd>Searches protein sequences with a sequence motif</dd>

 <dt><a href="/man-pages/man1/patmatmotifs.1e.html"><strong>patmatmotifs.1e</strong></a></dt>

 <dd>Scan a protein sequence with motifs from the prosite database</dd>

 <dt><a href="/man-pages/man1/patmattest.1e.html"><strong>patmattest.1e</strong></a></dt>

 <dd>Test of pattern matching</dd>

 <dt><a href="/man-pages/man1/pepcoil.1e.html"><strong>pepcoil.1e</strong></a></dt>

 <dd>Predicts coiled coil regions in protein sequences</dd>

 <dt><a href="/man-pages/man1/pepdigest.1e.html"><strong>pepdigest.1e</strong></a></dt>

 <dd>Reports on protein proteolytic enzyme or reagent cleavage sites</dd>

 <dt><a href="/man-pages/man1/pepinfo.1e.html"><strong>pepinfo.1e</strong></a></dt>

 <dd>Plot amino acid properties of a protein sequence in parallel.</dd>

 <dt><a href="/man-pages/man1/pepnet.1e.html"><strong>pepnet.1e</strong></a></dt>

 <dd>Draw a helical net for a protein sequence</dd>

 <dt><a href="/man-pages/man1/pepstats.1e.html"><strong>pepstats.1e</strong></a></dt>

 <dd>Calculates statistics of protein properties</dd>

 <dt><a href="/man-pages/man1/pepwheel.1e.html"><strong>pepwheel.1e</strong></a></dt>

 <dd>Draw a helical wheel diagram for a protein sequence</dd>

 <dt><a href="/man-pages/man1/pepwindow.1e.html"><strong>pepwindow.1e</strong></a></dt>

 <dd>Draw a hydropathy plot for a protein sequence</dd>

 <dt><a href="/man-pages/man1/pepwindowall.1e.html"><strong>pepwindowall.1e</strong></a></dt>

 <dd>Draw kyte-doolittle hydropathy plot for a protein alignment</dd>

 <dt><a href="/man-pages/man1/plotcon.1e.html"><strong>plotcon.1e</strong></a></dt>

 <dd>Plot conservation of a sequence alignment</dd>

 <dt><a href="/man-pages/man1/plotorf.1e.html"><strong>plotorf.1e</strong></a></dt>

 <dd>Plot potential open reading frames in a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/polydot.1e.html"><strong>polydot.1e</strong></a></dt>

 <dd>Draw dotplots for all-against-all comparison of a sequence set</dd>

 <dt><a href="/man-pages/man1/preg.1e.html"><strong>preg.1e</strong></a></dt>

 <dd>Regular expression search of protein sequence(s)</dd>

 <dt><a href="/man-pages/man1/prettyplot.1e.html"><strong>prettyplot.1e</strong></a></dt>

 <dd>Draw a sequence alignment with pretty formatting</dd>

 <dt><a href="/man-pages/man1/prettyseq.1e.html"><strong>prettyseq.1e</strong></a></dt>

 <dd>Write a nucleotide sequence and its translation to file</dd>

 <dt><a href="/man-pages/man1/prima.1e.html"><strong>prima.1e</strong></a></dt>

 <dd>Selects primers for pcr and dna amplification.</dd>

 <dt><a href="/man-pages/man1/primers.1e.html"><strong>primers.1e</strong></a></dt>

 <dd>Simple version of primer3 to pick pcr primers</dd>

 <dt><a href="/man-pages/man1/primersearch.1e.html"><strong>primersearch.1e</strong></a></dt>

 <dd>Search dna sequences for matches with primer pairs</dd>

 <dt><a href="/man-pages/man1/printsextract.1e.html"><strong>printsextract.1e</strong></a></dt>

 <dd>Extract data from prints database for use by pscan</dd>

 <dt><a href="/man-pages/man1/profit.1e.html"><strong>profit.1e</strong></a></dt>

 <dd>Scan one or more sequences with a simple frequency matrix</dd>

 <dt><a href="/man-pages/man1/prophecy.1e.html"><strong>prophecy.1e</strong></a></dt>

 <dd>Create frequency matrix or profile from a multiple alignment</dd>

 <dt><a href="/man-pages/man1/prophet.1e.html"><strong>prophet.1e</strong></a></dt>

 <dd>Scan one or more sequences with a gribskov or henikoff profile</dd>

 <dt><a href="/man-pages/man1/prosextract.1e.html"><strong>prosextract.1e</strong></a></dt>

 <dd>Processes the prosite motif database for use by patmatmotifs</dd>

 <dt><a href="/man-pages/man1/pscan.1e.html"><strong>pscan.1e</strong></a></dt>

 <dd>Scans protein sequence(s) with fingerprints from the prints database</dd>

 <dt><a href="/man-pages/man1/psiphi.1e.html"><strong>psiphi.1e</strong></a></dt>

 <dd>Calculates phi and psi torsion angles from protein coordinates</dd>

 <dt><a href="/man-pages/man1/rebaseextract.1e.html"><strong>rebaseextract.1e</strong></a></dt>

 <dd>Process the rebase database for use by restriction enzyme applications</dd>

 <dt><a href="/man-pages/man1/recoder.1e.html"><strong>recoder.1e</strong></a></dt>

 <dd>Find restriction sites to remove (mutate) with no translation change</dd>

 <dt><a href="/man-pages/man1/redata.1e.html"><strong>redata.1e</strong></a></dt>

 <dd>Retrieve information from rebase restriction enzyme database</dd>

 <dt><a href="/man-pages/man1/remap.1e.html"><strong>remap.1e</strong></a></dt>

 <dd>Display restriction enzyme binding sites in a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/restover.1e.html"><strong>restover.1e</strong></a></dt>

 <dd>Find restriction enzymes producing a specific overhang</dd>

 <dt><a href="/man-pages/man1/restrict.1e.html"><strong>restrict.1e</strong></a></dt>

 <dd>Report restriction enzyme cleavage sites in a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/revseq.1e.html"><strong>revseq.1e</strong></a></dt>

 <dd>Reverse and complement a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/seealso.1e.html"><strong>seealso.1e</strong></a></dt>

 <dd>Finds programs with similar function to a specified program</dd>

 <dt><a href="/man-pages/man1/seqcount.1e.html"><strong>seqcount.1e</strong></a></dt>

 <dd>Reads and counts sequences</dd>

 <dt><a href="/man-pages/man1/seqgo.1e.html"><strong>seqgo.1e</strong></a></dt>

 <dd>Extract a list of go terms from a sequence entry</dd>

 <dt><a href="/man-pages/man1/seqinfo.1e.html"><strong>seqinfo.1e</strong></a></dt>

 <dd>Returns sequence information</dd>

 <dt><a href="/man-pages/man1/seqmatchall.1e.html"><strong>seqmatchall.1e</strong></a></dt>

 <dd>All-against-all word comparison of a sequence set</dd>

 <dt><a href="/man-pages/man1/seqret.1e.html"><strong>seqret.1e</strong></a></dt>

 <dd>Reads and writes (returns) sequences</dd>

 <dt><a href="/man-pages/man1/seqretall.1e.html"><strong>seqretall.1e</strong></a></dt>

 <dd>Reads and writes (returns) a set of sequences one at a time</dd>

 <dt><a href="/man-pages/man1/seqretallfeat.1e.html"><strong>seqretallfeat.1e</strong></a></dt>

 <dd>Reads and writes (returns) one or more sequences</dd>

 <dt><a href="/man-pages/man1/seqretset.1e.html"><strong>seqretset.1e</strong></a></dt>

 <dd>Reads and writes (returns) a set of sequences all at once</dd>

 <dt><a href="/man-pages/man1/seqretsetall.1e.html"><strong>seqretsetall.1e</strong></a></dt>

 <dd>Reads and writes (returns) many sets of sequences</dd>

 <dt><a href="/man-pages/man1/seqretsingle.1e.html"><strong>seqretsingle.1e</strong></a></dt>

 <dd>Reads and writes (returns) a single sequence</dd>

 <dt><a href="/man-pages/man1/seqretsplit.1e.html"><strong>seqretsplit.1e</strong></a></dt>

 <dd>Reads sequences and writes them to individual files</dd>

 <dt><a href="/man-pages/man1/seqrettype.1e.html"><strong>seqrettype.1e</strong></a></dt>

 <dd>Reads and writes (returns) sequences</dd>

 <dt><a href="/man-pages/man1/seqtax.1e.html"><strong>seqtax.1e</strong></a></dt>

 <dd>Extract a list of taxonomy ids from a sequence entry</dd>

 <dt><a href="/man-pages/man1/seqxref.1e.html"><strong>seqxref.1e</strong></a></dt>

 <dd>Retrieve all database cross-references for a sequence entry</dd>

 <dt><a href="/man-pages/man1/seqxrefall.1e.html"><strong>seqxrefall.1e</strong></a></dt>

 <dd>Retrieve all cross-references for a sequence record</dd>

 <dt><a href="/man-pages/man1/seqxrefget.1e.html"><strong>seqxrefget.1e</strong></a></dt>

 <dd>Retrieve all cross-referenced data for a sequence entry</dd>

 <dt><a href="/man-pages/man1/servertell.1e.html"><strong>servertell.1e</strong></a></dt>

 <dd>Display information about a public server</dd>

 <dt><a href="/man-pages/man1/showalign.1e.html"><strong>showalign.1e</strong></a></dt>

 <dd>Display a multiple sequence alignment in pretty format</dd>

 <dt><a href="/man-pages/man1/showdb.1e.html"><strong>showdb.1e</strong></a></dt>

 <dd>Displays information on configured databases</dd>

 <dt><a href="/man-pages/man1/showfeat.1e.html"><strong>showfeat.1e</strong></a></dt>

 <dd>Display features of a sequence in pretty format</dd>

 <dt><a href="/man-pages/man1/showorf.1e.html"><strong>showorf.1e</strong></a></dt>

 <dd>Display a nucleotide sequence and translation in pretty format</dd>

 <dt><a href="/man-pages/man1/showpep.1e.html"><strong>showpep.1e</strong></a></dt>

 <dd>Displays protein sequences with features in pretty format</dd>

 <dt><a href="/man-pages/man1/showseq.1e.html"><strong>showseq.1e</strong></a></dt>

 <dd>Displays sequences with features in pretty format</dd>

 <dt><a href="/man-pages/man1/showserver.1e.html"><strong>showserver.1e</strong></a></dt>

 <dd>Displays information on configured servers</dd>

 <dt><a href="/man-pages/man1/shuffleseq.1e.html"><strong>shuffleseq.1e</strong></a></dt>

 <dd>Shuffles a set of sequences maintaining composition</dd>

 <dt><a href="/man-pages/man1/sigcleave.1e.html"><strong>sigcleave.1e</strong></a></dt>

 <dd>Reports on signal cleavage sites in a protein sequence</dd>

 <dt><a href="/man-pages/man1/silent.1e.html"><strong>silent.1e</strong></a></dt>

 <dd>Find restriction sites to insert (mutate) with no translation change</dd>

 <dt><a href="/man-pages/man1/sirna.1e.html"><strong>sirna.1e</strong></a></dt>

 <dd>Finds sirna duplexes in mrna</dd>

 <dt><a href="/man-pages/man1/sixpack.1e.html"><strong>sixpack.1e</strong></a></dt>

 <dd>Display a dna sequence with 6-frame translation and orfs</dd>

 <dt><a href="/man-pages/man1/sizeseq.1e.html"><strong>sizeseq.1e</strong></a></dt>

 <dd>Sort sequences by size</dd>

 <dt><a href="/man-pages/man1/skipredundant.1e.html"><strong>skipredundant.1e</strong></a></dt>

 <dd>Remove redundant sequences from an input set</dd>

 <dt><a href="/man-pages/man1/skipseq.1e.html"><strong>skipseq.1e</strong></a></dt>

 <dd>Reads and writes (returns) sequences, skipping first few</dd>

 <dt><a href="/man-pages/man1/splitsource.1e.html"><strong>splitsource.1e</strong></a></dt>

 <dd>Split sequence(s) into original source sequences</dd>

 <dt><a href="/man-pages/man1/splitter.1e.html"><strong>splitter.1e</strong></a></dt>

 <dd>Split sequence(s) into smaller sequences</dd>

 <dt><a href="/man-pages/man1/sqltest.1e.html"><strong>sqltest.1e</strong></a></dt>

 <dd>Demonstration of sql server access</dd>

 <dt><a href="/man-pages/man1/stretcher.1e.html"><strong>stretcher.1e</strong></a></dt>

 <dd>Needleman-wunsch rapid global alignment of two sequences</dd>

 <dt><a href="/man-pages/man1/stssearch.1e.html"><strong>stssearch.1e</strong></a></dt>

 <dd>Search a dna database for matches with a set of sts primers</dd>

 <dt><a href="/man-pages/man1/supermatcher.1e.html"><strong>supermatcher.1e</strong></a></dt>

 <dd>Calculate approximate local pair-wise alignments of larger sequences</dd>

 <dt><a href="/man-pages/man1/syco.1e.html"><strong>syco.1e</strong></a></dt>

 <dd>Draw synonymous codon usage statistic plot for a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/taxget.1e.html"><strong>taxget.1e</strong></a></dt>

 <dd>Get taxon(s)</dd>

 <dt><a href="/man-pages/man1/taxgetdown.1e.html"><strong>taxgetdown.1e</strong></a></dt>

 <dd>Get descendants of taxon(s)</dd>

 <dt><a href="/man-pages/man1/taxgetrank.1e.html"><strong>taxgetrank.1e</strong></a></dt>

 <dd>Get parents of taxon(s)</dd>

 <dt><a href="/man-pages/man1/taxgetspecies.1e.html"><strong>taxgetspecies.1e</strong></a></dt>

 <dd>Get all species under taxon(s)</dd>

 <dt><a href="/man-pages/man1/taxgetup.1e.html"><strong>taxgetup.1e</strong></a></dt>

 <dd>Get parents of taxon(s)</dd>

 <dt><a href="/man-pages/man1/taxseq.1e.html"><strong>taxseq.1e</strong></a></dt>

 <dd>Identify sequence entries containing taxonomy ids</dd>

 <dt><a href="/man-pages/man1/tcode.1e.html"><strong>tcode.1e</strong></a></dt>

 <dd>Identify protein-coding regions using fickett testcode statistic</dd>

 <dt><a href="/man-pages/man1/testplot.1e.html"><strong>testplot.1e</strong></a></dt>

 <dd>Test plot</dd>

 <dt><a href="/man-pages/man1/textget.1e.html"><strong>textget.1e</strong></a></dt>

 <dd>Get text data entries</dd>

 <dt><a href="/man-pages/man1/textsearch.1e.html"><strong>textsearch.1e</strong></a></dt>

 <dd>Search the textual description of sequence(s)</dd>

 <dt><a href="/man-pages/man1/texttotext.1e.html"><strong>texttotext.1e</strong></a></dt>

 <dd>Convert a plain text format to a different format</dd>

 <dt><a href="/man-pages/man1/texttoxml.1e.html"><strong>texttoxml.1e</strong></a></dt>

 <dd>Convert a plain text format to an xml format</dd>

 <dt><a href="/man-pages/man1/textvalid.1e.html"><strong>textvalid.1e</strong></a></dt>

 <dd>Check a plain text format is valid</dd>

 <dt><a href="/man-pages/man1/tfextract.1e.html"><strong>tfextract.1e</strong></a></dt>

 <dd>Process transfac transcription factor database for use by tfscan</dd>

 <dt><a href="/man-pages/man1/tfm.1e.html"><strong>tfm.1e</strong></a></dt>

 <dd>Displays full documentation for an application</dd>

 <dt><a href="/man-pages/man1/tfscan.1e.html"><strong>tfscan.1e</strong></a></dt>

 <dd>Identify transcription factor binding sites in dna sequences</dd>

 <dt><a href="/man-pages/man1/tmap.1e.html"><strong>tmap.1e</strong></a></dt>

 <dd>Predict and plot transmembrane segments in protein sequences</dd>

 <dt><a href="/man-pages/man1/tranalign.1e.html"><strong>tranalign.1e</strong></a></dt>

 <dd>Generate an alignment of nucleic coding regions from aligned proteins</dd>

 <dt><a href="/man-pages/man1/transeq.1e.html"><strong>transeq.1e</strong></a></dt>

 <dd>Translate nucleic acid sequences</dd>

 <dt><a href="/man-pages/man1/treetypedisplay.1e.html"><strong>treetypedisplay.1e</strong></a></dt>

 <dd>Test of graphics</dd>

 <dt><a href="/man-pages/man1/trimest.1e.html"><strong>trimest.1e</strong></a></dt>

 <dd>Remove poly-a tails from nucleotide sequences</dd>

 <dt><a href="/man-pages/man1/trimseq.1e.html"><strong>trimseq.1e</strong></a></dt>

 <dd>Remove unwanted characters from start and end of sequence(s)</dd>

 <dt><a href="/man-pages/man1/trimspace.1e.html"><strong>trimspace.1e</strong></a></dt>

 <dd>Remove extra whitespace from an ascii text file</dd>

 <dt><a href="/man-pages/man1/twofeat.1e.html"><strong>twofeat.1e</strong></a></dt>

 <dd>Finds neighbouring pairs of features in sequence(s)</dd>

 <dt><a href="/man-pages/man1/union.1e.html"><strong>union.1e</strong></a></dt>

 <dd>Concatenate multiple sequences into a single sequence</dd>

 <dt><a href="/man-pages/man1/urlget.1e.html"><strong>urlget.1e</strong></a></dt>

 <dd>Get urls of data resources</dd>

 <dt><a href="/man-pages/man1/vectorstrip.1e.html"><strong>vectorstrip.1e</strong></a></dt>

 <dd>Removes vectors from the ends of nucleotide sequence(s)</dd>

 <dt><a href="/man-pages/man1/water.1e.html"><strong>water.1e</strong></a></dt>

 <dd>Smith-waterman local alignment of sequences</dd>

 <dt><a href="/man-pages/man1/whichdb.1e.html"><strong>whichdb.1e</strong></a></dt>

 <dd>Search all sequence databases for an entry and retrieve it</dd>

 <dt><a href="/man-pages/man1/wobble.1e.html"><strong>wobble.1e</strong></a></dt>

 <dd>Plot third base position variability in a nucleotide sequence</dd>

 <dt><a href="/man-pages/man1/wordcount.1e.html"><strong>wordcount.1e</strong></a></dt>

 <dd>Count and extract unique words in molecular sequence(s)</dd>

 <dt><a href="/man-pages/man1/wordfinder.1e.html"><strong>wordfinder.1e</strong></a></dt>

 <dd>Match large sequences against one or more other sequences</dd>

 <dt><a href="/man-pages/man1/wordmatch.1e.html"><strong>wordmatch.1e</strong></a></dt>

 <dd>Finds regions of identity (exact matches) of two sequences</dd>

 <dt><a href="/man-pages/man1/wossdata.1e.html"><strong>wossdata.1e</strong></a></dt>

 <dd>Finds programs by edam data</dd>

 <dt><a href="/man-pages/man1/wossinput.1e.html"><strong>wossinput.1e</strong></a></dt>

 <dd>Finds programs by edam input data</dd>

 <dt><a href="/man-pages/man1/wossname.1e.html"><strong>wossname.1e</strong></a></dt>

 <dd>Finds programs by keywords in their short description</dd>

 <dt><a href="/man-pages/man1/wossoperation.1e.html"><strong>wossoperation.1e</strong></a></dt>

 <dd>Finds programs by edam operation</dd>

 <dt><a href="/man-pages/man1/wossoutput.1e.html"><strong>wossoutput.1e</strong></a></dt>

 <dd>Finds programs by edam output data</dd>

 <dt><a href="/man-pages/man1/wossparam.1e.html"><strong>wossparam.1e</strong></a></dt>

 <dd>Finds programs by edam parameter</dd>

 <dt><a href="/man-pages/man1/wosstopic.1e.html"><strong>wosstopic.1e</strong></a></dt>

 <dd>Finds programs by edam topic</dd>

 <dt><a href="/man-pages/man1/xmltotext.1e.html"><strong>xmltotext.1e</strong></a></dt>

 <dd>Convert an xml document to a plain text format</dd>

 <dt><a href="/man-pages/man1/xmltoxml.1e.html"><strong>xmltoxml.1e</strong></a></dt>

 <dd>Convert an xml document to a different format</dd>

 <dt><a href="/man-pages/man1/xmltransform.1e.html"><strong>xmltransform.1e</strong></a></dt>

 <dd>Convert an xml document to a different format using a supplied xslt file</dd>

 <dt><a href="/man-pages/man1/yank.1e.html"><strong>yank.1e</strong></a></dt>

 <dd>Add a sequence reference (a full usa) to a list file</dd>

</dl>

</li>

</ul>
  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
