<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>alimask: Add mask line to a multiple sequence alignment</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Add mask line to a multiple sequence alignment">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="alimask (1) manual">
  <meta name="twitter:description" content="Add mask line to a multiple sequence alignment">
  <meta name="twitter:image" content="https://www.carta.tech/images/hmmer-alimask-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/alimask.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="alimask (1) manual" />
  <meta property="og:description" content="Add mask line to a multiple sequence alignment" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/hmmer-alimask-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">alimask<small> (1)</small></h1>
        <p class="lead">Add mask line to a multiple sequence alignment</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/alimask.1.html">
      <span itemprop="name">alimask: Add mask line to a multiple sequence alignment</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/hmmer/">
      <span itemprop="name">hmmer</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/alimask.1.html">
      <span itemprop="name">alimask: Add mask line to a multiple sequence alignment</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>alimask</strong> <em>[options]</em> <em>&lt;msafile&gt;</em> <em>&lt;postmsafile&gt;</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>alimask</strong> is used to apply a mask line to a multiple sequence alignment, based on provided alignment or model coordinates. When <strong>hmmbuild</strong> receives a masked alignment as input, it produces a profile model in which the emission probabilities at masked positions are set to match the background frequency, rather than being set based on observed frequencies in the alignment. Position-specific insertion and deletion rates are not altered, even in masked regions. <strong>alimask</strong> autodetects input format, and produces masked alignments in Stockholm format. <em>&lt;msafile&gt;</em> may contain only one sequence alignment.</p><p>A common motivation for masking a region in an alignment is that the region contains a simple tandem repeat that is observed to cause an unacceptably high rate of false positive hits.</p><p>In the simplest case, a mask range is given in coordinates relative to the input alignment, using <strong>--alirange</strong><em> &lt;s&gt;</em><strong>.</strong> However it is more often the case that the region to be masked has been identified in coordinates relative to the profile model (e.g. based on recognizing a simple repeat pattern in false hit alignments or in the HMM logo). Not all alignment columns are converted to match state positions in the profile (see the <em>--symfrac</em> flag for <strong>hmmbuild</strong> for discussion), so model positions do not necessarily match up to alignment column positions. To remove the burden of converting model positions to alignment positions, <strong>alimask</strong> accepts the mask range input in model coordinates as well, using <strong>--modelrange</strong><em> &lt;s&gt;</em><strong>.</strong> When using this flag, <em>alimask</em> determines which alignment positions would be identified by <strong>hmmbuild</strong> as match states, a process that requires that all <strong>hmmbuild</strong> flags impacting that decision be supplied to <strong>alimask</strong>. It is for this reason that many of the <strong>hmmbuild</strong> flags are also used by <strong>alimask</strong>.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and all available options.</p>
  </dd>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Direct the summary output to file <em>&lt;f&gt;</em>, rather than to <em>stdout</em>.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR SPECIFYING MASK RANGE</h2>
        <div class="sectioncontent">
<p>A single mask range is given as a dash-separated pair, like <em>--modelrange 10-20</em> and multiple ranges may be submitted as a comma-separated list, <em>--modelrange 10-20,30-42</em>.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--modelrange</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Supply the given range(s) in model coordinates.</p>
  </dd>
  <dt>
    <p><strong>--alirange</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Supply the given range(s) in alignment coordinates.</p>
  </dd>
  <dt>
    <p><strong>--apendmask</strong></p>
  </dt>
  <dd>
    <p>Add to the existing mask found with the alignment. The default is to overwrite any existing mask.</p>
  </dd>
  <dt>
    <p><strong>--model2ali</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Rather than actually produce the masked alignment, simply print model range(s) corresponding to input alignment range(s).</p>
  </dd>
  <dt>
    <p><strong>--ali2model</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Rather than actually produce the masked alignment, simply print alignment range(s) corresponding to input model range(s).</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR SPECIFYING THE ALPHABET</h2>
        <div class="sectioncontent">
<p>The alphabet type (amino, DNA, or RNA) is autodetected by default, by looking at the composition of the <em>msafile</em>. Autodetection is normally quite reliable, but occasionally alphabet type may be ambiguous and autodetection can fail (for instance, on tiny toy alignments of just a few residues). To avoid this, or to increase robustness in automated analysis pipelines, you may specify the alphabet type of <em>msafile</em> with these options.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--amino</strong></p>
  </dt>
  <dd>
    <p>Specify that all sequences in <em>msafile</em> are proteins.</p>
  </dd>
  <dt>
    <p><strong>--dna</strong></p>
  </dt>
  <dd>
    <p>Specify that all sequences in <em>msafile</em> are DNAs.</p>
  </dd>
  <dt>
    <p><strong>--rna</strong></p>
  </dt>
  <dd>
    <p>Specify that all sequences in <em>msafile</em> are RNAs.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING PROFILE CONSTRUCTION</h2>
        <div class="sectioncontent">
<p>These options control how consensus columns are defined in an alignment.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--fast</strong></p>
  </dt>
  <dd>
    <p>Define consensus columns as those that have a fraction &gt;= <strong>symfrac</strong> of residues as opposed to gaps. (See below for the <strong>--symfrac</strong> option.) This is the default.</p>
  </dd>
  <dt>
    <p><strong>--hand</strong></p>
  </dt>
  <dd>
    <p>Define consensus columns in next profile using reference annotation to the multiple alignment. This allows you to define any consensus columns you like.</p>
  </dd>
  <dt>
    <p><strong>--symfrac</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Define the residue fraction threshold necessary to define a consensus column when using the <strong>--fast</strong> option. The default is 0.5. The symbol fraction in each column is calculated after taking relative sequence weighting into account, and ignoring gap characters corresponding to ends of sequence fragments (as opposed to internal insertions/deletions). Setting this to 0.0 means that every alignment column will be assigned as consensus, which may be useful in some cases. Setting it to 1.0 means that only columns that include 0 gaps (internal insertions/deletions) will be assigned as consensus.</p>
  </dd>
  <dt>
    <p><strong>--fragthresh</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>We only want to count terminal gaps as deletions if the aligned sequence is known to be full-length, not if it is a fragment (for instance, because only part of it was sequenced). HMMER uses a simple rule to infer fragments: if the sequence length L is less than or equal to a fraction <em>&lt;x&gt;</em> times the alignment length in columns, then the sequence is handled as a fragment. The default is 0.5. Setting <strong>--fragthresh</strong><em>0</em> will define no (nonempty) sequence as a fragment; you might want to do this if you know you've got a carefully curated alignment of full-length sequences. Setting <strong>--fragthresh</strong><em>1</em> will define all sequences as fragments; you might want to do this if you know your alignment is entirely composed of fragments, such as translated short reads in metagenomic shotgun data.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING RELATIVE WEIGHTS</h2>
        <div class="sectioncontent">
<p>HMMER uses an ad hoc sequence weighting algorithm to downweight closely related sequences and upweight distantly related ones. This has the effect of making models less biased by uneven phylogenetic representation. For example, two identical sequences would typically each receive half the weight that one sequence would.  These options control which algorithm gets used.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--wpb</strong></p>
  </dt>
  <dd>
    <p>Use the Henikoff position-based sequence weighting scheme [Henikoff and Henikoff, J. Mol. Biol. 243:574, 1994].  This is the default.</p>
  </dd>
  <dt>
    <p><strong>--wgsc</strong></p>
  </dt>
  <dd>
    <p>Use the Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et al, J. Mol. Biol. 235:1067, 1994].</p>
  </dd>
  <dt>
    <p><strong>--wblosum</strong></p>
  </dt>
  <dd>
    <p>Use the same clustering scheme that was used to weight data in calculating BLOSUM subsitution matrices [Henikoff and Henikoff, Proc. Natl. Acad. Sci 89:10915, 1992]. Sequences are single-linkage clustered at an identity threshold (default 0.62; see <em>--wid</em>) and within each cluster of c sequences, each sequence gets relative weight 1/c.</p>
  </dd>
  <dt>
    <p><strong>--wnone</strong></p>
  </dt>
  <dd>
    <p>No relative weights. All sequences are assigned uniform weight.</p>
  </dd>
  <dt>
    <p><strong>--wid</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the identity threshold used by single-linkage clustering when using <em>--wblosum</em>. Invalid with any other weighting scheme. Default is 0.62.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--informat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Declare that the input <em>msafile</em> is in format <em>&lt;s&gt;</em>. Currently the accepted multiple alignment sequence file formats include Stockholm, Aligned FASTA, Clustal, NCBI PSI-BLAST, PHYLIP, Selex, and UCSC SAM A2M. Default is to autodetect the format of the file.</p>
  </dd>
  <dt>
    <p><strong>--seed</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Seed the random number generator with <em>&lt;n&gt;</em>, an integer &gt;= 0. If <em>&lt;n&gt;</em> is nonzero, any stochastic simulations will be reproducible; the same command will give the same results. If <em>&lt;n&gt;</em> is 0, the random number generator is seeded arbitrarily, and stochastic simulations will vary from run to run of the same command. The default seed is 42.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO alimask&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/hmmer.1.html"><strong>hmmer</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the HMMER package.</p><p>For complete documentation, see the user guide that came with your HMMER distribution (Userguide.pdf); or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2013 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="aliki-rt.1.html"><span aria-hidden="true">&larr;</span> aliki-rt.1: Measurement tool for impulse responses</a></li>
   <li class="next"><a href="alistat.1.html">alistat.1:  <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
