<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>amap: Protein multiple alignment by sequence annealing</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Protein multiple alignment by sequence annealing">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="amap (1) manual">
  <meta name="twitter:description" content="Protein multiple alignment by sequence annealing">
  <meta name="twitter:image" content="https://www.carta.tech/images/amap-align-amap-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/amap.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="amap (1) manual" />
  <meta property="og:description" content="Protein multiple alignment by sequence annealing" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/amap-align-amap-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">amap<small> (1)</small></h1>
        <p class="lead">Protein multiple alignment by sequence annealing</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/amap.1.html">
      <span itemprop="name">amap: Protein multiple alignment by sequence annealing</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/amap-align/">
      <span itemprop="name">amap-align</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/amap.1.html">
      <span itemprop="name">amap: Protein multiple alignment by sequence annealing</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>amap</strong> [<strong></strong><em>OPTION</em>] [<em>MFAFILE</em>] [<em>MFAFILE</em>]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>AMAP is a tool to perform multiple alignment of peptidic sequences. It utilizes posterior decoding, and a sequence-annealing alignment, instead of the traditional progressive alignment method. It is the only alignment program that allows one to control the sensitivity / specificity tradeoff. It is based on the ProbCons source code, but uses alignment metric accuracy and eliminates the consistency transformation.</p><p>In its default configuration, AMAP is tuned to maximize the expected Alignment Metric Accuracy (AMA) score - a new alignment accuracy measure, based on a metric for the multiple-alignment space, which integrates sensitivity and specificity into a single balanced measure. AMA is defined as the fraction of correctly aligned residues (either to another residue or to a gap) out of the total number of residues in all the sequences.</p><p><strong>amap</strong> aligns sequences provided in MFA format. This format consists of multiple sequences. Each sequence in MFA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (&ldquo;&gt;&rdquo;) symbol in the first column.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p><strong>-clustalw</strong></p><p>use CLUSTALW output format instead of MFA</p><p><strong>-c</strong> <strong>--consistency</strong> <em>REPS</em></p><p>use 0 &lt;= <em>REPS</em> &lt;= 5 (default: 0) passes of consistency transformation</p><p><strong>-ir</strong> <strong>--iterative-refinement</strong> <em>REPS</em></p><p>use 0 &lt;= <em>REPS</em> &lt;=1000 (default: 0) passes of iterative-refinement</p><p><strong>-pre</strong> <strong>--pre-training</strong> <em>REPS</em></p><p>use 0 &lt;= <em>REPS</em> &lt;= 20 (default: 0) rounds of pretraining</p><p><strong>-pairs</strong></p><p>generate all-pairs pairwise alignments</p><p><strong>-viterbi</strong></p><p>use Viterbi algorithm to generate all pairs (automatically enables <strong>-pairs</strong>)</p><p><strong>-v</strong> <strong>--verbose</strong></p><p>Report progress while aligning (default: off)</p><p><strong>-annot</strong> <em>FILENAME</em></p><p>write annotation for multiple alignment to <em>FILENAME</em></p><p><strong>-t</strong> <strong>--train</strong> <em>FILENAME</em></p><p>compute EM transition probabilities, store in <em>FILENAME</em> (default: no training)</p><p><strong>-e</strong> <strong>--emissions</strong></p><p>also reestimate emission probabilities (default: off)</p><p><strong>-p</strong> <strong>--paramfile</strong> <em>FILENAME</em></p><p>read parameters from <em>FILENAME</em> (default: )</p><p><strong>-a</strong> <strong>--alignment-order</strong></p><p>print sequences in alignment order rather than input order (default: off)</p><p><strong>-g</strong> <strong>--gap-factor</strong> <em>GF</em></p><p>use <em>GF</em> as the gap-factor parameter, set to 0 for best sensitivity, higher values for better specificity (default: 0.5)</p><p><strong>-w</strong> <strong>--edge-weight-threshold</strong> <em>W</em></p><p>stop the sequence annealing process when best edge has lower weight than <em>W</em>, set to 0 for best sensitivity, higher values for better specificity (default: 0)</p><p><strong>-prog</strong> <strong>--progressive</strong></p><p>use progressive alignment instead of sequence annealing alignment (default: off)</p><p><strong>-noreorder</strong> <strong>--no-edge-reordering</strong></p><p>disable reordering of edges during sequence annealing alignment (default: off)</p><p><strong>-maxstep</strong> <strong>--use-max-stepsize</strong></p><p>use maximum improvement step size instead of tGf edge ranking (default: off)</p><p><strong>-print</strong> <strong>--print-posteriors</strong></p><p>only print the posterior probability matrices (default: off)</p><p><strong>-gui</strong> <em>START</em> <em>STEP</em></p><p>print output for the AMAP Display Java based GUI (default: ) starting at weight <em>START</em> (default: infinity) with step size <em>STEP</em> (default: )</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">
<p>To run AMAP with the default options change to the <em>align</em> directory and type:</p><p><strong>% amap &lt;multi-fasta-file-name&gt;</strong></p><p>If no file name is provided the list of options are printed.</p><p>In order to use the AMAP Display run AMAP with the -gui option, and save the output to a file, then use the file as the input to AmapDisplay. For example, type:</p><p><strong>% align/amap -gui examples/BB12020.tfa &gt; examples/BB12020.tfa.out</strong></p><p><strong>% java -jar display/AmapDisplay.jar examples/BB12020.tfa.out</strong></p><p>(on Debian systems, the <em>examples</em> directory is in <em>/usr/share/doc/amap-align/examples</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">NOTE</h2>
        <div class="sectioncontent">
<p>In older versions ( &lt;&nbsp;2.0-1) of the package for Debian(TM) systems, the <strong>amap</strong> command was renamed <strong>amap-align</strong> because there was already another tool called <strong>amap</strong> (which performs some computer network diagnostics). A symbolic link <strong>amap-align</strong> is still provided for upgrade purposes but will be removed in Debian releases posterior to Etch (Debian&nbsp;4.0).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO amap&hellip;</h2>
        <div class="sectioncontent">
<p>The current version of AMAP uses the PROBCONS 1.09 code base for some of the input/output procedures, and for the calculation of posterior probabilities (see PROBCONS.README in <em>/usr/share/doc/amap-align/</em>). Future releases might implement the algorithm using a new independent code base.</p><p>On Debian(TM) systems, <a href="../man1/probcons.1.html"><strong>probcons</strong>(1)</a> is available in the probcons package.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REFERENCES</h2>
        <div class="sectioncontent">
<p>For more details on AMAP and AMA, see Schwartz, Ariel S., Myers, Eugene W., and Pachter, Lior. Alignment Metric Accuracy (Submitted for publication). For more details on sequence-annealing, see Schwartz, Ariel S. and Pachter, Lior. Multiple Alignment by Sequence Annealing (Submitted for publication).</p><p>PROBCONS was published in Do, C.B., Mahabhashyam, M.S.P., Brudno, M., and Batzoglou, S. 2005. PROBCONS: Probabilistic Consistency-based Multiple Sequence Alignment. Genome Research 15: 330-340.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p><strong>Ariel Schwartz</strong> &lt;sariel@cs.berkeley.edu&gt;</p><ul>
<li><p>Upstream author of AMAP</p></li>
</ul><p><strong>Chuong Do</strong></p><ul>
<li><p>Wrote Probcons, on which AMAP is based.</p></li>
</ul><p><strong>Charles Plessy&gt;</strong> &lt;charles-debian-nospam@plessy.org&gt;</p><ul>
<li><p>Wrote this manpage in DocBook XML for the Debian distribution.</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">
<p>AMAP, PROBCONS, and this manual page have been made freely available as PUBLIC DOMAIN software and hence are not subject to copyright in the United States. This system and/or any portion of the source code may be used, modified, or redistributed without restrictions. AMAP, PROBCONS and this manual page are distributed WITHOUT WARRANTY, express or implied. The authors accept NO LEGAL LIABILITY OR RESPONSIBILITY for loss due to reliance on the program.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="alurestream.1.html"><span aria-hidden="true">&larr;</span> alurestream.1: Example program for sound streaming with alure</a></li>
   <li class="next"><a href="amarok.1.html">amarok.1: The audio player for kde <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
