<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>anfo-tool: Process native anfo binary files</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Process native anfo binary files">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="anfo-tool (1) manual">
  <meta name="twitter:description" content="Process native anfo binary files">
  <meta name="twitter:image" content="https://www.carta.tech/images/anfo-anfo-tool-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/anfo-tool.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="anfo-tool (1) manual" />
  <meta property="og:description" content="Process native anfo binary files" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/anfo-anfo-tool-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">anfo-tool<small> (1)</small></h1>
        <p class="lead">Process native anfo binary files</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/anfo-tool.1.html">
      <span itemprop="name">anfo-tool: Process native anfo binary files</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/anfo/">
      <span itemprop="name">anfo</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/anfo-tool.1.html">
      <span itemprop="name">anfo-tool: Process native anfo binary files</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>anfo-tool [</strong> <em>option</em> <strong>|</strong> <em>pattern</em> <strong>... ]</strong></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>anfo-tool</strong> is used to filter, process and convert the files created by <strong>anfo</strong>. Every pattern on the command line is wildcard expanded, then for every input file (or the standard input, if no pattern is given), <strong>anfo-tool</strong> builds a chain of input filters, it then merges these input streams in one of several ways, splits the result up into multiple output streams, each of which can have a chain of output filter applied.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<h3>General Options</h3>
<p>These options apply globally and modify the behavior of the whole program.  They can be placed anywhere in the command line.</p>
<dl class='dl-vertical'>
  <dt>
    -V, --version
  </dt>
  <dd>
    <p>Print version number and exit.</p>
  </dd>
  <dt>
    -q, --quiet
  </dt>
  <dd>
    <p>Suppress all output except fatal error messages.</p>
  </dd>
  <dt>
    -v, --verbose
  </dt>
  <dd>
    <p>Produce more output, including progress indicators for most operations.</p>
  </dd>
  <dt>
    --debug
  </dt>
  <dd>
    <p>Produce debugging output in addition to progress information.</p>
  </dd>
  <dt>
    -n, --dry-run
  </dt>
  <dd>
    <p>Parse command line, optionally print a description of the intended operations, then exit.</p>
  </dd>
  <dt>
    --vmem X
  </dt>
  <dd>
    <p>Limit virtual memory to <em>X</em> megabytes.  If memory runs out, <em>anfo-tool</em> tries to free up memory by forgetting about big files, e.g. genomes. Use this option to avoid swapping or out-of-memory conditions when operations involve big or multiple genomes.</p>
  </dd>

</dl>

<h3>Setting Parameters</h3>
<p>A parameter can be set multiple times on the command line and will overwrite previous settings.  Any filter option that needs a parameter picks up the last definition that appeared before the filter option.</p>
<dl class='dl-vertical'>
  <dt>
    --set-slope S
  </dt>
  <dd>
    <p>Set the slope parameter to <em>S</em>. The <em>slope</em> is used together with the <em>intercept</em> where filters apply to alignment scores; alignments scoring no worse than <em>slope * (length - intercept)</em> are considered good.  The default is 7.5.</p>
  </dd>
  <dt>
    --set-intercept L
  </dt>
  <dd>
    <p>Set the intercept parameter to <em>L</em>. The <em>intercept</em> is used together with the <em>slope</em> where filters apply to alignment scores; alignments scoring no worse than <em>slope * (length - intercept)</em> are considered good.  The default is 20.</p>
  </dd>
  <dt>
    --set-context C
  </dt>
  <dd>
    <p>Set the context parameter to <em>C</em>. The context is the number of surrounding bases of the reference included when printing alignments in text form.  The default is 0.</p>
  </dd>
  <dt>
    --set-genome G
  </dt>
  <dd>
    <p>Set the genome parameter to <em>G</em>. Many filters will only consider the best alignments to this specific genome if it is set.  If no genome is set, the globally best alignment is used.</p>
  </dd>
  <dt>
    --clear-genome
  </dt>
  <dd>
    <p>Clear the genome parameter.  Filters apply to the globally best alignment afterwards.</p>
  </dd>

</dl>

<h3>Filter Options</h3>
<p>Filters can be applied before merging the inputs or after splitting the back up.</p>
<dl class='dl-vertical'>
  <dt>
    -s, --sort-pos=n
  </dt>
  <dd>
    <p>Sort by alignment position while buffering no more than <em>n</em> MiB in memory.  If a genome is set, alignments to that genome are used.</p>
  </dd>
  <dt>
    -S, --sort-name=n
  </dt>
  <dd>
    <p>Sort by read name while buffering no more than <em>n</em> MiB in memory.</p>
  </dd>
  <dt>
    -l, --filter-length=L
  </dt>
  <dd>
    <p>Retain alignments only for reads of at least <em>L</em> bases length.  The reads themselves are kept.</p>
  </dd>
  <dt>
    -f, --filter-score
  </dt>
  <dd>
    <p>Retain alignments only if their score is good enough. <em>Uses</em>slope<em>and</em>intercept<em>.</em></p>
  </dd>
  <dt>
    --filter-mapq=Q
  </dt>
  <dd>
    <p>Remove alignments with mapping quality below <em>Q</em>.</p>
  </dd>
  <dt>
    -h, --filter-hit=SEQ
  </dt>
  <dd>
    <p>Keep only reads that have a hit to a sequence named <em>SEQ</em>. If <em>SEQ</em> is empty, reads are kept if they have any hit.  If the <em>genome</em> parameter is set, only hits to that genome count.</p>
  </dd>
  <dt>
    --delete-hit=SEQ
  </dt>
  <dd>
    <p>Delete alignments to <em>SEQ</em>. If <em>SEQ</em> is empty, all alignments are deleted.  If the <em>genome</em> parameter is set, only alignments to that genome are deleted.</p>
  </dd>
  <dt>
    --filter-qual=Q
  </dt>
  <dd>
    <p>Mask out bases with quality below <em>Q</em>. Such a base is replaced by the <em>N</em> ambiguity code.</p>
  </dd>
  <dt>
    --multiplicity=N
  </dt>
  <dd>
    <p>Keep only reads of molecules that have been sequenced at least <em>N</em> times.  Reads are considered to come from the same original molecule if their aligned coordinates are identical.</p>
  </dd>
  <dt>
    --subsample=F
  </dt>
  <dd>
    <p>Subsample a fraction <em>F</em> of the results.  Every read is independently and randomly choosen to be kept or not.</p>
  </dd>
  <dt>
    --inside-regions=FILE
  </dt>
  <dd>
    <p>Read a list of regions from <em>FILE</em>, then keep only alignments that overlap an annotated region.</p>
  </dd>
  <dt>
    --outside-regions=FILE
  </dt>
  <dd>
    <p>Read a list of regions from <em>FILE</em>, then keep only alignments that do not overlap an annotated region.</p>
  </dd>

</dl>

<h3>Special Filters</h3>

<dl class='dl-vertical'>
  <dt>
    -d, --rmdup=Q
  </dt>
  <dd>
    <p>Remove PCR duplicates, clamp quality scores to <em>Q</em>. Two reads are considered to be duplicates, if their aligned coordinates are identical.  If a <em>genome</em> is set, the best alignment to that genome is used, else the globally best alignment.  Both alignments must be good, as determined by <em>slope</em>and<em>intercept</em>. For a set of duplicates, a consensus is called, generally increasing the quality scores.  If a resulting quality score exceeds <em>Q</em>, it is set to <em>Q</em>. This filter requires the input to be sorted by alignment coordinate on the selected <em>genome</em>.</p><p><em>--duct-tape=NAME</em> Duct-tape overlapping alignments into contigs and call a consensus for them.  If a <em>genome</em> is set, alignments to that genome are used, else the globally best alignments.  This filter requires input to be sorted by alignment coordinate on the genome.  Output is a set of contigs, every position gets assigned a consensus base, a quality score and likelihoods for every possible diallele.  (It is called duct-taping because it kind of looks like an assembly, but is not nearly as solid.)</p>
  </dd>
  <dt>
    --edit-header=ED
  </dt>
  <dd>
    <p>Invoke the editor <em>ED</em> on the text representaion of the stream's header.  This can be used to clean up header that have accumulated too much cruft.</p>
  </dd>

</dl>

<h3>Merging Filters</h3>
<p>Exactly one merging filter should be given on the command line, all filter options occuring before that are part of the input filter chains, all further filters become output chains.  If no merging filter is given, <em>--concat</em> is assumed, and all filters are input filters.</p>
<dl class='dl-vertical'>
  <dt>
    -c, --concat
  </dt>
  <dd>
    <p>Concatenate all input streams in the order they appear on the command line.</p>
  </dd>
  <dt>
    -m, --merge
  </dt>
  <dd>
    <p>Merge sorted input streams, producing a sorted result.  All inputs must be sorted in the same way.</p>
  </dd>
  <dt>
    -j, --join
  </dt>
  <dd>
    <p>Join input streams and retain the single best hits to each genome. Every input stream must contain a record for every read, reads are buffered in memory until all of their hits are collected.  This way, joining works well if all inputs are nearly in the same order.  If reads are missing from some streams, joining them will waste memory.</p>
  </dd>
  <dt>
    --mega-merge
  </dt>
  <dd>
    <p>Merge many streams such as those produced by running <strong>anfo-sge</strong>. Streams that operated on the same reads are joined, then everything is merged.</p>
  </dd>

</dl>

<h3>Output Options</h3>
<p>If an output option is given on the command line, the current output filter chain is ended and a new one is started.  If no output option is given, a textual representation of the final stream is written to <em>stdout</em>. All output options accept <em>-</em> to write to stdout.</p>
<dl class='dl-vertical'>
  <dt>
    -o, --output FILE
  </dt>
  <dd>
    <p>Write native binary stream (a compressed protobuf message) to <em>FILE</em>. Writing a binary stream and reading it back in is lossless.</p>
  </dd>
  <dt>
    --output-text FILE
  </dt>
  <dd>
    <p>Write protobuf text stream to <em>FILE</em>. If the necessary genomes are available, a textual representation of the alignments is included.  If the <em>context</em> parameter is set, that many additional bases of the reference upstream and downstream from the alignment are included.</p>
  </dd>
  <dt>
    --output-sam=FILE
  </dt>
  <dd>
    <p>Write alignments in SAM format to <em>FILE</em>.</p>
  </dd>
  <dt>
    --output-glz FILE
  </dt>
  <dd>
    <p>Write contigs in GLZ 0.9 format to <em>FILE</em>. Generating GLZ only works after application of <em>--duct-tape</em>, every contigs becomes a GLZ record.</p>
  </dd>
  <dt>
    --output-3aln FILE
  </dt>
  <dd>
    <p>Write contigs in a table based format to <em>FILE</em>. The format is still subject to change, see the source code for detailed documentation.</p>
  </dd>
  <dt>
    --output-fasta FILE
  </dt>
  <dd>
    <p>Write alignments(!) in FastA format to <em>FILE</em>. Alignments are writte as pair of reference and query sequence, aligned coordinates are indicated in the description of the query sequence.  If the <em>context</em> parameter is set, that many additional bases of the reference upstream and downstream from the alignment are included.  This format is not suggested for any serious use, it exists to support legacy applications.</p>
  </dd>
  <dt>
    --output-fastq FILE
  </dt>
  <dd>
    <p>Write sequences(!) in FastQ format to <em>FILE</em>. Writing FastQ effectively reconstitutes the input to <strong>ANFO</strong> if no filtering was done on the results.</p>
  </dd>
  <dt>
    --output-table FILE
  </dt>
  <dd>
    <p>Write per-alignment statistics to <em>FILE</em>. The file has three colums:Ã‚sequence length, alignment score, difference to next best alignment.  It is mainly useful to analyze/visualize the distribution of alignment scores.</p>
  </dd>
  <dt>
    --stats FILE
  </dt>
  <dd>
    <p>Write simple statistics to <em>FILE</em>. This results in some simple summary statistics of a whole stream: number of aligned sequences, average length, GC content.</p>
  </dd>

</dl>


        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ENVIRONMENT</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    ANFO_PATH
  </dt>
  <dd>
    <p>Colon separated list of directories searched for genome and index files.</p>
  </dd>
  <dt>
    ANFO_TEMP
  </dt>
  <dd>
    <p>Temporary space used for sorting of large files.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FILES</h2>
        <div class="sectioncontent">
<p><em>/etc/popt</em></p><p>The system wide configuration file for <a href="../man3/popt.3.html"><strong>popt</strong>(3)</a>. <strong>anfo-tool</strong> identifies itself as "anfo-tool" to popt.</p><p><em>~/.popt</em></p><p>Per user configuration file for <a href="../man3/popt.3.html"><strong>popt</strong>(3)</a>.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>The command line of this tools is way too complicated and its semantics are counterintuitive.  Using <strong>anfo-tool</strong> is probably best avoided in most cases, the <strong>guile</strong> bindings should provide a much more scalable and easier to understand interface.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Udo Stenzel &lt;udo_stenzel@eva.mpg.de&gt;</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO anfo-tool&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man1/anfo.1.html"><strong>anfo</strong>(1)</a>, <a href="../man1/fa2dna.1.html"><strong>fa2dna</strong>(1)</a> <a href="../man3/popt.3.html"><strong>popt</strong>(3)</a>, <strong>fasta</strong>(5)</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="anfo.1.html"><span aria-hidden="true">&larr;</span> anfo.1: Find best alignment of short reads to database</a></li>
   <li class="next"><a href="animate-im6.1.html">animate-im6.1: Animates an image or image sequence on any x server. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
