<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>aragorn: Detect trna genes in nucleotide sequences</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Detect trna genes in nucleotide sequences">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="aragorn (1) manual">
  <meta name="twitter:description" content="Detect trna genes in nucleotide sequences">
  <meta name="twitter:image" content="https://www.carta.tech/images/aragorn-aragorn-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/aragorn.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="aragorn (1) manual" />
  <meta property="og:description" content="Detect trna genes in nucleotide sequences" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/aragorn-aragorn-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">aragorn<small> (1)</small></h1>
        <p class="lead">Detect trna genes in nucleotide sequences</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/aragorn.1.html">
      <span itemprop="name">aragorn: Detect trna genes in nucleotide sequences</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/aragorn/">
      <span itemprop="name">aragorn</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/aragorn.1.html">
      <span itemprop="name">aragorn: Detect trna genes in nucleotide sequences</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>aragorn</strong> [<em>OPTION</em>]... <em>FILE</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p><strong>-m</strong></p><p>Search for tmRNA genes.</p><p><strong>-t</strong></p><p>Search for tRNA genes. By default, all are detected. If one of <strong>-m</strong> or <strong>-t</strong> is specified, then the other is not detected unless specified as well.</p><p><strong>-mt</strong></p><p>Search for Metazoan mitochondrial tRNA genes. tRNA genes with introns not detected. <strong>-i</strong>, <strong>-sr</strong> switchs ignored. Composite Metazoan mitochondrial genetic code used.</p><p><strong>-mtmam</strong></p><p>Search for Mammalian mitochondrial tRNA genes. <strong>-i</strong>, <strong>-sr</strong> switchs ignored. <strong>-tv</strong> switch set. Mammalian mitochondrial genetic code used.</p><p><strong>-mtx</strong></p><p>Same as <strong>-mt</strong> but low scoring tRNA genes are not reported.</p><p><strong>-mtd</strong></p><p>Overlapping metazoan mitochondrial tRNA genes on opposite strands are reported.</p><p><strong>-gc</strong>[<em>num</em>]</p><p>Use the GenBank transl_table = [<em>num</em>] genetic code. Individual modifications can be appended using <em>,BBB</em>=&lt;aa&gt; B = A,C,G, or T. &lt;aa&gt; is the three letter code for an amino-acid. More than one modification can be specified. eg <strong>-gcvert</strong>,aga=Trp,agg=Trp uses the Vertebrate Mitochondrial code and the codons AGA and AGG changed to Tryptophan.</p><p><strong>-gcstd</strong></p><p>Use standard genetic code.</p><p><strong>-gcmet</strong></p><p>Use composite Metazoan mitochondrial genetic code.</p><p><strong>-gcvert</strong></p><p>Use Vertebrate mitochondrial genetic code.</p><p><strong>-gcinvert</strong></p><p>Use Invertebrate mitochondrial genetic code.</p><p><strong>-gcyeast</strong></p><p>Use Yeast mitochondrial genetic code.</p><p><strong>-gcprot</strong></p><p>Use Mold/Protozoan/Coelenterate mitochondrial genetic code.</p><p><strong>-gcciliate</strong></p><p>Use Ciliate genetic code.</p><p><strong>-gcflatworm</strong></p><p>Use Echinoderm/Flatworm mitochondrial genetic code</p><p><strong>-gceuplot</strong></p><p>Use Euplotid genetic code.</p><p><strong>-gcbact</strong></p><p>Use Bacterial/Plant Chloroplast genetic code.</p><p><strong>-gcaltyeast</strong></p><p>Use alternative Yeast genetic code.</p><p><strong>-gcascid</strong></p><p>Use Ascidian Mitochondrial genetic code.</p><p><strong>-gcaltflat</strong></p><p>Use alternative Flatworm Mitochondrial genetic code.</p><p><strong>-gcblep</strong></p><p>Use Blepharisma genetic code.</p><p><strong>-gcchloroph</strong></p><p>Use Chlorophycean Mitochondrial genetic code.</p><p><strong>-gctrem</strong></p><p>Use Trematode Mitochondrial genetic code.</p><p><strong>-gcscen</strong></p><p>Use Scenedesmus obliquus Mitochondrial genetic code.</p><p><strong>-gcthraust</strong></p><p>Use Thraustochytrium Mitochondrial genetic code.</p><p><strong>-tv</strong></p><p>Do not search for mitochondrial TV replacement loop tRNA genes. Only relevant if <strong>-mt</strong> used.</p><p><strong>-c7</strong></p><p>Search for tRNA genes with 7 base C-loops only.</p><p><strong>-i</strong></p><p>Search for tRNA genes with introns in anticodon loop with maximum length 3000 bases. Minimum intron length is 0 bases. Ignored if <strong>-m</strong> is specified.</p><p><strong>-i</strong>[<em>max</em>]</p><p>Search for tRNA genes with introns in anticodon loop with maximum length [<em>max</em>] bases. Minimum intron length is 0 bases. Ignored if <strong>-m</strong> is specified.</p><p><strong>-i</strong>[<em>min</em>],[<em>max</em>]</p><p>Search for tRNA genes with introns in anticodon loop with maximum length [<em>max</em>] bases, and minimum length [<em>min</em>] bases. Ignored if <strong>-m</strong> is specified.</p><p><strong>-io</strong></p><p>Same as <strong>-i</strong>, but allow tRNA genes with long introns to overlap shorter tRNA genes.</p><p><strong>-if</strong></p><p>Same as <strong>-i</strong>, but fix intron between positions 37 and 38 on C-loop (one base after anticodon).</p><p><strong>-ifo</strong></p><p>Same as <strong>-if</strong> and <strong>-io</strong> combined.</p><p><strong>-ir</strong></p><p>Same as <strong>-i</strong>, but report tRNA genes with minimum length [<em>min</em>] bases rather than search for tRNA genes with minimum length [<em>min</em>] bases. With this switch, [<em>min</em>] acts as an output filter, minimum intron length for searching is still 0 bases.</p><p><strong>-c</strong></p><p>Assume that each sequence has a circular topology. Search wraps around each end. Default setting.</p><p><strong>-l</strong></p><p>Assume that each sequence has a linear topology. Search does not wrap.</p><p><strong>-d</strong></p><p>Double. Search both strands of each sequence. Default setting.</p><p><strong>-s</strong> or <strong>-s+</strong></p><p>Single. Do not search the complementary (antisense) strand of each sequence.</p><p><strong>-sc</strong> or <strong>-s-</strong></p><p>Single complementary. Do not search the sense strand of each sequence.</p><p><strong>-ps</strong></p><p>Lower scoring thresholds to 95% of default levels.</p><p><strong>-ps</strong>[<em>num</em>]</p><p>Change scoring thresholds to [<em>num</em>] percent of default levels.</p><p><strong>-rp</strong></p><p>Flag possible pseudogenes (score &lt; 100 or tRNA anticodon loop &lt;&gt; 7 bases long). Note that genes with score &lt; 100 will not be detected or flagged if scoring thresholds are not also changed to below 100% (see -ps switch).</p><p><strong>-seq</strong></p><p>Print out primary sequence.</p><p><strong>-br</strong></p><p>Show secondary structure of tRNA gene primary sequence using round brackets.</p><p><strong>-fasta</strong></p><p>Print out primary sequence in fasta format.</p><p><strong>-fo</strong></p><p>Print out primary sequence in fasta format only (no secondary structure).</p><p><strong>-fon</strong></p><p>Same as <strong>-fo</strong>, with sequence and gene numbering in header.</p><p><strong>-fos</strong></p><p>Same as <strong>-fo</strong>, with no spaces in header.</p><p><strong>-fons</strong></p><p>Same as <strong>-fo</strong>, with sequence and gene numbering, but no spaces.</p><p><strong>-w</strong></p><p>Print out in Batch mode.</p><p><strong>-ss</strong></p><p>Use the stricter canonical 1-2 bp spacer1 and 1 bp spacer2. Ignored if <strong>-mt</strong> set. Default is to allow 3 bp spacer1 and 0-2 bp spacer2, which may degrade selectivity.</p><p><strong>-v</strong></p><p>Verbose. Prints out information during search to STDERR.</p><p><strong>-a</strong></p><p>Print out tRNA domain for tmRNA genes.</p><p><strong>-a7</strong></p><p>Restrict tRNA astem length to a maximum of 7 bases</p><p><strong>-aa</strong></p><p>Display message if predicted iso-acceptor species does not match species in sequence name (if present).</p><p><strong>-j</strong></p><p>Display 4-base sequence on 3&apos; end of astem regardless of predicted amino-acyl acceptor length.</p><p><strong>-jr</strong></p><p>Allow some divergence of 3&apos; amino-acyl acceptor sequence from NCCA.</p><p><strong>-jr4</strong></p><p>Allow some divergence of 3&apos; amino-acyl acceptor sequence from NCCA, and display 4 bases.</p><p><strong>-q</strong></p><p>Do not print configuration line (which switchs and files were used).</p><p><strong>-rn</strong></p><p>Repeat sequence name before summary information.</p><p><strong>-O</strong> [<em>outfile</em>]</p><p>Print output to <em>. If [&apos;outfile</em>] already exists, it is overwritten. By default all output goes to stdout.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>aragorn detects tRNA, mtRNA, and tmRNA genes. A minimum requirement is at least a 32 bit compiler architecture (variable types int and unsigned int are at least 4 bytes long).</p><p>[<em>FILE</em>] is assumed to contain one or more sequences in FASTA format. Results of the search are printed to STDOUT. All switches are optional and case-insensitive. Unless -i is specified, tRNA genes containing introns are not detected.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p>Bjorn Canback &lt;bcanback@acgt.se&gt;, Dean Laslett &lt;gaiaquark@gmail.com&gt;</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REFERENCES</h2>
        <div class="sectioncontent">
<p>Laslett, D. and Canback, B. (2004) ARAGORN, a program for the detection of transfer RNA and transfer-messenger RNA genes in nucleotide sequences Nucleic Acids Research, 32;11-16</p><p>Laslett, D. and Canback, B. (2008) ARWEN: a program to detect tRNA genes in metazoan mitochondrial nucleotide sequences Bioinformatics, 24(2); 172-175.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="ara.1.html"><span aria-hidden="true">&larr;</span> ara.1: A utility for doing boolean regexp queries on the the debian package database</a></li>
   <li class="next"><a href="arandr.1.html">arandr.1: Visual front end for xrandr 1.2 <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
