<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>bcftools: Utilities for the sequence alignment/map (sam) format  bcftools - utilities for the binary call format (bcf) and vcf</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Utilities for the sequence alignment/map (sam) format  bcftools - utilities for the binary call format (bcf) and vcf">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="bcftools (1) manual">
  <meta name="twitter:description" content="Utilities for the sequence alignment/map (sam) format  bcftools - utilities for the binary call format (bcf) and vcf">
  <meta name="twitter:image" content="https://www.carta.tech/images/samtools-bcftools-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/bcftools.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="bcftools (1) manual" />
  <meta property="og:description" content="Utilities for the sequence alignment/map (sam) format  bcftools - utilities for the binary call format (bcf) and vcf" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/samtools-bcftools-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">bcftools<small> (1)</small></h1>
        <p class="lead">Utilities for the sequence alignment/map (sam) format  bcftools - utilities for the binary call format (bcf) and vcf</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/bcftools.1.html">
      <span itemprop="name">bcftools: Utilities for the sequence alignment/map (sam) format  bcftools - utilities for the binary call format (bcf) and vcf</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/samtools/">
      <span itemprop="name">samtools</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/bcftools.1.html">
      <span itemprop="name">bcftools: Utilities for the sequence alignment/map (sam) format  bcftools - utilities for the binary call format (bcf) and vcf</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>samtools view -bt ref_list.txt -o aln.bam aln.sam.gz</p><p>samtools sort aln.bam aln.sorted</p><p>samtools index aln.sorted.bam</p><p>samtools idxstats aln.sorted.bam</p><p>samtools view aln.sorted.bam chr2:20,100,000-20,200,000</p><p>samtools merge out.bam in1.bam in2.bam in3.bam</p><p>samtools faidx ref.fasta</p><p>samtools pileup -vcf ref.fasta aln.sorted.bam</p><p>samtools mpileup -C50 -gf ref.fasta -r chr3:1,000-2,000 in1.bam in2.bam</p><p>samtools tview aln.sorted.bam ref.fasta</p><p>bcftools index in.bcf</p><p>bcftools view in.bcf chr2:100-200 &gt; out.vcf</p><p>bcftools view -Nvm0.99 in.bcf &gt; out.vcf 2&gt; out.afs</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>Samtools is a set of utilities that manipulate alignments in the BAM format. It imports from and exports to the SAM (Sequence Alignment/Map) format, does sorting, merging and indexing, and allows to retrieve reads in any regions swiftly.</p><p>Samtools is designed to work on a stream. It regards an input file `-' as the standard input (stdin) and an output file `-' as the standard output (stdout). Several commands can thus be combined with Unix pipes. Samtools always output warning and error messages to the standard error output (stderr).</p><p>Samtools is also able to open a BAM (not SAM) file on a remote FTP or HTTP server if the BAM file name starts with `ftp://' or `http://'. Samtools checks the current working directory for the index file and will download the index upon absence. Samtools does not retrieve the entire alignment file unless it is asked to do so.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SAMTOOLS COMMANDS AND OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>view</strong></p>
  </dt>
  <dd>
    <p>samtools view [-bchuHS] [-t in.refList] [-o output] [-f reqFlag] [-F skipFlag] [-q minMapQ] [-l library] [-r readGroup] [-R rgFile] &lt;in.bam&gt;|&lt;in.sam&gt; [region1 [...]]</p><p>Extract/print all or sub alignments in SAM or BAM format. If no region is specified, all the alignments will be printed; otherwise only alignments overlapping the specified regions will be output. An alignment may be given multiple times if it is overlapping several regions. A region can be presented, for example, in the following format: `chr2' (the whole chr2), `chr2:1000000' (region starting from 1,000,000bp) or `chr2:1,000,000-2,000,000' (region between 1,000,000 and 2,000,000bp including the end points). The coordinate is 1-based.</p><p><strong>OPTIONS:</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-b</strong></p>
  </dt>
  <dd>
    <p>Output in the BAM format.</p>
  </dd>
  <dt>
    <p><strong>-f</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Only output alignments with all bits in INT present in the FLAG field. INT can be in hex in the format of /^0x[0-9A-F]+/ [0]</p>
  </dd>
  <dt>
    <p><strong>-F</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Skip alignments with bits present in INT [0]</p>
  </dd>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Include the header in the output.</p>
  </dd>
  <dt>
    <p><strong>-H</strong></p>
  </dt>
  <dd>
    <p>Output the header only.</p>
  </dd>
  <dt>
    <p><strong>-l</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Only output reads in library STR [null]</p>
  </dd>
  <dt>
    <p><strong>-o</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>Output file [stdout]</p>
  </dd>
  <dt>
    <p><strong>-q</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Skip alignments with MAPQ smaller than INT [0]</p>
  </dd>
  <dt>
    <p><strong>-r</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Only output reads in read group STR [null]</p>
  </dd>
  <dt>
    <p><strong>-R</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>Output reads in read groups listed in <em>FILE</em> [null]</p>
  </dd>
  <dt>
    <p><strong>-s</strong><em></em><strong>FLOAT</strong></p>
  </dt>
  <dd>
    <p>Fraction of templates/pairs to subsample; the integer part is treated as the seed for the random number generator [-1]</p>
  </dd>
  <dt>
    <p><strong>-S</strong></p>
  </dt>
  <dd>
    <p>Input is in SAM. If @SQ header lines are absent, the <strong>`-t'</strong> option is required.</p>
  </dd>
  <dt>
    <p><strong>-c</strong></p>
  </dt>
  <dd>
    <p>Instead of printing the alignments, only count them and print the total number. All filter options, such as <strong>`-f',</strong> <strong>`-F'</strong> and <strong>`-q'</strong> , are taken into account.</p>
  </dd>
  <dt>
    <p><strong>-t</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>This file is TAB-delimited. Each line must contain the reference name and the length of the reference, one line for each distinct reference; additional fields are ignored. This file also defines the order of the reference sequences in sorting. If you run `samtools faidx &lt;ref.fa&gt;', the resultant index file <em>&lt;ref.fa&gt;.fai</em> can be used as this <em>&lt;in.ref_list&gt;</em> file.</p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Output uncompressed BAM. This option saves time spent on compression/decomprssion and is thus preferred when the output is piped to another samtools command.</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>tview</strong></p>
  </dt>
  <dd>
    <p>samtools tview [<strong>-p</strong> <em>chr:pos</em>] [<strong>-s</strong> <em>STR</em>] [<strong>-d</strong> <em>display</em>] &lt;in.sorted.bam&gt; [ref.fasta]</p><p>Text alignment viewer (based on the ncurses library). In the viewer, press `?' for help and press `g' to check the alignment start from a region in the format like `chr10:10,000,000' or `=10,000,000' when viewing the same reference sequence.</p><p><strong>Options:</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-d</strong><em></em><strong>display</strong></p>
  </dt>
  <dd>
    <p>Output as (H)tml or (C)urses or (T)ext</p>
  </dd>
  <dt>
    <p><strong>-p</strong><em></em><strong>chr:pos</strong></p>
  </dt>
  <dd>
    <p>Go directly to this position</p>
  </dd>
  <dt>
    <p><strong>-s</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Display only reads from this sample or read group</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>mpileup</strong></p>
  </dt>
  <dd>
    <p>samtools mpileup [<strong>-EBugp</strong>] [<strong>-C</strong> <em>capQcoef</em>] [<strong>-r</strong> <em>reg</em>] [<strong>-f</strong> <em>in.fa</em>] [<strong>-l</strong> <em>list</em>] [<strong>-M</strong> <em>capMapQ</em>] [<strong>-Q</strong> <em>minBaseQ</em>] [<strong>-q</strong> <em>minMapQ</em>] <em>in.bam</em> [<em>in2.bam</em> [<em>...</em>]]</p><p>Generate BCF or pileup for one or multiple BAM files. Alignment records are grouped by sample identifiers in @RG header lines. If sample identifiers are absent, each input file is regarded as one sample.</p><p>In the pileup format (without <strong>-u</strong>or<strong>-g</strong>), each line represents a genomic position, consisting of chromosome name, coordinate, reference base, read bases, read qualities and alignment mapping qualities. Information on match, mismatch, indel, strand, mapping quality and start and end of a read are all encoded at the read base column. At this column, a dot stands for a match to the reference base on the forward strand, a comma for a match on the reverse strand, a '&gt;' or '&lt;' for a reference skip, `ACGTN' for a mismatch on the forward strand and `acgtn' for a mismatch on the reverse strand. A pattern `&#92;+[0-9]+[ACGTNacgtn]+' indicates there is an insertion between this reference position and the next reference position. The length of the insertion is given by the integer in the pattern, followed by the inserted sequence. Similarly, a pattern `-[0-9]+[ACGTNacgtn]+' represents a deletion from the reference. The deleted bases will be presented as `*' in the following lines. Also at the read base column, a symbol `^' marks the start of a read. The ASCII of the character following `^' minus 33 gives the mapping quality. A symbol `$' marks the end of a read segment.</p><p><strong>Input Options:</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-6</strong></p>
  </dt>
  <dd>
    <p>Assume the quality is in the Illumina 1.3+ encoding. <strong>-A</strong> Do not skip anomalous read pairs in variant calling.</p>
  </dd>
  <dt>
    <p><strong>-B</strong></p>
  </dt>
  <dd>
    <p>Disable probabilistic realignment for the computation of base alignment quality (BAQ). BAQ is the Phred-scaled probability of a read base being misaligned. Applying this option greatly helps to reduce false SNPs caused by misalignments.</p>
  </dd>
  <dt>
    <p><strong>-b</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>List of input BAM files, one file per line [null]</p>
  </dd>
  <dt>
    <p><strong>-C</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Coefficient for downgrading mapping quality for reads containing excessive mismatches. Given a read with a phred-scaled probability q of being generated from the mapped position, the new mapping quality is about sqrt((INT-q)/INT)*INT. A zero value disables this functionality; if enabled, the recommended value for BWA is 50. [0]</p>
  </dd>
  <dt>
    <p><strong>-d</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>At a position, read maximally <em>INT</em> reads per input BAM. [250]</p>
  </dd>
  <dt>
    <p><strong>-E</strong></p>
  </dt>
  <dd>
    <p>Extended BAQ computation. This option helps sensitivity especially for MNPs, but may hurt specificity a little bit.</p>
  </dd>
  <dt>
    <p><strong>-f</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>The <strong>faidx</strong>-indexed reference file in the FASTA format. The file can be optionally compressed by <strong>razip</strong>. [null]</p>
  </dd>
  <dt>
    <p><strong>-l</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>BED or position list file containing a list of regions or sites where pileup or BCF should be generated [null]</p>
  </dd>
  <dt>
    <p><strong>-q</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Minimum mapping quality for an alignment to be used [0]</p>
  </dd>
  <dt>
    <p><strong>-Q</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Minimum base quality for a base to be considered [13]</p>
  </dd>
  <dt>
    <p><strong>-r</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Only generate pileup in region <em>STR</em> [all sites]</p>
  </dd>
  <dt>
    <p><strong>Output Options:</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>-D</strong></p>
  </dt>
  <dd>
    <p>Output per-sample read depth</p>
  </dd>
  <dt>
    <p><strong>-g</strong></p>
  </dt>
  <dd>
    <p>Compute genotype likelihoods and output them in the binary call format (BCF).</p>
  </dd>
  <dt>
    <p><strong>-S</strong></p>
  </dt>
  <dd>
    <p>Output per-sample Phred-scaled strand bias P-value</p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Similar to <strong>-g</strong> except that the output is uncompressed BCF, which is preferred for piping.</p>
  </dd>
  <dt>
    <p><strong>Options for Genotype Likelihood Computation (for -g or -u):</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>-e</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Phred-scaled gap extension sequencing error probability. Reducing <em>INT</em> leads to longer indels. [20]</p>
  </dd>
  <dt>
    <p><strong>-h</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Coefficient for modeling homopolymer errors. Given an <em>l</em>-long homopolymer run, the sequencing error of an indel of size <em>s</em> is modeled as <em>INT</em>*<em>s</em>/<em>l</em>. [100]</p>
  </dd>
  <dt>
    <p><strong>-I</strong></p>
  </dt>
  <dd>
    <p>Do not perform INDEL calling</p>
  </dd>
  <dt>
    <p><strong>-L</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Skip INDEL calling if the average per-sample depth is above <em>INT</em>. [250]</p>
  </dd>
  <dt>
    <p><strong>-o</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Phred-scaled gap open sequencing error probability. Reducing <em>INT</em> leads to more indel calls. [40]</p>
  </dd>
  <dt>
    <p><strong>-p</strong></p>
  </dt>
  <dd>
    <p>Apply -m and -F thresholds per sample to increase sensitivity of calling. By default both options are applied to reads pooled from all samples.</p>
  </dd>
  <dt>
    <p><strong>-P</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Comma dilimited list of platforms (determined by <strong>@RG-PL</strong>) from which indel candidates are obtained. It is recommended to collect indel candidates from sequencing technologies that have low indel error rate such as ILLUMINA. [all]</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>reheader</strong></p>
  </dt>
  <dd>
    <p>samtools reheader &lt;in.header.sam&gt; &lt;in.bam&gt;</p><p>Replace the header in <em>in.bam</em> with the header in <em>in.header.sam.</em> This command is much faster than replacing the header with a BAM-&gt;SAM-&gt;BAM conversion.</p>
  </dd>
  <dt>
    <p><strong>cat</strong></p>
  </dt>
  <dd>
    <p>samtools cat [-h header.sam] [-o out.bam] &lt;in1.bam&gt; &lt;in2.bam&gt; [ ... ]</p><p>Concatenate BAMs. The sequence dictionary of each input BAM must be identical, although this command does not check this. This command uses a similar trick to <strong>reheader</strong> which enables fast BAM concatenation.</p>
  </dd>
  <dt>
    <p><strong>sort</strong></p>
  </dt>
  <dd>
    <p>samtools sort [-nof] [-m maxMem] &lt;in.bam&gt; &lt;out.prefix&gt;</p><p>Sort alignments by leftmost coordinates. File <em>&lt;out.prefix&gt;.bam</em> will be created. This command may also create temporary files <em>&lt;out.prefix&gt;.%d.bam</em> when the whole alignment cannot be fitted into memory (controlled by option -m).</p><p><strong>OPTIONS:</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-o</strong></p>
  </dt>
  <dd>
    <p>Output the final alignment to the standard output.</p>
  </dd>
  <dt>
    <p><strong>-n</strong></p>
  </dt>
  <dd>
    <p>Sort by read names rather than by chromosomal coordinates</p>
  </dd>
  <dt>
    <p><strong>-f</strong></p>
  </dt>
  <dd>
    <p>Use <em>&lt;out.prefix&gt;</em> as the full output path and do not append <em>.bam</em> suffix.</p>
  </dd>
  <dt>
    <p><strong>-m</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Approximately the maximum required memory. [500000000]</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>merge</strong></p>
  </dt>
  <dd>
    <p>samtools merge [-nur1f] [-h inh.sam] [-R reg] &lt;out.bam&gt; &lt;in1.bam&gt; &lt;in2.bam&gt; [...]</p><p>Merge multiple sorted alignments. The header reference lists of all the input BAM files, and the @SQ headers of <em>inh.sam</em>, if any, must all refer to the same set of reference sequences. The header reference list and (unless overridden by <strong>-h</strong>) `@' headers of <em>in1.bam</em> will be copied to <em>out.bam</em>, and the headers of other files will be ignored.</p><p><strong>OPTIONS:</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-1</strong></p>
  </dt>
  <dd>
    <p>Use zlib compression level 1 to comrpess the output</p>
  </dd>
  <dt>
    <p><strong>-f</strong></p>
  </dt>
  <dd>
    <p>Force to overwrite the output file if present.</p>
  </dd>
  <dt>
    <p><strong>-h</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>Use the lines of <em>FILE</em> as `@' headers to be copied to <em>out.bam</em>, replacing any header lines that would otherwise be copied from <em>in1.bam</em>. (<em>FILE</em> is actually in SAM format, though any alignment records it may contain are ignored.)</p>
  </dd>
  <dt>
    <p><strong>-n</strong></p>
  </dt>
  <dd>
    <p>The input alignments are sorted by read names rather than by chromosomal coordinates</p>
  </dd>
  <dt>
    <p><strong>-R</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Merge files in the specified region indicated by <em>STR</em> [null]</p>
  </dd>
  <dt>
    <p><strong>-r</strong></p>
  </dt>
  <dd>
    <p>Attach an RG tag to each alignment. The tag value is inferred from file names.</p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Uncompressed BAM output</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>index</strong></p>
  </dt>
  <dd>
    <p>samtools index &lt;aln.bam&gt;</p><p>Index sorted alignment for fast random access. Index file <em>&lt;aln.bam&gt;.bai</em> will be created.</p>
  </dd>
  <dt>
    <p><strong>idxstats</strong></p>
  </dt>
  <dd>
    <p>samtools idxstats &lt;aln.bam&gt;</p><p>Retrieve and print stats in the index file. The output is TAB delimited with each line consisting of reference sequence name, sequence length, # mapped reads and # unmapped reads.</p>
  </dd>
  <dt>
    <p><strong>faidx</strong></p>
  </dt>
  <dd>
    <p>samtools faidx &lt;ref.fasta&gt; [region1 [...]]</p><p>Index reference sequence in the FASTA format or extract subsequence from indexed reference sequence. If no region is specified, <strong>faidx</strong> will index the file and create <em>&lt;ref.fasta&gt;.fai</em> on the disk. If regions are speficified, the subsequences will be retrieved and printed to stdout in the FASTA format. The input file can be compressed in the <strong>RAZF</strong> format.</p>
  </dd>
  <dt>
    <p><strong>fixmate</strong></p>
  </dt>
  <dd>
    <p>samtools fixmate &lt;in.nameSrt.bam&gt; &lt;out.bam&gt;</p><p>Fill in mate coordinates, ISIZE and mate related flags from a name-sorted alignment.</p>
  </dd>
  <dt>
    <p><strong>rmdup</strong></p>
  </dt>
  <dd>
    <p>samtools rmdup [-sS] &lt;input.srt.bam&gt; &lt;out.bam&gt;</p><p>Remove potential PCR duplicates: if multiple read pairs have identical external coordinates, only retain the pair with highest mapping quality. In the paired-end mode, this command <strong>ONLY</strong> works with FR orientation and requires ISIZE is correctly set. It does not work for unpaired reads (e.g. two ends mapped to different chromosomes or orphan reads).</p><p><strong>OPTIONS:</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-s</strong></p>
  </dt>
  <dd>
    <p>Remove duplicate for single-end reads. By default, the command works for paired-end reads only.</p>
  </dd>
  <dt>
    <p><strong>-S</strong></p>
  </dt>
  <dd>
    <p>Treat paired-end reads and single-end reads.</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>calmd</strong></p>
  </dt>
  <dd>
    <p>samtools calmd [-EeubSr] [-C capQcoef] &lt;aln.bam&gt; &lt;ref.fasta&gt;</p><p>Generate the MD tag. If the MD tag is already present, this command will give a warning if the MD tag generated is different from the existing tag. Output SAM by default.</p><p><strong>OPTIONS:</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-A</strong></p>
  </dt>
  <dd>
    <p>When used jointly with <strong>-r</strong> this option overwrites the original base quality.</p>
  </dd>
  <dt>
    <p><strong>-e</strong></p>
  </dt>
  <dd>
    <p>Convert a the read base to = if it is identical to the aligned reference base. Indel caller does not support the = bases at the moment.</p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Output uncompressed BAM</p>
  </dd>
  <dt>
    <p><strong>-b</strong></p>
  </dt>
  <dd>
    <p>Output compressed BAM</p>
  </dd>
  <dt>
    <p><strong>-S</strong></p>
  </dt>
  <dd>
    <p>The input is SAM with header lines</p>
  </dd>
  <dt>
    <p><strong>-C</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Coefficient to cap mapping quality of poorly mapped reads. See the <strong>pileup</strong> command for details. [0]</p>
  </dd>
  <dt>
    <p><strong>-r</strong></p>
  </dt>
  <dd>
    <p>Compute the BQ tag (without -A) or cap base quality by BAQ (with -A).</p>
  </dd>
  <dt>
    <p><strong>-E</strong></p>
  </dt>
  <dd>
    <p>Extended BAQ calculation. This option trades specificity for sensitivity, though the effect is minor.</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>targetcut</strong></p>
  </dt>
  <dd>
    <p>samtools targetcut [-Q minBaseQ] [-i inPenalty] [-0 em0] [-1 em1] [-2 em2] [-f ref] &lt;in.bam&gt;</p><p>This command identifies target regions by examining the continuity of read depth, computes haploid consensus sequences of targets and outputs a SAM with each sequence corresponding to a target. When option <strong>-f</strong> is in use, BAQ will be applied. This command is <strong>only</strong> designed for cutting fosmid clones from fosmid pool sequencing [Ref. Kitzman et al. (2010)].</p>
  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <p><strong>phase</strong></p>
  </dt>
  <dd>
    <p>samtools phase [-AF] [-k len] [-b prefix] [-q minLOD] [-Q minBaseQ] &lt;in.bam&gt;</p><p>Call and phase heterozygous SNPs. <strong>OPTIONS:</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-A</strong></p>
  </dt>
  <dd>
    <p>Drop reads with ambiguous phase.</p>
  </dd>
  <dt>
    <p><strong>-b</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Prefix of BAM output. When this option is in use, phase-0 reads will be saved in file <strong>STR</strong>.0.bam and phase-1 reads in <strong>STR</strong>.1.bam. Phase unknown reads will be randomly allocated to one of the two files. Chimeric reads with switch errors will be saved in <strong>STR</strong>.chimeric.bam. [null]</p>
  </dd>
  <dt>
    <p><strong>-F</strong></p>
  </dt>
  <dd>
    <p>Do not attempt to fix chimeric reads.</p>
  </dd>
  <dt>
    <p><strong>-k</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Maximum length for local phasing. [13]</p>
  </dd>
  <dt>
    <p><strong>-q</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Minimum Phred-scaled LOD to call a heterozygote. [40]</p>
  </dd>
  <dt>
    <p><strong>-Q</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Minimum base quality to be used in het calling. [13]</p>
  </dd>

</dl>

  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BCFTOOLS COMMANDS AND OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>view</strong></p>
  </dt>
  <dd>
    <p><strong>bcftools view</strong> [<strong>-AbFGNQSucgv</strong>] [<strong>-D</strong> <em>seqDict</em>] [<strong>-l</strong> <em>listLoci</em>] [<strong>-s</strong> <em>listSample</em>] [<strong>-i</strong> <em>gapSNPratio</em>] [<strong>-t</strong> <em>mutRate</em>] [<strong>-p</strong> <em>varThres</em>] [<strong>-m</strong> <em>varThres</em>] [<strong>-P</strong> <em>prior</em>] [<strong>-1</strong> <em>nGroup1</em>] [<strong>-d</strong> <em>minFrac</em>] [<strong>-U</strong> <em>nPerm</em>] [<strong>-X</strong> <em>permThres</em>] [<strong>-T</strong> <em>trioType</em>] <em>in.bcf</em> [<em>region</em>]</p><p>Convert between BCF and VCF, call variant candidates and estimate allele frequencies.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>Input/Output Options:</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>-A</strong></p>
  </dt>
  <dd>
    <p>Retain all possible alternate alleles at variant sites. By default, the view command discards unlikely alleles.</p>
  </dd>
  <dt>
    <p><strong>-b</strong></p>
  </dt>
  <dd>
    <p>Output in the BCF format. The default is VCF.</p>
  </dd>
  <dt>
    <p><strong>-D</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>Sequence dictionary (list of chromosome names) for VCF-&gt;BCF conversion [null]</p>
  </dd>
  <dt>
    <p><strong>-F</strong></p>
  </dt>
  <dd>
    <p>Indicate PL is generated by r921 or before (ordering is different).</p>
  </dd>
  <dt>
    <p><strong>-G</strong></p>
  </dt>
  <dd>
    <p>Suppress all individual genotype information.</p>
  </dd>
  <dt>
    <p><strong>-l</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>List of sites at which information are outputted [all sites]</p>
  </dd>
  <dt>
    <p><strong>-N</strong></p>
  </dt>
  <dd>
    <p>Skip sites where the REF field is not A/C/G/T</p>
  </dd>
  <dt>
    <p><strong>-Q</strong></p>
  </dt>
  <dd>
    <p>Output the QCALL likelihood format</p>
  </dd>
  <dt>
    <p><strong>-s</strong><em></em><strong>FILE</strong></p>
  </dt>
  <dd>
    <p>List of samples to use. The first column in the input gives the sample names and the second gives the ploidy, which can only be 1 or 2. When the 2nd column is absent, the sample ploidy is assumed to be 2. In the output, the ordering of samples will be identical to the one in <em>FILE</em>. [null]</p>
  </dd>
  <dt>
    <p><strong>-S</strong></p>
  </dt>
  <dd>
    <p>The input is VCF instead of BCF.</p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Uncompressed BCF output (force -b).</p>
  </dd>
  <dt>
    <p><strong>Consensus/Variant Calling Options:</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>-c</strong></p>
  </dt>
  <dd>
    <p>Call variants using Bayesian inference. This option automatically invokes option <strong>-e</strong>.</p>
  </dd>
  <dt>
    <p><strong>-d</strong><em></em><strong>FLOAT</strong></p>
  </dt>
  <dd>
    <p>When <strong>-v</strong> is in use, skip loci where the fraction of samples covered by reads is below FLOAT. [0]</p>
  </dd>
  <dt>
    <p><strong>-e</strong></p>
  </dt>
  <dd>
    <p>Perform max-likelihood inference only, including estimating the site allele frequency, testing Hardy-Weinberg equlibrium and testing associations with LRT.</p>
  </dd>
  <dt>
    <p><strong>-g</strong></p>
  </dt>
  <dd>
    <p>Call per-sample genotypes at variant sites (force -c)</p>
  </dd>
  <dt>
    <p><strong>-i</strong><em></em><strong>FLOAT</strong></p>
  </dt>
  <dd>
    <p>Ratio of INDEL-to-SNP mutation rate [0.15]</p>
  </dd>
  <dt>
    <p><strong>-m</strong><em></em><strong>FLOAT</strong></p>
  </dt>
  <dd>
    <p>New model for improved multiallelic and rare-variant calling. Another ALT allele is accepted if P(chi^2) of LRT exceeds the FLOAT threshold. The parameter seems robust and the actual value usually does not affect the results much; a good value to use is 0.99. This is the recommended calling method. [0]</p>
  </dd>
  <dt>
    <p><strong>-p</strong><em></em><strong>FLOAT</strong></p>
  </dt>
  <dd>
    <p>A site is considered to be a variant if P(ref|D)&lt;FLOAT [0.5]</p>
  </dd>
  <dt>
    <p><strong>-P</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Prior or initial allele frequency spectrum. If STR can be <em>full</em>, <em>cond2</em>, <em>flat</em> or the file consisting of error output from a previous variant calling run.</p>
  </dd>
  <dt>
    <p><strong>-t</strong><em></em><strong>FLOAT</strong></p>
  </dt>
  <dd>
    <p>Scaled muttion rate for variant calling [0.001]</p>
  </dd>
  <dt>
    <p><strong>-T</strong><em></em><strong>STR</strong></p>
  </dt>
  <dd>
    <p>Enable pair/trio calling. For trio calling, option <strong>-s</strong> is usually needed to be applied to configure the trio members and their ordering. In the file supplied to the option <strong>-s</strong>, the first sample must be the child, the second the father and the third the mother. The valid values of <em>STR</em> are `pair', `trioauto', `trioxd' and `trioxs', where `pair' calls differences between two input samples, and `trioxd' (`trioxs') specifies that the input is from the X chromosome non-PAR regions and the child is a female (male). [null]</p>
  </dd>
  <dt>
    <p><strong>-v</strong></p>
  </dt>
  <dd>
    <p>Output variant sites only (force -c)</p>
  </dd>
  <dt>
    <p><strong>Contrast Calling and Association Test Options:</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>-1</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Number of group-1 samples. This option is used for dividing the samples into two groups for contrast SNP calling or association test. When this option is in use, the following VCF INFO will be outputted: PC2, PCHI2 and QCHI2. [0]</p>
  </dd>
  <dt>
    <p><strong>-U</strong><em></em><strong>INT</strong></p>
  </dt>
  <dd>
    <p>Number of permutations for association test (effective only with <strong>-1</strong>) [0]</p>
  </dd>
  <dt>
    <p><strong>-X</strong><em></em><strong>FLOAT</strong></p>
  </dt>
  <dd>
    <p>Only perform permutations for P(chi^2)&lt;FLOAT (effective only with <strong>-U</strong>) [0.01]</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>index</strong></p>
  </dt>
  <dd>
    <p><strong>bcftools index</strong> <em>in.bcf</em></p><p>Index sorted BCF for random access.</p>
  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <p><strong>cat</strong></p>
  </dt>
  <dd>
    <p><strong>bcftools cat</strong> <em>in1.bcf</em> [<em>in2.bcf </em>[<em>...</em>]]]</p><p>Concatenate BCF files. The input files are required to be sorted and have identical samples appearing in the same order.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SAM FORMAT</h2>
        <div class="sectioncontent">
<p>Sequence Alignment/Map (SAM) format is TAB-delimited. Apart from the header lines, which are started with the `@' symbol, each alignment line consists of:</p><table class="table table-striped">
<tr>
<th>Col</th>
<th>Field</th>
<th>Description</th>
<th></th>
<th></th>
<th></th></tr>

<tr>
<td>1</td>
<td>QNAME</td>
<td>Query template/pair NAME</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>2</td>
<td>FLAG</td>
<td>bitwise FLAG</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>3</td>
<td>RNAME</td>
<td>Reference sequence NAME</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>4</td>
<td>POS</td>
<td>1-based leftmost POSition/coordinate of clipped sequence</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>5</td>
<td>MAPQ</td>
<td>MAPping Quality (Phred-scaled)</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>6</td>
<td>CIAGR</td>
<td>extended CIGAR string</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>7</td>
<td>MRNM</td>
<td>Mate Reference sequence NaMe (`=' if same as RNAME)</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>8</td>
<td>MPOS</td>
<td>1-based Mate POSistion</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>9</td>
<td>TLEN</td>
<td>inferred Template LENgth (insert size)</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>10</td>
<td>SEQ</td>
<td>query SEQuence on the same strand as the reference</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>11</td>
<td>QUAL</td>
<td>query QUALity (ASCII-33 gives the Phred base quality)</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>12+</td>
<td>OPT</td>
<td>variable OPTional fields in the format TAG:VTYPE:VALUE</td>
<td></td>
<td></td>
<td></td></tr>
</table><p>Each bit in the FLAG field is defined as:</p><table class="table table-striped">
<tr>
<th>Flag</th>
<th>Chr</th>
<th>Description</th>
<th></th>
<th></th>
<th></th></tr>

<tr>
<td>0x0001</td>
<td>p</td>
<td>the read is paired in sequencing</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0002</td>
<td>P</td>
<td>the read is mapped in a proper pair</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0004</td>
<td>u</td>
<td>the query sequence itself is unmapped</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0008</td>
<td>U</td>
<td>the mate is unmapped</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0010</td>
<td>r</td>
<td>strand of the query (1 for reverse)</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0020</td>
<td>R</td>
<td>strand of the mate</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0040</td>
<td>1</td>
<td>the read is the first read in a pair</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0080</td>
<td>2</td>
<td>the read is the second read in a pair</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0100</td>
<td>s</td>
<td>the alignment is not primary</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0200</td>
<td>f</td>
<td>the read fails platform/vendor quality checks</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>0x0400</td>
<td>d</td>
<td>the read is either a PCR or an optical duplicate</td>
<td></td>
<td></td>
<td></td></tr>
</table><p>where the second column gives the string representation of the FLAG field.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">VCF FORMAT</h2>
        <div class="sectioncontent">
<p>The Variant Call Format (VCF) is a TAB-delimited format with each data line consists of the following fields:</p><table class="table table-striped">
<tr>
<th>Col</th>
<th>Field</th>
<th>Description</th>
<th></th>
<th></th>
<th></th></tr>

<tr>
<td>1</td>
<td>CHROM</td>
<td>CHROMosome name</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>2</td>
<td>POS</td>
<td>the left-most POSition of the variant</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>3</td>
<td>ID</td>
<td>unique variant IDentifier</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>4</td>
<td>REF</td>
<td>the REFerence allele</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>5</td>
<td>ALT</td>
<td>the ALTernate allele(s), separated by comma</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>6</td>
<td>QUAL</td>
<td>variant/reference QUALity</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>7</td>
<td>FILTER</td>
<td>FILTers applied</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>8</td>
<td>INFO</td>
<td>INFOrmation related to the variant, separated by semi-colon</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>9</td>
<td>FORMAT</td>
<td>FORMAT of the genotype fields, separated by colon (optional)</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>10+</td>
<td>SAMPLE</td>
<td>SAMPLE genotypes and per-sample information (optional)</td>
<td></td>
<td></td>
<td></td></tr>
</table><p>The following table gives the <strong>INFO</strong> tags used by samtools and bcftools.</p><table class="table table-striped">
<tr>
<th>Tag</th>
<th>Format</th>
<th>Description</th>
<th></th>
<th></th>
<th></th></tr>

<tr>
<td>AF1</td>
<td>double</td>
<td>Max-likelihood estimate of the site allele frequency (AF) of the first ALT allele</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>DP</td>
<td>int</td>
<td>Raw read depth (without quality filtering)</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>DP4</td>
<td>int[4]</td>
<td># high-quality reference forward bases, ref reverse, alternate for and alt rev bases</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>FQ</td>
<td>int</td>
<td>Consensus quality. Positive: sample genotypes different; negative: otherwise</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>MQ</td>
<td>int</td>
<td>Root-Mean-Square mapping quality of covering reads</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>PC2</td>
<td>int[2]</td>
<td>Phred probability of AF in group1 samples being larger (,smaller) than in group2</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>PCHI2</td>
<td>double</td>
<td>Posterior weighted chi^2 P-value between group1 and group2 samples</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>PV4</td>
<td>double[4]</td>
<td>P-value for strand bias, baseQ bias, mapQ bias and tail distance bias</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>QCHI2</td>
<td>int</td>
<td>Phred-scaled PCHI2</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>RP</td>
<td>int</td>
<td># permutations yielding a smaller PCHI2</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>CLR</td>
<td>int</td>
<td>Phred log ratio of genotype likelihoods with and without the trio/pair constraint</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>UGT</td>
<td>string</td>
<td>Most probable genotype configuration without the trio constraint</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>CGT</td>
<td>string</td>
<td>Most probable configuration with the trio constraint</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>VDB</td>
<td>float</td>
<td>Tests variant positions within reads. Intended for filtering RNA-seq artifacts around splice sites</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>RPB</td>
<td>float</td>
<td>Mann-Whitney rank-sum test for tail distance bias</td>
<td></td>
<td></td>
<td></td></tr>

<tr>
<td>HWE</td>
<td>float</td>
<td>Hardy-Weinberg equilibrium test, Wigginton et al., PMID: 15789306</td>
<td></td>
<td></td>
<td></td></tr>
</table>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">
<ul>
<li><p>Import SAM to BAM when <strong>@SQ</strong> lines are present in the header:</p>
<pre>
  samtools view -bS aln.sam &gt; aln.bam
</pre>
<p>If <strong>@SQ</strong> lines are absent:</p>
<pre>
  samtools faidx ref.fa
  samtools view -bt ref.fa.fai aln.sam &gt; aln.bam
</pre>
<p>where <em>ref.fa.fai</em> is generated automatically by the <strong>faidx</strong> command.</p></li><li><p>Attach the <strong>RG</strong> tag while merging sorted alignments:</p>
<pre>
  perl -e 'print "@RG&#92;tID:ga&#92;tSM:hs&#92;tLB:ga&#92;tPL:Illumina&#92;n@RG&#92;tID:454&#92;tSM:hs&#92;tLB:454&#92;tPL:454&#92;n"' &gt; rg.txt
  samtools merge -rh rg.txt merged.bam ga.bam 454.bam
</pre>
<p>The value in a <strong>RG</strong> tag is determined by the file name the read is coming from. In this example, in the <em>merged.bam</em>, reads from <em>ga.bam</em> will be attached <em>RG:Z:ga</em>, while reads from <em>454.bam</em> will be attached <em>RG:Z:454</em>.</p></li><li><p>Call SNPs and short INDELs for one diploid individual:</p>
<pre>
  samtools mpileup -ugf ref.fa aln.bam | bcftools view -bvcg - &gt; var.raw.bcf
  bcftools view var.raw.bcf | vcfutils.pl varFilter -D 100 &gt; var.flt.vcf
</pre>
<p>The <strong>-D</strong> option of varFilter controls the maximum read depth, which should be adjusted to about twice the average read depth.  One may consider to add <strong>-C50</strong> to <strong>mpileup</strong> if mapping quality is overestimated for reads containing excessive mismatches. Applying this option usually helps <strong>BWA-short</strong> but may not other mappers.</p></li><li><p>Generate the consensus sequence for one diploid individual:</p>
<pre>
  samtools mpileup -uf ref.fa aln.bam | bcftools view -cg - | vcfutils.pl vcf2fq &gt; cns.fq
</pre>
</li><li><p>Call somatic mutations from a pair of samples:</p>
<pre>
  samtools mpileup -DSuf ref.fa aln.bam | bcftools view -bvcgT pair - &gt; var.bcf
</pre>
<p>In the output INFO field, <em>CLR</em> gives the Phred-log ratio between the likelihood by treating the two samples independently, and the likelihood by requiring the genotype to be identical. This <em>CLR</em> is effectively a score measuring the confidence of somatic calls. The higher the better.</p></li><li><p>Call de novo and somatic mutations from a family trio:</p>
<pre>
  samtools mpileup -DSuf ref.fa aln.bam | bcftools view -bvcgT pair -s samples.txt - &gt; var.bcf
</pre>
<p>File <em>samples.txt</em> should consist of three lines specifying the member and order of samples (in the order of child-father-mother). Similarly, <em>CLR</em> gives the Phred-log likelihood ratio with and without the trio constraint. <em>UGT</em> shows the most likely genotype configuration without the trio constraint, and <em>CGT</em> gives the most likely genotype configuration satisfying the trio constraint.</p></li><li><p>Phase one individual:</p>
<pre>
  samtools calmd -AEur aln.bam ref.fa | samtools phase -b prefix - &gt; phase.out
</pre>
<p>The <strong>calmd</strong> command is used to reduce false heterozygotes around INDELs.</p></li><li><p>Call SNPs and short indels for multiple diploid individuals:</p>
<pre>
  samtools mpileup -P ILLUMINA -ugf ref.fa *.bam | bcftools view -bcvg - &gt; var.raw.bcf
  bcftools view var.raw.bcf | vcfutils.pl varFilter -D 2000 &gt; var.flt.vcf
</pre>
<p>Individuals are identified from the <strong>SM</strong> tags in the <strong>@RG</strong> header lines. Individuals can be pooled in one alignment file; one individual can also be separated into multiple files. The <strong>-P</strong> option specifies that indel candidates should be collected only from read groups with the <strong>@RG-PL</strong> tag set to <em>ILLUMINA</em>. Collecting indel candidates from reads sequenced by an indel-prone technology may affect the performance of indel calling.</p><p>Note that there is a new calling model which can be invoked by</p>
<pre>
    bcftools view -m0.99  ...
</pre>
<p>which fixes some severe limitations of the default method.</p><p>For filtering, best results seem to be achieved by first applying the <em>SnpGap</em> filter and then applying some machine learning approach</p>
<pre>
    vcf-annotate -f SnpGap=n
    vcf filter ...
</pre>
<p>Both can be found in the <strong>vcftools</strong> and <strong>htslib</strong> package (links below).</p></li><li><p>Derive the allele frequency spectrum (AFS) on a list of sites from multiple individuals:</p>
<pre>
  samtools mpileup -Igf ref.fa *.bam &gt; all.bcf
  bcftools view -bl sites.list all.bcf &gt; sites.bcf
  bcftools view -cGP cond2 sites.bcf &gt; /dev/null 2&gt; sites.1.afs
  bcftools view -cGP sites.1.afs sites.bcf &gt; /dev/null 2&gt; sites.2.afs
  bcftools view -cGP sites.2.afs sites.bcf &gt; /dev/null 2&gt; sites.3.afs
  ......
</pre>
<p>where <em>sites.list</em> contains the list of sites with each line consisting of the reference sequence name and position. The following <strong>bcftools</strong> commands estimate AFS by EM.</p></li><li><p>Dump BAQ applied alignment for other SNP callers:</p>
<pre>
  samtools calmd -bAr aln.bam &gt; aln.baq.bam
</pre>
<p>It adds and corrects the <strong>NM</strong> and <strong>MD</strong> tags at the same time. The <strong>calmd</strong> command also comes with the <strong>-C</strong> option, the same as the one in <strong>pileup</strong> and <strong>mpileup</strong>. Apply if it helps.</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">LIMITATIONS</h2>
        <div class="sectioncontent">
<ul>
<li><p>Unaligned words used in bam_import.c, bam_endian.h, bam.c and bam_aux.c.</p></li><li><p>Samtools paired-end rmdup does not work for unpaired reads (e.g. orphan reads or ends mapped to different chromosomes). If this is a concern, please use Picard's MarkDuplicate which correctly handles these cases, although a little slower.</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Heng Li from the Sanger Institute wrote the C version of samtools. Bob Handsaker from the Broad Institute implemented the BGZF library and Jue Ruan from Beijing Genomics Institute wrote the RAZF library. John Marshall and Petr Danecek contribute to the source code and various people from the 1000 Genomes Project have contributed to the SAM format specification.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO bcftools&hellip;</h2>
        <div class="sectioncontent">
<p>Samtools website: &lt;http://samtools.sourceforge.net&gt;</p><p>Samtools latest source: &lt;https://github.com/samtools/samtools&gt;</p><p>VCFtools website with stable link to VCF specification: &lt;http://vcftools.sourceforge.net&gt;</p><p>HTSlib website: &lt;https://github.com/samtools/htslib&gt;</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="bcfg2.1.html"><span aria-hidden="true">&larr;</span> bcfg2.1: Bcfg2 client tool</a></li>
   <li class="next"><a href="bchunk.1.html">bchunk.1: Cd image format conversion from bin/cue to iso/cdr <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
