<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>clustalw: Multiple alignment of nucleic acid and protein sequences</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Multiple alignment of nucleic acid and protein sequences">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="clustalw (1) manual">
  <meta name="twitter:description" content="Multiple alignment of nucleic acid and protein sequences">
  <meta name="twitter:image" content="https://www.carta.tech/images/clustalw-clustalw-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/clustalw.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="clustalw (1) manual" />
  <meta property="og:description" content="Multiple alignment of nucleic acid and protein sequences" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/clustalw-clustalw-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">clustalw<small> (1)</small></h1>
        <p class="lead">Multiple alignment of nucleic acid and protein sequences</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/clustalw.1.html">
      <span itemprop="name">clustalw: Multiple alignment of nucleic acid and protein sequences</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/clustalw/">
      <span itemprop="name">clustalw</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/clustalw.1.html">
      <span itemprop="name">clustalw: Multiple alignment of nucleic acid and protein sequences</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>clustalw</strong> [<strong>-infile</strong>] <em>file.ext</em> [<strong>OPTIONS</strong>] <strong>clustalw</strong> [<strong>-help</strong> | <strong>-fullhelp</strong>]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>Clustal&nbsp;W is a general purpose multiple alignment program for DNA or proteins.</p><p>The program performs simultaneous alignment of many nucleotide or amino acid sequences. It is typically run interactively, providing a menu and an online help. If you prefer to use it in command-line (batch) mode, you will have to give several options, the minimum being <strong>-infile</strong>.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<h3>DATA (sequences)</h3>
<p><strong> -infile=</strong><strong></strong><em>file.ext</em><strong> </strong></p><p>Input sequences.</p><p><strong> -profile1=</strong><strong></strong><em>file.ext</em><strong> and -profile2=</strong><strong></strong><em>file.ext</em><strong> </strong></p><p>Profiles (old alignment)</p>
<h3>VERBS (do things)</h3>
<p><strong>-options</strong></p><p>List the command line parameters.</p><p><strong>-help or -check</strong></p><p>Outline the command line params.</p><p><strong>-fullhelp</strong></p><p>Output full help content.</p><p><strong>-align</strong></p><p>Do full multiple alignment.</p><p><strong>-tree</strong></p><p>Calculate NJ tree.</p><p><strong>-pim</strong></p><p>Output percent identity matrix (while calculating the tree).</p><p><strong> -bootstrap</strong><strong></strong><em>=n</em><strong> </strong></p><p>Bootstrap a NJ tree (<em>n</em>= number of bootstraps; def. = 1000).</p><p><strong>-convert</strong></p><p>Output the input sequences in a different file format.</p>
<h3>PARAMETERS (set things)</h3>
<p><strong>General settings:</strong></p><p><strong>-interactive</strong></p><p>Read command line, then enter normal interactive menus.</p><p><strong>-quicktree</strong></p><p>Use FAST algorithm for the alignment guide tree.</p><p><strong>-type=</strong></p><p><em>PROTEIN</em> or <em>DNA</em> sequences.</p><p><strong>-negative</strong></p><p>Protein alignment with negative values in matrix.</p><p><strong>-outfile=</strong></p><p>Sequence alignment file name.</p><p><strong>-output=</strong></p><p><em>GCG</em>, <em>GDE</em>, <em>PHYLIP</em>, <em>PIR</em> or <em>NEXUS</em>.</p><p><strong>-outputorder=</strong></p><p><em>INPUT</em> or <em>ALIGNED</em></p><p><strong>-case</strong></p><p><em>LOWER</em> or <em>UPPER</em> (for GDE output only).</p><p><strong>-seqnos=</strong></p><p><em>OFF</em> or <em>ON</em> (for Clustal output only).</p><p><strong>-seqnos_range=</strong></p><p><em>OFF</em> or <em>ON</em> (NEW: for all output formats).</p><p><strong>-range=</strong><strong></strong><em>m</em><strong>,</strong><strong></strong><em>n</em></p><p>Sequence range to write starting <em>m</em> to <em>m</em>+<em>n</em>.</p><p><strong>-maxseqlen=</strong><strong></strong><em>n</em></p><p>Maximum allowed input sequence length.</p><p><strong>-quiet</strong></p><p>Reduce console output to minimum.</p><p><strong>-stats=</strong><strong></strong><em>file</em></p><p>Log some alignments statistics to <em>file</em>.</p><p><strong>Fast Pairwise Alignments:</strong></p><p><strong>-ktuple=</strong><strong></strong><em>n</em></p><p>Word size.</p><p><strong>-topdiags=</strong><strong></strong><em>n</em></p><p>Number of best diags.</p><p><strong>-window=</strong><strong></strong><em>n</em></p><p>Window around best diags.</p><p><strong>-pairgap=</strong><strong></strong><em>n</em></p><p>Gap penalty.</p><p><strong>-score</strong></p><p><em>PERCENT</em> or <em>ABSOLUTE</em>.</p><p><strong>Slow Pairwise Alignments:</strong></p><p><strong>-pwmatrix=</strong></p><p>:Protein weight matrix=<em>BLOSUM</em>, <em>PAM</em>, <em>GONNET</em>, <em>ID</em> or <em>filename</em></p><p><strong>-pwdnamatrix=</strong></p><p>DNA weight matrix=<em>BLOSUM</em>IUB, <em>BLOSUM</em>CLUSTALW or <em>BLOSUM</em>filename.</p><p><strong>-pwgapopen=</strong><strong></strong><em>f</em></p><p>Gap opening penalty.</p><p><strong>-pwgapext=</strong><strong></strong><em>f</em></p><p>Gap extension penalty.</p><p><strong>Multiple Alignments:</strong></p><p><strong>-newtree=</strong></p><p>File for new guide tree.</p><p><strong>-usetree=</strong></p><p>File for old guide tree.</p><p><strong>-matrix=</strong></p><p>Protein weight matrix=<em>BLOSUM</em>, <em>PAM</em>, <em>GONNET</em>, <em>ID</em> or <em>filename</em>.</p><p><strong>-dnamatrix=</strong></p><p>DNA weight matrix=<em>IUB</em>, <em>CLUSTALW</em> or <em>filename</em>.</p><p><strong>-gapopen=</strong><strong></strong><em>f</em></p><p>Gap opening penalty.</p><p><strong>-gapext=</strong><strong></strong><em>f</em></p><p>Gap extension penalty.</p><p><strong>-engaps</strong></p><p>No end gap separation pen.</p><p><strong>-gapdist=</strong><strong></strong><em>n</em></p><p>Gap separation pen. range.</p><p><strong>-nogap</strong></p><p>Residue-specific gaps off.</p><p><strong>-nohgap</strong></p><p>Hydrophilic gaps off.</p><p><strong>-hgapresidues=</strong></p><p>List hydrophilic res.</p><p><strong>-maxdiv=</strong><strong></strong><em>n</em></p><p>Percent identity for delay.</p><p><strong>-type=</strong></p><p><em>PROTEIN</em> or <em>DNA</em></p><p><strong>-transweight=</strong><strong></strong><em>f</em></p><p>Transitions weighting.</p><p><strong>-iteration=</strong></p><p><em>NONE</em> or <em>TREE</em> or <em>ALIGNMENT</em>.</p><p><strong>-numiter=</strong><strong></strong><em>n</em></p><p>Maximum number of iterations to perform.</p><p><strong>Profile Alignments:</strong></p><p><strong>-profile</strong></p><p>Merge two alignments by profile alignment.</p><p><strong>-newtree1=</strong></p><p>File for new guide tree for profile1.</p><p><strong>-newtree2=</strong></p><p>File for new guide tree for profile2.</p><p><strong>-usetree1=</strong></p><p>File for old guide tree for profile1.</p><p><strong>-usetree2=</strong></p><p>File for old guide tree for profile2.</p><p><strong>Sequence to Profile Alignments:</strong></p><p><strong>-sequences</strong></p><p>Sequentially add profile2 sequences to profile1 alignment.</p><p><strong>-newtree=</strong></p><p>File for new guide tree.</p><p><strong>-usetree=</strong></p><p>File for old guide tree.</p><p><strong>Structure Alignments:</strong></p><p><strong>-nosecstr1</strong></p><p>Do not use secondary structure-gap penalty mask for profile 1.</p><p><strong>-nosecstr2</strong></p><p>Do not use secondary structure-gap penalty mask for profile 2.</p><p><strong>-secstrout=</strong><strong></strong><em>STRUCTURE</em><strong> or </strong><strong></strong><em>MASK</em><strong> or </strong><strong></strong><em>BOTH</em><strong> or </strong><strong></strong><em>NONE</em></p><p>Output in alignment file.</p><p><strong>-helixgap=</strong><strong></strong><em>n</em></p><p>Gap penalty for helix core residues.</p><p><strong>-strandgap=</strong><strong></strong><em>n</em></p><p>Gap penalty for strand core residues.</p><p><strong>loopgap=</strong><strong></strong><em>n</em></p><p>Gap penalty for loop regions.</p><p><strong>-terminalgap=</strong><strong></strong><em>n</em></p><p>Gap penalty for structure termini.</p><p><strong>-helixendin=</strong><strong></strong><em>n</em></p><p>Number of residues inside helix to be treated as terminal.</p><p><strong>-helixendout=</strong><strong></strong><em>n</em></p><p>Number of residues outside helix to be treated as terminal.</p><p><strong>-strandendin=</strong><strong></strong><em>n</em></p><p>Number of residues inside strand to be treated as terminal.</p><p><strong>-strandendout=</strong><strong></strong><em>n</em></p><p>Number of residues outside strand to be treated as terminal.</p><p><strong>Trees:</strong></p><p><strong>-outputtree=</strong><strong></strong><em>nj</em><strong> OR </strong><strong></strong><em>phylip</em><strong> OR </strong><strong></strong><em>dist</em><strong> OR </strong><strong></strong><em>nexus</em></p><p><strong>-seed=</strong><strong></strong><em>n</em></p><p>Seed number for bootstraps.</p><p><strong>-kimura</strong></p><p>Use Kimura&apos;s correction.</p><p><strong>-tossgaps</strong></p><p>Ignore positions with gaps.</p><p><strong>-bootlabels=</strong><strong></strong><em>node</em></p><p>Position of bootstrap values in tree display.</p><p><strong>-clustering=</strong></p><p>NJ or UPGMA.</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>The Clustal bug tracking system can be found at \m[blue]<strong>http://bioinf.ucd.ie/bugzilla/buglist.cgi?quicksearch=clustal</strong>\m[].</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO clustalw&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man1/clustalx.1.html"><strong>clustalx</strong>(1)</a>.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REFERENCES</h2>
        <div class="sectioncontent">
<p>Larkin MA, Blackshields G, Brown NP, Chenna R, McGettigan PA, McWilliam H, Valentin F, Wallace IM, Wilm A, Lopez R, Thompson JD, Gibson TJ, Higgins DG. (2007). \m[blue]<strong>Clustal W and Clustal X version 2.0.</strong>\m[]\s-2\u[1]\d\s+2 Bioinformatics, 23, 2947-2948.</p><p>Chenna R, Sugawara H, Koike T, Lopez R, Gibson TJ, Higgins DG, Thompson JD. (2003). \m[blue]<strong>Multiple sequence alignment with the Clustal series of programs.</strong>\m[]\s-2\u[2]\d\s+2 Nucleic Acids Res., 31, 3497-3500.</p><p>Jeanmougin F, Thompson JD, Gouy M, Higgins DG, Gibson TJ. (1998). \m[blue]<strong>Multiple sequence alignment with Clustal X</strong>\m[]\s-2\u[3]\d\s+2. Trends Biochem Sci., 23, 403-405.</p><p>Thompson JD, Gibson TJ, Plewniak F, Jeanmougin F, Higgins DG. (1997). \m[blue]<strong>The CLUSTAL_X windows interface: flexible strategies for multiple sequence alignment aided by quality analysis tools.</strong>\m[]\s-2\u[4]\d\s+2 Nucleic Acids Res., 25, 4876-4882.</p><p>Higgins DG, Thompson JD, Gibson TJ. (1996). \m[blue]<strong>Using CLUSTAL for multiple sequence alignments.</strong>\m[]\s-2\u[5]\d\s+2 Methods Enzymol., 266, 383-402.</p><p>Thompson JD, Higgins DG, Gibson TJ. (1994). \m[blue]<strong>CLUSTAL W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap penalties and weight matrix choice.</strong>\m[]\s-2\u[6]\d\s+2 Nucleic Acids Res., 22, 4673-4680.</p><p>Higgins DG. (1994). \m[blue]<strong>CLUSTAL V: multiple alignment of DNA and protein sequences.</strong>\m[]\s-2\u[7]\d\s+2 Methods Mol Biol., 25, 307-318</p><p>Higgins DG, Bleasby AJ, Fuchs R. (1992). \m[blue]<strong>CLUSTAL V: improved software for multiple sequence alignment.</strong>\m[]\s-2\u[8]\d\s+2 Comput. Appl. Biosci., 8, 189-191.</p><p>Higgins,D.G. and Sharp,P.M. (1989). \m[blue]<strong>Fast and sensitive multiple sequence alignments on a microcomputer.</strong>\m[]\s-2\u[9]\d\s+2 Comput. Appl. Biosci., 5, 151-153.</p><p>Higgins,D.G. and Sharp,P.M. (1988). \m[blue]<strong>CLUSTAL: a package for performing multiple sequence alignment on a microcomputer.</strong>\m[]\s-2\u[10]\d\s+2 Gene, 73, 237-244.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p><strong>Des Higgins</strong></p><p>Copyright holder for Clustal.</p><p><strong>Julie Thompson</strong></p><p>Copyright holder for Clustal.</p><p><strong>Toby Gibson</strong></p><p>Copyright holder for Clustal.</p><p><strong>Charles Plessy</strong> &lt;plessy@debian.org&gt;</p><p>Prepared this manpage in DocBook XML for the Debian distribution.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">
<p>Copyright &copy; 1988&ndash;2010 Des Higgins, Julie Thompson & Toby Giboson (Clustal)</p><p>Copyright &copy; 2008&ndash;2010 Charles Plessy (This manpage)</p><p>This program is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.</p><p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.</p><p>You should have received a copy of the GNU Lesser General Public License along with this program. If not, see http://www.gnu.org/licenses/, or on Debian systems, /usr/share/common-licenses/LGPL-3.</p><p>This manual page and its XML source can be used, modified, and redistributed as if it were in public domain.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">NOTES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
     1.
  </dt>
  <dd>
    <p>Clustal W and Clustal X version 2.0.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/17846036</p>
  </dd>
  <dt>
     2.
  </dt>
  <dd>
    <p>Multiple sequence alignment with the Clustal series of programs.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/12824352</p>
  </dd>
  <dt>
     3.
  </dt>
  <dd>
    <p>Multiple sequence alignment with Clustal X</p><p>http://www.ncbi.nlm.nih.gov/pubmed/9810230</p>
  </dd>
  <dt>
     4.
  </dt>
  <dd>
    <p>The CLUSTAL_X windows interface: flexible strategies for multiple sequence alignment aided by quality analysis tools.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/9396791</p>
  </dd>
  <dt>
     5.
  </dt>
  <dd>
    <p>Using CLUSTAL for multiple sequence alignments.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/8743695</p>
  </dd>
  <dt>
     6.
  </dt>
  <dd>
    <p>CLUSTAL W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap penalties and weight matrix choice.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/7984417</p>
  </dd>
  <dt>
     7.
  </dt>
  <dd>
    <p>CLUSTAL V: multiple alignment of DNA and protein sequences.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/8004173</p>
  </dd>
  <dt>
     8.
  </dt>
  <dd>
    <p>CLUSTAL V: improved software for multiple sequence alignment.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/1591615</p>
  </dd>
  <dt>
     9.
  </dt>
  <dd>
    <p>Fast and sensitive multiple sequence alignments on a microcomputer.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/2720464</p>
  </dd>
  <dt>
    10.
  </dt>
  <dd>
    <p>CLUSTAL: a package for performing multiple sequence alignment on a microcomputer.</p><p>http://www.ncbi.nlm.nih.gov/pubmed/3243435</p>
  </dd>

</dl>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="clustalo.1.html"><span aria-hidden="true">&larr;</span> clustalo.1: General purpose multiple sequence alignment program for proteins</a></li>
   <li class="next"><a href="clustalx.1.html">clustalx.1: A multiple sequence alignment program <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
