<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>cmalign: Align sequences to a covariance model</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Align sequences to a covariance model">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="cmalign (1) manual">
  <meta name="twitter:description" content="Align sequences to a covariance model">
  <meta name="twitter:image" content="https://www.carta.tech/images/infernal-cmalign-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/cmalign.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="cmalign (1) manual" />
  <meta property="og:description" content="Align sequences to a covariance model" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/infernal-cmalign-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">cmalign<small> (1)</small></h1>
        <p class="lead">Align sequences to a covariance model</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmalign.1.html">
      <span itemprop="name">cmalign: Align sequences to a covariance model</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/infernal/">
      <span itemprop="name">infernal</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmalign.1.html">
      <span itemprop="name">cmalign: Align sequences to a covariance model</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>cmalign</strong></p>
  </dt>
  <dd>
    <p><em>[options]</em> <em>&lt;cmfile&gt;</em> <em>&lt;seqfile&gt;</em></p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>cmalign</strong> aligns the RNA sequences in <em>&lt;seqfile&gt;</em> to the covariance model (CM) in <em>&lt;cmfile&gt;.</em> The new alignment is output to <em>stdout</em> in Stockholm format, but can be redirected to a file <em>&lt;f&gt;</em> with the <strong>-o</strong><em> &lt;f&gt;</em> option.</p><p>Either <em>&lt;cmfile&gt;</em> or <em>&lt;seqfile&gt;</em> (but not both) may be '-' (dash), which means reading this input from <em>stdin</em> rather than a file.</p><p>The sequence file <em>&lt;seqfile&gt;</em> must be in FASTA or Genbank format.</p><p><strong>cmalign</strong> uses an HMM banding technique to accelerate alignment by default as described below for the <strong>--hbanded</strong> option. HMM banding can be turned off with the <strong>--nonbanded</strong> option.</p><p>By default, <strong>cmalign</strong> computes the alignment with maximum expected accuracy that is consistent with constraints (bands) derived from an HMM, using a banded version of the Durbin/Holmes optimal accuracy algorithm. This behavior can be changed with the <strong>--cyk</strong> or <strong>--sample</strong> options.</p><p><strong>cmalign</strong> takes special care to correctly align truncated sequences, where some nucleotides from the beginning (5') and/or end (3') of the actual full length biological sequence are not present in the input sequence (see DL Kolbe and SR Eddy, Bioinformatics, 25:1236-1243, 2009). This behavior is on by default, but can be turned off with <strong>--notrunc.</strong> In previous versions of <strong>cmalign</strong> the <strong>--sub</strong> option was required to appropriately handle truncated sequences. The <strong>--sub</strong> option is still available in this version, but the new default method for handling truncated sequences should be as good or superior to the sub method in nearly all cases.</p><p>The <strong>--mapali</strong><em> &lt;s&gt;</em> option allows inclusion of the fixed training alignment used to build the CM from file <em>&lt;s&gt;</em> within the output alignment of <strong>cmalign.</strong></p><p>It is possible to merge two or more alignments created by the same CM using the Easel miniapp <strong>esl-alimerge</strong> (included in the easel/miniapps/ subdirectory of Infernal). Previous versions of <strong>cmalign</strong> included options to merge alignments but they were deprecated upon development of <strong>esl-alimerge,</strong> which is significantly more memory efficient.</p><p>By default, <strong>cmalign</strong> will output the alignment to stdout. The alignment can be redirected to an output file <em>&lt;f&gt;</em> with the <strong>-o</strong><em> &lt;f&gt;</em> option. With <strong>-o,</strong> information on each aligned sequence, including score and model alignment boundaries will be printed to stdout (more on this below).</p><p>The output alignment will be in Stockholm format by default. This can be changed to Pfam, aligned FASTA (AFA), A2M, Clustal, or Phylip format using the <strong>--outformat</strong><em> &lt;s&gt;</em> option, where <em>&lt;s&gt;</em> is the name of the desired format.  As a special case, if the output alignment is large (more than 10,000 sequences or more than 10,000,000 total nucleotides) than the output format will be Pfam format, with each sequence appearing on a single line, for reasons of memory efficiency. For alignments larger than this, using <strong>--ileaved</strong> will force interleaved Stockholm format, but the user should be aware that this may require a lot of memory. <strong>--ileaved</strong> will only work for alignments up to 100,000 sequences or 100,000,000 total nucleotides.</p><p>If the output alignment format is Stockholm or Pfam, the output alignment will be annotated with posterior probabilities which estimate the confidence level of each aligned nucleotide.  This annotation appears as lines beginning with "#=GR &lt;seq name&gt; PP", one per sequence, each immediately below the corresponding aligned sequence "&lt;seq name&gt;". Characters in PP lines have 12 possible values: "0-9", "*", or ".". If ".", the position corresponds to a gap in the sequence. A value of "0" indicates a posterior probability of between 0.0 and 0.05, "1" indicates between 0.05 and 0.15, "2" indicates between 0.15 and 0.25 and so on up to "9" which indicates between 0.85 and 0.95. A value of "*" indicates a posterior probability of between 0.95 and 1.0. Higher posterior probabilities correspond to greater confidence that the aligned nucleotide belongs where it appears in the alignment.  With <strong>--nonbanded,</strong> the calculation of the posterior probabilities considers all possible alignments of the target sequence to the CM. Without <strong>--nonbanded</strong> (i.e. in default mode), the calculation considers only possible alignments within the HMM bands. Further, the posterior probabilities are conditional on the truncation mode of the alignment. For example, if the sequence alignment is truncated 5', a PP value of "9" indicates between 0.85 and 0.95 of all 5' truncated alignments include the given nucleotide at the given position. The posterior annotation can be turned off with the <strong>--noprob</strong> option. If <strong>--small</strong> is enabled, posterior annotation must also be turned off using <strong>--noprob.</strong></p><p>The tabular output that is printed to stdout if the <strong>-o</strong> option is used includes one line per sequence and twelve fields per line: "idx": the index of the sequence in the input file, "seq name": the sequence name; "length": the length of the sequence; "cm from" and "cm to": the model start and end positions of the alignment; "trunc": "no" if the sequence is not truncated, "5'" if the beginning of the sequence truncated 5', "3'" if the end of the sequence is truncated, and "5'&3'" if both the beginning and the end are truncated; "bit sc": the bit score of the alignment, "avg pp" the average posterior probability of all aligned nucleotides in the alignment; "band calc", "alignment" and "total": the time in seconds required for calculating HMM bands, computing the alignment, and complete processing of the sequence, respectively; "mem (Mb)": the size in Mb of all dynamic programming matrices required for aligning the sequence.  This tabular data can be saved to file <em>&lt;f&gt;</em> with the <strong>--sfile</strong><em> &lt;f&gt;</em> option.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and available options.</p>
  </dd>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save the alignment in Stockholm format to a file <em>&lt;f&gt;.</em> The default is to write it to standard output.</p>
  </dd>
  <dt>
    <p><strong>-g</strong></p>
  </dt>
  <dd>
    <p>Configure the model for global alignment of the query model to the target sequences. By default, the model is configured for local alignment. Local alignments can contain large insertions and deletions called "local ends" in the structure to be penalized differently than normal indels. These are annotated as "~" columns in the RF line of the output alignment. The <strong>-g</strong> option can be used to disallow these local ends. The <strong>-g</strong> option is required if the <strong>--sub</strong> option is also used.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR CONTROLLING THE ALIGNMENT ALGORITHM</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--optacc</strong></p>
  </dt>
  <dd>
    <p>Align sequences using the Durbin/Holmes optimal accuracy algorithm. This is the default. The optimal accuracy alignment will be constrained by HMM bands for acceleration unless the <strong>--nonbanded</strong> option is enabled. The optimal accuracy algorithm determines the alignment that maximizes the posterior probabilities of the aligned nucleotides within it. The posterior probabilites are determined using (possibly HMM banded) variants of the Inside and Outside algorithms.</p>
  </dd>
  <dt>
    <p><strong>--cyk</strong></p>
  </dt>
  <dd>
    <p>Do not use the Durbin/Holmes optimal accuracy alignment to align the sequences, instead use the CYK algorithm which determines the optimally scoring (maximum likelihood) alignment of the sequence to the model, given the HMM bands (unless <strong>--nonbanded</strong> is also enabled).</p>
  </dd>
  <dt>
    <p><strong>--sample</strong></p>
  </dt>
  <dd>
    <p>Sample an alignment from the posterior distribution of alignments. The posterior distribution is determined using an HMM banded (unless <strong>--nonbanded)</strong> variant of the Inside algorithm.</p>
  </dd>
  <dt>
    <p><strong>--seed</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Seed the random number generator with <em>&lt;n&gt;,</em> an integer &gt;= 0. This option can only be used in combination with <strong>--sample.</strong> If <em>&lt;n&gt;</em> is nonzero, stochastic sampling of alignments will be reproducible; the same command will give the same results. If <em>&lt;n&gt;</em> is 0, the random number generator is seeded arbitrarily, and stochastic samplings may vary from run to run of the same command. The default seed is 181.</p>
  </dd>
  <dt>
    <p><strong>--notrunc</strong></p>
  </dt>
  <dd>
    <p>Turn off truncated alignment algorithms. All sequences in the input file will be assumed to be full length, unless <strong>--sub</strong> is also used, in which case the program can still handle truncated sequences but will use an alternative strategy for their alignment.</p>
  </dd>
  <dt>
    <p><strong>--sub</strong></p>
  </dt>
  <dd>
    <p>Turn on the sub model construction and alignment procedure. For each sequence, an HMM is first used to predict the model start and end consensus columns, and a new sub CM is constructed that only models consensus columns from start to end. The sequence is then aligned to this sub CM.  Sub alignment is an older method than the default one for aligning sequences that are possibly truncated. By default, <strong>cmalign</strong> uses special DP algorithms to handle truncated sequences which should be more accurate than the sub method in most cases. <strong>--sub</strong> is still included as an option mainly for testing against this default truncated sequence handling.  This "sub CM" procedure is not the same as the "sub CMs" described by Weinberg and Ruzzo.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR CONTROLLING SPEED AND MEMORY REQUIREMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--hbanded</strong></p>
  </dt>
  <dd>
    <p>This option is turned on by default. Accelerate alignment by pruning away regions of the CM DP matrix that are deemed negligible by an HMM. First, each sequence is scored with a CM plan 9 HMM derived from the CM using the Forward and Backward HMM algorithms to calculate posterior probabilities that each nucleotide aligns to each state of the HMM. These posterior probabilities are used to derive constraints (bands) on the CM DP matrix. Finally, the target sequence is aligned to the CM using the banded DP matrix, during which cells outside the bands are ignored. Usually most of the full DP matrix lies outside the bands (often more than 95%), making this technique faster because fewer DP calculations are required, and more memory efficient because only cells within the bands need be allocated.</p><p>Importantly, HMM banding sacrifices the guarantee of determining the optimally accurarte or optimal alignment, which will be missed if it lies outside the bands. The tau paramater is the amount of probability mass considered negligible during HMM band calculation; lower values of tau yield greater speedups but also a greater chance of missing the optimal alignment. The default tau is 1E-7, determined empirically as a good tradeoff between sensitivity and speed, though this value can be changed with the <strong>--tau  &lt;x&gt;</strong> option. The level of acceleration increases with both the length and primary sequence conservation level of the family. For example, with the default tau of 1E-7, tRNA models (low primary sequence conservation with length of about 75 nucleotides) show about 10X acceleration, and SSU bacterial rRNA models (high primary sequence conservation with length of about 1500 nucleotides) show about 700X. HMM banding can be turned off with the <strong>--nonbanded</strong> option.</p>
  </dd>
  <dt>
    <p><strong>--tau</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the tail loss probability used during HMM band calculation to <em>&lt;x&gt;.</em> This is the amount of probability mass within the HMM posterior probabilities that is considered negligible. The default value is 1E-7. In general, higher values will result in greater acceleration, but increase the chance of missing the optimal alignment due to the HMM bands.</p>
  </dd>
  <dt>
    <p><strong>--mxsize</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the maximum allowable total DP matrix size to <em>&lt;x&gt;</em> megabytes. By default this size is 1028 Mb. This should be large enough for the vast majority of alignments, however if it is not <strong>cmalign</strong> will attempt to iteratively tighten the HMM bands it uses to constrain the alignment by raising the tau parameter and recalculating the bands until the total matrix size needed falls below <em>&lt;x&gt;</em> megabytes or the maximum allowable tau value (0.05 by default, but changeable with <strong>--maxtau)</strong> is reached. At each iteration of band tightening, tau is multiplied by a 2.0. The band tightening strategy can be turned off with the <strong>--fixedtau</strong> option. If the maximum tau is reached and the required matrix size still exceeds <em>&lt;x&gt;</em> or if HMM banding is not being used and the required matrix size exceeds <em>&lt;x&gt;</em> then <strong>cmalign</strong> will exit prematurely and report an error message that the matrix exceeded its maximum allowable size. In this case, the <strong>--mxsize</strong> can be used to raise the size limit or the maximum tau can be raised with <strong>--maxtau.</strong> The limit will commonly be exceeded when the <strong>--nonbanded</strong> option is used without the <strong>--small</strong> option, but can still occur when <strong>--nonbanded</strong> is not used. Note that if <strong>cmalign</strong> is being run in <em>&lt;n&gt;</em> multiple threads on a multicore machine then each thread may have an allocated matrix of up to size <em>&lt;x&gt;</em> Mb at any given time.</p>
  </dd>
  <dt>
    <p><strong>--fixedtau</strong></p>
  </dt>
  <dd>
    <p>Turn off the HMM band tightening strategy described in the explanation of the <strong>--mxsize</strong> option above.</p>
  </dd>
  <dt>
    <p><strong>--maxtau</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the maximum allowed value for tau during band tightening, described in the explanation of <strong>--mxsize</strong> above, to <em>&lt;x&gt;.</em> By default this value is 0.05.</p>
  </dd>
  <dt>
    <p><strong>--nonbanded</strong></p>
  </dt>
  <dd>
    <p>Turns off HMM banding. The returned alignment is guaranteed to be the globally optimally accurate one (by default) or the globally optimally scoring one (if <strong>--cyk</strong> is enabled). The <strong>--small</strong> option is recommended in combination with this option, because standard alignment without HMM banding requires a lot of memory (see <strong>--small</strong> ).</p>
  </dd>
  <dt>
    <p><strong>--small</strong></p>
  </dt>
  <dd>
    <p>Use the divide and conquer CYK alignment algorithm described in SR Eddy, BMC Bioinformatics 3:18, 2002. The <strong>--nonbanded</strong> option must be used in combination with this options. Also, it is recommended whenever <strong>--nonbanded</strong> is used that <strong>--small</strong> is also used  because standard CM alignment without HMM banding requires a lot of memory, especially for large RNAs. <strong>--small</strong> allows CM alignment within practical memory limits, reducing the memory required for alignment LSU rRNA, the largest known RNAs, from 150 Gb to less than 300 Mb. This option can only be used in combination with <strong>--nonbanded,</strong> <strong>--notrunc,</strong> and <strong>--cyk.</strong></p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONAL OUTPUT FILES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--sfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Dump per-sequence alignment score and timig information to file <em>&lt;f&gt;.</em> The format of this file is described above (it's the same data in the same format as the tabular stdout output when the <strong>-o</strong> option is used).</p>
  </dd>
  <dt>
    <p><strong>--tfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Dump tabular sequence tracebacks for each individual sequence to a file <em>&lt;f&gt;.</em> Primarily useful for debugging.</p>
  </dd>
  <dt>
    <p><strong>--ifile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Dump per-sequence insert information to file <em>&lt;f&gt;.</em> The format of the file is described by "#"-prefixed comment lines included at the top of the file <em>&lt;f&gt;.</em> The insert information is valid even when the <strong>--matchonly</strong> option is used.</p>
  </dd>
  <dt>
    <p><strong>--elfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Dump per-sequence EL state (local end) insert information to file <em>&lt;f&gt;.</em> The format of the file is described by "#"-prefixed comment lines included at the top of the file <em>&lt;f&gt;.</em> The EL insert information is valid even when the <strong>--matchonly</strong> option is used.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--mapali</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Reads the alignment from file <em>&lt;f&gt;</em> used to build the model aligns it as a single object to the CM; e.g. the alignment in <em>&lt;f&gt;</em> is held fixed. This allows you to align sequences to a model with <strong>cmalign</strong> and view them in the context of an existing trusted multiple alignment. <em>&lt;f&gt;</em> must be the alignment file that the CM was built from. The program verifies that the checksum of the file matches that of the file used to construct the CM. A similar option to this one was called <strong>--withali</strong> in previous versions of <strong>cmalign.</strong></p>
  </dd>
  <dt>
    <p><strong>--mapstr</strong></p>
  </dt>
  <dd>
    <p>Must be used in combination with <strong>--mapali</strong><em> &lt;f&gt;</em><strong>.</strong> Propogate structural information for any pseudoknots that exist in <em>&lt;f&gt;</em> to the output alignment. A similar option to this one was called <strong>--withstr</strong> in previous versions of <strong>cmalign.</strong></p>
  </dd>
  <dt>
    <p><strong>--informat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Assert that the input <em>&lt;seqfile&gt;</em> is in format <em>&lt;s&gt;.</em> Do not run Babelfish format autodection. This increases the reliability of the program somewhat, because the Babelfish can make mistakes; particularly recommended for unattended, high-throughput runs of Infernal. Acceptable formats are: FASTA, GENBANK, and DDBJ. <em>&lt;s&gt;</em> is case-insensitive.</p>
  </dd>
  <dt>
    <p><strong>--outformat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Specify the output alignment format as <em>&lt;s&gt;.</em> Acceptable formats are: Pfam, AFA, A2M, Clustal, and Phylip. AFA is aligned fasta. Only Pfam and Stockholm alignment formats will include consensus structure annotation and posterior probability annotation of aligned residues.</p>
  </dd>
  <dt>
    <p><strong>--dnaout</strong></p>
  </dt>
  <dd>
    <p>Output the alignments as DNA sequence alignments, instead of RNA ones.</p>
  </dd>
  <dt>
    <p><strong>--noprob</strong></p>
  </dt>
  <dd>
    <p>Do not annotate the output alignment with posterior probabilities.</p>
  </dd>
  <dt>
    <p><strong>--matchonly</strong></p>
  </dt>
  <dd>
    <p>Only include match columns in the output alignment, do not include any insertions relative to the consensus model. This option may be useful when creating very large alignments that require a lot of memory and disk space, most of which is necessary only to deal with insert columns that are gaps in most sequences.</p>
  </dd>
  <dt>
    <p><strong>--ileaved</strong></p>
  </dt>
  <dd>
    <p>Output the alignment in interleaved Stockholm format of a fixed width that may be more convenient for examination. This was the default output alignment format of previous versions of <strong>cmalign.</strong> Note that <strong>cmalign</strong> requires more memory when this option is used. For this reason, <strong>--ileaved</strong> will only work for alignments of up to 100,000 sequences or a total of 100,000,000 aligned nucleotides.</p>
  </dd>
  <dt>
    <p><strong>--regress</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Save an additional copy of the output alignment with no author information to file <em>&lt;s&gt;.</em></p>
  </dd>
  <dt>
    <p><strong>--verbose</strong></p>
  </dt>
  <dd>
    <p>Output additional information in the tabular scores output (output to stdout if <strong>-o</strong> is used, or to <em>&lt;f&gt;</em> if <strong>--sfile</strong><em> &lt;f&gt;</em> is used). These are mainly useful for testing and debugging.</p>
  </dd>
  <dt>
    <p><strong>--cpu</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Specify that <em>&lt;n&gt;</em> parallel CPU workers be used. If <em>&lt;n&gt;</em> is set as "0", then the program will be run in serial mode, without using threads. You can also control this number by setting an environment variable, <em>INFERNAL_NCPU.</em> This option will only be available if the machine on which Infernal was built is capable of using POSIX threading (see the Installation section of the user guide for more information).</p>
  </dd>
  <dt>
    <p><strong>--mpi</strong></p>
  </dt>
  <dd>
    <p>Run as an MPI parallel program. This option will only be available if Infernal has been configured and built with the "--enable-mpi" flag (see the Installation section of the user guide for more information).</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO cmalign&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/infernal.1.html"><strong>infernal</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the Infernal package.</p><p>For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf); or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2014 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your Infernal source distribution, or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="cmake-gui.1.html"><span aria-hidden="true">&larr;</span> cmake-gui.1: Cmake gui command-line reference</a></li>
   <li class="next"><a href="cmatrix.1.html">cmatrix.1: Simulates the display from "the matrix" <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
