<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>cmbuild: Construct covariance model(s) from structurally annotated rna multiple sequence alignment(s)</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Construct covariance model(s) from structurally annotated rna multiple sequence alignment(s)">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="cmbuild (1) manual">
  <meta name="twitter:description" content="Construct covariance model(s) from structurally annotated rna multiple sequence alignment(s)">
  <meta name="twitter:image" content="https://www.carta.tech/images/infernal-cmbuild-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/cmbuild.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="cmbuild (1) manual" />
  <meta property="og:description" content="Construct covariance model(s) from structurally annotated rna multiple sequence alignment(s)" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/infernal-cmbuild-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">cmbuild<small> (1)</small></h1>
        <p class="lead">Construct covariance model(s) from structurally annotated rna multiple sequence alignment(s)</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmbuild.1.html">
      <span itemprop="name">cmbuild: Construct covariance model(s) from structurally annotated rna multiple sequence alignment(s)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/infernal/">
      <span itemprop="name">infernal</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmbuild.1.html">
      <span itemprop="name">cmbuild: Construct covariance model(s) from structurally annotated rna multiple sequence alignment(s)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>cmbuild</strong> <em>[options]</em> <em>&lt;cmfile_out&gt;</em> <em>&lt;msafile&gt;</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>For each multiple sequence alignment in <em>&lt;msafile&gt;</em> build a covariance model and save it to a new file <em>&lt;cmfile_out&gt;.</em></p><p>The alignment file must be in Stockholm or SELEX format, and must contain consensus secondary structure annotation. <strong>cmbuild</strong> uses the consensus structure to determine the architecture of the CM.</p><p><em>&lt;msafile&gt;</em> may be '-' (dash), which means reading this input from <em>stdin</em> rather than a file. To use '-', you must also specify the alignment file format with <strong>--informat</strong><em> &lt;s&gt;</em><strong>,</strong> as in <strong>--informat stockholm</strong> (because of a current limitation in our implementation, MSA file formats cannot be autodetected in a nonrewindable input stream.)</p><p><em>&lt;cmfile_out&gt;</em> may not be '-' <em>(stdout),</em> because sending the CM file to <em>stdout</em> would conflict with the other text output of the program.</p><p>In addition to writing CM(s) to <em>&lt;cmfile_out&gt;,</em> <strong>cmbuild</strong> also outputs a single line for each model created to stdout. Each line has the following fields: "aln": the index of the alignment used to build the CM; "idx": the index of the CM in the <em>&lt;cmfile_out&gt;;</em> "name": the name of the CM; "nseq": the number of sequences in the alignment used to build the CM; "eff_nseq": the effective number of sequences used to build the model; "alen": the length of the alignment used to build the CM; "clen": the number of columns from the alignment defined as consensus (match) columns; "bps": the number of basepairs in the CM; "bifs": the number of bifurcations in the CM; "rel entropy: CM": the total relative entropy of the model divided by the number of consensus columns; "rel entropy: HMM": the total relative entropy of the model ignoring secondary structure divided by the number of consensus columns. "description": description of the model/alignment.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and available options.</p>
  </dd>
  <dt>
    <p><strong>-n</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Name the new CM <em>&lt;s&gt;.</em> The default is to use the name of the alignment (if one is present in the <em>&lt;msafile&gt;),</em> or, failing that, the name of the <em>&lt;msafile&gt;.</em> If <em>&lt;msafile&gt;</em> contains more than one alignment, <em>-n</em> doesn't work, and every alignment must have a name annotated in the <em>&lt;msafile&gt;</em> (as in Stockholm #=GF ID annotation).</p>
  </dd>
  <dt>
    <p><strong>-F</strong></p>
  </dt>
  <dd>
    <p>Allow <em>&lt;cmfile_out&gt;</em> to be overwritten. Without this option, if <em>&lt;cmfile_out&gt;</em> already exists, <strong>cmbuild</strong> exits with an error.</p>
  </dd>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Direct the summary output to file <em>&lt;f&gt;,</em> rather than to <em>stdout.</em></p>
  </dd>
  <dt>
    <p><strong>-O</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>After each model is constructed, resave annotated source alignments to a file <em>&lt;f&gt;</em> in Stockholm format.  Sequences are annoted with what relative sequence weights were assigned.  The alignments are also annotated with a reference annotation line indicating which columns were assigned as consensus. If the source alignment had reference annotation ("#=GC RF") it will be replaced with the consensus residue of the model for consensus columns and '.' for insert columns, unless the <strong>--hand</strong> option was used for specifying consensus positions, in which case it will be unchanged.</p><p><strong>--devhelp</strong> Print help, as with <strong>-h ,</strong> but also include expert options that are not displayed with <strong>-h .</strong> These expert options are not expected to be relevant for the vast majority of users and so are not described in the manual page.  The only resources for understanding what they actually do are the brief one-line descriptions output when <strong>--devhelp</strong> is enabled, and the source code.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING MODEL CONSTRUCTION</h2>
        <div class="sectioncontent">
<p>These options control how consensus columns are defined in an alignment.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--fast</strong></p>
  </dt>
  <dd>
    <p>Define consensus columns automatically as those that have a fraction &gt;= <strong>symfrac</strong> of residues as opposed to gaps. (See below for the <strong>--symfrac</strong> option.) This is the default.</p>
  </dd>
  <dt>
    <p><strong>--hand</strong></p>
  </dt>
  <dd>
    <p>Use reference coordinate annotation (#=GC RF line, in Stockholm) to determine which columns are consensus, and which are inserts. Any non-gap character indicates a consensus column. (For example, mark consensus columns with "x", and insert columns with ".".) This option was called <strong>--rf</strong> in previous versions of Infernal (0.1 through 1.0.2).</p>
  </dd>
  <dt>
    <p><strong>--symfrac</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Define the residue fraction threshold necessary to define a consensus column when not using <strong>--hand.</strong> The default is 0.5. The symbol fraction in each column is calculated after taking relative sequence weighting into account.  Setting this to 0.0 means that every alignment column will be assigned as consensus, which may be useful in some cases. Setting it to 1.0 means that only columns that include 0 gaps will be assigned as consensus. This option replaces the <strong>--gapthresh</strong><em> &lt;y&gt;</em> option from previous versions of Infernal (0.1 through 1.0.2), with <em>&lt;x&gt;</em> equal to (1.0 - <em>&lt;y&gt;).</em> For example to reproduce behavior for a command of <strong>cmbuild --gapthresh  0.8</strong> in a previous version, use <strong>cmbuild --symfrac  0.2</strong> with this version.</p>
  </dd>
  <dt>
    <p><strong>--noss</strong></p>
  </dt>
  <dd>
    <p>Ignore the secondary structure annotation, if any, in <em>&lt;msafile&gt;</em> and build a CM with zero basepairs. This model will be similar to a profile HMM and the <strong>cmsearch</strong> and <strong>cmscan</strong> programs will use HMM algorithms which are faster than CM ones for this model. Additionally, a zero basepair model need not be calibrated with <strong>cmcalibrate</strong> prior to running <strong>cmsearch</strong> with it. The <strong>--noss</strong> option must be used if there is no secondary structure annotation in <strong>&lt;msafile&gt;.</strong></p>
  </dd>
  <dt>
    <p><strong>--rsearch</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Parameterize emission scores a la RSEARCH, using the RIBOSUM matrix in file <em>&lt;f&gt;.</em> With <strong>--rsearch</strong> enabled, all alignments in <em>&lt;msafile&gt;</em> must contain exactly one sequence or the <strong>--call</strong> option must also be enabled. All positions in each sequence will be considered consensus "columns".  Actually, the emission scores for these models will not be identical to RIBOSUM scores due of differences in the modelling strategy between Infernal and RSEARCH, but they will be as similar as possible.  RIBOSUM matrix files are included with Infernal in the "matrices/" subdirectory of the top-level "infernal-xxx" directory.  RIBOSUM matrices are substitution score matrices trained specifically for structural RNAs with separate single stranded residue and base pair substitution scores. For more information see the RSEARCH publication (Klein and Eddy, BMC Bioinformatics 4:44, 2003).</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER MODEL CONSTRUCTION OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--null</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Read a null model from <em>&lt;f&gt;.</em> The null model defines the probability of each RNA nucleotide in background sequence, the default is to use 0.25 for each nucleotide. The format of null files is specified in the user guide.</p>
  </dd>
  <dt>
    <p><strong>--prior</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Read a Dirichlet prior from <em>&lt;f&gt;,</em> replacing the default mixture Dirichlet. The format of prior files is specified in the user guide.</p>
  </dd>

</dl>
<p>Use <strong>--devhelp</strong> to see additional, otherwise undocumented, model construction options.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING RELATIVE WEIGHTS</h2>
        <div class="sectioncontent">
<p><strong>cmbuild</strong> uses an ad hoc sequence weighting algorithm to downweight closely related sequences and upweight distantly related ones. This has the effect of making models less biased by uneven phylogenetic representation. For example, two identical sequences would typically each receive half the weight that one sequence would.  These options control which algorithm gets used.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--wpb</strong></p>
  </dt>
  <dd>
    <p>Use the Henikoff position-based sequence weighting scheme [Henikoff and Henikoff, J. Mol. Biol. 243:574, 1994].  This is the default.</p>
  </dd>
  <dt>
    <p><strong>--wgsc</strong></p>
  </dt>
  <dd>
    <p>Use the Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et al, J. Mol. Biol. 235:1067, 1994].</p>
  </dd>
  <dt>
    <p><strong>--wnone</strong></p>
  </dt>
  <dd>
    <p>Turn sequence weighting off; e.g. explicitly set all sequence weights to 1.0.</p>
  </dd>
  <dt>
    <p><strong>--wgiven</strong></p>
  </dt>
  <dd>
    <p>Use sequence weights as given in annotation in the input alignment file. If no weights were given, assume they are all 1.0.  The default is to determine new sequence weights by the Gerstein/Sonnhammer/Chothia algorithm, ignoring any annotated weights.</p>
  </dd>
  <dt>
    <p><strong>--wblosum</strong></p>
  </dt>
  <dd>
    <p>Use the BLOSUM filtering algorithm to weight the sequences, instead of the default GSC weighting. Cluster the sequences at a given percentage identity (see <strong>--wid);</strong> assign each cluster a total weight of 1.0, distributed equally amongst the members of that cluster.</p>
  </dd>
  <dt>
    <p><strong>--wid</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Controls the behavior of the <em>--wblosum</em> weighting option by setting the percent identity for clustering the alignment to <em>&lt;x&gt;.</em></p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING EFFECTIVE SEQUENCE NUMBER</h2>
        <div class="sectioncontent">
<p>After relative weights are determined, they are normalized to sum to a total effective sequence number, <em>eff_nseq.</em> This number may be the actual number of sequences in the alignment, but it is almost always smaller than that. The default entropy weighting method <em>(--eent)</em> reduces the effective sequence number to reduce the information content (relative entropy, or average expected score on true homologs) per consensus position. The target relative entropy is controlled by a two-parameter function, where the two parameters are settable with <em>--ere</em> and <em>--esigma.</em></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--eent</strong></p>
  </dt>
  <dd>
    <p>Use the entropy weighting strategy to determine the effective sequence number that gives a target mean match state relative entropy. This option is the default, and can be turned off with <strong>--enone.</strong> The default target mean match state relative entropy is 0.59 bits for models with at least 1 basepair and 0.38 bits for models with zero basepairs, but changed with <strong>--ere.</strong> The default of 0.59 or 0.38 bits is automatically changed if the total relative entropy of the model (summed match state relative entropy) is less than a cutoff, which is is 6.0 bits by default, but can be changed with the expert, undocumented <strong>--eX</strong> option. If you really want to play with that option, consult the source code.</p>
  </dd>
  <dt>
    <p><strong>--enone</strong></p>
  </dt>
  <dd>
    <p>Turn off the entropy weighting strategy. The effective sequence number is just the number of sequences in the alignment.</p>
  </dd>
  <dt>
    <p><strong>--ere</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the target mean match state relative entropy as <em>&lt;x&gt;.</em> By default the target relative entropy per match position is 0.59 bits for models with at least 1 basepair and 0.38 for models with zero basepairs.</p>
  </dd>
  <dt>
    <p><strong>--eminseq</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Define the minimum allowed effective sequence number as <em>&lt;x&gt;.</em></p>
  </dd>
  <dt>
    <p><strong>--ehmmre</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the target HMM mean match state relative entropy as <em>&lt;x&gt;.</em> Entropy for basepairing match states is calculated using marginalized basepair emission probabilities.</p>
  </dd>
  <dt>
    <p><strong>--eset</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the effective sequence number for entropy weighting as <em>&lt;x&gt;.</em></p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING FILTER P7 HMM CONSTRUCTION</h2>
        <div class="sectioncontent">
<p>For each CM that <strong>cmbuild</strong> constructs, an accompanying filter p7 HMM is built from the input alignment as well. These options control filter HMM construction:</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--p7ere</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the target mean match state relative entropy for the filter p7 HMM as <em>&lt;x&gt;.</em> By default the target relative entropy per match position is 0.38 bits.</p>
  </dd>
  <dt>
    <p><strong>--p7ml</strong></p>
  </dt>
  <dd>
    <p>Use a maximum likelihood p7 HMM built from the CM as the filter HMM. This HMM will be as similar as possible to the CM (while necessarily ignorant of secondary structure).</p>
  </dd>

</dl>
<p>Use <strong>--devhelp</strong> to see additional, otherwise undocumented, filter HMM construction options.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING FILTER P7 HMM CALIBRATION</h2>
        <div class="sectioncontent">
<p>After building each filter HMM, <strong>cmbuild</strong> determines appropriate E-value parameters to use during filtering in <strong>cmsearch</strong> and <strong>cmscan</strong> by sampling a set of sequences and searching them with each HMM filter configuration and algorithm.</p><p><strong>--EmN</strong><em> &lt;n&gt;</em> Set the number of sampled sequences for local MSV filter HMM calibration to <em>&lt;n&gt;.</em> 200 by default.</p><p><strong>--EvN</strong><em> &lt;n&gt;</em> Set the number of sampled sequences for local Viterbi filter HMM calibration to <em>&lt;n&gt;.</em> 200 by default.</p><p><strong>--ElfN</strong><em> &lt;n&gt;</em> Set the number of sampled sequences for local Forward filter HMM calibration to <em>&lt;n&gt;.</em> 200 by default.</p><p><strong>--EgfN</strong><em> &lt;n&gt;</em> Set the number of sampled sequences for glocal Forward filter HMM calibration to <em>&lt;n&gt;.</em> 200 by default.</p><p>Use <strong>--devhelp</strong> to see additional, otherwise undocumented, filter HMM calibration options.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR REFINING THE INPUT ALIGNMENT</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--refine</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Attempt to refine the alignment before building the CM using expectation-maximization (EM). A CM is first built from the initial alignment as usual. Then, the sequences in the alignment are realigned optimally (with the HMM banded CYK algorithm, optimal means optimal given the bands) to the CM, and a new CM is built from the resulting alignment. The sequences are then realigned to the new CM, and a new CM is built from that alignment. This is continued until convergence, specifically when the alignments for two successive iterations are not significantly different (the summed bit scores of all the sequences in the alignment changes less than 1% between two successive iterations). The final alignment (the alignment used to build the CM that gets written to <em>&lt;cmfile_out&gt;)</em> is written to <em>&lt;f&gt;.</em></p>
  </dd>
  <dt>
    <p><strong>-l</strong></p>
  </dt>
  <dd>
    <p>With <strong>--refine,</strong> turn on the local alignment algorithm, which allows the alignment to span two or more subsequences if necessary (e.g. if the structures of the query model and target sequence are only partially shared), allowing certain large insertions and deletions in the structure to be penalized differently than normal indels. The default is to globally align the query model to the target sequences.</p>
  </dd>
  <dt>
    <p><strong>--gibbs</strong></p>
  </dt>
  <dd>
    <p>Modifies the behavior of <strong>--refine</strong> so Gibbs sampling is used instead of EM. The difference is that during the alignment stage the alignment is not necessarily optimal, instead an alignment (parsetree) for each sequences is sampled from the posterior distribution of alignments as determined by the Inside algorithm. Due to this sampling step <strong>--gibbs</strong> is non-deterministic, so different runs with the same alignment may yield different results. This is not true when <strong>--refine</strong> is used without the <strong>--gibbs</strong> option, in which case the final alignment and CM will always be the same. When <strong>--gibbs</strong> is enabled, the <strong>--seed  &lt;n&gt;</strong> option can be used to seed the random number generator predictably, making the results reproducible. The goal of the <strong>--gibbs</strong> option is to help expert RNA alignment curators refine structural alignments by allowing them to observe alternative high scoring alignments.</p>
  </dd>
  <dt>
    <p><strong>--seed</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Seed the random number generator with <em>&lt;n&gt;,</em> an integer &gt;= 0. This option can only be used in combination with <strong>--gibbs.</strong> If <em>&lt;n&gt;</em> is nonzero, stochastic sampling of alignments will be reproducible; the same command will give the same results. If <em>&lt;n&gt;</em> is 0, the random number generator is seeded arbitrarily, and stochastic samplings may vary from run to run of the same command. The default seed is 0.</p>
  </dd>
  <dt>
    <p><strong>--cyk</strong></p>
  </dt>
  <dd>
    <p>With <strong>--refine,</strong> align with the CYK algorithm. By default the optimal accuracy algorithm is used. There is more information on this in the <strong>cmalign</strong> manual page.</p>
  </dd>
  <dt>
    <p><strong>--notrunc</strong></p>
  </dt>
  <dd>
    <p>With <strong>--refine,</strong> turn off the the truncated alignment algorithm. There is more information on this in the <strong>cmalign</strong> manual page.</p>
  </dd>

</dl>
<p>Use <strong>--devhelp</strong> to see additional, otherwise undocumented, alignment refinement options as well as other output file options and options for building multiple models for a single alignment.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO cmbuild&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/infernal.1.html"><strong>infernal</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the Infernal package.</p><p>For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf); or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2014 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your Infernal source distribution, or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="cmatrix.1.html"><span aria-hidden="true">&larr;</span> cmatrix.1: Simulates the display from "the matrix"</a></li>
   <li class="next"><a href="cmcalibrate.1.html">cmcalibrate.1: Fit exponential tails for covariance model e-value determination <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
