<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>cmcalibrate: Fit exponential tails for covariance model e-value determination</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Fit exponential tails for covariance model e-value determination">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="cmcalibrate (1) manual">
  <meta name="twitter:description" content="Fit exponential tails for covariance model e-value determination">
  <meta name="twitter:image" content="https://www.carta.tech/images/infernal-cmcalibrate-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/cmcalibrate.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="cmcalibrate (1) manual" />
  <meta property="og:description" content="Fit exponential tails for covariance model e-value determination" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/infernal-cmcalibrate-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">cmcalibrate<small> (1)</small></h1>
        <p class="lead">Fit exponential tails for covariance model e-value determination</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmcalibrate.1.html">
      <span itemprop="name">cmcalibrate: Fit exponential tails for covariance model e-value determination</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/infernal/">
      <span itemprop="name">infernal</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmcalibrate.1.html">
      <span itemprop="name">cmcalibrate: Fit exponential tails for covariance model e-value determination</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>cmcalibrate</strong> <em>[options]</em> <em>cmfile</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>cmcalibrate</strong> determines exponential tail parameters for E-value determination by generating random sequences, searching them with the CM and collecting the scores of the resulting hits. A histogram of the bit scores of the hits is fit to an exponential tail, and the parameters of the fitted tail are saved to the CM file. The exponential tail parameters are then used to estimate the statistical significance of hits found in <strong>cmsearch</strong> and <strong>cmscan.</strong></p><p>A CM file must be calibrated with <strong>cmcalibrate</strong> before it can be used in <strong>cmsearch</strong> or <strong>cmscan,</strong> with a single exception: it is not necessary to calibrate CM files that include only models with zero basepairs before running <strong>cmsearch.</strong></p><p><strong>cmcalibrate</strong> is very slow. It takes a couple of hours to calibrate a single average sized CM on a single CPU. <strong>cmcalibrate</strong> will run in parallel on all available cores if Infernal was built on a system that supports POSIX threading (see the Installation section of the user guide for more information). Using <strong>&lt;n&gt;</strong> cores will result in roughly <strong>&lt;n&gt;</strong> -fold acceleration versus a single CPU.  MPI (Message Passing Interface) can be also be used for parallelization with the <strong>--mpi</strong> option if Infernal was built with MPI enabled, but using more than 161 processors is not recommended because increasing past 161 won't accelerate the calibration.  See the Installation seciton of the user guide for more information.</p><p>The <strong>--forecast</strong> option can be used to estimate how long the program will take to run for a given <em>cmfile</em> on the current machine. To predict the running time on <em>&lt;n&gt;</em> processors with MPI, additionally use the <strong>--nforecast</strong><em> &lt;n&gt;</em> option.</p><p>The random sequences searched in <strong>cmcalibrate</strong> are generated by an HMM that was trained on real genomic sequences with various GC contents. The goal is to have the GC distributions in the random sequences be similar to those in actual genomic sequences.</p><p>Four rounds of searches and subsequent exponential tail fits are performed, one each for the four different CM algorithms that can be used in <strong>cmsearch</strong> and <strong>cmscan:</strong> glocal CYK, glocal Inside, local CYK and local Inside.</p><p>The E-values parameters determined by <strong>cmcalibrate</strong> are only used by the <strong>cmsearch</strong> and <strong>cmscan</strong> programs. If you are not going to use these programs then do not waste time calibrating your models.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and available options.</p>
  </dd>
  <dt>
    <p><strong>-L</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the total length of random sequences to search to <em>&lt;x&gt;</em> megabases (Mb). By default, <em>&lt;x&gt; is</em> 1.6 Mb. Increasing <em>&lt;x&gt;</em> will make the exponential tail fits more precise and E-values more accurate, but will take longer (doubling <em>&lt;x&gt;</em> will roughly double the running time). Decreasing <em>&lt;x&gt;</em> is not recommended as it will make the fits less precise and the E-values less accurate.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR PREDICTING REQUIRED TIME AND MEMORY</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--forecast</strong></p>
  </dt>
  <dd>
    <p>Predict the running time of the calibration of <em>cmfile</em> (with provided options) on the current machine and exit. The calibration is not performed. The predictions should be considered rough estimates. If multithreading is enabled (see Installation section of user guide), the timing will take into account the number of available cores.</p>
  </dd>
  <dt>
    <p><strong>--nforecast</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>With <strong>--forecast,</strong> specify that <em>&lt;n&gt;</em> processors will be used for the calibration. This might be useful for predicting the running time of an MPI run with <em>&lt;n&gt;</em> processors.</p>
  </dd>
  <dt>
    <p><strong>--memreq</strong></p>
  </dt>
  <dd>
    <p>Predict the amount of required memory for calibrating <em>cmfile</em> (with provided options) on the current machine and exit. The calibration is not performed.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING EXPONENTIAL TAIL FITS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--gtailn</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>fit the exponential tail for glocal Inside and glocal CYK to the <em>&lt;n&gt;</em> highest scores in the histogram tail, where <em>&lt;n&gt;</em> is <em>&lt;x&gt;</em> times the number of Mb searched. The default value of <em>&lt;x&gt;</em> is 250. The value 250 was chosen because it works well empirically relative to other values.</p>
  </dd>
  <dt>
    <p><strong>--ltailn</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>fit the exponential tail for local Inside and local CYK to the <em>&lt;n&gt;</em> highest scores in the histogram tail, where <em>&lt;n&gt;</em> is <em>&lt;x&gt;</em> times the number of Mb searched. The default value of <em>&lt;x&gt;</em> is 750. The value 750 was chosen because it works well empirically relative to other values.</p>
  </dd>
  <dt>
    <p><strong>--tailp</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Ignore the <strong>--gtailn</strong> and <strong>--ltailn</strong> prefixed options and fit the <em>&lt;x&gt;</em> fraction tail of the histogram to an exponential tail, for all search modes.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONAL OUTPUT FILES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--hfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save the histograms fit to file <em>&lt;f&gt;.</em> The format of this file is two space delimited columns per line. The first column is the x-axis values of bit scores of each bin. The second column is the y-axis values of number of hits per bin. Each series is delimited by a line with a single character "&". The file will contain one series for each of the four exponential tail fits in the following order: glocal CYK, glocal Inside, local CYK, and local Inside.</p>
  </dd>
  <dt>
    <p><strong>--sfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save survival plot information to file <em>&lt;f&gt;.</em> The format of this file is two space delimited columns per line. The first column is the x-axis values of bit scores of each bin. The second column is the y-axis values of fraction of hits that meet or exceed the score for each bin. Each series is delimited by a line with a single character "&". The file will contain three series of data for each of the four CM search modes in the following order: glocal CYK, glocal Inside, local CYK, and local Inside. The first series is the empirical survival plot from the histogram of hits to the random sequence. The second series is the exponential tail fit to the empirical distribution. The third series is the exponential tail fit if lambda were fixed and set as the natural log of 2 (0.691314718).</p>
  </dd>
  <dt>
    <p><strong>--qqfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save quantile-quantile plot information to file <em>&lt;f&gt;.</em> The format of this file is two space delimited columns per line. The first column is the x-axis values, and the second column is the y-axis values. The distance of the points from the identity line (y=x) is a measure of how good the exponential tail fit is, the closer the points are to the identity line, the better the fit is. Each series is delimited by a line with a single character "&". The file will contain one series of empirical data for each of the four exponential tail fits in the following order: glocal CYK, glocal Inside, local CYK and local Inside.</p>
  </dd>
  <dt>
    <p><strong>--ffile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save space delimited statistics of different exponential tail fits to file <em>&lt;f&gt;.</em> The file will contain the lambda and mu values for exponential tails fit to histogram tails of different sizes. The fields in the file are labelled informatively.</p>
  </dd>
  <dt>
    <p><strong>--xfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save a list of the scores in each fit histogram tail to file <em>&lt;f&gt;.</em> Each line of this file will have a different score indicating one hit existed in the tail with that score.  Each series is delimited by a line with a single character "&". The file will contain one series for each of the four exponential tail fits in the following order: glocal CYK, glocal Inside, local CYK, and local Inside.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--seed</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Seed the random number generator with <em>&lt;n&gt;,</em> an integer &gt;= 0. If <em>&lt;n&gt;</em> is nonzero, stochastic simulations will be reproducible; the same command will give the same results. If <em>&lt;n&gt;</em> is 0, the random number generator is seeded arbitrarily, and stochastic simulations will vary from run to run of the same command. The default seed is 181.</p>
  </dd>
  <dt>
    <p><strong>--beta</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>By default query-dependent banding (QDB) is used to accelerate the CM search algorithms with a beta tail loss probability of 1E-15. This beta value can be changed to <em>&lt;x&gt;</em> with <strong>--beta</strong><em> &lt;x&gt;</em><strong>.</strong> The beta parameter is the amount of probability mass excluded during band calculation, higher values of beta give greater speedups but sacrifice more accuracy than lower values. The default value used is 1E-15. (For more information on QDB see Nawrocki and Eddy, PLoS Computational Biology 3(3): e56.)</p>
  </dd>
  <dt>
    <p><strong>--nonbanded</strong></p>
  </dt>
  <dd>
    <p>Turn off QDB during E-value calibration. This will slow down calibration.</p>
  </dd>
  <dt>
    <p><strong>--nonull3</strong></p>
  </dt>
  <dd>
    <p>Turn off the null3 post hoc additional null model. This is not recommended unless you plan on using the same option to <strong>cmsearch</strong> and/or <strong>cmscan.</strong></p>
  </dd>
  <dt>
    <p><strong>--random</strong></p>
  </dt>
  <dd>
    <p>Use the background null model of the CM to generate the random sequences, instead of the more realistic HMM. Unless the CM was built using the <strong>--null</strong> option to <strong>cmbuild,</strong> the background null model will be 25% each A, C, G and U.</p>
  </dd>
  <dt>
    <p><strong>--gc</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Generate the random sequences using the nucleotide distribution from the sequence file <em>&lt;f&gt;.</em></p>
  </dd>
  <dt>
    <p><strong>--cpu</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Specify that <em>&lt;n&gt;</em> parallel CPU workers be used. If <em>&lt;n&gt;</em> is set as "0", then the program will be run in serial mode, without using threads. You can also control this number by setting an environment variable, <em>INFERNAL_NCPU.</em> This option will only be available if the machine on which Infernal was built is capable of using POSIX threading (see the Installation section of the user guide for more information).</p>
  </dd>
  <dt>
    <p><strong>--mpi</strong></p>
  </dt>
  <dd>
    <p>Run as an MPI parallel program. This option will only be available if Infernal has been configured and built with the "--enable-mpi" flag (see the Installation section of the user guide for more information).</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO cmcalibrate&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/infernal.1.html"><strong>infernal</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the Infernal package.</p><p>For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf); or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2014 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your Infernal source distribution, or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="cmbuild.1.html"><span aria-hidden="true">&larr;</span> cmbuild.1: Construct covariance model(s) from structurally annotated rna multiple sequence alignment(s)</a></li>
   <li class="next"><a href="cmconvert.1.html">cmconvert.1: Convert infernal covariance model files <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
