<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>cmemit: Sample sequences from a covariance model</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Sample sequences from a covariance model">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="cmemit (1) manual">
  <meta name="twitter:description" content="Sample sequences from a covariance model">
  <meta name="twitter:image" content="https://www.carta.tech/images/infernal-cmemit-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/cmemit.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="cmemit (1) manual" />
  <meta property="og:description" content="Sample sequences from a covariance model" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/infernal-cmemit-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">cmemit<small> (1)</small></h1>
        <p class="lead">Sample sequences from a covariance model</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmemit.1.html">
      <span itemprop="name">cmemit: Sample sequences from a covariance model</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/infernal/">
      <span itemprop="name">infernal</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmemit.1.html">
      <span itemprop="name">cmemit: Sample sequences from a covariance model</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>cmemit</strong> <em>[options]</em> <em>&lt;cmfile&gt;</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>The <strong>cmemit</strong> program samples (emits) sequences from the covariance model(s) in <em>&lt;cmfile&gt;,</em> and writes them to output. Sampling sequences may be useful for a variety of purposes, including creating synthetic true positives for benchmarks or tests.</p><p>The default is to sample ten unaligned sequence from each CM. Alternatively, with the <strong>-c</strong> option, you can emit a single majority-rule consensus sequence; or with the <strong>-a</strong> option, you can emit an alignment.</p><p>The <em>&lt;cmfile&gt;</em> may contain a library of CMs, in which case each CM will be used in turn.</p><p><em>&lt;cmfile&gt;</em> may be '-' (dash), which means reading this input from <em>stdin</em> rather than a file.</p><p>For models with zero basepairs, sequences are sampled from the profile HMM filter instead of the CM. However, since these models will be nearly identical (unless special options were used in <strong>cmbuild</strong> to prevent this), using the HMM instead of the CM will not change the output in a significant way, unless the <strong>-l</strong> option is used. With <strong>-l,</strong> the HMM will be configured for equiprobable model begin and end positions, while the CM will not. You can force <strong>cmemit</strong> to always sample from the CM with the <strong>--nohmmonly</strong> option.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and available options.</p>
  </dd>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save the synthetic sequences to file <em>&lt;f&gt;</em> rather than writing them to stdout.</p>
  </dd>
  <dt>
    <p><strong>-N</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Generate <em>&lt;n&gt;</em> sequences. The default value for <em>&lt;n&gt;</em> is 10.</p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Write the generated sequences in unaligned format (FASTA). This is the default behavior.</p>
  </dd>
  <dt>
    <p><strong>-a</strong></p>
  </dt>
  <dd>
    <p>Write the generated sequences in an aligned format (STOCKHOLM) with consensus structure annotation rather than FASTA. Other output formats are possible with the <strong>--outformat</strong> option.</p>
  </dd>
  <dt>
    <p><strong>-c</strong></p>
  </dt>
  <dd>
    <p>Predict a single majority-rule consensus sequence instead of sampling sequences from the CM\'s probability distribution. Highly conserved residues (base paired residues that score higher than 3.0 bits, or single stranded residues that score higher than 1.0 bits) are shown in upper case; others are shown in lower case.</p>
  </dd>
  <dt>
    <p><strong>-e</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Embed the CM emitted sequences in a larger randomly generated sequence of length <em>&lt;n&gt;</em> generated from an HMM that was trained on real genomic sequences with various GC contents (the same HMM used by <strong>cmcalibrate).</strong> You can use the <strong>--iid</strong> option to generate 25% A, C, G, and U sequence instead. The CM emitted sequence will begin at a random position within the larger sequence and will be included in its entirety unless the <strong>--u5p</strong> or <strong>--u3p</strong> options are used. When <strong>-e</strong> is used in combination with <strong>--u5p,</strong> the CM emitted sequence will always begin at position 1 of the larger sequence and will be truncated 5'. When used in combination <strong>--u3p</strong> the CM emitted sequence will always end at position <em>&lt;n&gt;</em> of the larger sequence and will be truncated 3'.</p>
  </dd>
  <dt>
    <p><strong>-l</strong></p>
  </dt>
  <dd>
    <p>Configure the CMs into local mode before emitting sequences. By default the model will be in global mode. In local mode, large insertions and deletions are more common than in global mode.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR TRUNCATING EMITTED SEQUENCES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--u5p</strong></p>
  </dt>
  <dd>
    <p>Truncate all emitted sequences at a randomly chosen start position <strong>&lt;n&gt;,</strong> by only outputting residues beginning at <strong>&lt;n&gt;.</strong> A different start point is randomly chosen for each sequence.</p>
  </dd>
  <dt>
    <p><strong>--u3p</strong></p>
  </dt>
  <dd>
    <p>Truncate all emitted sequences at a randomly chosen end position <strong>&lt;n&gt;,</strong> by only outputting residues up to position <strong>&lt;n&gt;.</strong> A different end point is randomly chosen for each sequence.</p>
  </dd>
  <dt>
    <p><strong>--a5p</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>In combination with the <strong>-a</strong> option, truncate the emitted alignment at a randomly chosen start match position <em>&lt;n&gt;,</em> by only outputting alignment columns for positions after match state <em>&lt;n&gt;</em> - 1. <em>&lt;n&gt;</em> must be an integer between 0 and the consensus length of the model (which can be determined using the <strong>cmstat</strong> program. As a special case, using 0 as <em>&lt;n&gt;</em> will result in a randomly chosen start position.</p>
  </dd>
  <dt>
    <p><strong>--a3p</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>In combination with the <strong>-a</strong> option, truncate the emitted alignment at a randomly chosen end match position <em>&lt;n&gt;,</em> by only outputting alignment columns for positions before match state <em>&lt;n&gt;</em> + 1. <em>&lt;n&gt;</em> must be an integer between 1 and the consensus length of the model (which can be determined using the <strong>cmstat</strong> program). As a special case, using 0 as <em>&lt;n&gt;</em> will result in a randomly chosen end position.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--seed</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Seed the random number generator with <em>&lt;n&gt;,</em> an integer &gt;= 0. If <em>&lt;n&gt;</em> is nonzero, stochastic sampling of sequences will be reproducible; the same command will give the same results. If <em>&lt;n&gt;</em> is 0, the random number generator is seeded arbitrarily, and stochastic samplings will vary from run to run of the same command. The default seed is 0.</p>
  </dd>
  <dt>
    <p><strong>--iid</strong></p>
  </dt>
  <dd>
    <p>With <strong>-e,</strong> generate the larger sequences as 25% each A, C, G and U.</p>
  </dd>
  <dt>
    <p><strong>--rna</strong></p>
  </dt>
  <dd>
    <p>Specify that the emitted sequences be output as RNA sequences. This is true by default.</p>
  </dd>
  <dt>
    <p><strong>--dna</strong></p>
  </dt>
  <dd>
    <p>Specify that the emitted sequences be output as DNA sequences. By default, the output alphabet is RNA.</p>
  </dd>
  <dt>
    <p><strong>--idx</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Specify that the emitted sequences be named starting with <em>&lt;modelname&gt;.&lt;n&gt;.</em> By default <em>&lt;n&gt;</em> is 1.</p>
  </dd>
  <dt>
    <p><strong>--outformat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>With <strong>-a,</strong> specify the output alignment format as <em>&lt;s&gt;.</em> Acceptable formats are: Pfam, AFA, A2M, Clustal, and Phylip. AFA is aligned fasta. Only Pfam and Stockholm alignment formats will include consensus structure annotation.</p>
  </dd>
  <dt>
    <p><strong>--tfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Dump tabular sequence parsetrees (tracebacks) for each emitted sequence to file <em>&lt;f&gt;.</em> Primarily useful for debugging.</p>
  </dd>
  <dt>
    <p><strong>--exp</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Exponentiate the emission and transition probabilities of the CM by <em>&lt;x&gt;</em> and then renormalize those distributions before emitting sequences. This option changes the CM probability distribution of parsetrees relative to default. With <em>&lt;x&gt;</em> less than 1.0 the emitted sequences will tend to have lower bit scores upon alignment to the CM. With &lt;x&gt; greater than 1.0, the emitted sequences will tend to have higher bit scores upon alignment to the CM. This bit score difference will increase as &lt;x&gt; moves further away from 1.0 in either direction. If &lt;x&gt; equals 1.0, this option has no effect relative to default. This option is useful for generating sequences that are either more difficult ( <em>&lt;x&gt;</em> &lt; 1.0) or easier ( <em>&lt;x&gt;</em> &gt; 1.0) for the CM to distinguish as homologous from background, random sequence.</p>
  </dd>
  <dt>
    <p><strong>--hmmonly</strong></p>
  </dt>
  <dd>
    <p>Emit from the filter profile HMM instead of the CM.</p>
  </dd>
  <dt>
    <p><strong>--nohmmonly</strong></p>
  </dt>
  <dd>
    <p>Never emit from the filter profile HMM, always use the CM, even for models with zero basepairs.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO cmemit&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/infernal.1.html"><strong>infernal</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the Infernal package.</p><p>For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf); or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2014 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your Infernal source distribution, or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="cmdtool.1.html"><span aria-hidden="true">&larr;</span> cmdtool.1: Run a shell (or other program) in an openwindows enhanced terminal window</a></li>
   <li class="next"><a href="mew-beta-bin-cmew.1.html">cmew.1: Create mew's database file <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
