<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>cmscan: Search sequence(s) against a covariance model database</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Search sequence(s) against a covariance model database">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="cmscan (1) manual">
  <meta name="twitter:description" content="Search sequence(s) against a covariance model database">
  <meta name="twitter:image" content="https://www.carta.tech/images/infernal-cmscan-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/cmscan.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="cmscan (1) manual" />
  <meta property="og:description" content="Search sequence(s) against a covariance model database" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/infernal-cmscan-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">cmscan<small> (1)</small></h1>
        <p class="lead">Search sequence(s) against a covariance model database</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmscan.1.html">
      <span itemprop="name">cmscan: Search sequence(s) against a covariance model database</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/infernal/">
      <span itemprop="name">infernal</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/cmscan.1.html">
      <span itemprop="name">cmscan: Search sequence(s) against a covariance model database</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>cmscan</strong> <em>[options]</em> <em>&lt;cmdb&gt;</em> <em>&lt;seqfile&gt;</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>cmscan</strong> is used to search sequences against collections of covariance models. For each sequence in <em>&lt;seqfile&gt;,</em> use that query sequence to search the target database of CMs in <em>&lt;cmdb&gt;,</em> and output ranked lists of the CMs with the most significant matches to the sequence.</p><p>The <em>&lt;seqfile&gt;</em> may contain more than one query sequence. It can be in FASTA format, or several other common sequence file formats (genbank, embl, and among others), or in alignment file formats (stockholm, aligned fasta, and others). See the <em>--qformat</em> option for a complete list.</p><p>The <em>&lt;cmdb&gt;</em> needs to be press'ed using <strong>cmpress</strong> before it can be searched with <strong>cmscan.</strong> This creates four binary files, suffixed .i1{fimp}. Additionally, <em>&lt;cmdb&gt;</em> must have been calibrated for E-values with <strong>cmcalibrate</strong> before being press'ed with <strong>cmpress.</strong></p><p>The query <em>&lt;seqfile&gt;</em> may be '-' (a dash character), in which case the query sequences are read from a &lt;stdin&gt; pipe instead of from a file. The <em>&lt;cmdb&gt;</em> cannot be read from a &lt;stdin&gt; stream, because it needs to have those four auxiliary binary files generated by <strong>cmpress.</strong></p><p>The output format is designed to be human-readable, but is often so voluminous that reading it is impractical, and parsing it is a pain. The <strong>--tblout</strong> option saves output in a simple tabular format that is concise and easier to parse. The <strong>-o</strong> option allows redirecting the main output, including throwing it away in /dev/null.</p><p><strong>cmscan</strong> reexamines the 5' and 3' termini of target sequences using specialized algorithms for detection of <em>truncated</em> hits, in which part of the 5' and/or 3' end of the actual full length homologous sequence is missing in the target sequence file. These types of hits will be most common in sequence files consisting of unassembled sequencing reads. By default, any 5' truncated hit is required to include the first residue of the target sequence it derives from in <em>&lt;seqfile&gt;,</em> and any 3' truncated hit is required to include the final residue of the target sequence it derives from. Any 5' and 3' truncated hit must include the first and final residue of the target sequence it derives from. The <strong>--anytrunc</strong> option will relax the requirements for hit inclusion of sequence endpoints, and truncated hits are allowed to start and stop at any positions of target sequences. Importantly though, with <strong>--anytrunc,</strong> hit E-values will be less accurate because model calibration does not consider the possibility of truncated hits, so use it with caution. The <strong>--notrunc</strong> option can be used to turn off truncated hit detection. <strong>--notrunc</strong> will reduce the running time of <strong>cmscan,</strong> most significantly for target <em>&lt;seqfile&gt;</em> files that include many short sequences. Truncated hit detection is automatically turned off when the <strong>--max,</strong> <strong>--nohmm,</strong> <strong>--qdb,</strong> or <strong>--nonbanded</strong> options are used because it relies on the use of an accelerated HMM banded alignment strategy that is turned off by any of those options.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and all available options.</p>
  </dd>
  <dt>
    <p><strong>-g</strong></p>
  </dt>
  <dd>
    <p>Turn on the <em>glocal</em> alignment algorithm, global with respect to the query model and local with respect to the target database. By default, the local alignment algorithm is used which is local with respect to both the target sequence and the model. In local mode, the alignment to span two or more subsequences if necessary (e.g. if the structures of the query model and target sequence are only partially shared), allowing certain large insertions and deletions in the structure to be penalized differently than normal indels. Local mode performs better on empirical benchmarks and is significantly more sensitive for remote homology detection. Empirically, glocal searches return many fewer hits than local searches, so glocal may be desired for some applications.</p>
  </dd>
  <dt>
    <p><strong>-Z</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Calculate E-values as if the search space size was <em>&lt;x&gt;</em> megabases (Mb). Without the use of this option, the search space size changes for each query sequence, it is defined as the length of the current query sequence times 2 (because both strands of the sequence will be searched) times the number of CMs in <em>&lt;cmdb&gt;.</em></p>
  </dd>
  <dt>
    <p><strong>--devhelp</strong></p>
  </dt>
  <dd>
    <p>Print help, as with <strong>-h ,</strong> but also include expert options that are not displayed with <strong>-h .</strong> These expert options are not expected to be relevant for the vast majority of users and so are not described in the manual page. The only resources for understanding what they actually do are the brief one-line descriptions output when <strong>--devhelp</strong> is enabled, and the source code.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR CONTROLLING OUTPUT</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Direct the main human-readable output to a file <em>&lt;f&gt;</em> instead of the default stdout.</p>
  </dd>
  <dt>
    <p><strong>--tblout</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save a simple tabular (space-delimited) file summarizing the hits found, with one data line per hit. The format of this file is described in the Infernal user guide.</p>
  </dd>
  <dt>
    <p><strong>--acc</strong></p>
  </dt>
  <dd>
    <p>Use accessions instead of names in the main output, where available for profiles and/or sequences.</p>
  </dd>
  <dt>
    <p><strong>--noali</strong></p>
  </dt>
  <dd>
    <p>Omit the alignment section from the main output. This can greatly reduce the output volume.</p>
  </dd>
  <dt>
    <p><strong>--notextw</strong></p>
  </dt>
  <dd>
    <p>Unlimit the length of each line in the main output. The default is a limit of 120 characters per line, which helps in displaying the output cleanly on terminals and in editors, but can truncate target profile description lines.</p>
  </dd>
  <dt>
    <p><strong>--textw</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the main output's line length limit to <em>&lt;n&gt;</em> characters per line. The default is 120.</p>
  </dd>
  <dt>
    <p><strong>--verbose</strong></p>
  </dt>
  <dd>
    <p>Include extra search pipeline statistics in the main output, including filter survival statistics for truncated hit detection and number of envelopes discarded due to matrix size overflows.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING REPORTING THRESHOLDS</h2>
        <div class="sectioncontent">
<p>Reporting thresholds control which hits are reported in output files (the main output and <em>--tblout)</em> Hits are ranked by statistical significance (E-value). By default, all hits with an E-value &lt;= 10 are reported. The following options allow you to change the default E-value reporting thresholds, or to use bit score thresholds instead.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-E</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>In the per-target output, report target sequences with an E-value of &lt;= <em>&lt;x&gt;.</em> The default is 10.0, meaning that on average, about 10 false positives will be reported per query, so you can see the top of the noise and decide for yourself if it's really noise.</p>
  </dd>
  <dt>
    <p><strong>-T</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Instead of thresholding per-CM output on E-value, report target sequences with a bit score of &gt;= <em>&lt;x&gt;.</em></p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR INCLUSION THRESHOLDS</h2>
        <div class="sectioncontent">
<p>Inclusion thresholds are stricter than reporting thresholds. Inclusion thresholds control which hits are considered to be reliable enough to be included in a possible subsequent search round, or marked as significant ("!") as opposed to questionable ("?") in hit output.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--incE</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Use an E-value of &lt;= <em>&lt;x&gt;</em> as the hit inclusion threshold. The default is 0.01, meaning that on average, about 1 false positive would be expected in every 100 searches with different query sequences.</p>
  </dd>
  <dt>
    <p><strong>--incT</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Instead of using E-values for setting the inclusion threshold, instead use a bit score of &gt;= <em>&lt;x&gt;</em> as the hit inclusion threshold. By default this option is unset.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR MODEL-SPECIFIC SCORE THRESHOLDING</h2>
        <div class="sectioncontent">
<p>Curated CM databases may define specific bit score thresholds for each CM, superseding any thresholding based on statistical significance alone.</p><p>To use these options, the profile must contain the appropriate (GA, TC, and/or NC) optional score threshold annotation; this is picked up by <strong>cmbuild</strong> from Stockholm format alignment files. Each thresholding option has a score of <em>&lt;x&gt;</em> bits, and acts as if <strong>-T</strong><em> &lt;x&gt;</em> <strong>--incT</strong><em> &lt;x&gt;</em> has been applied specifically using each model's curated thresholds.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--cut_ga</strong></p>
  </dt>
  <dd>
    <p>Use the GA (gathering) bit scores in the model to set hit reporting and inclusion thresholds. GA thresholds are generally considered to be the reliable curated thresholds defining family membership; for example, in Rfam, these thresholds define what gets included in Rfam Full alignments based on searches with Rfam Seed models.</p>
  </dd>
  <dt>
    <p><strong>--cut_nc</strong></p>
  </dt>
  <dd>
    <p>Use the NC (noise cutoff) bit score thresholds in the model to set hit reporting and inclusion thresholds. NC thresholds are generally considered to be the score of the highest-scoring known false positive.</p>
  </dd>
  <dt>
    <p><strong>--cut_tc</strong></p>
  </dt>
  <dd>
    <p>Use the TC (trusted cutoff) bit score thresholds in the model to set hit reporting and inclusion thresholds. TC thresholds are generally considered to be the score of the lowest-scoring known true positive that is above all known false positives.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING THE ACCELERATION PIPELINE</h2>
        <div class="sectioncontent">
<p>Infernal 1.1 searches are accelerated in a six-stage filter pipeline. The first five stages use a profile HMM to define envelopes that are passed to the stage six CM CYK filter. Any envelopes that survive all filters are assigned final scores using the the CM Inside algorithm.</p><p>The profile HMM filter is built by the <strong>cmbuild</strong> program and is stored in <em>&lt;cmfile&gt;.</em></p><p>Each successive filter is slower than the previous one, but better than it at disciminating between subsequences that may contain high-scoring CM hits and those that do not. The first three HMM filter stages are the same as those used in HMMER3.  Stage 1 (F1) is the local HMM SSV filter modified for long sequences. Stage 2 (F2) is the local HMM Viterbi filter. Stage 3 (F3) is the local HMM Forward filter. Each of the first three stages uses the profile HMM in local mode, which allows a target subsequence to align to any region of the HMM. Stage 4 (F4) is a glocal HMM filter, which requires a target subsequence to align to the full-length profile HMM. Stage 5 (F5) is the glocal HMM envelope definition filter, which uses HMMER3's domain identification heursitics to define envelope boundaries. After each stage from 2 to 5 a bias filter step (F2b, F3b, F4b, and F5b) is used to remove sequences that appear to have passed the filter due to biased composition alone. Any envelopes that survive stages F1 through F5b are then passed with the local CM CYK filter. The CYK filter uses constraints (bands) derived from an HMM alignment of the envelope to reduce the number of required calculations and save time.  Any envelopes that pass CYK are scored with the local CM Inside algorithm, again using HMM bands for acceleration.</p><p>The default filter thresholds that define the minimum score required for a subsequence to survive each stage are defined based on the size of the search space (Z), which is defined as the length of the current query sequence times 2 (because both strands will be searched) times the number of profiles in <em>&lt;cmdb&gt;.</em> However, if either the <strong>-Z</strong><em> &lt;x&gt;</em> or <strong>--FZ</strong><em> &lt;x&gt;</em> options are used then the search space will be considered to be <em>&lt;x&gt;</em> for purposes of defining the filter thresholds.</p><p>For larger databases, the filters are more strict leading to more acceleration but potentially a greater loss of sensitivity. The rationale is that for larger databases, hits must have higher scores to achieve statistical significance, so stricter filtering that removes lower scoring insignificant hits is acceptable.</p><p>The P-value thresholds for all possible search space sizes and all filter stages are listed next. (A P-value threshold of 0.01 means that roughly 1% of the highest scoring nonhomologous subsequence are expected to pass the filter.) Z is defined as the number of nucleotides in the complete target sequence file times 2 because both strands will be searched with each model.</p><p>If Z is less than 2 Mb: F1 is 0.35; F2 and F2b are off; F3, F3b, F4, F4b and F5 are 0.02; F6 is 0.0001.</p><p>If Z is between 2 Mb and 20 Mb: F1 is 0.35; F2 and F2b are off; F3, F3b, F4, F4b and F5 are 0.005; F6 is 0.0001.</p><p>If Z is between 20 Mb and 200 Mb: F1 is 0.35; F2 and F2b are 0.15; F3, F3b, F4, F4b and F5 are 0.003; F6 is 0.0001.</p><p>If Z is between 200 Mb and 2 Gb: F1 is 0.15; F2 and F2b are 0.15; F3, F3b, F4, F4b, F5, and F5b are 0.0008; and F6 is 0.0001.</p><p>If Z is between 2 Gb and 20 Gb: F1 is 0.15; F2 and F2b are 0.15; F3, F3b, F4, F4b, F5, and F5b are 0.0002; and F6 is 0.0001.</p><p>If Z is more than 20 Gb: F1 is 0.06; F2 and F2b are 0.02; F3, F3b, F4, F4b, F5, and F5b are 0.0002; and F6 is 0.0001.</p><p>These thresholds were chosen based on performance on an internal benchmark testing many different possible settings.</p><p>There are five options for controlling the general filtering level. These options are, in order from least strict (slowest but most sensitive) to most strict (fastest but least sensitive): <strong>--max,</strong> <strong>--nohmm,</strong> <strong>--mid,</strong> <strong>--default,</strong> (this is the default setting) <strong>--rfam.</strong> and <strong>--hmmonly.</strong> With <strong>--default</strong> the filter thresholds will be database-size dependent. See the explanation of each of these individual options below for more information.</p><p>Additionally, an expert user can precisely control each filter stage score threshold with the <strong>--F1,</strong> <strong>--F1b,</strong> <strong>--F2,</strong> <strong>--F2b,</strong> <strong>--F3,</strong> <strong>--F3b,</strong> <strong>--F4,</strong> <strong>--F4b,</strong> <strong>--F5,</strong> <strong>--F5b,</strong> and <strong>--F6</strong> options. As well as turn each stage on or off with the <strong>--noF1,</strong> <strong>--doF1b,</strong> <strong>--noF2,</strong> <strong>--noF2b,</strong> <strong>--noF3,</strong> <strong>--noF3b,</strong> <strong>--noF4,</strong> <strong>--noF4b,</strong> <strong>--noF5,</strong> and <strong>--noF6.</strong> options. These options are only displayed if the <strong>--devhelp</strong> option is used to keep the number of displayed options with <strong>-h</strong> reasonable, and because they are only expected to be useful to a small minority of users.</p><p>As a special case, for any models in <em>&lt;cmfile&gt;</em> which have zero basepairs, profile HMM searches are run instead of CM searches. HMM algorithms are more efficient than CM algorithms, and the benefit of CM algorithms is lost for models with no secondary structure (zero basepairs). These profile HMM searches will run significantly faster than the CM searches. You can force HMM-only searches with the <strong>--hmmonly</strong> option. For more information on HMM-only searches see the user guide.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--max</strong></p>
  </dt>
  <dd>
    <p>Turn off all filters, and run non-banded Inside on every full-length target sequence. This increases sensitivity somewhat, at an extremely large cost in speed.</p>
  </dd>
  <dt>
    <p><strong>--nohmm</strong></p>
  </dt>
  <dd>
    <p>Turn off all HMM filter stages (F1 through F5b). The CYK filter, using QDBs, will be run on every full-length target sequence and will enforce a P-value threshold of 0.0001. Each subsequence that survives CYK will be passed to Inside, which will also use QDBs (but a looser set). This increases sensitivity somewhat, at a very large cost in speed.</p>
  </dd>
  <dt>
    <p><strong>--mid</strong></p>
  </dt>
  <dd>
    <p>Turn off the HMM SSV and Viterbi filter stages (F1 through F2b). Set remaining HMM filter thresholds (F3 through F5b) to 0.02 by default, but changeable to <em>&lt;x&gt;</em> with <strong>--Fmid</strong><em> &lt;x&gt;</em> sequence. This may increase sensitivity, at a significant cost in speed.</p>
  </dd>
  <dt>
    <p><strong>--default</strong></p>
  </dt>
  <dd>
    <p>Use the default filtering strategy. This option is on by default. The filter thresholds are determined based on the database size.</p>
  </dd>
  <dt>
    <p><strong>--rfam</strong></p>
  </dt>
  <dd>
    <p>Use a strict filtering strategy devised for large databases (more than 20 Gb). This will accelerate the search at a potential cost to sensitivity.</p>
  </dd>
  <dt>
    <p><strong>--hmmonly</strong></p>
  </dt>
  <dd>
    <p>Only use the filter profile HMM for searches, do not use the CM. Only filter stages F1 through F3 will be executed, using strict P-value thresholds (0.02 for F1, 0.001 for F2 and 0.00001 for F3). Additionally a bias composition filter is used after the F1 stage (with P=0.02 survival threshold). Any hit that survives all stages and has an HMM E-value or bit score above the reporting threshold will be output. The user can change the HMM-only filter thresholds and options with <strong>--hmmF1,</strong> <strong>--hmmF2,</strong> <strong>--hmmF3,</strong> <strong>--hmmnobias,</strong> <strong>--hmmnonull2,</strong> and <strong>--hmmmax.</strong> By default, searches for any model with zero basepairs will be run in HMM-only mode. This can be turned off, forcing CM searches for these models with the <strong>--nohmmonly</strong> option.</p>
  </dd>
  <dt>
    <p><strong>--FZ</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set filter thresholds as the defaults used if the database were <strong>&lt;x&gt;</strong> megabases (Mb). If used with <strong>&lt;x&gt;</strong> greater than 20000 (20 Gb) this option has the same effect as <strong>--rfam.</strong></p>
  </dd>
  <dt>
    <p><strong>--Fmid</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>With the <strong>--mid</strong> option set the HMM filter thresholds (F3 through F5b) to <em>&lt;x&gt;.</em> By default, <em>&lt;x&gt;</em> is 0.02.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--notrunc</strong></p>
  </dt>
  <dd>
    <p>Turn off truncated hit detection.</p>
  </dd>
  <dt>
    <p><strong>--anytrunc</strong></p>
  </dt>
  <dd>
    <p>Allow truncated hits to begin and end at any position in a target sequence. By default, 5' truncated hits must include the first residue of their target sequence and 3' truncated hits must include the final residue of their target sequence. With this option you may observe fewer full length hits that extend to the beginning and end of the query CM.</p>
  </dd>
  <dt>
    <p><strong>--nonull3</strong></p>
  </dt>
  <dd>
    <p>Turn off the null3 CM score corrections for biased composition. This correction is not used during the HMM filter stages.</p>
  </dd>
  <dt>
    <p><strong>--mxsize</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the maximum allowable CM DP matrix size to <em>&lt;x&gt;</em> megabytes. By default this size is 128 Mb. This should be large enough for the vast majority of searches, especially with smaller models. If <strong>cmsearch</strong> encounters an envelope in the CYK or Inside stage that requires a larger matrix, the envelope will be discounted from consideration. This behavior is like an additional filter that prevents expensive (slow) CM DP calculations, but at a potential cost to sensitivity. Note that if <strong>cmsearch</strong> is being run in <em>&lt;n&gt;</em> multiple threads on a multicore machine then each thread may have an allocated matrix of up to size <em>&lt;x&gt;</em> Mb at any given time.</p>
  </dd>
  <dt>
    <p><strong>--smxsize</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the maximum allowable CM search DP matrix size to <em>&lt;x&gt;</em> megabytes. By default this size is 128 Mb. This option is only relevant if the CM will not use HMM banded matrices, i.e. if the <strong>--max,</strong> <strong>--nohmm,</strong> <strong>--qdb,</strong> <strong>--fqdb,</strong> <strong>--nonbanded,</strong> or <strong>--fnonbanded</strong> options are also used. Note that if <strong>cmsearch</strong> is being run in <em>&lt;n&gt;</em> multiple threads on a multicore machine then each thread may have an allocated matrix of up to size <em>&lt;x&gt;</em> Mb at any given time.</p>
  </dd>
  <dt>
    <p><strong>--cyk</strong></p>
  </dt>
  <dd>
    <p>Use the CYK algorithm, not Inside, to determine the final score of all hits.</p>
  </dd>
  <dt>
    <p><strong>--acyk</strong></p>
  </dt>
  <dd>
    <p>Use the CYK algorithm to align hits. By default, the Durbin/Holmes optimal accuracy algorithm is used, which finds the alignment that maximizes the expected accuracy of all aligned residues.</p>
  </dd>
  <dt>
    <p><strong>--wcx</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>For each CM, set the W parameter, the expected maximum length of a hit, to <em>&lt;x&gt;</em> times the consensus length of the model. By default, the W parameter is read from the CM file and was calculated based on the transition probabilities of the model by <strong>cmbuild.</strong> You can find out what the default W is for a model using <strong>cmstat.</strong> This option should be used with caution as it impacts the filtering pipeline at several different stages in nonobvious ways. It is only recommended for expert users searching for hits that are much longer than any of the homologs used to build the model in <strong>cmbuild,</strong> e.g. ones with large introns or other large insertions. It cannot be used in combination with the <strong>--nohmm,</strong> <strong>--fqdb</strong> or <strong>--qdb</strong> options because in those cases W is limited by query-dependent bands.</p>
  </dd>
  <dt>
    <p><strong>--toponly</strong></p>
  </dt>
  <dd>
    <p>Only search the top (Watson) strand of target sequences in <em>&lt;seqfile&gt;.</em> By default, both strands are searched. This will halve the search space size (Z).</p>
  </dd>
  <dt>
    <p><strong>--bottomonly</strong></p>
  </dt>
  <dd>
    <p>Only search the bottom (Crick) strand of target sequences in <em>&lt;seqfile&gt;.</em> By default, both strands are searched. This will halve the search space size (Z).</p>
  </dd>
  <dt>
    <p><strong>--qformat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Assert that the query sequence database file is in format <em>&lt;s&gt;.</em> Accepted formats include <em>fasta,</em> <em>embl,</em> <em>genbank,</em> <em>ddbj,</em> <em>stockholm,</em> <em>pfam,</em> <em>a2m,</em> <em>afa,</em> <em>clustal,</em> and <em>phylip</em> The default is to autodetect the format of the file.</p>
  </dd>
  <dt>
    <p><strong>--glist</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Configure a subset of models from <em>&lt;cmfile&gt;</em> in glocal alignment mode, instead of local mode, namely the models listed in file <em>&lt;f&gt;.</em> Configure all other models (those not listed in <em>&lt;f&gt;)</em> in local mode. This option is incompatible with <em>-g.</em> File <em>&lt;f&gt;</em> must list valid names of models from <em>&lt;cmfile&gt;,</em> each separated by any whitespace character (e.g. a newline character).</p>
  </dd>
  <dt>
    <p><strong>--cpu</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the number of parallel worker threads to <em>&lt;n&gt;.</em> By default, Infernal sets this to the number of CPU cores it detects in your machine - that is, it tries to maximize the use of your available processor cores. Setting <em>&lt;n&gt;</em> higher than the number of available cores is of little if any value, but you may want to set it to something less. You can also control this number by setting an environment variable, <em>INFERNAL_NCPU.</em> This option is only available if Infernal was compiled with POSIX threads support. This is the default, but it may have been turned off at compile-time for your site or machine for some reason.</p>
  </dd>
  <dt>
    <p><strong>--stall</strong></p>
  </dt>
  <dd>
    <p>For debugging the MPI master/worker version: pause after start, to enable the developer to attach debuggers to the running master and worker(s) processes. Send SIGCONT signal to release the pause. (Under gdb: <em>(gdb) signal SIGCONT)</em> (Only available if optional MPI support was enabled at compile-time.)</p>
  </dd>
  <dt>
    <p><strong>--mpi</strong></p>
  </dt>
  <dd>
    <p>Run in MPI master/worker mode, using <em>mpirun.</em> (Only available if optional MPI support was enabled at compile-time.)</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO cmscan&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/infernal.1.html"><strong>infernal</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the Infernal package.</p><p>For complete documentation, see the user guide that came with your Infernal distribution (Userguide.pdf); or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2014 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your Infernal source distribution, or see the Infernal web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
The Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="cmpress.1.html"><span aria-hidden="true">&larr;</span> cmpress.1: Prepare a covariance model database for cmscan</a></li>
   <li class="next"><a href="cmsearch.1.html">cmsearch.1: Search covariance model(s) against a sequence database <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
