<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>convert_project: Convert assembly and sequencing file types</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Convert assembly and sequencing file types">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="convert_project (1) manual">
  <meta name="twitter:description" content="Convert assembly and sequencing file types">
  <meta name="twitter:image" content="https://www.carta.tech/images/mira-assembler-convert_project-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/convert_project.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="convert_project (1) manual" />
  <meta property="og:description" content="Convert assembly and sequencing file types" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/mira-assembler-convert_project-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">convert_project<small> (1)</small></h1>
        <p class="lead">Convert assembly and sequencing file types</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/convert_project.1.html">
      <span itemprop="name">convert_project: Convert assembly and sequencing file types</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/mira-assembler/">
      <span itemprop="name">mira-assembler</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/convert_project.1.html">
      <span itemprop="name">convert_project: Convert assembly and sequencing file types</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This program is part of the MIRA assembler package. It is used to convert project file types into other types.   Please check out the documentation below for more detailed information about convert_project.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    convert_project
  </dt>
  <dd>
    <p>[-f &lt;fromtype&gt;] [-t &lt;totype&gt; [-t &lt;totype&gt; ...]] [-aChimMsuZ] [-AcflnNoqrtvxXyz {...}] {infile} {outfile} [&lt;totype&gt; &lt;totype&gt; ...]</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-f</strong> &lt;fromtype&gt;</p>
  </dt>
  <dd>
    <p>load this type of project files, where fromtype is:</p>
  </dd>
  <dt>
    <p>caf</p>
  </dt>
  <dd>
    <p>a complete assembly or single sequences from CAF</p>
  </dd>
  <dt>
    <p>maf</p>
  </dt>
  <dd>
    <p>a complete assembly or single sequences from CAF</p>
  </dd>
  <dt>
    <p>fasta</p>
  </dt>
  <dd>
    <p>sequences from a FASTA file</p>
  </dd>
  <dt>
    <p>fastq</p>
  </dt>
  <dd>
    <p>sequences from a FASTQ file</p>
  </dd>
  <dt>
    <p>gbf</p>
  </dt>
  <dd>
    <p>sequences from a GBF file</p>
  </dd>
  <dt>
    <p>phd</p>
  </dt>
  <dd>
    <p>sequences from a PHD file</p>
  </dd>
  <dt>
    <p>fofnexp</p>
  </dt>
  <dd>
    <p>sequences in EXP files from file of filenames</p>
  </dd>
  <dt>
    <p><strong>-t</strong> &lt;totype&gt;</p>
  </dt>
  <dd>
    <p>write the sequences/assembly to this type (multiple mentions of <strong>-t</strong> are allowed):</p>
  </dd>
  <dt>
    <p>ace</p>
  </dt>
  <dd>
    <p>sequences or complete assembly to ACE</p>
  </dd>
  <dt>
    <p>caf</p>
  </dt>
  <dd>
    <p>sequences or complete assembly to CAF</p>
  </dd>
  <dt>
    <p>maf</p>
  </dt>
  <dd>
    <p>sequences or complete assembly to MAF</p>
  </dd>
  <dt>
    <p>sam</p>
  </dt>
  <dd>
    <p>complete assembly to SAM</p>
  </dd>
  <dt>
    <p>samnbb</p>
  </dt>
  <dd>
    <p>like above, but leaving out reference (backbones) in mapping assemblies</p>
  </dd>
  <dt>
    <p>gbf</p>
  </dt>
  <dd>
    <p>sequences or consensus to GBF</p>
  </dd>
  <dt>
    <p>gff3</p>
  </dt>
  <dd>
    <p>consensus to GFF3</p>
  </dd>
  <dt>
    <p>wig</p>
  </dt>
  <dd>
    <p>assembly coverage info to wiggle file</p>
  </dd>
  <dt>
    <p>gcwig</p>
  </dt>
  <dd>
    <p>assembly gc content info to wiggle file</p>
  </dd>
  <dt>
    <p>fasta</p>
  </dt>
  <dd>
    <p>sequences or consensus to FASTA file (qualities to</p><ul>
<li><p>.qual)</p></li>
</ul>
  </dd>
  <dt>
    <p>fastq</p>
  </dt>
  <dd>
    <p>sequences or consensus to FASTQ file</p>
  </dd>
  <dt>
    <p>exp</p>
  </dt>
  <dd>
    <p>sequences or complete assembly to EXP files in</p><ul>
<li><p>directories. Complete assemblies are suited for gap4 import as directed assembly. Note: using caf2gap to import into gap4 is recommended though</p></li>
</ul>
  </dd>
  <dt>
    <p>text</p>
  </dt>
  <dd>
    <p>complete assembly to text alignment (only when <strong>-f</strong> is</p><ul>
<li><p>caf, maf or gbf)</p></li>
</ul>
  </dd>
  <dt>
    <p>html</p>
  </dt>
  <dd>
    <p>complete assembly to HTML (only when <strong>-f</strong> is caf, maf or</p><ul>
<li><p>gbf)</p></li>
</ul>
  </dd>
  <dt>
    <p>tcs</p>
  </dt>
  <dd>
    <p>complete assembly to tcs</p>
  </dd>
  <dt>
    <p>hsnp</p>
  </dt>
  <dd>
    <p>surrounding of SNP tags (SROc, SAOc, SIOc) to HTML (only when <strong>-f</strong> is caf, maf or gbf)</p>
  </dd>
  <dt>
    <p>asnp</p>
  </dt>
  <dd>
    <p>analysis of SNP tags (only when <strong>-f</strong> is caf, maf or gbf)</p>
  </dd>
  <dt>
    <p>cstats</p>
  </dt>
  <dd>
    <p>contig statistics file like from MIRA (only when source contains contigs)</p>
  </dd>
  <dt>
    <p>crlist</p>
  </dt>
  <dd>
    <p>contig read list file like from MIRA (only when source contains contigs)</p>
  </dd>
  <dt>
    <p>maskedfasta</p>
  </dt>
  <dd>
    <p>reads where sequencing vector is masked out (with X) to FASTA file (qualities to .qual)</p>
  </dd>
  <dt>
    <p>scaf</p>
  </dt>
  <dd>
    <p>sequences or complete assembly to single sequences CAF</p>
  </dd>
  <dt>
    <p><strong>-a</strong></p>
  </dt>
  <dd>
    <p>Append to target files instead of rewriting</p>
  </dd>
  <dt>
    <p><strong>-A</strong> &lt;string&gt;</p>
  </dt>
  <dd>
    <p>String with MIRA parameters to be parsed Useful when setting parameters affecting consensus calling like <strong>-CO</strong>:mrpg etc. E.g.: <strong>-a</strong> "454_SETTINGS <strong>-CO</strong>:mrpg=3"</p>
  </dd>
  <dt>
    <p><strong>-b</strong></p>
  </dt>
  <dd>
    <p>Blind data Replaces all bases in reads/contigs with a 'c'</p>
  </dd>
  <dt>
    <p><strong>-C</strong></p>
  </dt>
  <dd>
    <p>Perform hard clip to reads When reading formats which define clipping points, will</p><ul>
<li><p>save only the unclipped part into the result file.</p></li><li><p>Applies only to files/formats which do not contain</p></li><li><p>contigs.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-d</strong></p>
  </dt>
  <dd>
    <p>Delete gap only columns When output is contigs: delete columns that are</p><ul>
<li><p>entirely gaps (like after having deleted reads during editing in gap4 or similar)</p></li><li><p>When output is reads: delete gaps in reads</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-F</strong></p>
  </dt>
  <dd>
    <p>Filter to read groups Special use case, do not use yet.</p>
  </dd>
  <dt>
    <p><strong>-m</strong></p>
  </dt>
  <dd>
    <p>Make contigs (only for <strong>-t</strong> = caf or maf) Encase single reads as contig singlets into the CAF/MAF file.</p>
  </dd>
  <dt>
    <p><strong>-n</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>when given, selects only reads or contigs given by name in that file.</p>
  </dd>
  <dt>
    <p><strong>-i</strong></p>
  </dt>
  <dd>
    <p>when <strong>-n</strong> is used, inverts the selection</p>
  </dd>
  <dt>
    <p><strong>-o</strong></p>
  </dt>
  <dd>
    <p>fastq quality Offset (only for <strong>-f</strong> = 'fastq') Offset of quality values in FASTQ file. Default of 0 tries to automatically recognise.</p>
  </dd>
  <dt>
    <p><strong>-Q</strong> &lt;quality&gt;</p>
  </dt>
  <dd>
    <p>Set default quality for bases in file types without quality values Furthermore, do not stop if expected quality files are missing (e.g. '.fasta')</p>
  </dd>
  <dt>
    <p><strong>-R</strong> &lt;name&gt;</p>
  </dt>
  <dd>
    <p>Rename contigs/singlets/reads with given name string to which a counter is appended. Known bug: will create duplicate names if input</p><ul>
<li><p>contains contigs/singlets as well as free reads, i.e. reads not in contigs nor singlets.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-S</strong> &lt;name&gt;</p>
  </dt>
  <dd>
    <p>(name)Scheme for renaming reads, important for paired-ends Only 'solexa' is currently supported.</p>
  </dd>

</dl>
<h3>The following switches work only when input (CAF or MAF) contains contigs.</h3>
<p>Beware: CAF and MAf can also contain just reads.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-M</strong></p>
  </dt>
  <dd>
    <p>Do not extract contigs (or their consensus), but the sequence of the reads they are composed of.</p>
  </dd>
  <dt>
    <p><strong>-N</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>like <strong>-n</strong>, but sorts output according to order given in file.</p>
  </dd>
  <dt>
    <p><strong>-r</strong> [cCqf]</p>
  </dt>
  <dd>
    <p>Recalculate consensus and / or consensus quality values and / or SNP feature tags. 'c' recalc cons & cons qualities (with IUPAC) 'C' recalc cons & cons qualities (forcing non-IUPAC) 'q' recalc consensus qualities only 'f' recalc SNP features Note: only the last of cCq is relevant, f works as a</p><ul>
<li><p>switch and can be combined with cQq (e.g. "-r C <strong>-r</strong> f")</p></li><li><p>Note: if the CAF/MAF contains multiple strains, recalculation of cons & cons qualities is forced, you</p></li><li><p>can just influence whether IUPACs are used or not.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-s</strong></p>
  </dt>
  <dd>
    <p>split output into multiple files instead of creating a single file</p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>'fillUp strain genomes' Fill holes in the genome of one strain (N or @) with sequence from a consensus of other strains Takes effect only with <strong>-r</strong> and <strong>-t</strong> gbf or fasta/q in FASTA/Q: bases filled up are in lower case in GBF: bases filled up are in upper case</p>
  </dd>
  <dt>
    <p><strong>-q</strong> &lt;integer&gt;</p>
  </dt>
  <dd>
    <p>Defines minimum quality a consensus base of a strain must have, consensus bases below this will be 'N' Default: 0 Only used with <strong>-r</strong>, and <strong>-f</strong> is caf/maf and <strong>-t</strong> is (fasta</p><ul>
<li><p>or gbf)</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-v</strong></p>
  </dt>
  <dd>
    <p>Print version number and exit</p>
  </dd>
  <dt>
    <p><strong>-x</strong> &lt;integer&gt;</p>
  </dt>
  <dd>
    <p>Minimum contig or read length When loading, discard all contigs / reads with a length less than this value. Default: 0 (=switched off) Note: not applied to reads in contigs!</p>
  </dd>
  <dt>
    <p><strong>-X</strong> &lt;integer&gt;</p>
  </dt>
  <dd>
    <p>Similar to <strong>-x</strong> but applies only to reads and then to the clipped length.</p>
  </dd>
  <dt>
    <p><strong>-y</strong> &lt;integer&gt;</p>
  </dt>
  <dd>
    <p>Minimum average contig coverage When loading, discard all contigs with an average coverage less than this value. Default: 1</p>
  </dd>
  <dt>
    <p><strong>-z</strong> &lt;integer&gt;</p>
  </dt>
  <dd>
    <p>Minimum number of reads in contig When loading, discard all contigs with a number of reads less than this value. Default: 0 (=switched off)</p>
  </dd>
  <dt>
    <p><strong>-l</strong> &lt;integer&gt;</p>
  </dt>
  <dd>
    <p>when output as text or HTML: number of bases shown in one alignment line. Default: 60.</p>
  </dd>
  <dt>
    <p><strong>-c</strong> &lt;character&gt;</p>
  </dt>
  <dd>
    <p>when output as text or HTML: character used to pad endgaps. Default: ' ' (blank)</p>
  </dd>

</dl>
<p>Aliases: caf2html, exp2fasta, ... etc. Any combination of "&lt;validfromtype&gt;2&lt;validtotype&gt;" can be used as program name (also using links) so as that convert_project automatically sets <strong>-f</strong> and <strong>-t</strong> accordingly.</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">
<ul>
<li><p>convert_project source.maf dest.sam</p></li><li><p>convert_project source.caf dest.fasta wig ace</p></li><li><p>convert_project -x 2000 -y 10 source.caf dest.caf</p></li><li><p>caf2html -l 100 -c . source.caf dest</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO convert_project&hellip;</h2>
        <div class="sectioncontent">
<p>A more extensive documentation is provided in the mira-doc package and can be found at /usr/share/doc/mira-assembler/DefinitiveGuideToMIRA.html.</p><p>You can also subscribe one of the MIRA mailing lists at</p><ul>
<li><p>http://www.chevreux.org/mira_mailinglists.html</p></li>
</ul><p>After subscribing, mail general questions to the MIRA talk mailing list:</p><ul>
<li><p>mira_talk@freelists.org</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>To report bugs or ask for features, please use the new ticketing system at:</p><ul>
<li><p>http://sourceforge.net/apps/trac/mira-assembler/</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>The author of the mira code is Bastien Chevreux &lt;bach@chevreux.org&gt;</p><p>This manual page was written by Andreas Tille &lt;tille@debian.org&gt; but can be freely used for any other distribution.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="convert_octree.1.html"><span aria-hidden="true">&larr;</span> convert_octree.1: Convert between octomap octree file formats</a></li>
   <li class="next"><a href="convert_seq.1.html">convert_seq.1: Conversion of sequence and alignment formats <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
