<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>emma: Multiple sequence alignment (clustalw wrapper)</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Multiple sequence alignment (clustalw wrapper)">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="emma (1e) manual">
  <meta name="twitter:description" content="Multiple sequence alignment (clustalw wrapper)">
  <meta name="twitter:image" content="https://www.carta.tech/images/emboss-emma-1e.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1e/emma.1e.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="emma (1e) manual" />
  <meta property="og:description" content="Multiple sequence alignment (clustalw wrapper)" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/emboss-emma-1e.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">emma<small> (1e)</small></h1>
        <p class="lead">Multiple sequence alignment (clustalw wrapper)</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1e/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1e/emma.1e.html">
      <span itemprop="name">emma: Multiple sequence alignment (clustalw wrapper)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/emboss/">
      <span itemprop="name">emboss</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1e/emma.1e.html">
      <span itemprop="name">emma: Multiple sequence alignment (clustalw wrapper)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>emma</strong> <strong>-sequence&nbsp;</strong><strong></strong><em>seqall</em> [<strong>-onlydend&nbsp;</strong><strong></strong><em>toggle</em>] <strong>-dend&nbsp;</strong><strong></strong><em>toggle</em> <strong>-dendfile&nbsp;</strong><strong></strong><em>infile</em> [<strong>-slow&nbsp;</strong><strong></strong><em>toggle</em>] <strong>-pwmatrix&nbsp;</strong><strong></strong><em>list</em> <strong>-pwdnamatrix&nbsp;</strong><strong></strong><em>list</em> <strong>-usermatrix&nbsp;</strong><strong></strong><em>variable</em> <strong>-pairwisedatafile&nbsp;</strong><strong></strong><em>infile</em> <strong>-matrix&nbsp;</strong><strong></strong><em>list</em> <strong>-usermamatrix&nbsp;</strong><strong></strong><em>variable</em> <strong>-dnamatrix&nbsp;</strong><strong></strong><em>list</em> <strong>-umamatrix&nbsp;</strong><strong></strong><em>variable</em> <strong>-mamatrixfile&nbsp;</strong><strong></strong><em>infile</em> <strong>-pwgapopen&nbsp;</strong><strong></strong><em>float</em> <strong>-pwgapextend&nbsp;</strong><strong></strong><em>float</em> <strong>-ktup&nbsp;</strong><strong></strong><em>integer</em> <strong>-gapw&nbsp;</strong><strong></strong><em>integer</em> <strong>-topdiags&nbsp;</strong><strong></strong><em>integer</em> <strong>-window&nbsp;</strong><strong></strong><em>integer</em> <strong>-nopercent&nbsp;</strong><strong></strong><em>boolean</em> [<strong>-gapopen&nbsp;</strong><strong></strong><em>float</em>] [<strong>-gapextend&nbsp;</strong><strong></strong><em>float</em>] [<strong>-endgaps&nbsp;</strong><strong></strong><em>boolean</em>] [<strong>-gapdist&nbsp;</strong><strong></strong><em>integer</em>] <strong>-norgap&nbsp;</strong><strong></strong><em>boolean</em> <strong>-hgapres&nbsp;</strong><strong></strong><em>string</em> <strong>-nohgap&nbsp;</strong><strong></strong><em>boolean</em> [<strong>-maxdiv&nbsp;</strong><strong></strong><em>integer</em>] <strong>-outseq&nbsp;</strong><strong></strong><em>seqoutset</em> <strong>-dendoutfile&nbsp;</strong><strong></strong><em>outfile</em> <strong>emma</strong> <strong>-help</strong></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>emma</strong> is a command line program from EMBOSS (&ldquo;the European Molecular Biology Open Software Suite&rdquo;). It is part of the "Alignment:Multiple" command group(s).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<h3>Input section</h3>
<p><strong>-sequence</strong> <em>seqall</em></p><p><strong>-onlydend</strong> <em>toggle</em></p><p>Default value: N</p><p><strong>-dend</strong> <em>toggle</em></p><p>Default value: N</p><p><strong>-dendfile</strong> <em>infile</em></p><p><strong>-slow</strong> <em>toggle</em></p><p>A distance is calculated between every pair of sequences and these are used to construct the dendrogram which guides the final multiple alignment. The scores are calculated from separate pairwise alignments. These can be calculated using 2 methods: dynamic programming (slow but accurate) or by the method of Wilbur and Lipman (extremely fast but approximate). The slow-accurate method is fine for short sequences but will be VERY SLOW for many (e.g. &gt;100) long (e.g. &gt;1000 residue) sequences. Default value: Y</p>
<h3>Pairwise align options</h3>
<p><strong>-pwmatrix</strong> <em>list</em></p><p>The scoring table which describes the similarity of each amino acid to each other. There are three &apos;in-built&apos; series of weight matrices offered. Each consists of several matrices which work differently at different evolutionary distances. To see the exact details, read the documentation. Crudely, we store several matrices in memory, spanning the full range of amino acid distance (from almost identical sequences to highly divergent ones). For very similar sequences, it is best to use a strict weight matrix which only gives a high score to identities and the most favoured conservative substitutions. For more divergent sequences, it is appropriate to use &apos;softer&apos; matrices which give a high score to many other frequent substitutions. 1) BLOSUM (Henikoff). These matrices appear to be the best available for carrying out data base similarity (homology searches). The matrices used are: Blosum80, 62, 45 and 30. 2) PAM (Dayhoff). These have been extremely widely used since the late &apos;70s. We use the PAM 120, 160, 250 and 350 matrices. 3) GONNET . These matrices were derived using almost the same procedure as the Dayhoff one (above) but are much more up to date and are based on a far larger data set. They appear to be more sensitive than the Dayhoff series. We use the GONNET 40, 80, 120, 160, 250 and 350 matrices. We also supply an identity matrix which gives a score of 1.0 to two identical amino acids and a score of zero otherwise. This matrix is not very useful. Default value: b</p><p><strong>-pwdnamatrix</strong> <em>list</em></p><p>The scoring table which describes the scores assigned to matches and mismatches (including IUB ambiguity codes). Default value: i</p><p><strong>-usermatrix</strong> <em>variable</em></p><p><strong>-pairwisedatafile</strong> <em>infile</em></p>
<h3>Matrix options</h3>
<p><strong>-matrix</strong> <em>list</em></p><p>This gives a menu where you are offered a choice of weight matrices. The default for proteins is the PAM series derived by Gonnet and colleagues. Note, a series is used! The actual matrix that is used depends on how similar the sequences to be aligned at this alignment step are. Different matrices work differently at each evolutionary distance. There are three &apos;in-built&apos; series of weight matrices offered. Each consists of several matrices which work differently at different evolutionary distances. To see the exact details, read the documentation. Crudely, we store several matrices in memory, spanning the full range of amino acid distance (from almost identical sequences to highly divergent ones). For very similar sequences, it is best to use a strict weight matrix which only gives a high score to identities and the most favoured conservative substitutions. For more divergent sequences, it is appropriate to use &apos;softer&apos; matrices which give a high score to many other frequent substitutions. 1) BLOSUM (Henikoff). These matrices appear to be the best available for carrying out data base similarity (homology searches). The matrices used are: Blosum80, 62, 45 and 30. 2) PAM (Dayhoff). These have been extremely widely used since the late &apos;70s. We use the PAM 120, 160, 250 and 350 matrices. 3) GONNET . These matrices were derived using almost the same procedure as the Dayhoff one (above) but are much more up to date and are based on a far larger data set. They appear to be more sensitive than the Dayhoff series. We use the GONNET 40, 80, 120, 160, 250 and 350 matrices. We also supply an identity matrix which gives a score of 1.0 to two identical amino acids and a score of zero otherwise. This matrix is not very useful. Alternatively, you can read in your own (just one matrix, not a series). Default value: b</p><p><strong>-usermamatrix</strong> <em>variable</em></p><p><strong>-dnamatrix</strong> <em>list</em></p><p>This gives a menu where a single matrix (not a series) can be selected. Default value: i</p><p><strong>-umamatrix</strong> <em>variable</em></p><p><strong>-mamatrixfile</strong> <em>infile</em></p>
<h3>Additional section</h3>

<h3>Slow align options</h3>
<p><strong>-pwgapopen</strong> <em>float</em></p><p>The penalty for opening a gap in the pairwise alignments. Default value: 10.0</p><p><strong>-pwgapextend</strong> <em>float</em></p><p>The penalty for extending a gap by 1 residue in the pairwise alignments. Default value: 0.1</p>
<h3>Fast align options</h3>
<p><strong>-ktup</strong> <em>integer</em></p><p>This is the size of exactly matching fragment that is used. INCREASE for speed (max= 2 for proteins; 4 for DNA), DECREASE for sensitivity. For longer sequences (e.g. &gt;1000 residues) you may need to increase the default. Default value: @($(acdprotein)?1:2)</p><p><strong>-gapw</strong> <em>integer</em></p><p>This is a penalty for each gap in the fast alignments. It has little affect on the speed or sensitivity except for extreme values. Default value: @($(acdprotein)?3:5)</p><p><strong>-topdiags</strong> <em>integer</em></p><p>The number of k-tuple matches on each diagonal (in an imaginary dot-matrix plot) is calculated. Only the best ones (with most matches) are used in the alignment. This parameter specifies how many. Decrease for speed; increase for sensitivity. Default value: @($(acdprotein)?5:4)</p><p><strong>-window</strong> <em>integer</em></p><p>This is the number of diagonals around each of the &apos;best&apos; diagonals that will be used. Decrease for speed; increase for sensitivity. Default value: @($(acdprotein)?5:4)</p><p><strong>-nopercent</strong> <em>boolean</em></p><p>Default value: N</p>
<h3>Gap options</h3>
<p><strong>-gapopen</strong> <em>float</em></p><p>The penalty for opening a gap in the alignment. Increasing the gap opening penalty will make gaps less frequent. Default value: 10.0</p><p><strong>-gapextend</strong> <em>float</em></p><p>The penalty for extending a gap by 1 residue. Increasing the gap extension penalty will make gaps shorter. Terminal gaps are not penalised. Default value: 5.0</p><p><strong>-endgaps</strong> <em>boolean</em></p><p>End gap separation: treats end gaps just like internal gaps for the purposes of avoiding gaps that are too close (set by &apos;gap separation distance&apos;). If you turn this off, end gaps will be ignored for this purpose. This is useful when you wish to align fragments where the end gaps are not biologically meaningful. Default value: Y</p><p><strong>-gapdist</strong> <em>integer</em></p><p>Gap separation distance: tries to decrease the chances of gaps being too close to each other. Gaps that are less than this distance apart are penalised more than other gaps. This does not prevent close gaps; it makes them less frequent, promoting a block-like appearance of the alignment. Default value: 8</p><p><strong>-norgap</strong> <em>boolean</em></p><p>Residue specific penalties: amino acid specific gap penalties that reduce or increase the gap opening penalties at each position in the alignment or sequence. As an example, positions that are rich in glycine are more likely to have an adjacent gap than positions that are rich in valine. Default value: N</p><p><strong>-hgapres</strong> <em>string</em></p><p>This is a set of the residues &apos;considered&apos; to be hydrophilic. It is used when introducing Hydrophilic gap penalties. Default value: GPSNDQEKR</p><p><strong>-nohgap</strong> <em>boolean</em></p><p>Hydrophilic gap penalties: used to increase the chances of a gap within a run (5 or more residues) of hydrophilic amino acids; these are likely to be loop or random coil regions where gaps are more common. The residues that are &apos;considered&apos; to be hydrophilic are set by &apos;-hgapres&apos;. Default value: N</p><p><strong>-maxdiv</strong> <em>integer</em></p><p>This switch, delays the alignment of the most distantly related sequences until after the most closely related sequences have been aligned. The setting shows the percent identity level required to delay the addition of a sequence; sequences that are less identical than this level to any other sequences will be aligned later. Default value: 30</p>
<h3>Output section</h3>
<p><strong>-outseq</strong> <em>seqoutset</em></p><p><strong>-dendoutfile</strong> <em>outfile</em></p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>Bugs can be reported to the Debian Bug Tracking system (http://bugs.debian.org/emboss), or directly to the EMBOSS developers (http://sourceforge.net/tracker/?group_id=93650&atid=605031).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO emma&hellip;</h2>
        <div class="sectioncontent">
<p>emma is fully documented via the <strong>tfm</strong>(1) system.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p><strong>Debian Med Packaging Team</strong> &lt;debian-med-packaging@lists.alioth.debian.org&gt;</p><p>Wrote the script used to autogenerate this manual page.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">
<p>This manual page was autogenerated from an Ajax Control Definition of the EMBOSS package. It can be redistributed under the same terms as EMBOSS itself.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="embossversion.1e.html"><span aria-hidden="true">&larr;</span> embossversion.1e: Reports the current emboss version number</a></li>
   <li class="next"><a href="emowse.1e.html">emowse.1e: Search protein sequences by digest fragment molecular weight <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
