<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>exact-tandems: Package for sequence alignment of multiple genomes</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Package for sequence alignment of multiple genomes">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="exact-tandems (1) manual">
  <meta name="twitter:description" content="Package for sequence alignment of multiple genomes">
  <meta name="twitter:image" content="https://www.carta.tech/images/mummer-exact-tandems-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/exact-tandems.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="exact-tandems (1) manual" />
  <meta property="og:description" content="Package for sequence alignment of multiple genomes" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/mummer-exact-tandems-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">exact-tandems<small> (1)</small></h1>
        <p class="lead">Package for sequence alignment of multiple genomes</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/exact-tandems.1.html">
      <span itemprop="name">exact-tandems: Package for sequence alignment of multiple genomes</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/mummer/">
      <span itemprop="name">mummer</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/exact-tandems.1.html">
      <span itemprop="name">exact-tandems: Package for sequence alignment of multiple genomes</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>mummer-annotate</strong> &lt;gapfile&gt;<em>&lt;datafile&gt;</em></p><p><strong>combineMUMs</strong> &lt;RefSequence&gt;<em>&lt;MatchSequences&gt;</em>&lt;GapsFile&gt;</p><p><strong>delta-filter</strong> [options]<em>&lt;deltafile&gt;</em></p><p><strong>dnadiff</strong> [options]<em>&lt;reference&gt;</em>&lt;query&gt; or [options]<em>-d</em>&lt;delta<em>file&gt;</em></p><p><strong>exact-tandems</strong> &lt;file&gt;<em>&lt;min-match-len&gt;</em></p><p><strong>gaps</strong></p><p><strong>mapview</strong> [options]<em>&lt;coords</em>file&gt;<em>[UTR</em>coords]<em>[CDS</em>coords]</p><p><strong>mgaps</strong> [-d<em>&lt;DiagDiff&gt;]</em>[-f<em>&lt;DiagFactor&gt;]</em>[-l<em>&lt;MatchLen&gt;]</em>[-s<em>&lt;MaxSeparation&gt;]</em></p><p><strong>mummer</strong> [<em>options</em>]<em>&lt;reference-file&gt;</em>&lt;query-files&gt;</p><p><strong>mummerplot</strong> [options]<em>&lt;match</em>file&gt;</p><p><strong>nucmer</strong> [options]<em>&lt;Reference&gt;</em>&lt;Query&gt;</p><p><strong>nucmer2xfig</strong></p><p><strong>promer</strong> [options]<em>&lt;Reference&gt;</em>&lt;Query&gt;</p><p><strong>repeat-match</strong> [options]<em>&lt;genome-file&gt;</em></p><p><strong>run-mummer1</strong> &lt;fasta<em>reference&gt;</em>&lt;fasta<em>query&gt;</em>&lt;prefix&gt;<em>[-r]</em></p><p><strong>run-mummer3</strong> &lt;fasta<em>reference&gt;</em>&lt;multi-fasta<em>query&gt;</em>&lt;prefix&gt;</p><p><strong>show-aligns</strong> [options]<em>&lt;deltafile&gt;</em>&lt;ref<em>ID&gt;</em>&lt;qry<em>ID&gt;</em></p><p>Input is the .delta output of either the "nucmer" or the "promer" program passed on the command line.</p><p>Output is to stdout, and consists of all the alignments between the query and reference sequences identified on the command line.</p><p>NOTE: No sorting is done by default, therefore the alignments will be ordered as found in the &lt;deltafile&gt; input.</p><p><strong>show-coords</strong> [options]<em>&lt;deltafile&gt;</em></p><p><strong>show-snps</strong> [options]<em>&lt;deltafile&gt;</em></p><p><strong>show-tiling</strong> [options]<em>&lt;deltafile&gt;</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p>All tools (exept for gaps) obey to the -h, --help, -V and --version options as one would expect. This help is excellent and makes these man pages basically obsolete.</p><p><strong>combineMUMs</strong> Combines MUMs in &lt;GapsFile&gt; by extending matches off ends and between MUMs.  &lt;RefSequence&gt; is a fasta file of the reference sequence.  &lt;MatchSequences&gt; is a multi-fasta file of the sequences matched against the reference</p>
<pre>
  -D      Only output to stdout the difference positions
          and characters
  -n      Allow matches only between nucleotides, i.e., ACGTs
  -N num  Break matches at &lt;num&gt; or more consecutive non-ACGTs
  -q tag  Used to label query match
  -r tag  Used to label reference match
  -S      Output all differences in strings
  -t      Label query matches with query fasta header
  -v num  Set verbose level for extra output
  -W file Reset the default output filename witherrors.gaps
  -x      Don't output .cover files
  -e      Set error-rate cutoff to e (e.g. 0.02 is two percent)
</pre>
<p><strong>dnadiff</strong> Run comparative analysis of two sequence sets using nucmer and its associated utilities with recommended parameters. See MUMmer documentation for a more detailed description of the output. Produces the following output files:</p>
<pre>
    .report  - Summary of alignments, differences and SNPs
    .delta   - Standard nucmer alignment output
    .1delta  - 1-to-1 alignment from delta-filter -1
    .mdelta  - M-to-M alignment from delta-filter -m
    .1coords - 1-to-1 coordinates from show-coords -THrcl .1delta
    .mcoords - M-to-M coordinates from show-coords -THrcl .mdelta
    .snps    - SNPs from show-snps -rlTHC .1delta
    .rdiff   - Classified ref breakpoints from show-diff -rH .mdelta
    .qdiff   - Classified qry breakpoints from show-diff -qH .mdelta
    .unref   - Unaligned reference IDs and lengths (if applicable)
    .unqry   - Unaligned query IDs and lengths (if applicable)
</pre>
<p>MANDATORY:</p>
<pre>
    reference       Set the input reference multi-FASTA filename
    query           Set the input query multi-FASTA filename
      or
    delta file      Unfiltered .delta alignment file from nucmer
</pre>
<p>OPTIONS:</p>
<pre>
    -d|delta        Provide precomputed delta file for analysis
    -h
    --help          Display help information and exit
    -p|prefix       Set the prefix of the output files (default "out")
    -V
    --version       Display the version information and exit
</pre>
<p><strong>delta-filter</strong></p>
<pre>
  -e float    For switches -g -r -q, keep repeats within e percent
              of the best LIS score [0, 100], no repeats by default
  -g          Global alignment using length*identity weighted LIS.
              For every reference-query pair, leave only the aligns
              which form the longest mutually consistent set
  -h          Display help information
  -i float    Set the minimum alignment identity [0, 100], default 0
  -l int      Set the minimum alignment length, default 0
  -q          Query alignment using length*identity weighted LIS.
              For each query, leave only the aligns which form the
              longest consistent set for the query
  -r          Reference alignment using length*identity weighted LIS.
              For each reference, leave only the aligns which form
              the longest consistent set for the reference
  -u float    Set the minimum alignment uniqueness, i.e. percent of
              the alignment matching to unique reference AND query
              sequence [0, 100], default 0
  -o float    Set the maximum alignment overlap for -r and -q options
              as a percent of the alignment length [0, 100], default 100
</pre>

<pre>
  Reads a delta alignment file from either nucmer or promer and
</pre>
<p>filters the alignments based on the command-line switches, leaving only the desired alignments which are output to stdout in the same delta format as the input. For multiple switches, order of operations is as follows: -i -l -u -q -r -g. If an alignment is excluded by a preceding operation, it will be ignored by the succeeding operations</p>
<pre>
  An important distinction between the -g option and the -r -q
</pre>
<p>options is that -g requires the alignments to be mutually consistent in their order, while the -r -q options are not required to be mutually consistent and therefore tolerate translocations, inversions, etc. Thus, -r provides a one-to-many, -q a many-to-one, -r -q a one-to-one local mapping, and -g a one-to-one global mapping of reference and query bases respectively.</p><p><strong>mapview</strong></p>
<pre>
  -h
</pre>

<pre>
  --help   Display help information and exit
</pre>

<pre>
  -m|mag   Set the magnification at which the figure is rendered,
           this is an option for fig2dev which is used to generate
           the PDF and PS files (default 1.0)
</pre>

<pre>
  -n|num   Set the number of output files used to partition the
           output, this is to avoid generating files that are too
           large to display (default 10)
</pre>

<pre>
  -p|prefix  Set the output file prefix
           (default "PROMER_graph or NUCMER_graph")
</pre>

<pre>
  -v
  --verbose  Verbose logging of the processed files
</pre>

<pre>
  -V
  --version  Display the version information and exit
</pre>

<pre>
  -x1 coord  Set the lower coordinate bound of the display
</pre>

<pre>
  -x2 coord  Set the upper coordinate bound of the display
</pre>

<pre>
  -g|ref     If the input file is provided by 'mgaps', set the
             reference sequence ID (as it appears in the first column
             of the UTR/CDS coords file)
</pre>

<pre>
  -I         Display the name of query sequences
</pre>

<pre>
  -Ir        Display the name of reference genes
</pre>
<p><strong>mummer</strong> Find and output (to stdout) the positions and length of all sufficiently long maximal matches of a substring in &lt;query-file&gt; and &lt;reference-file&gt;</p>
<pre>
  -mum           compute maximal matches that are unique in both sequences
  -mumcand       same as -mumreference
  -mumreference  compute maximal matches that are unique in
</pre>
<p> 		 the reference-sequence but not necessarily 		 in the query-sequence (default)</p>
<pre>
  -maxmatch      compute all maximal matches regardless of their uniqueness
  -n             match only the characters a, c, g, or t
                 they can be in upper or in lower case
  -l             set the minimum length of a match
                 if not set, the default value is 20
  -b             compute forward and reverse complement matches
  -r             only compute reverse complement matches
  -s             show the matching substrings
  -c             report the query-position of a reverse complement match
                 relative to the original query sequence
  -F             force 4 column output format regardless of the number of
                 reference sequence inputs
  -L             show the length of the query sequences on the header line
</pre>
<p><strong>nuncmer</strong></p>
<pre>
    nucmer generates nucleotide alignments between two mutli-FASTA input
    files. Two output files are generated. The .cluster output file lists
    clusters of matches between each sequence. The .delta file lists the
    distance between insertions and deletions that produce maximal scoring
    alignments between each sequence.
</pre>
<p><em>MANDATORY:</em></p>
<pre>
    Reference     Set the input reference multi-FASTA filename
    Query         Set the input query multi-FASTA filename
</pre>

<pre>
  --mum           Use anchor matches that are unique in both the reference
                  and query
  --mumcand       Same as --mumreference
  --mumreference  Use anchor matches that are unique in in the reference
                  but not necessarily unique in the query (default behavior)
  --maxmatch      Use all anchor matches regardless of their uniqueness
</pre>

<pre>
  -b|breaklen     Set the distance an alignment extension will attempt to
                  extend poor scoring regions before giving up (default 200)
  -c|mincluster   Sets the minimum length of a cluster of matches (default 65)
  --[no]delta     Toggle the creation of the delta file (default --delta)
  --depend        Print the dependency information and exit
  -d|diagfactor   Set the clustering diagonal difference separation factor
                  (default 0.12)
  --[no]extend    Toggle the cluster extension step (default --extend)
  -f
  --forward       Use only the forward strand of the Query sequences
  -g|maxgap       Set the maximum gap between two adjacent matches in a
                  cluster (default 90)
  -h
  --help          Display help information and exit
  -l|minmatch     Set the minimum length of a single match (default 20)
  -o
  --coords        Automatically generate the original NUCmer1.1 coords
                  output file using the 'show-coords' program
  --[no]optimize  Toggle alignment score optimization, i.e. if an alignment
                  extension reaches the end of a sequence, it will backtrack
                  to optimize the alignment score instead of terminating the
                  alignment at the end of the sequence (default --optimize)
  -p|prefix       Set the prefix of the output files (default "out")
  -r
  --reverse       Use only the reverse complement of the Query sequences
  --[no]simplify  Simplify alignments by removing shadowed clusters. Turn
                  this option off if aligning a sequence to itself to look
                  for repeats (default --simplify)
</pre>
<p><strong>promer</strong></p>
<pre>
    promer generates amino acid alignments between two mutli-FASTA DNA input
    files. Two output files are generated. The .cluster output file lists
    clusters of matches between each sequence. The .delta file lists the
    distance between insertions and deletions that produce maximal scoring
    alignments between each sequence. The DNA input is translated into all 6
    reading frames in order to generate the output, but the output coordinates
    reference the original DNA input.
</pre>
<p><em>MANDATORY:</em></p>
<pre>
    Reference     Set the input reference multi-FASTA DNA file
    Query         Set the input query multi-FASTA DNA file
</pre>

<pre>
  --mum           Use anchor matches that are unique in both the reference
                  and query
  --mumcand       Same as --mumreference
  --mumreference  Use anchor matches that are unique in in the reference
                  but not necessarily unique in the query (default behavior)
  --maxmatch      Use all anchor matches regardless of their uniqueness
</pre>

<pre>
  -b|breaklen     Set the distance an alignment extension will attempt to
                  extend poor scoring regions before giving up, measured in
                  amino acids (default 60)
  -c|mincluster   Sets the minimum length of a cluster of matches, measured in
                  amino acids (default 20)
  --[no]delta     Toggle the creation of the delta file (default --delta)
  --depend        Print the dependency information and exit
  -d|diagfactor   Set the clustering diagonal difference separation factor
                  (default .11)
  --[no]extend    Toggle the cluster extension step (default --extend)
  -g|maxgap       Set the maximum gap between two adjacent matches in a
                  cluster, measured in amino acids (default 30)
  -l|minmatch     Set the minimum length of a single match, measured in amino
                  acids (default 6)
  -m|masklen      Set the maximum bookend masking lenth, measured in amino
                  acids (default 8)
  -o
  --coords        Automatically generate the original PROmer1.1 ".coords"
                  output file using the "show-coords" program
  --[no]optimize  Toggle alignment score optimization, i.e. if an alignment
                  extension reaches the end of a sequence, it will backtrack
                  to optimize the alignment score instead of terminating the
                  alignment at the end of the sequence (default --optimize)
</pre>

<pre>
  -p|prefix       Set the prefix of the output files (default "out")
  -x|matrix       Set the alignment matrix number to 1 [BLOSUM 45],
                  2 [BLOSUM 62] or 3 [BLOSUM 80] (default 2)
</pre>
<p><strong>repeat-match</strong> Find all maximal exact matches in &lt;genome-file&gt;</p>
<pre>
  -E    Use exhaustive (slow) search to find matches
  -f    Forward strand only, don't use reverse complement
  -n #  Set minimum exact match length to #
  -t    Only output tandem repeats
  -V #  Set level of verbose (debugging) printing to #
</pre>
<p><strong>show-aligns</strong></p>
<pre>
  -h      Display help information
  -q      Sort alignments by the query start coordinate
  -r      Sort alignments by the reference start coordinate
  -w int  Set the screen width - default is 60
  -x int  Set the matrix type - default is 2 (BLOSUM 62),
          other options include 1 (BLOSUM 45) and 3 (BLOSUM 80)
          note: only has effect on amino acid alignments
</pre>
<p><strong>show-coords</strong></p>
<pre>
  -b          Merges overlapping alignments regardless of match dir
              or frame and does not display any idenitity information.
  -B          Switch output to btab format
  -c          Include percent coverage information in the output
  -d          Display the alignment direction in the additional
              FRM columns (default for promer)
  -g          Deprecated option. Please use 'delta-filter' instead
  -h          Display help information
  -H          Do not print the output header
  -I float    Set minimum percent identity to display
  -k          Knockout (do not display) alignments that overlap
              another alignment in a different frame by more than 50%
              of their length, AND have a smaller percent similarity
              or are less than 75% of the size of the other alignment
              (promer only)
  -l          Include the sequence length information in the output
  -L long     Set minimum alignment length to display
  -o          Annotate maximal alignments between two sequences, i.e.
              overlaps between reference and query sequences
  -q          Sort output lines by query IDs and coordinates
  -r          Sort output lines by reference IDs and coordinates
  -T          Switch output to tab-delimited format
</pre>

<pre>
  Input is the .delta output of either the "nucmer" or the
</pre>
<p>"promer" program passed on the command line.</p>
<pre>
  Output is to stdout, and consists of a list of coordinates,
</pre>
<p>percent identity, and other useful information regarding the alignment data contained in the .delta file used as input.</p>
<pre>
  NOTE: No sorting is done by default, therefore the alignments
</pre>
<p>will be ordered as found in the &lt;deltafile&gt; input.</p><p><strong>show-snps</strong></p>
<pre>
  -C            Do not report SNPs from alignments with an ambiguous
                mapping, i.e. only report SNPs where the [R] and [Q]
                columns equal 0 and do not output these columns
  -h            Display help information
  -H            Do not print the output header
  -I            Do not report indels
  -l            Include sequence length information in the output
  -q            Sort output lines by query IDs and SNP positions
  -r            Sort output lines by reference IDs and SNP positions
  -S            Specify which alignments to report by passing
                'show-coords' lines to stdin
  -T            Switch to tab-delimited format
  -x int        Include x characters of surrounding SNP context in the
                output, default 0
</pre>

<pre>
  Input is the .delta output of either the nucmer or promer program
</pre>
<p>passed on the command line.</p>
<pre>
  Output is to stdout, and consists of a list of SNPs (or amino acid
</pre>
<p>substitutions for promer) with positions and other useful info. Output will be sorted with -r by default and the [BUFF] column will always refer to the sequence whose positions have been sorted. This value specifies the distance from this SNP to the nearest mismatch (end of alignment, indel, SNP, etc) in the same alignment, while the [DIST] column specifies the distance from this SNP to the nearest sequence end. SNPs for which the [R] and [Q] columns are greater than 0 should be evaluated with caution, as these columns specify the number of other alignments which overlap this position. Use -C to assure SNPs are only reported from unique alignment regions.</p><p><strong>show-tiling</strong></p>
<pre>
  -a          Describe the tiling path by printing the tab-delimited
              alignment region coordinates to stdout
  -c          Assume the reference sequences are circular, and allow
              tiled contigs to span the origin
  -g int      Set maximum gap between clustered alignments [-1, INT_MAX]
              A value of -1 will represent infinity
              (nucmer default = 1000)
              (promer default = -1)
  -i float    Set minimum percent identity to tile [0.0, 100.0]
              (nucmer default = 90.0)
              (promer default = 55.0)
  -l int      Set minimum length contig to report [-1, INT_MAX]
              A value of -1 will represent infinity
              (common default = 1)
  -p file     Output a pseudo molecule of the query contigs to 'file'
  -R          Deal with repetitive contigs by randomly placing them
              in one of their copy locations (implies -V 0)
  -t file     Output a TIGR style contig list of each query sequence
              that sufficiently matches the reference (non-circular)
  -u file     Output the tab-delimited alignment region coordinates
              of the unusable contigs to 'file'
  -v float    Set minimum contig coverage to tile [0.0, 100.0]
              (nucmer default = 95.0) sum of individual alignments
              (promer default = 50.0) extent of syntenic region
  -V float    Set minimum contig coverage difference [0.0, 100.0]
              i.e. the difference needed to determine one alignment
              is 'better' than another alignment
              (nucmer default = 10.0) sum of individual alignments
              (promer default = 30.0) extent of syntenic region
  -x          Describe the tiling path by printing the XML contig
              linking information to stdout
</pre>

<pre>
  Input is the .delta output of the nucmer program, run on very
</pre>
<p>similar sequence data, or the .delta output of the promer program, run on divergent sequence data.</p>
<pre>
  Output is to stdout, and consists of the predicted location of
</pre>
<p>each aligning query contig as mapped to the reference sequences. These coordinates reference the extent of the entire query contig, even when only a certain percentage of the contig was actually aligned (unless the -a option is used). Columns are, start in ref, end in ref, distance to next contig, length of this contig, alignment coverage, identity, orientation, and ID respectively.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO exact-tandems&hellip;</h2>
        <div class="sectioncontent">
<p><strong>http://mummer.sourceforge.net/</strong></p><p>Open source MUMmer 3.0 is described in</p><p><em>Versatile and open software for comparing large genomes.</em> S. Kurtz, A. Phillippy, A.L. Delcher, M. Smoot, M. Shumway, C. Antonescu, and S.L. Salzberg, Genome Biology (2004), 5:R12.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>mummer was written by S. Kurtz, A. Phillippy, A.L. Delcher, M. Smoot, M. Shumway, C. Antonescu, and S.L. Salzberg.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="ex4_initiator.1.html"><span aria-hidden="true">&larr;</span> ex4_initiator.1: Generic ctn manual page</a></li>
   <li class="next"><a href="exaile.1.html">exaile.1: Music manager and player <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
