<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>fa2htgs: Formatter for high throughput genome sequencing project submissions</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Formatter for high throughput genome sequencing project submissions">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="fa2htgs (1) manual">
  <meta name="twitter:description" content="Formatter for high throughput genome sequencing project submissions">
  <meta name="twitter:image" content="https://www.carta.tech/images/ncbi-tools-bin-fa2htgs-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/fa2htgs.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="fa2htgs (1) manual" />
  <meta property="og:description" content="Formatter for high throughput genome sequencing project submissions" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/ncbi-tools-bin-fa2htgs-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">fa2htgs<small> (1)</small></h1>
        <p class="lead">Formatter for high throughput genome sequencing project submissions</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/fa2htgs.1.html">
      <span itemprop="name">fa2htgs: Formatter for high throughput genome sequencing project submissions</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/ncbi-tools-bin/">
      <span itemprop="name">ncbi-tools-bin</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/fa2htgs.1.html">
      <span itemprop="name">fa2htgs: Formatter for high throughput genome sequencing project submissions</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>fa2htgs</strong> [\|<strong>-</strong>\|] [\|<strong>-6</strong>&nbsp;<em>str</em>\|] [\|<strong>-7</strong>&nbsp;<em>str</em>\|] [\|<strong>-A</strong>&nbsp;<em>filename</em>\|] [\|<strong>-C</strong>&nbsp;<em>str</em>\|] [\|<strong>-D</strong>\|] [\|<strong>-L</strong>&nbsp;<em>filename</em>\|] [\|<strong>-M</strong>&nbsp;<em>str</em>\|] [\|<strong>-N</strong>\|] [\|<strong>-O</strong>&nbsp;<em>filename</em>\|] [\|<strong>-P</strong>&nbsp;<em>str</em>\|] [\|<strong>-Q</strong>&nbsp;<em>filename</em>\|] [\|<strong>-S</strong>&nbsp;<em>str</em>\|] [\|<strong>-T</strong>&nbsp;<em>filename</em>\|] [\|<strong>-X</strong>\|] [\|<strong>-a</strong>&nbsp;<em>str</em>\|] [\|<strong>-b</strong>&nbsp;<em>N</em>\|] [\|<strong>-c</strong>&nbsp;<em>str</em>\|] [\|<strong>-d</strong>&nbsp;<em>str</em>\|] [\|<strong>-e</strong>&nbsp;<em>filename</em>\|] [\|<strong>-f</strong>\|] <strong>-g</strong>&nbsp;<em>str</em> [\|<strong>-h</strong>&nbsp;<em>str</em>\|] [\|<strong>-i</strong>&nbsp;<em>filename</em>\|] [\|<strong>-k</strong>&nbsp;<em>str</em>\|] [\|<strong>-l</strong>&nbsp;<em>N</em>\|] [\|<strong>-m</strong>\|] [\|<strong>-n</strong>&nbsp;<em>str</em>\|] [\|<strong>-o</strong>&nbsp;<em>filename</em>\|] [\|<strong>-p</strong>&nbsp;<em>N</em>\|] [\|<strong>-q</strong>\|] [\|<strong>-r</strong>&nbsp;<em>str</em>\|] <strong>-s</strong>&nbsp;<em>str</em> [\|<strong>-t</strong>&nbsp;<em>filename</em>\|] [\|<strong>-u</strong>\|] [\|<strong>-v</strong>\|] [\|<strong>-w</strong>\|] [\|<strong>-x</strong>&nbsp;<em>str</em>\|]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>fa2htgs</strong> is a program used to generate Seq-submits (an ASN.1 sequence submission file) for high throughput genome sequencing projects.</p><p><strong>fa2htgs</strong> will read a FASTA file (or an Ace Contig file with Phrap sequence quality values), a Sequin submission template file, (to get contact and citation information for the submission), and a series of command line arguments (see below).  This program will then combines these information to make a submission suitable for GenBank. Once you have generated your submission file, you need to follow the submission protocol (see the README present on your FTP account or mailed out to your Center).</p><p><strong>fa2htgs</strong> is intended for the automation by scripts for bulk submission of unannotated genome sequence. It can easily be extended from its current simple form to allow more complicated processing.  A submission prepared with <strong>fa2htgs</strong> can also be read into <a href="../man1/Psequin.1.html"><strong>Psequin</strong>(1)</a>, and then annotated more extensively.</p><p>Questions and concerns about this processing protocol, or how to use this tool should be forwarded to &lt;htgs@ncbi.nlm.nih.gov&gt;.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p>A summary of options is included below.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-</strong></p>
  </dt>
  <dd>
    <p>Print usage message</p>
  </dd>
  <dt>
    <p><strong>-6</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>SP6 clone (e.g., Contig1,left)</p>
  </dd>
  <dt>
    <p><strong>-7</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>T7 clone (e.g., Contig2,right)</p>
  </dd>
  <dt>
    <p><strong>-A</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Filename for accession list input (mutually exclusive with <strong>-T</strong> and <strong>-i</strong>).  The input file contains a tab-delimited table with three to five columns, which are accession number, start position, stop position, and (optionally) length and strand.  If start &gt; stop, the minus strand on the referenced accession is used.  A gap is indicated by the word "gap" instead of an accession, 0 for the start and stop positions, and a number for the length.</p>
  </dd>
  <dt>
    <p><strong>-C</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Clone library name (will appear as <strong>/clone-lib="</strong><em>str</em><strong>"</strong> on the source feature)</p>
  </dd>
  <dt>
    <p><strong>-D</strong></p>
  </dt>
  <dd>
    <p>HTGS_DRAFT sequence</p>
  </dd>
  <dt>
    <p><strong>-L</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Read phrap contig order from <em>filename</em>.  This is a tab-delimited file that can be used to drive the order of contigs (normally specified by <strong>-P</strong>), as well as indicating the SP6 and T7 ends.  It can also be used when contigs are known to be in opposite orientation. For example:</p>
<pre>
    Contig2     +       1       SP6     left
    Contig3     +       1
    Contig1     -               T7      right

</pre>
<p>The first column is the contig name, the second is the orientation, the third is the fragment_group, the fourth indicates the SP6 or T7 end, and the fifth says which side of SP6 or T7 end had vector removed.</p>
  </dd>
  <dt>
    <p><strong>-M</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Map name (will appear as <strong>/map="</strong><em>str</em><strong>"</strong> on the source feature)</p>
  </dd>
  <dt>
    <p><strong>-N</strong></p>
  </dt>
  <dd>
    <p>Annotate assembly_fragments</p>
  </dd>
  <dt>
    <p><strong>-O</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Read comment from <em>filename</em> (100-character-per-line maximum; <strong>~</strong> is a linebreak and <strong>`~</strong> is a literal <strong>~</strong>.  You can check the format with <strong>PSequin</strong>(1).)</p>
  </dd>
  <dt>
    <p><strong>-P</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Contigs to use, separated by commas.  If <strong>-P</strong> is not indicated with the <strong>-T</strong> option, then the fragments will go in in the order that they are in the ace file (which is appropriate for a phase 1 record, but not for a phase 2 or 3).  If you need to set the order of the segments of the ace file, you need to set it with the <strong>-P</strong> flag, like this: <strong>-P "Contig1,Contig4,Contig3,Contig2,Contig5"</strong></p>
  </dd>
  <dt>
    <p><strong>-Q</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Read quality scores from <em>filename</em></p>
  </dd>
  <dt>
    <p><strong>-S</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Strain name</p>
  </dd>
  <dt>
    <p><strong>-T</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Filename for phrap input (mutually exclusive with <strong>-A</strong> and <strong>-i</strong>)</p>
  </dd>
  <dt>
    <p><strong>-X</strong></p>
  </dt>
  <dd>
    <p>The coordinates in the input file are on the resulting segmented sequence.  (Bases 1 through <em>n</em> of each accession are used.) Otherwise, the coordinates are on the individual accessions, which need not start at base 1 of the record.</p>
  </dd>
  <dt>
    <p><strong>-a</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>GenBank accession; use if and only if updating a sequence.</p>
  </dd>
  <dt>
    <p><strong>-b</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>Gap length (default = 100; anything from 0 to 1000000000 is legal)</p>
  </dd>
  <dt>
    <p><strong>-c</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Clone name (will appear as <strong>/clone</strong> in the source feature; can be the same as <strong>-s</strong>)</p>
  </dd>
  <dt>
    <p><strong>-d</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Title for sequence (will appear in GenBank <strong>DEFINITION</strong> line)</p>
  </dd>
  <dt>
    <p><strong>-e</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Log errors to <em>filename</em></p>
  </dd>
  <dt>
    <p><strong>-f</strong></p>
  </dt>
  <dd>
    <p>htgs_fulltop keyword</p>
  </dd>
  <dt>
    <p><strong>-g</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Genome Center tag (probably the same as your login name on the NCBI FTP server)</p>
  </dd>
  <dt>
    <p><strong>-h</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Chromosome (will appear as <strong>/chromosome</strong> in the source feature)</p>
  </dd>
  <dt>
    <p><strong>-i</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Filename for fasta input (default is stdin; mutually exclusive with <strong>-A</strong> and <strong>-T</strong>)</p>
  </dd>
  <dt>
    <p><strong>-k</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Add the supplied string as a keyword.</p>
  </dd>
  <dt>
    <p><strong>-l</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>Length of sequence in bp (default = 0). The length is checked against the actual number of bases we get. For phase 1 and 2 sequence it is also used to estimate gap lengths. For phase 1 and 2 records, it is important to use a number GREATER than the amount of provided nucleotide, otherwise this will generate false `gaps'.  Here is assumed that the putative full length of the BAC or cosmid will be used.  There should be at least 20 to 30 `n' in between the segments (you can check for these in Sequin), as this will ensure proper behavior when this sequence is used with BLAST.  Otherwise `artifactual' unrelated segment neighbors may be brought into proximity of each other.</p>
  </dd>
  <dt>
    <p><strong>-m</strong></p>
  </dt>
  <dd>
    <p>Take comment from template</p>
  </dd>
  <dt>
    <p><strong>-n</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Organism name (default = Homo sapiens)</p>
  </dd>
  <dt>
    <p><strong>-o</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Filename for asn.1 output (default = stdout)</p>
  </dd>
  <dt>
    <p><strong>-p</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>HTGS phase:</p>
<dl class='dl-vertical'>
  <dt>
    1
  </dt>
  <dd>
    <p>A collection of unordered contigs with gaps of unknown length.  A Phase 1 record must at the very least have two segments with one gap. (default)</p>
  </dd>
  <dt>
    2
  </dt>
  <dd>
    <p>A series of ordered contigs, possibly with known gap lengths.  This could be a single sequence without gaps, if the sequence has ambiguities to resolve.</p>
  </dd>
  <dt>
    3
  </dt>
  <dd>
    <p>A single contiguous sequence.  This sequence is finished, but not necessarily annotated.</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-q</strong></p>
  </dt>
  <dd>
    <p>htgs_cancelled keyword</p>
  </dd>
  <dt>
    <p><strong>-r</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Remark for update (brief comment describing the nature of the update, such as "new sequence", "new citation", or "updated features")</p>
  </dd>
  <dt>
    <p><strong>-s</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Sequence name.  The sequence must have a name that is unique within the genome center. We use the combination of the genome center name (<strong>-g</strong> argument) and the sequence name (<strong>-s</strong>) to track this sequence and to talk to you about it.  The name can have any form you like but must be unique within your center.</p>
  </dd>
  <dt>
    <p><strong>-t</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Filename for Seq-submit template (default = template.sub)</p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Take biosource from template</p>
  </dd>
  <dt>
    <p><strong>-v</strong></p>
  </dt>
  <dd>
    <p>htgs_activefin keyword</p>
  </dd>
  <dt>
    <p><strong>-w</strong></p>
  </dt>
  <dd>
    <p>Whole Genome Shotgun flag</p>
  </dd>
  <dt>
    <p><strong>-x</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Secondary accession numbers, separated by commas, s.t. U10000,L11000.</p>
  </dd>

</dl>
<p>In some cases a large segment will supersede another or group of other accession numbers (records).  These records which are no longer wanted in GenBank should be made secondary. Using the <strong>-x</strong> argument you can list the Accession Numbers you want to make secondary.  This will instruct us to remove the accession number(s) from GenBank, and will no longer be part of the GenBank release. They will nonetheless be available from Entrez.</p><p><strong>GREAT CARE</strong> should be taken when using this argument!!!  Improper use of accession numbers here will result in the inappropriate withdrawal of GenBank records from GenBank, EMBL and DDBJ.  We provide this parameter as a convenience to submitting centers, but this may need to be removed if it is not used carefully.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>The National Center for Biotechnology Information.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO fa2htgs&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man1/Psequin.1.html"><strong>Psequin</strong>(1)</a>, /usr/share/doc/ncbi-tools-bin/README.fa2htgs.gz</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="fa2dna.1.html"><span aria-hidden="true">&larr;</span> fa2dna.1: Format fasta database for use with anfo</a></li>
   <li class="next"><a href="faad.1.html">faad.1: Faad  process an advanced audio codec stream <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
