<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>fasttreeMP: Create phylogenetic trees from alignments of nucleotide or protein sequences (openmp version)</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Create phylogenetic trees from alignments of nucleotide or protein sequences (openmp version)">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="fasttreeMP (1) manual">
  <meta name="twitter:description" content="Create phylogenetic trees from alignments of nucleotide or protein sequences (openmp version)">
  <meta name="twitter:image" content="https://www.carta.tech/images/fasttree-fasttreeMP-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/fasttreeMP.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="fasttreeMP (1) manual" />
  <meta property="og:description" content="Create phylogenetic trees from alignments of nucleotide or protein sequences (openmp version)" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/fasttree-fasttreeMP-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">fasttreeMP<small> (1)</small></h1>
        <p class="lead">Create phylogenetic trees from alignments of nucleotide or protein sequences (openmp version)</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/fasttreeMP.1.html">
      <span itemprop="name">fasttreeMP: Create phylogenetic trees from alignments of nucleotide or protein sequences (openmp version)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/fasttree/">
      <span itemprop="name">fasttree</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/fasttreeMP.1.html">
      <span itemprop="name">fasttreeMP: Create phylogenetic trees from alignments of nucleotide or protein sequences (openmp version)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>fasttreeMP infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences. It handles alignments with up to a million of sequences in a reasonable amount of time and memory.</p><p>fasttreeMP is more accurate than PhyML 3 with default settings, and much more accurate than the distance-matrix methods that are traditionally used for large alignments. fasttreeMP uses the Jukes-Cantor or generalized time-reversible (GTR) models of nucleotide evolution and the JTT (Jones-Taylor-Thornton 1992) model of amino acid evolution. To account for the varying rates of evolution across sites, fasttreeMP uses a single rate for each site (the "CAT" approximation). To quickly estimate the reliability of each split in the tree, fasttreeMP computes local support values with the Shimodaira-Hasegawa test (these are the same as PhyML 3's "SH-like local supports").</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>fasttreeMP protein_alignment &gt; tree</strong></p><p><strong>fasttreeMP </strong><strong>-nt</strong> nucleotide_alignment &gt; tree</p><p><strong>fasttreeMP </strong><strong>-nt</strong> <strong>-gtr</strong> &lt; nucleotide_alignment &gt; tree</p><p>fasttreeMP accepts alignments in fasta or phylip interleaved formats</p><h3>Common options (must be before the alignment file):</h3>
<p><strong>-quiet</strong> to suppress reporting information <strong>-nopr</strong> to suppress progress indicator <strong>-log</strong> logfile <strong>--</strong> save intermediate trees, settings, and model details <strong>-fastest</strong> <strong>--</strong> speed up the neighbor joining phase & reduce memory usage</p><ul>
<li><p>(recommended for &gt;50,000 sequences) <strong>-n</strong> &lt;number&gt; to analyze multiple alignments (phylip format only)</p></li><li><p>(use for global bootstrap, with seqboot and CompareToBootstrap.pl) <strong>-nosupport</strong> to not compute support values <strong>-intree</strong> newick_file to set the starting tree(s) <strong>-intree1</strong> newick_file to use this starting tree for all the alignments</p></li><li><p>(for faster global bootstrap on huge alignments) <strong>-pseudo</strong> to use pseudocounts (recommended for highly gapped sequences) <strong>-gtr</strong> <strong>--</strong> generalized time-reversible model (nucleotide alignments only) <strong>-wag</strong> <strong>--</strong> Whelan-And-Goldman 2001 model (amino acid alignments only) <strong>-quote</strong> <strong>--</strong> allow spaces and other restricted characters (but not ' characters) in</p></li><li><p>sequence names and quote names in the output tree (fasta input only; fasttreeMP will not be able to read these trees back in <strong>-noml</strong> to turn off maximum-likelihood <strong>-nome</strong> to turn off minimum-evolution NNIs and SPRs</p></li><li><p>(recommended if running additional ML NNIs with <strong>-intree</strong>) <strong>-nome</strong> <strong>-mllen</strong> with <strong>-intree</strong> to optimize branch lengths for a fixed topology <strong>-cat</strong> # to specify the number of rate categories of sites (default 20)</p></li><li><p>or <strong>-nocat</strong> to use constant rates <strong>-gamma</strong> <strong>--</strong> after optimizing the tree under the CAT approximation,</p></li><li><p>rescale the lengths to optimize the Gamma20 likelihood <strong>-constraints</strong> constraintAlignment to constrain the topology search</p></li><li><p>constraintAlignment should have 1s or 0s to indicates splits <strong>-expert</strong> <strong>--</strong> see more options</p></li>
</ul>
<h3>Detailed usage for fasttreeMP 2.1.4 SSE3:</h3>
<p>fasttreeMP [-nt] [-n 100] [-quote] [-pseudo | <strong>-pseudo</strong> 1.0]</p><ul>
<li><p>[-boot 1000 | <strong>-nosupport]</strong> [-intree starting_trees_file | <strong>-intree1</strong> starting_tree_file] [-quiet | <strong>-nopr]</strong> [-nni 10] [-spr 2] [-noml | <strong>-mllen</strong> | <strong>-mlnni</strong> 10] [-mlacc 2] [-cat 20 | <strong>-nocat]</strong> [-gamma] [-slow | <strong>-fastest]</strong> [-2nd | <strong>-no2nd]</strong> [-slownni] [-seed 1253] [-top | <strong>-notop]</strong> [-topm 1.0 [-close 0.75] [-refresh 0.8]] [-matrix Matrix | <strong>-nomatrix]</strong> [-nj | <strong>-bionj]</strong> [-wag] [-nt] [-gtr] [-gtrrates ac ag at cg ct gt] [-gtrfreq A C G T] [ <strong>-constraints</strong> constraintAlignment [ <strong>-constraintWeight</strong> 100.0 ] ] [-log logfile]</p></li><li><p>[ alignment_file ]</p></li><li><p>&gt; newick_tree</p></li>
</ul><p>or</p><p>fasttreeMP [-nt] [-matrix Matrix | <strong>-nomatrix]</strong> [-rawdist] <strong>-makematrix</strong> [alignment]</p><ul>
<li><p>[-n 100] &gt; phylip_distance_matrix</p></li><li><p>fasttreeMP supports fasta or phylip interleaved alignments By default fasttreeMP expects protein alignments,  use <strong>-nt</strong> for nucleotides fasttreeMP reads standard input if no alignment file is given</p></li>
</ul>
<h3>Input/output options:</h3>
<p><strong>-n</strong> <strong>--</strong> read in multiple alignments in. This only</p><ul>
<li><p>works with phylip interleaved format. For example, you can use it with the output from phylip's seqboot. If you use <strong>-n</strong>, fasttreeMP will write 1 tree per line to standard output. <strong>-intree</strong> newickfile <strong>--</strong> read the starting tree in from newickfile.</p></li><li><p>Any branch lengths in the starting trees are ignored. <strong>-intree</strong> with <strong>-n</strong> will read a separate starting tree for each alignment. <strong>-intree1</strong> newickfile <strong>--</strong> read the same starting tree for each alignment <strong>-quiet</strong> <strong>--</strong> do not write to standard error during normal operation (no progress</p></li><li><p>indicator, no options summary, no likelihood values, etc.) <strong>-nopr</strong> <strong>--</strong> do not write the progress indicator to stderr <strong>-log</strong> logfile <strong>--</strong> save intermediate trees so you can extract</p></li><li><p>the trees and restart long-running jobs if they crash <strong>-log</strong> also reports the per-site rates (1 means slowest category) <strong>-quote</strong> <strong>--</strong> quote sequence names in the output and allow spaces, commas,</p></li><li><p>parentheses, and colons in them but not ' characters (fasta files only)</p></li>
</ul>
<h3>Distances:</h3>
<ul>
<li><p>Default: For protein sequences, log-corrected distances and an</p></li><li><p>amino acid dissimilarity matrix derived from BLOSUM45</p></li><li><p>or for nucleotide sequences, Jukes-Cantor distances To specify a different matrix, use <strong>-matrix</strong> FilePrefix or <strong>-nomatrix</strong> Use <strong>-rawdist</strong> to turn the log-correction off or to use %different instead of Jukes-Cantor <strong>-pseudo</strong> [weight] <strong>--</strong> Use pseudocounts to estimate distances between</p></li><li><p>sequences with little or no overlap. (Off by default.) Recommended if analyzing the alignment has sequences with little or no overlap. If the weight is not specified, it is 1.0</p></li>
</ul>
<h3>Topology refinement:</h3>
<ul>
<li><p>By default, fasttreeMP tries to improve the tree with up to 4*log2(N) rounds of minimum-evolution nearest-neighbor interchanges (NNI), where N is the number of unique sequences, 2 rounds of subtree-prune-regraft (SPR) moves (also min. evo.), and up to 2*log(N) rounds of maximum-likelihood NNIs. Use <strong>-nni</strong> to set the number of rounds of min. evo. NNIs, and <strong>-spr</strong> to set the rounds of SPRs. Use <strong>-noml</strong> to turn off both min-evo NNIs and SPRs (useful if refining</p></li><li><p>an approximately maximum-likelihood tree with further NNIs)</p></li><li><p>Use <strong>-sprlength</strong> set the maximum length of a SPR move (default 10) Use <strong>-mlnni</strong> to set the number of rounds of maximum-likelihood NNIs Use <strong>-mlacc</strong> 2 or <strong>-mlacc</strong> 3 to always optimize all 5 branches at each NNI,</p></li><li><p>and to optimize all 5 branches in 2 or 3 rounds</p></li><li><p>Use <strong>-mllen</strong> to optimize branch lengths without ML NNIs Use <strong>-mllen</strong> <strong>-nome</strong> with <strong>-intree</strong> to optimize branch lengths on a fixed topology Use <strong>-slownni</strong> to turn off heuristics to avoid constant subtrees (affects both</p></li><li><p>ML and ME NNIs)</p></li>
</ul>
<h3>Maximum likelihood model options:</h3>
<p><strong>-wag</strong> <strong>--</strong> Whelan-And-Goldman 2001 model instead of (default) Jones-Taylor-Thorton 1992 model (a.a. only) <strong>-gtr</strong> <strong>--</strong> generalized time-reversible instead of (default) Jukes-Cantor (nt only) <strong>-cat</strong> # <strong>--</strong> specify the number of rate categories of sites (default 20) <strong>-nocat</strong> <strong>--</strong> no CAT model (just 1 category) <strong>-gamma</strong> <strong>--</strong> after the final round of optimizing branch lengths with the CAT model,</p><ul>
<li><p>report the likelihood under the discrete gamma model with the same number of categories. fasttreeMP uses the same branch lengths but optimizes the gamma shape parameter and the scale of the lengths. The final tree will have rescaled lengths. Used with <strong>-log</strong>, this also generates per-site likelihoods for use with CONSEL, see GammaLogToPaup.pl and documentation on the fasttreeMP web site.</p></li>
</ul>
<h3>Support value options:</h3>
<ul>
<li><p>By default, fasttreeMP computes local support values by resampling the site likelihoods 1,000 times and the Shimodaira Hasegawa test. If you specify <strong>-nome</strong>, it will compute minimum-evolution bootstrap supports instead In either case, the support values are proportions ranging from 0 to 1</p></li><li><p>Use <strong>-nosupport</strong> to turn off support values or <strong>-boot</strong> 100 to use just 100 resamples Use <strong>-seed</strong> to initialize the random number generator</p></li>
</ul>
<h3>Searching for the best join:</h3>
<ul>
<li><p>By default, fasttreeMP combines the 'visible set' of fast neighbor-joining with</p></li><li><p>local hill-climbing as in relaxed neighbor-joining <strong>-slow</strong> <strong>--</strong> exhaustive search (like NJ or BIONJ, but different gap handling) <strong>-slow</strong> takes half an hour instead of 8 seconds for 1,250 proteins <strong>-fastest</strong> <strong>--</strong> search the visible set (the top hit for each node) only</p></li><li><p>Unlike the original fast neighbor-joining, <strong>-fastest</strong> updates visible(C) after joining A and B if join(AB,C) is better than join(C,visible(C)) <strong>-fastest</strong> also updates out-distances in a very lazy way, <strong>-fastest</strong> sets <strong>-2nd</strong> on as well, use <strong>-fastest</strong> <strong>-no2nd</strong> to avoid this</p></li>
</ul>
<h3>Top-hit heuristics:</h3>
<ul>
<li><p>By default, fasttreeMP uses a top-hit list to speed up search Use <strong>-notop</strong> (or <strong>-slow</strong>) to turn this feature off</p></li><li><p>and compare all leaves to each other, and all new joined nodes to each other <strong>-topm</strong> 1.0 <strong>--</strong> set the top-hit list size to parameter*sqrt(N)</p></li><li><p>fasttreeMP estimates the top m hits of a leaf from the top 2*m hits of a 'close' neighbor, where close is defined as d(seed,close) &lt; 0.75 * d(seed, hit of rank 2*m), and updates the top-hits as joins proceed <strong>-close</strong> 0.75 <strong>--</strong> modify the close heuristic, lower is more conservative <strong>-refresh</strong> 0.8 <strong>--</strong> compare a joined node to all other nodes if its</p></li><li><p>top-hit list is less than 80% of the desired length, or if the age of the top-hit list is log2(m) or greater <strong>-2nd</strong> or <strong>-no2nd</strong> to turn 2nd-level top hits heuristic on or off</p></li><li><p>This reduces memory usage and running time but may lead to marginal reductions in tree quality. (By default, <strong>-fastest</strong> turns on <strong>-2nd</strong>.)</p></li>
</ul>
<h3>Join options:</h3>
<p><strong>-nj</strong>: regular (unweighted) neighbor-joining (default) <strong>-bionj</strong>: weighted joins as in BIONJ</p><ul>
<li><p>fasttreeMP will also weight joins during NNIs</p></li>
</ul>
<h3>Constrained topology search options:</h3>
<p><strong>-constraints</strong> alignmentfile <strong>--</strong> an alignment with values of 0, 1, and -</p><ul>
<li><p>Not all sequences need be present. A column of 0s and 1s defines a constrained split. Some constraints may be violated (see 'violating constraints:' in standard error). <strong>-constraintWeight</strong> <strong>--</strong> how strongly to weight the constraints. A value of 1</p></li><li><p>means a penalty of 1 in tree length for violating a constraint Default: 100.0</p></li>
</ul><p>For more information, see http://www.microbesonline.org/fasttree/</p><ul>
<li><p>or the comments in the source code</p></li><li><p>fasttreeMP protein_alignment &gt; tree fasttreeMP <strong>-nt</strong> nucleotide_alignment &gt; tree fasttreeMP <strong>-nt</strong> <strong>-gtr</strong> &lt; nucleotide_alignment &gt; tree</p></li>
</ul><p>fasttreeMP accepts alignments in fasta or phylip interleaved formats</p>
<h3>Common options (must be before the alignment file):</h3>
<p><strong>-quiet</strong> to suppress reporting information <strong>-nopr</strong> to suppress progress indicator <strong>-log</strong> logfile <strong>--</strong> save intermediate trees, settings, and model details <strong>-fastest</strong> <strong>--</strong> speed up the neighbor joining phase & reduce memory usage</p><ul>
<li><p>(recommended for &gt;50,000 sequences) <strong>-n</strong> &lt;number&gt; to analyze multiple alignments (phylip format only)</p></li><li><p>(use for global bootstrap, with seqboot and CompareToBootstrap.pl) <strong>-nosupport</strong> to not compute support values <strong>-intree</strong> newick_file to set the starting tree(s) <strong>-intree1</strong> newick_file to use this starting tree for all the alignments</p></li><li><p>(for faster global bootstrap on huge alignments) <strong>-pseudo</strong> to use pseudocounts (recommended for highly gapped sequences) <strong>-gtr</strong> <strong>--</strong> generalized time-reversible model (nucleotide alignments only) <strong>-wag</strong> <strong>--</strong> Whelan-And-Goldman 2001 model (amino acid alignments only) <strong>-quote</strong> <strong>--</strong> allow spaces and other restricted characters (but not ' characters) in</p></li><li><p>sequence names and quote names in the output tree (fasta input only; fasttreeMP will not be able to read these trees back in <strong>-noml</strong> to turn off maximum-likelihood <strong>-nome</strong> to turn off minimum-evolution NNIs and SPRs</p></li><li><p>(recommended if running additional ML NNIs with <strong>-intree</strong>) <strong>-nome</strong> <strong>-mllen</strong> with <strong>-intree</strong> to optimize branch lengths for a fixed topology <strong>-cat</strong> # to specify the number of rate categories of sites (default 20)</p></li><li><p>or <strong>-nocat</strong> to use constant rates <strong>-gamma</strong> <strong>--</strong> after optimizing the tree under the CAT approximation,</p></li><li><p>rescale the lengths to optimize the Gamma20 likelihood <strong>-constraints</strong> constraintAlignment to constrain the topology search</p></li><li><p>constraintAlignment should have 1s or 0s to indicates splits <strong>-expert</strong> <strong>--</strong> see more options</p></li>
</ul><p>For more information, see http://www.microbesonline.org/fasttree/</p>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="fasttree.1.html"><span aria-hidden="true">&larr;</span> fasttree.1: Create phylogenetic trees from alignments of nucleotide or protein sequences</a></li>
   <li class="next"><a href="fastx_artifacts_filter.1.html">fastx_artifacts_filter.1: Fastq/a artifacts filter <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
