<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>genome-music-play: Run the full suite of music tools sequentially.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Run the full suite of music tools sequentially.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="genome-music-play (1p) manual">
  <meta name="twitter:description" content="Run the full suite of music tools sequentially.">
  <meta name="twitter:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-genome-music-play-1p.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1p/genome-music-play.1p.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="genome-music-play (1p) manual" />
  <meta property="og:description" content="Run the full suite of music tools sequentially." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-genome-music-play-1p.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">genome-music-play<small> (1p)</small></h1>
        <p class="lead">Run the full suite of music tools sequentially.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/genome-music-play.1p.html">
      <span itemprop="name">genome-music-play: Run the full suite of music tools sequentially.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libgenome-model-tools-music-perl/">
      <span itemprop="name">libgenome-model-tools-music-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/genome-music-play.1p.html">
      <span itemprop="name">genome-music-play: Run the full suite of music tools sequentially.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">genome music play</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">VERSION</h2>
        <div class="sectioncontent">
<p>This document describes genome music play version 0.04 (2013-05-14 at 16:03:04)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>genome music play --bam-list=? --roi-file=? --reference-sequence=? --output-dir=? --maf-file=? --pathway-file=? [--numeric-clinical-data-file=?] [--categorical-clinical-data-file=?] [--mutation-matrix-file=?] [--permutations=?] [--normal-min-depth=?] [--tumor-min-depth=?] [--min-mapq=?] [--show-skipped] [--genes-to-ignore=?] [--bmr=?] [--max-proximity=?] [--bmr-modifier-file=?] [--numerical-data-test-method=?] [--skip-low-mr-genes] [--max-fdr=?] [--genetic-data-type=?] [--wu-annotation-headers] [--bmr-groups=?] [--separate-truncations] [--merge-concurrent-muts] [--skip-non-coding] [--skip-silent] [--min-mut-genes-per-path=?] [--glm-model-file=?] [--processors=?] [--aa-range=?] [--nuc-range=?] [--reference-build=?] [--show-known-hits] [--glm-clinical-data-file=?] [--use-maf-in-glm] [--omimaa-dir=?] [--cosmic-dir=?] [--verbose] [--clinical-correlation-matrix-file=?]</p><p>This tool takes as parameters all the information required to run the individual tools. An example usage is:</p>
<pre>
 ... music play &#92;
        --bam-list input/bams_to_analyze.txt &#92;
        --numeric-clinical-data-file input/numeric_clinical_data.csv &#92;
        --maf-file input/myMAF.tsv &#92;
        --output-dir play_output_dir &#92;
        --pathway-file input/pathway_db &#92;
        --reference-sequence input/refseq/all_sequences.fa &#92;
        --roi-file input/all_coding_regions.bed &#92;
        --genetic-data-type gene
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REQUIRED ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    bam-list  <em>Text</em>
  </dt>
  <dd>
    <p>Tab delimited list of \s-1BAM\s0 files [sample_name normal_bam tumor_bam]</p>
  </dd>
  <dt>
    roi-file  <em>Text</em>
  </dt>
  <dd>
    <p>Tab delimited list of ROIs [chr start stop gene_name]</p>
  </dd>
  <dt>
    reference-sequence  <em>Text</em>
  </dt>
  <dd>
    <p>Path to reference sequence in \s-1FASTA\s0 format</p>
  </dd>
  <dt>
    output-dir  <em>Text</em>
  </dt>
  <dd>
    <p>Directory where output files and subdirectories will be written</p>
  </dd>
  <dt>
    maf-file  <em>Text</em>
  </dt>
  <dd>
    <p>List of mutations using \s-1TCGA\s0 \s-1MAF\s0 specifications v2.3</p>
  </dd>
  <dt>
    pathway-file  <em>Text</em>
  </dt>
  <dd>
    <p>Tab-delimited file of pathway information</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONAL ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    numeric-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Table of samples (y) vs. numeric clinical data category (x)</p>
  </dd>
  <dt>
    categorical-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Table of samples (y) vs. categorical clinical data category (x)</p>
  </dd>
  <dt>
    mutation-matrix-file  <em>Text</em>
  </dt>
  <dd>
    <p>Optionally store the sample-vs-gene matrix used during calculations.</p>
  </dd>
  <dt>
    permutations  <em>Number</em>
  </dt>
  <dd>
    <p>Number of permutations used to determine P-values</p>
  </dd>
  <dt>
    normal-min-depth  <em>Integer</em>
  </dt>
  <dd>
    <p>The minimum read depth to consider a Normal \s-1BAM\s0 base as covered</p>
  </dd>
  <dt>
    tumor-min-depth  <em>Integer</em>
  </dt>
  <dd>
    <p>The minimum read depth to consider a Tumor \s-1BAM\s0 base as covered</p>
  </dd>
  <dt>
    min-mapq  <em>Integer</em>
  </dt>
  <dd>
    <p>The minimum mapping quality of reads to consider towards read depth counts</p>
  </dd>
  <dt>
    show-skipped  <em>Boolean</em>
  </dt>
  <dd>
    <p>Report each skipped mutation, not just how many Default value 'false' (--noshow-skipped) if not specified</p>
  </dd>
  <dt>
    genes-to-ignore  <em>Text</em>
  </dt>
  <dd>
    <p>Comma-delimited list of genes to ignore for background mutation rates</p>
  </dd>
  <dt>
    bmr  <em>Number</em>
  </dt>
  <dd>
    <p>Background mutation rate in the targeted regions</p>
  </dd>
  <dt>
    max-proximity  <em>Text</em>
  </dt>
  <dd>
    <p>Maximum \s-1AA\s0 distance between 2 mutations</p>
  </dd>
  <dt>
    bmr-modifier-file  <em>Text</em>
  </dt>
  <dd>
    <p>Tab delimited list of values per gene that modify \s-1BMR\s0 before testing [gene_name bmr_modifier]</p>
  </dd>
  <dt>
    numerical-data-test-method  <em>Text</em>
  </dt>
  <dd>
    <p>Either 'cor' for Pearson Correlation or 'wilcox' for the Wilcoxon Rank-Sum Test for numerical clinical data. Default value 'cor' if not specified</p>
  </dd>
  <dt>
    skip-low-mr-genes  <em>Boolean</em>
  </dt>
  <dd>
    <p>Skip testing genes with MRs lower than the background \s-1MR\s0 Default value 'true' if not specified</p>
  </dd>
  <dt>
    max-fdr  <em>Number</em>
  </dt>
  <dd>
    <p>The maximum allowed false discovery rate for a gene to be considered an \s-1SMG\s0 Default value '0.2' if not specified</p>
  </dd>
  <dt>
    genetic-data-type  <em>Text</em>
  </dt>
  <dd>
    <p>Data in matrix file must be either \*(L"gene\*(R" or \*(L"variant\*(R" type data</p>
  </dd>
  <dt>
    wu-annotation-headers  <em>Boolean</em>
  </dt>
  <dd>
    <p>Use this to default to wustl annotation format headers</p>
  </dd>
  <dt>
    bmr-groups  <em>Integer</em>
  </dt>
  <dd>
    <p>Number of clusters of samples with comparable BMRs Default value '1' if not specified</p>
  </dd>
  <dt>
    separate-truncations  <em>Boolean</em>
  </dt>
  <dd>
    <p>Group truncational mutations as a separate category Default value 'false' (--noseparate-truncations) if not specified</p>
  </dd>
  <dt>
    merge-concurrent-muts  <em>Boolean</em>
  </dt>
  <dd>
    <p>Multiple mutations of a gene in the same sample are treated as 1 Default value 'false' (--nomerge-concurrent-muts) if not specified</p>
  </dd>
  <dt>
    skip-non-coding  <em>Boolean</em>
  </dt>
  <dd>
    <p>Skip non-coding mutations from the provided \s-1MAF\s0 file Default value 'true' if not specified</p>
  </dd>
  <dt>
    skip-silent  <em>Boolean</em>
  </dt>
  <dd>
    <p>Skip silent mutations from the provided \s-1MAF\s0 file Default value 'true' if not specified</p>
  </dd>
  <dt>
    min-mut-genes-per-path  <em>Integer</em>
  </dt>
  <dd>
    <p>Pathways with fewer mutated genes than this will be ignored Default value '1' if not specified</p>
  </dd>
  <dt>
    glm-model-file  <em>Text</em>
  </dt>
  <dd>
    <p>File outlining the type of model, response variable, covariants, etc. for the \s-1GLM\s0 analysis. (See \s-1DESCRIPTION\s0).</p>
  </dd>
  <dt>
    processors  <em>Integer</em>
  </dt>
  <dd>
    <p>Number of processors to use in \s-1SMG\s0 (requires 'foreach' and 'doMC' R packages) Default value '1' if not specified</p>
  </dd>
  <dt>
    aa-range  <em>Integer</em>
  </dt>
  <dd>
    <p>Set how close a 'near' match is when searching for amino acid near hits Default value '2' if not specified</p>
  </dd>
  <dt>
    nuc-range  <em>Integer</em>
  </dt>
  <dd>
    <p>Set how close a 'near' match is when searching for nucleotide position near hits Default value '5' if not specified</p>
  </dd>
  <dt>
    reference-build  <em>Text</em>
  </dt>
  <dd>
    <p>Put either \*(L"Build36\*(R" or \*(L"Build37\*(R" Default value 'Build37' if not specified</p>
  </dd>
  <dt>
    show-known-hits  <em>Boolean</em>
  </dt>
  <dd>
    <p>When a finding is novel, show known \s-1AA\s0 in that gene Default value 'true' if not specified</p>
  </dd>
  <dt>
    glm-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Clinical traits, mutational profiles, other mixed clinical data (See \s-1DESCRIPTION\s0).</p>
  </dd>
  <dt>
    use-maf-in-glm  <em>Boolean</em>
  </dt>
  <dd>
    <p>Set this flag to use the variant matrix created from the \s-1MAF\s0 file as variant input to \s-1GLM\s0 analysis. Default value 'false' (--nouse-maf-in-glm) if not specified</p>
  </dd>
  <dt>
    omimaa-dir  <em>Path</em>
  </dt>
  <dd>
    <p>omim amino acid mutation database folder</p>
  </dd>
  <dt>
    cosmic-dir  <em>Path</em>
  </dt>
  <dd>
    <p>cosmic amino acid mutation database folder</p>
  </dd>
  <dt>
    verbose  <em>Boolean</em>
  </dt>
  <dd>
    <p>turn on to display larger working output Default value 'true' if not specified</p>
  </dd>
  <dt>
    clinical-correlation-matrix-file  <em>Text</em>
  </dt>
  <dd>
    <p>Optionally store the sample-vs-gene matrix used internally during calculations.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This command can be used to run all of the MuSiC analysis tools on a set of data. Please see the individual tools for further description of the parameters.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p> Thomas B. Mooney, M.S.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">CREDITS</h2>
        <div class="sectioncontent">
<p>Please see the credits for <strong>genome-music</strong>(1).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO genome-music-play&hellip;</h2>
        <div class="sectioncontent">
<p><strong>genome-music</strong>(1), <strong>genome-music-path-scan</strong>(1), <strong>genome-music-smg</strong>(1), <strong>genome-music-clinical-correlation</strong>(1), <strong>genome-music-mutation-relation</strong>(1), <strong>genome-music-cosmic-omim</strong>(1), <strong>genome-music-proximity</strong>(1), <strong>genome-music-pfam</strong>(1)</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="genome-music-pfam.1p.html"><span aria-hidden="true">&larr;</span> genome-music-pfam.1p: Add pfam annotation to a maf file</a></li>
   <li class="next"><a href="genome-music-plot.1p.html">genome-music-plot.1p: Generate relevant plots and visualizations for music. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
