<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>genome-music-cosmic-omim: Compare the amino acid changes of supplied mutations to cosmic and omim databases.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Compare the amino acid changes of supplied mutations to cosmic and omim databases.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="genome-music-cosmic-omim (1p) manual">
  <meta name="twitter:description" content="Compare the amino acid changes of supplied mutations to cosmic and omim databases.">
  <meta name="twitter:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-genome-music-cosmic-omim-1p.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1p/genome-music-cosmic-omim.1p.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="genome-music-cosmic-omim (1p) manual" />
  <meta property="og:description" content="Compare the amino acid changes of supplied mutations to cosmic and omim databases." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-genome-music-cosmic-omim-1p.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">genome-music-cosmic-omim<small> (1p)</small></h1>
        <p class="lead">Compare the amino acid changes of supplied mutations to cosmic and omim databases.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/genome-music-cosmic-omim.1p.html">
      <span itemprop="name">genome-music-cosmic-omim: Compare the amino acid changes of supplied mutations to cosmic and omim databases.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libgenome-model-tools-music-perl/">
      <span itemprop="name">libgenome-model-tools-music-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/genome-music-cosmic-omim.1p.html">
      <span itemprop="name">genome-music-cosmic-omim: Compare the amino acid changes of supplied mutations to cosmic and omim databases.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">genome music cosmic-omim</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">VERSION</h2>
        <div class="sectioncontent">
<p>This document describes genome music cosmic-omim (2013-05-14 at 16:03:04)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>genome music cosmic-omim --maf-file=? --output-file=? --reference-build=? [--omimaa-dir=?] [--cosmic-dir=?] [--verbose] [--wu-annotation-headers] [--aa-range=?] [--nuc-range=?] [--show-known-hits]</p>
<pre>
 ... music cosmic-omim &#92;
        --maf-file input_dir/myMAF.tsv &#92;
        --output-file output_dir/myMAF_output.tsv &#92;
        --no-verbose

 ... music cosmic-omim &#92;
        --maf-file input_dir/myMAF.tsv &#92;
        --output-file output_dir/myMAF_output.tsv &#92;
        --omimaa-dir omim_dir/ &#92;
        --cosmic-dir cosmic_dir/ &#92;
        --no-verbose
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REQUIRED ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    maf-file  <em>Path</em>
  </dt>
  <dd>
    <p>list of annotated mutations in \s-1MAF\s0 format (or any file with MAF+annotation headers)</p>
  </dd>
  <dt>
    output-file  <em>Path</em>
  </dt>
  <dd>
    <p>Output file contains the input file with two columns appended to the end, corresponding to cosmic and omim mutation comparisons, respectively</p>
  </dd>
  <dt>
    reference-build  <em>Text</em>
  </dt>
  <dd>
    <p>Put either \*(L"Build36\*(R" or \*(L"Build37\*(R" Default value 'Build37' if not specified</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONAL ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    omimaa-dir  <em>Path</em>
  </dt>
  <dd>
    <p>omim amino acid mutation database folder</p>
  </dd>
  <dt>
    cosmic-dir  <em>Path</em>
  </dt>
  <dd>
    <p>cosmic amino acid mutation database folder</p>
  </dd>
  <dt>
    verbose  <em>Boolean</em>
  </dt>
  <dd>
    <p>Use this to display the larger working output Default value 'false' (--noverbose) if not specified</p>
  </dd>
  <dt>
    wu-annotation-headers  <em>Boolean</em>
  </dt>
  <dd>
    <p>Use this if input \s-1MAF\s0 contains \s-1WUSTL\s0 annotation format headers Default value 'false' (--nowu-annotation-headers) if not specified</p>
  </dd>
  <dt>
    aa-range  <em>Integer</em>
  </dt>
  <dd>
    <p>Set how close a 'near' match is when searching for amino acid near hits Default value '2' if not specified</p>
  </dd>
  <dt>
    nuc-range  <em>Integer</em>
  </dt>
  <dd>
    <p>Set how close a 'near' match is when searching for nucleotide position near hits Default value '5' if not specified</p>
  </dd>
  <dt>
    show-known-hits  <em>Boolean</em>
  </dt>
  <dd>
    <p>When a finding is novel, show known \s-1AA\s0 in that gene Default value 'true' if not specified</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This tool looks at the amino acid changes for the given set of mutations and compares the genomic coordinates as well as the affected amino acid to the coordinates and amino acids of all cancer-specific mutations listed in the Cosmic and \s-1OMIM\s0 databases. The database files are specially prepared for this task and provided with the MuSiC suite. The tool reports various types of matches, including matches within \*(L"near proximity\*(R", where \*(L"near proximity\*(R" is currently defined as a linear \s-1DNA\s0 distance of 5 bases or 2 amino acids. (This type of matching helps to account for the possibility of subtle differences in reported positions for variants due to differences in transcript definitions or other things of this nature.) Any site without a match in a particular databases is reported as \*(L"novel\*(R" with respect to that database.</p><p>The output of this script returns each row the original input \s-1MAF\s0 file with two columns appended to the end of each, one column for each of the databases. Also included is a \s-1STDOUT\s0 printout of a summary of what was found in the input \s-1MAF\s0. Neither output can be suppressed in the current version. The --verbose option is used to display working notes that are useful for various purposes in debugging potential \s-1MAF\s0 problems. The Omim and Cosmic directories must point to the output of the downloader, named appropriately, as they don't recognize the \s-1OMIM\s0 database in the raw download format.</p><p>This tool only compares build 36 or build 37 coordinates that are specified in Cosmic to the coordinates in your maf file. This is a weakness of the Cosmic database (not all position entries are currently available for both builds) that is out of our control.</p><p>In addition to the standard version 2.3 \s-1MAF\s0 headers, there needs to be 3 columns appended. These column headers in the \s-1MAF\s0 must have these names in the header in order for the tool to find them:</p>
<pre>
   transcript_name - the transcript name, such as \s-1NM_000028\s0
</pre>
<p> amino_acid_change - the amino acid change, such as p.R290H</p>
<pre>
        c_position - the nucleotide position changed, such as c.869
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">LICENSE</h2>
        <div class="sectioncontent">
<p>Copyright (C) 2010-2011 Washington University in St. Louis.</p><p>It is released under the Lesser \s-1GNU\s0 Public License (\s-1LGPL\s0) version 3.  See the associated \s-1LICENSE\s0 file in this distribution.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p> Brian Dunford-Shore  David Larson, Ph.D.  Michael C. Wendl, Ph.D.  William Schierding, M.S.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">CREDITS</h2>
        <div class="sectioncontent">
<p>This tool depends on copies of data from the following databases, packaged in a form useable for quick analysis:</p><p> * COSMIC - http://www.sanger.ac.uk/genetics/CGP/cosmic/  * OMIM - http://www.ncbi.nlm.nih.gov/omim</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO genome-music-cosmic-omim&hellip;</h2>
        <div class="sectioncontent">
<p><strong>genome-music</strong>(1), <a href="../man1/genome.1.html"><strong>genome</strong>(1)</a></p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="genome-music-clinical-correlation.1p.html"><span aria-hidden="true">&larr;</span> genome-music-clinical-correlation.1p: Correlate phenotypic traits against mutated genes, or against individual variants</a></li>
   <li class="next"><a href="genome-music-galaxy.1p.html">genome-music-galaxy.1p: Run the full suite of music tools sequentially. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
