<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>genome-music-clinical-correlation: Correlate phenotypic traits against mutated genes, or against individual variants</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Correlate phenotypic traits against mutated genes, or against individual variants">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="genome-music-clinical-correlation (1p) manual">
  <meta name="twitter:description" content="Correlate phenotypic traits against mutated genes, or against individual variants">
  <meta name="twitter:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-genome-music-clinical-correlation-1p.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1p/genome-music-clinical-correlation.1p.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="genome-music-clinical-correlation (1p) manual" />
  <meta property="og:description" content="Correlate phenotypic traits against mutated genes, or against individual variants" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-genome-music-clinical-correlation-1p.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">genome-music-clinical-correlation<small> (1p)</small></h1>
        <p class="lead">Correlate phenotypic traits against mutated genes, or against individual variants</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/genome-music-clinical-correlation.1p.html">
      <span itemprop="name">genome-music-clinical-correlation: Correlate phenotypic traits against mutated genes, or against individual variants</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libgenome-model-tools-music-perl/">
      <span itemprop="name">libgenome-model-tools-music-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/genome-music-clinical-correlation.1p.html">
      <span itemprop="name">genome-music-clinical-correlation: Correlate phenotypic traits against mutated genes, or against individual variants</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">genome music clinical-correlation</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">VERSION</h2>
        <div class="sectioncontent">
<p>This document describes genome music clinical-correlation version 0.04 (2013-05-14 at 16:03:04)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>genome music clinical-correlation --bam-list=? --output-file=? [--maf-file=?] [--glm-clinical-data-file=?] [--use-maf-in-glm] [--skip-non-coding] [--skip-silent] [--clinical-correlation-matrix-file=?] [--input-clinical-correlation-matrix-file=?] [--genetic-data-type=?] [--numeric-clinical-data-file=?] [--numerical-data-test-method=?] [--categorical-clinical-data-file=?] [--glm-model-file=?]</p>
<pre>
 ... music clinical-correlation &#92;
        --bam-list /path/myBamList.tsv &#92;
        --maf-file /path/myMAF.tsv &#92;
        --numeric-clinical-data-file /path/myNumericData.tsv &#92;
        --genetic-data-type &apos;gene&apos; &#92;
        --output-file /path/output_file

 ... music clinical-correlation &#92;
        --maf-file /path/myMAF.tsv &#92;
        --bam-list /path/myBamList.tsv &#92;
        --numeric-clinical-data-file /path/myNumericData.tsv &#92;
        --categorical-clinical-data-file /path/myClassData.tsv &#92;
        --genetic-data-type &apos;gene&apos; &#92;
        --output-file /path/output_file

 ... music clinical-correlation &#92;
        --maf-file /path/myMAF.tsv &#92;
        --bam-list /path/myBamList.tsv &#92;
        --output-file /path/output_file &#92;
        --glm-model-file /path/model.tsv &#92;
        --glm-clinical-data-file /path/glm_clinical_data.tsv &#92;
        --use-maf-in-glm
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REQUIRED ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    bam-list  <em>Text</em>
  </dt>
  <dd>
    <p>Tab delimited list of \s-1BAM\s0 files [sample_name, normal_bam, tumor_bam] (See Description)</p>
  </dd>
  <dt>
    output-file  <em>Text</em>
  </dt>
  <dd>
    <p>Results of clinical-correlation tool. Will have suffix added for data type</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONAL ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    maf-file  <em>Text</em>
  </dt>
  <dd>
    <p>List of mutations using \s-1TCGA\s0 \s-1MAF\s0 specification v2.3</p>
  </dd>
  <dt>
    glm-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Clinical traits, mutational profiles, other mixed clinical data (See \s-1DESCRIPTION\s0)</p>
  </dd>
  <dt>
    use-maf-in-glm  <em>Boolean</em>
  </dt>
  <dd>
    <p>Create a variant matrix from the \s-1MAF\s0 file as variant input to \s-1GLM\s0 analysis. Default value 'false' (--nouse-maf-in-glm) if not specified</p>
  </dd>
  <dt>
    skip-non-coding  <em>Boolean</em>
  </dt>
  <dd>
    <p>Skip non-coding mutations from the provided \s-1MAF\s0 file Default value 'true' if not specified</p>
  </dd>
  <dt>
    skip-silent  <em>Boolean</em>
  </dt>
  <dd>
    <p>Skip silent mutations from the provided \s-1MAF\s0 file Default value 'true' if not specified</p>
  </dd>
  <dt>
    clinical-correlation-matrix-file  <em>Text</em>
  </dt>
  <dd>
    <p>Specify a file to store the sample-vs-gene matrix created during calculations</p>
  </dd>
  <dt>
    input-clinical-correlation-matrix-file  <em>Text</em>
  </dt>
  <dd>
    <p>Instead of creating this from the \s-1MAF\s0, input the sample-vs-gene matrix for calculations</p>
  </dd>
  <dt>
    genetic-data-type  <em>Text</em>
  </dt>
  <dd>
    <p>Correlate clinical data to \*(L"gene\*(R" or \*(L"variant\*(R" level data Default value 'gene' if not specified</p>
  </dd>
  <dt>
    numeric-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Table of samples (y) vs. numeric clinical data category (x)</p>
  </dd>
  <dt>
    numerical-data-test-method  <em>Text</em>
  </dt>
  <dd>
    <p>Either 'cor' for Pearson Correlation or 'wilcox' for the Wilcoxon Rank-Sum Test for numerical clinical data Default value 'cor' if not specified</p>
  </dd>
  <dt>
    categorical-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Table of samples (y) vs. categorical clinical data category (x)</p>
  </dd>
  <dt>
    glm-model-file  <em>Text</em>
  </dt>
  <dd>
    <p>File outlining the type of model, response variable, covariants, etc. for the \s-1GLM\s0 analysis. (See \s-1DESCRIPTION\s0)</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This command relates clinical traits and mutational data. Either one can perform correlation analysis between mutations recorded in a \s-1MAF\s0 and the particular phenotypic traits recorded in clinical data files for the same samples, or one can run a generalized linear model (\s-1GLM\s0) analysis on the same types of data.</p><p>The clinical data files for correlation must be separated between numeric and categoric data and must follow these conventions:</p><ul>
<li><p>Headers are required</p></li><li><p>Each file must include at least 1 sample_id column and 1 attribute column, with the format being [sample_id  clinical_data_attribute_1 clinical_data_attribute_2 ...]</p></li><li><p>The sample \s-1ID\s0 must match the sample \s-1ID\s0 listed in the \s-1MAF\s0 under \*(L"Tumor_Sample_Barcode\*(R" for relating the mutations of this sample.</p></li>
</ul><p>Note the importance of the headers: the header for each clinical_data_attribute will appear in the output file to denote relationships with the mutation data from the \s-1MAF\s0.</p><p>Internally, the input data is fed into an R script which calculates a P-value representing the probability that the correlation seen between the mutations in each gene (or variant) and each phenotype trait are random. Lower P-values indicate lower randomness, or likely true correlations.</p><p>The results are saved to the output filename given with a suffix appended; \*(L".numeric.csv\*(R" will be appended for results derived from numeric clinical data, and \*(L".categorical.csv\*(R" will be appended for results derived from categorical clinical data. Also, \*(L".glm.csv\*(R" will be appended to the output filename for \s-1GLM\s0 results.</p><p>The \s-1GLM\s0 analysis accepts a mixed numeric and categoric clinical data file, input using the parameter --glm-clinical-data-file. \s-1GLM\s0 clinical data must adhere to the formats described above for the correlation clinical data files. \s-1GLM\s0 also requires the user to input a --glm-model-file. This file requires specific headers and defines the analysis to be performed rather exactly. Here are the conventions required for this file:</p><ul>
<li><p>Columns must be ordered as such:</p></li><li><p>[ analysis_type    clinical_data_trait_name    variant/gene_name   covariates  memo ]</p></li><li><p>The 'analysis_type' column must contain either \*(L"Q\*(R", indicating a quantative trait, or \*(L"B\*(R", indicating a binary trait will be examined.</p></li><li><p>The 'clinical_data_trait_name' is the name of a clinical data trait defined by being a header in the --glm-clinical-data-file.</p></li><li><p>The 'variant/gene_name' can either be the name of one or more columns from the --glm-clinical-data-file, or the name of one or more mutated gene names from the \s-1MAF\s0, separated by \*(L"|\*(R". If this column is left blank, or instead contains \*(L"\s-1NA\s0\*(R", then each column from either the variant mutation matrix (--use-maf-in-glm) or alternatively the --glm-clinical-data-file is used consecutively as the variant column in independent analyses.</p></li><li><p>'covariates' are the names of one or more columns from the --glm-clinical-data-file, separated by \*(L"+\*(R".</p></li><li><p>'memo' is any note deemed useful to the user. It will be printed in the output data file for reference.</p></li>
</ul><p>\s-1GLM\s0 analysis may be performed using solely the data input into --glm-clinical-data-file, as described above, or alternatively, mutational data from the \s-1MAF\s0 may be included as variants in the \s-1GLM\s0 analysis, as also described above. Use the --use-maf-in-glm flag to include the mutation matrix derived from the maf as variant data.</p><p>Note that all input files for both correlation and \s-1GLM\s0 analysis must be tab-separated.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    --bam-list
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Provide a file containing sample names and normal/tumor \s-1BAM\s0 locations for each. Use the tab- delimited format [sample_name normal_bam tumor_bam] per line. This tool only needs sample_name, so all other columns can be skipped. The sample_name must be the same as the tumor sample names used in the \s-1MAF\s0 file (16th column, with the header Tumor_Sample_Barcode).
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">LICENSE</h2>
        <div class="sectioncontent">
<p>Copyright (C) 2010-2011 Washington University in St. Louis.</p><p>It is released under the Lesser \s-1GNU\s0 Public License (\s-1LGPL\s0) version 3.  See the associated \s-1LICENSE\s0 file in this distribution.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p> Nathan D. Dees, Ph.D.  Qunyuan Zhang, Ph.D.  William Schierding, M.S.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO genome-music-clinical-correlation&hellip;</h2>
        <div class="sectioncontent">
<p><strong>genome-music</strong>(1), <a href="../man1/genome.1.html"><strong>genome</strong>(1)</a></p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="genome-music-bmr-calc-wig-covg.1p.html"><span aria-hidden="true">&larr;</span> genome-music-bmr-calc-wig-covg.1p: Count covered bases per-gene for each given wiggle track format file.</a></li>
   <li class="next"><a href="genome-music-cosmic-omim.1p.html">genome-music-cosmic-omim.1p: Compare the amino acid changes of supplied mutations to cosmic and omim databases. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
