<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>genome-music: Mutational significance in cancer (cancer mutation analysis)</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Mutational significance in cancer (cancer mutation analysis)">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="genome-music (1p) manual">
  <meta name="twitter:description" content="Mutational significance in cancer (cancer mutation analysis)">
  <meta name="twitter:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-genome-music-1p.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1p/genome-music.1p.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="genome-music (1p) manual" />
  <meta property="og:description" content="Mutational significance in cancer (cancer mutation analysis)" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-genome-music-1p.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">genome-music<small> (1p)</small></h1>
        <p class="lead">Mutational significance in cancer (cancer mutation analysis)</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/genome-music.1p.html">
      <span itemprop="name">genome-music: Mutational significance in cancer (cancer mutation analysis)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libgenome-model-tools-music-perl/">
      <span itemprop="name">libgenome-model-tools-music-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/genome-music.1p.html">
      <span itemprop="name">genome-music: Mutational significance in cancer (cancer mutation analysis)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">genome music</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">VERSION</h2>
        <div class="sectioncontent">
<p>This document describes genome music version 0.04 (2013-05-14 at 16:03:04)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>genome music ...</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>The MuSiC suite is a set of tools aimed at discovering the significance of somatic mutations found within a given cohort of cancer samples, and with respect to a variety of external data sources. The standard inputs required are:</p>
<dl class='dl-vertical'>
  <dt>
    1. mapped reads in \s-1BAM\s0 format
  </dt>
  <dd>
    
  </dd>
  <dt>
    2. predicted or validated SNVs or indels in mutation annotation format (\s-1MAF\s0)
  </dt>
  <dd>
    
  </dd>
  <dt>
    3. a list of regions of interest (typically the boundaries of coding exons)
  </dt>
  <dd>
    
  </dd>
  <dt>
    4. any relevant numeric or categorical clinical data.
  </dt>
  <dd>
    
  </dd>

</dl>
<p>The formats for inputs 3. and 4. are:</p>
<dl class='dl-vertical'>
  <dt>
    3. Regions of Interest File:
  </dt>
  <dd>
    <ul>
<li><p>Do not use headers</p></li><li><p>4 columns, which are [chromosome  start-position(1-based)  stop-position(1-based)  gene_name]</p></li>
</ul>
  </dd>
  <dt>
    4. Clinical Data Files:
  </dt>
  <dd>
    <ul>
<li><p>Headers are required</p></li><li><p>At least 1 sample_id column and 1 attribute column, with the format being [sample_id  clinical_data_attribute  clinical_data_attribute  ...]</p></li><li><p>The sample_id must match the sample_id listed in the \s-1MAF\s0 under \*(L"Tumor_Sample_Barcode\*(R" for relating the mutations of this sample.</p></li><li><p>The header for each clinical_data_attribute will appear in the output file to denote relationships with the mutation data from the \s-1MAF\s0.</p></li>
</ul>
  </dd>

</dl>
<p>Descriptions for the usage of each tool (each sub-command) can be found separately.</p><p>The <strong>play</strong> command runs all of the sub-commands serially on a selected input set.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SUB-COMMANDS</h2>
        <div class="sectioncontent">
<h3>\s-1GENERAL\s0</h3>

<dl class='dl-vertical'>
  <dt>
    <strong>bmr</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Calculate gene coverages and background mutation rates.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>clinical-correlation</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Correlate phenotypic traits against mutated genes, or against individual variants
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>cosmic-omim</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Compare the amino acid changes of supplied mutations to \s-1COSMIC\s0 and \s-1OMIM\s0 databases.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>galaxy</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Run the full suite of MuSiC tools sequentially.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>mutation-relation</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Identify relationships of mutation concurrency or mutual exclusivity in genes across cases.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>path-scan</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Find signifcantly mutated pathways in a cohort given a list of somatic mutations.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>pfam</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Add Pfam annotation to a \s-1MAF\s0 file
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>play</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Run the full suite of MuSiC tools sequentially.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>plot</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Generate relevant plots and visualizations for MuSiC.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>proximity</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Perform a proximity analysis on a list of mutations.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>smg</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Identify significantly mutated genes.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

<dl class='dl-vertical'>
  <dt>
    <strong>survival</strong>
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Create survival plots and P-values for clinical and mutational phenotypes.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>


        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">LICENSE</h2>
        <div class="sectioncontent">
<p>Copyright (C) 2007-2011 Washington University in St. Louis.</p><p>It is released under the Lesser \s-1GNU\s0 Public License (\s-1LGPL\s0) version 3.  See the associated \s-1LICENSE\s0 file in this distribution.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p>This software is developed by the analysis and engineering teams at The Genome Institute at Washington University School of Medicine in St. Louis, with funding from the National Human Genome Research Institute.  Richard K. Wilson, P.I.</p><p>The primary authors of the MuSiC suite are:</p>
<pre>
 Nathan D. Dees, Ph.D.
 Cyriac Kandoth, Ph.D.
 Dan Koboldt, M.S.
 William Schierding, M.S.
 Michael Wendl, Ph.D.
 Qunyuan Zhang, Ph.D.
 Thomas B. Mooney, M.S.
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">CREDITS</h2>
        <div class="sectioncontent">
<p>The MuSiC suite uses tabix, by Heng Li.  See http://samtools.sourceforge.net/tabix.shtml.</p><p>MuSiC depends on copies of data from the following databases, packaged in a form useable for quick analysis:</p><p> * KEGG - http://www.genome.jp/kegg/  * COSMIC - http://www.sanger.ac.uk/genetics/CGP/cosmic/  * OMIM - http://www.ncbi.nlm.nih.gov/omim  * Pfam - http://pfam.sanger.ac.uk/  * SMART - http://smart.embl-heidelberg.de/  * SUPERFAMILY - http://supfam.cs.bris.ac.uk/SUPERFAMILY/  * PatternScan - http://www.expasy.ch/prosite/</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>For defects with any software in the genome namespace, contact  genome-dev ~at~ genome.wustl.edu.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO genome-music&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man1/genome.1.html"><strong>genome</strong>(1)</a></p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="genlist.1p.html"><span aria-hidden="true">&larr;</span> genlist.1p: </a></li>
   <li class="next"><a href="genome-music-bmr.1p.html">genome-music-bmr.1p: Calculate gene coverages and background mutation rates. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
