<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>gmt-music-smg: Identify significantly mutated genes.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Identify significantly mutated genes.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="gmt-music-smg (1p) manual">
  <meta name="twitter:description" content="Identify significantly mutated genes.">
  <meta name="twitter:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-gmt-music-smg-1p.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1p/gmt-music-smg.1p.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="gmt-music-smg (1p) manual" />
  <meta property="og:description" content="Identify significantly mutated genes." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-gmt-music-smg-1p.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">gmt-music-smg<small> (1p)</small></h1>
        <p class="lead">Identify significantly mutated genes.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/gmt-music-smg.1p.html">
      <span itemprop="name">gmt-music-smg: Identify significantly mutated genes.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libgenome-model-tools-music-perl/">
      <span itemprop="name">libgenome-model-tools-music-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/gmt-music-smg.1p.html">
      <span itemprop="name">gmt-music-smg: Identify significantly mutated genes.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">gmt music smg</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">VERSION</h2>
        <div class="sectioncontent">
<p>This document describes gmt music smg version 0.04 (2013-05-14 at 16:03:04)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>gmt music smg --gene-mr-file=? --output-file=? [--max-fdr=?] [--skip-low-mr-genes] [--bmr-modifier-file=?] [--processors=?]</p>
<pre>
 ... music smg &#92;
      --gene-mr-file output_dir/gene_mrs &#92;
      --output-file output_dir/smgs
</pre>
<p>(A \*(L"gene-mr-file\*(R" can be generated using the tool \*(L"music bmr calc-bmr\*(R".)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REQUIRED ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    gene-mr-file  <em>Text</em>
  </dt>
  <dd>
    <p>File with per-gene mutation rates (Created using \*(L"music bmr calc-bmr\*(R")</p>
  </dd>
  <dt>
    output-file  <em>Text</em>
  </dt>
  <dd>
    <p>Output file that will list significantly mutated genes and their p-values</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONAL ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    max-fdr  <em>Number</em>
  </dt>
  <dd>
    <p>The maximum allowed false discovery rate for a gene to be considered an \s-1SMG\s0 Default value '0.2' if not specified</p>
  </dd>
  <dt>
    skip-low-mr-genes  <em>Boolean</em>
  </dt>
  <dd>
    <p>Skip testing genes with MRs lower than the background \s-1MR\s0 Default value 'true' if not specified</p>
  </dd>
  <dt>
    bmr-modifier-file  <em>Text</em>
  </dt>
  <dd>
    <p>Tab delimited multipliers per gene that modify \s-1BMR\s0 before testing [gene_name bmr_modifier]</p>
  </dd>
  <dt>
    processors  <em>Integer</em>
  </dt>
  <dd>
    <p>Number of processors to use (requires 'foreach' and 'doMC' R packages) Default value '1' if not specified</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This script runs R-based statistical tools to identify Significantly Mutated Genes (SMGs), when given per-gene mutation rates categorized by mutation type, and the overall background mutation rates (BMRs) for each of those categories (gene_mr_file, created using \*(L"music bmr calc-bmr\*(R").</p><p>P-values and false discovery rates (FDRs) for each gene in gene_mr_file is calculated using three tests: Fisher's Combined P-value test (\s-1FCPT\s0), Likelihood Ratio test (\s-1LRT\s0), and the Convolution test (\s-1CT\s0). For a gene, if its \s-1FDR\s0 for at least 2 of these tests is &lt;= max_fdr, it will be output as an \s-1SMG\s0. Another output file with prefix \*(L"_detailed\*(R" will have p-values and FDRs for all genes.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    --bmr-modifier-file
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    The user can provide a \s-1BMR\s0 modifier for each gene in the \s-1ROI\s0 file, which is a multiplier for the categorized background mutation rates, before testing them against the gene's categorized mutation rates. Such a file can be used to correct for regional or systematic bias in mutation rates across the genome that may be correlated to CpG deamination or \s-1DNA\s0 repair processes like transcription-coupled repair or mismatch repair. Mutation rates have also been associated with \s-1DNA\s0 replication timing, where higher mutation rates are seen in late replicating regions. Note that the same per-gene multiplier is used on each mutation category of \s-1BMR\s0. Any genes from the \s-1ROI\s0 file that are not in the \s-1BMR\s0 modifier file will be tested against unmodified overall BMRs per mutation category. \s-1BMR\s0 modifiers of &lt;=0 are not permitted, because that's just silly.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>
  <dt>
    --skip-low-mr-genes
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Genes with consistently lower MRs than the BMRs across mutation categories, may show up in the results as an \s-1SMG\s0 (by \s-1CT\s0 or \s-1LRT\s0). If such genes are not of interest, they may be assigned a p-value of 1. This should also speed things up. Genes with higher Indel or Truncation rates than the background will not be skipped even if the gene's overall \s-1MR\s0 is lower than the \s-1BMR\s0. If bmr-modifiers are applied, this step uses the modified BMRs instead.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p> Qunyuan Zhang, Ph.D.  Cyriac Kandoth, Ph.D.  Nathan D. Dees, Ph.D.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="gmt-music-proximity.1p.html"><span aria-hidden="true">&larr;</span> gmt-music-proximity.1p: Perform a proximity analysis on a list of mutations.</a></li>
   <li class="next"><a href="gmt-music-survival.1p.html">gmt-music-survival.1p: Create survival plots and p-values for clinical and mutational phenotypes. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
