<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>gmt-music-bmr-calc-covg: Uses calcroicovg.c to count covered bases per-gene for each given tumor-normal pair of bams.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Uses calcroicovg.c to count covered bases per-gene for each given tumor-normal pair of bams.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="gmt-music-bmr-calc-covg (1p) manual">
  <meta name="twitter:description" content="Uses calcroicovg.c to count covered bases per-gene for each given tumor-normal pair of bams.">
  <meta name="twitter:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-gmt-music-bmr-calc-covg-1p.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1p/gmt-music-bmr-calc-covg.1p.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="gmt-music-bmr-calc-covg (1p) manual" />
  <meta property="og:description" content="Uses calcroicovg.c to count covered bases per-gene for each given tumor-normal pair of bams." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-gmt-music-bmr-calc-covg-1p.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">gmt-music-bmr-calc-covg<small> (1p)</small></h1>
        <p class="lead">Uses calcroicovg.c to count covered bases per-gene for each given tumor-normal pair of bams.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/gmt-music-bmr-calc-covg.1p.html">
      <span itemprop="name">gmt-music-bmr-calc-covg: Uses calcroicovg.c to count covered bases per-gene for each given tumor-normal pair of bams.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libgenome-model-tools-music-perl/">
      <span itemprop="name">libgenome-model-tools-music-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/gmt-music-bmr-calc-covg.1p.html">
      <span itemprop="name">gmt-music-bmr-calc-covg: Uses calcroicovg.c to count covered bases per-gene for each given tumor-normal pair of bams.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">gmt music bmr calc-covg</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">VERSION</h2>
        <div class="sectioncontent">
<p>This document describes gmt music bmr calc-covg version 0.04 (2013-05-14 at 16:03:05)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>gmt music bmr calc-covg --roi-file=? --reference-sequence=? --bam-list=? --output-dir=? [--cmd-list-file=?] [--cmd-prefix=?] [--normal-min-depth=?] [--tumor-min-depth=?] [--min-mapq=?]</p><p>General usage:</p>
<pre>
 ... music bmr calc-covg &#92;
    --bam-list input_dir/bam_list &#92;
    --output-dir output_dir/ &#92;
    --reference-sequence input_dir/all_sequences.fa &#92;
    --roi-file input_dir/all_coding_exons.tsv
</pre>
<p>To create a list of commands that will allow the processing of each tumor-normal pair in parallel with an \s-1LSF\s0 job scheduler:</p><p> ... music bmr calc-covg &#92;     --bam-list input_dir/bam_list &#92;     --output-dir output_dir/ &#92;     --reference-sequence input_dir/all_sequences.fa &#92;     --roi-file input_dir/all_coding_exons.tsv &#92;     --cmd_list_file parallelizable_commands &#92;     --cmd_prefix bsub</p><p>In the above case, the commands printed into the output file \*(L"parallelizable_commands\*(R" can be run in parallel. After they complete, rerun this script as printed directly below (--cmd_list_file and --cmd_prefix have been removed) to merge the parallelized calculations:</p><p> ... music bmr calc-covg &#92;     --bam-list input_dir/bam_list &#92;     --output-dir output_dir/ &#92;     --reference-sequence input_dir/all_sequences.fa &#92;     --roi-file input_dir/all_coding_exons.tsv</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REQUIRED ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    roi-file  <em>Text</em>
  </dt>
  <dd>
    <p>Tab delimited list of ROIs [chr start stop gene_name] (See Description)</p>
  </dd>
  <dt>
    reference-sequence  <em>Text</em>
  </dt>
  <dd>
    <p>Path to reference sequence in \s-1FASTA\s0 format</p>
  </dd>
  <dt>
    bam-list  <em>Text</em>
  </dt>
  <dd>
    <p>Tab delimited list of \s-1BAM\s0 files [sample_name normal_bam tumor_bam] (See Description)</p>
  </dd>
  <dt>
    output-dir  <em>Text</em>
  </dt>
  <dd>
    <p>Directory where output files and subdirectories will be written</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONAL ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    cmd-list-file  <em>Text</em>
  </dt>
  <dd>
    <p>A file to write calcRoiCovg commands to (See Description)</p>
  </dd>
  <dt>
    cmd-prefix  <em>Text</em>
  </dt>
  <dd>
    <p>A command that submits a job to your cluster (See Description)</p>
  </dd>
  <dt>
    normal-min-depth  <em>Integer</em>
  </dt>
  <dd>
    <p>The minimum read depth to consider a Normal \s-1BAM\s0 base as covered</p>
  </dd>
  <dt>
    tumor-min-depth  <em>Integer</em>
  </dt>
  <dd>
    <p>The minimum read depth to consider a Tumor \s-1BAM\s0 base as covered</p>
  </dd>
  <dt>
    min-mapq  <em>Integer</em>
  </dt>
  <dd>
    <p>The minimum mapping quality of reads to consider towards read depth counts</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This script counts bases with sufficient coverage in the ROIs of each gene in the given pairs of tumor-normal \s-1BAM\s0 files and categorizes them into - \s-1AT\s0, \s-1CG\s0 (non-CpG), and CpG counts. It also adds up these base-counts across all ROIs of each gene for each sample, but covered bases that lie within overlapping ROIs are not counted more than once towards these total counts.</p><p>By default, this script runs a C-based tool named calcRoiCovg for each sample one after another, taking ~30 mins per sample to generate per-ROI covered base counts. If the results of calcRoiCovg for a sample already exists in the output subdirectory roi_covgs, re-calculation is skipped. This allows you to run your own calcRoiCovg jobs in parallel or on multiple machines (Keep reading).</p><p>Speed things up by running calcRoiCovg jobs in parallel: If a compute cluster or multiple machines are available, run this script twice as follows:</p><ul>
<li><p>Define cmd-list-file and cmd-prefix to generate a file with commands that can be submitted to a cluster or run manually. These jobs will write per-ROI base counts in a subdirectory roi_covgs.</p></li><li><p>After all the parallelized calcRoiCovg jobs are completed, run this script again to add them up and generate the final per-gene base counts in a subdirectory gene_covgs. Remember to remove the cmd-list-file and cmd-prefix arguments or you will just be re-creating a list of commands.</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    --roi-file
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    The regions of interest (ROIs) of each gene are typically regions targeted for sequencing or are merged exon loci (from multiple transcripts) of genes with 2-bp flanks (splice junctions). ROIs from the same chromosome must be listed adjacent to each other in this file. This allows the underlying C-based code to run much more efficiently and avoid re-counting bases seen in overlapping ROIs (for overall covered base counts). For per-gene base counts, an overlapping base will be counted each time it appears in an \s-1ROI\s0 of the same gene. To avoid this, be sure to merge together overlapping ROIs of the same gene. BEDtools' mergeBed can help if used per gene.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>
  <dt>
    --reference-sequence
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    The reference sequence in \s-1FASTA\s0 format. If a reference sequence index is not found next to this file (a .fai file), it will be created.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>
  <dt>
    --bam-list
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Provide a file containing sample names and normal/tumor \s-1BAM\s0 locations for each. Use the tab- delimited format [sample_name normal_bam tumor_bam] per line. Additional columns like clinical data are allowed, but ignored. The sample_name must be the same as the tumor sample names used in the \s-1MAF\s0 file (16th column, with the header Tumor_Sample_Barcode).
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>
  <dt>
    --output-dir
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Specify an output directory where the following will be created/written: roi_covgs: Subdirectory containing per-ROI covered base counts for each sample. gene_covgs: Subdirectory containing per-gene covered base counts for each sample. total_covgs: File containing the overall non-overlapping coverages per sample.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>
  <dt>
    --cmd-list-file
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Specify a file into which a list of calcRoiCovg jobs will be written to. These can be scheduled in parallel, and will write per-ROI covered base-counts into the output subdirectory roi_covgs. If cmd-list-file is left unspecified, this script runs calcRoiCovg per sample one after another, taking ~30 mins per sample, but it skips samples whose output is already in roi_covgs.
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>
  <dt>
    --cmd-prefix
  </dt>
  <dd>
    
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">LICENSE</h2>
        <div class="sectioncontent">
<p>Copyright (C) 2010-2011 Washington University in St. Louis.</p><p>It is released under the Lesser \s-1GNU\s0 Public License (\s-1LGPL\s0) version 3.  See the associated \s-1LICENSE\s0 file in this distribution.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p> Cyriac Kandoth, Ph.D.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO gmt-music-bmr-calc-covg&hellip;</h2>
        <div class="sectioncontent">
<p><strong>genome-music-bmr</strong>(1), <strong>genome-music</strong>(1), <a href="../man1/genome.1.html"><strong>genome</strong>(1)</a></p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="gmt-music-bmr-calc-bmr.1p.html"><span aria-hidden="true">&larr;</span> gmt-music-bmr-calc-bmr.1p: Calculates mutation rates given per-gene coverage (from "music bmr calc-covg"), and a mutation list</a></li>
   <li class="next"><a href="gmt-music-bmr-calc-covg-helper.1p.html">gmt-music-bmr-calc-covg-helper.1p: Uses calcroicovg.c to count covered bases per-gene for a tumor-normal pair of bams. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
