<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>gmt-music-survival: Create survival plots and p-values for clinical and mutational phenotypes.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Create survival plots and p-values for clinical and mutational phenotypes.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="gmt-music-survival (1p) manual">
  <meta name="twitter:description" content="Create survival plots and p-values for clinical and mutational phenotypes.">
  <meta name="twitter:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-gmt-music-survival-1p.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1p/gmt-music-survival.1p.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="gmt-music-survival (1p) manual" />
  <meta property="og:description" content="Create survival plots and p-values for clinical and mutational phenotypes." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/libgenome-model-tools-music-perl-gmt-music-survival-1p.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">gmt-music-survival<small> (1p)</small></h1>
        <p class="lead">Create survival plots and p-values for clinical and mutational phenotypes.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/gmt-music-survival.1p.html">
      <span itemprop="name">gmt-music-survival: Create survival plots and p-values for clinical and mutational phenotypes.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/libgenome-model-tools-music-perl/">
      <span itemprop="name">libgenome-model-tools-music-perl</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1p/gmt-music-survival.1p.html">
      <span itemprop="name">gmt-music-survival: Create survival plots and p-values for clinical and mutational phenotypes.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">gmt music survival</h2>
        <div class="sectioncontent">

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">VERSION</h2>
        <div class="sectioncontent">
<p>This document describes gmt music survival version 0.04 (2013-05-14 at 16:03:05)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>gmt music survival --bam-list=? --output-dir=? [--maf-file=?] [--skip-silent] [--genetic-data-type=?] [--numeric-clinical-data-file=?] [--categorical-clinical-data-file=?] [--glm-clinical-data-file=?] [--phenotypes-to-include=?] [--legend-placement=?] [--skip-non-coding]</p>
<pre>
 ... music survival &#92;
        --bam-list /path/myBamList.tsv &#92;
        --maf-file /path/myMAF.tsv &#92;
        --numeric-clinical-data-file /path/myNumericData.tsv &#92;
        --categorical-clinical-data-file /path/myClassData.tsv &#92;
        --output-dir /path/output_directory

 ... music survival &#92;
        --bam-list /path/myBamList.tsv &#92;
        --maf-file /path/myMAF.tsv &#92;
        --glm-clinical-data-file /path/myGLMClinicalData.tsv &#92;
        --output-dir /path/output_directory

 ... music survival &#92;
        --bam-list /path/myBamList.tsv &#92;
        --maf-file /path/myMAF.tsv &#92;
        --genetic-data-type &apos;gene&apos; &#92;
        --glm-clinical-data-file /path/myGlmClinicalData.tsv &#92;
        --phenotypes-to-include &apos;Race,Gender,TP53&apos; &#92;
        --output-dir /path/output_directory
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REQUIRED ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    bam-list  <em>Text</em>
  </dt>
  <dd>
    <p>List of sample names to be included in the analysis. (See Description)</p>
  </dd>
  <dt>
    output-dir  <em>Text</em>
  </dt>
  <dd>
    <p>Directory where output files will be written</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONAL ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    maf-file  <em>Text</em>
  </dt>
  <dd>
    <p>List of mutations in \s-1MAF\s0 format</p>
  </dd>
  <dt>
    skip-silent  <em>Boolean</em>
  </dt>
  <dd>
    <p>Skip silent mutations from the provided \s-1MAF\s0 file Default value 'true' if not specified</p>
  </dd>
  <dt>
    genetic-data-type  <em>Text</em>
  </dt>
  <dd>
    <p>Correlate clinical data to \*(L"gene\*(R" or \*(L"variant\*(R" level data Default value 'gene' if not specified</p>
  </dd>
  <dt>
    numeric-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Table of samples (y) vs. numeric clinical data category (x)</p>
  </dd>
  <dt>
    categorical-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Table of samples (y) vs. categorical clinical data category (x)</p>
  </dd>
  <dt>
    glm-clinical-data-file  <em>Text</em>
  </dt>
  <dd>
    <p>Clinical traits, mutational profiles, other mixed clinical data (See \s-1DESCRIPTION\s0).</p>
  </dd>
  <dt>
    phenotypes-to-include  <em>Text</em>
  </dt>
  <dd>
    <p>Include only these genes and/or phenotypes in the anlaysis. (COMMA-DELIMITED)</p>
  </dd>
  <dt>
    legend-placement  <em>Text</em>
  </dt>
  <dd>
    <p>Choose one of 'bottomleft', 'topleft', 'topright', or 'bottomright'. Default value 'bottomleft' if not specified</p>
  </dd>
  <dt>
    skip-non-coding  <em>Boolean</em>
  </dt>
  <dd>
    <p>Skip non-coding mutations from the provided \s-1MAF\s0 file Default value 'true' if not specified</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This command performs survival analysis and plots survival curves for mutational data, as well as any clinical traits of interest as specified via the --phenotypes-to-include input parameter. The analyses performed include the Kaplan-Meier estimator followed by the Cox Proportional Hazards model. Outputs for each gene/clinical trait analyzed include survival curves, a hazard ratio (with confidence intervals), and P-values and FDRs describing the significance of the difference between survivors and non-survivors.</p><p>All clinical data files are searched for the required (case insensitive) \*(L"vital_status\*(R" and \*(L"days_to_last_followup\*(R" columns which are paired to phenotypes via sample IDs for the survival analysis. The first column of all clinical data files \s-1MUST\s0 contain the sample IDs, same as in other MuSiC tools. By default, analysis is performed on every gene present in the \s-1MAF\s0. Optionally, the analysis may be limited to only specific genes by listing them (comma delimited) after the --phenotypes-to-include input parameter. Survival analysis may also be performed on other columns in the clinical data file by adding the column headers to the list of entries specified after the --phenotypes-to-include input parameter.</p><p>Here are some general guildelines for creating clinical data input files:</p><ul>
<li><p>Headers are required.</p></li><li><p>The first column of each clinical data file must contain sample IDs which match those in both the --bam-list and the \s-1MAF\s0 variant list (in the \s-1MAF\s0, this is the Tumor_Sample_Barcode column, specifically).</p></li><li><p>In at least one of the clinical data files input, columns with headers \*(L"vital_status\*(R" and \*(L"days_to_last_followup\*(R" (case insensitive) must exist. \*(L"vital_status\*(R" must be delineated by 1's and 0's, where 0 denotes 'living', and 1 denotes 'deceased'.</p></li>
</ul><p>Note that all input files must be tab-separated.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ARGUMENTS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    --bam-list
  </dt>
  <dd>
    
<dl class='dl-vertical'>
  <dt>
    Provide a file containing sample names and normal/tumor \s-1BAM\s0 locations for each. Use the tab- delimited format [sample_name normal_bam tumor_bam] per line. This tool only needs sample_name, so all other columns can be skipped. The sample_name must be the same as the tumor sample names used in the \s-1MAF\s0 file (16th column, with the header Tumor_Sample_Barcode).
  </dt>
  <dd>
    
  </dd>

</dl>

  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">LICENSE</h2>
        <div class="sectioncontent">
<p>Copyright (C) 2010-2011 Washington University in St. Louis.</p><p>It is released under the Lesser \s-1GNU\s0 Public License (\s-1LGPL\s0) version 3.  See the associated \s-1LICENSE\s0 file in this distribution.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p> Nathan D. Dees, Ph.D.  Qunyuan Zhang, Ph.D.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO gmt-music-survival&hellip;</h2>
        <div class="sectioncontent">
<p><strong>genome-music</strong>(1), <a href="../man1/genome.1.html"><strong>genome</strong>(1)</a></p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="gmt-music-smg.1p.html"><span aria-hidden="true">&larr;</span> gmt-music-smg.1p: Identify significantly mutated genes.</a></li>
   <li class="next"><a href="gpgmailtunl.1p.html">gpgmailtunl.1p: Encrypts an email message into the body of another email. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
