<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>gt-hop: Cognate sequence-based homopolymer error correction.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Cognate sequence-based homopolymer error correction.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="gt-hop (1) manual">
  <meta name="twitter:description" content="Cognate sequence-based homopolymer error correction.">
  <meta name="twitter:image" content="https://www.carta.tech/images/genometools-gt-hop-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/gt-hop.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="gt-hop (1) manual" />
  <meta property="og:description" content="Cognate sequence-based homopolymer error correction." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/genometools-gt-hop-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">gt-hop<small> (1)</small></h1>
        <p class="lead">Cognate sequence-based homopolymer error correction.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/gt-hop.1.html">
      <span itemprop="name">gt-hop: Cognate sequence-based homopolymer error correction.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/genometools/">
      <span itemprop="name">genometools</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/gt-hop.1.html">
      <span itemprop="name">gt-hop: Cognate sequence-based homopolymer error correction.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>gt hop</strong> -&lt;mode&gt; -c &lt;encseq&gt; -map &lt;sam/bam&gt; -reads &lt;fastq&gt; [options...]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>-c</strong> [<em>string</em>]</p><p>cognate sequence (encoded using gt encseq encode) (default: undefined)</p><p><strong>-map</strong> [<em>string</em>]</p><p>mapping of reads to the cognate sequence it must be in SAM/BAM format, and sorted by coordinate (can be prepared e.g. using: samtools sort) (default: undefined)</p><p><strong>-sam</strong> [<em>yes|no</em>]</p><p>mapping file is SAM default: BAM (default: no)</p><p><strong>-aggressive</strong> [<em>yes|no</em>]</p><p>correct as much as possible (default: no)</p><p><strong>-moderate</strong> [<em>yes|no</em>]</p><p>mediate between sensitivity and precision (default: no)</p><p><strong>-conservative</strong> [<em>yes|no</em>]</p><p>correct only most likely errors (default: no)</p><p><strong>-expert</strong> [<em>yes|no</em>]</p><p>manually select correction criteria (default: no)</p><p><strong>-reads</strong></p><p>uncorrected read file(s) in FastQ format; the corrected reads are output in the currect working directory in files which are named as the input files, each prepended by a prefix (see -outprefix option) -reads allows one to output the reads in the same order as in the input and is mandatory if the SAM contains more than a single primary alignment for each read (e.g. output of bwasw) see also -o option as an alternative</p><p><strong>-outprefix</strong> [<em>string</em>]</p><p>prefix for output filenames (corrected reads)when -reads is specified the prefix is prepended to each input filename (default: hop_)</p><p><strong>-o</strong> [<em>string</em>]</p><p>output file for corrected reads (see also -reads/-outprefix) if -o is used, reads are output in a single file in the order they are found in the SAM file (which usually differ from the original order) this will only work if the reads were aligned with a software which only includes 1 alignment for each read (e.g. bwa) (default: undefined)</p><p><strong>-hmin</strong> [<em>value</em>]</p><p>minimal homopolymer length in cognate sequence (default: 3)</p><p><strong>-read-hmin</strong> [<em>value</em>]</p><p>minimal homopolymer length in reads (default: 2)</p><p><strong>-qmax</strong> [<em>value</em>]</p><p>maximal average quality of homopolymer in a read (default: 120)</p><p><strong>-altmax</strong> [<em>value</em>]</p><p>max support of alternate homopol. length; e.g. 0.8 means: do not correct any read if homop. length in more than 80% of the reads has the same value, different from the cognate if altmax is set to 1.0 reads are always corrected (default: 0.800000)</p><p><strong>-cogmin</strong> [<em>value</em>]</p><p>min support of cognate sequence homopol. length; e.g. 0.1 means: do not correct any read if cognate homop. length is not present in at least 10% of the reads if cogmin is set to 0.0 reads are always corrected (default: 0.100000)</p><p><strong>-mapqmin</strong> [<em>value</em>]</p><p>minimal mapping quality (default: 21)</p><p><strong>-covmin</strong> [<em>value</em>]</p><p>minimal coverage; e.g. 5 means: do not correct any read if coverage (number of reads mapped over whole homopolymer) is less than 5 if covmin is set to 1 reads are always corrected (default: 1)</p><p><strong>-allow-muliple</strong> [<em>yes|no</em>]</p><p>allow multiple corrections in a read (default: no)</p><p><strong>-clenmax</strong> [<em>value</em>]</p><p>maximal correction length default: unlimited (default: undefined)</p><p><strong>-ann</strong> [<em>string</em>]</p><p>annotation of cognate sequence it must be sorted by coordinates on the cognate sequence (this can be e.g. done using: gt gff3 -sort) if -ann is used, corrections will be limited to homopolymers startingor ending inside the feature type indicated by -ft optionformat: sorted GFF3 (default: undefined)</p><p><strong>-ft</strong> [<em>string</em>]</p><p>feature type to use when -ann option is specified (default: CDS)</p><p><strong>-v</strong> [<em>yes|no</em>]</p><p>be verbose (default: no)</p><p><strong>-help</strong></p><p>display help for basic options and exit</p><p><strong>-help+</strong></p><p>display help for all options and exit</p><p><strong>-version</strong></p><p>display version information and exit</p><p>Correction mode:</p><p>One of the options <em>-aggressive</em>, <em>-moderate</em>, <em>-conservative</em> or <em>-expert</em> must be selected.</p><p>The <em>-aggressive</em>, <em>-moderate</em> and <em>-conservative</em> modes are presets of the criteria by which it is decided if an observed discrepancy in homopolymer length between cognate sequence and a read shall be corrected or not. A description of the single criteria is provided by using the <em>-help+</em>&apos; option. The presets are equivalent to the following settings:</p>
<pre>
                    -aggressive    -moderate      -conservative
-hmin               3              3              3
-read-hmin          1              1              2
-altmax             1.00           0.99           0.80
-refmin             0.00           0.00           0.10
-mapqmin            0              10             21
-covmin             1              1              1
-clenmax            unlimited      unlimited      unlimited
-allow-multiple     yes            yes            no
</pre>
<p>The aggressive mode tries to maximize the sensitivity, the conservative mode to minimize the false positives. An even more conservative set of corrections can be achieved using the <em>-ann</em> option (see <em>-help+</em>).</p><p>The <em>-expert</em> mode allows one to manually set each parameter; the default values are the same as in the <em>-conservative</em> mode.</p><p>(Finally, for evaluation purposes only, the <em>-state-of-truth</em> mode can be used: this mode assumes that the sequenced genome has been specified as cognate sequence and outputs an ideal list of corrections.)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REPORTING BUGS</h2>
        <div class="sectioncontent">
<p>Report bugs to &lt;gt-users@genometools.org&gt;.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="gt-gtf_to_gff3.1.html"><span aria-hidden="true">&larr;</span> gt-gtf_to_gff3.1: Parse gtf2.2 file and convert it to gff3.</a></li>
   <li class="next"><a href="gt-id_to_md5.1.html">gt-id_to_md5.1: Change sequence ids in given gff3 files to md5 fingerprints of the corresponding sequences. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
