<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>gt-ltrdigest: Identifies and annotates sequence features in ltr retrotransposon candidates.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Identifies and annotates sequence features in ltr retrotransposon candidates.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="gt-ltrdigest (1) manual">
  <meta name="twitter:description" content="Identifies and annotates sequence features in ltr retrotransposon candidates.">
  <meta name="twitter:image" content="https://www.carta.tech/images/genometools-gt-ltrdigest-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/gt-ltrdigest.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="gt-ltrdigest (1) manual" />
  <meta property="og:description" content="Identifies and annotates sequence features in ltr retrotransposon candidates." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/genometools-gt-ltrdigest-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">gt-ltrdigest<small> (1)</small></h1>
        <p class="lead">Identifies and annotates sequence features in ltr retrotransposon candidates.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/gt-ltrdigest.1.html">
      <span itemprop="name">gt-ltrdigest: Identifies and annotates sequence features in ltr retrotransposon candidates.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/genometools/">
      <span itemprop="name">genometools</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/gt-ltrdigest.1.html">
      <span itemprop="name">gt-ltrdigest: Identifies and annotates sequence features in ltr retrotransposon candidates.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>gt ltrdigest</strong> [option ...] gff3_file</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>-outfileprefix</strong> [<em>string</em>]</p><p>prefix for output files (e.g. <em>foo</em> will create files called <em>foo_*.csv</em> and <em>foo_*.fas</em>) Omit this option for GFF3 output only. (default: undefined)</p><p><strong>-metadata</strong> [<em>yes|no</em>]</p><p>output metadata (run conditions) to separate file (default: yes)</p><p><strong>-seqnamelen</strong> [<em>value</em>]</p><p>set maximal length of sequence names in FASTA headers (e.g. for clustalw or similar tools) (default: 20)</p><p><strong>-pptlen</strong> [<em>start</em> <em>end</em>]</p><p>required PPT length range (default: [8..30])</p><p><strong>-uboxlen</strong> [<em>start</em> <em>end</em>]</p><p>required U-box length range (default: [3..30])</p><p><strong>-uboxdist</strong> [<em>value</em>]</p><p>allowed U-box distance range from PPT (default: 0)</p><p><strong>-pptradius</strong> [<em>value</em>]</p><p>radius around beginning of 3&apos; LTR to search for PPT (default: 30)</p><p><strong>-pptrprob</strong> [<em>value</em>]</p><p>purine emission probability inside PPT (default: 0.970000)</p><p><strong>-pptyprob</strong> [<em>value</em>]</p><p>pyrimidine emission probability inside PPT (default: 0.030000)</p><p><strong>-pptgprob</strong> [<em>value</em>]</p><p>background G emission probability outside PPT (default: 0.250000)</p><p><strong>-pptcprob</strong> [<em>value</em>]</p><p>background C emission probability outside PPT (default: 0.250000)</p><p><strong>-pptaprob</strong> [<em>value</em>]</p><p>background A emission probability outside PPT (default: 0.250000)</p><p><strong>-ppttprob</strong> [<em>value</em>]</p><p>background T emission probability outside PPT (default: 0.250000)</p><p><strong>-pptuprob</strong> [<em>value</em>]</p><p>U/T emission probability inside U-box (default: 0.910000)</p><p><strong>-trnas</strong> [<em>filename</em>]</p><p>tRNA library in multiple FASTA format for PBS detection Omit this option to disable PBS search. (default: undefined)</p><p><strong>-pbsalilen</strong> [<em>start</em> <em>end</em>]</p><p>required PBS/tRNA alignment length range (default: [11..30])</p><p><strong>-pbsoffset</strong> [<em>start</em> <em>end</em>]</p><p>allowed PBS offset from LTR boundary range (default: [0..5])</p><p><strong>-pbstrnaoffset</strong> [<em>start</em> <em>end</em>]</p><p>allowed PBS/tRNA 3&apos; end alignment offset range (default: [0..5])</p><p><strong>-pbsmaxedist</strong> [<em>value</em>]</p><p>maximal allowed PBS/tRNA alignment unit edit distance (default: 1)</p><p><strong>-pbsradius</strong> [<em>value</em>]</p><p>radius around end of 5&apos; LTR to search for PBS (default: 30)</p><p><strong>-hmms</strong></p><p>profile HMM models for domain detection (separate by spaces, finish with --) in HMMER3 format Omit this option to disable pHMM search.</p><p><strong>-pdomevalcutoff</strong> [<em>value</em>]</p><p>global E-value cutoff for pHMM search default 1E-6 (default: 0.000001)</p><p><strong>-pdomcutoff</strong> [<em>...</em>]</p><p>model-specific score cutoff choose from TC (trusted cutoff) | GA (gathering cutoff) | NONE (no cutoffs) (default: NONE)</p><p><strong>-aliout</strong> [<em>yes|no</em>]</p><p>output pHMM to amino acid sequence alignments (default: no)</p><p><strong>-aaout</strong> [<em>yes|no</em>]</p><p>output amino acid sequences for protein domain hits (default: no)</p><p><strong>-allchains</strong> [<em>yes|no</em>]</p><p>output features from all chains and unchained features, labeled with chain numbers (default: no)</p><p><strong>-maxgaplen</strong> [<em>value</em>]</p><p>maximal allowed gap size between fragments (in amino acids) when chaining pHMM hits for a protein domain (default: 50)</p><p><strong>-pbsmatchscore</strong> [<em>value</em>]</p><p>match score for PBS/tRNA alignments (default: 5)</p><p><strong>-pbsmismatchscore</strong> [<em>value</em>]</p><p>mismatch score for PBS/tRNA alignments (default: -10)</p><p><strong>-pbsinsertionscore</strong> [<em>value</em>]</p><p>insertion score for PBS/tRNA alignments (default: -20)</p><p><strong>-pbsdeletionscore</strong> [<em>value</em>]</p><p>deletion score for PBS/tRNA alignments (default: -20)</p><p><strong>-v</strong> [<em>yes|no</em>]</p><p>be verbose (default: no)</p><p><strong>-o</strong> [<em>filename</em>]</p><p>redirect output to specified file (default: undefined)</p><p><strong>-gzip</strong> [<em>yes|no</em>]</p><p>write gzip compressed output file (default: no)</p><p><strong>-bzip2</strong> [<em>yes|no</em>]</p><p>write bzip2 compressed output file (default: no)</p><p><strong>-force</strong> [<em>yes|no</em>]</p><p>force writing to output file (default: no)</p><p><strong>-seqfile</strong> [<em>filename</em>]</p><p>set the sequence file from which to take the sequences (default: undefined)</p><p><strong>-encseq</strong> [<em>filename</em>]</p><p>set the encoded sequence indexname from which to take the sequences (default: undefined)</p><p><strong>-seqfiles</strong></p><p>set the sequence files from which to extract the features use <em>--</em> to terminate the list of sequence files</p><p><strong>-matchdesc</strong> [<em>yes|no</em>]</p><p>search the sequence descriptions from the input files for the desired sequence IDs (in GFF3), reporting the first match (default: no)</p><p><strong>-matchdescstart</strong> [<em>yes|no</em>]</p><p>exactly match the sequence descriptions from the input files for the desired sequence IDs (in GFF3) from the beginning to the first whitespace (default: no)</p><p><strong>-usedesc</strong> [<em>yes|no</em>]</p><p>use sequence descriptions to map the sequence IDs (in GFF3) to actual sequence entries. If a description contains a sequence range (e.g., III:1000001..2000000), the first part is used as sequence ID (<em>III</em>) and the first range position as offset (<em>1000001</em>) (default: no)</p><p><strong>-regionmapping</strong> [<em>string</em>]</p><p>set file containing sequence-region to sequence file mapping (default: undefined)</p><p><strong>-help</strong></p><p>display help for basic options and exit</p><p><strong>-help+</strong></p><p>display help for all options and exit</p><p><strong>-version</strong></p><p>display version information and exit</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REPORTING BUGS</h2>
        <div class="sectioncontent">
<p>Report bugs to &lt;gt-users@genometools.org&gt;.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="gt-ltrclustering.1.html"><span aria-hidden="true">&larr;</span> gt-ltrclustering.1: Cluster features of ltrs.</a></li>
   <li class="next"><a href="gt-ltrharvest.1.html">gt-ltrharvest.1: Predict ltr retrotransposons. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
