<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>gustaf: Gustaf - generic multi-split alignment finder: tool for split-read mapping allowing multiple splits.  synopsis gustaf [options] genome fasta file read fasta file  description gustaf uses seqans stellar to find splits as local matches on different strands or chromosomes. criteria and penalties to chain these matches can be specified. output file contains the breakpoints along the best chain. the genome file is used as database input, the read file as query input. all stellar options are supported. see stellar documentation for stellar parameters and options. (c) 2011-2012 by kathrin trappe -h, --help displays this help message. --version display version information gustaf options: main options: -tp, --transpen int interchromosomal translocation penalty default: 5. -ip, --invpen int inversion penalty default: 5. -op, --orderpen int intrachromosomal order change penalty default: 0. -oth, --overlapthresh double allowed overlap between matches default: 0.5. -gth, --gapthresh int allowed gap length between matches default: 10. -ith, --initgapthresh int allowed initial or ending gap length at begin and end of read default: 15. -st, --support int number of supporting reads default: 2. input options: -m, --matchfile file file of (stellar) matches valid filetypes are: gff and gff. output options: -bpo, --breakpointout file name of breakpoint output file. valid filetypes are: gff and txt. default: breakpoints.gff. -j, --jobname str job/queue name default: . -do, --dots enable graph output in dot format stellar options: main options: -e, --epsilon num maximal error rate (max 0.25). in range [0.0000001..0.25]. default: 0.05. -l, --minlength num minimal length of epsilon-matches. in range [0..inf]. default: 100. -f, --forward search only in forward strand of database. -r, --reverse search only in reverse complement of database. -a, --alphabet str alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). one of dna, dna5, rna, rna5, protein, and char. -v, --verbose set verbosity mode. filtering options: -k, --kmer num length of the q-grams (max 32). in range [1..32]. -rp, --repeatperiod num maximal period of low complexity repeats to be filtered. default: 1. -rl, --repeatlength num minimal length of low complexity repeats to be filtered. default: 1000. -c, --abundancecut num k-mer overabundance cut ratio. in range [0..1]. default: 1. verification options: -x, --xdrop num maximal x-drop for extension. default: 5. -vs, --verification str verification strategy: exact or bestlocal or bandedglobal one of exact, bestlocal, and bandedglobal. default: exact. -dt, --disablethresh num maximal number of verified matches before disabling verification for one query sequence (default infinity). in range [0..inf]. -n, --nummatches num maximal number of kept matches per query and database. if stellar finds more matches, only the longest ones are kept. default: 50. -s, --sortthresh num number of matches triggering removal of duplicates. choose a smaller value for saving space. default: 500.  version gustaf version: 1.0 last update july 2012</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Gustaf - generic multi-split alignment finder: tool for split-read mapping allowing multiple splits.  synopsis gustaf [options] genome fasta file read fasta file  description gustaf uses seqans stellar to find splits as local matches on different strands or chromosomes. criteria and penalties to chain these matches can be specified. output file contains the breakpoints along the best chain. the genome file is used as database input, the read file as query input. all stellar options are supported. see stellar documentation for stellar parameters and options. (c) 2011-2012 by kathrin trappe -h, --help displays this help message. --version display version information gustaf options: main options: -tp, --transpen int interchromosomal translocation penalty default: 5. -ip, --invpen int inversion penalty default: 5. -op, --orderpen int intrachromosomal order change penalty default: 0. -oth, --overlapthresh double allowed overlap between matches default: 0.5. -gth, --gapthresh int allowed gap length between matches default: 10. -ith, --initgapthresh int allowed initial or ending gap length at begin and end of read default: 15. -st, --support int number of supporting reads default: 2. input options: -m, --matchfile file file of (stellar) matches valid filetypes are: gff and gff. output options: -bpo, --breakpointout file name of breakpoint output file. valid filetypes are: gff and txt. default: breakpoints.gff. -j, --jobname str job/queue name default: . -do, --dots enable graph output in dot format stellar options: main options: -e, --epsilon num maximal error rate (max 0.25). in range [0.0000001..0.25]. default: 0.05. -l, --minlength num minimal length of epsilon-matches. in range [0..inf]. default: 100. -f, --forward search only in forward strand of database. -r, --reverse search only in reverse complement of database. -a, --alphabet str alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). one of dna, dna5, rna, rna5, protein, and char. -v, --verbose set verbosity mode. filtering options: -k, --kmer num length of the q-grams (max 32). in range [1..32]. -rp, --repeatperiod num maximal period of low complexity repeats to be filtered. default: 1. -rl, --repeatlength num minimal length of low complexity repeats to be filtered. default: 1000. -c, --abundancecut num k-mer overabundance cut ratio. in range [0..1]. default: 1. verification options: -x, --xdrop num maximal x-drop for extension. default: 5. -vs, --verification str verification strategy: exact or bestlocal or bandedglobal one of exact, bestlocal, and bandedglobal. default: exact. -dt, --disablethresh num maximal number of verified matches before disabling verification for one query sequence (default infinity). in range [0..inf]. -n, --nummatches num maximal number of kept matches per query and database. if stellar finds more matches, only the longest ones are kept. default: 50. -s, --sortthresh num number of matches triggering removal of duplicates. choose a smaller value for saving space. default: 500.  version gustaf version: 1.0 last update july 2012">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="gustaf (1) manual">
  <meta name="twitter:description" content="Gustaf - generic multi-split alignment finder: tool for split-read mapping allowing multiple splits.  synopsis gustaf [options] genome fasta file read fasta file  description gustaf uses seqans stellar to find splits as local matches on different strands or chromosomes. criteria and penalties to chain these matches can be specified. output file contains the breakpoints along the best chain. the genome file is used as database input, the read file as query input. all stellar options are supported. see stellar documentation for stellar parameters and options. (c) 2011-2012 by kathrin trappe -h, --help displays this help message. --version display version information gustaf options: main options: -tp, --transpen int interchromosomal translocation penalty default: 5. -ip, --invpen int inversion penalty default: 5. -op, --orderpen int intrachromosomal order change penalty default: 0. -oth, --overlapthresh double allowed overlap between matches default: 0.5. -gth, --gapthresh int allowed gap length between matches default: 10. -ith, --initgapthresh int allowed initial or ending gap length at begin and end of read default: 15. -st, --support int number of supporting reads default: 2. input options: -m, --matchfile file file of (stellar) matches valid filetypes are: gff and gff. output options: -bpo, --breakpointout file name of breakpoint output file. valid filetypes are: gff and txt. default: breakpoints.gff. -j, --jobname str job/queue name default: . -do, --dots enable graph output in dot format stellar options: main options: -e, --epsilon num maximal error rate (max 0.25). in range [0.0000001..0.25]. default: 0.05. -l, --minlength num minimal length of epsilon-matches. in range [0..inf]. default: 100. -f, --forward search only in forward strand of database. -r, --reverse search only in reverse complement of database. -a, --alphabet str alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). one of dna, dna5, rna, rna5, protein, and char. -v, --verbose set verbosity mode. filtering options: -k, --kmer num length of the q-grams (max 32). in range [1..32]. -rp, --repeatperiod num maximal period of low complexity repeats to be filtered. default: 1. -rl, --repeatlength num minimal length of low complexity repeats to be filtered. default: 1000. -c, --abundancecut num k-mer overabundance cut ratio. in range [0..1]. default: 1. verification options: -x, --xdrop num maximal x-drop for extension. default: 5. -vs, --verification str verification strategy: exact or bestlocal or bandedglobal one of exact, bestlocal, and bandedglobal. default: exact. -dt, --disablethresh num maximal number of verified matches before disabling verification for one query sequence (default infinity). in range [0..inf]. -n, --nummatches num maximal number of kept matches per query and database. if stellar finds more matches, only the longest ones are kept. default: 50. -s, --sortthresh num number of matches triggering removal of duplicates. choose a smaller value for saving space. default: 500.  version gustaf version: 1.0 last update july 2012">
  <meta name="twitter:image" content="https://www.carta.tech/images/seqan-apps-gustaf-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/gustaf.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="gustaf (1) manual" />
  <meta property="og:description" content="Gustaf - generic multi-split alignment finder: tool for split-read mapping allowing multiple splits.  synopsis gustaf [options] genome fasta file read fasta file  description gustaf uses seqans stellar to find splits as local matches on different strands or chromosomes. criteria and penalties to chain these matches can be specified. output file contains the breakpoints along the best chain. the genome file is used as database input, the read file as query input. all stellar options are supported. see stellar documentation for stellar parameters and options. (c) 2011-2012 by kathrin trappe -h, --help displays this help message. --version display version information gustaf options: main options: -tp, --transpen int interchromosomal translocation penalty default: 5. -ip, --invpen int inversion penalty default: 5. -op, --orderpen int intrachromosomal order change penalty default: 0. -oth, --overlapthresh double allowed overlap between matches default: 0.5. -gth, --gapthresh int allowed gap length between matches default: 10. -ith, --initgapthresh int allowed initial or ending gap length at begin and end of read default: 15. -st, --support int number of supporting reads default: 2. input options: -m, --matchfile file file of (stellar) matches valid filetypes are: gff and gff. output options: -bpo, --breakpointout file name of breakpoint output file. valid filetypes are: gff and txt. default: breakpoints.gff. -j, --jobname str job/queue name default: . -do, --dots enable graph output in dot format stellar options: main options: -e, --epsilon num maximal error rate (max 0.25). in range [0.0000001..0.25]. default: 0.05. -l, --minlength num minimal length of epsilon-matches. in range [0..inf]. default: 100. -f, --forward search only in forward strand of database. -r, --reverse search only in reverse complement of database. -a, --alphabet str alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). one of dna, dna5, rna, rna5, protein, and char. -v, --verbose set verbosity mode. filtering options: -k, --kmer num length of the q-grams (max 32). in range [1..32]. -rp, --repeatperiod num maximal period of low complexity repeats to be filtered. default: 1. -rl, --repeatlength num minimal length of low complexity repeats to be filtered. default: 1000. -c, --abundancecut num k-mer overabundance cut ratio. in range [0..1]. default: 1. verification options: -x, --xdrop num maximal x-drop for extension. default: 5. -vs, --verification str verification strategy: exact or bestlocal or bandedglobal one of exact, bestlocal, and bandedglobal. default: exact. -dt, --disablethresh num maximal number of verified matches before disabling verification for one query sequence (default infinity). in range [0..inf]. -n, --nummatches num maximal number of kept matches per query and database. if stellar finds more matches, only the longest ones are kept. default: 50. -s, --sortthresh num number of matches triggering removal of duplicates. choose a smaller value for saving space. default: 500.  version gustaf version: 1.0 last update july 2012" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/seqan-apps-gustaf-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">gustaf<small> (1)</small></h1>
        <p class="lead">Gustaf - generic multi-split alignment finder: tool for split-read mapping allowing multiple splits.  synopsis gustaf [options] genome fasta file read fasta file  description gustaf uses seqans stellar to find splits as local matches on different strands or chromosomes. criteria and penalties to chain these matches can be specified. output file contains the breakpoints along the best chain. the genome file is used as database input, the read file as query input. all stellar options are supported. see stellar documentation for stellar parameters and options. (c) 2011-2012 by kathrin trappe -h, --help displays this help message. --version display version information gustaf options: main options: -tp, --transpen int interchromosomal translocation penalty default: 5. -ip, --invpen int inversion penalty default: 5. -op, --orderpen int intrachromosomal order change penalty default: 0. -oth, --overlapthresh double allowed overlap between matches default: 0.5. -gth, --gapthresh int allowed gap length between matches default: 10. -ith, --initgapthresh int allowed initial or ending gap length at begin and end of read default: 15. -st, --support int number of supporting reads default: 2. input options: -m, --matchfile file file of (stellar) matches valid filetypes are: gff and gff. output options: -bpo, --breakpointout file name of breakpoint output file. valid filetypes are: gff and txt. default: breakpoints.gff. -j, --jobname str job/queue name default: . -do, --dots enable graph output in dot format stellar options: main options: -e, --epsilon num maximal error rate (max 0.25). in range [0.0000001..0.25]. default: 0.05. -l, --minlength num minimal length of epsilon-matches. in range [0..inf]. default: 100. -f, --forward search only in forward strand of database. -r, --reverse search only in reverse complement of database. -a, --alphabet str alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). one of dna, dna5, rna, rna5, protein, and char. -v, --verbose set verbosity mode. filtering options: -k, --kmer num length of the q-grams (max 32). in range [1..32]. -rp, --repeatperiod num maximal period of low complexity repeats to be filtered. default: 1. -rl, --repeatlength num minimal length of low complexity repeats to be filtered. default: 1000. -c, --abundancecut num k-mer overabundance cut ratio. in range [0..1]. default: 1. verification options: -x, --xdrop num maximal x-drop for extension. default: 5. -vs, --verification str verification strategy: exact or bestlocal or bandedglobal one of exact, bestlocal, and bandedglobal. default: exact. -dt, --disablethresh num maximal number of verified matches before disabling verification for one query sequence (default infinity). in range [0..inf]. -n, --nummatches num maximal number of kept matches per query and database. if stellar finds more matches, only the longest ones are kept. default: 50. -s, --sortthresh num number of matches triggering removal of duplicates. choose a smaller value for saving space. default: 500.  version gustaf version: 1.0 last update july 2012</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/gustaf.1.html">
      <span itemprop="name">gustaf: Gustaf - generic multi-split alignment finder: tool for split-read mapping allowing multiple splits.  synopsis gustaf [options] genome fasta file read fasta file  description gustaf uses seqans stellar to find splits as local matches on different strands or chromosomes. criteria and penalties to chain these matches can be specified. output file contains the breakpoints along the best chain. the genome file is used as database input, the read file as query input. all stellar options are supported. see stellar documentation for stellar parameters and options. (c) 2011-2012 by kathrin trappe -h, --help displays this help message. --version display version information gustaf options: main options: -tp, --transpen int interchromosomal translocation penalty default: 5. -ip, --invpen int inversion penalty default: 5. -op, --orderpen int intrachromosomal order change penalty default: 0. -oth, --overlapthresh double allowed overlap between matches default: 0.5. -gth, --gapthresh int allowed gap length between matches default: 10. -ith, --initgapthresh int allowed initial or ending gap length at begin and end of read default: 15. -st, --support int number of supporting reads default: 2. input options: -m, --matchfile file file of (stellar) matches valid filetypes are: gff and gff. output options: -bpo, --breakpointout file name of breakpoint output file. valid filetypes are: gff and txt. default: breakpoints.gff. -j, --jobname str job/queue name default: . -do, --dots enable graph output in dot format stellar options: main options: -e, --epsilon num maximal error rate (max 0.25). in range [0.0000001..0.25]. default: 0.05. -l, --minlength num minimal length of epsilon-matches. in range [0..inf]. default: 100. -f, --forward search only in forward strand of database. -r, --reverse search only in reverse complement of database. -a, --alphabet str alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). one of dna, dna5, rna, rna5, protein, and char. -v, --verbose set verbosity mode. filtering options: -k, --kmer num length of the q-grams (max 32). in range [1..32]. -rp, --repeatperiod num maximal period of low complexity repeats to be filtered. default: 1. -rl, --repeatlength num minimal length of low complexity repeats to be filtered. default: 1000. -c, --abundancecut num k-mer overabundance cut ratio. in range [0..1]. default: 1. verification options: -x, --xdrop num maximal x-drop for extension. default: 5. -vs, --verification str verification strategy: exact or bestlocal or bandedglobal one of exact, bestlocal, and bandedglobal. default: exact. -dt, --disablethresh num maximal number of verified matches before disabling verification for one query sequence (default infinity). in range [0..inf]. -n, --nummatches num maximal number of kept matches per query and database. if stellar finds more matches, only the longest ones are kept. default: 50. -s, --sortthresh num number of matches triggering removal of duplicates. choose a smaller value for saving space. default: 500.  version gustaf version: 1.0 last update july 2012</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/seqan-apps/">
      <span itemprop="name">seqan-apps</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/gustaf.1.html">
      <span itemprop="name">gustaf: Gustaf - generic multi-split alignment finder: tool for split-read mapping allowing multiple splits.  synopsis gustaf [options] genome fasta file read fasta file  description gustaf uses seqans stellar to find splits as local matches on different strands or chromosomes. criteria and penalties to chain these matches can be specified. output file contains the breakpoints along the best chain. the genome file is used as database input, the read file as query input. all stellar options are supported. see stellar documentation for stellar parameters and options. (c) 2011-2012 by kathrin trappe -h, --help displays this help message. --version display version information gustaf options: main options: -tp, --transpen int interchromosomal translocation penalty default: 5. -ip, --invpen int inversion penalty default: 5. -op, --orderpen int intrachromosomal order change penalty default: 0. -oth, --overlapthresh double allowed overlap between matches default: 0.5. -gth, --gapthresh int allowed gap length between matches default: 10. -ith, --initgapthresh int allowed initial or ending gap length at begin and end of read default: 15. -st, --support int number of supporting reads default: 2. input options: -m, --matchfile file file of (stellar) matches valid filetypes are: gff and gff. output options: -bpo, --breakpointout file name of breakpoint output file. valid filetypes are: gff and txt. default: breakpoints.gff. -j, --jobname str job/queue name default: . -do, --dots enable graph output in dot format stellar options: main options: -e, --epsilon num maximal error rate (max 0.25). in range [0.0000001..0.25]. default: 0.05. -l, --minlength num minimal length of epsilon-matches. in range [0..inf]. default: 100. -f, --forward search only in forward strand of database. -r, --reverse search only in reverse complement of database. -a, --alphabet str alphabet type of input sequences (dna, rna, dna5, rna5, protein, char). one of dna, dna5, rna, rna5, protein, and char. -v, --verbose set verbosity mode. filtering options: -k, --kmer num length of the q-grams (max 32). in range [1..32]. -rp, --repeatperiod num maximal period of low complexity repeats to be filtered. default: 1. -rl, --repeatlength num minimal length of low complexity repeats to be filtered. default: 1000. -c, --abundancecut num k-mer overabundance cut ratio. in range [0..1]. default: 1. verification options: -x, --xdrop num maximal x-drop for extension. default: 5. -vs, --verification str verification strategy: exact or bestlocal or bandedglobal one of exact, bestlocal, and bandedglobal. default: exact. -dt, --disablethresh num maximal number of verified matches before disabling verification for one query sequence (default infinity). in range [0..inf]. -n, --nummatches num maximal number of kept matches per query and database. if stellar finds more matches, only the longest ones are kept. default: 50. -s, --sortthresh num number of matches triggering removal of duplicates. choose a smaller value for saving space. default: 500.  version gustaf version: 1.0 last update july 2012</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    <nav>
  <ul class="pager">
   <li class="previous"><a href="gusload.1.html"><span aria-hidden="true">&larr;</span> gusload.1: Load a gus-format patch file into the awe32 ram</a></li>
   <li class="next"><a href="guvcview.1.html">guvcview.1: Gtk+ base uvc viewer <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
