<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>hmm2search: </title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="hmm2search (1) manual">
  <meta name="twitter:description" content="">
  <meta name="twitter:image" content="https://www.carta.tech/images/hmmer2-hmm2search-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/hmm2search.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="hmm2search (1) manual" />
  <meta property="og:description" content="" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/hmmer2-hmm2search-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">hmm2search<small> (1)</small></h1>
        <p class="lead"></p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/hmm2search.1.html">
      <span itemprop="name">hmm2search: </span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/hmmer2/">
      <span itemprop="name">hmmer2</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/hmm2search.1.html">
      <span itemprop="name">hmm2search: </span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>hmm2search</strong> <em>[options]</em> <em>hmmfile</em> <em>seqfile</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>hmm2search</strong> reads an HMM from <em>hmmfile</em> and searches <em>seqfile</em> for significantly similar sequence matches.</p><p><em>seqfile</em> will be looked for first in the current working directory, then in a directory named by the environment variable <em>BLASTDB.</em> This lets users use existing BLAST databases, if BLAST has been configured for the site.</p><p><strong>hmm2search</strong> may take minutes or even hours to run, depending on the size of the sequence database. It is a good idea to redirect the output to a file.</p><p>The output consists of four sections: a ranked list of the best scoring sequences, a ranked list of the best scoring domains, alignments for all the best scoring domains, and a histogram of the scores. A sequence score may be higher than a domain score for the same sequence if there is more than one domain in the sequence; the sequence score takes into account all the domains. All sequences scoring above the <em>-E</em> and <em>-T</em> cutoffs are shown in the first list, then <em>every</em> domain found in this list is shown in the second list of domain hits. If desired, E-value and bit score thresholds may also be applied to the domain list using the <em>--domE</em> and <em>--domT</em> options.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Print brief help; includes version number and summary of all options, including expert options.</p>
  </dd>
  <dt>
    <p><strong>-A</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Limits the alignment output to the <em>&lt;n&gt;</em> best scoring domains. <strong>-A0</strong> shuts off the alignment output and can be used to reduce the size of output files.</p>
  </dd>
  <dt>
    <p><strong>-E</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the E-value cutoff for the per-sequence ranked hit list to <em>&lt;x&gt;,</em> where <em>&lt;x&gt;</em> is a positive real number. The default is 10.0. Hits with E-values better than (less than) this threshold will be shown.</p>
  </dd>
  <dt>
    <p><strong>-T</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the bit score cutoff for the per-sequence ranked hit list to <em>&lt;x&gt;,</em> where <em>&lt;x&gt;</em> is a real number. The default is negative infinity; by default, the threshold is controlled by E-value and not by bit score. Hits with bit scores better than (greater than) this threshold will be shown.</p>
  </dd>
  <dt>
    <p><strong>-Z</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Calculate the E-value scores as if we had seen a sequence database of <em>&lt;n&gt;</em> sequences. The default is the number of sequences seen in your database file <em>&lt;seqfile&gt;.</em></p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXPERT OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--compat</strong></p>
  </dt>
  <dd>
    <p>Use the output format of HMMER 2.1.1, the 1998-2001 public release; provided so 2.1.1 parsers don't have to be rewritten.</p>
  </dd>
  <dt>
    <p><strong>--cpu</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the maximum number of CPUs that the program will run on. The default is to use all CPUs in the machine. Overrides the HMMER_NCPU environment variable. Only affects threaded versions of HMMER (the default on most systems).</p>
  </dd>
  <dt>
    <p><strong>--cut_ga</strong></p>
  </dt>
  <dd>
    <p>Use Pfam GA (gathering threshold) score cutoffs. Equivalent to --globT &lt;GA1&gt; --domT &lt;GA2&gt;, but the GA1 and GA2 cutoffs are read from the HMM file. hmm2build puts these cutoffs there if the alignment file was annotated in a Pfam-friendly alignment format (extended SELEX or Stockholm format) and the optional GA annotation line was present. If these cutoffs are not set in the HMM file, <strong>--cut_ga</strong> doesn't work.</p>
  </dd>
  <dt>
    <p><strong>--cut_tc</strong></p>
  </dt>
  <dd>
    <p>Use Pfam TC (trusted cutoff) score cutoffs. Equivalent to --globT &lt;TC1&gt; --domT &lt;TC2&gt;, but the TC1 and TC2 cutoffs are read from the HMM file. hmm2build puts these cutoffs there if the alignment file was annotated in a Pfam-friendly alignment format (extended SELEX or Stockholm format) and the optional TC annotation line was present. If these cutoffs are not set in the HMM file, <strong>--cut_tc</strong> doesn't work.</p>
  </dd>
  <dt>
    <p><strong>--cut_nc</strong></p>
  </dt>
  <dd>
    <p>Use Pfam NC (noise cutoff) score cutoffs. Equivalent to --globT &lt;NC1&gt; --domT &lt;NC2&gt;, but the NC1 and NC2 cutoffs are read from the HMM file. hmm2build puts these cutoffs there if the alignment file was annotated in a Pfam-friendly alignment format (extended SELEX or Stockholm format) and the optional NC annotation line was present. If these cutoffs are not set in the HMM file, <strong>--cut_nc</strong> doesn't work.</p>
  </dd>
  <dt>
    <p><strong>--domE</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the E-value cutoff for the per-domain ranked hit list to <em>&lt;x&gt;,</em> where <em>&lt;x&gt;</em> is a positive real number. The default is infinity; by default, all domains in the sequences that passed the first threshold will be reported in the second list, so that the number of domains reported in the per-sequence list is consistent with the number that appear in the per-domain list.</p>
  </dd>
  <dt>
    <p><strong>--domT</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the bit score cutoff for the per-domain ranked hit list to <em>&lt;x&gt;,</em> where <em>&lt;x&gt;</em> is a real number. The default is negative infinity; by default, all domains in the sequences that passed the first threshold will be reported in the second list, so that the number of domains reported in the per-sequence list is consistent with the number that appear in the per-domain list. <em>Important note:</em> only one domain in a sequence is absolutely controlled by this parameter, or by <strong>--domT.</strong> The second and subsequent domains in a sequence have a de facto bit score threshold of 0 because of the details of how HMMER works. HMMER requires at least one pass through the main model per sequence; to do more than one pass (more than one domain) the multidomain alignment must have a better score than the single domain alignment, and hence the extra domains must contribute positive score. See the Users' Guide for more detail.</p>
  </dd>
  <dt>
    <p><strong>--forward</strong></p>
  </dt>
  <dd>
    <p>Use the Forward algorithm instead of the Viterbi algorithm to determine the per-sequence scores. Per-domain scores are still determined by the Viterbi algorithm. Some have argued that Forward is a more sensitive algorithm for detecting remote sequence homologues; my experiments with HMMER have not confirmed this, however.</p>
  </dd>
  <dt>
    <p><strong>--informat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Assert that the input <em>seqfile</em> is in format <em>&lt;s&gt;;</em> do not run Babelfish format autodection. This increases the reliability of the program somewhat, because the Babelfish can make mistakes; particularly recommended for unattended, high-throughput runs of HMMER. Valid format strings include FASTA, GENBANK, EMBL, GCG, PIR, STOCKHOLM, SELEX, MSF, CLUSTAL, and PHYLIP. See the User's Guide for a complete list.</p>
  </dd>
  <dt>
    <p><strong>--null2</strong></p>
  </dt>
  <dd>
    <p>Turn off the post hoc second null model. By default, each alignment is rescored by a postprocessing step that takes into account possible biased composition in either the HMM or the target sequence. This is almost essential in database searches, especially with local alignment models. There is a very small chance that this postprocessing might remove real matches, and in these cases <strong>--null2</strong> may improve sensitivity at the expense of reducing specificity by letting biased composition hits through.</p>
  </dd>
  <dt>
    <p><strong>--pvm</strong></p>
  </dt>
  <dd>
    <p>Run on a Parallel Virtual Machine (PVM). The PVM must already be running. The client program <strong>hmm2search-pvm</strong> must be installed on all the PVM nodes. Optional PVM support must have been compiled into HMMER.</p>
  </dd>
  <dt>
    <p><strong>--xnu</strong></p>
  </dt>
  <dd>
    <p>Turn on XNU filtering of target protein sequences. Has no effect on nucleic acid sequences. In trial experiments, <strong>--xnu</strong> appears to perform less well than the default post hoc null2 model.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO hmm2search&hellip;</h2>
        <div class="sectioncontent">
<p>Master man page, with full list of and guide to the individual man pages: see <a href="../man1/hmmer2.1.html"><strong>hmmer2</strong>(1)</a>.</strong></p><p>For complete documentation, see the user guide (ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf); or see the HMMER web page, http://hmmer.janelia.org/.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
Freely distributed under the GNU General Public License (GPL).
</pre>
<p>See the file COPYING in your distribution for details on redistribution conditions.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
Sean Eddy
HHMI/Dept. of Genetics
Washington Univ. School of Medicine
4566 Scott Ave.
St Louis, MO 63110 USA
http://www.genetics.wustl.edu/eddy/
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="hmm2pfam-pvm.1.html"><span aria-hidden="true">&larr;</span> hmm2pfam-pvm.1: Hmmer utilities for pvm clusters</a></li>
   <li class="next"><a href="hmm2search-pvm.1.html">hmm2search-pvm.1: Hmmer utilities for pvm clusters <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
