<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>hmm2build: </title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="hmm2build (1) manual">
  <meta name="twitter:description" content="">
  <meta name="twitter:image" content="https://www.carta.tech/images/hmmer2-hmm2build-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/hmm2build.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="hmm2build (1) manual" />
  <meta property="og:description" content="" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/hmmer2-hmm2build-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">hmm2build<small> (1)</small></h1>
        <p class="lead"></p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/hmm2build.1.html">
      <span itemprop="name">hmm2build: </span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/hmmer2/">
      <span itemprop="name">hmmer2</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/hmm2build.1.html">
      <span itemprop="name">hmm2build: </span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>hmm2build</strong> <em>[options]</em> <em>hmmfile</em> <em>alignfile</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>hmm2build</strong> reads a multiple sequence alignment file <em>alignfile</em> , builds a new profile HMM, and saves the HMM in <em>hmmfile.</em></p><p><em>alignfile</em> may be in ClustalW, GCG MSF, SELEX, Stockholm, or aligned FASTA alignment format. The format is automatically detected.</p><p>By default, the model is configured to find one or more nonoverlapping alignments to the complete model: multiple global alignments with respect to the model, and local with respect to the sequence. This is analogous to the behavior of the <strong>hmmls</strong> program of HMMER 1. To configure the model for multiple <em>local</em> alignments with respect to the model and local with respect to the sequence, a la the old program <strong>hmmfs,</strong> use the <strong>-f</strong> (fragment) option. More rarely, you may want to configure the model for a single global alignment (global with respect to both model and sequence), using the <strong>-g</strong> option; or to configure the model for a single local/local alignment (a la standard Smith/Waterman, or the old <strong>hmmsw</strong> program), use the <strong>-s</strong> option.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-f</strong></p>
  </dt>
  <dd>
    <p>Configure the model for finding multiple domains per sequence, where each domain can be a local (fragmentary) alignment. This is analogous to the old <strong>hmmfs</strong> program of HMMER 1.</p>
  </dd>
  <dt>
    <p><strong>-g</strong></p>
  </dt>
  <dd>
    <p>Configure the model for finding a single global alignment to a target sequence, analogous to the old <strong>hmms</strong> program of HMMER 1.</p>
  </dd>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Print brief help; includes version number and summary of all options, including expert options.</p>
  </dd>
  <dt>
    <p><strong>-n</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Name this HMM <em>&lt;s&gt;.</em> <em>&lt;s&gt;</em> can be any string of non-whitespace characters (e.g. one "word"). There is no length limit (at least not one imposed by HMMER; your shell will complain about command line lengths first).</p>
  </dd>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Re-save the starting alignment to <em>&lt;f&gt;,</em> in Stockholm format. The columns which were assigned to match states will be marked with x's in an #=RF annotation line. If either the <strong>--hand</strong> or <strong>--fast</strong> construction options were chosen, the alignment may have been slightly altered to be compatible with Plan 7 transitions, so saving the final alignment and comparing to the starting alignment can let you view these alterations. See the User's Guide for more information on this arcane side effect.</p>
  </dd>
  <dt>
    <p><strong>-s</strong></p>
  </dt>
  <dd>
    <p>Configure the model for finding a single local alignment per target sequence. This is analogous to the standard Smith/Waterman algorithm or the <strong>hmmsw</strong> program of HMMER 1.</p>
  </dd>
  <dt>
    <p><strong>-A</strong></p>
  </dt>
  <dd>
    <p>Append this model to an existing <em>hmmfile</em> rather than creating <em>hmmfile.</em> Useful for building HMM libraries (like Pfam).</p>
  </dd>
  <dt>
    <p><strong>-F</strong></p>
  </dt>
  <dd>
    <p>Force overwriting of an existing <em>hmmfile.</em> Otherwise HMMER will refuse to clobber your existing HMM files, for safety's sake.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXPERT OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--amino</strong></p>
  </dt>
  <dd>
    <p>Force the sequence alignment to be interpreted as amino acid sequences. Normally HMMER autodetects whether the alignment is protein or DNA, but sometimes alignments are so small that autodetection is ambiguous. See <strong>--nucleic.</strong></p>
  </dd>
  <dt>
    <p><strong>--archpri</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the "architecture prior" used by MAP architecture construction to <em>&lt;x&gt;,</em> where <em>&lt;x&gt;</em> is a probability between 0 and 1. This parameter governs a geometric prior distribution over model lengths. As <em>&lt;x&gt;</em> increases, longer models are favored a priori. As <em>&lt;x&gt;</em> decreases, it takes more residue conservation in a column to make a column a "consensus" match column in the model architecture. The 0.85 default has been chosen empirically as a reasonable setting.</p>
  </dd>
  <dt>
    <p><strong>--binary</strong></p>
  </dt>
  <dd>
    <p>Write the HMM to <em>hmmfile</em> in HMMER binary format instead of readable ASCII text.</p>
  </dd>
  <dt>
    <p><strong>--cfile</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save the observed emission and transition counts to <em>&lt;f&gt;</em> after the architecture has been determined (e.g. after residues/gaps have been assigned to match, delete, and insert states). This option is used in HMMER development for generating data files useful for training new Dirichlet priors. The format of count files is documented in the User's Guide.</p>
  </dd>
  <dt>
    <p><strong>--fast</strong></p>
  </dt>
  <dd>
    <p>Quickly and heuristically determine the architecture of the model by assigning all columns will more than a certain fraction of gap characters to insert states. By default this fraction is 0.5, and it can be changed using the <strong>--gapmax</strong> option. The default construction algorithm is a maximum a posteriori (MAP) algorithm, which is slower.</p>
  </dd>
  <dt>
    <p><strong>--gapmax</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Controls the <em>--fast</em> model construction algorithm, but if <em>--fast</em> is not being used, has no effect. If a column has more than a fraction <em>&lt;x&gt;</em> of gap symbols in it, it gets assigned to an insert column. <em>&lt;x&gt;</em> is a frequency from 0 to 1, and by default is set to 0.5. Higher values of <em>&lt;x&gt;</em> mean more columns get assigned to consensus, and models get longer; smaller values of <em>&lt;x&gt;</em> mean fewer columns get assigned to consensus, and models get smaller. <em>&lt;x&gt;</em></p>
  </dd>
  <dt>
    <p><strong>--hand</strong></p>
  </dt>
  <dd>
    <p>Specify the architecture of the model by hand: the alignment file must be in SELEX or Stockholm format, and the reference annotation line (#=RF in SELEX, #=GC RF in Stockholm) is used to specify the architecture. Any column marked with a non-gap symbol (such as an 'x', for instance) is assigned as a consensus (match) column in the model.</p>
  </dd>
  <dt>
    <p><strong>--idlevel</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Controls both the determination of effective sequence number and the behavior of the <em>--wblosum</em> weighting option. The sequence alignment is clustered by percent identity, and the number of clusters at a cutoff threshold of <em>&lt;x&gt;</em> is used to determine the effective sequence number. Higher values of <em>&lt;x&gt;</em> give more clusters and higher effective sequence numbers; lower values of <em>&lt;x&gt;</em> give fewer clusters and lower effective sequence numbers. <em>&lt;x&gt;</em> is a fraction from 0 to 1, and by default is set to 0.62 (corresponding to the clustering level used in constructing the BLOSUM62 substitution matrix).</p>
  </dd>
  <dt>
    <p><strong>--informat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Assert that the input <em>seqfile</em> is in format <em>&lt;s&gt;;</em> do not run Babelfish format autodection. This increases the reliability of the program somewhat, because the Babelfish can make mistakes; particularly recommended for unattended, high-throughput runs of HMMER. Valid format strings include FASTA, GENBANK, EMBL, GCG, PIR, STOCKHOLM, SELEX, MSF, CLUSTAL, and PHYLIP. See the User's Guide for a complete list.</p>
  </dd>
  <dt>
    <p><strong>--noeff</strong></p>
  </dt>
  <dd>
    <p>Turn off the effective sequence number calculation, and use the true number of sequences instead. This will usually reduce the sensitivity of the final model (so don't do it without good reason!)</p>
  </dd>
  <dt>
    <p><strong>--nucleic</strong></p>
  </dt>
  <dd>
    <p>Force the alignment to be interpreted as nucleic acid sequence, either RNA or DNA. Normally HMMER autodetects whether the alignment is protein or DNA, but sometimes alignments are so small that autodetection is ambiguous. See <strong>--amino.</strong></p>
  </dd>
  <dt>
    <p><strong>--null</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Read a null model from <em>&lt;f&gt;.</em> The default for protein is to use average amino acid frequencies from Swissprot 34 and p1 = 350/351; for nucleic acid, the default is to use 0.25 for each base and p1 = 1000/1001. For documentation of the format of the null model file and further explanation of how the null model is used, see the User's Guide.</p>
  </dd>
  <dt>
    <p><strong>--pam</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Apply a heuristic PAM- (substitution matrix-) based prior on match emission probabilities instead of the default mixture Dirichlet. The substitution matrix is read from <em>&lt;f&gt;.</em> See <strong>--pamwgt.</strong></p><p>The default Dirichlet state transition prior and insert emission prior are unaffected. Therefore in principle you could combine <strong>--prior</strong> with <strong>--pam</strong> but this isn't recommended, as it hasn't been tested. ( <strong>--pam</strong> itself hasn't been tested much!)</p>
  </dd>
  <dt>
    <p><strong>--pamwgt</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Controls the weight on a PAM-based prior. Only has effect if <strong>--pam</strong> option is also in use. <em>&lt;x&gt;</em> is a positive real number, 20.0 by default. <em>&lt;x&gt;</em> is the number of "pseudocounts" contriubuted by the heuristic prior. Very high values of <em>&lt;x&gt;</em> can force a scoring system that is entirely driven by the substitution matrix, making HMMER somewhat approximate Gribskov profiles.</p>
  </dd>
  <dt>
    <p><strong>--pbswitch</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>For alignments with a very large number of sequences, the GSC, BLOSUM, and Voronoi weighting schemes are slow; they're O(N^2) for N sequences. Henikoff position-based weights (PB weights) are more efficient. At or above a certain threshold sequence number <em>&lt;n&gt;</em> <strong>hmm2build</strong> will switch from GSC, BLOSUM, or Voronoi weights to PB weights. To disable this switching behavior (at the cost of compute time, set <em>&lt;n&gt;</em> to be something larger than the number of sequences in your alignment. <em>&lt;n&gt;</em> is a positive integer; the default is 1000.</p>
  </dd>
  <dt>
    <p><strong>--prior</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Read a Dirichlet prior from <em>&lt;f&gt;,</em> replacing the default mixture Dirichlet. The format of prior files is documented in the User's Guide, and an example is given in the Demos directory of the HMMER distribution.</p>
  </dd>
  <dt>
    <p><strong>--swentry</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Controls the total probability that is distributed to local entries into the model, versus starting at the beginning of the model as in a global alignment. <em>&lt;x&gt;</em> is a probability from 0 to 1, and by default is set to 0.5. Higher values of <em>&lt;x&gt;</em> mean that hits that are fragments on their left (N or 5'-terminal) side will be penalized less, but complete global alignments will be penalized more. Lower values of <em>&lt;x&gt;</em> mean that fragments on the left will be penalized more, and global alignments on this side will be favored. This option only affects the configurations that allow local alignments, e.g. <strong>-s</strong> and <strong>-f;</strong> unless one of these options is also activated, this option has no effect. You have independent control over local/global alignment behavior for the N/C (5'/3') termini of your target sequences using <strong>--swentry</strong> and <strong>--swexit.</strong></p>
  </dd>
  <dt>
    <p><strong>--swexit</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Controls the total probability that is distributed to local exits from the model, versus ending an alignment at the end of the model as in a global alignment. <em>&lt;x&gt;</em> is a probability from 0 to 1, and by default is set to 0.5. Higher values of <em>&lt;x&gt;</em> mean that hits that are fragments on their right (C or 3'-terminal) side will be penalized less, but complete global alignments will be penalized more. Lower values of <em>&lt;x&gt;</em> mean that fragments on the right will be penalized more, and global alignments on this side will be favored. This option only affects the configurations that allow local alignments, e.g. <strong>-s</strong> and <strong>-f;</strong> unless one of these options is also activated, this option has no effect. You have independent control over local/global alignment behavior for the N/C (5'/3') termini of your target sequences using <strong>--swentry</strong> and <strong>--swexit.</strong></p>
  </dd>
  <dt>
    <p><strong>--verbose</strong></p>
  </dt>
  <dd>
    <p>Print more possibly useful stuff, such as the individual scores for each sequence in the alignment.</p>
  </dd>
  <dt>
    <p><strong>--wblosum</strong></p>
  </dt>
  <dd>
    <p>Use the BLOSUM filtering algorithm to weight the sequences, instead of the default. Cluster the sequences at a given percentage identity (see <strong>--idlevel);</strong> assign each cluster a total weight of 1.0, distributed equally amongst the members of that cluster.</p>
  </dd>
  <dt>
    <p><strong>--wgsc</strong></p>
  </dt>
  <dd>
    <p>Use the Gerstein/Sonnhammer/Chothia ad hoc sequence weighting algorithm. This is already the default, so this option has no effect (unless it follows another option in the -\-w family, in which case it overrides it).</p>
  </dd>
  <dt>
    <p><strong>--wme</strong></p>
  </dt>
  <dd>
    <p>Use the Krogh/Mitchison maximum entropy algorithm to "weight" the sequences. This supercedes the Eddy/Mitchison/Durbin maximum discrimination algorithm, which gives almost identical weights but is less robust. ME weighting seems to give a marginal increase in sensitivity over the default GSC weights, but takes a fair amount of time.</p>
  </dd>
  <dt>
    <p><strong>--wnone</strong></p>
  </dt>
  <dd>
    <p>Turn off all sequence weighting.</p>
  </dd>
  <dt>
    <p><strong>--wpb</strong></p>
  </dt>
  <dd>
    <p>Use the Henikoff position-based weighting scheme.</p>
  </dd>
  <dt>
    <p><strong>--wvoronoi</strong></p>
  </dt>
  <dd>
    <p>Use the Sibbald/Argos Voronoi sequence weighting algorithm in place of the default GSC weighting.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO hmm2build&hellip;</h2>
        <div class="sectioncontent">
<p>Master man page, with full list of and guide to the individual man pages: see <a href="../man1/hmmer2.1.html"><strong>hmmer2</strong>(1)</a>.</strong></p><p>For complete documentation, see the user guide (ftp://selab.janelia.org/pub/software/hmmer/2.3.2/Userguide.pdf); or see the HMMER web page, http://hmmer.janelia.org/.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 1992-2003 HHMI/Washington University School of Medicine.
Freely distributed under the GNU General Public License (GPL).
</pre>
<p>See the file COPYING in your distribution for details on redistribution conditions.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
Sean Eddy
HHMI/Dept. of Genetics
Washington Univ. School of Medicine
4566 Scott Ave.
St Louis, MO 63110 USA
http://www.genetics.wustl.edu/eddy/
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="hmm2align.1.html"><span aria-hidden="true">&larr;</span> hmm2align.1: </a></li>
   <li class="next"><a href="hmm2calibrate.1.html">hmm2calibrate.1:  <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
