<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>hmmbuild: Construct profile hmm(s) from multiple sequence alignment(s)</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Construct profile hmm(s) from multiple sequence alignment(s)">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="hmmbuild (1) manual">
  <meta name="twitter:description" content="Construct profile hmm(s) from multiple sequence alignment(s)">
  <meta name="twitter:image" content="https://www.carta.tech/images/hmmer-hmmbuild-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/hmmbuild.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="hmmbuild (1) manual" />
  <meta property="og:description" content="Construct profile hmm(s) from multiple sequence alignment(s)" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/hmmer-hmmbuild-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">hmmbuild<small> (1)</small></h1>
        <p class="lead">Construct profile hmm(s) from multiple sequence alignment(s)</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/hmmbuild.1.html">
      <span itemprop="name">hmmbuild: Construct profile hmm(s) from multiple sequence alignment(s)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/hmmer/">
      <span itemprop="name">hmmer</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/hmmbuild.1.html">
      <span itemprop="name">hmmbuild: Construct profile hmm(s) from multiple sequence alignment(s)</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>hmmbuild</strong> <em>[options]</em> <em>&lt;hmmfile_out&gt;</em> <em>&lt;msafile&gt;</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>For each multiple sequence alignment in <em>&lt;msafile&gt;</em> build a profile HMM and save it to a new file <em>&lt;hmmfile_out&gt;</em>.</p><p><em>&lt;msafile&gt;</em> may be '-' (dash), which means reading this input from <em>stdin</em> rather than a file. To use '-', you must also specify the alignment file format with <strong>--informat</strong><em> &lt;s&gt;</em><strong>,</strong> as in <strong>--informat stockholm</strong> (because of a current limitation in our implementation, MSA file formats cannot be autodetected in a nonrewindable input stream.)</p><p><em>&lt;hmmfile_out&gt;</em> may not be '-' <em>(stdout)</em>, because sending the HMM file to <em>stdout</em> would conflict with the other text output of the program.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and all available options.</p>
  </dd>
  <dt>
    <p><strong>-n</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Name the new profile <em>&lt;s&gt;</em>. The default is to use the name of the alignment (if one is present in the <em>msafile</em>, or, failing that, the name of the <em>hmmfile</em>. If <em>msafile</em> contains more than one alignment, <em>-n</em> doesn't work, and every alignment must have a name annotated in the <em>msafile</em> (as in Stockholm #=GF ID annotation).</p>
  </dd>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Direct the summary output to file <em>&lt;f&gt;</em>, rather than to <em>stdout</em>.</p>
  </dd>
  <dt>
    <p><strong>-O</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>After each model is constructed, resave annotated, possibly modified source alignments to a file <em>&lt;f&gt;</em> in Stockholm format. The alignments are annotated with a reference annotation line indicating which columns were assigned as consensus, and sequences are annotated with what relative sequence weights were assigned. Some residues of the alignment may have been shifted to accommodate restrictions of the Plan7 profile architecture, which disallows transitions between insert and delete states.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR SPECIFYING THE ALPHABET</h2>
        <div class="sectioncontent">
<p>The alphabet type (amino, DNA, or RNA) is autodetected by default, by looking at the composition of the <em>msafile</em>. Autodetection is normally quite reliable, but occasionally alphabet type may be ambiguous and autodetection can fail (for instance, on tiny toy alignments of just a few residues). To avoid this, or to increase robustness in automated analysis pipelines, you may specify the alphabet type of <em>msafile</em> with these options.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--amino</strong></p>
  </dt>
  <dd>
    <p>Specify that all sequences in <em>msafile</em> are proteins.</p>
  </dd>
  <dt>
    <p><strong>--dna</strong></p>
  </dt>
  <dd>
    <p>Specify that all sequences in <em>msafile</em> are DNAs.</p>
  </dd>
  <dt>
    <p><strong>--rna</strong></p>
  </dt>
  <dd>
    <p>Specify that all sequences in <em>msafile</em> are RNAs.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING PROFILE CONSTRUCTION</h2>
        <div class="sectioncontent">
<p>These options control how consensus columns are defined in an alignment.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--fast</strong></p>
  </dt>
  <dd>
    <p>Define consensus columns as those that have a fraction &gt;= <strong>symfrac</strong> of residues as opposed to gaps. (See below for the <strong>--symfrac</strong> option.) This is the default.</p>
  </dd>
  <dt>
    <p><strong>--hand</strong></p>
  </dt>
  <dd>
    <p>Define consensus columns in next profile using reference annotation to the multiple alignment. This allows you to define any consensus columns you like.</p>
  </dd>
  <dt>
    <p><strong>--symfrac</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Define the residue fraction threshold necessary to define a consensus column when using the <strong>--fast</strong> option. The default is 0.5. The symbol fraction in each column is calculated after taking relative sequence weighting into account, and ignoring gap characters corresponding to ends of sequence fragments (as opposed to internal insertions/deletions). Setting this to 0.0 means that every alignment column will be assigned as consensus, which may be useful in some cases. Setting it to 1.0 means that only columns that include 0 gaps (internal insertions/deletions) will be assigned as consensus.</p>
  </dd>
  <dt>
    <p><strong>--fragthresh</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>We only want to count terminal gaps as deletions if the aligned sequence is known to be full-length, not if it is a fragment (for instance, because only part of it was sequenced). HMMER uses a simple rule to infer fragments: if the sequence length L is less than or equal to a fraction <em>&lt;x&gt;</em> times the alignment length in columns, then the sequence is handled as a fragment. The default is 0.5. Setting <strong>--fragthresh</strong><em>0</em> will define no (nonempty) sequence as a fragment; you might want to do this if you know you've got a carefully curated alignment of full-length sequences. Setting <strong>--fragthresh</strong><em>1</em> will define all sequences as fragments; you might want to do this if you know your alignment is entirely composed of fragments, such as translated short reads in metagenomic shotgun data.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING RELATIVE WEIGHTS</h2>
        <div class="sectioncontent">
<p>HMMER uses an ad hoc sequence weighting algorithm to downweight closely related sequences and upweight distantly related ones. This has the effect of making models less biased by uneven phylogenetic representation. For example, two identical sequences would typically each receive half the weight that one sequence would.  These options control which algorithm gets used.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--wpb</strong></p>
  </dt>
  <dd>
    <p>Use the Henikoff position-based sequence weighting scheme [Henikoff and Henikoff, J. Mol. Biol. 243:574, 1994].  This is the default.</p>
  </dd>
  <dt>
    <p><strong>--wgsc</strong></p>
  </dt>
  <dd>
    <p>Use the Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et al, J. Mol. Biol. 235:1067, 1994].</p>
  </dd>
  <dt>
    <p><strong>--wblosum</strong></p>
  </dt>
  <dd>
    <p>Use the same clustering scheme that was used to weight data in calculating BLOSUM subsitution matrices [Henikoff and Henikoff, Proc. Natl. Acad. Sci 89:10915, 1992]. Sequences are single-linkage clustered at an identity threshold (default 0.62; see <em>--wid</em>) and within each cluster of c sequences, each sequence gets relative weight 1/c.</p>
  </dd>
  <dt>
    <p><strong>--wnone</strong></p>
  </dt>
  <dd>
    <p>No relative weights. All sequences are assigned uniform weight.</p>
  </dd>
  <dt>
    <p><strong>--wid</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the identity threshold used by single-linkage clustering when using <em>--wblosum</em>. Invalid with any other weighting scheme. Default is 0.62.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING EFFECTIVE SEQUENCE NUMBER</h2>
        <div class="sectioncontent">
<p>After relative weights are determined, they are normalized to sum to a total effective sequence number, <em>eff_nseq</em>. This number may be the actual number of sequences in the alignment, but it is almost always smaller than that. The default entropy weighting method <em>(--eent)</em> reduces the effective sequence number to reduce the information content (relative entropy, or average expected score on true homologs) per consensus position. The target relative entropy is controlled by a two-parameter function, where the two parameters are settable with <em>--ere</em> and <em>--esigma</em>.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--eent</strong></p>
  </dt>
  <dd>
    <p>Adjust effective sequence number to achieve a specific relative entropy per position (see <em>--ere</em>). This is the default.</p>
  </dd>
  <dt>
    <p><strong>--eclust</strong></p>
  </dt>
  <dd>
    <p>Set effective sequence number to the number of single-linkage clusters at a specific identity threshold (see <em>--eid</em>). This option is not recommended; it's for experiments evaluating how much better <strong>--eent</strong> is.</p>
  </dd>
  <dt>
    <p><strong>--enone</strong></p>
  </dt>
  <dd>
    <p>Turn off effective sequence number determination and just use the actual number of sequences. One reason you might want to do this is to try to maximize the relative entropy/position of your model, which may be useful for short models.</p>
  </dd>
  <dt>
    <p><strong>--eset</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Explicitly set the effective sequence number for all models to <em>&lt;x&gt;</em>.</p>
  </dd>
  <dt>
    <p><strong>--ere</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the minimum relative entropy/position target to <em>&lt;x&gt;</em>. Requires <strong>--eent</strong>. Default depends on the sequence alphabet. For protein sequences, it is 0.59 bits/position; for nucleotide sequences, it is 0.45 bits/position.</p>
  </dd>
  <dt>
    <p><strong>--esigma</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the minimum relative entropy contributed by an entire model alignment, over its whole length. This has the effect of making short models have higher relative entropy per position than <em>--ere</em> alone would give. The default is 45.0 bits.</p>
  </dd>
  <dt>
    <p><strong>--eid</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the fractional pairwise identity cutoff used by single linkage clustering with the <strong>--eclust</strong> option. The default is 0.62.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING PRIORS</h2>
        <div class="sectioncontent">
<p>By default, weighted counts are converted to mean posterior probability parameter estimates using mixture Dirichlet priors. Default mixture Dirichlet prior parameters for protein models and for nucleic acid (RNA and DNA) models are built in. The following options allow you to override the default priors.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--pnone</strong></p>
  </dt>
  <dd>
    <p>Don't use any priors. Probability parameters will simply be the observed frequencies, after relative sequence weighting.</p>
  </dd>
  <dt>
    <p><strong>--plaplace</strong></p>
  </dt>
  <dd>
    <p>Use a Laplace +1 prior in place of the default mixture Dirichlet prior.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING E-VALUE CALIBRATION</h2>
        <div class="sectioncontent">
<p>The location parameters for the expected score distributions for MSV filter scores, Viterbi filter scores, and Forward scores require three short random sequence simulations.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--EmL</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the sequence length in simulation that estimates the location parameter mu for MSV filter E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--EmN</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the number of sequences in simulation that estimates the location parameter mu for MSV filter E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--EvL</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the sequence length in simulation that estimates the location parameter mu for Viterbi filter E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--EvN</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the number of sequences in simulation that estimates the location parameter mu for Viterbi filter E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--EfL</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the sequence length in simulation that estimates the location parameter tau for Forward E-values. Default is 100.</p>
  </dd>
  <dt>
    <p><strong>--EfN</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the number of sequences in simulation that estimates the location parameter tau for Forward E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--Eft</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the tail mass fraction to fit in the simulation that estimates the location parameter tau for Forward evalues. Default is 0.04.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--cpu</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the number of parallel worker threads to <em>&lt;n&gt;</em>. By default, HMMER sets this to the number of CPU cores it detects in your machine - that is, it tries to maximize the use of your available processor cores. Setting <em>&lt;n&gt;</em> higher than the number of available cores is of little if any value, but you may want to set it to something less. You can also control this number by setting an environment variable, <em>HMMER_NCPU</em>.</p><p>This option is only available if HMMER was compiled with POSIX threads support. This is the default, but it may have been turned off for your site or machine for some reason.</p>
  </dd>
  <dt>
    <p><strong>--informat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Declare that the input <em>msafile</em> is in format <em>&lt;s&gt;</em>. Currently the accepted multiple alignment sequence file formats include Stockholm, Aligned FASTA, Clustal, NCBI PSI-BLAST, PHYLIP, Selex, and UCSC SAM A2M. Default is to autodetect the format of the file.</p>
  </dd>
  <dt>
    <p><strong>--seed</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Seed the random number generator with <em>&lt;n&gt;</em>, an integer &gt;= 0. If <em>&lt;n&gt;</em> is nonzero, any stochastic simulations will be reproducible; the same command will give the same results. If <em>&lt;n&gt;</em> is 0, the random number generator is seeded arbitrarily, and stochastic simulations will vary from run to run of the same command. The default seed is 42.</p>
  </dd>
  <dt>
    <p><strong>--w_beta</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Window length tail mass. The upper bound, <em>W</em>, on the length at which nhmmer expects to find an instance of the model is set such that the fraction of all sequences generated by the model with length <em>&gt;= W</em> is less than <em>&lt;x&gt;</em>. The default is 1e-7.</p>
  </dd>
  <dt>
    <p><strong>--w_length</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Override the model instance length upper bound, <em>W</em>, which is otherwise controlled by <strong>--w_beta</strong>. It should be larger than the model length. The value of <em>W</em> is used deep in the acceleration pipeline, and modest changes are not expected to impact results (though larger values of <em>W</em> do lead to longer run time).</p>
  </dd>
  <dt>
    <p><strong>--mpi</strong></p>
  </dt>
  <dd>
    <p>Run as a parallel MPI program. Each alignment is assigned to a MPI worker node for construction. (Therefore, the maximum parallelization cannot exceed the number of alignments in the input <em>msafile</em>.) This is useful when building large profile libraries. This option is only available if optional MPI capability was enabled at compile-time.</p>
  </dd>
  <dt>
    <p><strong>--stall</strong></p>
  </dt>
  <dd>
    <p>For debugging MPI parallelization: arrest program execution immediately after start, and wait for a debugger to attach to the running process and release the arrest.</p>
  </dd>
  <dt>
    <p><strong>--maxinsertlen</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Restrict insert length parameterization such that the expected insert length at each position of the model is no more than <em>&lt;n&gt;</em>.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO hmmbuild&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/hmmer.1.html"><strong>hmmer</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the HMMER package.</p><p>For complete documentation, see the user guide that came with your HMMER distribution (Userguide.pdf); or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2013 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="hmmalign.1.html"><span aria-hidden="true">&larr;</span> hmmalign.1: Align sequences to a profile hmm</a></li>
   <li class="next"><a href="hmmconvert.1.html">hmmconvert.1: Convert profile file to a hmmer format <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
