<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>hmmer: Profile hmms for biological sequence analysis</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Profile hmms for biological sequence analysis">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="hmmer (1) manual">
  <meta name="twitter:description" content="Profile hmms for biological sequence analysis">
  <meta name="twitter:image" content="https://www.carta.tech/images/hmmer-hmmer-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/hmmer.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="hmmer (1) manual" />
  <meta property="og:description" content="Profile hmms for biological sequence analysis" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/hmmer-hmmer-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">hmmer<small> (1)</small></h1>
        <p class="lead">Profile hmms for biological sequence analysis</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/hmmer.1.html">
      <span itemprop="name">hmmer: Profile hmms for biological sequence analysis</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/hmmer/">
      <span itemprop="name">hmmer</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/hmmer.1.html">
      <span itemprop="name">hmmer: Profile hmms for biological sequence analysis</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>hmmalign</strong></p>
<pre>
  Align sequences to a profile
</pre>
<p><strong>hmmbuild</strong></p>
<pre>
  Construct profile(s) from multiple sequence alignment(s)
</pre>
<p><strong>hmmconvert</strong></p>
<pre>
  Convert profile file to various HMMER and non-HMMER formats
</pre>
<p><strong>hmmemit</strong></p>
<pre>
  Sample sequences from a profile
</pre>
<p><strong>hmmfetch</strong></p>
<pre>
  Retrieve profile HMM(s) from a file
</pre>
<p><strong>hmmpgmd</strong></p>
<pre>
  Daemon used for hmmer.org web server, for phmmer-, hmmsearch-, and hmmscan-like search
</pre>
<p><strong>hmmpress</strong></p>
<pre>
  Prepare an HMM database for hmmscan
</pre>
<p><strong>hmmscan</strong></p>
<pre>
  Search protein sequence(s) against a protein profile database
</pre>
<p><strong>hmmsearch</strong></p>
<pre>
  Search protein profile(s) against a protein sequence database
</pre>
<p><strong>hmmsim</strong></p>
<pre>
  Collect profile score distributions on random sequences
</pre>
<p><strong>hmmstat</strong></p>
<pre>
  Summary statistics for a profile file
</pre>
<p><strong>jackhmmer</strong></p>
<pre>
  Iterative search of a protein sequence against a protein sequence database
</pre>
<p><strong>nhmmer</strong></p>
<pre>
  Search nucleotide sequence(s), alignment(s), or profile(s) against a nucleotide sequence database
</pre>
<p><strong>nhmmscan</strong></p>
<pre>
  Search nucleotide sequence(s) against a nucleotide profile database
</pre>
<p><strong>phmmer</strong></p>
<pre>
  Search a protein sequence(s) against a protein sequence database
</pre>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>HMMER is a suite of several programs for biological sequence alignment and database homology search. It uses probabilistic models called "profile hidden Markov models" (profile HMMs) to represent the likely evolutionary homologs of a single sequence or a multiple alignment of a sequence family. A main avenue of research is to improve the evolutionary predictive models in HMMER to be able to recognize and accurately align increasingly remote homologs, distant in time.</p><p>HMMER is also used as an organizational tool, to group the exponentially growing number of biological sequences into a vastly smaller set of well-annotated sequence families. New sequences can be annotated by comparison against curated sequence family databases of prebuilt HMMER profiles, in addition or instead of comparison to the entire sequence database. Databases such as Pfam, SMART, and TIGRfams, among others, are based on this principle.</p><p>HMMER is used in three main modes: to search a sequence database for new homologs of a sequence or a sequence family; to search a profile database (like Pfam) to find what known family a query sequence belongs to, or what domains it has; and to automatically construct large multiple alignments (i.e. with an effectively unlimited number of sequences) using a profile representative of a sequence family.</p><p>Suppose you have a multiple sequence alignment of a sequence family of interest, and you want to search a sequence database for additional homologs. The <strong>hmmbuild</strong> program builds profile(s) from multiple alignment(s). The <strong>hmmsearch</strong> program searches protein profile(s) against a protein sequence database, and <strong>nhmmer</strong> searches nucleotide profile(s) against a nucleotide sequence database.</p><p>Suppose you have a single sequence of interest, and you want to search a sequence database for additional homologs. The <strong>phmmer</strong> program searches a single protein sequence against a protein sequence database. The <strong>jackhmmer</strong> program does the same thing but iteratively -- homologs detected in a previous round are incorporated into a new profile, and the new profile is searched again. <strong>phmmer</strong> is used like BLASTP, and <strong>jackhmmer</strong> is used like a protein PSI-BLAST. The <strong>nhmmer</strong> program searches a single nucleotide sequence against a nucleotide sequence.</p><p>Suppose you have sequence(s) that you want to analyze using a HMMER-based profile HMM database like Pfam (http://pfam.sanger.ac.uk). The <strong>hmmpress</strong> program formats a profile HMM flatfile (such as the file you would download from Pfam) into a HMMER binary database. The <strong>hmmscan</strong> program searches protein sequence(s) against that database. The <strong>nhmmscan</strong> program can similarly search nucleotide sequence(s) against a pressed database of nucleotide profiles, such as from Dfam (http://dfam.janelia.org).</p><p>Suppose you want to align lots of sequences. You can construct a manageably small alignment of a representative set of sequences, build a profile with <strong>hmmbuild</strong>, and use the <strong>hmmalign</strong> program to align any number of sequences to that profile.</p><p>HMMER also includes some auxiliary tools for working with large profile databases. <strong>hmmfetch</strong> fetches one or more profiles from a database. <strong>hmmstat</strong> prints summary statistics about a profile file.</p><p>For compatibility with other profile software and previous versions of HMMER, the <strong>hmmconvert</strong> program converts profiles to a few other formats. We intend to add more support for other formats over time.</p><p>The <strong>hmmemit</strong> program generates (simulates) "homologous" sequences by sampling from a profile. It can also generate a "consensus" sequence.</p><p>The <strong>hmmsim</strong> program is a simulator used for collecting statistics about score distributions on random sequences.</p><p>Each program has its own man page.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO hmmer&hellip;</h2>
        <div class="sectioncontent">
<p>This is a summary man page for the entire HMMER3 package. See individual man pages <strong>[<a href="../man1/hmmbuild.1.html"><strong>hmmbuild</strong>(1)</a></strong>, for example] for usage, options, and description of each program in the package.</p><p>For complete documentation, see the user guide that came with your HMMER distribution (Userguide.pdf); or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2013 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="hmmemit.1.html"><span aria-hidden="true">&larr;</span> hmmemit.1: Sample sequences from a profile hmm</a></li>
   <li class="next"><a href="hmmer2.1.html">hmmer2.1:  <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
