<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>htseq-qa: Perform simple quality assesment of high-throughput sequencing reads  the python script htseq-qa takes a file with sequencing reads (either raw or aligned reads) and produces a pdf file with useful plots to assess the technical quality of a run.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Perform simple quality assesment of high-throughput sequencing reads  the python script htseq-qa takes a file with sequencing reads (either raw or aligned reads) and produces a pdf file with useful plots to assess the technical quality of a run.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="htseq-qa (1) manual">
  <meta name="twitter:description" content="Perform simple quality assesment of high-throughput sequencing reads  the python script htseq-qa takes a file with sequencing reads (either raw or aligned reads) and produces a pdf file with useful plots to assess the technical quality of a run.">
  <meta name="twitter:image" content="https://www.carta.tech/images/python-htseq-htseq-qa-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/htseq-qa.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="htseq-qa (1) manual" />
  <meta property="og:description" content="Perform simple quality assesment of high-throughput sequencing reads  the python script htseq-qa takes a file with sequencing reads (either raw or aligned reads) and produces a pdf file with useful plots to assess the technical quality of a run." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/python-htseq-htseq-qa-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">htseq-qa<small> (1)</small></h1>
        <p class="lead">Perform simple quality assesment of high-throughput sequencing reads  the python script htseq-qa takes a file with sequencing reads (either raw or aligned reads) and produces a pdf file with useful plots to assess the technical quality of a run.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/htseq-qa.1.html">
      <span itemprop="name">htseq-qa: Perform simple quality assesment of high-throughput sequencing reads  the python script htseq-qa takes a file with sequencing reads (either raw or aligned reads) and produces a pdf file with useful plots to assess the technical quality of a run.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/python-htseq/">
      <span itemprop="name">python-htseq</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/htseq-qa.1.html">
      <span itemprop="name">htseq-qa: Perform simple quality assesment of high-throughput sequencing reads  the python script htseq-qa takes a file with sequencing reads (either raw or aligned reads) and produces a pdf file with useful plots to assess the technical quality of a run.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">PLOT</h2>
        <div class="sectioncontent">
<p>Here is a typical plot: [image]</p><p>The plot is made from a SAM file, which contained aligned and unalignable reads. The left column is made from the non-aligned, the right column from the aligned reads. The header informs you about the name of the SAM file, and the number of reads.</p><p>The upper row shows how often which base was called for each position in the read. In this sample, the non-alignable reads have a clear excess in A. The aligned reads have a balance between complementing reads: A and C (reddish colours) have equal levels, and so do C and G (greenish colours). The sequences seem to be AT rich. Furthermore, nearly all aligned reads start with a T, followed by an A, and then, a C in 70% and an A in 30% of the reads. Such an imbalance would be reason for concern if it has no good explanation. Here, the reason is that the fragmentation of the sample was done by enzyme digestion.</p><p>The lower half shows the abundance of base-call quality scores at the different positions in the read. Nearly all aligned reads have a quality of 34 over their whole length, while for the non-aligned reads, some reads have lower quality scores towards their ends.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">USAGE</h2>
        <div class="sectioncontent">
<p>Note that <strong>htseq-qa</strong> needs matplotlib to produce the plot, so you need to install this module, as described <em>here</em> on the matplotlib web site.</p><p>After you have installed HTSeq (see <em>install</em>) and matplotlib, you can run <strong>htseq-qa</strong> from the command line:</p>
<pre>
htseq-qa [options] read_file
</pre>
<p>If the file <strong>htseq-qa</strong> is not in your path, you can, alternatively, call the script with</p>
<pre>
python -m HTSeq.scripts.qa [options] read_file
</pre>
<p>The <em>read_file</em> is either a FASTQ file or a SAM file. For a SAM file, a plot with two columns is produced as above, for a FASTQ file, you get only one column.</p><p>The output is written into a file with the same name as <em>read_file</em>, with the suffix <strong>.pdf</strong> added. View it with a PDF viewer such as the Acrobat Reader.</p><h3>Options</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>-t &lt;type&gt;, --type=&lt;type&gt;</strong></p>
  </dt>
  <dd>
    <p>The file type of the <em>read_file</em>. Supported values for <em>&lt;type&gt;</em> are:</p><ul>
<li><p><strong>sam</strong>: a SAM file (Note that the <em>SAMtools</em> contain Perl scripts to convert most alignment formats to SAM)</p></li><li><p><strong>solexa-export</strong>: an <strong>_export.txt</strong> file as produced by the SolexaPipeline software after aligning with Eland (<strong>htseq-qa</strong> expects the new Solexa quality encoding as produced by version 1.3 or newer of the SolexaPipeline)</p></li><li><p><strong>fastq</strong>: a FASTQ file with standard (Sanger or Phred) quality encoding</p></li><li><p><strong>solexa-fastq</strong>: a FASTQ file with Solexa quality encoding, as produced by the SolexaPipeline after base-calling with Bustard (<strong>htseq-qa</strong> expects the new Solexa quality encoding as produced by version 1.3 or newer of the SolexaPipeline)</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-o &lt;outfile&gt;, --outfile=&lt;outfile&gt;</strong></p>
  </dt>
  <dd>
    <p>output filename (default is <em>&lt;read_file&gt;\(ga\(ga</em>.pdf\(ga\(ga)</p>
  </dd>
  <dt>
    <p><strong>-r &lt;readlen&gt;, --readlength=&lt;readlen&gt;</strong></p>
  </dt>
  <dd>
    <p>the maximum read length (when not specified, the script guesses from the file</p>
  </dd>
  <dt>
    <p><strong>-g &lt;gamma&gt;, --gamma=&lt;gamma&gt;</strong></p>
  </dt>
  <dd>
    <p>the gamma factor for the contrast adjustment of the quality score plot</p>
  </dd>
  <dt>
    <p><strong>-n, --nosplit</strong></p>
  </dt>
  <dd>
    <p>do not split reads in unaligned and aligned ones, i.e., produce a one-column plot</p>
  </dd>
  <dt>
    <p><strong>-m, --maxqual</strong></p>
  </dt>
  <dd>
    <p>the maximum quality score that appears in the data (default: 40)</p>
  </dd>
  <dt>
    <p><strong>-h, --help</strong></p>
  </dt>
  <dd>
    <p>Show a usage summary and exit</p>
  </dd>

</dl>


        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Simon Anders</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">
<p>2010, Simon Anders</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="htseq-count.1.html"><span aria-hidden="true">&larr;</span> htseq-count.1: Count the number of reads in a sam alignment file that map to gff features  given a file with aligned sequencing reads and a list of genomic features, a common task is to count how many reads map to each feature.  a feature is here an interval (i.e., a range of positions) on a chromosome or a union of such intervals.  in the case of rna-seq, the features are typically genes, where each gene is considered here as the union of all its exons. one may also consider each exon as a feature, e.g., in order to check for alternative splicing. for comparative chip-seq, the features might be binding region from a pre-determined list.  special care must be taken to decide how to deal with reads that overlap more than one feature. the htseq-count script allows to choose between three modes. of course, if none of these fits your needs, you can write your own script with htseq. see the chapter tour for a step-by-step guide on how to do so.  the three overlap resolution modes of htseq-count work as follows. for each position i in the read, a set s(i) is defined as the set of all features overlapping position i. then, consider the set s, which is (with i running through all position within the read) the union of all the sets s(i) for mode union. the intersection of all the sets s(i) for mode intersection-strict. the intersection of all non-empty sets s(i) for mode intersection-nonempty.  if s contains precisely one feature, the read is counted for this feature. if it contains more than one feature, the read is counted as ambiguous (and not counted for any features), and if s is empty, the read is counted as no_feature.  the following figure illustrates the effect of these three modes: [image]</a></li>
   <li class="next"><a href="htsserver.1.html">htsserver.1: Offline browser server : copy websites to a local directory <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
