<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>insegt: Intersecting second generation sequencing data with annotation  synopsis insegt [options] aligments-file annotations-file  description insegt is a tool to analyze alignments of rna-seq reads (single-end or paired-end) by using gene-annotations. input to insegt is a sam file containing the alignments and a file containing the annotations of the reference genome, either in gff or gtf format. -h, --help displays this help message. --version display version information options: : -ro, --read-output file output filename for read-output, which contains the mapped annotations followed by their parent annotation. valid filetype is: gff. -ao, --anno-output file output filename for anno-output, which contains the annotations similar to the gff input and additionally the counts of the mapped reads and the normalized expression levels in rpkm. valid filetype is: gff. -to, --tuple-output file output filename for tuple-output, which contains exon tuples connected by reads or matepairs. valid filetype is: gff. -fo, --fusion-output str output filename for fusion-output, which contains exon tuple of gene fusions (advanced option, currently no output port for knime). one of gff. -n, --ntuple int ntuple default: 2. -o, --offset-interval int offset to short alignment-intervals for search. default: 5. -t, --threshold-gaps int threshold for allowed gaps in alignment (not introns). default: 5. -c, --threshold-count int threshold for min. count of tuple for output. default: 1. -r, --threshold-rpkm double threshold for min. rpkm of tuple for output. default: 0.0. -m, --max-tuple create only maxtuple (which are spanned by the whole read). -e, --exact-ntuple create only tuple of exact length n. by default all tuple up to the given length are computed (if -m is set, -e will be ignored). -u, --unknown-orientation orientation of reads is unknown.  examples</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Intersecting second generation sequencing data with annotation  synopsis insegt [options] aligments-file annotations-file  description insegt is a tool to analyze alignments of rna-seq reads (single-end or paired-end) by using gene-annotations. input to insegt is a sam file containing the alignments and a file containing the annotations of the reference genome, either in gff or gtf format. -h, --help displays this help message. --version display version information options: : -ro, --read-output file output filename for read-output, which contains the mapped annotations followed by their parent annotation. valid filetype is: gff. -ao, --anno-output file output filename for anno-output, which contains the annotations similar to the gff input and additionally the counts of the mapped reads and the normalized expression levels in rpkm. valid filetype is: gff. -to, --tuple-output file output filename for tuple-output, which contains exon tuples connected by reads or matepairs. valid filetype is: gff. -fo, --fusion-output str output filename for fusion-output, which contains exon tuple of gene fusions (advanced option, currently no output port for knime). one of gff. -n, --ntuple int ntuple default: 2. -o, --offset-interval int offset to short alignment-intervals for search. default: 5. -t, --threshold-gaps int threshold for allowed gaps in alignment (not introns). default: 5. -c, --threshold-count int threshold for min. count of tuple for output. default: 1. -r, --threshold-rpkm double threshold for min. rpkm of tuple for output. default: 0.0. -m, --max-tuple create only maxtuple (which are spanned by the whole read). -e, --exact-ntuple create only tuple of exact length n. by default all tuple up to the given length are computed (if -m is set, -e will be ignored). -u, --unknown-orientation orientation of reads is unknown.  examples">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="insegt (1) manual">
  <meta name="twitter:description" content="Intersecting second generation sequencing data with annotation  synopsis insegt [options] aligments-file annotations-file  description insegt is a tool to analyze alignments of rna-seq reads (single-end or paired-end) by using gene-annotations. input to insegt is a sam file containing the alignments and a file containing the annotations of the reference genome, either in gff or gtf format. -h, --help displays this help message. --version display version information options: : -ro, --read-output file output filename for read-output, which contains the mapped annotations followed by their parent annotation. valid filetype is: gff. -ao, --anno-output file output filename for anno-output, which contains the annotations similar to the gff input and additionally the counts of the mapped reads and the normalized expression levels in rpkm. valid filetype is: gff. -to, --tuple-output file output filename for tuple-output, which contains exon tuples connected by reads or matepairs. valid filetype is: gff. -fo, --fusion-output str output filename for fusion-output, which contains exon tuple of gene fusions (advanced option, currently no output port for knime). one of gff. -n, --ntuple int ntuple default: 2. -o, --offset-interval int offset to short alignment-intervals for search. default: 5. -t, --threshold-gaps int threshold for allowed gaps in alignment (not introns). default: 5. -c, --threshold-count int threshold for min. count of tuple for output. default: 1. -r, --threshold-rpkm double threshold for min. rpkm of tuple for output. default: 0.0. -m, --max-tuple create only maxtuple (which are spanned by the whole read). -e, --exact-ntuple create only tuple of exact length n. by default all tuple up to the given length are computed (if -m is set, -e will be ignored). -u, --unknown-orientation orientation of reads is unknown.  examples">
  <meta name="twitter:image" content="https://www.carta.tech/images/seqan-apps-insegt-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/insegt.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="insegt (1) manual" />
  <meta property="og:description" content="Intersecting second generation sequencing data with annotation  synopsis insegt [options] aligments-file annotations-file  description insegt is a tool to analyze alignments of rna-seq reads (single-end or paired-end) by using gene-annotations. input to insegt is a sam file containing the alignments and a file containing the annotations of the reference genome, either in gff or gtf format. -h, --help displays this help message. --version display version information options: : -ro, --read-output file output filename for read-output, which contains the mapped annotations followed by their parent annotation. valid filetype is: gff. -ao, --anno-output file output filename for anno-output, which contains the annotations similar to the gff input and additionally the counts of the mapped reads and the normalized expression levels in rpkm. valid filetype is: gff. -to, --tuple-output file output filename for tuple-output, which contains exon tuples connected by reads or matepairs. valid filetype is: gff. -fo, --fusion-output str output filename for fusion-output, which contains exon tuple of gene fusions (advanced option, currently no output port for knime). one of gff. -n, --ntuple int ntuple default: 2. -o, --offset-interval int offset to short alignment-intervals for search. default: 5. -t, --threshold-gaps int threshold for allowed gaps in alignment (not introns). default: 5. -c, --threshold-count int threshold for min. count of tuple for output. default: 1. -r, --threshold-rpkm double threshold for min. rpkm of tuple for output. default: 0.0. -m, --max-tuple create only maxtuple (which are spanned by the whole read). -e, --exact-ntuple create only tuple of exact length n. by default all tuple up to the given length are computed (if -m is set, -e will be ignored). -u, --unknown-orientation orientation of reads is unknown.  examples" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/seqan-apps-insegt-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">insegt<small> (1)</small></h1>
        <p class="lead">Intersecting second generation sequencing data with annotation  synopsis insegt [options] aligments-file annotations-file  description insegt is a tool to analyze alignments of rna-seq reads (single-end or paired-end) by using gene-annotations. input to insegt is a sam file containing the alignments and a file containing the annotations of the reference genome, either in gff or gtf format. -h, --help displays this help message. --version display version information options: : -ro, --read-output file output filename for read-output, which contains the mapped annotations followed by their parent annotation. valid filetype is: gff. -ao, --anno-output file output filename for anno-output, which contains the annotations similar to the gff input and additionally the counts of the mapped reads and the normalized expression levels in rpkm. valid filetype is: gff. -to, --tuple-output file output filename for tuple-output, which contains exon tuples connected by reads or matepairs. valid filetype is: gff. -fo, --fusion-output str output filename for fusion-output, which contains exon tuple of gene fusions (advanced option, currently no output port for knime). one of gff. -n, --ntuple int ntuple default: 2. -o, --offset-interval int offset to short alignment-intervals for search. default: 5. -t, --threshold-gaps int threshold for allowed gaps in alignment (not introns). default: 5. -c, --threshold-count int threshold for min. count of tuple for output. default: 1. -r, --threshold-rpkm double threshold for min. rpkm of tuple for output. default: 0.0. -m, --max-tuple create only maxtuple (which are spanned by the whole read). -e, --exact-ntuple create only tuple of exact length n. by default all tuple up to the given length are computed (if -m is set, -e will be ignored). -u, --unknown-orientation orientation of reads is unknown.  examples</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/insegt.1.html">
      <span itemprop="name">insegt: Intersecting second generation sequencing data with annotation  synopsis insegt [options] aligments-file annotations-file  description insegt is a tool to analyze alignments of rna-seq reads (single-end or paired-end) by using gene-annotations. input to insegt is a sam file containing the alignments and a file containing the annotations of the reference genome, either in gff or gtf format. -h, --help displays this help message. --version display version information options: : -ro, --read-output file output filename for read-output, which contains the mapped annotations followed by their parent annotation. valid filetype is: gff. -ao, --anno-output file output filename for anno-output, which contains the annotations similar to the gff input and additionally the counts of the mapped reads and the normalized expression levels in rpkm. valid filetype is: gff. -to, --tuple-output file output filename for tuple-output, which contains exon tuples connected by reads or matepairs. valid filetype is: gff. -fo, --fusion-output str output filename for fusion-output, which contains exon tuple of gene fusions (advanced option, currently no output port for knime). one of gff. -n, --ntuple int ntuple default: 2. -o, --offset-interval int offset to short alignment-intervals for search. default: 5. -t, --threshold-gaps int threshold for allowed gaps in alignment (not introns). default: 5. -c, --threshold-count int threshold for min. count of tuple for output. default: 1. -r, --threshold-rpkm double threshold for min. rpkm of tuple for output. default: 0.0. -m, --max-tuple create only maxtuple (which are spanned by the whole read). -e, --exact-ntuple create only tuple of exact length n. by default all tuple up to the given length are computed (if -m is set, -e will be ignored). -u, --unknown-orientation orientation of reads is unknown.  examples</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/seqan-apps/">
      <span itemprop="name">seqan-apps</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/insegt.1.html">
      <span itemprop="name">insegt: Intersecting second generation sequencing data with annotation  synopsis insegt [options] aligments-file annotations-file  description insegt is a tool to analyze alignments of rna-seq reads (single-end or paired-end) by using gene-annotations. input to insegt is a sam file containing the alignments and a file containing the annotations of the reference genome, either in gff or gtf format. -h, --help displays this help message. --version display version information options: : -ro, --read-output file output filename for read-output, which contains the mapped annotations followed by their parent annotation. valid filetype is: gff. -ao, --anno-output file output filename for anno-output, which contains the annotations similar to the gff input and additionally the counts of the mapped reads and the normalized expression levels in rpkm. valid filetype is: gff. -to, --tuple-output file output filename for tuple-output, which contains exon tuples connected by reads or matepairs. valid filetype is: gff. -fo, --fusion-output str output filename for fusion-output, which contains exon tuple of gene fusions (advanced option, currently no output port for knime). one of gff. -n, --ntuple int ntuple default: 2. -o, --offset-interval int offset to short alignment-intervals for search. default: 5. -t, --threshold-gaps int threshold for allowed gaps in alignment (not introns). default: 5. -c, --threshold-count int threshold for min. count of tuple for output. default: 1. -r, --threshold-rpkm double threshold for min. rpkm of tuple for output. default: 0.0. -m, --max-tuple create only maxtuple (which are spanned by the whole read). -e, --exact-ntuple create only tuple of exact length n. by default all tuple up to the given length are computed (if -m is set, -e will be ignored). -u, --unknown-orientation orientation of reads is unknown.  examples</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    <nav>
  <ul class="pager">
   <li class="previous"><a href="insdseqget.1.html"><span aria-hidden="true">&larr;</span> insdseqget.1: Format sequences from genbank as an xml insdset</a></li>
   <li class="next"><a href="insert_brackets.1.html">insert_brackets.1: Add delimiters before/after text <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
