<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>jackhmmer: Iteratively search sequence(s) against a protein database</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Iteratively search sequence(s) against a protein database">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="jackhmmer (1) manual">
  <meta name="twitter:description" content="Iteratively search sequence(s) against a protein database">
  <meta name="twitter:image" content="https://www.carta.tech/images/hmmer-jackhmmer-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/jackhmmer.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="jackhmmer (1) manual" />
  <meta property="og:description" content="Iteratively search sequence(s) against a protein database" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/hmmer-jackhmmer-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">jackhmmer<small> (1)</small></h1>
        <p class="lead">Iteratively search sequence(s) against a protein database</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/jackhmmer.1.html">
      <span itemprop="name">jackhmmer: Iteratively search sequence(s) against a protein database</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/hmmer/">
      <span itemprop="name">hmmer</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/jackhmmer.1.html">
      <span itemprop="name">jackhmmer: Iteratively search sequence(s) against a protein database</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>jackhmmer</strong> <em>[options]</em> <em>&lt;seqfile&gt;</em> <em>&lt;seqdb&gt;</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>jackhmmer</strong> iteratively searches each query sequence in <em>&lt;seqfile&gt;</em> against the target sequence(s) in <em>&lt;seqdb&gt;</em>. The first iteration is identical to a <strong>phmmer</strong> search. For the next iteration, a multiple alignment of the query together with all target sequences satisfying <em>inclusion thresholds</em> is assembled, a profile is constructed from this alignment (identical to using <strong>hmmbuild</strong> on the alignment), and profile search of the <em>&lt;seqdb&gt;</em> is done (identical to an <strong>hmmsearch</strong> with the profile).</p><p>The query <em>&lt;seqfile&gt;</em> may be '-' (a dash character), in which case the query sequences are read from a &lt;stdin&gt; pipe instead of from a file. The <em>&lt;seqdb&gt;</em> cannot be read from a &lt;stdin&gt; stream, because <strong>jackhmmer</strong> needs to do multiple passes over the database.</p><p>The output format is designed to be human-readable, but is often so voluminous that reading it is impractical, and parsing it is a pain. The <strong>--tblout</strong> and <strong>--domtblout</strong> options save output in simple tabular formats that are concise and easier to parse. The <strong>-o</strong> option allows redirecting the main output, including throwing it away in /dev/null.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and all available options.</p>
  </dd>
  <dt>
    <p><strong>-N</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the maximum number of iterations to <em>&lt;n&gt;</em>. The default is 5. If N=1, the result is equivalent to a <strong>phmmer</strong> search.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING OUTPUT</h2>
        <div class="sectioncontent">
<p>By default, output for each iteration appears on stdout in a somewhat human readable, somewhat parseable format. These options allow redirecting that output or saving additional kinds of output to files, including checkpoint files for each iteration.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Direct the human-readable output to a file <em>&lt;f&gt;</em>.</p>
  </dd>
  <dt>
    <p><strong>-A</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>After the final iteration, save an annotated multiple alignment of all hits satisfying inclusion thresholds (also including the original query) to <em>&lt;f&gt;</em> in Stockholm format.</p>
  </dd>
  <dt>
    <p><strong>--tblout</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>After the final iteration, save a tabular summary of top sequence hits to <em>&lt;f&gt;</em> in a readily parseable, columnar, whitespace-delimited format.</p>
  </dd>
  <dt>
    <p><strong>--domtblout</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>After the final iteration, save a tabular summary of top domain hits to <em>&lt;f&gt;</em> in a readily parseable, columnar, whitespace-delimited format.</p>
  </dd>
  <dt>
    <p><strong>--chkhmm</strong><em> &lt;prefix&gt;</em></p>
  </dt>
  <dd>
    <p>At the start of each iteration, checkpoint the query HMM, saving it to a file named <em>&lt;prefix&gt;-&lt;n&gt;.hmm</em> where <em>&lt;n&gt;</em> is the iteration number (from 1..N).</p>
  </dd>
  <dt>
    <p><strong>--chkali</strong><em> &lt;prefix&gt;</em></p>
  </dt>
  <dd>
    <p>At the end of each iteration, checkpoint an alignment of all domains satisfying inclusion thresholds (e.g. what will become the query HMM for the next iteration), saving it to a file named <em>&lt;checkpoint file prefix&gt;-&lt;n&gt;.sto</em> in Stockholm format, where <em>&lt;n&gt;</em> is the iteration number (from 1..N).</p>
  </dd>
  <dt>
    <p><strong>--acc</strong></p>
  </dt>
  <dd>
    <p>Use accessions instead of names in the main output, where available for profiles and/or sequences.</p>
  </dd>
  <dt>
    <p><strong>--noali</strong></p>
  </dt>
  <dd>
    <p>Omit the alignment section from the main output. This can greatly reduce the output volume.</p>
  </dd>
  <dt>
    <p><strong>--notextw</strong></p>
  </dt>
  <dd>
    <p>Unlimit the length of each line in the main output. The default is a limit of 120 characters per line, which helps in displaying the output cleanly on terminals and in editors, but can truncate target profile description lines.</p>
  </dd>
  <dt>
    <p><strong>--textw</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the main output's line length limit to <em>&lt;n&gt;</em> characters per line. The default is 120.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING SINGLE SEQUENCE SCORING (FIRST ITERATION)</h2>
        <div class="sectioncontent">
<p>By default, the first iteration uses a search model constructed from a single query sequence. This model is constructed using a standard 20x20 substitution matrix for residue probabilities, and two additional parameters for position-independent gap open and gap extend probabilities. These options allow the default single-sequence scoring parameters to be changed.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--popen</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the gap open probability for a single sequence query model to <em>&lt;x&gt;</em>. The default is 0.02. <em>&lt;x&gt;</em> must be &gt;= 0 and &lt; 0.5.</p>
  </dd>
  <dt>
    <p><strong>--pextend</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the gap extend probability for a single sequence query model to <em>&lt;x&gt;</em>. The default is 0.4. <em>&lt;x&gt;</em> must be &gt;= 0 and &lt; 1.0.</p>
  </dd>
  <dt>
    <p><strong>--mx</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Obtain residue alignment probabilities from the built-in substitution matrix named <em>&lt;s&gt;</em>. Several standard matrices are built-in, and do not need to be read from files. The matrix name <em>&lt;s&gt;</em> can be PAM30, PAM70, PAM120, PAM240, BLOSUM45, BLOSUM50, BLOSUM62, BLOSUM80, or BLOSUM90. Only one of the <strong>--mx</strong> and <strong>--mxfile</strong> options may be used.</p>
  </dd>
  <dt>
    <p><strong>--mxfile</strong><em> &lt;mxfile&gt;</em></p>
  </dt>
  <dd>
    <p>Obtain residue alignment probabilities from the substitution matrix in file <em>&lt;mxfile&gt;</em>. The default score matrix is BLOSUM62 (this matrix is internal to HMMER and does not have to be available as a file). The format of a substitution matrix <em>&lt;mxfile&gt;</em> is the standard format accepted by BLAST, FASTA, and other sequence analysis software.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING REPORTING THRESHOLDS</h2>
        <div class="sectioncontent">
<p>Reporting thresholds control which hits are reported in output files (the main output, <em>--tblout</em>, and <em>--domtblout</em>). In each iteration, sequence hits and domain hits are ranked by statistical significance (E-value) and output is generated in two sections called per-target and per-domain output. In per-target output, by default, all sequence hits with an E-value &lt;= 10 are reported. In the per-domain output, for each target that has passed per-target reporting thresholds, all domains satisfying per-domain reporting thresholds are reported. By default, these are domains with conditional E-values of &lt;= 10. The following options allow you to change the default E-value reporting thresholds, or to use bit score thresholds instead.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-E</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Report sequences with E-values &lt;= <em>&lt;x&gt;</em> in per-sequence output. The default is 10.0.</p>
  </dd>
  <dt>
    <p><strong>-T</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Use a bit score threshold for per-sequence output instead of an E-value threshold (any setting of <strong>-E</strong> is ignored). Report sequences with a bit score of &gt;= <em>&lt;x&gt;</em>. By default this option is unset.</p>
  </dd>
  <dt>
    <p><strong>-Z</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Declare the total size of the database to be <em>&lt;x&gt;</em> sequences, for purposes of E-value calculation. Normally E-values are calculated relative to the size of the database you actually searched (e.g. the number of sequences in <em>target_seqdb</em>). In some cases (for instance, if you've split your target sequence database into multiple files for parallelization of your search), you may know better what the actual size of your search space is.</p>
  </dd>
  <dt>
    <p><strong>--domE</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Report domains with conditional E-values &lt;= <em>&lt;x&gt;</em> in per-domain output, in addition to the top-scoring domain per significant sequence hit. The default is 10.0.</p>
  </dd>
  <dt>
    <p><strong>--domT</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Use a bit score threshold for per-domain output instead of an E-value threshold (any setting of <strong>--domT</strong> is ignored). Report domains with a bit score of &gt;= <em>&lt;x&gt;</em> in per-domain output, in addition to the top-scoring domain per significant sequence hit. By default this option is unset.</p>
  </dd>
  <dt>
    <p><strong>--domZ</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Declare the number of significant sequences to be <em>&lt;x&gt;</em> sequences, for purposes of conditional E-value calculation for additional domain significance. Normally conditional E-values are calculated relative to the number of sequences passing per-sequence reporting threshold.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING INCLUSION THRESHOLDS</h2>
        <div class="sectioncontent">
<p>Inclusion thresholds control which hits are included in the multiple alignment and profile constructed for the next search iteration. By default, a sequence must have a per-sequence E-value of &lt;= 0.001 (see <strong>-E</strong> option) to be included, and any additional domains in it besides the top-scoring one must have a conditional E-value of &lt;= 0.001 (see <strong>--domE</strong> option). The difference between reporting thresholds and inclusion thresholds is that inclusion thresholds control which hits actually get used in the next iteration (or the final output multiple alignment if the <em>-A</em> option is used), whereas reporting thresholds control what you see in output. Reporting thresholds are generally more loose so you can see borderline hits in the top of the noise that might be of interest.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--incE</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Include sequences with E-values &lt;= <em>&lt;x&gt;</em> in subsequent iteration or final alignment output by <em>-A</em>. The default is 0.001.</p>
  </dd>
  <dt>
    <p><strong>--incT</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Use a bit score threshold for per-sequence inclusion instead of an E-value threshold (any setting of <strong>--incE</strong> is ignored). Include sequences with a bit score of &gt;= <em>&lt;x&gt;</em>. By default this option is unset.</p>
  </dd>
  <dt>
    <p><strong>--incdomE</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Include domains with conditional E-values &lt;= <em>&lt;x&gt;</em> in subsequent iteration or final alignment output by <em>-A</em>, in addition to the top-scoring domain per significant sequence hit. The default is 0.001.</p>
  </dd>
  <dt>
    <p><strong>--incdomT</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Use a bit score threshold for per-domain inclusion instead of an E-value threshold (any setting of <strong>--incT</strong> is ignored). Include domains with a bit score of &gt;= <em>&lt;x&gt;</em>. By default this option is unset.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING ACCELERATION HEURISTICS</h2>
        <div class="sectioncontent">
<p>HMMER3 searches are accelerated in a three-step filter pipeline: the MSV filter, the Viterbi filter, and the Forward filter. The first filter is the fastest and most approximate; the last is the full Forward scoring algorithm, slowest but most accurate. There is also a bias filter step between MSV and Viterbi. Targets that pass all the steps in the acceleration pipeline are then subjected to postprocessing -- domain identification and scoring using the Forward/Backward algorithm.</p><p>Essentially the only free parameters that control HMMER's heuristic filters are the P-value thresholds controlling the expected fraction of nonhomologous sequences that pass the filters. Setting the default thresholds higher will pass a higher proportion of nonhomologous sequence, increasing sensitivity at the expense of speed; conversely, setting lower P-value thresholds will pass a smaller proportion, decreasing sensitivity and increasing speed. Setting a filter's P-value threshold to 1.0 means it will passing all sequences, and effectively disables the filter.</p><p>Changing filter thresholds only removes or includes targets from consideration; changing filter thresholds does not alter bit scores, E-values, or alignments, all of which are determined solely in postprocessing.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--max</strong></p>
  </dt>
  <dd>
    <p>Maximum sensitivity.  Turn off all filters, including the bias filter, and run full Forward/Backward postprocessing on every target. This increases sensitivity slightly, at a large cost in speed.</p>
  </dd>
  <dt>
    <p><strong>--F1</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>First filter threshold; set the P-value threshold for the MSV filter step.  The default is 0.02, meaning that roughly 2% of the highest scoring nonhomologous targets are expected to pass the filter.</p>
  </dd>
  <dt>
    <p><strong>--F2</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Second filter threshold; set the P-value threshold for the Viterbi filter step.  The default is 0.001.</p>
  </dd>
  <dt>
    <p><strong>--F3</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Third filter threshold; set the P-value threshold for the Forward filter step.  The default is 1e-5.</p>
  </dd>
  <dt>
    <p><strong>--nobias</strong></p>
  </dt>
  <dd>
    <p>Turn off the bias filter. This increases sensitivity somewhat, but can come at a high cost in speed, especially if the query has biased residue composition (such as a repetitive sequence region, or if it is a membrane protein with large regions of hydrophobicity). Without the bias filter, too many sequences may pass the filter with biased queries, leading to slower than expected performance as the computationally intensive Forward/Backward algorithms shoulder an abnormally heavy load.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING PROFILE CONSTRUCTION (LATER ITERATIONS)</h2>
        <div class="sectioncontent">
<p>These options control how consensus columns are defined in multiple alignments when building profiles. By default, <strong>jackhmmer</strong> always includes your original query sequence in the alignment result at every iteration, and consensus positions are defined by that query sequence: that is, a default <strong>jackhmmer</strong> profile is always the same length as your original query, at every iteration.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--fast</strong></p>
  </dt>
  <dd>
    <p>Define consensus columns as those that have a fraction &gt;= <strong>symfrac</strong> of residues as opposed to gaps. (See below for the <strong>--symfrac</strong> option.) Although this is the default profile construction option elsewhere (in <strong>hmmbuild</strong>, in particular), it may have undesirable effects in <strong>jackhmmer</strong>, because a profile could iteratively walk in sequence space away from your original query, leaving few or no consensus columns corresponding to its residues.</p>
  </dd>
  <dt>
    <p><strong>--hand</strong></p>
  </dt>
  <dd>
    <p>Define consensus columns in next profile using reference annotation to the multiple alignment. <strong>jackhmmer</strong> propagates reference annotation from the previous profile to the multiple alignment, and thence to the next profile. This is the default.</p>
  </dd>
  <dt>
    <p><strong>--symfrac</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Define the residue fraction threshold necessary to define a consensus column when using the <strong>--fast</strong> option. The default is 0.5. The symbol fraction in each column is calculated after taking relative sequence weighting into account, and ignoring gap characters corresponding to ends of sequence fragments (as opposed to internal insertions/deletions). Setting this to 1.0 means that every alignment column will be assigned as consensus, which may be useful in some cases. Setting it to 0.0 is a bad idea, because no columns will be assigned as consensus, and you'll get a model of zero length.</p>
  </dd>
  <dt>
    <p><strong>--fragthresh</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>We only want to count terminal gaps as deletions if the aligned sequence is known to be full-length, not if it is a fragment (for instance, because only part of it was sequenced). HMMER uses a simple rule to infer fragments: if the sequence length L is less than or equal to a fraction <em>&lt;x&gt;</em> times the alignment length in columns, then the sequence is handled as a fragment. The default is 0.5. Setting <strong>--fragthresh</strong><em>0</em> will define no (nonempty) sequence as a fragment; you might want to do this if you know you've got a carefully curated alignment of full-length sequences. Setting <strong>--fragthresh</strong><em>1</em> will define all sequences as fragments; you might want to do this if you know your alignment is entirely composed of fragments, such as translated short reads in metagenomic shotgun data.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING RELATIVE WEIGHTS</h2>
        <div class="sectioncontent">
<p>Whenever a profile is built from a multiple alignment, HMMER uses an ad hoc sequence weighting algorithm to downweight closely related sequences and upweight distantly related ones. This has the effect of making models less biased by uneven phylogenetic representation. For example, two identical sequences would typically each receive half the weight that one sequence would (and this is why <strong>jackhmmer</strong> isn't concerned about always including your original query sequence in each iteration's alignment, even if it finds it again in the database you're searching). These options control which algorithm gets used.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--wpb</strong></p>
  </dt>
  <dd>
    <p>Use the Henikoff position-based sequence weighting scheme [Henikoff and Henikoff, J. Mol. Biol. 243:574, 1994].  This is the default.</p>
  </dd>
  <dt>
    <p><strong>--wgsc</strong></p>
  </dt>
  <dd>
    <p>Use the Gerstein/Sonnhammer/Chothia weighting algorithm [Gerstein et al, J. Mol. Biol. 235:1067, 1994].</p>
  </dd>
  <dt>
    <p><strong>--wblosum</strong></p>
  </dt>
  <dd>
    <p>Use the same clustering scheme that was used to weight data in calculating BLOSUM subsitution matrices [Henikoff and Henikoff, Proc. Natl. Acad. Sci 89:10915, 1992]. Sequences are single-linkage clustered at an identity threshold (default 0.62; see <em>--wid</em>) and within each cluster of c sequences, each sequence gets relative weight 1/c.</p>
  </dd>
  <dt>
    <p><strong>--wnone</strong></p>
  </dt>
  <dd>
    <p>No relative weights. All sequences are assigned uniform weight.</p>
  </dd>
  <dt>
    <p><strong>--wid</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the identity threshold used by single-linkage clustering when using <em>--wblosum</em>. Invalid with any other weighting scheme. Default is 0.62.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING EFFECTIVE SEQUENCE NUMBER</h2>
        <div class="sectioncontent">
<p>After relative weights are determined, they are normalized to sum to a total effective sequence number, <em>eff_nseq</em>. This number may be the actual number of sequences in the alignment, but it is almost always smaller than that. The default entropy weighting method <em>(--eent)</em> reduces the effective sequence number to reduce the information content (relative entropy, or average expected score on true homologs) per consensus position. The target relative entropy is controlled by a two-parameter function, where the two parameters are settable with <em>--ere</em> and <em>--esigma</em>.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--eent</strong></p>
  </dt>
  <dd>
    <p>Adjust effective sequence number to achieve a specific relative entropy per position (see <em>--ere</em>). This is the default.</p>
  </dd>
  <dt>
    <p><strong>--eclust</strong></p>
  </dt>
  <dd>
    <p>Set effective sequence number to the number of single-linkage clusters at a specific identity threshold (see <em>--eid</em>). This option is not recommended; it's for experiments evaluating how much better <strong>--eent</strong> is.</p>
  </dd>
  <dt>
    <p><strong>--enone</strong></p>
  </dt>
  <dd>
    <p>Turn off effective sequence number determination and just use the actual number of sequences. One reason you might want to do this is to try to maximize the relative entropy/position of your model, which may be useful for short models.</p>
  </dd>
  <dt>
    <p><strong>--eset</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Explicitly set the effective sequence number for all models to <em>&lt;x&gt;</em>.</p>
  </dd>
  <dt>
    <p><strong>--ere</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the minimum relative entropy/position target to <em>&lt;x&gt;</em>. Requires <strong>--eent</strong>. Default depends on the sequence alphabet; for protein sequences, it is 0.59 bits/position.</p>
  </dd>
  <dt>
    <p><strong>--esigma</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the minimum relative entropy contributed by an entire model alignment, over its whole length. This has the effect of making short models have higher relative entropy per position than <em>--ere</em> alone would give. The default is 45.0 bits.</p>
  </dd>
  <dt>
    <p><strong>--eid</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the fractional pairwise identity cutoff used by single linkage clustering with the <strong>--eclust</strong> option. The default is 0.62.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING PRIORS</h2>
        <div class="sectioncontent">
<p>In profile construction, by default, weighted counts are converted to mean posterior probability parameter estimates using mixture Dirichlet priors.  Default mixture Dirichlet prior parameters for protein models and for nucleic acid (RNA and DNA) models are built in. The following options allow you to override the default priors.</p><p><strong>--pnone</strong> Don't use any priors. Probability parameters will simply be the observed frequencies, after relative sequence weighting.</p><p><strong>--plaplace</strong> Use a Laplace +1 prior in place of the default mixture Dirichlet prior.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING E-VALUE CALIBRATION</h2>
        <div class="sectioncontent">
<p>Estimating the location parameters for the expected score distributions for MSV filter scores, Viterbi filter scores, and Forward scores requires three short random sequence simulations.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--EmL</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the sequence length in simulation that estimates the location parameter mu for MSV filter E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--EmN</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the number of sequences in simulation that estimates the location parameter mu for MSV filter E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--EvL</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the sequence length in simulation that estimates the location parameter mu for Viterbi filter E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--EvN</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the number of sequences in simulation that estimates the location parameter mu for Viterbi filter E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--EfL</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the sequence length in simulation that estimates the location parameter tau for Forward E-values. Default is 100.</p>
  </dd>
  <dt>
    <p><strong>--EfN</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the number of sequences in simulation that estimates the location parameter tau for Forward E-values. Default is 200.</p>
  </dd>
  <dt>
    <p><strong>--Eft</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Sets the tail mass fraction to fit in the simulation that estimates the location parameter tau for Forward evalues. Default is 0.04.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--nonull2</strong></p>
  </dt>
  <dd>
    <p>Turn off the null2 score corrections for biased composition.</p>
  </dd>
  <dt>
    <p><strong>-Z</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Assert that the total number of targets in your searches is <em>&lt;x&gt;</em>, for the purposes of per-sequence E-value calculations, rather than the actual number of targets seen.</p>
  </dd>
  <dt>
    <p><strong>--domZ</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Assert that the total number of targets in your searches is <em>&lt;x&gt;</em>, for the purposes of per-domain conditional E-value calculations, rather than the number of targets that passed the reporting thresholds.</p>
  </dd>
  <dt>
    <p><strong>--seed</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Seed the random number generator with <em>&lt;n&gt;</em>, an integer &gt;= 0. If <em>&lt;n&gt;</em> is &gt;0, any stochastic simulations will be reproducible; the same command will give the same results. If <em>&lt;n&gt;</em> is 0, the random number generator is seeded arbitrarily, and stochastic simulations will vary from run to run of the same command. The default seed is 42.</p>
  </dd>
  <dt>
    <p><strong>--qformat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Declare that the input <em>query_seqfile</em> is in format <em>&lt;s&gt;</em>. Accepted sequence file formats include FASTA, EMBL, GenBank, DDBJ, UniProt, Stockholm, and SELEX. Default is to autodetect the format of the file.</p>
  </dd>
  <dt>
    <p><strong>--tformat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Declare that the input <em>target_seqdb</em> is in format <em>&lt;s&gt;</em>. Accepted sequence file formats include FASTA, EMBL, GenBank, DDBJ, UniProt, Stockholm, and SELEX. Default is to autodetect the format of the file.</p>
  </dd>
  <dt>
    <p><strong>--cpu</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the number of parallel worker threads to <em>&lt;n&gt;</em>. By default, HMMER sets this to the number of CPU cores it detects in your machine - that is, it tries to maximize the use of your available processor cores. Setting <em>&lt;n&gt;</em> higher than the number of available cores is of little if any value, but you may want to set it to something less. You can also control this number by setting an environment variable, <em>HMMER_NCPU</em>.</p><p>This option is only available if HMMER was compiled with POSIX threads support. This is the default, but it may have been turned off at compile-time for your site or machine for some reason.</p>
  </dd>
  <dt>
    <p><strong>--stall</strong></p>
  </dt>
  <dd>
    <p>For debugging the MPI master/worker version: pause after start, to enable the developer to attach debuggers to the running master and worker(s) processes. Send SIGCONT signal to release the pause. (Under gdb: <em>(gdb) signal SIGCONT</em>) (Only available if optional MPI support was enabled at compile-time.)</p>
  </dd>
  <dt>
    <p><strong>--mpi</strong></p>
  </dt>
  <dd>
    <p>Run in MPI master/worker mode, using <em>mpirun</em>. (Only available if optional MPI support was enabled at compile-time.)</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO jackhmmer&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/hmmer.1.html"><strong>hmmer</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the HMMER package.</p><p>For complete documentation, see the user guide that came with your HMMER distribution (Userguide.pdf); or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2013 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="jackdaemon.1.html"><span aria-hidden="true">&larr;</span> jackdaemon.1: Autoradio jack daemon</a></li>
   <li class="next"><a href="jackd1-jackrec.1.html">jackrec.1: Jack toolkit client for recording audio <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
