<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>macs2: Model-based analysis for chip-sequencing</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Model-based analysis for chip-sequencing">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="macs2 (1) manual">
  <meta name="twitter:description" content="Model-based analysis for chip-sequencing">
  <meta name="twitter:image" content="https://www.carta.tech/images/macs-macs2-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/macs2.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="macs2 (1) manual" />
  <meta property="og:description" content="Model-based analysis for chip-sequencing" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/macs-macs2-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">macs2<small> (1)</small></h1>
        <p class="lead">Model-based analysis for chip-sequencing</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/macs2.1.html">
      <span itemprop="name">macs2: Model-based analysis for chip-sequencing</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/macs/">
      <span itemprop="name">macs</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/macs2.1.html">
      <span itemprop="name">macs2: Model-based analysis for chip-sequencing</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>macs2</strong> <em>&lt;-t tfile&gt; </em>[<em>-n name</em>] [<em>-g genomesize</em>] [<em>options</em>]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>Example: macs2 <strong>-t</strong> ChIP.bam <strong>-c</strong> Control.bam <strong>-f</strong> BAM <strong>-g</strong> hs <strong>-n</strong> test <strong>-B</strong> <strong>-q</strong> 0.01</p><p>or example for broad peak calling: macs2 <strong>-t</strong> ChIP.bam <strong>-c</strong> Control.bam <strong>--broad</strong> <strong>-g</strong> hs</p><p>macs2 <strong>--</strong> Model-based Analysis for ChIP-Sequencing</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--version</strong></p>
  </dt>
  <dd>
    <p>show program's version number and exit</p>
  </dd>
  <dt>
    <p><strong>-h</strong>, <strong>--help</strong></p>
  </dt>
  <dd>
    <p>show this help message and exit.</p>
  </dd>
  <dt>
    <p><strong>-t</strong> TFILE, <strong>--treatment</strong>=<em>TFILE</em></p>
  </dt>
  <dd>
    <p>ChIP-seq treatment file. REQUIRED.</p>
  </dd>
  <dt>
    <p><strong>-c</strong> CFILE, <strong>--control</strong>=<em>CFILE</em></p>
  </dt>
  <dd>
    <p>Control file.</p>
  </dd>
  <dt>
    <p><strong>-n</strong> NAME, <strong>--name</strong>=<em>NAME</em></p>
  </dt>
  <dd>
    <p>Experiment name, which will be used to generate output file names. DEFAULT: "NA"</p>
  </dd>
  <dt>
    <p><strong>-f</strong> FORMAT, <strong>--format</strong>=<em>FORMAT</em></p>
  </dt>
  <dd>
    <p>Format of tag file, "AUTO", "BED" or "ELAND" or "ELANDMULTI" or "ELANDEXPORT" or "SAM" or "BAM" or "BOWTIE". The default AUTO option will let MACS decide which format the file is. Please check the definition in 00README file if you choose ELAND/ELANDMULTI/ELANDEXPORT/SAM/BAM/BOWTIE. DEFAULT: "AUTO"</p>
  </dd>
  <dt>
    <p><strong>-g</strong> GSIZE, <strong>--gsize</strong>=<em>GSIZE</em></p>
  </dt>
  <dd>
    <p>Effective genome size. It can be 1.0e+9 or 1000000000, or shortcuts:'hs' for human (2.7e9), 'mm' for mouse (1.87e9), 'ce' for C. elegans (9e7) and 'dm' for fruitfly (1.2e8), Default:hs</p>
  </dd>
  <dt>
    <p><strong>-s</strong> TSIZE, <strong>--tsize</strong>=<em>TSIZE</em></p>
  </dt>
  <dd>
    <p>Tag size. This will overide the auto detected tag size. DEFAULT: Not set</p>
  </dd>
  <dt>
    <p><strong>--bw</strong>=<em>BW</em></p>
  </dt>
  <dd>
    <p>Band width. This value is only used while building the shifting model. DEFAULT: 300</p>
  </dd>
  <dt>
    <p><strong>-q</strong> QVALUE, <strong>--qvalue</strong>=<em>QVALUE</em></p>
  </dt>
  <dd>
    <p>Minimum FDR (q-value) cutoff for peak detection. DEFAULT: 0.01</p>
  </dd>
  <dt>
    <p><strong>-p</strong> PVALUE, <strong>--pvalue</strong>=<em>PVALUE</em></p>
  </dt>
  <dd>
    <p>Pvalue cutoff for peak detection. When set (e.g. <strong>-q</strong> 0.05 or <strong>-q</strong> 1e-5), qvalue cutoff will be ignored. Default is not set.</p>
  </dd>
  <dt>
    <p><strong>-m</strong> MFOLD, <strong>--mfold</strong>=<em>MFOLD</em></p>
  </dt>
  <dd>
    <p>Select the regions within MFOLD range of highconfidence enrichment ratio against background to build model. The regions must be lower than upper limit, and higher than the lower limit. DEFAULT:10,30</p>
  </dd>
  <dt>
    <p><strong>--nolambda</strong></p>
  </dt>
  <dd>
    <p>If True, MACS will use fixed background lambda as local lambda for every peak region. Normally, MACS calculates a dynamic local lambda to reflect the local bias due to potential chromatin structure.</p>
  </dd>
  <dt>
    <p><strong>--slocal</strong>=<em>SMALLLOCAL</em></p>
  </dt>
  <dd>
    <p>The small nearby region in basepairs to calculate dynamic lambda. This is used to capture the bias near the peak summit region. Invalid if there is no control data. If you set this to 0, MACS will skip slocal lambda calculation. *Note* that MACS will always perform a d-size local lambda calculation. The final local bias should be the maximum of the lambda value from d, slocal, and llocal size windows. DEFAULT: 1000</p>
  </dd>
  <dt>
    <p><strong>--llocal</strong>=<em>LARGELOCAL</em></p>
  </dt>
  <dd>
    <p>The large nearby region in basepairs to calculate dynamic lambda. This is used to capture the surround bias. If you set this to 0, MACS will skip llocal lambda calculation. *Note* that MACS will always perform a d-size local lambda calculation. The final local bias should be the maximum of the lambda value from d, slocal, and llocal size windows. DEFAULT: 10000.</p>
  </dd>
  <dt>
    <p><strong>--auto-bimodal</strong></p>
  </dt>
  <dd>
    <p>Whether turn on the auto pair model process. If set, when MACS failed to build paired model, it will use the nomodel settings, the '--shiftsize' parameter to shift and extend each tags. Not to use this automate fixation is a default behavior now. DEFAULT: False</p>
  </dd>
  <dt>
    <p><strong>--nomodel</strong></p>
  </dt>
  <dd>
    <p>Whether or not to build the shifting model. If True, MACS will not build model. by default it means shifting size = 100, try to set shiftsize to change it. DEFAULT: False</p>
  </dd>
  <dt>
    <p><strong>--shiftsize</strong>=<em>SHIFTSIZE</em></p>
  </dt>
  <dd>
    <p>The arbitrary shift size in bp. When nomodel is true, MACS will use this value as 1/2 of fragment size. DEFAULT: 100</p>
  </dd>
  <dt>
    <p><strong>--keep-dup</strong>=<em>KEEPDUPLICATES</em></p>
  </dt>
  <dd>
    <p>It controls the MACS behavior towards duplicate tags at the exact same location <strong>--</strong> the same coordination and the same strand. The default 'auto' option makes MACS calculate the maximum tags at the exact same location based on binomal distribution using 1e-5 as pvalue cutoff; and the 'all' option keeps every tags. If an integer is given, at most this number of tags will be kept at the same location. Default: auto</p>
  </dd>
  <dt>
    <p><strong>--to-large</strong></p>
  </dt>
  <dd>
    <p>When set, scale the small sample up to the bigger sample. By default, the bigger dataset will be scaled down towards the smaller dataset, which will lead to smaller p/qvalues and more specific results. Keep in mind that scaling down will bring down background noise more. DEFAULT: False</p>
  </dd>
  <dt>
    <p><strong>--down-sample</strong></p>
  </dt>
  <dd>
    <p>When set, random sampling method will scale down the bigger sample. By default, MACS uses linear scaling. Warning: This option will make your result unstable and irreproducible since each time, random reads would be selected. Consider to use 'randsample' script instead. DEFAULT: False</p>
  </dd>
  <dt>
    <p><strong>--shift-control</strong></p>
  </dt>
  <dd>
    <p>When set, control tags will be shifted just as ChIP tags according to their strand before the extension of d, slocal and llocal. By default, control tags are extended centered at their current positions regardless of strand. You may consider to turn this option on while comparing two ChIP datasets of different condition but the same factor. DEFAULT: False</p>
  </dd>
  <dt>
    <p><strong>--half-ext</strong></p>
  </dt>
  <dd>
    <p>When set, MACS extends 1/2 d size for each fragment centered at its middle point. DEFAULT: False</p>
  </dd>
  <dt>
    <p><strong>-B</strong>, <strong>--bdg</strong></p>
  </dt>
  <dd>
    <p>Whether or not to save extended fragment pileup, local lambda and score tracks at every bp into a bedGraph file. DEFAULT: False</p>
  </dd>
  <dt>
    <p><strong>--broad</strong></p>
  </dt>
  <dd>
    <p>If set, MACS will try to call broad peaks by linking nearby highly enriched regions. The linking region is controlled by another cutoff through <strong>--linking-cutoff</strong>. The maximum linking region length is 4 times of d from MACS. DEFAULT: False</p>
  </dd>
  <dt>
    <p><strong>--broad-cutoff</strong>=<em>BROADCUTOFF</em></p>
  </dt>
  <dd>
    <p>Cutoff for broad region. This option is not available unless <strong>--broad</strong> is set. If <strong>-p</strong> is set, this is a pvalue cutoff, otherwise, it's a qvalue cutoff. DEFAULT: 0.1</p>
  </dd>
  <dt>
    <p><strong>--verbose</strong>=<em>VERBOSE</em></p>
  </dt>
  <dd>
    <p>Set verbose level. 0: only show critical message, 1: show additional warning message, 2: show process information, 3: show debug messages. DEFAULT:2</p>
  </dd>

</dl>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="macroexecutor.1.html"><span aria-hidden="true">&larr;</span> macroexecutor.1: Manual page for macroexecutor 3.3.1</a></li>
   <li class="next"><a href="macs2diff.1.html">macs2diff.1: Differential analysis for chip-sequencing <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
