<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>melting: Nearest-neighbor computation of nucleic acid hybridation</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Nearest-neighbor computation of nucleic acid hybridation">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="melting (1) manual">
  <meta name="twitter:description" content="Nearest-neighbor computation of nucleic acid hybridation">
  <meta name="twitter:image" content="https://www.carta.tech/images/melting-melting-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/melting.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="melting (1) manual" />
  <meta property="og:description" content="Nearest-neighbor computation of nucleic acid hybridation" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/melting-melting-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">melting<small> (1)</small></h1>
        <p class="lead">Nearest-neighbor computation of nucleic acid hybridation</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/melting.1.html">
      <span itemprop="name">melting: Nearest-neighbor computation of nucleic acid hybridation</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/melting/">
      <span itemprop="name">melting</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/melting.1.html">
      <span itemprop="name">melting: Nearest-neighbor computation of nucleic acid hybridation</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>melting</strong> [<em>options</em>]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>Melting</strong> computes, for a nucleic acid duplex, the enthalpy and the entropy of the helix-coil transition, and then its melting temperature. Three types of hybridisation are possible: DNA/DNA, DNA/RNA, and RNA/RNA. The program uses the method of nearest-neighbors. The set of thermodynamic parameters can be easely changed, for instance following an experimental breakthrough. Melting is a free program in both sense of the term. It comes with no cost and it is open-source. In addition it is coded in ISO C and can be compiled on any operating system. Some perl scripts are provided to show how melting can be used as a block to construct more ambitious programs.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p>The options are treated sequentially. If there is a conflict between the value of two options, the latter normally erases the former.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-A</strong><em>file.nn</em></p>
  </dt>
  <dd>
    <p>Informs the program to use <em>file.nn</em> as an alternative set of nearest-neighbor parameters, rather than the default for the specified hybridisation type. The standard distribution of melting provides some files ready-to-use: <em>all97a.nn</em> (Allawi et al 1997), <em>bre86a.nn</em> (Breslauer et al 1986), <em>san96a.nn</em> (SantaLucia et al 1996), <em>sug96a.nn</em> (Sugimoto et al 1996) <em>san04a.nn</em> (Santalucia et al 2004) (DNA/DNA), <em>fre86a.nn</em> (Freier et al 1986), <em>xia98a.nn</em> (Xia et al 1998), (RNA/RNA), and <em>sug95a.nn</em> (Sugimoto et al 1995), (DNA/RNA).</p><p>The program will look for the file in a directory specified during the installation. However, if an environment variable NN_PATH is defined, melting will search in this one first. Be careful, the option <strong>-A</strong> changes the default parameter set defined by the option <strong>-H.</strong></p>
  </dd>
  <dt>
    <p><strong>-C</strong><em>complementary_sequence</em></p>
  </dt>
  <dd>
    <p>Enters the complementary sequence, from 3' to 5'. This option is mandatory if there are mismatches between the two strands. If it is not used, the program will compute it as the complement of the sequence entered with the option <strong>-S.</strong></p>
  </dd>
  <dt>
    <p><strong>-D</strong><em>dnadnade.nn</em></p>
  </dt>
  <dd>
    <p>Informs the program to use the file <em>dnadnade.nn</em> to compute the contribution of dangling ends to the thermodynamic of helix-coil transition. The dangling ends are not taken into account by the approximative mode.</p>
  </dd>
  <dt>
    <p><strong>-F</strong><em>factor</em></p>
  </dt>
  <dd>
    <p>This is the a correction factor used to modulate the effect of the  nucleic acid concentration in the computation of the melting temperature. See section ALGORITHM for details.</p>
  </dd>
  <dt>
    <p><strong>-G</strong><em>x.xxe-xx</em></p>
  </dt>
  <dd>
    <p>Magnesium  concentration  (No maximum concentration for the moment). The effect</p>
<pre>
   of  ions  on  thermodynamic  stability  of nucleic  acid duplexes is complex,
   and the correctÂ­ing functions are  at  best rough  approximations.The published
   Tm  correction formula for divalent Mg2+ ions of  Owczarzy  et al(2008) can
   take in account the competitive binding of monovalent and divalent ions on DNA.
   However this formula is only for DNA duplexes.
</pre>

  </dd>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Displays a short help and quit with EXIT_SUCCESS.</p>
  </dd>
  <dt>
    <p><strong>-H</strong><em>hybridisation_type</em></p>
  </dt>
  <dd>
    <p>Specifies the hybridisation type. This will set the nearest-neighbor set to use if no alternative set is provided by the option <strong>-A</strong> (remember the options are read sequentially). Moreover this parameter determines the equation to use if the sequence length exceeds the limit of application of the nearest-neighbor approach (arbitrarily set up by the author). Possible values are <em>dnadna,</em> <em>dnarna</em> and <em>rnadna</em> (synonymous), and <em>rnarna.</em> For reasons of compatibility the values of the previous versions of melting <em>A,B,C,F,R,S,T,U,W</em> are still available although <strong>strongly</strong> deprecated. Use the option <strong>-A</strong> to require an alternative set of thermodynamic parameters. IMPORTANT: If the duplex is a DNA/RNA heteroduplex, the sequence of the DNA strand has to be entered with the option <strong>-S.</strong></p>
  </dd>
  <dt>
    <p><strong>-I</strong><em>input_file</em></p>
  </dt>
  <dd>
    <p>Provides the name of an input file containing the parameters of the run. The input has to contain one parameter per line, formatted as in the command line. The order is not important, as well as blank lines. example:</p><p>###beginning###</p><p>-Hdnadna</p><p>-Asug96a.nn</p><p>-SAGCTCGACTC</p><p>-CTCGAGGTGAG</p><p>-N0.2</p><p>-P0.0001</p><p>-v</p><p>-Ksan96a</p><p>###end###</p>
  </dd>
  <dt>
    <p><strong>-i</strong><em>file.nn</em></p>
  </dt>
  <dd>
    
<pre>
  Informs  the  program to use file.nn as an alternative set  of  inosine pair
  parameters, rather than  the  default  for the specified hybridisation type.
  The standard distribution of melting provides some  files ready-to-use:  san05a.nn
</pre>
<p> (Santalucia et al 2005) for deoxyinosine in DNA duplexes, bre07a.nn (Brent M Znosko</p>
<pre>
  et al 2007)for inosine in RNA duplexes. Note  that  not all the inosine mismatched
  wobble's pairs have been investigated. Therefore it could be impossible to  compute
  the Tm of a duplex with inosine pairs. Moreover, those inosine pairs are not taken
  into account by the  approximative mode.
</pre>

  </dd>
  <dt>
    <p><strong>-K</strong><em>salt_correction</em></p>
  </dt>
  <dd>
    <p>Permits to chose another correction for the concentration in sodium. Currently, one can chose between <em>wet91a, san96a, san98a.</em> See section ALGORITHM. TP. BI. "-k" "x.xxe-xx"</p>
<pre>
   Potassium  concentration  (No maximum concentration for the moment). The effect of ions
   on  thermodynamic  stability  of nucleic  acid duplexes is complex, and the correctÂ­ing
   functions are  at  best rough  approximations.The published  Tm  correction formula for
   sodium ions of Owczarzy et al (2008)is therefore also applicable to buffers containing Tris or
   KCl. Monovalent K+, Na+, Tris+ ions  stabilize  DNA duplexes
   with similar potency, and their effects on duplex stability are additive. However this formula
   is only for DNA duplexes.
</pre>

  </dd>
  <dt>
    <p><strong>-L</strong></p>
  </dt>
  <dd>
    <p>Prints the legal informations and quit with EXIT_SUCCESS.</p>
  </dd>
  <dt>
    <p><strong>-M</strong><em>dnadnamm.nn</em></p>
  </dt>
  <dd>
    <p>Informs the program to use the file <em>dnadnamm.nn</em> to compute the contribution of mismatches to the thermodynamic of helix-coil transition. Note that not all the mismatched Crick's pairs have been investigated. Therefore it could be impossible to compute the Tm of a mismatched duplex. Moreover, those mismatches are not taken into account by the approximative mode.</p>
  </dd>
  <dt>
    <p><strong>-N</strong><em>x.xxe-xx</em></p>
  </dt>
  <dd>
    <p>Sodium concentration (between 0 and 10 M). The effect of ions on thermodynamic</p>
<pre>
  stability of nucleic acid duplexes is complex, and the correcting functions
  are at best rough approximations. Moreover, they are generally reliable only
  for [Na+] belonging to [0.1,10M]. If there are no other ions in
  solution, we can use only the sodium correction. In the other case, we use the Owczarzy's
  algorithm.
</pre>

  </dd>
  <dt>
    <p><strong>-O</strong><em>output_file</em></p>
  </dt>
  <dd>
    <p>The output is directed to this file instead of the standard output. The name of the file can be omitted. An automatic name is then generated, of the form meltingYYYYMMMDD_HHhMMm.out (of course, on POSIX compliant systems, you can emulate this with the redirection of stdout to a file constructed with the program date).</p>
  </dd>
  <dt>
    <p><strong>-P</strong><em>x.xxe-xx</em></p>
  </dt>
  <dd>
    <p>Concentration of the nucleic acid strand in excess (between 0 and 0.1 M).</p>
  </dd>
  <dt>
    <p><strong>-p</strong></p>
  </dt>
  <dd>
    <p>Return the directory supposed to contain the sets of calorimetric parameters and quit with EXIT_SUCCESS. If the environment variable NN_PATH is set, it is returned. Otherwise, the value defined by default during the compilation is returned.</p>
  </dd>
  <dt>
    <p><strong>-q</strong></p>
  </dt>
  <dd>
    <p>Turn off the interactive correction of wrongly entered parameter. Useful for run through a server, or a batch script. Default is OFF (i.e. interactive on). The switch works in both sens. Therefore if <strong>-q</strong> has been set in an input file, another <strong>-q</strong> on the command line will switch the quiet mode OFF (same thing if two <strong>-q</strong> are set on the same command line).</p>
  </dd>
  <dt>
    <p><strong>-S</strong><em>sequence</em></p>
  </dt>
  <dd>
    <p>Sequence of one strand of the nucleic acid duplex, entered 5' to 3'. IMPORTANT: If it is a DNA/RNA heteroduplex, the sequence of the DNA strand has to be entered. Uridine and thymidine are considered as identical. The bases can be upper or lowercase.</p>
  </dd>
  <dt>
    <p><strong>-T</strong><em>xxx</em></p>
  </dt>
  <dd>
    <p>Size threshold before approximative computation. The nearest-neighbour approach will be used only if the length of the sequence is inferior to this threshold.</p>
  </dd>
  <dt>
    <p><strong>-t</strong><em>x.xxe-xx</em></p>
  </dt>
  <dd>
    <p>Tris buffer  concentration  (No maximum concentration for the moment).</p>
<pre>
   The effect  of  ions  on  thermodynamic  stability  of nucleic  acid
   duplexes is complex, and the correctÂ­ing functions are  at  best
   rough  approximations.The published  Tm  correction formula for sodium ions of
   Owczarzy et al(2008)is therefore also applicable to buffers containing Tris or
   KCl. Monovalent K+, Na+, Tris+ ions  stabilize  DNA duplexes with similar potency, and
   their effects on duplex stability are additive. However this formula is only for DNA
   duplexes. Be careful, the Tris+ ion concentration is about half of the total tris buffer
   concentration.
</pre>

  </dd>
  <dt>
    <p><strong>-v</strong></p>
  </dt>
  <dd>
    <p>Control the verbose mode, issuing a lot more information about the current run (try it once to see if you can get something interesting). Default is OFF. The switch works in both sens. Therefore if <strong>-v</strong> has been set in an input file, another <strong>-v</strong> on the command line will switch the verbose mode OFF (same thing if two <strong>-v</strong> are set on the same command line).</p>
  </dd>
  <dt>
    <p><strong>-V</strong></p>
  </dt>
  <dd>
    <p>Displays the version number and quit with EXIT_SUCCESS.</p>
  </dd>
  <dt>
    <p><strong>-x</strong></p>
  </dt>
  <dd>
    <p>Force the program to compute an approximative tm, based on G+C content. This option has to be used with caution. Note that such a calcul is increasingly incorrect when the length of the duplex decreases. Moreover, it does not take into account nucleic acid concentration, which is a strong mistake.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ALGORITHM</h2>
        <div class="sectioncontent">
<h3>Thermodynamics of helix-coil transition of nucleic acid</h3>
<p>The nearest-neighbor approach is based on the fact that the helix-coil transition works as a zipper. After an initial attachment, the hybridisation propagates laterally. Therefore, the process depends on the adjacent nucleotides on each strand (the Crick's pairs). Two duplexes with the same base pairs could have different stabilities, and on the contrary, two duplexes with different sequences but identical sets of Crick's pairs will have the same thermodynamics properties (see Sugimoto et al. 1994). This program first computes the hybridisation enthalpy and entropy from the elementary parameters of each Crick's pair.</p><p>DeltaH = deltaH(initiation) + SUM(deltaH(Crick's pair))</p><p>DeltaS = deltaS(initiation) + SUM(deltaS(Crick's pair))</p><p>See Wetmur J.G. (1991) and SantaLucia (1998) for deep reviews on the nucleic acid hybridisation and on the different set of nearest-neighbor parameters.</p>
<h3>Effect of mismatches and dangling ends</h3>
<p>The mismatching pairs are also taken into account. However the thermodynamic parameters are still not available for every possible cases (notably when both positions are mismatched). In such a case, the program, unable to compute any relevant result, will quit with a warning.</p><p>The two first and positions cannot be mismatched. in such a case, the result is unpredictable, and all cases are possible. for instance (see Allawi and SanLucia 1997), the duplex</p><p>A          T</p><p> GTGAGCTCAT</p><p> TACTCGAGTG</p><p>T          A</p><p>is more stable than</p><p>AGTGAGCTCATT</p><p>TTACTCGAGTGA</p><p>The dangling ends, that is the umatched terminal nucleotides, can be taken into account.</p>
<h3>Example</h3>
<p>DeltaH(</p><p>AGCGATGAA-</p><p>-CGCTGCTTT</p><p>) = DeltaH(AG/-C)+DeltaH(A-/TT)</p><p>+DeltaH(initG/C)+DeltaH(initA/T)</p><p>+DeltaH(GC/CG)+DeltaH(CG/GC)+2xDeltaH(GA/CT)+DeltaH(AA/TT)</p><p>+Delta(AT/TG mismatch) +DeltaG(TC/GG mismatch)</p><p>(The same computation is performed for DeltaS)</p>
<h3>The melting temperature</h3>
<p>Then the melting temperature is computed by the following formula:</p><p>Tm = DeltaH / (DeltaS + Rx ln ([nucleic acid]/F))</p><p><em>Tm in K</em> (for [Na+] = 1 M )</p>
<pre>
     + f([Na+]) - 273.15
</pre>
<p><em>correction</em> for the salt concentration (if there are only sodium cations in the solution)and to get the temperature in degree Celsius. (In fact some corrections are directly included in the DeltaS see that of SanLucia 1998)</p>
<h3>Correction for the concentration of nucleic acid</h3>
<p>If the concentration of the two strands are similar, F is 1 in case of self-complementary oligonucleotides, 4 otherwise. If one strand is in excess (for instance in PCR experiment), F is 2 (Actually the formula would have to use the difference of concentrations rather than the total concentration, but if the excess is sufficient, the total concentration can be assumed to be identical to the concentration of the strand in excess).</p><p>Note however, MELTING makes the assumption of no self-assembly, <em>i.e.</em> the computation does not take any entropic term to correct for self-complementarity.</p>
<h3>Correction for the concentration of salt</h3>
<p>If there are only sodium ions in the solution, we can use the following corrections:</p><p>The correction can be chosen between <em>wet91a,</em> presented in Wetmur 1991 <em>i.e.</em></p><p>16.6 x log([Na+] / (1 + 0.7 x [Na+])) + 3.85</p><p><em>san96a</em> presented in SantaLucia et al. 1996 <em>i.e.</em></p><p>12.5 x log[Na+]</p><p>and <em>san98a</em> presented in SantaLucia 1998 <em>i.e.</em> a correction of the entropic term without modification of enthalpy</p><p>DeltaS = DeltaS([Na+]=1M) + 0.368 x (N-1) x ln[Na+]</p><p>Where N is the length of the duplex (SantaLucia 1998 actually used 'N' the number of non-terminal phosphates, that is effectively equal to our N-1). CAUTION, this correction is meant to correct entropy values expressed in cal.mol-1.K-1!!!</p>
<h3>Correction for the concentration of ions when other monovalent ions such as Tris+ and K+ or divalent Mg2+ ions are added</h3>
<p>If there are only Na+ ions, we can use the correction for the concentration of salt(see above). In the opposite case , we will use the magnesium and monovalent ions correction from Owczarzy et al (2008). (only for DNA duplexes)</p><p>[Mon+] = [Na+] + [K+] + [Tris+]</p><p>Where [Tris+] = [Tris buffer]/2. (in the option -t, it is the Tris buffer concentration which is entered).</p><p>If [Mon+] = 0, the divalent ions are the only ions present  and the melting temperature is :</p><p>1/Tm(Mg2+) = 1/Tm(1M Na+) + a - b x ln([Mg2+]) + Fgc x (c + d x ln([Mg2+]) + 1/(2 x (Nbp - 1)) x (- e +f x ln([Mg2+]) + g x ln([Mg2+]) x ln([Mg2+]))</p><p>where : a = 3.92/100000. b = 9.11/1000000. c = 6.26/100000. d = 1.42/100000. e = 4.82/10000. f = 5.25/10000. g = 8.31/100000. Fgc is the fraction of GC base pairs in the sequence and Nbp is the length of the sequence (Number of base pairs).</p><p>If [Mon+] &gt; 0, there are several cases because we can have a competitive DNA binding between monovalent and divalent cations  :</p><p>If the ratio [Mg2+]^(0.5)/[Mon+] is inferior to 0.22, monovalent ion influence is dominant, divalent cations can be disregarded and the melting temperature is :</p><p>1/Tm(Mg2+) = 1/Tm(1M Na+) + (4.29 x Fgc - 3.95) x 1/100000 x ln([mon+]) + 9.40 x 1/1000000 x ln([Mon+]) x ln([Mon+])</p><p>where : Fgc is the fraction of GC base pairs in the sequence.</p><p>If the ratio [Mg2+]^(0.5)/[Mon+] is included in [0.22, 6[, we must take in account both Mg2+ and monovalent cations concentrations. The melting temperature is :</p><p>1/Tm(Mg2+) = 1/Tm(1M Na+) + a - b x ln([Mg2+]) + Fgc x (c + d x ln([Mg2+]) + 1/(2 x (Nbp - 1)) x (- e + f x ln([Mg2+]) + g x ln([Mg2+]) x ln([Mg2+]))</p><p>where : a = 3.92/100000 x (0.843 - 0.352 x [Mon+]0.5 x ln([Mon+])).</p>
<pre>
        b = 9.11/1000000.
</pre>
<p>	c = 6.26/100000.</p>
<pre>
        d = 1.42/100000 x (1.279 - 4.03/1000 x ln([mon+]) - 8.03/1000 x
</pre>
<p>	ln([mon+] x ln([mon+]). 	e = 4.82/10000. 	f = 5.25/10000. 	g = 8.31/100000 x (0.486 - 0.258 x ln([mon+]) + 5.25/1000 x ln([mon+] x ln([mon+] x ln([mon+]).</p><p>Fgc is the fraction of GC base pairs in the sequence and Nbp is the length of the sequence (Number of base pairs).</p><p>Finally, if the ratio [Mg2+]^(0.5)/[Mon+] is superior to 6, divalent ion influence is dominant, monovalent cations can be disregarded and the melting temperature is :</p><p>1/Tm(Mg2+) = 1/Tm(1M Na+) + a - b x ln([Mg2+]) + Fgc x (c + d x ln([Mg2+]) + 1/(2 x (Nbp - 1)) x (- e + f x ln([Mg2+]) + g x ln([Mg2+]) x ln([Mg2+]))</p><p>where : a = 3.92/100000. b = 9.11/1000000. c = 6.26/100000. d = 1.42/100000. e = 4.82/10000. f = 5.25/10000. g = 8.31/100000.</p><p>Fgc is the fraction of GC base pairs in the sequence and Nbp is the length of the sequence (Number of base pairs).</p>
<h3>Long sequences</h3>
<p>It is important to realise that the nearest-neighbor approach has been established on small oligonucleotides. Therefore the use of melting in the non-approximative mode is really accurate only for relatively short sequences (Although if the sequences are two short, let's say &lt; 6 bp, the influence of extremities becomes too important and the reliability decreases a lot). For long sequences an approximative mode has been designed. This mode is launched if the sequence length is higher than the value given by the option -T (the default threshold is 60 bp).</p><p>The melting temperature is computed by the following formulas:</p><p>DNA/DNA:</p><p>Tm = 81.5+16.6*log10([Na+]/(1+0.7[Na+]))+0.41%GC-500/size</p><p>DNA/RNA:</p><p>Tm = 67+16.6*log10([Na+]/(1.0+0.7[Na+]))+0.8%GC-500/size</p><p>RNA/RNA:</p><p>Tm = 78+16.6*log10([Na+]/(1.0+0.7[Na+]))+0.7%GC-500/size</p><p>This mode is nevertheless <strong>strongly disencouraged.</strong></p>
<h3>Miscellaneous comments</h3>
<p>Melting is currently accurate only when the hybridisation is performed at pH 71.</p><p>The computation is valid only for the hybridisations performed in aqueous medium. Therefore the use of denaturing agents such as formamide completely invalidates the results.</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REFERENCES</h2>
        <div class="sectioncontent">
<p>Allawi H.T., SantaLucia J. (1997). Thermodynamics and NMR of internal G.T mismatches in DNA. <em>Biochemistry</em> 36: 10581-10594</p><p>Allawi H.T., SantaLucia J. (1998). Nearest Neighbor thermodynamics parameters for internal G.A mismatches in DNA. <em>Biochemistry</em> 37: 2170-2179</p><p>Allawi H.T., SantaLucia J. (1998). Thermodynamics of internal C.T mismatches in DNA. <em>Nucleic Acids Res</em> 26: 2694-2701.</p><p>Allawi H.T., SantaLucia J. (1998). Nearest Neighbor thermodynamics of internal A.C mismatches in DNA: sequence dependence and pH effects. <em>Biochemistry</em> 37: 9435-9444.</p><p>Bommarito S., Peyret N., SantaLucia J. (2000). Thermodynamic parameters for DNA sequences with dangling ends. <em>Nucleic Acids Res</em> 28: 1929-1934</p><p>Breslauer K.J., Frank R., BlÃ¯Â¿Â½ker H., Marky L.A. (1986). Predicting DNA duplex stability from the base sequence. <em>Proc Natl Acad Sci USA</em> 83: 3746-3750</p><p>Freier S.M., Kierzek R., Jaeger J.A., Sugimoto N., Caruthers M.H., Neilson T., Turner D.H. (1986). Improved free-energy parameters for predictions of RNA duplex stability. <em>Biochemistry</em> 83:9373-9377</p><p>Owczarzy R., Moreira B.G., You Y., Behlke M.B., Walder J.A. (2008) Predicting stability of DNA duplexes in solutions containing Magnesium and Monovalent Cations. Biochemistry 47: 5336-5353.</p><p>Peyret N., Seneviratne P.A., Allawi H.T., SantaLucia J. (1999). Nearest Neighbor thermodynamics and NMR of DNA sequences with internal A.A, C.C, G.G and T.T mismatches. dependence and pH effects. <em>Biochemistry</em> 38: 3468-3477</p><p>SantaLucia J. Jr, Allawi H.T., Seneviratne P.A. (1996). Improved nearest-neighbor parameters for predicting DNA duplex stability. <em>Biochemistry</em> 35: 3555-3562</p><p>Sugimoto N., Katoh M., Nakano S., Ohmichi T., Sasaki M. (1994). RNA/DNA hybrid duplexes with identical nearest-neighbor base-pairs hve identical stability. <em>FEBS Letters</em> 354: 74-78</p><p>Sugimoto N., Nakano S., Katoh M., Matsumura A., Nakamuta H., Ohmichi T., Yoneyama M., Sasaki M. (1995). Thermodynamic parameters to predict stability of RNA/DNA hybrid duplexes. <em>Biochemistry</em> 34: 11211-11216</p><p>Sugimoto N., Nakano S., Yoneyama M., Honda K. (1996). Improved thermodynamic parameters and helix initiation factor to predict stability of DNA duplexes. <em>Nuc Acids Res</em> 24: 4501-4505</p><p>Watkins N.E., Santalucia J. Jr. (2005). Nearest-neighbor t- hermodynamics of deoxyinosine pairs in DNA duplexes. Nucleic Acids Research 33: 6258-6267</p><p>Wright D.J., Rice J.L., Yanker D.M., Znosko B.M. (2007). Nearest neighbor parameters for inosine-uridine pairs in RNA duplexes. Biochemistry 46: 4625-4634</p><p>Xia T., SantaLucia J., Burkard M.E., Kierzek R., Schroeder S.J., Jiao X., Cox C., Turner D.H. (1998). Thermodynamics parameters for an expanded nearest-neighbor model for formation of RNA duplexes with Watson-Crick base pairs. <em>Biochemistry</em> 37: 14719-14735</p><p>For review see:</p><p>SantaLucia J. (1998) A unified view of polymer, dumbbell, and oligonucleotide DNA nearest-neighbor thermodynamics. <em>Proc Natl Acad Sci USA</em> 95: 1460-1465</p><p>SantaLucia  J., Hicks Donald (2004) The Thermodynamics of DNA structural motifs. Annu. Rev. Biophys. Struct. 33: 415 -440</p><p>Wetmur J.G. (1991) DNA probes: applications of the principles of nucleic acid hybridization. <em>Crit Rev Biochem Mol Biol</em> 26: 227-259</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FILES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><em>*.nn</em></p>
  </dt>
  <dd>
    <p>Files containing the nearest-neighbor parameters, enthalpy and entropy, for each Crick's pair. They have to be placed in a directory defined during the compilation or targeted by the environment variable NN_PATH.</p>
  </dd>
  <dt>
    <p><em>tkmelting.pl</em></p>
  </dt>
  <dd>
    <p>A Graphical User Interface written in Perl/Tk is available for those who prefer the 'button and menu' approach.</p>
  </dd>
  <dt>
    <p><em>*.pl</em></p>
  </dt>
  <dd>
    <p>Scripts are available to use MELTING iteratively. For instance, the script multi.pl permits to predict the Tm of several duplexes in one shot. The script profil.pl allow an interactive computation along a sequence, by sliding a window of specified width.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO melting&hellip;</h2>
        <div class="sectioncontent">
<p>New versions and related material can be found at http://www.ebi.ac.uk/compneur/melting/ and at at https://sourceforge.net/projects/melting/</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">KNOWN BUGS</h2>
        <div class="sectioncontent">
<p>The infiles have to be ended by a blank line because otherwise the last line is not decoded.</p><p>If an infile is called, containing the address of another input file, it does not care of this latter. If it is its own address, the program quit (is it a bug or a feature?).</p><p>In interactive mode, a sequence can be entered on several lines with a backslash</p><p>AGCGACGAGCTAGCCTA&#92;</p><p>AGGACCTATACGAC</p><p>If by mistake it is entered as</p><p>AGCGACGAGCTAGCCTA&#92;AGGACCTATACGAC</p><p>The backslash will be considered as an illegal character. Here again, I do not think it is actually a bug (even if it is unlikely, there is a small probability that the backslash could actually be a mistyped base).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">
<p>Melting is copyright (C) 1997, 2013 by Nicolas Le NovÃ¨re and Marine Dumousseau</p><p>This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.</p><p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.</p><p>You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ACKNOWLEDGEMENTS</h2>
        <div class="sectioncontent">
<p>Nicolas Joly is an efficient and kind debugger and advisor.  Catherine Letondal wrote the HTML interface to melting. Thanks to Nirav Merchant, Taejoon Kwon, Leo Schalkwyk, Mauro Petrillo, Andrew Thompson, Wong Chee Hong, Ivano Zara for their bug fixes and comments.  Thanks to Richard Owczarzy for his magnesium correction. Thanks to Charles Plessy for the graphical interface files. Markus Piotrowski updated TkMELTING to cover version 4.3.  Finally thanks to the usenet helpers, particularly Olivier Dehon and Nicolas Chuche.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p>Nicolas Le NovÃ¨re Babraham Institute, Babraham Research Campus Babraham CB22 3AT Cambridge United-Kingdom. n.lenovere@gmail.com</p><p>Marine Dumousseau EMBL-EBI, Wellcome-Trust Genome Campus Hinxton CB10 1SD Cambridge United-Kingdom. marine@ebi.ac.uk</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">HISTORY</h2>
        <div class="sectioncontent">
<p>See the file ChangeLog for the changes of the versions 4 and more recent.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="meltbuild.1.html"><span aria-hidden="true">&larr;</span> meltbuild.1: Compiles melt sources into dvi, postscript or pdf files</a></li>
   <li class="next"><a href="meltpp.1.html">meltpp.1: The melt pre-processor <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
