<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>nhmmer: Search dna/rna queries against a dna/rna sequence database</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Search dna/rna queries against a dna/rna sequence database">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="nhmmer (1) manual">
  <meta name="twitter:description" content="Search dna/rna queries against a dna/rna sequence database">
  <meta name="twitter:image" content="https://www.carta.tech/images/hmmer-nhmmer-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/nhmmer.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="nhmmer (1) manual" />
  <meta property="og:description" content="Search dna/rna queries against a dna/rna sequence database" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/hmmer-nhmmer-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">nhmmer<small> (1)</small></h1>
        <p class="lead">Search dna/rna queries against a dna/rna sequence database</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/nhmmer.1.html">
      <span itemprop="name">nhmmer: Search dna/rna queries against a dna/rna sequence database</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/hmmer/">
      <span itemprop="name">hmmer</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/nhmmer.1.html">
      <span itemprop="name">nhmmer: Search dna/rna queries against a dna/rna sequence database</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>nhmmer</strong> <em>[options]</em> <em>&lt;queryfile&gt;</em> <em>&lt;seqdb&gt;</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>nhmmer</strong> is used to search one or more nucleotide queries against a nucleotide sequence database. For each query in <em>&lt;queryfile&gt;</em>, use that query to search the target database of sequences in <em>&lt;seqdb&gt;</em>, and output a ranked list of the hits with the most significant matches to the query. A query may be either a profile model built using <strong>hmmbuild</strong>, a sequence alignment, or a single sequence. Sequence based queries can be in a number of formats (see <strong>--qformat</strong>), and can typically be autodetected. Note that only <em>Stockholm</em> format supports the use of multiple sequence-based queries.</p><p>Either the query <em>&lt;queryfile&gt;</em> or the target <em>&lt;seqdb&gt;</em> may be '-' (a dash character), in which case the query file or target database input will be read from a &lt;stdin&gt; pipe instead of from a file. Only one input source can come through &lt;stdin&gt;, not both. If the query is sequence-based and passed via &lt;stdin&gt;, the <strong>--qformat</strong> flag must be used. If the <em>&lt;queryfile&gt;</em> contains more than one query, then <em>&lt;seqdb&gt;</em> cannot come from &lt;stdin&gt;, because we can't rewind the streaming target database to search it with another profile.</p><p>If the query is sequence-based, and not from &lt;stdin&gt;, a new file containing the HMM(s) built from the input(s) in <em>&lt;queryfile&gt;</em> may optionally be produced, with the filename set using the <strong>--hmmout</strong> flag.</p><p>The output format is designed to be human-readable, but is often so voluminous that reading it is impractical, and parsing it is a pain. The <strong>--tblout</strong> option saves output in a simple tabular format that is concise and easier to parse. The <strong>-o</strong> option allows redirecting the main output, including throwing it away in /dev/null.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help; print a brief reminder of command line usage and all available options.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR CONTROLLING OUTPUT</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-o</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Direct the main human-readable output to a file <em>&lt;f&gt;</em> instead of the default stdout.</p>
  </dd>
  <dt>
    <p><strong>-A</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save a multiple alignment of all significant hits (those satisfying <em>inclusion thresholds</em>) to the file <em>&lt;f&gt;</em>.</p>
  </dd>
  <dt>
    <p><strong>--tblout</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save a simple tabular (space-delimited) file summarizing the per-target output, with one data line per homologous target sequence found.</p>
  </dd>
  <dt>
    <p><strong>--dfamtblout</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save a tabular (space-delimited) file summarizing the per-hit output, similar to <strong>--tblout</strong> but more succinct.</p>
  </dd>
  <dt>
    <p><strong>--aliscoresout</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>Save to file a list of per-position scores for each hit. This is useful, for example, in identifying regions of high score density for use in resolving overlapping hits from different models.</p>
  </dd>
  <dt>
    <p><strong>--hmmout</strong><em> &lt;f&gt;</em></p>
  </dt>
  <dd>
    <p>If &lt;queryfile&gt; is sequence-based, the internally-computed HMM(s) are written to <em>&lt;f&gt;</em>.</p>
  </dd>
  <dt>
    <p><strong>--acc</strong></p>
  </dt>
  <dd>
    <p>Use accessions instead of names in the main output, where available for profiles and/or sequences.</p>
  </dd>
  <dt>
    <p><strong>--noali</strong></p>
  </dt>
  <dd>
    <p>Omit the alignment section from the main output. This can greatly reduce the output volume.</p>
  </dd>
  <dt>
    <p><strong>--notextw</strong></p>
  </dt>
  <dd>
    <p>Unlimit the length of each line in the main output. The default is a limit of 120 characters per line, which helps in displaying the output cleanly on terminals and in editors, but can truncate target profile description lines.</p>
  </dd>
  <dt>
    <p><strong>--textw</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the main output's line length limit to <em>&lt;n&gt;</em> characters per line. The default is 120.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING REPORTING THRESHOLDS</h2>
        <div class="sectioncontent">
<p>Reporting thresholds control which hits are reported in output files (the main output, <em>--tblout</em>, and <em>--dfamtblout</em>). Hits are ranked by statistical significance (E-value).</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-E</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Report target sequences with an E-value of &lt;= <em>&lt;x&gt;</em>. The default is 10.0, meaning that on average, about 10 false positives will be reported per query, so you can see the top of the noise and decide for yourself if it's really noise.</p>
  </dd>
  <dt>
    <p><strong>-T</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Instead of thresholding output on E-value, instead report target sequences with a bit score of &gt;= <em>&lt;x&gt;</em>.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR INCLUSION THRESHOLDS</h2>
        <div class="sectioncontent">
<p>Inclusion thresholds are stricter than reporting thresholds. Inclusion thresholds control which hits are considered to be reliable enough to be included in an output alignment or a subsequent search round, or marked as significant ("!") as opposed to questionable ("?") in hit output.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--incE</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Use an E-value of &lt;= <em>&lt;x&gt;</em> as the inclusion threshold. The default is 0.01, meaning that on average, about 1 false positive would be expected in every 100 searches with different query sequences.</p>
  </dd>
  <dt>
    <p><strong>--incT</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Instead of using E-values for setting the inclusion threshold, use a bit score of &gt;= <em>&lt;x&gt;</em> as the inclusion threshold. By default this option is unset.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS FOR MODEL-SPECIFIC SCORE THRESHOLDING</h2>
        <div class="sectioncontent">
<p>Curated profile databases may define specific bit score thresholds for each profile, superseding any thresholding based on statistical significance alone.</p><p>To use these options, the profile must contain the appropriate (GA, TC, and/or NC) optional score threshold annotation; this is picked up by <strong>hmmbuild</strong> from Stockholm format alignment files. For a nucleotide model, each thresholding option has a single per-hit threshold &lt;x&gt; This acts as if <strong>-T</strong><em>&lt;x&gt;</em> <strong>--incT</strong><em>&lt;x&gt;</em> has been applied specifically using each model's curated thresholds.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--cut_ga</strong></p>
  </dt>
  <dd>
    <p>Use the GA (gathering) bit score threshold in the model to set per-hit reporting and inclusion thresholds. GA thresholds are generally considered to be the reliable curated thresholds defining family membership; for example, in Dfam, these thresholds are applied when annotating a genome with a model of a family known to be found in that organism. They may allow for minimal expected false discovery rate.</p>
  </dd>
  <dt>
    <p><strong>--cut_nc</strong></p>
  </dt>
  <dd>
    <p>Use the NC (noise cutoff) bit score threshold in the model to set per-hit reporting and inclusion thresholds. NC thresholds are less stringent than GA; in the context of Pfam, they are generally used to store the score of the highest-scoring known false positive.</p>
  </dd>
  <dt>
    <p><strong>--cut_tc</strong></p>
  </dt>
  <dd>
    <p>Use the NC (trusted cutoff) bit score threshold in the model to set per-hit reporting and inclusion thresholds. TC thresholds are more stringent than GA, and are generally considered to be the score of the lowest-scoring known true positive that is above all known false positives; for example, in Dfam, these thresholds are applied when annotating a genome with a model of a family not known to be found in that organism.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS CONTROLLING THE ACCELERATION PIPELINE</h2>
        <div class="sectioncontent">
<p>HMMER3 searches are accelerated in a three-step filter pipeline: the scanning-SSV filter, the Viterbi filter, and the Forward filter. The first filter is the fastest and most approximate; the last is the full Forward scoring algorithm. There is also a bias filter step between SSV and Viterbi. Targets that pass all the steps in the acceleration pipeline are then subjected to postprocessing -- domain identification and scoring using the Forward/Backward algorithm.</p><p>Changing filter thresholds only removes or includes targets from consideration; changing filter thresholds does not alter bit scores, E-values, or alignments, all of which are determined solely in postprocessing.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--max</strong></p>
  </dt>
  <dd>
    <p>Turn off (nearly) all filters, including the bias filter, and run full Forward/Backward postprocessing on most of the target sequence. In contrast to <strong>hmmscan</strong>, the <em>--max</em> flag in <strong>nhmmscan</strong> sets the scanning-SSV filter threshold to 0.4, not 1.0. Use of this flag increases sensitivity somewhat, at a large cost in speed.</p>
  </dd>
  <dt>
    <p><strong>--F1</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the P-value threshold for the SSV filter step.  The default is 0.02, meaning that roughly 2% of the highest scoring nonhomologous targets are expected to pass the filter.</p>
  </dd>
  <dt>
    <p><strong>--F2</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the P-value threshold for the Viterbi filter step. The default is 0.001.</p>
  </dd>
  <dt>
    <p><strong>--F3</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Set the P-value threshold for the Forward filter step. The default is 1e-5.</p>
  </dd>
  <dt>
    <p><strong>--nobias</strong></p>
  </dt>
  <dd>
    <p>Turn off the bias filter. This increases sensitivity somewhat, but can come at a high cost in speed, especially if the query has biased residue composition (such as a repetitive sequence region, or if it is a membrane protein with large regions of hydrophobicity). Without the bias filter, too many sequences may pass the filter with biased queries, leading to slower than expected performance as the computationally intensive Forward/Backward algorithms shoulder an abnormally heavy load.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OTHER OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--tformat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Assert that the target sequence database file is in format <em>&lt;s&gt;</em>. Accepted formats include <em>fasta</em>, <em>embl</em>, <em>genbank</em>, <em>ddbj</em>, <em>uniprot</em>, <em>stockholm</em>, <em>pfam</em>, <em>a2m</em>, and <em>afa</em>. The default is to autodetect the format of the file.</p>
  </dd>
  <dt>
    <p><strong>--qformat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Declare that the input <em>queryfile</em> is in format <em>&lt;s&gt;</em>. This is used when the query is sequence-based, rather than made up of profile model(s). Currently the accepted multiple alignment sequence file formats include Stockholm, Aligned FASTA, Clustal, NCBI PSI-BLAST, PHYLIP, Selex, and UCSC SAM A2M. Default is to autodetect the format of the file.</p>
  </dd>
  <dt>
    <p><strong>--nonull2</strong></p>
  </dt>
  <dd>
    <p>Turn off the null2 score corrections for biased composition.</p>
  </dd>
  <dt>
    <p><strong>-Z</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>For the purposes of per-hit E-value calculations, Assert that the total size of the target database is <em>&lt;x&gt;</em> million nucleotides, rather than the actual number of targets seen.</p>
  </dd>
  <dt>
    <p><strong>--seed</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the random number seed to <em>&lt;n&gt;</em>. Some steps in postprocessing require Monte Carlo simulation.  The default is to use a fixed seed (42), so that results are exactly reproducible. Any other positive integer will give different (but also reproducible) results. A choice of 0 uses a randomly chosen seed.</p>
  </dd>
  <dt>
    <p><strong>--w_beta</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Window length tail mass. The upper bound, <em>W</em>, on the length at which nhmmer expects to find an instance of the model is set such that the fraction of all sequences generated by the model with length <em>&gt;= W</em> is less than <em>&lt;x&gt;</em>. The default is 1e-7. This flag may be used to override the value of <em>W</em> established for the model by <strong>hmmbuild</strong>, or when the query is sequence-based.</p>
  </dd>
  <dt>
    <p><strong>--w_length</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Override the model instance length upper bound, <em>W</em>, which is otherwise controlled by <strong>--w_beta</strong>. It should be larger than the model length. The value of <em>W</em> is used deep in the acceleration pipeline, and modest changes are not expected to impact results (though larger values of <em>W</em> do lead to longer run time). This flag may be used to override the value of <em>W</em> established for the model by <strong>hmmbuild</strong>, or when the query is sequence-based.</p>
  </dd>
  <dt>
    <p><strong>--toponly</strong></p>
  </dt>
  <dd>
    <p>Only search the top strand. By default both the query sequence and its reverse-complement are searched.</p>
  </dd>
  <dt>
    <p><strong>--bottomonly</strong></p>
  </dt>
  <dd>
    <p>Only search the bottom (reverse-complement) strand. By default both the query sequence and its reverse-complement are searched.</p>
  </dd>
  <dt>
    <p><strong>--cpu</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Set the number of parallel worker threads to <em>&lt;n&gt;</em>. By default, HMMER sets this to the number of CPU cores it detects in your machine - that is, it tries to maximize the use of your available processor cores. Setting <em>&lt;n&gt;</em> higher than the number of available cores is of little if any value, but you may want to set it to something less. You can also control this number by setting an environment variable, <em>HMMER_NCPU</em>.</p><p>This option is only available if HMMER was compiled with POSIX threads support. This is the default, but it may have been turned off at compile-time for your site or machine for some reason.</p>
  </dd>
  <dt>
    <p><strong>--stall</strong></p>
  </dt>
  <dd>
    <p>For debugging the MPI master/worker version: pause after start, to enable the developer to attach debuggers to the running master and worker(s) processes. Send SIGCONT signal to release the pause. (Under gdb: <em>(gdb) signal SIGCONT</em>) (Only available if optional MPI support was enabled at compile-time.)</p>
  </dd>
  <dt>
    <p><strong>--mpi</strong></p>
  </dt>
  <dd>
    <p>Run in MPI master/worker mode, using <em>mpirun</em>. (Only available if optional MPI support was enabled at compile-time.)</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO nhmmer&hellip;</h2>
        <div class="sectioncontent">
<p>See <a href="../man1/hmmer.1.html"><strong>hmmer</strong>(1)</a></strong> for a master man page with a list of all the individual man pages for programs in the HMMER package.</p><p>For complete documentation, see the user guide that came with your HMMER distribution (Userguide.pdf); or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">

<pre>
Copyright (C) 2013 Howard Hughes Medical Institute.
Freely distributed under the GNU General Public License (GPLv3).
</pre>
<p>For additional information on copyright and licensing, see the file called COPYRIGHT in your HMMER source distribution, or see the HMMER web page ().</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">

<pre>
Eddy/Rivas Laboratory
Janelia Farm Research Campus
19700 Helix Drive
Ashburn VA 20147 USA
http://eddylab.org
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="ngraph.1.html"><span aria-hidden="true">&larr;</span> ngraph.1: Creating scientific 2-dimensional graphs</a></li>
   <li class="next"><a href="nhmmscan.1.html">nhmmscan.1: Search nucleotide sequence(s) against a nucleotide profile database <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
