<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>phytime: Bayesian estimation of divergence times from large sequence alignments</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Bayesian estimation of divergence times from large sequence alignments">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="phytime (1) manual">
  <meta name="twitter:description" content="Bayesian estimation of divergence times from large sequence alignments">
  <meta name="twitter:image" content="https://www.carta.tech/images/phyml-phytime-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/phytime.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="phytime (1) manual" />
  <meta property="og:description" content="Bayesian estimation of divergence times from large sequence alignments" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/phyml-phytime-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">phytime<small> (1)</small></h1>
        <p class="lead">Bayesian estimation of divergence times from large sequence alignments</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/phytime.1.html">
      <span itemprop="name">phytime: Bayesian estimation of divergence times from large sequence alignments</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/phyml/">
      <span itemprop="name">phyml</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/phytime.1.html">
      <span itemprop="name">phytime: Bayesian estimation of divergence times from large sequence alignments</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>Bayesian estimation of divergence times from molecular sequences relies on sophisticated Markov chain Monte Carlo techniques, and Metropolis-Hastings (MH) samplers have been successfully used in that context. This approach involves heavy computational burdens that can hinder the analysis of large phylogenomic data sets. Reliable estimation of divergence times can also be extremely time consuming, if not impossible, for sequence alignments that convey weak or conflicting phylogenetic signals, emphasizing the need for more efficient sampling methods. This article describes a new approach that estimates the posterior density of substitution rates and node times. The prior distribution of rates accounts for their potential autocorrelation along lineages, whereas priors on node ages are modeled with uniform densities. Also, the likelihood function is approximated by a multivariate normal density. The combination of these components leads to convenient mathematical simplifications, allowing the posterior distribution of rates and times to be estimated using a Gibbs sampling algorithm. The analysis of four real-world data sets shows that this sampler outperforms the standard MH approach and demonstrates the suitability of this new method for analyzing large and/or difficult data sets.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>phytime</strong> [command args]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p>All the options below are optional except '-i','-u' and '--calibration'.</p><p>Command options:</p><ul>
<li><p>-i (or <strong>--input</strong>) seq_file_name</p></li><li><p>seq_file_name is the name of the nucleotide or amino-acid sequence file in PHYLIP format.</p></li>
</ul><p><strong>-d</strong> (or <strong>--datatype</strong>) data_type</p><ul>
<li><p>data_type is 'nt' for nucleotide (default), 'aa' for amino-acid sequences, or 'generic', (use NEXUS file format and the 'symbols' parameter here).</p></li>
</ul><p><strong>-q</strong> (or <strong>--sequential</strong>)</p>
<pre>
                Changes interleaved format (default) to sequential format.
</pre>
<p><strong>-m</strong> (or <strong>--model</strong>) model</p><ul>
<li><p>model : substitution model name. - Nucleotide-based models : HKY85 (default) | JC69 | K80 | F81 | F84 | TN93 | GTR | custom (*) (*) : for the custom option, a string of six digits identifies the model. For instance, 000000</p></li><li><p>corresponds to F81 (or JC69 provided the distribution of nucleotide frequencies is uniform). 012345 corresponds to GTR. This option can be used for encoding any model that is a nested within GTR.</p></li><li><p>- Amino-acid based models : LG (default) | WAG | JTT | MtREV | Dayhoff | DCMut | RtREV | CpREV | VT</p></li>
</ul>
<dl class='dl-vertical'>
  <dt>
    <p>Blosum62 | MtMam | MtArt | HIVw |</p>
  </dt>
  <dd>
    <p>HIVb | custom</p>
  </dd>

</dl>
<p><strong>--aa_rate_file</strong> filename</p><ul>
<li><p>filename is the name of the file that provides the amino acid substitution rate matrix in PAML format. It is compulsory to use this option when analysing amino acid sequences with the `custom' model.</p></li>
</ul><p><strong>--calibration</strong> filename</p><ul>
<li><p>filename is the name of the calibration file that provides a priori defined boundaries for node ages. Please read the manual for more information about the format of this file.</p></li>
</ul><p><strong>-t</strong> (or <strong>--ts</strong>/tv) ts/tv_ratio</p><ul>
<li><p>ts/tv_ratio : transition/transversion ratio. DNA sequences only. Can be a fixed positive value (ex:4.0) or e to get the maximum likelihood estimate.</p></li>
</ul><p><strong>-v</strong> (or <strong>--pinv</strong>) prop_invar</p><ul>
<li><p>prop_invar : proportion of invariable sites. Can be a fixed value in the [0,1] range or e to get the maximum likelihood estimate.</p></li>
</ul><p><strong>-c</strong> (or <strong>--nclasses</strong>) nb_subst_cat</p><ul>
<li><p>nb_subst_cat : number of relative substitution rate categories. Default : nb_subst_cat=4. Must be a positive integer.</p></li>
</ul><p><strong>-a</strong> (or <strong>--alpha</strong>) gamma</p><ul>
<li><p>gamma : distribution of the gamma distribution shape parameter. Can be a fixed positive value or e to get the maximum likelihood estimate.</p></li>
</ul><p><strong>-u</strong> (or <strong>--inputtree</strong>) user_tree_file</p><ul>
<li><p>user_tree_file : starting tree filename. The tree must be in Newick format.</p></li>
</ul><p><strong>--r_seed</strong> num</p><ul>
<li><p>num is the seed used to initiate the random number generator. Must be an integer.</p></li>
</ul><p><strong>--run_id</strong> ID_string</p><ul>
<li><p>Append the string ID_string at the end of each PhyML output file. This option may be useful when running simulations involving PhyML.</p></li>
</ul><p><strong>--quiet</strong></p><ul>
<li><p>No interactive question (for running in batch mode) and quiet output.</p></li>
</ul><p><strong>--no_memory_check</strong></p><ul>
<li><p>No interactive question for memory usage (for running in batch mode). Normal output otherwise.</p></li>
</ul><p><strong>--chain_len</strong> num</p><ul>
<li><p>num is the number of generations or runs of the Markov Chain Monte Carlo. Set to 1E+6 by default. Must be an integer.</p></li>
</ul><p><strong>--sample_freq</strong> num</p><ul>
<li><p>The chain is sampled every num generations. Set to 1E+3 by default. Must be an integer.</p></li>
</ul><p><strong>--no_data</strong></p><ul>
<li><p>Use this option to sample from the priors only (rather from the posterior joint density of the model parameters).</p></li>
</ul><p><strong>--fastlk</strong></p><ul>
<li><p>Use the multivariate normal approximation to the likelihood and speed up calculations</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO phytime&hellip;</h2>
        <div class="sectioncontent">

<pre>
        'Bayesian estimation of divergence times from large sequence alignments.'
        Stephane Guindon,
        Molecular Biology and Evolution 27(8):1768-81, 2010.
</pre>

<pre>
        Please cite this paper if you use this software in your publications.
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="phylip.1.html"><span aria-hidden="true">&larr;</span> phylip.1: Runs various phylogenies programs</a></li>
   <li class="next"><a href="phyutility.1.html">phyutility.1: Tools for manipulating sequences and phylogenies <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
