<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>proda: Multiple alignment of protein sequences with repeats and rearrangements</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Multiple alignment of protein sequences with repeats and rearrangements">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="proda (1) manual">
  <meta name="twitter:description" content="Multiple alignment of protein sequences with repeats and rearrangements">
  <meta name="twitter:image" content="https://www.carta.tech/images/proda-proda-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/proda.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="proda (1) manual" />
  <meta property="og:description" content="Multiple alignment of protein sequences with repeats and rearrangements" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/proda-proda-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">proda<small> (1)</small></h1>
        <p class="lead">Multiple alignment of protein sequences with repeats and rearrangements</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/proda.1.html">
      <span itemprop="name">proda: Multiple alignment of protein sequences with repeats and rearrangements</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/proda/">
      <span itemprop="name">proda</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/proda.1.html">
      <span itemprop="name">proda: Multiple alignment of protein sequences with repeats and rearrangements</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>proda</strong> [<strong></strong><em>option</em>] [<em>mfafile</em>] [&gt;&nbsp;<em>output</em>]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This manual page documents briefly the <strong>proda</strong> command.</p><p><strong>proda</strong> (Protein Domain Aligner) is public domain software for generating multiple alignments of protein sequences with repeats and rearrangements, e.g. proteins with multiple domains.</p><p>Given a set of protein sequences as input, ProDA first finds local pairwise alignments between all pairs of sequences, then forms blocks of alignable sequence fragments, and finally generates multiple alignments of the blocks. ProDA relies on many techniques used in <strong>probcons</strong> (&lt;http://probcons.stanford.edu&gt;), a recent multiple aligner that shows high accuracy in a number of popular benchmarks.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">INPUT FORMAT</h2>
        <div class="sectioncontent">
<p>Proda accepts input files in the MFA format. The MFA format is specified below:</p><p>\h'-04'&bull;\h'+03'the MFA format consists of multiple sequences;</p><p>\h'-04'&bull;\h'+03'each sequence in the MFA format begins with a single-line description, followed by lines of sequence data;</p><p>\h'-04'&bull;\h'+03'the description line is distinguished from the sequence data by a greater-than ("&gt;") symbol in the first column.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OUTPUT FORMAT</h2>
        <div class="sectioncontent">
<p>For a set of input sequences, Proda usually outputs several blocks in turn, each consists of alignable sequence fragments. Each block is followed by its multiple alignment.</p><p>A block is specified by listing its sequence fragments. Each fragment is output as sequence_name(start-end), where sequence_name is the name of the original sequence and start and end are positions at which the fragment begins and ends respectively.</p><p>Proda produces block alignments in the ClustalW (ALN) format described below:</p><p>\h'-04'&bull;\h'+03'the ClustalW format consists of a single header line followed by sequence data in blocks of 50 alignment positions;</p><p>\h'-04'&bull;\h'+03'each block consists of:</p><p>\h'-04'&bull;\h'+03'one line of data for each of the sequences in the alignment - in particular, the name of the sequence;</p><p>\h'-04'&bull;\h'+03'50 characters of the alignment;</p><p>\h'-04'&bull;\h'+03'one annotation line indicating fully conserved (*), strongly-conserved (:), or weakly-conserved columns (.);</p><p>\h'-04'&bull;\h'+03'the description line is distinguished from the sequence data by a greater-than ("&gt;") symbol in the first column.</p><h3>FASTA format for output</h3>
<p>If the -fasta option is specified, then, in addition to regular output, ProDA produces a file containing block alignments in the FASTA format. The output file has the same name as the first input file and extension ".fasta". Two consecutive block alignments are separated by a line containing character \'#\'.</p><p>The FASTA format is described below:</p><p>\h'-04'&bull;\h'+03'the FASTA format consists of all the sequences given in the input files;</p><p>\h'-04'&bull;\h'+03'each sequence in the FASTA format begins with a single-line description, followed by lines of sequence data;</p><p>\h'-04'&bull;\h'+03'the description line is distinguished from the sequence data by a greater-than ("&gt;") symbol in the first column;</p><p>\h'-04'&bull;\h'+03'aligned residues are in upper case, unaligned residues are in lower case.</p><p>Since a final alignment contains each sequence only once, this option should be used only if input sequences do not contain repeats.</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p><strong>-L</strong> [min_length]</p><p>Set minimal alignment length equal to [min_length].</p><p>ProDA finds alignments of length greater than or equal to a threshold L. By default, L = 30. This option sets the threshold to [min_length].</p><p><strong>-posterior</strong></p><p>Use posterior decoding when computing local pairwise alignments.</p><p>ProDA computes local pairwise alignments between two sequences using a pair-HMM and either Viterbi decoding or posterior decoding. The default option is using Viterbi decoding which is faster than posterior decoding but may be less accurate. Turning on this option instructs the aligner to use posterior decoding instead. In the example above, the output was generated with -posterior option turned on.</p><p><strong>-silent</strong></p><p>Do not report progress while aligning.</p><p>Turning on this option instructs the aligner not to report the progress while aligning. By default, ProDA reports the progress on all pairwise alignments, block generation, and on block alignment. Since some stages of the algorithm, especially pairwise alignment, may take long time, reporting progress makes the program look alive while running.</p><p><strong>-tran</strong></p><p>Use transitivity when forming blocks of alignable sequence fragments.</p><p>Two sequence fragments are directly alignable if they are parts of a local pairwise alignment. By default, two fragments are considered alignable if and only if they are directly alignable. Turning on this option instructs the aligner to consider two fragments alignable when they are directly alignable or when both of them are directly alignable to a third fragment.</p><p><strong>-fasta</strong></p><p>Use FASTA output format in addition to the ClustalW format.</p><p>When this option is turned on, the aligner generates output in the FASTA format and stores in a file with the same name as the first input file and extension ".fasta", in addition to the normal output to stdout. This option should be used only if input sequences do not contain repeats.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO proda&hellip;</h2>
        <div class="sectioncontent">
<p><strong>probcons</strong> (1)</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>This manual page was written by David Paleino &lt;d.paleino@gmail.com&gt; for the Debian(TM) system (but may be used by others). This man page is released under the same conditions as the software, that is Public Domain.</p><p>This software has been released in Public Domain by Phuong T.M., Do C.B., Edgar R.C. and Batzoglou S. in "Multiple alignment of protein sequences with repeats and rearrangements", Nucleic Acids Research 2006 - 34(20), 5932-5942</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">
<p>Copyright &copy; 2007 David Paleino</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="procmeter3_modules.1.html"><span aria-hidden="true">&larr;</span> procmeter3_modules.1: The modules that are used by the procmeter3, procmeter3-xaw, gprocmeter3, procmeter3-gtk1, procmeter3-gtk2, procmeter3-gtk3, procmeter3-log and procmeter3-lcd programs.</a></li>
   <li class="next"><a href="prodigal.1.html">prodigal.1: Microbial (bacterial and archaeal) gene finding program <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
