<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>prof: Secondary structure and solvent accessibility predictor</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Secondary structure and solvent accessibility predictor">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="prof (1) manual">
  <meta name="twitter:description" content="Secondary structure and solvent accessibility predictor">
  <meta name="twitter:image" content="https://www.carta.tech/images/profphd-prof-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/prof.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="prof (1) manual" />
  <meta property="og:description" content="Secondary structure and solvent accessibility predictor" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/profphd-prof-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">prof<small> (1)</small></h1>
        <p class="lead">Secondary structure and solvent accessibility predictor</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/prof.1.html">
      <span itemprop="name">prof: Secondary structure and solvent accessibility predictor</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/profphd/">
      <span itemprop="name">profphd</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/prof.1.html">
      <span itemprop="name">prof: Secondary structure and solvent accessibility predictor</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>prof [<em>\s-1INPUTFILE\s0</em>+] [\s-1OPTIONS\s0]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>Secondary structure is predicted by a system of neural networks rating at an expected average accuracy &gt; 72% for the three states helix, strand and loop (Rost & Sander, \s-1PNAS\s0, 1993 , 90, 7558-7562; Rost & Sander, \s-1JMB\s0, 1993 , 232, 584-599; and Rost & Sander, Proteins, 1994 , 19, 55-72; evaluation of accuracy). Evaluated on the same data set, PROFsec is rated at ten percentage points higher three-state accuracy than methods using only single sequence information, and at more than six percentage points higher than, e.g., a method using alignment information based on statistics (Levin, Pascarella, Argos & Garnier, Prot. Engng., 6, 849-54, 1993). PHDsec predictions have three main features:</p>
<dl class='dl-vertical'>
  <dt>
    1. improved accuracy through evolutionary information from multiple sequence alignments
  </dt>
  <dd>
    
  </dd>
  <dt>
    2. improved beta-strand prediction through a balanced training procedure
  </dt>
  <dd>
    
  </dd>
  <dt>
    3. more accurate prediction of secondary structure segments by using a multi-level system
  </dt>
  <dd>
    
  </dd>

</dl>
<p>Solvent accessibility is predicted by a neural network method rating at a correlation coefficient (correlation between experimentally observed and predicted relative solvent accessibility) of 0.54 cross-validated on a set of 238 globular proteins (Rost & Sander, Proteins, 1994, 20, 216-226; evaluation of accuracy). The output of the neural network codes for 10 states of relative accessibility. Expressed in units of the difference between prediction by homology modelling (best method) and prediction at random (worst method), PROFacc is some 26 percentage points superior to a comparable neural network using three output states (buried, intermediate, exposed) and using no information from multiple alignments.</p><p>Transmembrane helices in integral membrane proteins are predicted by a system of neural networks. The shortcoming of the network system is that often too long helices are predicted. These are cut by an empirical filter. The final prediction (Rost et al., Protein Science, 1995, 4, 521-533; evaluation of accuracy) has an expected per-residue accuracy of about 95%. The number of false positives, i.e., transmembrane helices predicted in globular proteins, is about 2%. The neural network prediction of transmembrane helices (PHDhtm) is refined by a dynamic programming-like algorithm. This method resulted in correct predictions of all transmembrane helices for 89% of the 131 proteins used in a cross-validation test; more than 98% of the transmembrane helices were correctly predicted. The output of this method is used to predict topology, i.e., the orientation of the N-term with respect to the membrane. The expected accuracy of the topology prediction is &gt; 86%. Prediction accuracy is higher than average for eukaryotic proteins and lower than average for prokaryotes. PHDtopology is more accurate than all other methods tested on identical data sets.</p><p>If no output file option (such as <strong>--fileRdb</strong> or <strong>--fileOut</strong>) is given the \s-1RDB\s0 formatted output is written into <em>./INPUTFILENAME.prof</em> where 'prof' replaces the extension of the input file.  In lack of extension '.prof' is appended to the input file name.</p><h3>Output format</h3>
<p>The \s-1RDB\s0 format is self-annotating, see example outputs in <em>/usr/share/profphd/prof/exa</em>.</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REFERENCES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    Rost, B. and Sander, C. (1994a). Combining evolutionary information and neural networks to predict protein secondary structure. Proteins, 19(1), 55-72.
  </dt>
  <dd>
    
  </dd>
  <dt>
    Rost, B. and Sander, C. (1994b). Conservation and prediction of solvent accessibility in protein families. Proteins, 20(3), 216-26.
  </dt>
  <dd>
    
  </dd>
  <dt>
    Rost, B., Casadio, R., Fariselli, P., and Sander, C. (1995). Transmembrane helices predicted at 95% accuracy. Protein Sci, 4(3), 521-33.
  </dt>
  <dd>
    
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p>See each keyword for more help.  Most of these are likely to be broken.</p>
<dl class='dl-vertical'>
  <dt>
    a
  </dt>
  <dd>
    <p>alternative connectivity patterns (default=3)</p>
  </dd>
  <dt>
    3
  </dt>
  <dd>
    <p>predict sec + acc + htm</p>
  </dd>
  <dt>
    acc
  </dt>
  <dd>
    <p>predict solvent accessibility, only</p>
  </dd>
  <dt>
    ali
  </dt>
  <dd>
    <p>add alignment to 'human-readable' \s-1PROF\s0 output file(s)</p>
  </dd>
  <dt>
    arch
  </dt>
  <dd>
    <p>system architecture (e.g.: SGI64|SGI5|SGI32|SUNMP|ALPHA)</p>
  </dd>
  <dt>
    ascii
  </dt>
  <dd>
    <p>write 'human-readable' \s-1PROF\s0 output file(s)</p>
  </dd>
  <dt>
    best
  </dt>
  <dd>
    <p>\s-1PROF\s0 with best accuracy and longest run-time</p>
  </dd>
  <dt>
    both
  </dt>
  <dd>
    <p>predict secondary structure and solvent accessibility</p>
  </dd>
  <dt>
    data
  </dt>
  <dd>
    <p>data=&lt;all|brief|normal|detail&gt;  for \s-1HTML\s0 out: only those parts of predictions written</p>
  </dd>
  <dt>
    debug
  </dt>
  <dd>
    <p>keep most intermediate files, print debugging messages</p>
  </dd>
  <dt>
    dirWork
  </dt>
  <dd>
    <p>work directory, default: a temporary directory from File::Temp::tempdir. Must be fully qualified path. Known to work.</p>
  </dd>
  <dt>
    doEval
  </dt>
  <dd>
    <p>\s-1DO\s0 evaluation for list (only for known structures and lists)</p>
  </dd>
  <dt>
    doFilterHssp
  </dt>
  <dd>
    <p>filter the input \s-1HSSP\s0 file       (excluding some pairs)</p>
  </dd>
  <dt>
    doHtmfil
  </dt>
  <dd>
    <p>\s-1DO\s0 filter the membrane prediction                  (default)</p>
  </dd>
  <dt>
    doHtmisit
  </dt>
  <dd>
    <p>\s-1DO\s0 check strength of predicted membrane helix      (default)</p>
  </dd>
  <dt>
    doHtmref
  </dt>
  <dd>
    <p>\s-1DO\s0 refine the membrane prediction                  (default)</p>
  </dd>
  <dt>
    doHtmtop
  </dt>
  <dd>
    <p>\s-1DO\s0 membrane helix topology                         (default)</p>
  </dd>
  <dt>
    dssp
  </dt>
  <dd>
    <p>convert \s-1PROF\s0 into \s-1DSSP\s0 format</p>
  </dd>
  <dt>
    expand
  </dt>
  <dd>
    <p>expand insertions when converting output to \s-1MSF\s0 format</p>
  </dd>
  <dt>
    fast
  </dt>
  <dd>
    <p>\s-1PROF\s0 with lowest accuracy and highest speed</p>
  </dd>
  <dt>
    fileCasp
  </dt>
  <dd>
    <p>name of \s-1PROF\s0 output in \s-1CASP\s0 format              (file.caspProf)</p>
  </dd>
  <dt>
    fileDssp
  </dt>
  <dd>
    <p>name of \s-1PROF\s0 output in \s-1DSSP\s0 format              (file.dsspProf)</p>
  </dd>
  <dt>
    fileHtml
  </dt>
  <dd>
    <p>name of \s-1PROF\s0 output in \s-1HTML\s0 format              (file.htmlProf)</p>
  </dd>
  <dt>
    fileMsf
  </dt>
  <dd>
    <p>name of \s-1PROF\s0 output in \s-1MSF\s0 format               (file.msfProf)</p>
  </dd>
  <dt>
    fileNotHtm
  </dt>
  <dd>
    <p>name of file flagging that no membrane helix was found</p>
  </dd>
  <dt>
    fileOut
  </dt>
  <dd>
    <p>name of \s-1PROF\s0 output in \s-1RDB\s0 format               (file.rdbProf) Known to work.</p>
  </dd>
  <dt>
    fileProf
  </dt>
  <dd>
    <p>name of \s-1PROF\s0 output in human readable format    (file.prof) Broken.</p>
  </dd>
  <dt>
    fileRdb
  </dt>
  <dd>
    <p>name of \s-1PROF\s0 output in \s-1RDB\s0 format               (file.rdbProf) Known to work.</p>
  </dd>
  <dt>
    fileSaf
  </dt>
  <dd>
    <p>name of \s-1PROF\s0 output in \s-1SAF\s0 format               (file.safProf)</p>
  </dd>
  <dt>
    filter
  </dt>
  <dd>
    <p>filter the input \s-1HSSP\s0 file       (excluding some pairs)</p>
  </dd>
  <dt>
    good
  </dt>
  <dd>
    <p>\s-1PROF\s0 with good accuracy and moderate speed</p>
  </dd>
  <dt>
    graph
  </dt>
  <dd>
    <p>add \s-1ASCII\s0 graph to 'human-readable' \s-1PROF\s0 output file(s)</p>
  </dd>
  <dt>
    htm
  </dt>
  <dd>
    <p>use: 'htm=&lt;N|0.N&gt;' gives minimal transmembrane helix detected default is 'htm=8' (resp. htm=0.8)  smaller numbers more false positives and fewer false negatives!</p>
  </dd>
  <dt>
    html   argument
  </dt>
  <dd>
    <p>'hmtl' or 'html=&lt;all|body|head&gt;' write \s-1HTML\s0 format of prediction 'html' will result in that the \s-1PROF\s0 output is converted to \s-1HTML\s0 'html=body' restricts \s-1HTML\s0 file to the \s-1HTML_BODY\s0 tag part 'html=head' restricts \s-1HTML\s0 file to the \s-1HTML_HEADER\s0 tag part 'html=all'  gives both \s-1HEADER\s0 and \s-1BODY\s0</p>
  </dd>
  <dt>
    keepConv
  </dt>
  <dd>
    <p>keep the conversion of the input file to \s-1HSSP\s0 format</p>
  </dd>
  <dt>
    keepFilter argument
  </dt>
  <dd>
    <p>&lt;*|doKeepFilter=1&gt;     keep the filtered \s-1HSSP\s0 file</p>
  </dd>
  <dt>
    keepHssp  argument
  </dt>
  <dd>
    <p>&lt;*|doKeepHssp=1&gt;         keep the intermediate \s-1HSSP\s0 file</p>
  </dd>
  <dt>
    keepNetDb argument
  </dt>
  <dd>
    <p>&lt;*|doKeepNetDb=1&gt;       keep the intermediate DbNet file(s)</p>
  </dd>
  <dt>
    list argument
  </dt>
  <dd>
    <p>&lt;*|isList=1&gt;      input file is list of files</p>
  </dd>
  <dt>
    msf
  </dt>
  <dd>
    <p>convert \s-1PROF\s0 into \s-1MSF\s0 format</p>
  </dd>
  <dt>
    nice
  </dt>
  <dd>
    <p>give 'nice-D' to set the nice value (priority) of the job</p>
  </dd>
  <dt>
    noProfHead
  </dt>
  <dd>
    <p>do \s-1NOT\s0 copy file with tables into local directory</p>
  </dd>
  <dt>
    noSearch
  </dt>
  <dd>
    <p>short for doSearchFile=0, i.e. no searching of \s-1DB\s0 files</p>
  </dd>
  <dt>
    noascii
  </dt>
  <dd>
    <p>surpress writing \s-1ASCII\s0 (i.e. human readable) result files</p>
  </dd>
  <dt>
    nohtml
  </dt>
  <dd>
    <p>surpress writing \s-1HTML\s0 result files</p>
  </dd>
  <dt>
    nonice
  </dt>
  <dd>
    <p>job will not be niced, i.e. not run with lower priority</p>
  </dd>
  <dt>
    notEval
  </dt>
  <dd>
    <p>\s-1DO\s0 \s-1NOT\s0 check accuracy even when known structures</p>
  </dd>
  <dt>
    notHtmfil
  </dt>
  <dd>
    <p>do \s-1NOT\s0 filter the membrane prediction</p>
  </dd>
  <dt>
    notHtmisit
  </dt>
  <dd>
    <p>do \s-1NOT\s0 check whether or not membrane helix strong enough</p>
  </dd>
  <dt>
    notHtmref
  </dt>
  <dd>
    <p>do \s-1NOT\s0 refine the membrane prediction</p>
  </dd>
  <dt>
    notHtmtop
  </dt>
  <dd>
    <p>do \s-1NOT\s0 membrane helix topology</p>
  </dd>
  <dt>
    nresPerLineAli
  </dt>
  <dd>
    <p>Number of characters used for \s-1MSF\s0 file. Default: 50.</p>
  </dd>
  <dt>
    numresMin
  </dt>
  <dd>
    <p>Minimal number of residues to run network, otherwise prd=symbolPrdShort. Default: 9.</p>
  </dd>
  <dt>
    optJury
  </dt>
  <dd>
    <p>Adds \s-1PHD\s0 to jury. Default: `normal,usePHD'. Many other parameters change the default for this one as a side-effect, the list is not comprehensive: phd, nophd, /^para(3|Both|Sec|Acc|Htm|CapH|CapE|CapHE)/, /^para?/, jct</p>
  </dd>
  <dt>
    para3
  </dt>
  <dd>
    <p>Parameter file for sec+acc+htm. Default: `&lt;\s-1DIRPROF\s0&gt;/net/PROFboth_best.par'.</p>
  </dd>
  <dt>
    paraAcc
  </dt>
  <dd>
    <p>Parameter file for acc. Default: `&lt;\s-1DIRPROF\s0&gt;/net/PROFacc_best.par'.</p>
  </dd>
  <dt>
    paraBoth
  </dt>
  <dd>
    <p>Parameter file for sec+acc. Default: `&lt;\s-1DIRPROF\s0&gt;/net/PROFboth_best.par'.</p>
  </dd>
  <dt>
    paraSec
  </dt>
  <dd>
    <p>Parameter file for sec. Default: `&lt;\s-1DIRPROF\s0&gt;/net/PROFsec_best.par'.</p>
  </dd>
  <dt>
    riSubAcc
  </dt>
  <dd>
    <p>Minimal reliability index (\s-1RI\s0) for subset PROFacc. Default: 4.</p>
  </dd>
  <dt>
    riSubSec
  </dt>
  <dd>
    <p>Minimal reliability index (\s-1RI\s0) for subset PROFsec. Default: 5.</p>
  </dd>
  <dt>
    riSubSym
  </dt>
  <dd>
    <p>Symbol for residues predicted with \s-1RI\s0 &lt; riSubSec/Acc. Default: `.'.</p>
  </dd>
  <dt>
    s_k_i_p
  </dt>
  <dd>
    <p>problems, manual, hints, notation, txt, known, \s-1DONE\s0, Date, date, aa, Lhssp, numaa, code</p>
  </dd>
  <dt>
    saf
  </dt>
  <dd>
    <p>convert \s-1PROF\s0 into \s-1SAF\s0 format</p>
  </dd>
  <dt>
    scrAddHelp
  </dt>
  <dd>
    
  </dd>
  <dt>
    scrGoal
  </dt>
  <dd>
    <p>neural network switching</p>
  </dd>
  <dt>
    scrHelpTxt
  </dt>
  <dd>
    <p>Input file formats accepted:       hssp,dssp,msf,saf,fastamul,pirmul,fasta,pir,gcg,swiss</p>
  </dd>
  <dt>
    scrIn
  </dt>
  <dd>
    <p>list_of_files (or single file) parameter_file</p>
  </dd>
  <dt>
    scrName
  </dt>
  <dd>
    <p>prof</p>
  </dd>
  <dt>
    scrNarg
  </dt>
  <dd>
    <p>2</p>
  </dd>
  <dt>
    sec
  </dt>
  <dd>
    <p>predict secondary structure,   only</p>
  </dd>
  <dt>
    silent
  </dt>
  <dd>
    <p>no information written to screen - this is the default</p>
  </dd>
  <dt>
    skipMissing
  </dt>
  <dd>
    <p>do not abort if input file missing!</p>
  </dd>
  <dt>
    sourceFile
  </dt>
  <dd>
    <p>prof</p>
  </dd>
  <dt>
    test
  </dt>
  <dd>
    <p>is just a test (faster)</p>
  </dd>
  <dt>
    translate-jobid-in-param-values
  </dt>
  <dd>
    <p>String 'jobid' gets substituted with $par{jobid}</p>
  </dd>
  <dt>
    tst
  </dt>
  <dd>
    <p>quick run through program, low accuracy</p>
  </dd>
  <dt>
    user
  </dt>
  <dd>
    <p>user name</p>
  </dd>
  <dt>
    --version
  </dt>
  <dd>
    <p>Print version</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>B. Rost, Sander C, Fariselli P, Casadio R, Liu J, Yachdav G, Kajan L.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    Prediction from alignment in \s-1HSSP\s0 file for best results
  </dt>
  <dd>
    
<pre>
 prof /usr/share/profphd/prof/exa/1ppt.hssp fileRdb=/tmp/1ppt.hssp.prof
</pre>

  </dd>
  <dt>
    Prediction from a single sequence
  </dt>
  <dd>
    <p> prof /usr/share/profphd/prof/exa/1ppt.f fileRdb=/tmp/1ppt.f.rdbProf</p>
  </dd>
  <dt>
    phd.pl invocation
  </dt>
  <dd>
    <p> /usr/share/profphd/prof/embl/phd.pl /usr/share/profphd/prof/exa/1ppt.hssp htm fileOutPhd=/tmp/query.phdPred  fileOutRdb=/tmp/query.phdRdb  fileNotHtm=/tmp/query.phdNotHtm</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ENVIRONMENT</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    \s-1PROFPHDDIR\s0
  </dt>
  <dd>
    <p>Override package prof package dir <em>/usr/share/profphd</em>.</p>
  </dd>
  <dt>
    \s-1RGUTILSDIR\s0
  </dt>
  <dd>
    <p>Override location of librg-utils-perl <em>/usr/share/librg-utils-perl</em>.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FILES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <em>*.rdbProf</em>
  </dt>
  <dd>
    <p>default output file extension</p>
  </dd>
  <dt>
    <em>/usr/share/profphd/prof</em>
  </dt>
  <dd>
    <p>default data directory</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>Please report bugs at &lt;https://rostlab.org/bugzilla3/enter_bug.cgi?product=profphd&gt;.</p>
<dl class='dl-vertical'>
  <dt>
    Prediction from \s-1HSSP\s0 file fails when residue lines with exclamation marks `!' are present:
  </dt>
  <dd>
    <p>Use 'optJury=normal' and 'both' like this:  prof /tmp/1a3q.hssp fileRdb=/tmp/1a3q.hssp.profRdb optJury=normal both</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO prof&hellip;</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    Main website
  </dt>
  <dd>
    <p>&lt;http://www.predictprotein.org/&gt;</p>
  </dd>
  <dt>
    Documentation
  </dt>
  <dd>
    <p>&lt;http://www.predictprotein.org/docs.php&gt;</p>
  </dd>
  <dt>
    Community website
  </dt>
  <dd>
    <p>&lt;http://groups.google.com/group/PredictProtein&gt;</p>
  </dd>
  <dt>
    \s-1FTP\s0
  </dt>
  <dd>
    <p>&lt;ftp://rostlab.org/pub/cubic/downloads/prof&gt;</p>
  </dd>
  <dt>
    Newsgroups
  </dt>
  <dd>
    <p>&lt;http://groups.google.com/group/PredictProtein&gt;</p>
  </dd>

</dl>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="prodigal.1.html"><span aria-hidden="true">&larr;</span> prodigal.1: Microbial (bacterial and archaeal) gene finding program</a></li>
   <li class="next"><a href="profanity.1.html">profanity.1: A simple console based xmpp chat client. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
