<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>rabema_build_gold_standard: Rabema gold standard builder  synopsis rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-sam perfect.sam rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-bam perfect.bam  description this program allows to build a rabema gold standard. the input is a reference fasta file and a perfect sam/bam map (e.g. created using razers 3 in full-sensitivity mode). the input sam/bam file must be sorted by coordinate. the program will create a fasta index file ref.fa.fai for fast random access to the reference. -h, --help displays this help message. --version display version information -v, --verbose enable verbose output. -vv, --very-verbose enable even more verbose output. input / output: -o, --out-gsi gsi path to write the resulting gsi file to. valid filetypes are: gsi and gsi.gz. -r, --reference fasta path to load reference fasta from. valid filetypes are: fa and fasta. -s, --in-sam sam path to load the "perfect" sam file from. valid filetype is: sam. -b, --in-bam bam path to load the "perfect" bam file from. valid filetype is: bam. gold standard parameters: --oracle-mode enable oracle mode. this is used for simulated data when the input sam/bam file gives exactly one position that is considered as the true sample position. --match-n when set, n matches all characters without penalty. --distance-metric metric set distance metric. valid values: hamming, edit. default: edit. one of hamming and edit. default: edit. -e, --max-error rate maximal error rate to build gold standard for in percent. this parameter is an integer and relative to the read length. in case of oracle mode, the error rate for the read at the sampling position is used and rate is used as a cutoff threshold. default: 0.  return values a return value of 0 indicates success, any other value indicates an error.  examples rabema_build_gold_standard -e 4 -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with all mapping locations and a fasta reference ref.fa to gsi file out.gsi with a maximal error rate of 4. rabema_build_gold_standard --distance-metric edit -e 4 -o out.gsi -b in.bam -r ref.fa same as above, but using hamming instead of edit distance and bam as the input. rabema_build_gold_standard --oracle-mode -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with the original sample position, e.g. as exported by read simulator mason.  memory requirements from version 1.1, great care has been taken to keep the memory requirements as low as possible. there memory required is two times the size of the largest chromosome plus some constant memory for each match. for example, the memory usage for 100bp human genome reads at 5% error rate was 1.7gb. of this, roughly 400gb came from the chromosome and 1.3gb from the matches.  references m. holtgrewe, a.-k. emde, d. weese and k. reinert. a novel and well-defined benchmarking method for second generation read mapping, bmc bioinformatics 2011, 12:210. http://www.seqan.de/rabema rabema homepage http://www.seqan.de/mason mason homepage  version rabema_build_gold_standard version: 1.2.0 last update march 14, 2013</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Rabema gold standard builder  synopsis rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-sam perfect.sam rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-bam perfect.bam  description this program allows to build a rabema gold standard. the input is a reference fasta file and a perfect sam/bam map (e.g. created using razers 3 in full-sensitivity mode). the input sam/bam file must be sorted by coordinate. the program will create a fasta index file ref.fa.fai for fast random access to the reference. -h, --help displays this help message. --version display version information -v, --verbose enable verbose output. -vv, --very-verbose enable even more verbose output. input / output: -o, --out-gsi gsi path to write the resulting gsi file to. valid filetypes are: gsi and gsi.gz. -r, --reference fasta path to load reference fasta from. valid filetypes are: fa and fasta. -s, --in-sam sam path to load the "perfect" sam file from. valid filetype is: sam. -b, --in-bam bam path to load the "perfect" bam file from. valid filetype is: bam. gold standard parameters: --oracle-mode enable oracle mode. this is used for simulated data when the input sam/bam file gives exactly one position that is considered as the true sample position. --match-n when set, n matches all characters without penalty. --distance-metric metric set distance metric. valid values: hamming, edit. default: edit. one of hamming and edit. default: edit. -e, --max-error rate maximal error rate to build gold standard for in percent. this parameter is an integer and relative to the read length. in case of oracle mode, the error rate for the read at the sampling position is used and rate is used as a cutoff threshold. default: 0.  return values a return value of 0 indicates success, any other value indicates an error.  examples rabema_build_gold_standard -e 4 -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with all mapping locations and a fasta reference ref.fa to gsi file out.gsi with a maximal error rate of 4. rabema_build_gold_standard --distance-metric edit -e 4 -o out.gsi -b in.bam -r ref.fa same as above, but using hamming instead of edit distance and bam as the input. rabema_build_gold_standard --oracle-mode -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with the original sample position, e.g. as exported by read simulator mason.  memory requirements from version 1.1, great care has been taken to keep the memory requirements as low as possible. there memory required is two times the size of the largest chromosome plus some constant memory for each match. for example, the memory usage for 100bp human genome reads at 5% error rate was 1.7gb. of this, roughly 400gb came from the chromosome and 1.3gb from the matches.  references m. holtgrewe, a.-k. emde, d. weese and k. reinert. a novel and well-defined benchmarking method for second generation read mapping, bmc bioinformatics 2011, 12:210. http://www.seqan.de/rabema rabema homepage http://www.seqan.de/mason mason homepage  version rabema_build_gold_standard version: 1.2.0 last update march 14, 2013">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="rabema_build_gold_standard (1) manual">
  <meta name="twitter:description" content="Rabema gold standard builder  synopsis rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-sam perfect.sam rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-bam perfect.bam  description this program allows to build a rabema gold standard. the input is a reference fasta file and a perfect sam/bam map (e.g. created using razers 3 in full-sensitivity mode). the input sam/bam file must be sorted by coordinate. the program will create a fasta index file ref.fa.fai for fast random access to the reference. -h, --help displays this help message. --version display version information -v, --verbose enable verbose output. -vv, --very-verbose enable even more verbose output. input / output: -o, --out-gsi gsi path to write the resulting gsi file to. valid filetypes are: gsi and gsi.gz. -r, --reference fasta path to load reference fasta from. valid filetypes are: fa and fasta. -s, --in-sam sam path to load the "perfect" sam file from. valid filetype is: sam. -b, --in-bam bam path to load the "perfect" bam file from. valid filetype is: bam. gold standard parameters: --oracle-mode enable oracle mode. this is used for simulated data when the input sam/bam file gives exactly one position that is considered as the true sample position. --match-n when set, n matches all characters without penalty. --distance-metric metric set distance metric. valid values: hamming, edit. default: edit. one of hamming and edit. default: edit. -e, --max-error rate maximal error rate to build gold standard for in percent. this parameter is an integer and relative to the read length. in case of oracle mode, the error rate for the read at the sampling position is used and rate is used as a cutoff threshold. default: 0.  return values a return value of 0 indicates success, any other value indicates an error.  examples rabema_build_gold_standard -e 4 -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with all mapping locations and a fasta reference ref.fa to gsi file out.gsi with a maximal error rate of 4. rabema_build_gold_standard --distance-metric edit -e 4 -o out.gsi -b in.bam -r ref.fa same as above, but using hamming instead of edit distance and bam as the input. rabema_build_gold_standard --oracle-mode -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with the original sample position, e.g. as exported by read simulator mason.  memory requirements from version 1.1, great care has been taken to keep the memory requirements as low as possible. there memory required is two times the size of the largest chromosome plus some constant memory for each match. for example, the memory usage for 100bp human genome reads at 5% error rate was 1.7gb. of this, roughly 400gb came from the chromosome and 1.3gb from the matches.  references m. holtgrewe, a.-k. emde, d. weese and k. reinert. a novel and well-defined benchmarking method for second generation read mapping, bmc bioinformatics 2011, 12:210. http://www.seqan.de/rabema rabema homepage http://www.seqan.de/mason mason homepage  version rabema_build_gold_standard version: 1.2.0 last update march 14, 2013">
  <meta name="twitter:image" content="https://www.carta.tech/images/seqan-apps-rabema_build_gold_standard-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/rabema_build_gold_standard.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="rabema_build_gold_standard (1) manual" />
  <meta property="og:description" content="Rabema gold standard builder  synopsis rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-sam perfect.sam rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-bam perfect.bam  description this program allows to build a rabema gold standard. the input is a reference fasta file and a perfect sam/bam map (e.g. created using razers 3 in full-sensitivity mode). the input sam/bam file must be sorted by coordinate. the program will create a fasta index file ref.fa.fai for fast random access to the reference. -h, --help displays this help message. --version display version information -v, --verbose enable verbose output. -vv, --very-verbose enable even more verbose output. input / output: -o, --out-gsi gsi path to write the resulting gsi file to. valid filetypes are: gsi and gsi.gz. -r, --reference fasta path to load reference fasta from. valid filetypes are: fa and fasta. -s, --in-sam sam path to load the "perfect" sam file from. valid filetype is: sam. -b, --in-bam bam path to load the "perfect" bam file from. valid filetype is: bam. gold standard parameters: --oracle-mode enable oracle mode. this is used for simulated data when the input sam/bam file gives exactly one position that is considered as the true sample position. --match-n when set, n matches all characters without penalty. --distance-metric metric set distance metric. valid values: hamming, edit. default: edit. one of hamming and edit. default: edit. -e, --max-error rate maximal error rate to build gold standard for in percent. this parameter is an integer and relative to the read length. in case of oracle mode, the error rate for the read at the sampling position is used and rate is used as a cutoff threshold. default: 0.  return values a return value of 0 indicates success, any other value indicates an error.  examples rabema_build_gold_standard -e 4 -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with all mapping locations and a fasta reference ref.fa to gsi file out.gsi with a maximal error rate of 4. rabema_build_gold_standard --distance-metric edit -e 4 -o out.gsi -b in.bam -r ref.fa same as above, but using hamming instead of edit distance and bam as the input. rabema_build_gold_standard --oracle-mode -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with the original sample position, e.g. as exported by read simulator mason.  memory requirements from version 1.1, great care has been taken to keep the memory requirements as low as possible. there memory required is two times the size of the largest chromosome plus some constant memory for each match. for example, the memory usage for 100bp human genome reads at 5% error rate was 1.7gb. of this, roughly 400gb came from the chromosome and 1.3gb from the matches.  references m. holtgrewe, a.-k. emde, d. weese and k. reinert. a novel and well-defined benchmarking method for second generation read mapping, bmc bioinformatics 2011, 12:210. http://www.seqan.de/rabema rabema homepage http://www.seqan.de/mason mason homepage  version rabema_build_gold_standard version: 1.2.0 last update march 14, 2013" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/seqan-apps-rabema_build_gold_standard-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">rabema_build_gold_standard<small> (1)</small></h1>
        <p class="lead">Rabema gold standard builder  synopsis rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-sam perfect.sam rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-bam perfect.bam  description this program allows to build a rabema gold standard. the input is a reference fasta file and a perfect sam/bam map (e.g. created using razers 3 in full-sensitivity mode). the input sam/bam file must be sorted by coordinate. the program will create a fasta index file ref.fa.fai for fast random access to the reference. -h, --help displays this help message. --version display version information -v, --verbose enable verbose output. -vv, --very-verbose enable even more verbose output. input / output: -o, --out-gsi gsi path to write the resulting gsi file to. valid filetypes are: gsi and gsi.gz. -r, --reference fasta path to load reference fasta from. valid filetypes are: fa and fasta. -s, --in-sam sam path to load the "perfect" sam file from. valid filetype is: sam. -b, --in-bam bam path to load the "perfect" bam file from. valid filetype is: bam. gold standard parameters: --oracle-mode enable oracle mode. this is used for simulated data when the input sam/bam file gives exactly one position that is considered as the true sample position. --match-n when set, n matches all characters without penalty. --distance-metric metric set distance metric. valid values: hamming, edit. default: edit. one of hamming and edit. default: edit. -e, --max-error rate maximal error rate to build gold standard for in percent. this parameter is an integer and relative to the read length. in case of oracle mode, the error rate for the read at the sampling position is used and rate is used as a cutoff threshold. default: 0.  return values a return value of 0 indicates success, any other value indicates an error.  examples rabema_build_gold_standard -e 4 -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with all mapping locations and a fasta reference ref.fa to gsi file out.gsi with a maximal error rate of 4. rabema_build_gold_standard --distance-metric edit -e 4 -o out.gsi -b in.bam -r ref.fa same as above, but using hamming instead of edit distance and bam as the input. rabema_build_gold_standard --oracle-mode -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with the original sample position, e.g. as exported by read simulator mason.  memory requirements from version 1.1, great care has been taken to keep the memory requirements as low as possible. there memory required is two times the size of the largest chromosome plus some constant memory for each match. for example, the memory usage for 100bp human genome reads at 5% error rate was 1.7gb. of this, roughly 400gb came from the chromosome and 1.3gb from the matches.  references m. holtgrewe, a.-k. emde, d. weese and k. reinert. a novel and well-defined benchmarking method for second generation read mapping, bmc bioinformatics 2011, 12:210. http://www.seqan.de/rabema rabema homepage http://www.seqan.de/mason mason homepage  version rabema_build_gold_standard version: 1.2.0 last update march 14, 2013</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/rabema_build_gold_standard.1.html">
      <span itemprop="name">rabema_build_gold_standard: Rabema gold standard builder  synopsis rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-sam perfect.sam rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-bam perfect.bam  description this program allows to build a rabema gold standard. the input is a reference fasta file and a perfect sam/bam map (e.g. created using razers 3 in full-sensitivity mode). the input sam/bam file must be sorted by coordinate. the program will create a fasta index file ref.fa.fai for fast random access to the reference. -h, --help displays this help message. --version display version information -v, --verbose enable verbose output. -vv, --very-verbose enable even more verbose output. input / output: -o, --out-gsi gsi path to write the resulting gsi file to. valid filetypes are: gsi and gsi.gz. -r, --reference fasta path to load reference fasta from. valid filetypes are: fa and fasta. -s, --in-sam sam path to load the "perfect" sam file from. valid filetype is: sam. -b, --in-bam bam path to load the "perfect" bam file from. valid filetype is: bam. gold standard parameters: --oracle-mode enable oracle mode. this is used for simulated data when the input sam/bam file gives exactly one position that is considered as the true sample position. --match-n when set, n matches all characters without penalty. --distance-metric metric set distance metric. valid values: hamming, edit. default: edit. one of hamming and edit. default: edit. -e, --max-error rate maximal error rate to build gold standard for in percent. this parameter is an integer and relative to the read length. in case of oracle mode, the error rate for the read at the sampling position is used and rate is used as a cutoff threshold. default: 0.  return values a return value of 0 indicates success, any other value indicates an error.  examples rabema_build_gold_standard -e 4 -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with all mapping locations and a fasta reference ref.fa to gsi file out.gsi with a maximal error rate of 4. rabema_build_gold_standard --distance-metric edit -e 4 -o out.gsi -b in.bam -r ref.fa same as above, but using hamming instead of edit distance and bam as the input. rabema_build_gold_standard --oracle-mode -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with the original sample position, e.g. as exported by read simulator mason.  memory requirements from version 1.1, great care has been taken to keep the memory requirements as low as possible. there memory required is two times the size of the largest chromosome plus some constant memory for each match. for example, the memory usage for 100bp human genome reads at 5% error rate was 1.7gb. of this, roughly 400gb came from the chromosome and 1.3gb from the matches.  references m. holtgrewe, a.-k. emde, d. weese and k. reinert. a novel and well-defined benchmarking method for second generation read mapping, bmc bioinformatics 2011, 12:210. http://www.seqan.de/rabema rabema homepage http://www.seqan.de/mason mason homepage  version rabema_build_gold_standard version: 1.2.0 last update march 14, 2013</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/seqan-apps/">
      <span itemprop="name">seqan-apps</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/rabema_build_gold_standard.1.html">
      <span itemprop="name">rabema_build_gold_standard: Rabema gold standard builder  synopsis rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-sam perfect.sam rabema_build_gold_standard [options] --out-gsi out.gsi --reference ref.fa --in-bam perfect.bam  description this program allows to build a rabema gold standard. the input is a reference fasta file and a perfect sam/bam map (e.g. created using razers 3 in full-sensitivity mode). the input sam/bam file must be sorted by coordinate. the program will create a fasta index file ref.fa.fai for fast random access to the reference. -h, --help displays this help message. --version display version information -v, --verbose enable verbose output. -vv, --very-verbose enable even more verbose output. input / output: -o, --out-gsi gsi path to write the resulting gsi file to. valid filetypes are: gsi and gsi.gz. -r, --reference fasta path to load reference fasta from. valid filetypes are: fa and fasta. -s, --in-sam sam path to load the "perfect" sam file from. valid filetype is: sam. -b, --in-bam bam path to load the "perfect" bam file from. valid filetype is: bam. gold standard parameters: --oracle-mode enable oracle mode. this is used for simulated data when the input sam/bam file gives exactly one position that is considered as the true sample position. --match-n when set, n matches all characters without penalty. --distance-metric metric set distance metric. valid values: hamming, edit. default: edit. one of hamming and edit. default: edit. -e, --max-error rate maximal error rate to build gold standard for in percent. this parameter is an integer and relative to the read length. in case of oracle mode, the error rate for the read at the sampling position is used and rate is used as a cutoff threshold. default: 0.  return values a return value of 0 indicates success, any other value indicates an error.  examples rabema_build_gold_standard -e 4 -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with all mapping locations and a fasta reference ref.fa to gsi file out.gsi with a maximal error rate of 4. rabema_build_gold_standard --distance-metric edit -e 4 -o out.gsi -b in.bam -r ref.fa same as above, but using hamming instead of edit distance and bam as the input. rabema_build_gold_standard --oracle-mode -o out.gsi -s in.sam -r ref.fa build gold standard from a sam file in.sam with the original sample position, e.g. as exported by read simulator mason.  memory requirements from version 1.1, great care has been taken to keep the memory requirements as low as possible. there memory required is two times the size of the largest chromosome plus some constant memory for each match. for example, the memory usage for 100bp human genome reads at 5% error rate was 1.7gb. of this, roughly 400gb came from the chromosome and 1.3gb from the matches.  references m. holtgrewe, a.-k. emde, d. weese and k. reinert. a novel and well-defined benchmarking method for second generation read mapping, bmc bioinformatics 2011, 12:210. http://www.seqan.de/rabema rabema homepage http://www.seqan.de/mason mason homepage  version rabema_build_gold_standard version: 1.2.0 last update march 14, 2013</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    <nav>
  <ul class="pager">
   <li class="previous"><a href="rabbitvcs.1.html"><span aria-hidden="true">&larr;</span> rabbitvcs.1: Integrated version control for nautilus</a></li>
   <li class="next"><a href="rabema_evaluate.1.html">rabema_evaluate.1: Rabema evaluation  synopsis rabema_evaluate [options] --reference ref.fa --in-gsi in.gsi --in-sam mapping.sam rabema_evaluate [options] --reference ref.fa --in-gsi in.gsi --in-bam mapping.bam  description compare the sam/bam output mapping.sam/mapping.bam of any read mapper against the rabema gold standard previously built with rabema_build_gold_standard. the input is a reference fasta file, a gold standard interval (gsi) file and the sam/bam input to evaluate. the input sam/bam file must be sorted by queryname. the program will create a fasta index file ref.fa.fai for fast random access to the reference. -h, --help displays this help message. --version display version information -v, --verbose enable verbose output. -vv, --very-verbose enable even more verbose output. input / output: -r, --reference fasta path to load reference fasta from. valid filetypes are: fa and fasta. -g, --in-gsi gsi path to load gold standard intervals from. if compressed using gzip, the file will be decompressed on the fly. valid filetypes are: gsi and gsi.gz. -s, --in-sam sam path to load the read mapper sam output from. valid filetype is: sam. -b, --in-bam bam path to load the read mapper bam output from. valid filetype is: bam. --out-tsv tsv path to write the statistics to as tsv. valid filetype is: tsv. benchmark parameters: --oracle-mode enable oracle mode. this is used for simulated data when the input gsi file gives exactly one position that is considered as the true sample position. for simulated data. --only-unique-reads consider only reads that a single alignment in the mapping result file. usefull for precision computation. --match-n when set, n matches all characters without penalty. --distance-metric metric set distance metric. valid values: hamming, edit. default: edit. one of hamming and edit. default: edit. -e, --max-error rate maximal error rate to build gold standard for in percent. this parameter is an integer and relative to the read length. the error rate is ignored in oracle mode, here the distance of the read at the sample position is taken, individually for each read. default: 0 default: 0. -c, --benchmark-category cat set benchmark category. one of {all, all-best, any-best. default: all one of all, all-best, and any-best. default: all. --trust-nm when set, we trust the alignment and distance from sam/bam file and no realignment is performed. off by default. --ignore-paired-flags when set, we ignore all sam/bam flags related to pairing. this is necessary when analyzing sam from soap's soap2sam.pl script. --dont-panic do not stop program execution if an additional hit was found that indicates that the gold standard is incorrect. logging: --show-missed-intervals show details for each missed interval from the gsi. --show-invalid-hits show details for invalid hits (with too high error rate). --show-additional-hits show details for additional hits (low enough error rate but not in gold standard. --show-hits show details for hit intervals. --show-try-hit show details for each alignment in sam/bam input. the occurrence of "invalid" hits in the read mapper's output is not an error. if there are additional hits, however, this shows an error in the gold standard.  return values a return value of 0 indicates success, any other value indicates an error.  memory requirements from version 1.1, great care has been taken to keep the memory requirements as low as possible. the evaluation step needs to store the whole reference sequence in memory but little more memory. so, for the human genome, the memory requirements are below 4 gb, regardless of the size of the gsi or sam/bam file.  references m. holtgrewe, a.-k. emde, d. weese and k. reinert. a novel and well-defined benchmarking method for second generation read mapping, bmc bioinformatics 2011, 12:210. http://www.seqan.de/rabema rabema homepage http://www.seqan.de/mason mason homepage  version rabema_evaluate version: 1.2.0 last update march 14, 2013 <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
