<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>rasmol: Molecular graphics visualisation tool v2.7.5</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Molecular graphics visualisation tool v2.7.5">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="rasmol (1) manual">
  <meta name="twitter:description" content="Molecular graphics visualisation tool v2.7.5">
  <meta name="twitter:image" content="https://www.carta.tech/images/rasmol-rasmol-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/rasmol.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="rasmol (1) manual" />
  <meta property="og:description" content="Molecular graphics visualisation tool v2.7.5" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/rasmol-rasmol-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">rasmol<small> (1)</small></h1>
        <p class="lead">Molecular graphics visualisation tool v2.7.5</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/rasmol.1.html">
      <span itemprop="name">rasmol: Molecular graphics visualisation tool v2.7.5</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/rasmol/">
      <span itemprop="name">rasmol</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/rasmol.1.html">
      <span itemprop="name">rasmol: Molecular graphics visualisation tool v2.7.5</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>rasmol</strong> [\|<strong>-nodiplay</strong>\|] [\|[\|<strong>-</strong>.I<strong>format</strong>.RB<strong>\|]</strong> <em>filename \|]</em> [\|<strong>-script </strong>.I<strong>scriptfile</strong>\|]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FORMATS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-pdb</strong></p>
  </dt>
  <dd>
    <p>Protein Data Bank</p>
  </dd>
  <dt>
    <p><strong>-mdl</strong></p>
  </dt>
  <dd>
    <p>MDL's MOL File Format</p>
  </dd>
  <dt>
    <p><strong>-mol2</strong></p>
  </dt>
  <dd>
    <p>Tripos' Sybyl MOL2 Format</p>
  </dd>
  <dt>
    <p><strong>-xyz</strong></p>
  </dt>
  <dd>
    <p>MSC's XYZ (XMol) Format</p>
  </dd>
  <dt>
    <p><strong>-mopac</strong></p>
  </dt>
  <dd>
    <p>MOPAC Input or Output File Format</p>
  </dd>
  <dt>
    <p><strong>-alchemy</strong></p>
  </dt>
  <dd>
    <p>Alchemy File Format</p>
  </dd>
  <dt>
    <p><strong>-charmm</strong></p>
  </dt>
  <dd>
    <p>CHARMm File Format</p>
  </dd>
  <dt>
    <p><strong>-cif</strong></p>
  </dt>
  <dd>
    <p>IUCr CIF or CIF File Format</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">NOTICES</h2>
        <div class="sectioncontent">
<p>This software has been created from several sources.  Much of the code is from RasMol 2.6, as created by Roger Sayle. The torsion angle code, new POVRAY3 code and other features are derived from the RasMol2.6x1 revisions by Arne Mueller. The Ramachandran printer plot code was derived from fisipl created by Frances C. Bernstein.  See the Protein Data Bank program tape.</p><p>The code to display multiple molecules and to allow bond rotation is derived in large part from the UCB mods by Gary Grossman and Marco Molinaro, included with permission of Eileen Lewis of the ModularCHEM Consortium.</p><p>The CIF modifications make use of a library based in part on CBFlib by Paul J. Ellis and Herbert J. Bernstein. Parts of CBFlib is loosely based on the CIFPARSE software package from the NDB at Rutgers university. Please type the RasMol commands <strong>help copying,</strong> <strong>help general,</strong> <strong>help IUCR,</strong> <strong>help CBFlib,</strong>  and <strong>help CIFPARSE</strong> for applicable notices.  Please type <strong>help copyright</strong> for copyright notices.  If you use RasMol V2.6 or an earlier version, type the RasMol command <strong>help oldnotice.</strong></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYING</h2>
        <div class="sectioncontent">
<p>This version is based directly on RasMol version 2.7.4.2, on RasMol verion 2.7.4.2, on RasMol version 2.7.4, on RasMol version 2.7.3.1, on RasMol version 2.7.3, on RasMol version 2.7.2.1.1, Rasmol version 2.7.2, RasMol version 2.7.1.1 and RasTop version 1.3 and indirectly on the RasMol 2.5-ucb and 2.6-ucb versions and version 2.6_CIF.2, RasMol 2.6x1 and RasMol_2.6.4.</p><p>RasMol 2.7.5 may be distributed under the terms of the GNU General Public License (the GPL), see</p>
<pre>
          http://www.gnu.org/licenses/gpl.txt
</pre>
<p>or the file GPL or type the command <strong>help GPL</strong></p><p>or RasMol 2.7.5 may be distributed under the RASMOL license. See the file NOTICE or type the command <strong>help RASLIC</strong></p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>GPL</strong></p>
  </dt>
  <dd>
    
<pre>
                    GNU GENERAL PUBLIC LICENSE
                       Version 2, June 1991
</pre>
<p> Copyright (C) 1989, 1991 Free Software Foundation, Inc.</p>
<pre>
                       59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
</pre>
<p> Everyone is permitted to copy and distribute verbatim copies  of this license document, but changing it is not allowed.</p>
<pre>
                            Preamble
</pre>

<pre>
  The licenses for most software are designed to take away your
</pre>
<p>freedom to share and change it.  By contrast, the GNU General Public License is intended to guarantee your freedom to share and change free software--to make sure the software is free for all its users.  This General Public License applies to most of the Free Software Foundation's software and to any other program whose authors commit to using it.  (Some other Free Software Foundation software is covered by the GNU Library General Public License instead.)  You can apply it to your programs, too.</p>
<pre>
  When we speak of free software, we are referring to freedom, not
</pre>
<p>price.  Our General Public Licenses are designed to make sure that you have the freedom to distribute copies of free software (and charge for this service if you wish), that you receive source code or can get it if you want it, that you can change the software or use pieces of it in new free programs; and that you know you can do these things.</p>
<pre>
  To protect your rights, we need to make restrictions that forbid
</pre>
<p>anyone to deny you these rights or to ask you to surrender the rights. These restrictions translate to certain responsibilities for you if you distribute copies of the software, or if you modify it.</p>
<pre>
  For example, if you distribute copies of such a program, whether
</pre>
<p>gratis or for a fee, you must give the recipients all the rights that you have.  You must make sure that they, too, receive or can get the source code.  And you must show them these terms so they know their rights.</p>
<pre>
  We protect your rights with two steps: (1) copyright the software, and
</pre>
<p>(2) offer you this license which gives you legal permission to copy, distribute and/or modify the software.</p>
<pre>
  Also, for each author's protection and ours, we want to make certain
</pre>
<p>that everyone understands that there is no warranty for this free software.  If the software is modified by someone else and passed on, we want its recipients to know that what they have is not the original, so that any problems introduced by others will not reflect on the original authors' reputations.</p>
<pre>
  Finally, any free program is threatened constantly by software
</pre>
<p>patents.  We wish to avoid the danger that redistributors of a free program will individually obtain patent licenses, in effect making the program proprietary.  To prevent this, we have made it clear that any patent must be licensed for everyone's free use or not licensed at all.</p>
<pre>
  The precise terms and conditions for copying, distribution and
</pre>
<p>modification follow.</p>
<pre>
                    GNU GENERAL PUBLIC LICENSE
   TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION
</pre>

<pre>
  0. This License applies to any program or other work which contains
</pre>
<p>a notice placed by the copyright holder saying it may be distributed under the terms of this General Public License.  The "Program", below, refers to any such program or work, and a "work based on the Program" means either the Program or any derivative work under copyright law: that is to say, a work containing the Program or a portion of it, either verbatim or with modifications and/or translated into another language.  (Hereinafter, translation is included without limitation in the term "modification".)  Each licensee is addressed as "you".</p><p>Activities other than copying, distribution and modification are not covered by this License; they are outside its scope.  The act of running the Program is not restricted, and the output from the Program is covered only if its contents constitute a work based on the Program (independent of having been made by running the Program). Whether that is true depends on what the Program does.</p>
<pre>
  1. You may copy and distribute verbatim copies of the Program's
</pre>
<p>source code as you receive it, in any medium, provided that you conspicuously and appropriately publish on each copy an appropriate copyright notice and disclaimer of warranty; keep intact all the notices that refer to this License and to the absence of any warranty; and give any other recipients of the Program a copy of this License along with the Program.</p><p>You may charge a fee for the physical act of transferring a copy, and you may at your option offer warranty protection in exchange for a fee.</p>
<pre>
  2. You may modify your copy or copies of the Program or any portion
</pre>
<p>of it, thus forming a work based on the Program, and copy and distribute such modifications or work under the terms of Section 1 above, provided that you also meet all of these conditions:</p>
<pre>
    a) You must cause the modified files to carry prominent notices
    stating that you changed the files and the date of any change.
</pre>

<pre>
    b) You must cause any work that you distribute or publish, that in
    whole or in part contains or is derived from the Program or any
    part thereof, to be licensed as a whole at no charge to all third
    parties under the terms of this License.
</pre>

<pre>
    c) If the modified program normally reads commands interactively
    when run, you must cause it, when started running for such
    interactive use in the most ordinary way, to print or display an
    announcement including an appropriate copyright notice and a
    notice that there is no warranty (or else, saying that you provide
    a warranty) and that users may redistribute the program under
    these conditions, and telling the user how to view a copy of this
    License.  (Exception: if the Program itself is interactive but
    does not normally print such an announcement, your work based on
    the Program is not required to print an announcement.)
</pre>
<p>These requirements apply to the modified work as a whole.  If identifiable sections of that work are not derived from the Program, and can be reasonably considered independent and separate works in themselves, then this License, and its terms, do not apply to those sections when you distribute them as separate works.  But when you distribute the same sections as part of a whole which is a work based on the Program, the distribution of the whole must be on the terms of this License, whose permissions for other licensees extend to the entire whole, and thus to each and every part regardless of who wrote it.</p><p>Thus, it is not the intent of this section to claim rights or contest your rights to work written entirely by you; rather, the intent is to exercise the right to control the distribution of derivative or collective works based on the Program.</p><p>In addition, mere aggregation of another work not based on the Program with the Program (or with a work based on the Program) on a volume of a storage or distribution medium does not bring the other work under the scope of this License.</p>
<pre>
  3. You may copy and distribute the Program (or a work based on it,
</pre>
<p>under Section 2) in object code or executable form under the terms of Sections 1 and 2 above provided that you also do one of the following:</p>
<pre>
    a) Accompany it with the complete corresponding machine-readable
    source code, which must be distributed under the terms of Sections
    1 and 2 above on a medium customarily used for software interchange; or,
</pre>

<pre>
    b) Accompany it with a written offer, valid for at least three
    years, to give any third party, for a charge no more than your
    cost of physically performing source distribution, a complete
    machine-readable copy of the corresponding source code, to be
    distributed under the terms of Sections 1 and 2 above on a medium
    customarily used for software interchange; or,
</pre>

<pre>
    c) Accompany it with the information you received as to the offer
    to distribute corresponding source code.  (This alternative is
    allowed only for noncommercial distribution and only if you
    received the program in object code or executable form with such
    an offer, in accord with Subsection b above.)
</pre>
<p>The source code for a work means the preferred form of the work for making modifications to it.  For an executable work, complete source code means all the source code for all modules it contains, plus any associated interface definition files, plus the scripts used to control compilation and installation of the executable.  However, as a special exception, the source code distributed need not include anything that is normally distributed (in either source or binary form) with the major components (compiler, kernel, and so on) of the operating system on which the executable runs, unless that component itself accompanies the executable.</p><p>If distribution of executable or object code is made by offering access to copy from a designated place, then offering equivalent access to copy the source code from the same place counts as distribution of the source code, even though third parties are not compelled to copy the source along with the object code.</p>
<pre>
  4. You may not copy, modify, sublicense, or distribute the Program
</pre>
<p>except as expressly provided under this License.  Any attempt otherwise to copy, modify, sublicense or distribute the Program is void, and will automatically terminate your rights under this License. However, parties who have received copies, or rights, from you under this License will not have their licenses terminated so long as such parties remain in full compliance.</p>
<pre>
  5. You are not required to accept this License, since you have not
</pre>
<p>signed it.  However, nothing else grants you permission to modify or distribute the Program or its derivative works.  These actions are prohibited by law if you do not accept this License.  Therefore, by modifying or distributing the Program (or any work based on the Program), you indicate your acceptance of this License to do so, and all its terms and conditions for copying, distributing or modifying the Program or works based on it.</p>
<pre>
  6. Each time you redistribute the Program (or any work based on the
</pre>
<p>Program), the recipient automatically receives a license from the original licensor to copy, distribute or modify the Program subject to these terms and conditions.  You may not impose any further restrictions on the recipients' exercise of the rights granted herein. You are not responsible for enforcing compliance by third parties to this License.</p>
<pre>
  7. If, as a consequence of a court judgment or allegation of patent
</pre>
<p>infringement or for any other reason (not limited to patent issues), conditions are imposed on you (whether by court order, agreement or otherwise) that contradict the conditions of this License, they do not excuse you from the conditions of this License.  If you cannot distribute so as to satisfy simultaneously your obligations under this License and any other pertinent obligations, then as a consequence you may not distribute the Program at all.  For example, if a patent license would not permit royalty-free redistribution of the Program by all those who receive copies directly or indirectly through you, then the only way you could satisfy both it and this License would be to refrain entirely from distribution of the Program.</p><p>If any portion of this section is held invalid or unenforceable under any particular circumstance, the balance of the section is intended to apply and the section as a whole is intended to apply in other circumstances.</p><p>It is not the purpose of this section to induce you to infringe any patents or other property right claims or to contest validity of any such claims; this section has the sole purpose of protecting the integrity of the free software distribution system, which is implemented by public license practices.  Many people have made generous contributions to the wide range of software distributed through that system in reliance on consistent application of that system; it is up to the author/donor to decide if he or she is willing to distribute software through any other system and a licensee cannot impose that choice.</p><p>This section is intended to make thoroughly clear what is believed to be a consequence of the rest of this License.</p>
<pre>
  8. If the distribution and/or use of the Program is restricted in
</pre>
<p>certain countries either by patents or by copyrighted interfaces, the original copyright holder who places the Program under this License may add an explicit geographical distribution limitation excluding those countries, so that distribution is permitted only in or among countries not thus excluded.  In such case, this License incorporates the limitation as if written in the body of this License.</p>
<pre>
  9. The Free Software Foundation may publish revised and/or new versions
</pre>
<p>of the General Public License from time to time.  Such new versions will be similar in spirit to the present version, but may differ in detail to address new problems or concerns.</p><p>Each version is given a distinguishing version number.  If the Program specifies a version number of this License which applies to it and "any later version", you have the option of following the terms and conditions either of that version or of any later version published by the Free Software Foundation.  If the Program does not specify a version number of this License, you may choose any version ever published by the Free Software Foundation.</p>
<pre>
  10. If you wish to incorporate parts of the Program into other free
</pre>
<p>programs whose distribution conditions are different, write to the author to ask for permission.  For software which is copyrighted by the Free Software Foundation, write to the Free Software Foundation; we sometimes make exceptions for this.  Our decision will be guided by the two goals of preserving the free status of all derivatives of our free software and of promoting the sharing and reuse of software generally.</p>
<pre>
                            NO WARRANTY
</pre>

<pre>
  11. BECAUSE THE PROGRAM IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
</pre>
<p>FOR THE PROGRAM, TO THE EXTENT PERMITTED BY APPLICABLE LAW.  EXCEPT WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES PROVIDE THE PROGRAM "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.  THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE PROGRAM IS WITH YOU.  SHOULD THE PROGRAM PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL NECESSARY SERVICING, REPAIR OR CORRECTION.</p>
<pre>
  12. IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
</pre>
<p>WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR REDISTRIBUTE THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.</p>
<pre>
                     END OF TERMS AND CONDITIONS
</pre>

<pre>
            How to Apply These Terms to Your New Programs
</pre>

<pre>
  If you develop a new program, and you want it to be of the greatest
</pre>
<p>possible use to the public, the best way to achieve this is to make it free software which everyone can redistribute and change under these terms.</p>
<pre>
  To do so, attach the following notices to the program.  It is safest
</pre>
<p>to attach them to the start of each source file to most effectively convey the exclusion of warranty; and each file should have at least the "copyright" line and a pointer to where the full notice is found.</p>
<pre>
    &lt;one line to give the program's name and a brief idea of what it does.&gt;
    Copyright (C) &lt;year&gt;  &lt;name of author&gt;
</pre>

<pre>
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.
</pre>

<pre>
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
</pre>

<pre>
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
</pre>
<p>Also add information on how to contact you by electronic and paper mail.</p><p>If the program is interactive, make it output a short notice like this when it starts in an interactive mode:</p>
<pre>
    Gnomovision version 69, Copyright (C) year name of author
    Gnomovision comes with ABSOLUTELY NO WARRANTY; for details type `show w'.
    This is free software, and you are welcome to redistribute it
    under certain conditions; type `show c' for details.
</pre>
<p>The hypothetical commands `show w' and `show c' should show the appropriate parts of the General Public License.  Of course, the commands you use may be called something other than `show w' and `show c'; they could even be mouse-clicks or menu items--whatever suits your program.</p><p>You should also get your employer (if you work as a programmer) or your school, if any, to sign a "copyright disclaimer" for the program, if necessary.  Here is a sample; alter the names:</p>
<pre>
  Yoyodyne, Inc., hereby disclaims all copyright interest in the program
  `Gnomovision' (which makes passes at compilers) written by James Hacker.
</pre>

<pre>
  &lt;signature of Ty Coon&gt;, 1 April 1989
  Ty Coon, President of Vice
</pre>
<p>This General Public License does not permit incorporating your program into proprietary programs.  If your program is a subroutine library, you may consider it more useful to permit linking proprietary applications with the library.  If this is what you want to do, use the GNU Library General Public License instead of this License.</p>
  </dd>
  <dt>
    <p><strong>RASLIC</strong></p>
  </dt>
  <dd>
    <p>If you do not use the GPL, the following license terms apply:</p><p>RasMol License</p><p>Even though the authors of the various documents and software found here have made a good faith effort to ensure that the documents are correct and that the software performs according to its documentation, and we would greatly appreciate hearing of any problems you may encounter, the programs and documents any files created by the programs are provided **AS IS** without any warranty as to correctness, merchantability or fitness for any particular or general use.</p><p>THE RESPONSIBILITY FOR ANY ADVERSE CONSEQUENCES FROM THE USE OF PROGRAMS OR DOCUMENTS OR ANY FILE OR FILES CREATED BY USE OF THE PROGRAMS OR DOCUMENTS LIES SOLELY WITH THE USERS OF THE PROGRAMS OR DOCUMENTS OR FILE OR FILES AND NOT WITH AUTHORS OF THE PROGRAMS OR DOCUMENTS.</p><p>Subject to your acceptance of the conditions stated above, and your respect for the terms and conditions stated in the notices below, if you are not going to make any modifications or create derived works, you are given permission to freely copy and distribute this package, provided you do the following:</p>
<pre>
  1.  Either include the complete documentation, especially the file
</pre>
<p>NOTICE, with what you distribute or provide a clear indication where people can get a copy of the documentation; and</p>
<pre>
  2.  Please give credit where credit is due citing the version and
</pre>
<p>original authors properly; and</p>
<pre>
  3.  Please do not give anyone the impression that the original
</pre>
<p>authors are providing a warranty of any kind.</p><p>If you would like to use major pieces of RasMol in some other program, make modifications to RasMol, or in some other way make what a lawyer would call a "derived work", you are not only permitted to do so, you are encouraged to do so. In addition to the things we discussed above, please do the following:</p>
<pre>
  4.  Please explain in your documentation how what you did differs
</pre>
<p>from this version of RasMol; and</p>
<pre>
  5.  Please make your modified source code available.
</pre>
<p>This version of RasMol is _not_ in the public domain, but it is given freely to the community in the hopes of advancing science.  If you make changes, please make them in a responsible manner, and please offer us the opportunity to include those changes in future versions of RasMol.</p>
  </dd>
  <dt>
    <p><strong>General Notice</strong></p>
  </dt>
  <dd>
    <p>The following notice applies to this work as a whole and to the works included within it:</p><p>* Creative endeavors depend on the lively exchange of ideas. There are laws and customs which establish rights and responsibilities for authors and the users of what authors create. This notice is not intended to prevent you from using the software and documents in this package, but to ensure that there are no misunderstandings about terms and conditions of such use.</p><p>* Please read the following notice carefully. If you do not understand any portion of this notice, please seek appropriate professional legal advice before making use of the software and documents included in this software package. In addition to whatever other steps you may be obliged to take to respect the intellectual property rights of the various parties involved, if you do make use of the software and documents in this package, please give credit where credit is due by citing this package, its authors and the URL or other source from which you obtained it, or equivalent primary references in the literature with the same authors.</p><p>* Some of the software and documents included within this software package are the intellectual property of various parties, and placement in this package does not in any way imply that any such rights have in any way been waived or diminished.</p><p>* With respect to any software or documents for which a copyright exists, ALL RIGHTS ARE RESERVED TO THE OWNERS OF SUCH COPYRIGHT.</p><p>* Even though the authors of the various documents and software found here have made a good faith effort to ensure that the documents are correct and that the software performs according to its documentation, and we would greatly appreciate hearing of any problems you may encounter, the programs and documents and any files created by the programs are provided **AS IS** without any warranty as to correctness, merchantability or fitness for any particular or general use.</p><p>* THE RESPONSIBILITY FOR ANY ADVERSE CONSEQUENCES FROM THE USE OF PROGRAMS OR DOCUMENTS OR ANY FILE OR FILES CREATED BY USE OF THE PROGRAMS OR DOCUMENTS LIES SOLELY WITH THE USERS OF THE PROGRAMS OR DOCUMENTS OR FILE OR FILES AND NOT WITH AUTHORS OF THE PROGRAMS OR DOCUMENTS.</p><p>See the files GPL and RASLIC for two alternate ways to license this package.</p>
  </dd>
  <dt>
    <p><strong>RasMol V2.6 Notice</strong></p>
  </dt>
  <dd>
    <p>The following notice applies to RasMol V 2.6 and older RasMol versions.</p><p>Information in this document is subject to change without notice and does not represent a commitment on the part of the supplier. This package is sold/distributed subject to the condition that it shall not, by way of trade or otherwise, be lent, re-sold, hired out or otherwise circulated without the supplier's prior consent, in any form of packaging or cover other than that in which it was produced. No part of this manual or accompanying software may be reproduced, stored in a retrieval system on optical or magnetic disk, tape or any other medium, or transmitted in any form or by any means, electronic, mechanical, photocopying, recording or otherwise for any purpose other than the purchaser's personal use.</p><p>This product is not to be used in the planning, construction, maintenance, operation or use of any nuclear facility nor the flight, navigation or communication of aircraft or ground support equipment. The author shall not be liable, in whole or in part, for any claims or damages arising from such use, including death, bankruptcy or outbreak of war.</p>
  </dd>
  <dt>
    <p><strong>IUCR Policy</strong></p>
  </dt>
  <dd>
    <p><strong>The IUCr Policy for the Protection and the Promotion of the STAR File and</strong> <strong>CIF Standards for Exchanging and Archiving Electronic Data.</strong></p><p><strong>Overview</strong></p><p>The Crystallographic Information File (CIF)[1] is a standard for information interchange promulgated by the International Union of Crystallography (IUCr). CIF (Hall, Allen & Brown, 1991) is the recommended method for submitting publications to Acta Crystallographica Section C and reports of crystal structure determinations to other sections of Acta Crystallographica and many other journals. The syntax of a CIF is a subset of the more general STAR File[2] format. The CIF and STAR File approaches are used increasingly in the structural sciences for data exchange and archiving, and are having a significant influence on these activities in other fields.</p><p><strong>Statement of intent</strong></p><p>The IUCr's interest in the STAR File is as a general data interchange standard for science, and its interest in the CIF, a conformant derivative of the STAR File, is as a concise data exchange and archival standard for crystallography and structural science.</p><p><strong>Protection of the standards</strong></p><p>To protect the STAR File and the CIF as standards for interchanging and archiving electronic data, the IUCr, on behalf of the scientific community,</p>
<pre>
   * holds the copyrights on the standards themselves,
</pre>

<pre>
   * owns the associated trademarks and service marks, and
</pre>

<pre>
   * holds a patent on the STAR File.
</pre>
<p>These intellectual property rights relate solely to the interchange formats, not to the data contained therein, nor to the software used in the generation, access or manipulation of the data.</p><p><strong>Promotion of the standards</strong></p><p>The sole requirement that the IUCr, in its protective role, imposes on software purporting to process STAR File or CIF data is that the following conditions be met prior to sale or distribution.</p>
<pre>
   * Software claiming to read files written to either the STAR File or the
</pre>
<p>CIF standard must be able to extract the pertinent data from a file conformant to the STAR File syntax, or the CIF syntax, respectively.</p>
<pre>
   * Software claiming to write files in either the STAR File, or the CIF,
</pre>
<p>standard must produce files that are conformant to the STAR File syntax, or the CIF syntax, respectively.</p>
<pre>
   * Software claiming to read definitions from a specific data dictionary
</pre>
<p>approved by the IUCr must be able to extract any pertinent definition which is conformant to the dictionary definition language (DDL)[3] associated with that dictionary.</p><p>The IUCr, through its Committee on CIF Standards, will assist any developer to verify that software meets these conformance conditions.</p><p><strong>Glossary of terms</strong></p><p>[1] CIF:</p><p>is a data file conformant to the file syntax defined at http://www.iucr.org/iucr-top/cif/spec/index.html</p><p>[2] STAR File:</p><p>is a data file conformant to the file syntax defined at http://www.iucr.org/iucr-top/cif/spec/star/index.html</p><p>[3] DDL:</p><p>is a language used in a data dictionary to define data items in terms of "attributes". Dictionaries currently approved by the IUCr, and the DDL versions used to construct these dictionaries, are listed at http://www.iucr.org/iucr-top/cif/spec/ddl/index.html</p><p>Last modified: 30 September 2000</p><p>IUCr Policy Copyright (C) 2000 International Union of Crystallography</p>
  </dd>
  <dt>
    <p><strong>CBFLIB</strong></p>
  </dt>
  <dd>
    <p>The following Disclaimer Notice applies to CBFlib V0.1, from which this code in part is derived.</p><p>* The items furnished herewith were developed under the sponsorship of the U.S. Government. Neither the U.S., nor the U.S. D.O.E., nor the Leland Stanford Junior University, nor their employees, makes any warranty, express or implied, or assumes any liability or responsibility for accuracy, completeness or usefulness of any information, apparatus, product or process disclosed, or represents that its use will not infringe privately-owned rights. Mention of any product, its manufacturer, or suppliers shall not, nor is it intended to, imply approval, disapproval, or fitness for any particular use. The U.S. and the University at all times retain the right to use and disseminate the furnished items for any purpose whatsoever.</p><p>Notice 91 02 01</p>
  </dd>
  <dt>
    <p><strong>CIFPARSE</strong></p>
  </dt>
  <dd>
    <p>Portions of this software are loosely based on the CIFPARSE software package from the NDB at Rutgers University.  See</p>
<pre>
   http://ndbserver.rutgers.edu/NDB/mmcif/software
</pre>
<p>CIFPARSE is part of the NDBQUERY application, a program component of the Nucleic Acid Database Project [ H. M. Berman, W. K. Olson, D. L. Beveridge, J. K. Westbrook, A. Gelbin, T. Demeny, S. H. Shieh, A. R. Srinivasan, and B. Schneider. (1992). The Nucleic Acid Database: A Comprehensive Relational Database of Three-Dimensional Structures of Nucleic Acids. Biophys J., 63, 751-759.], whose cooperation is gratefully acknowledged, especially in the form of design concepts created by J. Westbrook.</p><p>Please be aware of the following notice in the CIFPARSE API:</p><p>This software is provided WITHOUT WARRANTY OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE OR ANY OTHER WARRANTY, EXPRESS OR IMPLIED. RUTGERS MAKE NO REPRESENTATION OR WARRANTY THAT THE SOFTWARE WILL NOT INFRINGE ANY PATENT, COPYRIGHT OR OTHER PROPRIETARY RIGHT.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>RasMol is a molecular graphics program intended for the visualisation of proteins, nucleic acids and small molecules. The program is aimed at display, teaching and generation of publication quality images.   RasMol runs on wide range of architectures and operating systems including Microsoft Windows, Apple Macintosh, UNIX and VMS systems. UNIX and VMS versions require an 8, 24 or 32 bit colour X Windows display (X11R4 or later).  The X Windows version of RasMol provides optional support for a hardware dials box and accelerated shared memory communication (via the XInput and MIT-SHM extensions) if available on the current X Server.</p><p>The program reads in a molecule coordinate file and interactively displays the molecule on the screen in a variety of colour schemes and molecule representations. Currently available representations include depth-cued wireframes, 'Dreiding' sticks, spacefilling (CPK) spheres, ball and stick, solid and strand biomolecular ribbons, atom labels and dot surfaces.</p><p>Up to 5 molecules may be loaded and displayed at once.  Any one or all of  the molecules may be rotated and translated.</p><p>The RasMol help facility can be accessed by typing "help &lt;topic&gt;" or "help &lt;topic&gt; &lt;subtopic&gt;" from the command line. A complete list of RasMol commands may be displayed by typing "help commands". A single question mark may also be used to abbreviate the keyword "help".  Please type "help notices" for important notices.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COMMANDS</h2>
        <div class="sectioncontent">
<p>RasMol allows the execution of interactive commands typed at the <strong>RasMol&gt;</strong> prompt in the terminal window. Each command must be given on a separate line. Keywords are case insensitive and may be entered in either upper or lower case letters. All whitespace characters are ignored except to separate keywords and their arguments.</p><p>All commands may be prefixed by a parenthesized <strong>atom expression</strong> to temporarily select certain atoms just for the execution of that one command.  After execution of the command, the previous selection is restored except for the commands <strong>select</strong> , <strong>restrict</strong> and <strong>script.</strong></p><p>The commands/keywords currently recognised by RasMol are given below.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>Backbone</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>backbone</strong> command permits the representation of a polypeptide backbone as a series of bonds connecting the adjacent alpha carbons of each amino acid in a chain. The display of these backbone 'bonds' is turned on and off by the command parameter in the same way as with the <strong>wireframe</strong> command. The command <strong>backbone off</strong> turns off the selected 'bonds', and <strong>backbone on</strong> or with a number turns them on. The number can be used to specify the cylinder radius of the representation in either Angstrom or RasMol units. A parameter value of 500 (2.0 Angstroms) or above results in a "Parameter value too large" error. Backbone objects may be coloured using the RasMol <strong>colour backbone</strong> command.</p><p>The reserved word backbone is also used as a predefined set ("help sets") and as a parameter to the <strong>set hbond</strong> and <strong>set ssbond</strong> commands. The RasMol command <strong>trace</strong> renders a smoothed backbone, in contrast to <strong>backbone</strong> which connects alpha carbons with straight lines.</p><p>The backbone may be displayed with dashed lines by use of the <strong>backbone dash</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Background</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>background</strong> command is used to set the colour of the "canvas" background. The colour may be given as either a colour name or a comma separated triple of Red, Green and Blue (RGB) components enclosed in square brackets. Typing the command <strong>help colours</strong> will give a list of the predefined colour names recognised by RasMol. When running under X Windows, RasMol also recognises colours in the X server's colour name database.</p><p>The <strong>background</strong> command is synonymous with the RasMol <strong>set background</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Bond</strong></p>
  </dt>
  <dd>
    <p>The RasMol command <strong>bond &lt;number&gt; &lt;number&gt; +</strong> adds the designated bond to the drawing, increasing the bond order if the bond already exists.  The command <strong>bond &lt;number&gt; &lt;number&gt; pick</strong> selects the two atoms specified by the atom serial numbers as the two ends of a bond around which the <strong>rotate bond &lt;angle&gt;</strong> command will be applied.  If no bond exists, it is created.</p><p>Rotation around a previously picked bond may be specified by the <strong>rotate bond &lt;angle&gt;</strong> command, or may also be controlled with the mouse, using the <strong>bond rotate on/off</strong> or the equivalent <strong>rotate bond on/off</strong> commands.</p>
  </dd>
  <dt>
    <p><strong>Bulgarian</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>Bulgarian</strong> command sets the menus and messages to the Bulgarian versions.</p><p>This command may not work correctly unless appropriate fonts have been installed.  The commands <strong>Bulgarian,</strong> <strong>Chinese,</strong> <strong>English,</strong> <strong>French,</strong> <strong>Italian,</strong> <strong>Russian</strong> and <strong>Spanish</strong> may be used to select Bulgarian, Chinese, English, French, Italian, Japanese, Russian and Spanish menus and messages if the appropriate fonts have been installed.</p>
  </dd>
  <dt>
    <p><strong>Cartoon</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>cartoon</strong> command does a display of a molecule <strong>ribbons</strong> as Richardson (MolScript) style protein <strong>cartoons,</strong> implemented as thick (deep) ribbons. The easiest way to obtain a cartoon representation of a protein is to use the <strong>Cartoons</strong> option on the <strong>Display</strong> menu. The <strong>cartoon</strong> command represents the currently selected residues as a deep ribbon with width specified by the command's argument. Using the command without a parameter results in the ribbon's width being taken from the protein's secondary structure, as described in the <strong>ribbons</strong> command. By default, the C-termini of beta-sheets are displayed as arrow heads. This may be enabled and disabled using the <strong>set cartoons</strong> command. The depth of the cartoon may be adjusted using the <strong>set cartoons &lt;number&gt;</strong> command. The <strong>set cartoons</strong> command without any parameters returns these two options to their default values.</p>
  </dd>
  <dt>
    <p><strong>Centre</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>centre</strong> command defines the point about which the <strong>rotate</strong> command and the scroll bars rotate the current molecule. Without a parameter the centre command resets the centre of rotation to be the centre of gravity of the molecule. If an atom expression is specified, RasMol rotates the molecule about the centre of gravity of the set of atoms specified by the expression. Hence, if a single atom is specified by the expression, that atom will remain 'stationary' during rotations.</p><p>Type <strong>help expression</strong> for more information on RasMol atom expressions.</p><p>Alternatively the centring may be given as a comma separated triple of [CenX, CenY, CenZ] offsets in RasMol units (1/250 of an Angstrom) from the centre of gravity.  The triple must be enclosed in square brackets.</p><p>The optional forms <strong>centre ... translate</strong> and <strong>centre ... center</strong> may be used to specify use of a translated centre of rotation (not necessarily in the centre of the canvas) or a centre of rotation which is placed at the centre of the canvas.  Starting with RasMol 2.7.2, the default is to center the new axis on the canvas.</p>
  </dd>
  <dt>
    <p><strong>Chinese</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>Chinese</strong> command sets the menus and messages to the Chinese versions.</p><p>This command may not work correctly unless appropriate fonts have been installed.  The commands <strong>Bulgarian,</strong> <strong>Chinese,</strong> <strong>English,</strong> <strong>French,</strong> <strong>Italian,</strong> <strong>Russian</strong> and <strong>Spanish</strong> may be used to select Bulgarian, Chinese, English, French, Italian, Japanese, Russian and Spanish menus and messages if the appropriate fonts have been installed.</p>
  </dd>
  <dt>
    <p><strong>Clipboard</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>clipboard</strong> command places a copy of the currently displayed image on the local graphics 'clipboard'. Note: this command is not yet supported on UNIX or VMS machines. It is intended to make transferring images between applications easier under Microsoft Windows or on an Apple Macintosh.</p><p>When using RasMol on a UNIX or VMS system this functionality may be achieved by generating a raster image in a format that can be read by the receiving program using the RasMol <strong>write</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Colour</strong></p>
  </dt>
  <dd>
    <p>Colour the atoms (or other objects) of the selected region. The colour may be given as either a colour name or a comma separated triple of Red, Green and Blue (RGB) components enclosed in square brackets. Typing the command <strong>help colours</strong> will give a list of all the predefined colour names recognised by RasMol.</p><p>Allowed objects are <strong>atoms,</strong> <strong>bonds,</strong> <strong>backbone,</strong> <strong>ribbons,</strong> <strong>labels,</strong> <strong>dots,</strong> <strong>hbonds,</strong> <strong>map,</strong> and <strong>ssbonds.</strong> If no object is specified, the default keyword <strong>atom</strong> is assumed. Some colour schemes are defined for certain object types. The colour scheme <strong>none</strong> can be applied to all objects except atoms and dots, stating that the selected objects have no colour of their own, but use the colour of their associated atoms (i.e. the atoms they connect). <strong>Atom</strong> objects can also be coloured by <strong>alt,</strong> <strong>amino,</strong> <strong>chain,</strong> <strong>charge,</strong> <strong>cpk,</strong> <strong>group,</strong> <strong>model,</strong> <strong>shapely,</strong> <strong>structure,</strong> <strong>temperature</strong> or <strong>user.</strong> Hydrogen bonds can also be coloured by <strong>type</strong> and dot surfaces can also be coloured by <strong>electrostatic potential.</strong> For more information type <strong>help colour &lt;colour&gt;.</strong> Map objects may be coloured by specific color of by nearest atom.</p>
  </dd>
  <dt>
    <p><strong>ColourMode</strong></p>
  </dt>
  <dd>
    <p>ColourMode allows the user to switch between using the new <strong>colour</strong> method. At present, the new coloring technique is the same as the old one, but to preserve compatibility for older scripts it may be wise to add a "colormode on" near the top of your script somewhere, if the script was designed for version 2.7.3 of RasMol or earlier. The new color method, when completed, aims to fix a few bugs in the coloring routines.</p>
  </dd>
  <dt>
    <p><strong>Connect</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>connect</strong> command is used to force RasMol to (re)calculate the connectivity of the current molecule. If the original input file contained connectivity information, this is discarded. The command <strong>connect false</strong> uses a fast heuristic algorithm that is suitable for determining bonding in large bio-molecules such as proteins and nucleic acids. The command <strong>connect true</strong> uses a slower more accurate algorithm based upon covalent radii that is more suitable to small molecules containing inorganic elements or strained rings. If no parameters are given, RasMol determines which algorithm to use based on the number of atoms in the input file. Greater than 255 atoms causes RasMol to use the faster implementation. This is the method used to determine bonding, if necessary, when a molecule is first read in using the <strong>load</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Defer</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>defer</strong> command adds the command given to the macro with given name, if no name is given, the command is added to the macro with a blank name. The command <strong>zap</strong> is a special case. In that case the macro is erased. If no name is given the command must begin with a selection, e.g. <strong>defer (selection).spacefill</strong></p><p>The deferred commands accumulated under the given name can be executed using the <strong>execute</strong> command</p>
  </dd>
  <dt>
    <p><strong>Define</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>define</strong> command allows the user to associate an arbitrary set of atoms with a unique identifier. This allows the definition of user-defined sets. These sets are declared statically, i.e. once defined the contents of the set do not change, even if the expression defining them depends on the current transformation and representation of the molecule.</p>
  </dd>
  <dt>
    <p><strong>Depth</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>depth</strong> command enables, disables or positions the back-clipping plane of the molecule. The program only draws those portions of the molecule that are closer to the viewer than the clipping plane. Integer values range from zero at the very back of the molecule to 100 which is completely in front of the molecule. Intermediate values determine the percentage of the molecule to be drawn.</p><p>This command interacts with the <strong>slab &lt;value&gt;</strong> command, which clips to the front of a given z-clipping plane.</p>
  </dd>
  <dt>
    <p><strong>Dots</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>dots</strong> command is used to generate a van der Waals' dot surface around the currently selected atoms. Dot surfaces display regularly spaced points on a sphere of van der Waals' radius about each selected atom. Dots that would are 'buried' within the van der Waals' radius of any other atom (selected or not) are not displayed. The command <strong>dots on</strong> deletes any existing dot surface and generates a dots surface around the currently selected atom set with a default dot density of 100. The command <strong>dots off</strong> deletes any existing dot surface. The dot density may be specified by providing a numeric parameter between 1 and 1000. This value approximately corresponds to the number of dots on the surface of a medium sized atom.</p><p>By default, the colour of each point on a dot surface is the colour of its closest atom at the time the surface is generated. The colour of the whole dot surface may be changed using the <strong>colour dots</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Echo</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>echo</strong> command is used to display a message in the RasMol command/terminal window. The string parameter may optionally be delimited in double quote characters. If no parameter is specified, the <strong>echo</strong> command displays a blank line. This command is particularly useful for displaying text from within a RasMol <strong>script</strong> file.</p>
  </dd>
  <dt>
    <p><strong>English</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>English</strong> command sets the menus and messages to the English versions.</p><p>This command may not work correctly unless appropriate fonts have been installed.  The commands <strong>Bulgarian,</strong> <strong>Chinese,</strong> <strong>English,</strong> <strong>French,</strong> <strong>Italian,</strong> <strong>Russian</strong> and <strong>Spanish</strong> may be used to select Bulgarian, Chinese, English, French, Italian, Japanese, Russian and Spanish menus and messages if the appropriate fonts have been installed.</p>
  </dd>
  <dt>
    <p><strong>Execute</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>execute</strong> command:</p><p>1.  saves the old poise of the molecule (translation, rotation and zoom)</p><p>2. executes the specified macro suppressing both screen updates and recording</p><p>3. animates motion of the newly rendered molecule linearly from the old poise to the new poise</p><p>The macro must have been previously defined by calls to the <strong>defer</strong> command.</p><p>The animation of the motion depends on the prior settings of the <strong>record</strong> command.</p>
  </dd>
  <dt>
    <p><strong>French</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>French</strong> command sets the menus and messages to the French versions.</p><p>This command may not work correctly unless appropriate fonts have been installed.  The commands <strong>Bulgarian,</strong> <strong>Chinese,</strong> <strong>English,</strong> <strong>French,</strong> <strong>Italian,</strong> <strong>Russian</strong> and <strong>Spanish</strong> may be used to select Bulgarian, Chinese, English, French, Italian, Japanese, Russian and Spanish menus and messages if the appropriate fonts have been installed.</p>
  </dd>
  <dt>
    <p><strong>HBonds</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>hbond</strong> command is used to represent the hydrogen bonding of the protein molecule's backbone. This information is useful in assessing the protein's secondary structure. Hydrogen bonds are represented as either dotted lines or cylinders between the donor and acceptor residues. The first time the <strong>hbond</strong> command is used, the program searches the structure of the molecule to find hydrogen bonded residues and reports the number of bonds to the user. The command <strong>hbonds on</strong> displays the selected 'bonds' as dotted lines, and the <strong>hbonds off</strong> turns off their display. The colour of hbond objects may be changed by the <strong>colour hbond</strong> command. Initially, each hydrogen bond has the colours of its connected atoms.</p><p>By default the dotted lines are drawn between the accepting oxygen and the donating nitrogen. By using the <strong>set hbonds</strong> command the alpha carbon positions of the appropriate residues may be used instead. This is especially useful when examining proteins in backbone representation.</p>
  </dd>
  <dt>
    <p><strong>Help</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>help</strong> command provides on-line help on the given topic.</p>
  </dd>
  <dt>
    <p><strong>Italian</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>Italian</strong> command sets the menus and messages to the Italian versions.</p><p>This command may not work correctly unless appropriate fonts have been installed.  The commands <strong>Bulgarian,</strong> <strong>Chinese,</strong> <strong>English,</strong> <strong>French,</strong> <strong>Italian,</strong> <strong>Russian</strong> and <strong>Spanish</strong> may be used to select Bulgarian, Chinese, English, French, Italian, Japanese, Russian and Spanish menus and messages if the appropriate fonts have been installed.</p>
  </dd>
  <dt>
    <p><strong>Japanese</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>Japanese</strong> command sets the menus and messages to the Japanese versions.</p><p>This command may not work correctly unless appropriate fonts have been installed.  The commands <strong>Bulgarian,</strong> <strong>Chinese,</strong> <strong>English,</strong> <strong>French,</strong> <strong>Italian,</strong> <strong>Russian</strong> and <strong>Spanish</strong> may be used to select Bulgarian, Chinese, English, French, Italian, Japanese, Russian and Spanish menus and messages if the appropriate fonts have been installed.</p>
  </dd>
  <dt>
    <p><strong>Label</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>label</strong> command allows an arbitrary formatted text string to be associated with each currently selected atom.  This string may contain embedded 'expansion specifiers' which display properties of the atom being labelled. An expansion specifier consists of a '%' character followed by a single alphabetic character specifying the property to be displayed. An actual '%' character may be displayed by using the expansion specifier '%%'.</p><p>Atom labelling for the currently selected atoms may be turned off with the command <strong>label off.</strong> By default, if no string is given as a parameter, RasMol uses labels appropriate for the current molecule.</p><p>The colour of each label may be changed using the <strong>colour label</strong> command. By default, each label is drawn in the same colour as the atom to which it is attached. The size and spacing of the displayed text may be changed using the <strong>set fontsize</strong> command.  The width of the strokes in the displayed text may be changed  using the <strong>set fontstroke</strong>  command.</p>
  </dd>
  <dt>
    <p><strong>Load</strong></p>
  </dt>
  <dd>
    <p>Load a molecule coordinate file into RasMol. Valid molecule file formats are <strong>pdb</strong> (Protein Data Bank format), <strong>mdl</strong> (Molecular Design Limited's MOL file format), <strong>alchemy</strong> (Tripos' Alchemy file format), <strong>mol2</strong> (Tripos' Sybyl Mol2 file format), <strong>charmm</strong> (CHARMm file format), <strong>xyz</strong> (MSC's XMol XYZ file format), <strong>mopac</strong> (J. P. Stewart's MOPAC file format) or <strong>cif</strong> (IUCr CIF or mmCIF file format). If no file format is specified, <strong>PDB,</strong> <strong>CIF,</strong> or <strong>mmCIF</strong> is assumed by default. Up to 20 molecules may be loaded at a time. If CHEM_COMP ligand models are included in an mmCIF file, they will be loaded as NMR models, first giving the all the NMR models for model coordinates if specified and then giving all the NMR models for ideal model coordinates.</p><p>To delete a molecule prior to loading another use the RasMol <strong>zap</strong> command.  To select a molecule for manipulation use the RasMol <strong>molecule &lt;n&gt;</strong> command.</p><p>The <strong>load</strong> command selects all the atoms in the molecule, centres it on the screen and renders it as a CPK coloured wireframe model. If the molecule contains no bonds (i.e. contains only alpha carbons), it is drawn as an alpha carbon backbone. If the file specifies fewer bonds than atoms, RasMol determines connectivity using the <strong>connect</strong> command.</p><p>The <strong>load inline</strong> command also allows the storing of atom coordinates in scripts to allow better integration with WWW browsers. A load command executed inside a script file may specify the keyword <strong>inline</strong> instead of a conventional filename. This option specifies that the coordinates of the molecule to load are stored in the same file as the currently executing commands.</p>
  </dd>
  <dt>
    <p><strong>Map</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map</strong> commands manipulate electron density maps in coordination with the display of molecules.  These commands are very memory intensive and may not work on machines with limited memory.  Each molecule may have as many maps as available memory permits.  Maps may be read from files or generated from Gaussian density distributions around atoms.</p><p><strong>map colour,</strong> to colour a map according to a given colour scheme, <strong>map generate,</strong> to generate a map from selected atoms based on pseudo-Gaussians, <strong>map level,</strong> to set the contouring level for selected maps, <strong>map load,</strong> to load a map from a file, <strong>map mask</strong> to designate a mask for the selected maps, <strong>map resolution,</strong> to set the resolution for contouring selected maps, <strong>map restrict,</strong> to select one or more maps and to disable all others, <strong>map save,</strong> to save map information to a file, <strong>map scale,</strong> <strong>control the scaling of pseudo-Gaussians when generating maps,</strong> <strong>map select,</strong> to select one or more maps, <strong>map show,</strong> to display information about one or more maps or about the parameters to be used in generating or loading the next map, <strong>map spacing,</strong> to set the spacing betwen contour lines of selected maps, <strong>map spread,</strong> to set the variance of the Gaussians for map generation as a fraction of the atomic radius, and <strong>map zap</strong> to delete previously generated or loaded maps.</p><p>The effect of <strong>map generate</strong> and <strong>map load</strong> commands is modified by the <strong>map mask</strong> command which limits the portion of the display space that can be considered for display of maps.</p>
  </dd>
  <dt>
    <p><strong>Map colour</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map colour</strong> command colours the selected maps according to the specified colour scheme.  The colour scheme may be a colour name or and RBG triple in brackets, or the keyword <strong>atom</strong> to cause the map points to be coloured by the color of the nearest atom.</p>
  </dd>
  <dt>
    <p><strong>Map generate</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map generate</strong> command generates a map from whatever atoms are currently selected, by summing electron densities approximated by Gaussian distributions. The height of each Gaussian is determined by the setting of the <strong>map scale</strong> command. In the default of map scale true, each Gaussian has a height proportional element type of the atom. If the optional 'LRSurf' parameter is given or if map scale false has been executed, each Gaussian is scaled so that the Gaussian contour level 1 is at the van der Waals radius. In either case a standard deviation determined by the most recently specified spread or resolution is used.  If a non-zero spread has been given the radius of the atom is multiplied by the spread to find the standard deviation.  The default is 2/3rds.  If a resolution has been given, the spread is inferred as 2/3rds of the resolution.</p><p>For example, if the resolution is given as 1., and the atom in question is a Carbon with a van der Waals radius of 468 RasMol units (1.87 Angstroms), the inferred spead is .6667, and the standard deviation of the Gaussian is taken as 1.25 Angstroms.</p><p>If the spread has been set to zero, the spread for each atom is determined from the van der Waals radius and the probe atom radius to simulate the effect of a Lee-Richards surface.</p><p>If no specific map was given by the map selector, the new map is given the next available map number.</p><p>If a specific map was given by the map selector, the new map replaces that map.  If more than one map was given by the map selector, the new map replaces the lowest numbered of the selected maps.  In any case the new map becomes the currently selected map.</p><p>The map is displayed as dots, mesh or a surface, depending on the last map rendering mode selected or the mode selected on the command itself.</p>
  </dd>
  <dt>
    <p><strong>Map level</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map level</strong> command sets the contour level to be used in creating subsequent representations of generated or loaded maps.  If the keyword MEAN in used the level is relative to the mean of the map data.  Otherwise the level is absolute.</p><p>In general, a lower level results in a map containing more of the displayed volume, while a higher level results in a map containing less of the displayed volume.</p>
  </dd>
  <dt>
    <p><strong>Map load</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map load</strong> command loads a map file into RasMol.  The valid formats are CCP4 map format and imgCIF format.</p><p>If no specific map was given by the map selector, the new map is given the next available map number.</p><p>If a specific map was given by the map selector, the new map replaces that map.  If more than one map was given by the map selector, the new map replaces the lowest numbered of the selected maps.  In any case the new map becomes the currently selected map.</p><p>The map is displayed as dots, mesh or a surface depending on the last map rendering mode selected.</p>
  </dd>
  <dt>
    <p><strong>Map mask</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map mask</strong> command specifies a mask to be used to limit the display space to be used for making representations of other maps or removes an earlier mask specification.</p><p>The 'selected' option indicates that the mask is to be created from the currently selected atoms.  The '&lt;number&gt;' option indicates that the mask is to be copied from the map of the number specified. The 'none' option removes the previously specified mask, if any.</p><p>The map selector specifies the map or maps to which the specified mask will the applied.  For example, 'map next mask selected' specifies that the currently selected atoms are to be used to generate a mask to be applied to any maps created by subsequent 'map load' or 'map generate' commands.</p><p>Any map may be used as a mask.  The portions of the mask map greater than than or equal to the average value of the mask map allow the values of the map being masked to be used as given.  The portions of the mask map lower than the average value of the mask map cause the values of the map being masked to be treated as if they were equal to the lowest data value of the map being masked.</p>
  </dd>
  <dt>
    <p><strong>Map resolution</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map resolution</strong> command specifies the resolution in RasMol units or, if a number containing a decimal point is given, the resolution in Angstroms to be used in generating and in representing maps.</p><p>The resolution is used at the map spacing for representations of maps, indicating the separation between contour levels (see the <strong>map spacing</strong> command) and to infer the map spread to be used in generated maps from selected atoms (see the <strong>map spread</strong> command).  The map spread is set to two thirds of the specified resolution.</p>
  </dd>
  <dt>
    <p><strong>Map restrict</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map restrict</strong> command selects particular maps to make them active for subsequent map commands.  This is similar to the <strong>map select</strong> command, but does disables the display of the maps that were not selected.</p>
  </dd>
  <dt>
    <p><strong>Map save</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map save</strong> command saves an imgCIF map file.</p><p>If no specific map was given by the map selector, the currently selected maps and their masks are written to the file, one map and mask pair per data block.</p>
  </dd>
  <dt>
    <p><strong>Map scale</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map scale</strong> command selects the scaling of pseudo-Gaussians in the <strong>map generate</strong> commands. In the default of map scale true, each Gaussian has a height proportional element type of the atom. If map scale false has been executed, each Gaussian is scaled so that the Gaussian contour level 1 is at the van der Waals radius. In either case a standard deviation determined by the most recently specified spread or resolution is used.</p>
  </dd>
  <dt>
    <p><strong>Map select</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map select</strong> command selects particular maps to make them active for subsequent map commands.  This is similar to the <strong>map restrict</strong> command, but does not disable the display of the maps that were not selected.</p><p>If the optional <strong>atom</strong> parameter is given, the command selects the atoms with centres closest to the map points.  The radius of the search may be specified by the parameter <strong>search_radius.</strong> The default is to look for atoms within 4 Angstroms plus the probe radius. If the optional <strong>within</strong> parameter is given, the new selection is taken from within the currently selected atoms.  If the options <strong>add</strong> parameter is given, the new selection is added to the currently selected atoms. The default is to search within all atoms.</p>
  </dd>
  <dt>
    <p><strong>Map show</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map show</strong> command causes information about the maps specified by the map selector to be written to the command window.</p>
  </dd>
  <dt>
    <p><strong>Map spacing</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map spacing</strong> command specifies the spacing to be used between contour lines in creating representations of maps.  The spacing is typically  given in Angstroms with a decimal point, but may also be specified in RasMol units (250ths of an Angstom) as an integer.  For maps loaded in grid coordinates that spacing is parallel to the cell edges.  The default spacing is one half Angstrom.</p>
  </dd>
  <dt>
    <p><strong>Map spread</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map spread</strong> command specifies the reciprocal of the number of standard deviations per radius to be used in generating maps as sums of Gaussians centered on atomic positions.  The default spread is one two thirds (i.e. each radius covers 1.5 standard deviations).</p><p>If the spread has been set to zero, the spread for each atom is determined from the van der Waals radius and the probe atom radius to simulate the effect of a Lee-Richards surface.</p>
  </dd>
  <dt>
    <p><strong>Map zap</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>map zap</strong> command removes the data and representations of the maps specified by the map selector.  The map numbers of maps that have not been removed are not changed.</p>
  </dd>
  <dt>
    <p><strong>Molecule</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>molecule</strong> command selects one of up to 5 previously loaded molecules for active manipulation.  While all the molcules are displayed and may be rotated collectively (see the <strong>rotate all</strong> command), only one molecule at a time time is active for manipulation by the commands which control the details of rendering.</p>
  </dd>
  <dt>
    <p><strong>Monitor</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>monitor</strong> command allows the display of distance monitors. A distance monitor is a dashed (dotted) line between an arbitrary pair of atoms, optionally labelled by the distance between them. The RasMol command <strong>monitor &lt;number&gt; &lt;number&gt;</strong> adds such a distance monitor between the two atoms specified by the atom serial numbers given as parameters</p><p>Distance monitors are turned off with the command <strong>monitors off.</strong> By default, monitors display the distance between its two end points as a label at the centre of the monitor. These distance labels may be turned off with the command <strong>set monitors off,</strong> and re-enabled with the command <strong>set monitors on.</strong> Like most other representations, the colour of a monitor is taken from the colour of its end points unless specified by the <strong>colour monitors</strong> command.</p><p>Distance monitors may also be added to a molecule interactively with the mouse, using the <strong>set picking monitor</strong> command. Clicking on an atom results in its being identified on the rasmol command line. In addition every atom picked increments a modulo counter such that, in monitor mode, every second atom displays the distance between this atom and the previous one. The shift key may be used to form distance monitors between a fixed atom and several consecutive positions. A distance monitor may also be removed (toggled) by selecting the appropriate pair of atom end points a second time.</p>
  </dd>
  <dt>
    <p><strong>Notoggle</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>NoToggle</strong> command enables or disables the use of the toggle ability that is used by some of the other RasMol commands. When no boolean value is specified, NoToggle mode is ENABLED. When NoToggle mode is ENABLED, all toggle functionality is DISABLED. To turn it off, one must explicitly set <strong>notoggle off.</strong></p><p>Some commands which use the toggle feature are: <strong>ColourMode.</strong> More functions that utilize this capability may be added at a later date.</p>
  </dd>
  <dt>
    <p><strong>Pause</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>pause</strong> command is used in script files to stop the script file for local manipulation by a mouse, until any key is pushed to restart the script file. <strong>Wait</strong> is synonymous with <strong>pause.</strong> This command may be executed in RasMol script files to suspend the sequential execution of commands and allow the user to examine the current image.   When RasMol executes a <strong>pause</strong> command in a script file, it suspends  execution of the rest of the file, refreshes the image on the screen and allows the manipulation of the image using the mouse and scroll  bars, or resizing of the graphics window.  Once a key is pressed, control returns to the script file at the line following the <strong>pause</strong> command.  While a script is suspended the molecule may be rotated, translated, scaled, slabbed and picked as usual, but all menu commands are disabled.</p>
  </dd>
  <dt>
    <p><strong>Play</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>play</strong> command specifies the recording medium from which to play back a movie. The playback frame start time is given in seconds to millisecond precision. Since we are working on computers, the medium is specified as a set of files, each marked with the playback frame start time in milliseconds as part of the name. The place in the name at which to look for the playback frame start time in milliseconds is marked by the characters "ssssss" with an appropriate number of digits.  RasMol accepts either upper or lower case s's or decimal digits to mark the place for the time.  The play off and play eject commands effectively remove the specified medium from use. If no medium is specified, play off suspends playing and play on resumes playing. Normally play starts immediately and runs to the end of the medium. However, if play off and/or or some combination of play from and play until is entered before <strong>play type medium,</strong> those settings will be used.</p><p>As of release 2.7.5, RasMol support play from scripts and data files.</p>
  </dd>
  <dt>
    <p><strong>Print</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>print</strong> command sends the currently displayed image to the local default printer using the operating system's native printer driver. Note: this command is not yet supported under UNIX or VMS. It is intended to take advantage of Microsoft Windows and Apple Macintosh printer drivers. For example, this allows images to be printed directly on a dot matrix printer.</p><p>When using RasMol on a UNIX or VMS system this functionality may be achieved by either generating a PostScript file using the RasMol <strong>write ps</strong> or <strong>write vectps</strong> commands and printing that or generating a raster image file and using a utility to dump that to the local printer.</p>
  </dd>
  <dt>
    <p><strong>Quit</strong></p>
  </dt>
  <dd>
    <p>Exit from the RasMol program. The RasMol commands <strong>exit</strong> and <strong>quit</strong> are synonymous, except within nested scripts.  In that case, <strong>exit</strong> terminates only the current level, while <strong>quit</strong> terminates all nested levels of scripts.</p>
  </dd>
  <dt>
    <p><strong>Record</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>record</strong> command specifies the recording medium to hold the movie. Since we are working on computers, the medium is specified as a template for a set of files, each marked with the playback frame start time in milliseconds (rather than as seconds to avoid embedding a decimal point) as part of the name. The place in the name to be replaced with the playback frame start time in milliseconds is marked by the characters "ssssss" with an appropriate number of digits. RasMol accepts either upper or lower case s's or decimal digits to mark the place for the time.  The record off commands remove the specified medium from use. If no medium is specified, record off suspends recording and record on resumes recording with the next available time on the same medium. The screen is the default medium and is, by default, on. Writing to disk must be explicitly specified so that the disk does not get filled up unintentionally. The type of a recording medium may be an image type such as gif, pict or png to record the actual screen images or script to record the RasMol commands used to generate the frames.</p><p>Normally recording starts at playback frame start time 0 seconds. A non-zero starting time in seconds can be specified with the <strong>record from</strong> command as in <strong>record from 25</strong> or <strong>record from 37.25</strong> to help in organizing scenes of movies to be assembled later in an appropriate order. The <strong>record until</strong> command allows an upper limit to be set on recording time in seconds. The default is to have no limit. Issuing the commands</p><p><strong>record from 600</strong></p><p><strong>record until 1800</strong></p><p>would result in a 20 minute movie segment intended to start 10 minutes into a longer movie. These commands allow control over rewriting selected time segments.</p>
  </dd>
  <dt>
    <p><strong>Refresh</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>refresh</strong> command redraws the current image.  This is useful in scripts to ensure application of a complex list of parameter changes.</p>
  </dd>
  <dt>
    <p><strong>Renumber</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>renumber</strong> command sequentially numbers the residues in a macromolecular chain. The optional parameter specifies the value of the first residue in the sequence. By default, this value is one. For proteins, each amino acid is numbered consecutively from the N terminus to the C terminus. For nucleic acids, each base is numbered from the 5' terminus to the 3' terminus. All chains in the current database are renumbered and gaps in the original sequence are ignored. The starting value for numbering may be negative.</p>
  </dd>
  <dt>
    <p><strong>Reset</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>reset</strong> command restores the original viewing transformation and centre of rotation. The scale is set to its default value, <strong>zoom 100,</strong> the centre of rotation is set to the geometric centre of the currently loaded molecule, <strong>centre all,</strong> this centre is translated to the middle of the screen and the viewpoint set to the default orientation.</p><p>This command should not be mistaken for the RasMol <strong>zap</strong> command which deletes the currently stored molecule, returning the program to its initial state.</p>
  </dd>
  <dt>
    <p><strong>Restrict</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>restrict</strong> command both defines the currently selected region of the molecule and disables the representation of (most of) those parts of the molecule no longer selected.  All subsequent RasMol commands that modify a molecule's colour or representation affect only the currently selected region. The parameter of a <strong>restrict</strong> command is a RasMol atom expression that is evaluated for every atom of the current molecule. This command is very similar to the RasMol <strong>select</strong> command, except <strong>restrict</strong> disables the <strong>wireframe,</strong> <strong>spacefill</strong> and <strong>backbone</strong> representations in the non-selected region.</p><p>Type "help expression" for more information on RasMol atom expressions or see section <strong>Atom Expressions.</strong></p>
  </dd>
  <dt>
    <p><strong>Ribbons</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>ribbons</strong> command displays the currently loaded protein or nucleic acid as a smooth solid "ribbon" surface passing along the backbone of the protein. The ribbon is drawn between each amino acid whose alpha carbon is currently selected. The colour of the ribbon is changed by the RasMol <strong>colour ribbon</strong> command. If the current ribbon colour is <strong>none</strong> (the default), the colour is taken from the alpha carbon at each position along its length.</p><p>The width of the ribbon at each position is determined by the optional parameter in the usual RasMol units. By default the width of the ribbon is taken from the secondary structure of the protein or a constant value of 720 (2.88 Angstroms) for nucleic acids. The default width of protein alpha helices and beta sheets is 380 (1.52 Angstroms) and 100 (0.4 Angstroms) for turns and random coil. The secondary structure assignment is either from the PDB file or calculated using the DSSP algorithm as used by the <strong>structure</strong> command. This command is similar to the RasMol command <strong>strands</strong> which renders the biomolecular ribbon as parallel depth-cued curves.</p>
  </dd>
  <dt>
    <p><strong>Rotate</strong></p>
  </dt>
  <dd>
    <p>Rotate the molecule about the specified axis. Permitted values for the axis parameter are "x", "y", "z" and "bond". The integer parameter states the angle in degrees for the structure to be rotated. For the X and Y axes, positive values move the closest point up and right, and negative values move it down and left, respectively. For the Z axis, a positive rotation acts clockwise and a negative angle anti-clockwise.</p><p>Alternatively, this command may be used to specify which rotations the mouse or dials will control.  If <strong>rotate bond true</strong> is selected, the horizontal scroll bar will control rotation around the axis selected by the <strong>bond src dst pick</strong> command.  If <strong>rotate all true</strong> is selected, and multiple molecules have been loaded, then all molecules will rotate together.  In all other cases, the mouseand dials control the the rotation of the molecule selected by the <strong>molecule n</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Russian</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>Russian</strong> command sets the menus and messages to the Russian versions.</p><p>This command may not work correctly unless appropriate fonts have been installed.  The commands <strong>Bulgarian,</strong> <strong>Chinese,</strong> <strong>English,</strong> <strong>French,</strong> <strong>Italian,</strong> <strong>Russian</strong> and <strong>Spanish</strong> may be used to select Bulgarian, Chinese, English, French, Italian, Japanese, Russian and Spanish menus and messages if the appropriate fonts have been installed.</p>
  </dd>
  <dt>
    <p><strong>Save</strong></p>
  </dt>
  <dd>
    <p>Save the currently selected set of atoms in a Protein Data Bank (PDB), MDL, Alchemy(tm) or XYZ format file. The distinction between this command and the RasMol <strong>write</strong> command has been dropped. The only difference is that without a format specifier the <strong>save</strong> command generates a <strong>PDB</strong> file and the <strong>write</strong> command generates a <strong>GIF</strong> image.</p>
  </dd>
  <dt>
    <p><strong>Script</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>script</strong> command reads a set of RasMol commands sequentially from a text file and executes them. This allows sequences of commonly used commands to be stored and performed by single command. A RasMol script file may contain a further script command up to a maximum "depth" of 10, allowing complicated sequences of actions to be executed. RasMol ignores all characters after the first '#' character on each line allowing the scripts to be annotated. Script files are often also annotated using the RasMol <strong>echo</strong> command.</p><p>The most common way to generate a RasMol script file is to use the <strong>write script</strong> or <strong>write rasmol</strong> commands to output the sequence of commands that are needed to regenerate the current view, representation and colouring of the currently displayed molecule.</p><p>The RasMol command <strong>source</strong> is synonymous with the <strong>script</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Select</strong></p>
  </dt>
  <dd>
    <p>Define the currently selected region of the molecule. All subsequent RasMol commands that manipulate a molecule or modify its colour or representation only affect the currently selected region. The parameter of a <strong>select</strong> command is a RasMol expression that is evaluated for every atom of the current molecule. The currently selected (active) region of the molecule are those atoms that cause the expression to evaluate true. To select the whole molecule use the RasMol command <strong>select all.</strong> The behaviour of the <strong>select</strong> command without any parameters is determined by the RasMol <strong>hetero</strong> and <strong>hydrogen</strong> parameters.</p><p>Type "help expression" for more information on RasMol atom expressions or see section <strong>Atom Expressions.</strong></p>
  </dd>
  <dt>
    <p><strong>Set</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set</strong> command allows the user to alter various internal program parameters such as those controlling rendering options. Each parameter has its own set or permissible parameter options. Typically, omitting the parameter option resets that parameter to its default value. A list of valid parameter names is given below.</p>
  </dd>
  <dt>
    <p><strong>Show</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>show</strong> command display details of the status of the currently loaded molecule. The command <strong>show information</strong> lists the molecule's name, classification, PDB code and the number of atoms, chains, groups it contains. If hydrogen bonding, disulphide bridges or secondary structure have been determined, the number of hbonds, ssbonds, helices, ladders and turns are also displayed, respectively. The command <strong>show centre</strong> shows any non-zero centering values selected by the <strong>centre [CenX, CenY, CenZ]</strong> command. The command <strong>show phipsi</strong> shows the phi and psi angles of the currently selected residues and the omega angles of cis peptide bonds. The command <strong>show RamPrint</strong> (or 'show RPP' or 'show RamachandranPrinterPlot') shows a simple Ramachandran printer plot in the style of Frances Bernstein's fisipl program.  The command <strong>show rotation</strong> (or 'show rot' or 'show 'rotate') shows the currently selected values of z, y, x and bond rotations, if any. The command <strong>show selected</strong> (or 'show selected group' or 'show selected chain' or  'show selected atom' ) shows the groups (default), chains or atoms of the current selection. The command <strong>show sequence</strong> lists the residues that comprise each chain of the molecule.  The command <strong>show symmetry</strong> shows the space group and unit cell of the molecule. The command <strong>show translation</strong> shows any non-zero translation values selected by the <strong>translate &lt;axis&gt; &lt;value&gt;</strong> command. The command <strong>show zoom</strong> shows any non-zero zoom value selected by the <strong>zoom &lt;value&gt;</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Slab</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>slab</strong> command enables, disables or positions the z-clipping plane of the molecule. The program only draws those portions of the molecule that are further from the viewer than the slabbing plane. Integer values range from zero at the very back of the molecule to 100 which is completely in front of the molecule. Intermediate values determine the percentage of the molecule to be drawn.</p><p>This command interacts with the <strong>depth &lt;value&gt;</strong> command, which clips to the rear of a given z-clipping plane.</p>
  </dd>
  <dt>
    <p><strong>Spacefill</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>spacefill</strong> command is used to represent all of the currently selected atoms as solid spheres. This command is used to produce both union-of-spheres and ball-and-stick models of a molecule. The command, <strong>spacefill true,</strong> the default, represents each atom as a sphere of van der Waals radius. The command <strong>spacefill off</strong> turns off the representation of the selected atom as spheres. A sphere radius may be specified as an integer in RasMol units (1/250th Angstrom) or a value containing a decimal point. A value of 500 (2.0 Angstroms) or greater results in a "Parameter value too large" error.</p><p>The <strong>temperature</strong> option sets the radius of each sphere to the value stored in its temperature field. Zero or negative values have no effect and values greater than 2.0 are truncated to 2.0.  The <strong>user</strong> option allows the radius of each sphere to be specified by additional lines in the molecule's PDB file using Raster 3D's COLOUR record extension.</p><p>The RasMol command <strong>cpk</strong> is synonymous with the <strong>spacefill</strong> command.</p><p>The RasMol command <strong>cpknew</strong> is synonymous with the <strong>spacefill</strong> command, except that a slightly different set of colours is used.</p>
  </dd>
  <dt>
    <p><strong>Spanish</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>Spanish</strong> command sets the menus and messages to the Spanish versions.</p><p>This command may not work correctly unless appropriate fonts have been installed.  The commands <strong>Bulgarian,</strong> <strong>Chinese,</strong> <strong>English,</strong> <strong>French,</strong> <strong>Italian,</strong> <strong>Russian</strong> and <strong>Spanish</strong> may be used to select Bulgarian, Chinese, English, French, Italian, Japanese, Russian and Spanish menus and messages if the appropriate fonts have been installed.</p>
  </dd>
  <dt>
    <p><strong>SSBonds</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>ssbonds</strong> command is used to represent the disulphide bridges of the protein molecule as either dotted lines or cylinders between the connected cysteines. The first time that the <strong>ssbonds</strong> command is used, the program searches the structure of the protein to find half-cysteine pairs (cysteines whose sulphurs are within 3 Angstroms of each other) and reports the number of bridges to the user. The command <strong>ssbonds on</strong> displays the selected "bonds" as dotted lines, and the command <strong>ssbonds off</strong> disables the display of ssbonds in the currently selected area. Selection of disulphide bridges is identical to normal bonds, and may be adjusted using the RasMol <strong>set bondmode</strong> command. The colour of disulphide bonds may be changed using the <strong>colour ssbonds</strong> command. By default, each disulphide bond has the colours of its connected atoms.</p><p>By default disulphide bonds are drawn between the sulphur atoms within the cysteine groups. By using the <strong>set ssbonds</strong> command the position of the cysteine's alpha carbons may be used instead.</p>
  </dd>
  <dt>
    <p><strong>Star</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>star</strong> command is used to represent all of the currently selected atoms as stars (six strokes, one each in the x, -x, y, -y, z and -z directions). The commands <strong>select not bonded</strong> followed by <strong>star 75</strong> are useful to mark unbonded atoms in a <strong>wireframe</strong> display with less overhead than provided by <strong>spacefill 75.</strong> This can be done automatically for all subsequent wireframe displays with the command <strong>set bondmode not bonded.</strong></p><p>The command <strong>star true,</strong> the default, represents each atom as a star with strokes length equal to van der Waals radius. The command <strong>star off</strong> turns off the representation of the selected atom as stars. A star stroke length may be specified as an integer in RasMol units (1/250th Angstrom) or a value containing a decimal point. A value of 500 (2.0 Angstroms) or greater results in a "Parameter value too large" error.</p><p>The <strong>temperature</strong> option sets the stroke length of each star to the value stored in its temperature field. Zero or negative values have no effect and values greater than 2.0 are truncated to 2.0.  The <strong>user</strong> option allows the stroke length of each star to be specified by additional lines in the molecule's PDB file using Raster 3D's COLOUR record extension.</p><p>The RasMol <strong>spacefill</strong> command can be used for more artistic rendering of atoms as spheres.</p>
  </dd>
  <dt>
    <p><strong>Stereo</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>stereo</strong> command provides side-by-side stereo display of images. Stereo viewing of a molecule may be turned on (and off) either by selecting <strong>Stereo</strong> from the <strong>Options</strong> menu, or by typing the commands <strong>stereo on</strong> or <strong>stereo off.</strong></p><p>Starting with RasMol version 2.7.2.1, the <strong>Stereo</strong> menu selection and the command <strong>stereo</strong> without arguments cycle from the initial state of <strong>stereo off</strong> to <strong>stereo on</strong> in cross-eyed mode to <strong>stereo on</strong> in wall-eyed mode and then back to <strong>stereo off.</strong></p><p>The separation angle between the two views may be adjusted with the <strong>set stereo [-] &lt;number&gt;</strong> command, where positive values result in crossed eye viewing and negative values in relaxed (wall-eyed) viewing. The inclusion of <strong>[-] &lt;number&gt;</strong> in the <strong>stereo</strong> command, as for example in <strong>stereo 3</strong> or <strong>stereo -5,</strong> also controls angle and direction.</p><p>The stereo command is only partially implemented. When stereo is turned on, the image is not properly recentred. (This can be done with a <strong>translate x -&lt;number&gt;</strong>  command.) It is not supported in vector PostScript output files, is not saved by the <strong>write script</strong> command, and in general is not yet properly interfaced with several other features of the program.</p>
  </dd>
  <dt>
    <p><strong>Strands</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>strands</strong> command displays the currently loaded protein or nucleic acid as a smooth "ribbon" of depth-cued curves passing along the backbone of the protein. The ribbon is composed of a number of strands that run parallel to one another along the peptide plane of each residue. The ribbon is drawn between each amino acid whose alpha carbon is currently selected. The colour of the ribbon is changed by the RasMol <strong>colour ribbon</strong> command. If the current ribbon colour is <strong>none</strong> (the default), the colour is taken from the alpha carbon at each position along its length. The central and outermost strands may be coloured independently using the <strong>colour ribbon1</strong> and <strong>colour ribbon2</strong> commands, respectively. The number of strands in the ribbon may be altered using the RasMol <strong>set strands</strong> command.</p><p>The width of the ribbon at each position is determined by the optional parameter in the usual RasMol units. By default the width of the ribbon is taken from the secondary structure of the protein or a constant value of 720 for nucleic acids (which produces a ribbon 2.88 Angstroms wide). The default width of protein alpha helices and beta sheets is 380 (1.52 Angstroms) and 100 (0.4 Angstroms) for turns and random coil. The secondary structure assignment is either from the PDB file or calculated using the DSSP algorithm as used by the <strong>structure</strong> command. This command is similar to the RasMol command <strong>ribbons</strong> which renders the biomolecular ribbon as a smooth shaded surface.</p>
  </dd>
  <dt>
    <p><strong>Structure</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>structure</strong> command calculates secondary structure assignments for the currently loaded protein. If the original PDB file contained structural assignment records (HELIX, SHEET and TURN) these are discarded. Initially, the hydrogen bonds of the current molecule are found, if this hasn't been done already. The secondary structure is then determined using Kabsch and Sander's DSSP algorithm. Once finished the program reports the number of helices, strands and turns found.</p>
  </dd>
  <dt>
    <p><strong>Surface</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>surface</strong> command renders a Lee-Richards molecular surface resulting from rolling a probe atom on the selected atoms. The value given specifies the radius of the probe. If given in the first form, the evolute of the surface of the probe is shown (the solvent excluded surface). If given in the second form, the envelope of the positions of the center of the probe is shown (the solvent accessible surface).</p>
  </dd>
  <dt>
    <p><strong>Trace</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>trace</strong> command displays a smooth spline between  consecutive alpha carbon positions.  This spline does not pass exactly through the alpha carbon position of each residue, but  follows the same path as <strong>ribbons,</strong> <strong>strands</strong> and <strong>cartoons.</strong> Note that residues may be displayed as <strong>ribbons,</strong> <strong>strands,</strong> <strong>cartoons</strong> or as a <strong>trace.</strong> Enabling one of these representations disables the others. However, a residue may be displayed simultaneously as backbone and as one of the above representations.  This may change in future versions of RasMol.  Prior to version 2.6, <strong>trace</strong> was synonymous with <strong>backbone.</strong></p><p><strong>Trace temperature</strong> displays the backbone as a wider cylinder at high temperature factors and thinner at lower.  This representation is useful to X-ray crystallographers and NMR spectroscopists.</p>
  </dd>
  <dt>
    <p><strong>Translate</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>translate</strong> command moves the position of the centre of the molecule on the screen. The axis parameter specifies along which axis the molecule is to be moved and the integer parameter specifies the absolute position of the molecule centre from the middle of the screen. Permitted values for the axis parameter are "x", "y" and "z". Displacement values must be between -100 and 100 which correspond to moving the current molecule just off the screen. A positive "x" displacement moves the molecule to the right, and a positive "y" displacement moves the molecule down the screen. The pair of commands <strong>translate x 0</strong> and <strong>translate y 0</strong> centres the molecule on the screen.</p>
  </dd>
  <dt>
    <p><strong>UnBond</strong></p>
  </dt>
  <dd>
    <p>The RasMol command <strong>unbond &lt;number&gt; &lt;number&gt;</strong> removes the designated bond from the drawing.</p><p>The command <strong>unbond</strong> without arguments removes a bond previously picked by the <strong>bond &lt;number&gt; &lt;number&gt; pick</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Wireframe</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>wireframe</strong> command represents each bond within the selected region of the molecule as a cylinder, a line or a depth-cued vector. The display of bonds as depth-cued vectors (drawn darker the further away from the viewer) is turned on by the command <strong>wireframe</strong> or <strong>wireframe on.</strong> The selected bonds are displayed as cylinders by specifying a radius either as an integer in RasMol units or containing a decimal point as a value in Angstroms.  A parameter value of 500 (2.0 Angstroms) or above results in an "Parameter value too large" error. Bonds may be coloured using the <strong>colour bonds</strong> command.</p><p>If the selected bonds involved atoms of alternate conformers then the bonds are narrowed in the middle to a radius of .8 of the specified radius (or to the radius specified as the optional second parameter).</p><p>Non-bonded atoms, which could become invisible in an ordinary <strong>wireframe</strong> display can be marked by a preceding <strong>set bondmode not bonded</strong> command.  If nearly co-linear bonds to atoms cause them to be difficult to see in a wireframe display, the <strong>set bondmode all</strong> command will add markers for <strong>all</strong> atoms in subsequent <strong>wireframe</strong> command executions.</p>
  </dd>
  <dt>
    <p><strong>Write</strong></p>
  </dt>
  <dd>
    <p>Write the current image to a file in a standard format. Currently supported image file formats include <strong>bmp</strong> (Microsoft bitmap) and <strong>gif</strong> (Compuserve GIF), <strong>iris</strong> (IRIS RGB), <strong>ppm</strong> (Portable Pixmap), <strong>ras</strong> (Sun rasterfile), <strong>ps</strong> and <strong>epsf</strong> (Encapsulated PostScript), <strong>monops</strong> (Monochrome Encapsulated PostScript), <strong>pict</strong> (Apple PICT), <strong>vectps</strong> (Vector Postscript).  The <strong>write</strong> command may also be used to generate command scripts for other graphics programs. The format <strong>script</strong> writes out a file containing the RasMol <strong>script</strong> commands to reproduce the current image. The format <strong>molscript</strong> writes out the commands required to render the current view of the molecule as ribbons in Per Kraulis' Molscript program and the format <strong>kinemage</strong> the commands for David Richardson's program Mage.  The following formats are useful for further processing: <strong>povray</strong> (POVRay 2), <strong>povray3</strong> (POVRay 3 -- under development), <strong>vrml</strong> (VRML file). Finally, several formats are provided to provide phi-psi data for listing or for <strong>phipsi</strong> (phi-psi data as an annotated list with cis omegas), <strong>ramachan</strong> and <strong>RDF</strong> and <strong>RamachandranDataFile</strong> (phi-psi data as columns of numbers for gnuplot), <strong>RPP</strong> and <strong>RamachandranPrinterPlot</strong> (phi-psi data as a printer plot).</p><p>The distinction between this command and the RasMol <strong>save</strong> command has been dropped. The only difference is that without a format specifier the <strong>save</strong> command generates a <strong>PDB</strong> file and the <strong>write</strong> command generates a <strong>GIF</strong> image.</p>
  </dd>
  <dt>
    <p><strong>Zap</strong></p>
  </dt>
  <dd>
    <p>Deletes the contents of the current database and resets parameter variables to their initial default state.</p>
  </dd>
  <dt>
    <p><strong>Zoom</strong></p>
  </dt>
  <dd>
    <p>Change the magnification of the currently displayed image. Boolean parameters either magnify or reset the scale of current molecule. An integer parameter specifies the desired magnification as a percentage of the default scale. The minimum parameter value is 10; the maximum parameter value is dependent upon the size of the molecule being displayed. For medium sized proteins this is about 500.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SET PARAMETERS</h2>
        <div class="sectioncontent">
<p>RasMol has a number of internal parameters that may be modified using the <strong>set</strong> command. These parameters control a number of program options such as rendering options and mouse button mappings.</p>
<pre>
    picking         play.fps        radius          record.aps
</pre>

<dl class='dl-vertical'>
  <dt>
    <p><strong>Set Ambient</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>ambient</strong> parameter is used to control the amount of ambient (or surrounding) light in the scene. The <strong>ambient</strong> value must be between 0 and 100. It controls the percentage intensity of the darkest shade of an object. For a solid object, this is the intensity of surfaces facing away from the light source or in shadow. For depth-cued objects this is the intensity of objects furthest from the viewer.</p><p>This parameter is commonly used to correct for monitors with different "gamma values" (brightness), to change how light or dark a hardcopy image appears when printed or to alter the feeling of depth for wireframe or ribbon representations.</p>
  </dd>
  <dt>
    <p><strong>Set Axes</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>axes</strong> parameter controls the display of orthogonal coordinate axes on the current display. The coordinate axes are those used in the molecule data file, and the origin is the centre of the molecule's bounding box. The <strong>set axes</strong> command is similar to the commands <strong>set boundbox</strong> and <strong>set unitcell</strong> that display the bounding box and the crystallographic unit cell, respectively.</p>
  </dd>
  <dt>
    <p><strong>Set Backfade</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>backfade</strong> parameter is used to control backfade to the specified background colour, rather than black.  This is controlled by the commands <strong>set backfade on</strong> and <strong>set backfade off.</strong> For example, this may be used to generate depth-cued images that fade to white, rather than black.</p>
  </dd>
  <dt>
    <p><strong>Set Background</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>background</strong> parameter is used to set the colour of the "canvas" background. The colour may be given as either a colour name or a comma separated triple of Red, Green, Blue (RGB) components enclosed in square brackets. Typing the command <strong>help colours</strong> will give a list of the predefined colour names recognised by RasMol. When running under X Windows, RasMol also recognises colours in the X server's colour name database.</p><p>The command <strong>set background</strong> is synonymous with the RasMol command <strong>background.</strong></p>
  </dd>
  <dt>
    <p><strong>Set BondMode</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set bondmode</strong> command controls the mechanism used to select individual bonds and modifies the display of bonded and non-bonded atoms by subsequent <strong>wireframe</strong> commands.</p><p>When using the <strong>select</strong> and <strong>restrict</strong> commands, a given bond will be selected if i) the bondmode is <strong>or</strong> and either of the connected atoms is selected, or ii) the bondmode is <strong>and</strong> and both atoms connected by the bond are selected. Hence an individual bond may be uniquely identified by using the command <strong>set bondmode and</strong> and then uniquely selecting the atoms at both ends.</p><p>The <strong>bondmode [all | none | not bonded]</strong> commands add <strong>star 75</strong> or <strong>spacefill 75</strong> markers for the designated atoms to <strong>wireframe</strong> displays.  Stars are used when the specified wireframe radius is zero.</p>
  </dd>
  <dt>
    <p><strong>Set Bonds</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>bonds</strong> parameter is used to control display of double and triple bonds as multiple  lines or cylinders.  Currently bond orders are only read from  MDL Mol files, Sybyl Mol2 format files, Tripos Alchemy format files, CIF and mmCIF,  and suitable PDB files.  Double (and triple)  bonds are specified in some  PDB files by specifying a given bond twice  (and three times) in CONECT records.  The  command <strong>set bonds on</strong> enables the display of bond orders, and  the command <strong>set bonds off</strong> disables them.</p>
  </dd>
  <dt>
    <p><strong>Set BoundBox</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>boundbox</strong> parameter controls the display of the current molecule's bounding box on the display. The bounding box is orthogonal to the data file's original coordinate axes. The <strong>set boundbox</strong> command is similar to the commands <strong>set axes</strong> and <strong>set unitcell</strong> that display orthogonal coordinate axes and the bounding box, respectively.</p>
  </dd>
  <dt>
    <p><strong>Set Cartoon</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>cartoon</strong> parameter is used to control display of the cartoon version of the <strong>ribbons</strong> display.  By default, the C-termini of beta-sheets are displayed as arrow heads. This may be enabled and disabled using the <strong>set cartoons &lt;boolean&gt;</strong> command. The depth of the cartoon may be adjusted using the <strong>cartoons &lt;number&gt;</strong> command. The <strong>set cartoons</strong> command without any parameters returns these two options to  their default values.</p>
  </dd>
  <dt>
    <p><strong>Set CisAngle</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>cisangle</strong> parameter controls the cutoff angle for identifying cis peptide  bonds.  If no value is given, the cutoff is set to 90 degrees.</p>
  </dd>
  <dt>
    <p><strong>Set Display</strong></p>
  </dt>
  <dd>
    <p>This command controls the display mode within RasMol. By default, <strong>set display normal,</strong> RasMol displays the molecule in the representation specified by the user. The command <strong>set display selected</strong> changes the display mode such that the molecule is temporarily drawn so as to indicate currently selected portion of the molecule. The user specified colour scheme and representation remains unchanged. In this representation all selected atoms are shown in yellow and all non selected atoms are shown in blue. The colour of the background is also changed to a dark grey to indicate the change of display mode. This command is typically only used by external Graphical User Interfaces (GUIs).</p>
  </dd>
  <dt>
    <p><strong>Set FontSize</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set fontsize</strong> command is used to control the size of the characters that form atom labels. This value corresponds to the height of the displayed character in pixels. The maximum value of <strong>fontsize</strong> is 48 pixels, and the default value is 8 pixels high. Fixed or proportional spacing may be selected by appending the "FS" or "PS" modifiers, respectively.  The default is "FS". To display atom labels on the screen use the RasMol <strong>label</strong> command and to change the colour of displayed labels, use the <strong>colour labels</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Set FontStroke</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set fontstroke</strong> command is used to control the size of the stroke width of the characters that form atom labels.  This value is the radius in pixels of cylinders used to form the strokes.  The special value of "0" is the default used for the normal single pixel stroke width, which allows for rapid drawing and rotation of the image. Non-zero values are provided to allow for more artistic atom labels for publication at the expense of extra time in rendering the image.</p><p>When wider strokes are used, a larger font size is recommend, e.g. by using the RasMol <strong>set fontsize 24 PS</strong> command, followed by <strong>set fontstroke 2</strong></p><p>To display atom labels on the screen use the RasMol <strong>label</strong> command, and to change the colour of displayed labels use the <strong>colour labels</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Set HBonds</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>hbonds</strong> parameter determines whether hydrogen bonds are drawn between the donor and acceptor atoms of the hydrogen bond, <strong>set hbonds sidechain</strong> or between the alpha carbon atoms of the protein backbone and between the phosphorous atoms of the nucleic acid backbone, <strong>set hbonds backbone.</strong> The actual display of hydrogen bonds is controlled by the <strong>hbonds</strong> command. Drawing hydrogen bonds between protein alpha carbons or nucleic acid phosphorous atoms is useful when the rest of the molecule is shown in only a schematic representation such as <strong>backbone,</strong> <strong>ribbons</strong> or <strong>strands.</strong> This parameter is similar to the RasMol <strong>ssbonds</strong> parameter.</p>
  </dd>
  <dt>
    <p><strong>Set Hetero</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>hetero</strong> parameter is used to modify the 'default' behaviour of the RasMol <strong>select</strong> command, i.e. the behaviour of <strong>select</strong> without any parameters. When this value is <strong>false,</strong> the default <strong>select</strong> region does not include any heterogeneous atoms (refer to the predefined set <strong>hetero</strong> ). When this value is <strong>true,</strong> the default <strong>select</strong> region may contain hetero atoms. This parameter is similar to the RasMol <strong>hydrogen</strong> parameter which determines whether hydrogen atoms should be included in the default set. If both <strong>hetero</strong> and <strong>hydrogen</strong> are <strong>true,</strong> <strong>select</strong> without any parameters is equivalent to <strong>select all.</strong></p>
  </dd>
  <dt>
    <p><strong>Set HourGlass</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>hourglass</strong> parameter allows the user to enable and disable the use of the 'hour glass' cursor used by RasMol to indicate that the program is currently busy drawing the next frame. The command <strong>set hourglass on</strong> enables the indicator, whilst <strong>set hourglass off</strong> prevents RasMol from changing the cursor. This is useful when spinning the molecule, running a sequence of commands from a script file or using interprocess communication to execute complex sequences of commands. In these cases a 'flashing' cursor may be distracting.</p>
  </dd>
  <dt>
    <p><strong>Set Hydrogen</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>hydrogen</strong> parameter is used to modify the "default" behaviour of the RasMol <strong>select</strong> command, i.e. the behaviour of <strong>select</strong> without any parameters. When this value is <strong>false,</strong> the default <strong>select</strong> region does not include any hydrogen, deuterium or tritium atoms (refer to the predefined set <strong>hydrogen</strong> ). When this value is <strong>true,</strong> the default <strong>select</strong> region may contain hydrogen atoms. This parameter is similar to the RasMol <strong>hetero</strong> parameter which determines whether heterogeneous atoms should be included in the default set. If both <strong>hydrogen</strong> and <strong>hetero</strong> are <strong>true,</strong> <strong>select</strong> without any parameters is equivalent to <strong>select all.</strong></p>
  </dd>
  <dt>
    <p><strong>Set Kinemage</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set kinemage</strong> command controls the amount of detail stored in a Kinemage output file generated by the RasMol <strong>write kinemage</strong> command. The output kinemage files are intended to be displayed by David Richardson's Mage program. <strong>set kinemage false,</strong> the default, only stores the currently displayed representation in the generated output file. The command <strong>set kinemage true,</strong> generates a more complex Kinemage that contains both the wireframe and backbone representations as well as the coordinate axes, bounding box and crystal unit cell.</p>
  </dd>
  <dt>
    <p><strong>Set Menus</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set menus</strong> command enables the canvas window's menu buttons or menu bar. This command is typically only used by graphical user interfaces or to create as large an image as possible when using Microsoft Windows.</p>
  </dd>
  <dt>
    <p><strong>Set Monitor</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set monitor</strong> command enables <strong>monitors.</strong> The distance monitor labels may be turned off with the command <strong>set monitor off,</strong> and re-enabled with the command <strong>set monitor on.</strong></p>
  </dd>
  <dt>
    <p><strong>Set Mouse</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set mouse</strong> command sets the rotation, translation, scaling and zooming mouse bindings. The default value is <strong>rasmol</strong> which is suitable for two button mice (for three button mice the second and third buttons are synonymous); X-Y rotation is controlled by the first button, and X-Y translation by the second. Additional functions are controlled by holding a modifier key on the keyboard. [Shift] and the first button performs scaling, [shift] and the second button performs Z-rotation, and [control] and the first mouse button controls the clipping plane. The <strong>insight</strong> and <strong>quanta</strong> options provide the same mouse bindings as other packages for experienced users.</p>
  </dd>
  <dt>
    <p><strong>Set Picking</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set picking</strong> series of commands affects how a user may interact with a molecule displayed on the screen in RasMol.</p><p><strong>Enabling/Disabling Atom Identification Picking:</strong> Clicking on an atom with the mouse results in identification and the display of its residue name, residue number, atom name, atom serial number and chain in the command window. This behavior may be disabled with the command <strong>set picking none</strong> and restored with the command <strong>set picking ident.</strong> The command <strong>set picking coord</strong> adds the atomic coordinates of the atom to the display.</p><p>Disabling picking, by using <strong>set picking off</strong> is useful when executing the <strong>pause</strong> command in RasMol scripts as it prevents the display of spurious message on the command line while the script is suspended.</p><p><strong>Measuring Distances, Angles and Torsions:</strong> Interactive measurement of distances, angles and torsions is achieved using the commands: <strong>set picking distance,</strong> <strong>set picking monitor,</strong> <strong>set picking angle</strong> and <strong>set picking torsion,</strong> respectively. In these modes, clicking on an atom results in it being identified on the rasmol command line. In addition every atom picked increments a modulo counter such that in distance mode, every second atom displays the distance (or distance monitor) between this atom and the previous one. In angle mode, every third atom displays the angle between the previous three atoms and in torsion mode every fourth atom displays the torsion between the last four atoms. By holding down the shift key while picking an atom, this modulo counter is not incremented and allows, for example, the distances of consecutive atoms from a fixed atom to be displayed. See the <strong>monitor</strong> command for how to control the display of distance monitor lines and labels.</p><p><strong>Labelling Atoms with the Mouse:</strong> The mouse may also be used to toggle the display of an atom label on a given atom. The RasMol command <strong>set picking label</strong> removes a label from a picked atom if it already has one or displays a concise label at that atom position otherwise.</p><p><strong>Centring Rotation with the Mouse:</strong> A molecule may be centred on a specified atom position using the RasMol commands <strong>set picking centre</strong> or <strong>set picking center.</strong> In this mode, picking an atom causes all futher rotations to be about that point.</p><p><strong>Picking a Bond as a Rotation Axis:</strong> Any bond may be picked as an axis of rotation for the portion of the molecule beyond the second atom selected.  This feature should be used with caution, since, naturally, it changes the conformation of the molecule.  After executing <strong>set picking bond</strong> or using the equivalent "Pick Bond" in the "Settings" menu, a bond to be rotated is picked with the same sort of mouse clicks as are used for picking atoms for a distance measurement.  Normally this should be done where a bond exists, but if no bond exists, it will be added.  The bond cannot be used for rotation if it is part of a ring of any size.  All bonds selected for rotation are remembered so that they can be properly reported when writing a script, but only the most recently selected bond may be actively rotated.</p><p><strong>Enabling Atom/Group/Chain Selection Picking:</strong> Atoms, groups and chains may be selected (as if with the <strong>select</strong> command), with the <strong>set picking atom,</strong> <strong>set picking group,</strong> <strong>set picking chain</strong> commands.  For each of these commands, the shift key may be used to have a new selection added to the old, and the control key may be used to have a new selection deleted from the old. When the <strong>set picking atom</strong> command is given, the mouse can be used to pick or to drag a box around the atoms for which selection is desired.  When the <strong>set picking group</strong> command is given, picking any an atom will cause selection of all atoms which agree in residue number with the picked atom, even if in different chains. When the <strong>set picking chain</strong> command is given, picking any atom will cause selection of all atoms which agree in chain identifier with the picked atom.</p>
  </dd>
  <dt>
    <p><strong>Set Play</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set play.fps</strong> command gives the number of frames per second for playback by the <strong>play</strong> command (default 24 frames per second).</p><p>In the current release of RasMol, the play timing is not controlled by this parameter.</p>
  </dd>
  <dt>
    <p><strong>Set Radius</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set radius</strong> command is used to alter the behaviour of the RasMol <strong>dots</strong> command depending upon the value of the <strong>solvent</strong> parameter. When <strong>solvent</strong> is <strong>true,</strong> the <strong>radius</strong> parameter controls whether a true van der Waals' surface is generated by the <strong>dots</strong> command. If the value of <strong>radius</strong> is anything other than zero, that value is used as the radius of each atom instead of its true vdW value. When the value of <strong>solvent</strong> is <strong>true,</strong> this parameter determines the 'probe sphere' (solvent) radius. The parameter may be given as an integer in rasmol units or containing a decimal point in Angstroms. The default value of this parameter is determined by the value of <strong>solvent</strong> and changing <strong>solvent</strong> resets <strong>radius</strong> to its new default value.</p>
  </dd>
  <dt>
    <p><strong>Set Record</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set record.aps</strong> gives the maximum on-screen velocity in Angstroms per second in animating translations, rotations and zooms (default 10 A/second).</p><p>The RasMol <strong>set record.aps</strong> command gives number of frames per second for recording by the <strong>record</strong> command (default 24 frames per second).</p><p>The RasMol <strong>set record.dwell</strong> command sets the time in seconds to dwell on a change in appearance (default .5 sec).</p>
  </dd>
  <dt>
    <p><strong>Set ShadePower</strong></p>
  </dt>
  <dd>
    <p>The <strong>shadepower</strong> parameter (adopted from RasTop) determines the shade repartition (the contrast) used in rendering solid objects. This value between 0 and 100 adjusts shading on an object surface oriented along the direction to the light source.  Changing the shadepower parameter does not change the maximum or the minimum values of this shading, as does changing the <strong>ambient</strong> parameter.  A value of 100 concentrates the light on the top of spheres, giving a highly specular, glassy rendering (see the <strong>specpower</strong> parameter). A value of 0 distributes the light on the entire object.</p><p>This implementation of shadepower differs from the one in RasTop only in the choice of range (0 to 100 versus -20 to 20 in RasTop).</p>
  </dd>
  <dt>
    <p><strong>Set Shadow</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set shadow</strong> command enables and disables ray-tracing of the currently rendered image. Currently only the spacefilling representation is shadowed or can cast shadows. Enabling shadowing will automatically disable the Z-clipping (slabbing) plane using the command <strong>slab off.</strong> Ray-tracing typically takes about several seconds for a moderately sized protein. It is recommended that shadowing be normally disabled whilst the molecule is being transformed or manipulated, and only enabled once an appropriate viewpoint is selected, to provide a greater impression of depth.</p>
  </dd>
  <dt>
    <p><strong>Set SlabMode</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>slabmode</strong> parameter controls the rendering method of objects cut by the slabbing (z-clipping) plane. Valid slabmode parameters are "reject", "half", "hollow", "solid" and "section".</p>
  </dd>
  <dt>
    <p><strong>Set Solvent</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set solvent</strong> command is used to control the behaviour of the RasMol <strong>dots</strong> command. Depending upon the value of the <strong>solvent</strong> parameter, the <strong>dots</strong> command either generates a van der Waals' or a solvent accessible surface around the currently selected set of atoms. Changing this parameter automatically resets the value of the RasMol <strong>radius</strong> parameter. The command <strong>set solvent false,</strong> the default value, indicates that a van der Waals' surface should be generated and resets the value of <strong>radius</strong> to zero. The command <strong>set solvent true</strong> indicates that a 'Connolly' or 'Richards' solvent accessible surface should be drawn and sets the <strong>radius</strong> parameter, the solvent radius, to 1.2 Angstroms (or 300 RasMol units).</p>
  </dd>
  <dt>
    <p><strong>Set Specular</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set specular</strong> command enables and disables the display of specular highlights on solid objects drawn by RasMol. Specular highlights appear as white reflections of the light source on the surface of the object. The current RasMol implementation uses an approximation function to generate this highlight.</p><p>The specular highlights on the surfaces of solid objects may be altered by using the specular reflection coefficient, which is altered using the RasMol <strong>set specpower</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Set SpecPower</strong></p>
  </dt>
  <dd>
    <p>The <strong>specpower</strong> parameter determines the shininess of solid objects rendered by RasMol. This value between 0 and 100 adjusts the reflection coefficient used in specular highlight calculations. The specular highlights are enabled and disabled by the RasMol <strong>set specular</strong> command. Values around 20 or 30 produce plastic looking surfaces. High values represent more shiny surfaces such as metals, while lower values produce more diffuse/dull surfaces.</p>
  </dd>
  <dt>
    <p><strong>Set SSBonds</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>ssbonds</strong> parameter determines whether disulphide bridges are drawn between the sulphur atoms in the sidechain (the default) or between the alpha carbon atoms in the backbone of the cysteines residues. The actual display of disulphide bridges is controlled by the <strong>ssbonds</strong> command. Drawing disulphide bridges between alpha carbons is useful when the rest of the protein is shown in only a schematic representation such as <strong>backbone,</strong> <strong>ribbons</strong> or <strong>strands.</strong> This parameter is similar to the RasMol <strong>hbonds</strong> parameter.</p>
  </dd>
  <dt>
    <p><strong>Set Stereo</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>set stereo</strong> parameter controls the separation between the left and right images. Turning stereo on and off doesn't reposition the centre of the molecule.</p><p>Stereo viewing of a molecule may be turned on (and off) either by selecting <strong>Stereo</strong> from the <strong>Options</strong> menu, or by typing the commands <strong>stereo on</strong> or <strong>stereo off.</strong></p><p>The separation angle between the two views may be adjusted with the <strong>set stereo [-] &lt;number&gt;</strong> command, where positive values result in crossed eye viewing and negative values in relaxed (wall-eyed) viewing. Currently, stereo viewing is not supported in <strong>vector PostScript</strong> output files.</p>
  </dd>
  <dt>
    <p><strong>Set Strands</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>strands</strong> parameter controls the number of parallel strands that are displayed in the ribbon representations of proteins. The permissible values for this parameter are 1, 2, 3, 4, 5 and 9. The default value is 5. The number of strands is constant for all ribbons being displayed. However, the ribbon width (the separation between strands) may be controlled on a residue by residue basis using the RasMol <strong>ribbons</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Set Transparent</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>transparent</strong> parameter controls the writing of transparent GIFs by the <strong>write gif &lt;filename&gt;</strong> command.  This may be controlled by the <strong>set transparent on</strong> and <strong>set transparent off</strong> commands.</p>
  </dd>
  <dt>
    <p><strong>Set UnitCell</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>unitcell</strong> parameter controls the display of the crystallographic unit cell on the current display. The crystal cell is only enabled if the appropriate crystal symmetry information is contained in the PDB, CIF or mmCIF data file. The RasMol command <strong>show symmetry</strong> display details of the crystal's space group and unit cell axes. The <strong>set unitcell</strong> command is similar to the commands <strong>set axes</strong> and <strong>set boundbox</strong> that display orthogonal coordinate axes and the bounding box, respectively.</p>
  </dd>
  <dt>
    <p><strong>Set VectPS</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>vectps</strong> parameter is use to control the way in which the RasMol <strong>write</strong> command generates vector PostScript output files. The command <strong>set vectps on</strong> enables the use of black outlines around spheres and cylinder bonds producing "cartoon-like" high resolution output. However, the current implementation of RasMol incorrectly cartoons spheres that are intersected by more than one other sphere. Hence "ball and stick" models are rendered correctly but not large spacefilling spheres models. Cartoon outlines can be disabled, the default, by the command <strong>set vectps off.</strong></p>
  </dd>
  <dt>
    <p><strong>Set Write</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>write</strong> parameter controls the use of the <strong>save</strong> and <strong>write</strong> commands within scripts, but it may only be executed from the command line.  By default, this value is <strong>false,</strong> prohibiting the generation of files in any scripts executed at start-up (such as those launched from a WWW browser). However, animators may start up RasMol interactively: type <strong>set write on</strong> and then execute a script to generate each frame using the source command.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ATOM EXPRESSIONS</h2>
        <div class="sectioncontent">
<p>RasMol atom expressions uniquely identify an arbitrary group of atoms within a molecule. Atom expressions are composed of either primitive expressions, predefined sets, comparison operators, <strong>within</strong> expressions, or logical (boolean) combinations of the above expression types.</p><p>The logical operators allow complex queries to be constructed out of simpler ones using the standard boolean connectives <strong>and,</strong> <strong>or</strong> and <strong>not.</strong> These may be abbreviated by the symbols "&", "|" and "!", respectively. Parentheses (brackets) may be used to alter the precedence of the operators. For convenience, a comma may also be used for boolean disjunction.</p><p>The atom expression is evaluated for each atom, hence <strong>protein and backbone</strong> selects protein backbone atoms, not the protein and [nucleic] acid backbone atoms!</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>Primitive Expressions</strong></p>
  </dt>
  <dd>
    <p>RasMol primitive expressions are the fundamental building blocks of atom expressions. There are two types of primitive expression. The first type is used to identify a given residue number or range of residue numbers. A single residue is identified by its number (position in the sequence), and a range is specified by lower and upper bounds separated by a hyphen character. For example <strong>select 5,6,7,8</strong> is also <strong>select 5-8.</strong> Note that this selects the given residue numbers in all macromolecule chains.</p><p>The second type of primitive expression specifies a sequence of fields that must match for a given atom. The first part specifies a residue (or group of residues) and an optional second part specifies the atoms within those residues. The first part consists of a residue name, optionally followed by a residue number and/or chain identifier.</p><p>The second part consists of a period character followed by an atom name.  An atom name may be up to four alphabetic or numeric characters. An optional semicolon followed by an alternate conformation identifier may be appended.  An optional slash followed by a model number may also be appended.</p><p>An asterisk may be used as a wild card for a whole field and a question mark as a single character wildcard.</p>
  </dd>
  <dt>
    <p><strong>Comparison Operators</strong></p>
  </dt>
  <dd>
    <p>Parts of a molecule may also be distinguished using equality, inequality and ordering operators on their properties. The format of such comparison expression is a property name, followed by a comparison operator and then an integer value.</p><p>The atom properties that may be used in RasMol are <strong>atomno</strong> for the atom serial number, <strong>elemno</strong> for the atom's atomic number (element), <strong>resno</strong> for the residue number, <strong>radius</strong> for the spacefill radius in RasMol units (or zero if not represented as a sphere) and <strong>temperature</strong> for the PDB isotropic temperature value.</p><p>The equality operator is denoted either "=" or "==". The inequality operator as either "&lt;&gt;", "!=" or "/=". The ordering operators are "&lt;" for less than, "&lt;=" for less than or equal to, "&gt;" for greater than, and "&gt;" for greater than or equal to.</p>
  </dd>
  <dt>
    <p><strong>Within Expressions</strong></p>
  </dt>
  <dd>
    <p>A RasMol <strong>within</strong> expression allows atoms to be selected on their proximity to another set of atoms. A <strong>within</strong> expression takes two parameters separated by a comma and surrounded by parentheses. The first argument is an integer value called the "cut-off" distance of the within expression and the second argument is any  valid atom expression. The cut-off distance is expressed in either integer RasMol units or Angstroms containing a decimal point. An atom is selected if it is within the cut-off distance of any of the atoms defined by the second argument. This allows complex expressions to be constructed containing nested <strong>within</strong> expressions.</p><p>For example, the command <strong>select within(3.2,backbone)</strong> selects any atom within a 3.2 Angstrom radius of any atom in a protein or nucleic acid backbone. <strong>Within</strong> expressions are particularly useful for selecting the atoms around an active site.</p>
  </dd>
  <dt>
    <p><strong>Predefined Sets</strong></p>
  </dt>
  <dd>
    <p>RasMol atom expressions may contain predefined sets. These sets are single keywords that represent portions of a molecule of interest. Predefined sets are often abbreviations of primitive atom expressions. In some cases the use of predefined sets allows selection of areas of a molecule that could not otherwise be distinguished. A list of the currently predefined sets is given below. In addition to the sets listed here, RasMol also treats element names (and their plurals) as predefined sets containing all atoms of that element type, i.e. the command <strong>select oxygen</strong> is equivalent to the command <strong>select elemno=8.</strong></p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">Predefined Sets</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>AT Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the atoms in the complementary nucleotides adenosine and thymidine (A and T, respectively). All nucleotides are classified as either the set <strong>at</strong> or the set <strong>cg</strong> This set is equivalent to the RasMol atom expressions <strong>a,t,</strong> and <strong>nucleic and not cg.</strong></p>
  </dd>
  <dt>
    <p><strong>Acidic Set</strong></p>
  </dt>
  <dd>
    <p>The set of acidic amino acids. These are the residue types Asp and Glu. All amino acids are classified as either <strong>acidic,</strong> <strong>basic</strong> <strong>or</strong> <strong>neutral.</strong> This set is equivalent to the RasMol atom expressions <strong>asp, glu</strong> and <strong>amino and not (basic or neutral).</strong></p>
  </dd>
  <dt>
    <p><strong>Acyclic Set</strong></p>
  </dt>
  <dd>
    <p>The set of atoms in amino acids not containing a cycle or ring. All amino acids are classified as either <strong>cyclic</strong> or <strong>acyclic.</strong> This set is equivalent to the RasMol atom expression <strong>amino and not cyclic.</strong></p>
  </dd>
  <dt>
    <p><strong>Aliphatic Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the aliphatic amino acids. These are the amino acids Ala, Gly, Ile, Leu and Val. This set is equivalent to the RasMol atom expression <strong>ala, gly, ile, leu, val.</strong></p>
  </dd>
  <dt>
    <p><strong>Alpha Set</strong></p>
  </dt>
  <dd>
    <p>The set of alpha carbons in the protein molecule. This set is approximately equivalent to the RasMol atom expression <strong>*.CA.</strong> This command should not be confused with the predefined set <strong>helix</strong> which contains the atoms in the amino acids of the protein's alpha helices.</p>
  </dd>
  <dt>
    <p><strong>Amino Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all the atoms contained in amino acid residues. This is useful for distinguishing the protein from the nucleic acid and heterogeneous atoms in the current molecule database.</p>
  </dd>
  <dt>
    <p><strong>Aromatic Set</strong></p>
  </dt>
  <dd>
    <p>The set of atoms in amino acids containing aromatic rings. These are the amino acids His, Phe, Trp and Tyr. Because they contain aromatic rings all members of this set are member of the predefined set <strong>cyclic.</strong> This set is equivalent to the RasMol atom expressions <strong>his, phe, trp, tyr</strong> and <strong>cyclic and not pro.</strong></p>
  </dd>
  <dt>
    <p><strong>Backbone Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the four atoms of each amino acid that form the polypeptide N-C-C-O backbone of proteins, and the atoms of the sugar phosphate backbone of nucleic acids. Use the RasMol predefined sets <strong>protein</strong> and <strong>nucleic</strong> to distinguish between the two forms of backbone. Atoms in nucleic acids and proteins are either <strong>backbone</strong> or <strong>sidechain.</strong> This set is equivalent to the RasMol expression <strong>(protein or nucleic) and not sidechain.</strong></p><p>The predefined set <strong>mainchain</strong> is synonymous with the set <strong>backbone.</strong></p>
  </dd>
  <dt>
    <p><strong>Basic Set</strong></p>
  </dt>
  <dd>
    <p>The set of basic amino acids. These are the residue types Arg, His and Lys. All amino acids are classified as either <strong>acidic,</strong> <strong>basic</strong> or <strong>neutral.</strong> This set is equivalent to the RasMol atom expressions <strong>arg, his, lys</strong> and <strong>amino and not (acidic or neutral).</strong></p>
  </dd>
  <dt>
    <p><strong>Bonded Set</strong></p>
  </dt>
  <dd>
    <p>This set contain all the atoms in the current molecule database that are bonded to at least one other atom.</p>
  </dd>
  <dt>
    <p><strong>Buried Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the atoms in those amino acids that tend (prefer) to be buried inside protein, away from contact with solvent molecules. This set refers to the amino acids preference and not the actual solvent accessibility for the current protein. All amino acids are classified as either <strong>surface</strong> or <strong>buried.</strong> This set is equivalent to the RasMol atom expression <strong>amino and not surface.</strong></p>
  </dd>
  <dt>
    <p><strong>CG Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the atoms in the complementary nucleotides cytidine and guanosine (C and G, respectively). All nucleotides are classified as either the set <strong>at</strong> or the set <strong>cg</strong> This set is equivalent to the RasMol atom expressions <strong>c,g</strong> and <strong>nucleic and not at.</strong></p>
  </dd>
  <dt>
    <p><strong>Charged Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the charged amino acids. These are the amino acids that are either <strong>acidic</strong> or <strong>basic.</strong> Amino acids are classified as being either <strong>charged</strong> or <strong>neutral.</strong> This set is equivalent to the RasMol atom expressions <strong>acidic or basic</strong> and <strong>amino and not neutral.</strong></p>
  </dd>
  <dt>
    <p><strong>Cyclic Set</strong></p>
  </dt>
  <dd>
    <p>The set of atoms in amino acids containing a cycle or rings. All amino acids are classified as either <strong>cyclic</strong> or <strong>acyclic.</strong> This set consists of the amino acids His, Phe, Pro, Trp and Tyr. The members of the predefined set <strong>aromatic</strong> are members of this set. The only cyclic but non-aromatic amino acid is proline. This set is equivalent to the RasMol atom expressions <strong>his, phe, pro, trp, tyr</strong> and <strong>aromatic or pro</strong> and <strong>amino and not acyclic.</strong></p>
  </dd>
  <dt>
    <p><strong>Cystine Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the atoms of cysteine residues that form part of a disulphide bridge, i.e. half cystines. RasMol automatically determines disulphide bridges, if neither the predefined set <strong>cystine</strong> nor the RasMol <strong>ssbonds</strong> command have been used since the molecule was loaded. The set of free cysteines may be determined using the RasMol atom expression <strong>cys and not cystine.</strong></p>
  </dd>
  <dt>
    <p><strong>Helix Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all atoms that form part of a protein alpha helix as determined by either the PDB file author or Kabsch and Sander's DSSP algorithm. By default, RasMol uses the secondary structure determination given in the PDB file if it exists. Otherwise, it uses the DSSP algorithm as used by the RasMol <strong>structure</strong> command.</p><p>This predefined set should not be confused with the predefined set <strong>alpha</strong> which contains the alpha carbon atoms of a protein.</p>
  </dd>
  <dt>
    <p><strong>Hetero Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all the heterogeneous atoms in the molecule. These are the atoms described by HETATM entries in the PDB file. These typically contain water, cofactors and other solvents and ligands. All <strong>hetero</strong> atoms are classified as either <strong>ligand</strong> or <strong>solvent</strong> atoms. These heterogeneous <strong>solvent</strong> atoms are further classified as either <strong>water</strong> or <strong>ions.</strong></p>
  </dd>
  <dt>
    <p><strong>Hydrogen Set</strong></p>
  </dt>
  <dd>
    <p>This predefined set contains all the hydrogen, deuterium and tritium atoms of the current molecule. This predefined set is equivalent to the RasMol atom expression <strong>elemno=1.</strong></p>
  </dd>
  <dt>
    <p><strong>Hydrophobic Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all the hydrophobic amino acids. These are the amino acids Ala, Leu, Val, Ile, Pro, Phe, Met and Trp. All amino acids are classified as either <strong>hydrophobic</strong> or <strong>polar.</strong> This set is equivalent to the RasMol atom expressions <strong>ala, leu, val, ile, pro, phe, met, trp</strong> and <strong>amino and not polar.</strong></p>
  </dd>
  <dt>
    <p><strong>Ions Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all the heterogeneous phosphate and sulphate ions in the current molecule data file. A large number of these ions are sometimes associated with protein and nucleic acid structures determined by X-ray crystallography. These atoms tend to clutter an image. All <strong>hetero</strong> atoms are classified as either <strong>ligand</strong> or <strong>solvent</strong> atoms. All <strong>solvent</strong> atoms are classified as either <strong>water</strong> or <strong>ions.</strong></p>
  </dd>
  <dt>
    <p><strong>Large Set</strong></p>
  </dt>
  <dd>
    <p>All amino acids are classified as either <strong>small,</strong> <strong>medium</strong> or <strong>large.</strong> This set is equivalent to the RasMol atom expression <strong>amino and not (small or medium).</strong></p>
  </dd>
  <dt>
    <p><strong>Ligand Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all the heterogeneous cofactor and ligand moieties that are contained in the current molecule data file.  This set is defined to be all <strong>hetero</strong> atoms that are not <strong>solvent</strong> atoms. Hence this set is equivalent to the RasMol atom expression <strong>hetero and not solvent.</strong></p>
  </dd>
  <dt>
    <p><strong>Medium Set</strong></p>
  </dt>
  <dd>
    <p>All amino acids are classified as either <strong>small,</strong> <strong>medium</strong> or <strong>large.</strong> This set is equivalent to the RasMol atom expression <strong>amino and not (large or small).</strong></p>
  </dd>
  <dt>
    <p><strong>Neutral Set</strong></p>
  </dt>
  <dd>
    <p>The set of neutral amino acids. All amino acids are classified as either <strong>acidic,</strong> <strong>basic</strong> or <strong>neutral.</strong> This set is equivalent to the RasMol atom expression <strong>amino and not (acidic or basic).</strong></p>
  </dd>
  <dt>
    <p><strong>Nucleic Set</strong></p>
  </dt>
  <dd>
    <p>The set of all atoms in nucleic acids, which consists of the four nucleotide bases adenosine, cytidine, guanosine and thymidine (A, C, G and T, respectively). All neucleotides are classified as either <strong>purine</strong> or <strong>pyrimidine.</strong> This set is equivalent to the RasMol atom expressions <strong>a,c,g,t</strong> and <strong>purine or pyrimidine.</strong> The symbols for RNA nucleotides (U, +U, I, 1MA, 5MC, OMC, 1MG, 2MG, M2G, 7MG, OMG, YG, H2U, 5MU, and PSU) are also recognized as members of this set.</p>
  </dd>
  <dt>
    <p><strong>Polar Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the polar amino acids. All amino acids are classified as either <strong>hydrophobic</strong> or <strong>polar.</strong> This set is equivalent to the RasMol atom expression <strong>amino and not hydrophobic.</strong></p>
  </dd>
  <dt>
    <p><strong>Protein Set</strong></p>
  </dt>
  <dd>
    <p>The set of all atoms in proteins. This consists of the RasMol predefined set <strong>amino</strong> and common post-translation modifications.</p>
  </dd>
  <dt>
    <p><strong>Purine Set</strong></p>
  </dt>
  <dd>
    <p>The set of purine nucleotides. These are the bases adenosine and guanosine (A and G, respectively). All nucleotides are either <strong>purines</strong> or <strong>pyrimidines.</strong> This set is equivalent to the RasMol atom expressions <strong>a,g</strong> and <strong>nucleic and not pyrimidine.</strong></p>
  </dd>
  <dt>
    <p><strong>Pyrimidine Set</strong></p>
  </dt>
  <dd>
    <p>The set of pyrimidine nucleotides. These are the bases cytidine and thymidine (C and T, respectively). All nucleotides are either <strong>purines</strong> or <strong>pyrimidines.</strong> This set is equivalent to the RasMol atom expressions <strong>c,t</strong> and <strong>nucleic and not purine.</strong></p>
  </dd>
  <dt>
    <p><strong>Selected Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the set of atoms in the currently selected region. The currently selected region is defined by the preceding <strong>select</strong> or <strong>restrict</strong> command and not the atom expression containing the <strong>selected</strong> keyword.</p>
  </dd>
  <dt>
    <p><strong>Sheet Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all atoms that form part of a protein beta sheet as determined by either the PDB file author or Kabsch and Sander's DSSP algorithm. By default, RasMol uses the secondary structure determination given in the PDB file if it exists. Otherwise, it uses the DSSP algorithm as used by the RasMol <strong>structure</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Sidechain Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the functional sidechains of any amino acids and the base of each nucleotide. These are the atoms not part of the polypeptide N-C-C-O backbone of proteins or the sugar phosphate backbone of nucleic acids. Use the RasMol predefined sets <strong>protein</strong> and <strong>nucleic</strong> to distinguish between the two forms of sidechain. Atoms in nucleic acids and proteins are either <strong>backbone</strong> or <strong>sidechain.</strong> This set is equivalent to the RasMol expression <strong>(protein or nucleic) and not backbone.</strong></p>
  </dd>
  <dt>
    <p><strong>Small Set</strong></p>
  </dt>
  <dd>
    <p>All amino acids are classified as either <strong>small,</strong> <strong>medium</strong> or <strong>large.</strong> This set is equivalent to the RasMol atom expression <strong>amino and not (medium or large).</strong></p>
  </dd>
  <dt>
    <p><strong>Solvent Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the solvent atoms in the molecule coordinate file. These are the heterogeneous water molecules, phosphate and sulphate ions. All <strong>hetero</strong> atoms are classified as either <strong>ligand</strong> or <strong>solvent</strong> atoms. All <strong>solvent</strong> atoms are classified as either <strong>water</strong> or <strong>ions.</strong> This set is equivalent to the RasMol atom expressions <strong>hetero and not ligand</strong> and <strong>water or ions.</strong></p>
  </dd>
  <dt>
    <p><strong>Surface Set</strong></p>
  </dt>
  <dd>
    <p>This set contains the atoms in those amino acids that tend (prefer) to be on the surface of proteins, in contact with solvent molecules. This set refers to the amino acids preference and not the actual solvent accessibility for the current protein. All amino acids are classified as either <strong>surface</strong> or <strong>buried.</strong> This set is equivalent to the RasMol atom expression <strong>amino and not buried.</strong></p>
  </dd>
  <dt>
    <p><strong>Turn Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all atoms that form part of a protein turns as determined by either the PDB file author or Kabsch and Sander's DSSP algorithm. By default, RasMol uses the secondary structure determination given in the PDB file if it exists. Otherwise, it uses the DSSP algorithm as used by the RasMol <strong>structure</strong> command.</p>
  </dd>
  <dt>
    <p><strong>Water Set</strong></p>
  </dt>
  <dd>
    <p>This set contains all the heterogeneous water molecules in the current database. A large number of water molecules are sometimes associated with protein and nucleic acid structures determined by X-ray crystallography. These atoms tend to clutter an image. All <strong>hetero</strong> atoms are classified as either <strong>ligand</strong> or <strong>solvent</strong> atoms. The <strong>solvent</strong> atoms are further classified as either <strong>water</strong> or <strong>ions.</strong></p>
  </dd>
  <dt>
    <p><strong>Set Summary</strong></p>
  </dt>
  <dd>
    <p>The table below summarises RasMol's classification of the common amino acids.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COLOUR SCHEMES</h2>
        <div class="sectioncontent">
<p>The RasMol <strong>colour</strong> command allows different objects (such as atoms, bonds and ribbon segments) to be given a specified colour. Typically this colour is either a RasMol predefined colour name or an RGB triple. Additionally RasMol also supports <strong>alt,</strong> <strong>amino,</strong> <strong>chain,</strong> <strong>charge,</strong> <strong>cpk,</strong> <strong>group,</strong> <strong>model,</strong> <strong>shapely,</strong> <strong>structure,</strong> <strong>temperature</strong> or <strong>user</strong> colour schemes for atoms, and <strong>hbond type</strong> colour scheme for hydrogen bonds and <strong>electrostatic potential</strong> colour scheme for dot surfaces. The 24 currently predefined colour names are Black, Blue, BlueTint, Brown, Cyan, Gold, Grey, Green, GreenBlue, GreenTint, HotPink, Magenta, Orange, Pink, PinkTint, Purple, Red, RedOrange, SeaGreen, SkyBlue, Violet, White, Yellow and YellowTint</p><p>If you frequently wish to use a colour not predefined, you can write a one-line script. For example, if you make the file <strong>grey.col</strong> containing the line, <strong>colour [180,180,180] #grey,</strong> then the command <strong>script grey.col</strong> colours the currently selected atom set grey.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>Alt Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>alt</strong> (Alternate Conformer) colour scheme codes the base structure with one colour and applies a limited number of colours to each alternate conformer.  In a RasMol built for 8-bit colour systems, 4 colours are allowed for alternate conformers. Otherwise, 8 colours are available.</p>
  </dd>
  <dt>
    <p><strong>Amino Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>amino</strong> colour scheme colours amino acids according to traditional amino acid properties. The purpose of colouring is to identify amino acids in an unusual or surprising environment. The outer parts of a protein that are polar are visible (bright) colours and non-polar residues darker. Most colours are hallowed by tradition. This colour scheme is similar to the <strong>shapely</strong> scheme.</p>
  </dd>
  <dt>
    <p><strong>Chain Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>chain</strong> colour scheme assigns each macromolecular chain a unique colour. This colour scheme is particularly useful for distinguishing the parts of multimeric structure or the individual 'strands' of a DNA chain. <strong>Chain</strong> can be selected from the RasMol <strong>Colours</strong> menu.</p>
  </dd>
  <dt>
    <p><strong>Charge Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>charge</strong> colour scheme colour codes each atom according to the charge value stored in the input file (or beta factor field of PDB files). High values are coloured in blue (positive) and lower values coloured in red (negative). Rather than use a fixed scale this scheme determines the maximum and minimum values of the charge/temperature field and interpolates from red to blue appropriately. Hence, green cannot be assumed to be 'no net charge' charge.</p><p>The difference between the <strong>charge</strong> and <strong>temperature</strong> colour schemes is that increasing temperature values proceed from blue to red, whereas increasing charge values go from red to blue.</p><p>If the charge/temperature field stores reasonable values it is possible to use the RasMol <strong>colour dots potential</strong> command to colour code a dot surface (generated by the <strong>dots</strong> command) by electrostatic potential.</p>
  </dd>
  <dt>
    <p><strong>CPK Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>cpk</strong> colour scheme is based upon the colours of the popular plastic spacefilling models which were developed by Corey, Pauling and later improved by Kultun. This colour scheme colours 'atom' objects by the atom (element) type. This is the scheme conventionally used by chemists. The assignment of the most commonly used element types to colours is given below.</p>
  </dd>
  <dt>
    <p><strong>Group Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>group</strong> colour scheme colour codes residues by their position in a macromolecular chain. Each chain is drawn as a smooth spectrum from blue through green, yellow and orange to red. Hence the N terminus of proteins and 5' terminus of nucleic acids are coloured red and the C terminus of proteins and 3' terminus of nucleic acids are drawn in blue. If a chain has a large number of heterogeneous molecules associated with it, the macromolecule may not be drawn in the full 'range' of the spectrum. <strong>Group</strong> can be selected from the RasMol <strong>Colours</strong> menu.</p><p>If a chain has a large number of heterogeneous molecules associated with it, the macromolecule may not be drawn in the full range of the spectrum. When RasMol performs group colouring it decides the range of colours it uses from the residue numbering given in the PDB file. Hence the lowest residue number is displayed in blue and the highest residue number is displayed as red. Unfortunately, if a PDB file contains a large number of heteroatoms, such as water molecules, that occupy the high residue numbers, the protein is displayed in the blue-green end of the spectrum and the waters in the yellow-red end of the spectrum. This is aggravated by there typically being many more water molecules than amino acid residues. The solution to this problem is to use the command <strong>set hetero off</strong> before applying the group colour scheme. This can also be achieved by toggling <strong>Hetero Atoms</strong> on the <strong>Options</strong> menu before selecting <strong>Group</strong> on the <strong>Colour</strong> menu. This command instructs RasMol to only use non-hetero residues in the group colour scaling.</p>
  </dd>
  <dt>
    <p><strong>NMR Model Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>model</strong> colour scheme codes each NMR model with a distinct colour.  The NMR model number is taken as a numeric value.  High values are coloured in blue and lower values coloured in red. Rather than use a fixed scale this scheme determines the maximum value of the NMR model number and interpolates from red to blue appropriately.</p>
  </dd>
  <dt>
    <p><strong>Shapely Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>shapely</strong> colour scheme colour codes residues by amino acid property. This scheme is based upon Bob Fletterick's "Shapely Models". Each amino acid and nucleic acid residue is given a unique colour. The <strong>shapely</strong> colour scheme is used by David Bacon's Raster3D program. This colour scheme is similar to the <strong>amino</strong> colour scheme.</p>
  </dd>
  <dt>
    <p><strong>Structure Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>structure</strong> colour scheme colours the molecule by protein secondary structure. Alpha helices are coloured magenta, [240,0,128], beta sheets are coloured yellow, [255,255,0], turns are coloured pale blue, [96,128,255] and all other residues are coloured white. The secondary structure is either read from the PDB file (HELIX, SHEET and TURN records), if available, or determined using Kabsch and Sander's DSSP algorithm. The RasMol <strong>structure</strong> command may be used to force DSSP's structure assignment to be used.</p>
  </dd>
  <dt>
    <p><strong>Temperature Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>temperature</strong> colour scheme colour codes each atom according to the anisotropic temperature (beta) value stored in the PDB file. Typically this gives a measure of the mobility/uncertainty of a given atom's position. High values are coloured in warmer (red) colours and lower values in colder (blue) colours. This feature is often used to associate a "scale" value [such as amino acid variability in viral mutants] with each atom in a PDB file, and colour the molecule appropriately.</p><p>The difference between the <strong>temperature</strong> and <strong>charge</strong> colour schemes is that increasing temperature values proceed from blue to red, whereas increasing charge values go from red to blue.</p>
  </dd>
  <dt>
    <p><strong>User Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>user</strong> colour scheme allows RasMol to use the colour scheme stored in the PDB file. The colours for each atom are stored in COLO records placed in the PDB data file. This convention was introduced by David Bacon's Raster3D program.</p>
  </dd>
  <dt>
    <p><strong>HBond Type Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>type</strong> colour scheme applies only to hydrogen bonds, hence is used in the command <strong>colour hbonds type.</strong> This scheme colour codes each hydrogen bond according to the distance along a protein chain between hydrogen bond donor and acceptor. This schematic representation was introduced by Belhadj-Mostefa and Milner-White. This representation gives a good insight into protein secondary structure (hbonds forming alpha helices appear red, those forming sheets appear yellow and those forming turns appear magenta).</p>
  </dd>
  <dt>
    <p><strong>Potential Colours</strong></p>
  </dt>
  <dd>
    <p>The RasMol <strong>potential</strong> colour scheme applies only to dot surfaces, hence is used in the command <strong>colour dots potential.</strong> This scheme colours each currently displayed dot by the electrostatic potential at that point in space. This potential is calculated using Coulomb's law taking the temperature/charge field of the input file to be the charge assocated with that atom. This is the same interpretation used by the <strong>colour charge</strong> command. Like the <strong>charge</strong> colour scheme low values are blue/white and high values are red.</p>
  </dd>
  <dt>
    <p><strong>Amino Acid Codes</strong></p>
  </dt>
  <dd>
    <p>The following table lists the names, single letter and three letter codes of each of the amino acids.</p>
  </dd>
  <dt>
    <p><strong>Booleans</strong></p>
  </dt>
  <dd>
    <p>A boolean parameter is a truth value. Valid boolean values are 'true' and 'false', and their synonyms 'on' and 'off'. Boolean parameters are commonly used by RasMol to either enable or disable a representation or option.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FILE FORMATS</h2>
        <div class="sectioncontent">
<p><strong>Protein Data Bank Files</strong></p><p>If you do not have the PDB documentation, you may find the following summary of the PDB file format useful. The Protein Data Bank is a computer-based archival database for macromolecular structures. The database was established in 1971 by Brookhaven National Laboratory, Upton, New York, as a public domain repository for resolved crystallographic structures. The Bank uses a uniform format to store atomic coordinates and partial bond connectivities as derived from crystallographic studies.  In 1999 the Protein Data Bank moved to the Research Collaboratory for Structural Biology.</p><p>PDB file entries consist of records of 80 characters each. Using the punched card analogy, columns 1 to 6 contain a record-type identifier, the columns 7 to 70 contain data. In older entries, columns 71 to 80 are normally blank, but may contain sequence information added by library management programs.  In new entries conforming to the 1996 PDB format, there is other information in those columns. The first four characters of the record identifier are sufficient to identify the type of record uniquely, and the syntax of each record is independent of the order of records within any entry for a particular macromolecule.</p><p>The only record types that are of major interest to the RasMol program are the ATOM and HETATM records which describe the position of each atom. ATOM/HETATM records contain standard atom names and residue abbreviations, along with sequence identifiers, coordinates in Angstrom units, occupancies and thermal motion factors. The exact details are given below as a FORTRAN format statement.  The "fmt" column indicates use of the field in all PDB formats, in the 1992 and earlier formats or in the 1996 and later formats.</p><p>Residues occur in order starting from the N-terminal residue for proteins and 5'-terminus for nucleic acids. If the residue sequence is known, certain atom serial numbers may be omitted to allow for future insertion of any missing atoms. Within each residue, atoms are ordered in a standard manner, starting with the backbone (N-C-C-O for proteins) and proceeding in increasing remoteness from the alpha carbon, along the side chain.</p><p>HETATM records are used to define post-translational modifications and cofactors associated with the main molecule. TER records are interpreted as breaks in the main molecule's backbone.</p><p>If present, RasMol also inspects HEADER, COMPND, HELIX, SHEET, TURN, CONECT, CRYST1, SCALE, MODEL, ENDMDL, EXPDTA and END records. Information such as the name, database code, revision date and classification of the molecule are extracted from HEADER and COMPND records, initial secondary structure assignments are taken from HELIX, SHEET and TURN records, and the end of the file may be indicated by an END record.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>RasMol Interpretation of PDB fields</strong></p>
  </dt>
  <dd>
    <p>Atoms located at 9999.000, 9999.000, 9999.000 are assumed to be Insight pseudo atoms and are ignored by RasMol. Atom names beginning ' Q' are also assumed to be pseudo atoms or position markers.</p><p>When a data file contains an NMR structure, multiple conformations may be placed in a single PDB file delimited by pairs of MODEL and ENDMDL records. RasMol displays all the NMR models contained in the file.</p><p>Residue names "CSH", "CYH" and "CSM" are considered pseudonyms for cysteine "CYS". Residue names "WAT", "H20", "SOL" and "TIP" are considered pseudonyms for water "HOH". The residue name "D20" is consider heavy water "DOD". The residue name "SUL" is considered a sulphate ion "SO4". The residue name "CPR" is considered to be cis-proline and is translated as "PRO". The residue name "TRY" is considered a pseudonym for tryptophan "TRP".</p><p>RasMol uses the HETATM fields to define the sets hetero, water, solvent and ligand. Any group with the name "HOH", "DOD", "SO4" or "PO4" (or aliased to one of these names by the preceding rules) is considered a solvent and is considered to be defined by a HETATM field.</p><p>RasMol only respects CONECT connectivity records in PDB files containing fewer than 256 atoms. This is explained in more detail in the section on determining molecule connectivity. CONECT records that define a bond more than once are interpreted as specifying the bond order of that bond, i.e. a bond specified twice is a double bond and a bond specified three (or more) times is a triple bond.  This is not a standard PDB feature.</p>
  </dd>
  <dt>
    <p><strong>PDB Colour Scheme Specification</strong></p>
  </dt>
  <dd>
    <p>RasMol also accepts the supplementary COLO record type in the PDB files. This record format was introduced by David Bacon's Raster3D program for specifying the colour scheme to be used when rendering the molecule. This extension is not currently supported by the PDB. The COLO record has the same basic record type as the ATOM and HETATM records described above.</p><p>Colours are assigned to atoms using a matching process. The Mask field is used in the matching process as follows. First RasMol reads in and remembers all the ATOM, HETATM and COLO records in input order. When the user-defined ('User') colour scheme is selected, RasMol goes through each remembered ATOM/HETATM record in turn, and searches for a COLO record that matches in all of columns 7 through 30. The first such COLO record to be found determines the colour and radius of the atom.</p><p>Note that the Red, Green and Blue components are in the same positions as the X, Y, and Z components of an ATOM or HETA record, and the van der Waals radius goes in the place of the Occupancy. The Red, Green and Blue components must all be in the range 0 to 1.</p><p>In order that one COLO record can provide colour and radius specifications for more than one atom (e.g. based on residue, atom type, or any other criterion for which labels can be given somewhere in columns 7 through 30), a 'don't-care' character, the hash mark "#" (number or sharp sign) is used. This character, when found in a COLO record, matches any character in the corresponding column in a ATOM/HETATM record. All other characters must match identically to count as a match. As an extension to the specification, any atom that fails to match a COLO record is displayed in white.</p>
  </dd>
  <dt>
    <p><strong>Multiple NMR Models</strong></p>
  </dt>
  <dd>
    <p>RasMol loads all of the NMR models from a PDB file no matter which command is used: <strong>load pdb &lt;filename&gt;</strong> or <strong>load nmrpdb &lt;filename&gt;</strong></p><p>Once multiple NMR conformations have been loaded they may be manipulated with the atom expression extensions described in <strong>Primitive Expressions.</strong> In particular, the command <strong>restrict */1</strong> will restrict the display to the first model only.</p>
  </dd>
  <dt>
    <p><strong>CIF and mmCIF Format Files</strong></p>
  </dt>
  <dd>
    <p>CIF is the IUCr standard for presentation of small molecules and mmCIF is intended as the replacement for the fixed-field PDB format for presentation of macromolecular structures. RasMol can accept data sets in either format.</p><p>There are many useful sites on the World Wide Web where information tools and software related to CIF, mmCIF and the PDB can be found. The following are good starting points for exploration:</p><p>The International Union of Crystallography (IUCr) provides access to software, dictionaries, policy statements and documentation relating to CIF and mmCIF at: IUCr, Chester, England (www.iucr.org/iucr-top/cif/) with many mirror sites.</p><p>The Nucleic Acid Database Project provides access to its entries, software and documentation, with an mmCIF page giving access to the dictionary and mmCIF software tools at Rutgers University, New Jersey, USA (http://ndbserver.rutgers.edu/NDB/mmcif) with many mirror sites.</p><p>This version of RasMol restricts CIF or mmCIF tag values to essentially the same conventions as are used for the fixed-field PDB format. Thus chain identifiers and alternate conformation identifiers are limited to a single character, atom names are limited to 4 characters, etc. RasMol interprets the following CIF and mmCIF tags: A search is made through multiple data blocks for the desired tags, so a single dataset may be composed from multiple data blocks, but multiple data sets may not be stacked in the same file.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">MACHINE-SPECIFIC SUPPORT</h2>
        <div class="sectioncontent">
<p>In the following sections, support for <strong>Monochrome X-Windows,</strong> <strong>Tcl/Tk IPC,</strong> <strong>UNIX sockets based IPC,</strong> <strong>Compiling RasWin with Borland and MetroWerks</strong> are described.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>Monochrome X-Windows Support</strong></p>
  </dt>
  <dd>
    <p>RasMol supports the many monochrome UNIX workstations typically found in academia, such as low-end SUN workstations and NCD X-terminals. The X11 version of RasMol (when compiled in 8 bit mode) now detects black and white X-Windows displays and enables dithering automatically. The use of run-time error diffusion dithering means that all display modes of RasMol are available when in monochrome mode. For best results, users should experiment with the set ambient command to ensure the maximum contrast in resulting images.</p>
  </dd>
  <dt>
    <p><strong>Tcl/Tk IPC support</strong></p>
  </dt>
  <dd>
    <p>Version 4 of Tk graphics library changed the protocol used to communicate between Tk applications. RasMol version 2.6 was modified such that it could communicate with both this new protocol and the previous version 3 protocol supported by RasMol v2.5. Although Tcl/Tk 3.x applications may only communicate with other 3.x applications and Tcl/Tk 4.x applications with other 4.x applications, these changes allow RasMol to communicate between processes with both protocols (potentially concurrently).</p>
  </dd>
  <dt>
    <p><strong>UNIX sockets based IPC</strong></p>
  </dt>
  <dd>
    <p>The UNIX implementation of RasMol supports BSD-style socket communication. An identical socket mechanism is also being developed for VMS, Apple Macintosh and Microsoft Windows systems. This should allow RasMol to interactively display results of a computation on a remote host. The current protocol acts as a TCP/IP server on port 21069 that executes command lines until either the command <strong>exit</strong> or the command <strong>quit</strong> is typed. The command exit from the RasMol server, the command <strong>quit</strong> both disconnects the current session and terminates RasMol. This functionality may be tested using the UNIX command <strong>telnet &lt;hostname&gt; 21069.</strong></p>
  </dd>
  <dt>
    <p><strong>Compiling RasWin with Borland and MetroWerks</strong></p>
  </dt>
  <dd>
    <p>A number of changes were made to the source code in the transition from version 2.5 to 2.6 to allow the Microsoft Windows version of RasMol to compile using the Borland C/C++ compiler. These fixes include name changes for the standard library and special code to avoid a bug in _fmemset. Additional changes were made in the transition from 2.6 to 2.7 to allow compilation with the MetroWerks compilers.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BIBLIOGRAPHY</h2>
        <div class="sectioncontent">
<p><strong>Molecular Graphics</strong></p><p>[1] Nelson Max, "Computer Representation of Molecular Surfaces", IEEE Computer Graphics and Applications, pp.21-29, August 1983.</p><p>[2] Arthur M. Lesk, "Protein Architecture: A Practical Approach", IRL Press Publishers, 1991.</p><p><strong>Molecular Graphics Programs</strong></p><p>[3] Per J. Kraulis, "MOLSCRIPT: A Program to Produce both Detailed and Schematic Plots of Protein Structures", Journal of Applied Crystallography, Vol.24, pp.946-950, 1991.</p><p>[4] David Bacon and Wayne F. Anderson, "A Fast Algorithm for Rendering Space-Filling Molecule Pictures", Journal of Molecular Graphics, Vol.6, No.4, pp.219-220, December 1988.</p><p>[5] David C. Richardson and Jane S. Richardson, "The Kinemage: A tool for Scientific Communication", Protein Science, Vol.1, No.1,pp.3-9, January 1992.</p><p>[6] Mike Carson, "RIBBONS 2.0", Journal of Applied Crystallography, Vol.24, pp.958-961, 1991.</p><p>[7] Conrad C. Huang, Eric F. Pettersen, Teri E. Klein, Thomas E. Ferrin and Robert Langridge, "Conic: A Fast Renderer for Space-Filling Molecules with Shadows", Journal of Molecular Graphics, Vol.9, No.4, pp.230-236, December 1991.</p><p><strong>Molecular Biology Algorithms</strong></p><p>[8] Wolfgang Kabsch and Christian Sander, "Dictionary of Protein Secondary Structure: Pattern Recognition of Hydrogen-Bonded and Geometrical Features", Biopolymers, Vol.22, pp.2577-2637, 1983.</p><p>[9] Michael L. Connolly, "Solvent-Accessible Surfaces of Proteins and Nucleic Acids", Science, Vol.221, No.4612, pp.709-713, August 1983.</p><p>[10] Khaled Belhadj-Mostefa, Ron Poet and E. James Milner-White, "Displaying Inter-Main Chain Hydrogen Bond Patterns in Proteins", Journal of Molecular Graphics, Vol.9, No.3, pp.194-197, September 1991.</p><p>[11] Mike Carson, "Ribbon Models of Macromolecules", Journal of Molecular Graphics, Vol.5, No.2, pp.103-106, June 1987.</p><p>[12] Mike Carson and Charles E. Bugg, "Algorithm for Ribbon Models of Proteins", Journal of Molecular Graphics, Vol.4, No.2, pp.121-122, June 1986.</p><p>[13] H. Iijima, J. B. Dunbar Jr. and G. Marshall, "Calibration of Effective van der Waals Atomic Contact Radii for Proteins and Peptides", Proteins: Structure, Functions and Genetics, Vol.2, pp.330-339,1987.</p><p><strong>Graphics Algorithms</strong></p><p>[14] J. Foley, A. van Dam, S. Feiner and J. Hughes, "Computer Graphics: Principles and Practice", 2nd Edition, Addison Wesley Publishers, 1990.</p><p>[15] J. Cleary and G. Wyvill, "Analysis of an Algorithm for Fast Ray Tracing using Uniform Space Subdivision", The Visual Computer, Vol.4, pp.65-83, 1988.</p><p>[16] Thomas Porter,"Spherical Shading", Computer Graphics Vol.12, ACM SIGGRAPH, pp.282-285, 1978.</p><p>[17] Jean-Michel Cense, "Exact Visibility Calculation for Space-Filling Molecular Models", Journal of Molecular Graphics, Vol.9, No.3, pp.191-193, September 1991.</p><p>[18] Chris Schafmeister, "Fast Algorithm for Generating CPK Images on Graphics Workstations", Journal of Molecular Graphics, Vol.8, No.4, pp.201-206, December 1990.</p><p>[19] Bruce A. Johnson, "MSURF: A Rapid and General Program for the Representation of Molecular Surfaces", Journal of Molecular Graphics, Vol.5, No.3, pp.167-169, September 1987.</p><p><strong>File Formats</strong></p><p>[20] Frances C. Bernstein et al., "The Protein Data Bank: A Computer-Based Archival File for Macromolecular Structures", Journal of Molecular Biology, Vol.112, pp.535-542, 1977.</p><p>[21] Arthur Dalby, James G. Nourse, W. Douglas Hounshell, Ann K. I. Gushurst, David L. Grier, Burton A. Leland and John Laufer, "Description of Several Chemical File Formats Used by Computer Programs Developed at Molecular Design Limited", Journal of Chemical Information and Computer Sciences, Vol.32, No.3, pp.244-255, 1992.</p><p>[22] Adobe Systems Inc., "PostScript Language Reference Manual", Addison-Wesley Publishers, Reading, Mass., 1985.</p><p>[23] Philip E. Bourne et al., "The Macromolecular Crystallographic Information File (mmCIF)", Meth. Enzymol. (1997) 277, 571-590.</p><p>[24] Sydney R. Hall, "The STAR File: a New Format for Electronic Data Transfer and Archiving", Journal of Chemical Information and Computer Sciences, Vol. 31, 326-333, 1991.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO rasmol&hellip;</h2>
        <div class="sectioncontent">
<p>The RasMol User Manual!</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>1992-1998 by Roger Sayle (rasmol@ggr.co.uk)</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="ras2tiff.1.html"><span aria-hidden="true">&larr;</span> ras2tiff.1: Create a tiff file from a sun rasterfile</a></li>
   <li class="next"><a href="rasmol-classic.1.html">rasmol-classic.1: Molecular graphics visualisation tool v2.7.5 <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
