<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Ray: Assemble genomes in parallel using the message-passing interface</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Assemble genomes in parallel using the message-passing interface">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="Ray (1) manual">
  <meta name="twitter:description" content="Assemble genomes in parallel using the message-passing interface">
  <meta name="twitter:image" content="https://www.carta.tech/images/ray-Ray-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/Ray.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="Ray (1) manual" />
  <meta property="og:description" content="Assemble genomes in parallel using the message-passing interface" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/ray-Ray-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">Ray<small> (1)</small></h1>
        <p class="lead">Assemble genomes in parallel using the message-passing interface</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/Ray.1.html">
      <span itemprop="name">Ray: Assemble genomes in parallel using the message-passing interface</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/ray/">
      <span itemprop="name">ray</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/Ray.1.html">
      <span itemprop="name">Ray: Assemble genomes in parallel using the message-passing interface</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong></strong> mpiexec <strong>-n</strong> NUMBER_OF_RANKS Ray <strong>-k</strong> KMERLENGTH <strong>-p</strong> l1_1.fastq l1_2.fastq <strong>-p</strong> l2_1.fastq l2_2.fastq <strong>-o</strong> test</p><p><strong></strong> mpiexec <strong>-n</strong> NUMBER_OF_RANKS Ray Ray.conf # with commands in a file</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION:</h2>
        <div class="sectioncontent">
<p>The Ray genome assembler is built on top of the RayPlatform, a generic plugin-based distributed and parallel compute engine that uses the message-passing interface for passing messages.</p><p>Ray targets several applications:</p><ul>
<li><p>- de novo genome assembly (with Ray vanilla) - de novo meta-genome assembly (with Ray Meta) - de novo transcriptome assembly (works, but not tested a lot) - quantification of contig abundances - quantification of microbiome consortia members (with Ray Communities) - quantification of transcript expression - taxonomy profiling of samples (with Ray Communities) - gene ontology profiling of samples (with Ray Ontologies) <strong>-help</strong></p></li><li><p>Displays this help page. <strong>-version</strong></p></li><li><p>Displays Ray version and compilation options.</p></li><li><p>Using a configuration file</p></li><li><p>Ray can be launched with mpiexec <strong>-n</strong> 16 Ray Ray.conf The configuration file can include comments (starting with #).</p></li><li><p>K-mer length <strong>-k</strong> kmerLength</p></li><li><p>Selects the length of k-mers. The default value is 21. It must be odd because reverse-complement vertices are stored together. The maximum length is defined at compilation by MAXKMERLENGTH Larger k-mers utilise more memory.</p></li><li><p>Inputs <strong>-p</strong> leftSequenceFile rightSequenceFile [averageOuterDistance standardDeviation]</p></li><li><p>Provides two files containing paired-end reads. averageOuterDistance and standardDeviation are automatically computed if not provided. <strong>-i</strong> interleavedSequenceFile [averageOuterDistance standardDeviation]</p></li><li><p>Provides one file containing interleaved paired-end reads. averageOuterDistance and standardDeviation are automatically computed if not provided. <strong>-s</strong> sequenceFile</p></li><li><p>Provides a file containing single-end reads.</p></li><li><p>Outputs <strong>-o</strong> outputDirectory</p></li><li><p>Specifies the directory for outputted files. Default is RayOutput</p></li><li><p>Assembly options (defaults work well) <strong>-disable-recycling</strong></p></li><li><p>Disables read recycling during the assembly reads will be set free in 3 cases: 1. the distance did not match for a pair 2. the read has not met its mate 3. the library population indicates a wrong placement see Constrained traversal of repeats with paired sequences. Sebastien Boisvert, Elenie Godzaridis, Francois Laviolette & Jacques Corbeil. First Annual RECOMB Satellite Workshop on Massively Parallel Sequencing, March 26-27 2011, Vancouver, BC, Canada. <strong>-disable-scaffolder</strong></p></li><li><p>Disables the scaffolder. <strong>-minimum-contig-length</strong> minimumContigLength</p></li><li><p>Changes the minimum contig length, default is 100 nucleotides <strong>-color-space</strong></p></li><li><p>Runs in color-space Needs csfasta files. Activated automatically if csfasta files are provided. <strong>-use-maximum-seed-coverage</strong> maximumSeedCoverageDepth</p></li><li><p>Ignores any seed with a coverage depth above this threshold. The default is 4294967295. <strong>-use-minimum-seed-coverage</strong> minimumSeedCoverageDepth</p></li><li><p>Sets the minimum seed coverage depth. Any path with a coverage depth lower than this will be discarded. The default is 0.</p></li><li><p>Distributed storage engine (all these values are for each MPI rank) <strong>-bloom-filter-bits</strong> bits</p></li><li><p>Sets the number of bits for the Bloom filter Default is 268435456 bits, 0 bits disables the Bloom filter. <strong>-hash-table-buckets</strong> buckets</p></li><li><p>Sets the initial number of buckets. Must be a power of 2 ! Default value: 268435456 <strong>-hash-table-buckets-per-group</strong> buckets</p></li><li><p>Sets the number of buckets per group for sparse storage Default value: 64, Must be between &gt;=1 and &lt;= 64 <strong>-hash-table-load-factor-threshold</strong> threshold</p></li><li><p>Sets the load factor threshold for real-time resizing Default value: 0.75, must be &gt;= 0.5 and &lt; 1 <strong>-hash-table-verbosity</strong></p></li><li><p>Activates verbosity for the distributed storage engine</p></li><li><p>Biological abundances <strong>-search</strong> searchDirectory</p></li><li><p>Provides a directory containing fasta files to be searched in the de Bruijn graph. Biological abundances will be written to RayOutput/BiologicalAbundances See Documentation/BiologicalAbundances.txt <strong>-one-color-per-file</strong></p></li><li><p>Sets one color per file instead of one per sequence. By default, each sequence in each file has a different color. For files with large numbers of sequences, using one single color per file may be more efficient.</p></li><li><p>Taxonomic profiling with colored de Bruijn graphs <strong>-with-taxonomy</strong> Genome-to-Taxon.tsv TreeOfLife-Edges.tsv Taxon-Names.tsv</p></li><li><p>Provides a taxonomy. Computes and writes detailed taxonomic profiles. See Documentation/Taxonomy.txt for details.</p></li>
</ul>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-gene-ontology</strong> OntologyTerms.txt</p>
  </dt>
  <dd>
    <p>Annotations.txt</p><ul>
<li><p>Provides an ontology and annotations. OntologyTerms.txt is fetched from http://geneontology.org Annotations.txt is a 2-column file (EMBL_CDS handle       &       gene ontology identifier) See Documentation/GeneOntology.txt</p></li><li><p>Other outputs <strong>-enable-neighbourhoods</strong></p></li><li><p>Computes contig neighborhoods in the de Bruijn graph Output file: RayOutput/NeighbourhoodRelations.txt <strong>-amos</strong></p></li><li><p>Writes the AMOS file called RayOutput/AMOS.afg An AMOS file contains read positions on contigs. Can be opened with software with graphical user interface. <strong>-write-kmers</strong></p></li><li><p>Writes k-mer graph to RayOutput/kmers.txt The resulting file is not utilised by Ray. The resulting file is very large. <strong>-write-read-markers</strong></p></li><li><p>Writes read markers to disk. <strong>-write-seeds</strong></p></li><li><p>Writes seed DNA sequences to RayOutput/Rank&lt;rank&gt;.RaySeeds.fasta <strong>-write-extensions</strong></p></li><li><p>Writes extension DNA sequences to RayOutput/Rank&lt;rank&gt;.RayExtensions.fasta <strong>-write-contig-paths</strong></p></li><li><p>Writes contig paths with coverage values to RayOutput/Rank&lt;rank&gt;.RayContigPaths.txt <strong>-write-marker-summary</strong></p></li><li><p>Writes marker statistics.</p></li><li><p>Memory usage <strong>-show-memory-usage</strong></p></li><li><p>Shows memory usage. Data is fetched from /proc on GNU/Linux Needs __linux__ <strong>-show-memory-allocations</strong></p></li><li><p>Shows memory allocation events</p></li><li><p>Algorithm verbosity <strong>-show-extension-choice</strong></p></li><li><p>Shows the choice made (with other choices) during the extension. <strong>-show-ending-context</strong></p></li><li><p>Shows the ending context of each extension. Shows the children of the vertex where extension was too difficult. <strong>-show-distance-summary</strong></p></li><li><p>Shows summary of outer distances used for an extension path. <strong>-show-consensus</strong></p></li><li><p>Shows the consensus when a choice is done.</p></li><li><p>Checkpointing <strong>-write-checkpoints</strong> checkpointDirectory</p></li><li><p>Write checkpoint files <strong>-read-checkpoints</strong> checkpointDirectory</p></li><li><p>Read checkpoint files <strong>-read-write-checkpoints</strong> checkpointDirectory</p></li><li><p>Read and write checkpoint files</p></li><li><p>Message routing for large number of cores <strong>-route-messages</strong></p></li><li><p>Enables the Ray message router. Disabled by default. Messages will be routed accordingly so that any rank can communicate directly with only a few others. Without <strong>-route-messages</strong>, any rank can communicate directly with any other rank. Files generated: Routing/Connections.txt, Routing/Routes.txt and Routing/RelayEvents.txt and Routing/Summary.txt <strong>-connection-type</strong> type</p></li><li><p>Sets the connection type for routes. Accepted values are debruijn, hypercube, polytope, group, random, kautz and complete. Default is debruijn.</p></li><li><p>debruijn: a full de Bruijn graph a given alphabet and diameter hypercube: a hypercube, alphabet is {0,1} and the vertices is a power of 2 polytope: a convex regular polytope, alphabet is {0,1,...,B-1} and the vertices is a power of B group: silly model where one representative per group can communicate with outsiders random: Erdos-Renyi model kautz: a full de Kautz graph, which is a subgraph of a de Bruijn graph complete: a full graph with all the possible connections</p></li><li><p>With the type debruijn, the number of ranks must be a power of something. Examples: 256 = 16*16, 512=8*8*8, 49=7*7, and so on. Otherwise, don't use debruijn routing but use another one With the type kautz, the number of ranks n must be n=(k+1)*k^(d-1) for some k and d <strong>-routing-graph-degree</strong> degree</p></li><li><p>Specifies the outgoing degree for the routing graph. See Documentation/Routing.txt</p></li><li><p>Hardware testing <strong>-test-network-only</strong></p></li><li><p>Tests the network and returns. <strong>-write-network-test-raw-data</strong></p></li><li><p>Writes one additional file per rank detailing the network test. <strong>-exchanges</strong> NumberOfExchanges</p></li><li><p>Sets the number of exchanges <strong>-disable-network-test</strong></p></li><li><p>Skips the network test.</p></li><li><p>Debugging <strong>-verify-message-integrity</strong></p></li><li><p>Checks message data reliability for any non-empty message. add '-D CONFIG_SSE_4_2' in the Makefile to use hardware instruction (SSE 4.2) <strong>-run-profiler</strong></p></li><li><p>Runs the profiler as the code runs. By default, only show granularity warnings. Running the profiler increases running times. <strong>-with-profiler-details</strong></p></li><li><p>Shows number of messages sent and received in each methods during in each time slices (epochs). Needs <strong>-run-profiler</strong>. <strong>-show-communication-events</strong></p></li><li><p>Shows all messages sent and received. <strong>-show-read-placement</strong></p></li><li><p>Shows read placement in the graph during the extension. <strong>-debug-bubbles</strong></p></li><li><p>Debugs bubble code. Bubbles can be due to heterozygous sites or sequencing errors or other (unknown) events <strong>-debug-seeds</strong></p></li><li><p>Debugs seed code. Seeds are paths in the graph that are likely unique. <strong>-debug-fusions</strong></p></li><li><p>Debugs fusion code. <strong>-debug-scaffolder</strong></p></li><li><p>Debug the scaffolder.</p></li>
</ul>
  </dd>

</dl>
<p>FILES</p><ul>
<li><p>Input files</p></li><li><p>Note: file format is determined with file extension.</p></li><li><p>.fasta .fasta.gz (needs HAVE_LIBZ=y at compilation) .fasta.bz2 (needs HAVE_LIBBZ2=y at compilation) .fastq .fastq.gz (needs HAVE_LIBZ=y at compilation) .fastq.bz2 (needs HAVE_LIBBZ2=y at compilation) .sff (paired reads must be extracted manually) .csfasta (color-space reads)</p></li><li><p>Outputted files</p></li><li><p>Scaffolds</p></li><li><p>RayOutput/Scaffolds.fasta</p></li><li><p>The scaffold sequences in FASTA format</p></li><li><p>RayOutput/ScaffoldComponents.txt</p></li><li><p>The components of each scaffold</p></li><li><p>RayOutput/ScaffoldLengths.txt</p></li><li><p>The length of each scaffold</p></li><li><p>RayOutput/ScaffoldLinks.txt</p></li><li><p>Scaffold links</p></li><li><p>Contigs</p></li><li><p>RayOutput/Contigs.fasta</p></li><li><p>Contiguous sequences in FASTA format</p></li><li><p>RayOutput/ContigLengths.txt</p></li><li><p>The lengths of contiguous sequences</p></li><li><p>Summary</p></li><li><p>RayOutput/OutputNumbers.txt</p></li><li><p>Overall numbers for the assembly</p></li><li><p>de Bruijn graph</p></li><li><p>RayOutput/CoverageDistribution.txt</p></li><li><p>The distribution of coverage values</p></li><li><p>RayOutput/CoverageDistributionAnalysis.txt</p></li><li><p>Analysis of the coverage distribution</p></li><li><p>RayOutput/degreeDistribution.txt</p></li><li><p>Distribution of ingoing and outgoing degrees</p></li><li><p>RayOutput/kmers.txt</p></li><li><p>k-mer graph, required option: <strong>-write-kmers</strong></p></li><li><p>The resulting file is not utilised by Ray. The resulting file is very large.</p></li><li><p>Assembly steps</p></li><li><p>RayOutput/SeedLengthDistribution.txt</p></li><li><p>Distribution of seed length</p></li><li><p>RayOutput/Rank&lt;rank&gt;.OptimalReadMarkers.txt</p></li><li><p>Read markers.</p></li><li><p>RayOutput/Rank&lt;rank&gt;.RaySeeds.fasta</p></li><li><p>Seed DNA sequences, required option: <strong>-write-seeds</strong></p></li><li><p>RayOutput/Rank&lt;rank&gt;.RayExtensions.fasta</p></li><li><p>Extension DNA sequences, required option: <strong>-write-extensions</strong></p></li><li><p>RayOutput/Rank&lt;rank&gt;.RayContigPaths.txt</p></li><li><p>Contig paths with coverage values, required option: <strong>-write-contig-paths</strong></p></li><li><p>Paired reads</p></li><li><p>RayOutput/LibraryStatistics.txt</p></li><li><p>Estimation of outer distances for paired reads</p></li><li><p>RayOutput/Library&lt;LibraryNumber&gt;.txt</p></li><li><p>Frequencies for observed outer distances (insert size + read lengths)</p></li><li><p>Partition</p></li><li><p>RayOutput/NumberOfSequences.txt</p></li><li><p>Number of reads in each file</p></li><li><p>RayOutput/SequencePartition.txt</p></li><li><p>Sequence partition</p></li><li><p>Ray software</p></li><li><p>RayOutput/RayVersion.txt</p></li><li><p>The version of Ray</p></li><li><p>RayOutput/RayCommand.txt</p></li><li><p>The exact same command provided</p></li><li><p>AMOS</p></li><li><p>RayOutput/AMOS.afg</p></li><li><p>Assembly representation in AMOS format, required option: <strong>-amos</strong></p></li><li><p>Communication</p></li><li><p>RayOutput/MessagePassingInterface.txt</p></li><li><p>Number of messages sent</p></li><li><p>RayOutput/NetworkTest.txt</p></li><li><p>Latencies in microseconds</p></li><li><p>RayOutput/Rank&lt;rank&gt;NetworkTestData.txt</p></li><li><p>Network test raw data</p></li>
</ul><p>DOCUMENTATION</p><ul>
<li><p>- mpiexec <strong>-n</strong> 1 Ray <strong>-help</strong>|less (always up-to-date) - This help page (always up-to-date) - The directory Documentation/ - Manual (Portable Document Format): InstructionManual.tex (in Documentation) - Mailing list archives: http://sourceforge.net/mailarchive/forum.php?forum_name=denovoassembler-users</p></li>
</ul><p>AUTHOR</p><ul>
<li><p>Written by Sebastien Boisvert.</p></li>
</ul><p>REPORTING BUGS</p><ul>
<li><p>Report bugs to denovoassembler-users@lists.sourceforge.net Home page: &lt;http://denovoassembler.sourceforge.net/&gt;</p></li>
</ul><p>COPYRIGHT</p><ul>
<li><p>This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.</p></li><li><p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.</p></li><li><p>You have received a copy of the GNU General Public License along with this program (see LICENSE).</p></li>
</ul><p>Ray 2.1.0</p><p>License for Ray: GNU General Public License version 3 RayPlatform version: 1.1.0 License for RayPlatform: GNU Lesser General Public License version 3</p><p>MAXKMERLENGTH: 32 KMER_U64_ARRAY_SIZE: 1 Maximum coverage depth stored by CoverageDepth: 4294967295 MAXIMUM_MESSAGE_SIZE_IN_BYTES: 4000 bytes FORCE_PACKING = n ASSERT = n HAVE_LIBZ = y HAVE_LIBBZ2 = y CONFIG_PROFILER_COLLECT = n CONFIG_CLOCK_GETTIME = n __linux__ = y _MSC_VER = n __GNUC__ = y RAY_32_BITS = n RAY_64_BITS = y MPI standard version: MPI 2.1 MPI library: Open-MPI 1.4.2 Compiler: GNU gcc/g++ 4.4.5</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="RTPredict.1.html"><span aria-hidden="true">&larr;</span> RTPredict.1: Library for lc/ms data management and analysis - tools</a></li>
   <li class="next"><a href="RebaseTensorImage.1.html">RebaseTensorImage.1: Part of ants registration suite <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
