<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>razers: Fast read mapping with sensitivity control  synopsis razers [options] genome file reads file razers [options] genome file mp-reads file1 mp-reads file2  description razers is a versatile full-sensitive read mapper based on a k-mer counting filter. it supports single and paired-end mapping, and optimally parametrizes the filter based on a user-defined minimal sensitivity. see http://www.seqan.de/projects/razers for more information. input to razers is a reference genome file and either one file with single-end reads or two files containing left or right mates of paired-end reads. (c) copyright 2009 by david weese. -h, --help displays this help message. --version display version information main options: -f, --forward map reads only to forward strands. -r, --reverse map reads only to reverse strands. -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num percent recognition rate. in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -o, --output file change output filename. default: reads file.razers. valid filetypes are: razers, eland, fa, fasta, and gff. -v, --verbose verbose mode. -vv, --vverbose very verbose mode. output format options: -a, --alignment dump the alignment for each match (only razer or fasta format). -pa, --purge-ambiguous purge reads with more than max-hits best matches. -dr, --distance-range num only consider matches with at most num more errors compared to the best. default: output all. -gn, --genome-naming num select how genomes are named (see naming section below). in range [0..1]. default: 0. -rn, --read-naming num select how reads are named (see naming section below). in range [0..2]. default: 0. -so, --sort-order num select how matches are sorted (see sorting section below). in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). in range [0..1]. default: 0. filtration options: -s, --shape bitstring manually set k-mer shape. default: 11111111111. -t, --threshold num manually set minimum k-mer count threshold. in range [1..inf]. -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime. verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file. -mcl, --min-clipped-len num set minimal read length for read clipping. in range [0..inf]. default: 0. -qih, --quality-in-header quality string in fasta header.  formats, naming, sorting, and coordinate schemes razers supports various output formats. the output format is detected automatically from the file name suffix. .razers razer format .fa, .fasta enhanced fasta format .eland eland format</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Fast read mapping with sensitivity control  synopsis razers [options] genome file reads file razers [options] genome file mp-reads file1 mp-reads file2  description razers is a versatile full-sensitive read mapper based on a k-mer counting filter. it supports single and paired-end mapping, and optimally parametrizes the filter based on a user-defined minimal sensitivity. see http://www.seqan.de/projects/razers for more information. input to razers is a reference genome file and either one file with single-end reads or two files containing left or right mates of paired-end reads. (c) copyright 2009 by david weese. -h, --help displays this help message. --version display version information main options: -f, --forward map reads only to forward strands. -r, --reverse map reads only to reverse strands. -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num percent recognition rate. in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -o, --output file change output filename. default: reads file.razers. valid filetypes are: razers, eland, fa, fasta, and gff. -v, --verbose verbose mode. -vv, --vverbose very verbose mode. output format options: -a, --alignment dump the alignment for each match (only razer or fasta format). -pa, --purge-ambiguous purge reads with more than max-hits best matches. -dr, --distance-range num only consider matches with at most num more errors compared to the best. default: output all. -gn, --genome-naming num select how genomes are named (see naming section below). in range [0..1]. default: 0. -rn, --read-naming num select how reads are named (see naming section below). in range [0..2]. default: 0. -so, --sort-order num select how matches are sorted (see sorting section below). in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). in range [0..1]. default: 0. filtration options: -s, --shape bitstring manually set k-mer shape. default: 11111111111. -t, --threshold num manually set minimum k-mer count threshold. in range [1..inf]. -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime. verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file. -mcl, --min-clipped-len num set minimal read length for read clipping. in range [0..inf]. default: 0. -qih, --quality-in-header quality string in fasta header.  formats, naming, sorting, and coordinate schemes razers supports various output formats. the output format is detected automatically from the file name suffix. .razers razer format .fa, .fasta enhanced fasta format .eland eland format">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="razers (1) manual">
  <meta name="twitter:description" content="Fast read mapping with sensitivity control  synopsis razers [options] genome file reads file razers [options] genome file mp-reads file1 mp-reads file2  description razers is a versatile full-sensitive read mapper based on a k-mer counting filter. it supports single and paired-end mapping, and optimally parametrizes the filter based on a user-defined minimal sensitivity. see http://www.seqan.de/projects/razers for more information. input to razers is a reference genome file and either one file with single-end reads or two files containing left or right mates of paired-end reads. (c) copyright 2009 by david weese. -h, --help displays this help message. --version display version information main options: -f, --forward map reads only to forward strands. -r, --reverse map reads only to reverse strands. -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num percent recognition rate. in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -o, --output file change output filename. default: reads file.razers. valid filetypes are: razers, eland, fa, fasta, and gff. -v, --verbose verbose mode. -vv, --vverbose very verbose mode. output format options: -a, --alignment dump the alignment for each match (only razer or fasta format). -pa, --purge-ambiguous purge reads with more than max-hits best matches. -dr, --distance-range num only consider matches with at most num more errors compared to the best. default: output all. -gn, --genome-naming num select how genomes are named (see naming section below). in range [0..1]. default: 0. -rn, --read-naming num select how reads are named (see naming section below). in range [0..2]. default: 0. -so, --sort-order num select how matches are sorted (see sorting section below). in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). in range [0..1]. default: 0. filtration options: -s, --shape bitstring manually set k-mer shape. default: 11111111111. -t, --threshold num manually set minimum k-mer count threshold. in range [1..inf]. -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime. verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file. -mcl, --min-clipped-len num set minimal read length for read clipping. in range [0..inf]. default: 0. -qih, --quality-in-header quality string in fasta header.  formats, naming, sorting, and coordinate schemes razers supports various output formats. the output format is detected automatically from the file name suffix. .razers razer format .fa, .fasta enhanced fasta format .eland eland format">
  <meta name="twitter:image" content="https://www.carta.tech/images/seqan-apps-razers-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/razers.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="razers (1) manual" />
  <meta property="og:description" content="Fast read mapping with sensitivity control  synopsis razers [options] genome file reads file razers [options] genome file mp-reads file1 mp-reads file2  description razers is a versatile full-sensitive read mapper based on a k-mer counting filter. it supports single and paired-end mapping, and optimally parametrizes the filter based on a user-defined minimal sensitivity. see http://www.seqan.de/projects/razers for more information. input to razers is a reference genome file and either one file with single-end reads or two files containing left or right mates of paired-end reads. (c) copyright 2009 by david weese. -h, --help displays this help message. --version display version information main options: -f, --forward map reads only to forward strands. -r, --reverse map reads only to reverse strands. -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num percent recognition rate. in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -o, --output file change output filename. default: reads file.razers. valid filetypes are: razers, eland, fa, fasta, and gff. -v, --verbose verbose mode. -vv, --vverbose very verbose mode. output format options: -a, --alignment dump the alignment for each match (only razer or fasta format). -pa, --purge-ambiguous purge reads with more than max-hits best matches. -dr, --distance-range num only consider matches with at most num more errors compared to the best. default: output all. -gn, --genome-naming num select how genomes are named (see naming section below). in range [0..1]. default: 0. -rn, --read-naming num select how reads are named (see naming section below). in range [0..2]. default: 0. -so, --sort-order num select how matches are sorted (see sorting section below). in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). in range [0..1]. default: 0. filtration options: -s, --shape bitstring manually set k-mer shape. default: 11111111111. -t, --threshold num manually set minimum k-mer count threshold. in range [1..inf]. -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime. verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file. -mcl, --min-clipped-len num set minimal read length for read clipping. in range [0..inf]. default: 0. -qih, --quality-in-header quality string in fasta header.  formats, naming, sorting, and coordinate schemes razers supports various output formats. the output format is detected automatically from the file name suffix. .razers razer format .fa, .fasta enhanced fasta format .eland eland format" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/seqan-apps-razers-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">razers<small> (1)</small></h1>
        <p class="lead">Fast read mapping with sensitivity control  synopsis razers [options] genome file reads file razers [options] genome file mp-reads file1 mp-reads file2  description razers is a versatile full-sensitive read mapper based on a k-mer counting filter. it supports single and paired-end mapping, and optimally parametrizes the filter based on a user-defined minimal sensitivity. see http://www.seqan.de/projects/razers for more information. input to razers is a reference genome file and either one file with single-end reads or two files containing left or right mates of paired-end reads. (c) copyright 2009 by david weese. -h, --help displays this help message. --version display version information main options: -f, --forward map reads only to forward strands. -r, --reverse map reads only to reverse strands. -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num percent recognition rate. in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -o, --output file change output filename. default: reads file.razers. valid filetypes are: razers, eland, fa, fasta, and gff. -v, --verbose verbose mode. -vv, --vverbose very verbose mode. output format options: -a, --alignment dump the alignment for each match (only razer or fasta format). -pa, --purge-ambiguous purge reads with more than max-hits best matches. -dr, --distance-range num only consider matches with at most num more errors compared to the best. default: output all. -gn, --genome-naming num select how genomes are named (see naming section below). in range [0..1]. default: 0. -rn, --read-naming num select how reads are named (see naming section below). in range [0..2]. default: 0. -so, --sort-order num select how matches are sorted (see sorting section below). in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). in range [0..1]. default: 0. filtration options: -s, --shape bitstring manually set k-mer shape. default: 11111111111. -t, --threshold num manually set minimum k-mer count threshold. in range [1..inf]. -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime. verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file. -mcl, --min-clipped-len num set minimal read length for read clipping. in range [0..inf]. default: 0. -qih, --quality-in-header quality string in fasta header.  formats, naming, sorting, and coordinate schemes razers supports various output formats. the output format is detected automatically from the file name suffix. .razers razer format .fa, .fasta enhanced fasta format .eland eland format</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/razers.1.html">
      <span itemprop="name">razers: Fast read mapping with sensitivity control  synopsis razers [options] genome file reads file razers [options] genome file mp-reads file1 mp-reads file2  description razers is a versatile full-sensitive read mapper based on a k-mer counting filter. it supports single and paired-end mapping, and optimally parametrizes the filter based on a user-defined minimal sensitivity. see http://www.seqan.de/projects/razers for more information. input to razers is a reference genome file and either one file with single-end reads or two files containing left or right mates of paired-end reads. (c) copyright 2009 by david weese. -h, --help displays this help message. --version display version information main options: -f, --forward map reads only to forward strands. -r, --reverse map reads only to reverse strands. -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num percent recognition rate. in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -o, --output file change output filename. default: reads file.razers. valid filetypes are: razers, eland, fa, fasta, and gff. -v, --verbose verbose mode. -vv, --vverbose very verbose mode. output format options: -a, --alignment dump the alignment for each match (only razer or fasta format). -pa, --purge-ambiguous purge reads with more than max-hits best matches. -dr, --distance-range num only consider matches with at most num more errors compared to the best. default: output all. -gn, --genome-naming num select how genomes are named (see naming section below). in range [0..1]. default: 0. -rn, --read-naming num select how reads are named (see naming section below). in range [0..2]. default: 0. -so, --sort-order num select how matches are sorted (see sorting section below). in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). in range [0..1]. default: 0. filtration options: -s, --shape bitstring manually set k-mer shape. default: 11111111111. -t, --threshold num manually set minimum k-mer count threshold. in range [1..inf]. -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime. verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file. -mcl, --min-clipped-len num set minimal read length for read clipping. in range [0..inf]. default: 0. -qih, --quality-in-header quality string in fasta header.  formats, naming, sorting, and coordinate schemes razers supports various output formats. the output format is detected automatically from the file name suffix. .razers razer format .fa, .fasta enhanced fasta format .eland eland format</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/seqan-apps/">
      <span itemprop="name">seqan-apps</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/razers.1.html">
      <span itemprop="name">razers: Fast read mapping with sensitivity control  synopsis razers [options] genome file reads file razers [options] genome file mp-reads file1 mp-reads file2  description razers is a versatile full-sensitive read mapper based on a k-mer counting filter. it supports single and paired-end mapping, and optimally parametrizes the filter based on a user-defined minimal sensitivity. see http://www.seqan.de/projects/razers for more information. input to razers is a reference genome file and either one file with single-end reads or two files containing left or right mates of paired-end reads. (c) copyright 2009 by david weese. -h, --help displays this help message. --version display version information main options: -f, --forward map reads only to forward strands. -r, --reverse map reads only to reverse strands. -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num percent recognition rate. in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -o, --output file change output filename. default: reads file.razers. valid filetypes are: razers, eland, fa, fasta, and gff. -v, --verbose verbose mode. -vv, --vverbose very verbose mode. output format options: -a, --alignment dump the alignment for each match (only razer or fasta format). -pa, --purge-ambiguous purge reads with more than max-hits best matches. -dr, --distance-range num only consider matches with at most num more errors compared to the best. default: output all. -gn, --genome-naming num select how genomes are named (see naming section below). in range [0..1]. default: 0. -rn, --read-naming num select how reads are named (see naming section below). in range [0..2]. default: 0. -so, --sort-order num select how matches are sorted (see sorting section below). in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). in range [0..1]. default: 0. filtration options: -s, --shape bitstring manually set k-mer shape. default: 11111111111. -t, --threshold num manually set minimum k-mer count threshold. in range [1..inf]. -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime. verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file. -mcl, --min-clipped-len num set minimal read length for read clipping. in range [0..inf]. default: 0. -qih, --quality-in-header quality string in fasta header.  formats, naming, sorting, and coordinate schemes razers supports various output formats. the output format is detected automatically from the file name suffix. .razers razer format .fa, .fasta enhanced fasta format .eland eland format</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    <nav>
  <ul class="pager">
   <li class="previous"><a href="raxmlHPC-PTHREADS-SSE3.1.html"><span aria-hidden="true">&larr;</span> raxmlHPC-PTHREADS-SSE3.1: Randomized axelerated maximum likelihood</a></li>
   <li class="next"><a href="razers3.1.html">razers3.1: Faster, fully sensitive read mapping  synopsis razers3 [options] genome file reads file razers3 [options] genome file pe-reads file1 pe-reads file2  description razers 3 is a versatile full-sensitive read mapper based on k-mer counting and seeding filters. it supports single and paired-end mapping, shared-memory parallelism, and optimally parametrizes the filter based on a user-defined minimal sensitivity. see http://www.seqan.de/projects/razers for more information. input to razers 3 is a reference genome file and either one file with single-end reads or two files containing left or right mates of paired-end reads. (c) copyright 2009-2013 by david weese. -h, --help displays this help message. --version display version information main options: -i, --percent-identity num percent identity threshold. in range [50..100]. default: 95. -rr, --recognition-rate num percent recognition rate. in range [80..100]. default: 99. -ng, --no-gaps allow only mismatches, no indels. default: allow both. -f, --forward map reads only to forward strands. -r, --reverse map reads only to reverse strands. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -o, --output file mapping result filename. default: reads file.razers. valid filetypes are: .razers, .eland, .fa, .fasta, .gff, .sam, and .afg. -v, --verbose verbose mode. -vv, --vverbose very verbose mode. paired-end options: -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. output format options: -a, --alignment dump the alignment for each match (only razer or fasta format). -pa, --purge-ambiguous purge reads with more than max-hits best matches. -dr, --distance-range num only consider matches with at most num more errors compared to the best. default: output all. -gn, --genome-naming num select how genomes are named (see naming section below). in range [0..1]. default: 0. -rn, --read-naming num select how reads are named (see naming section below). in range [0..3]. default: 0. --full-readid use the whole read id (don't clip after whitespace). -so, --sort-order num select how matches are sorted (see sorting section below). in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). in range [0..1]. default: 0. -ds, --dont-shrink-alignments disable alignment shrinking in sam. this is required for generating a gold mapping for rabema. filtration options: -fl, --filter str select k-mer filter. one of pigeonhole and swift. default: pigeonhole. -mr, --mutation-rate num set the percent mutation rate (pigeonhole). in range [0..20]. default: 5. -ol, --overlap-length num manually set the overlap length of adjacent k-mers (pigeonhole). in range [0..inf]. -pd, --param-dir dir read user-computed parameter files in the directory dir (swift). -t, --threshold num manually set minimum k-mer count threshold (swift). in range [1..inf]. -tl, --taboo-length num set taboo length (swift). in range [1..inf]. default: 1. -s, --shape bitstring manually set k-mer shape. -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. default: 1. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -lf, --load-factor num set the load factor for the open addressing k-mer index. in range [1..inf]. default: 1.6. verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file. -mf, --mismatch-file file write mismatch patterns to file. misc options: -cm, --compact-mult num multiply compaction treshold by this value after reaching and compacting. in range [0..inf]. default: 2.2. -ncf, --no-compact-frac num don't compact if in this last fraction of genome. in range [0..1]. default: 0.05. parallelism options: -pws, --parallel-window-size num collect candidates in windows of this length. in range [1..inf]. default: 500000. -pvs, --parallel-verification-size num verify candidates in packages of this size. in range [1..inf]. default: 100. -pvmpc, --parallel-verification-max-package-count num largest number of packages to create for verification per thread-1. in range [1..inf]. default: 100. -amms, --available-matches-memory-size num bytes of main memory available for storing matches. in range [-1..inf]. default: 0. -mhst, --match-histo-start-threshold num when to start histogram. in range [1..inf]. default: 5.  formats, naming, sorting, and coordinate schemes razers 3 supports various output formats. the output format is detected automatically from the file name suffix. .razers razer format .fa, .fasta enhanced fasta format .eland eland format <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
