<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>readseq: Reads and writes nucleic/protein sequences in various formats</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Reads and writes nucleic/protein sequences in various formats">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="readseq (1) manual">
  <meta name="twitter:description" content="Reads and writes nucleic/protein sequences in various formats">
  <meta name="twitter:image" content="https://www.carta.tech/images/readseq-readseq-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/readseq.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="readseq (1) manual" />
  <meta property="og:description" content="Reads and writes nucleic/protein sequences in various formats" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/readseq-readseq-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">readseq<small> (1)</small></h1>
        <p class="lead">Reads and writes nucleic/protein sequences in various formats</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/readseq.1.html">
      <span itemprop="name">readseq: Reads and writes nucleic/protein sequences in various formats</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/readseq/">
      <span itemprop="name">readseq</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/readseq.1.html">
      <span itemprop="name">readseq: Reads and writes nucleic/protein sequences in various formats</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>readseq</strong> <em>[-options] in.seq &gt; out.seq</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This manual page documents briefly the <strong>readseq</strong> command. This manual page was written for the Debian GNU/Linux distribution because the original program does not have a manual page. Instead, it has documentation in text form, see below.</p><p><strong>readseq</strong> reads and writes biosequences (nucleic/protein) in various formats. Data files may have multiple sequences. <strong>readseq</strong> is particularly useful as it automatically detects many sequence formats, and interconverts among them.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FORMATS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p>Formats which readseq currently understands:</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * IG/Stanford, used by Intelligenetics and others
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * GenBank/GB, genbank flatfile format
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * NBRF format
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * EMBL, EMBL flatfile format
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * GCG, single sequence format of GCG software
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * DNAStrider, for common Mac program
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * Fitch format, limited use
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * Pearson/Fasta, a common format used by Fasta programs and others
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * Zuker format, limited use. Input only.
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * Olsen, format printed by Olsen VMS sequence editor. Input only.
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * Phylip3.2, sequential format for Phylip programs
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * Phylip, interleaved format for Phylip programs (v3.3, v3.4)
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  * Plain/Raw, sequence data only (no name, document, numbering)
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  + MSF multi sequence format used by GCG software
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  + PAUP's multiple sequence (NEXUS) format
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  + PIR/CODATA format used by PIR
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  + ASN.1 format used by NCBI
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  + Pretty print with various options for nice looking output. Output only.
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  + LinAll format, limited use (LinAll and ConStruct programs)
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  + Vienna format used by ViennaRNA programs
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    <p>See the included "Formats" file for detail on file formats.</p>
  </dt>
  <dd>
    
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-help</strong></p>
  </dt>
  <dd>
    <p>Show summary of options.</p>
  </dd>
  <dt>
    <p><strong>-a[ll]</strong></p>
  </dt>
  <dd>
    <p>Select All sequences</p>
  </dd>
  <dt>
    <p><strong>-c[aselower]</strong></p>
  </dt>
  <dd>
    <p>Change to lower case</p>
  </dd>
  <dt>
    <p><strong>-C[ASEUPPER]</strong></p>
  </dt>
  <dd>
    <p>Change to UPPER CASE</p>
  </dd>
  <dt>
    <p><strong>-degap[=-]</strong></p>
  </dt>
  <dd>
    <p>Remove gap symbols</p>
  </dd>
  <dt>
    <p><strong>-i[tem=2,3,4]</strong></p>
  </dt>
  <dd>
    <p>Select Item number(s) from several</p>
  </dd>
  <dt>
    <p><strong>-l[ist]</strong></p>
  </dt>
  <dd>
    <p>List sequences only</p>
  </dd>
  <dt>
    <p><strong>-o[utput=]out.seq</strong></p>
  </dt>
  <dd>
    <p>Redirect Output</p>
  </dd>
  <dt>
    <p><strong>-p[ipe]</strong></p>
  </dt>
  <dd>
    <p>Pipe (command line, &lt;stdin, &gt;stdout)</p>
  </dd>
  <dt>
    <p><strong>-r[everse]</strong></p>
  </dt>
  <dd>
    <p>Change to Reverse-complement</p>
  </dd>
  <dt>
    <p><strong>-v[erbose]</strong></p>
  </dt>
  <dd>
    <p>Verbose progress</p>
  </dd>
  <dt>
    <p><strong>-f[ormat=]# Format number for output, or</strong></p>
  </dt>
  <dd>
    
<pre>
    -f[ormat=]Name Format name for output:
    1. IG/Stanford           11. Phylip3.2
    2. GenBank/GB            12. Phylip
    3. NBRF                  13. Plain/Raw
    4. EMBL                  14. PIR/CODATA
    5. GCG                   15. MSF
    6. DNAStrider            16. ASN.1
    7. Fitch                 17. PAUP/NEXUS
    8. Pearson/Fasta         18. Pretty (out-only)
    9. Zuker (in-only)       19. LinAll
   10. Olsen (in-only)       20. Vienna
</pre>
<p>Pretty format options:</p>
  </dd>
  <dt>
    <p><strong>-wid[th]=#</strong></p>
  </dt>
  <dd>
    <p>Sequence line width</p>
  </dd>
  <dt>
    <p><strong>-tab=#</strong></p>
  </dt>
  <dd>
    <p>Left indent</p>
  </dd>
  <dt>
    <p><strong>-col[space]=#</strong></p>
  </dt>
  <dd>
    <p>Column space within sequence line on output</p>
  </dd>
  <dt>
    <p><strong>-gap[count]</strong></p>
  </dt>
  <dd>
    <p>Count gap chars in sequence numbers</p>
  </dd>
  <dt>
    <p><strong>-nameleft, -nameright[=#]</strong></p>
  </dt>
  <dd>
    <p>Name on left/right side [=max width]</p>
  </dd>
  <dt>
    <p><strong>-nametop</strong></p>
  </dt>
  <dd>
    <p>Name at top/bottom</p>
  </dd>
  <dt>
    <p><strong>-numleft, -numright</strong></p>
  </dt>
  <dd>
    <p>Seq index on left/right side</p>
  </dd>
  <dt>
    <p><strong>-numtop, -numbot</strong></p>
  </dt>
  <dd>
    <p>Index on top/bottom</p>
  </dd>
  <dt>
    <p><strong>-match[=.]</strong></p>
  </dt>
  <dd>
    <p>Use match base for 2..n species</p>
  </dd>
  <dt>
    <p><strong>-inter[line=#]</strong></p>
  </dt>
  <dd>
    <p>Blank line(s) between sequence blocks</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    
<pre>
  readseq
      -- for interactive use
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  readseq my.1st.seq  my.2nd.seq  -all  -format=genbank  -output=my.gb
      -- convert all of two input files to one genbank format output file
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  readseq my.seq -all -form=pretty -nameleft=3 -numleft -numright -numtop -match
      -- output to standard output a file in a pretty format
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  readseq my.seq -item=9,8,3,2 -degap -CASE -rev -f=msf -out=my.rev
      -- select 4 items from input, degap, reverse, and uppercase them
</pre>

  </dt>
  <dd>
    
  </dd>
  <dt>
    
<pre>
  cat *.seq | readseq -pipe -all -format=asn &gt; bunch-of.asn
      -- pipe a bunch of data thru readseq, converting all to asn
</pre>

  </dt>
  <dd>
    
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO readseq&hellip;</h2>
        <div class="sectioncontent">
<p>The programs are documented fully in text form. See the files in <em>/usr/share/doc/readseq</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>This manual page was written by Stephane Bortzmeyer &lt;bortzmeyer@debian.org&gt;, for the Debian GNU/Linux system (but may be used by others).</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="readpst.1.html"><span aria-hidden="true">&larr;</span> readpst.1: Convert pst (ms outlook personal folders) files to mbox and other formats</a></li>
   <li class="next"><a href="realcsh.c.1.html">realcsh.c.1: A shell for running c code <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
