<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>RTEvaluation: Library for lc/ms data management and analysis - tools</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Library for lc/ms data management and analysis - tools">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="RTEvaluation (1) manual">
  <meta name="twitter:description" content="Library for lc/ms data management and analysis - tools">
  <meta name="twitter:image" content="https://www.carta.tech/images/topp-RTEvaluation-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/RTEvaluation.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="RTEvaluation (1) manual" />
  <meta property="og:description" content="Library for lc/ms data management and analysis - tools" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/topp-RTEvaluation-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">RTEvaluation<small> (1)</small></h1>
        <p class="lead">Library for lc/ms data management and analysis - tools</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/RTEvaluation.1.html">
      <span itemprop="name">RTEvaluation: Library for lc/ms data management and analysis - tools</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/topp/">
      <span itemprop="name">topp</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/RTEvaluation.1.html">
      <span itemprop="name">RTEvaluation: Library for lc/ms data management and analysis - tools</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>This manual page documents briefly the <strong>topp</strong> package that brings a set of binary tools to build the OpenMS proteomics pipeline (topp). OpenMS offers an infrastructure for the development of mass spectrometry-related software and powerful 2D and 3D visualization solutions. OpenMS offers analyses for various quantitation protocols, including label-free quantitation, SILAC, iTRAQ, SRM, SWATH… It provides build-in algorithms for de-novo identification and database search, as well as adapters to other state-of-the art tools like X!Tandem, Mascot, OMSSA… OpenMS supports the Proteomics Standard Initiative (PSI) formats for MS data and supports easy integration of tools into workflow engines like Knime, Galaxy, WS-Pgrade, and TOPPAS via the TOPPtools concept and a unified parameter handling.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AVAILABLE MODULES</h2>
        <div class="sectioncontent">
<p><strong>AccurateMassSearch</strong> -- Find potential HMDB ids within the given mass error window.</p><p><strong>AdditiveSeries</strong> -- Computes an additive series to quantify a peptide in a set of samples.</p><p><strong>BaselineFilter</strong> -- Removes the baseline from profile spectra using a top-hat filter.</p><p><strong>CVInspector</strong> -- A tool for visualization and validation of PSI mapping and CV files.</p><p><strong>CompNovo</strong> -- Performs a de novo peptide identification using the CompNovo engine.</p><p><strong>CompNovoCID</strong> -- Performs a de novo peptide identification using the CompNovo engine.</p><p><strong>ConsensusID</strong> -- Computes a consensus identification from peptide identifications of several identification engines.</p><p><strong>ConsensusMapNormalizer</strong> -- Normalizes maps of one consensusXML file.</p><p><strong>ConvertTSVToTraML</strong> -- Converts an OpenSWATH transition TSV file to a TraML file.</p><p><strong>ConvertTraMLToTSV</strong> -- Converts a TraML file to an OpenSWATH transition TSV file.</p><p><strong>DBExporter</strong> -- Exports data from an OpenMS database to a file.</p><p><strong>DBImporter</strong> -- Imports data to an OpenMS database.</p><p><strong>DTAExtractor</strong> -- Extracts spectra of an MS run file to several files in DTA format.</p><p><strong>DeMeanderize</strong> -- Orders the spectra of MALDI spotting plates correctly.</p><p><strong>Decharger</strong> -- Decharges and merges different feature charge variants of the same peptide.</p><p><strong>DecoyDatabase</strong> -- Create decoy peptide databases from normal ones.</p><p><strong>Digestor</strong> -- Digests a protein database in-silico.</p><p><strong>DigestorMotif</strong> -- digests a protein database in-silico.</p><p><strong>EICExtractor</strong> -- Extracts intensities from dedicates positions in a LC/MS map.</p><p><strong>ERPairFinder</strong> -- Util which can be used to evaluate pair ratios on enhanced resolution (zoom) scans.</p><p><strong>ExecutePipeline</strong> -- Executes workflows created by TOPPAS.</p><p><strong>FFEval</strong> -- Evaluation tool for feature detection algorithms.</p><p><strong>FalseDiscoveryRate</strong> -- Estimates the false discovery rate on peptide and protein level using decoy searches.</p><p><strong>FeatureFinderCentroided</strong> -- Detects two-dimensional features in LC-MS data.</p><p><strong>FeatureFinderIsotopeWavelet</strong> -- Detects two-dimensional features in LC-MS data.</p><p><strong>FeatureFinderMRM</strong> -- Detects two-dimensional features in LC-MS data.</p><p><strong>FeatureFinderMetabo</strong> -- Assembles metabolite features from singleton mass traces.</p><p><strong>FeatureFinderRaw</strong> -- Determination of peak ratios in LC-MS data.</p><p><strong>FeatureFinderSuperHirn</strong> -- Finds mass spectrometric features in mass spectra.</p><p><strong>FeatureLinkerLabeled</strong> -- Groups corresponding isotope-labeled features in a feature map.</p><p><strong>FeatureLinkerUnlabeled</strong> -- Groups corresponding features from multiple maps.</p><p><strong>FeatureLinkerUnlabeledQT</strong> -- Groups corresponding features from multiple maps.</p><p><strong>FileConverter</strong> -- Converts between different MS file formats.</p><p><strong>FileFilter</strong> -- Extracts or manipulates portions of data from peak, feature or consensus-feature files.</p><p><strong>FileInfo</strong> -- Shows basic information about the file, such as data ranges and file type.</p><p><strong>FileMerger</strong> -- Merges several MS files into one file.</p><p><strong>FuzzyDiff</strong> -- Compares two files, tolerating numeric differences.</p><p><strong>GenericWrapper</strong> -- Allows the generic wrapping of external tools.</p><p><strong>HighResPrecursorMassCorrector</strong> -- Corrects the precursor mz determined by the instrument software.</p><p><strong>IDConflictResolver</strong> -- Resolves ambiguous annotations of features with peptide identifications.</p><p><strong>IDDecoyProbability</strong> -- Estimates peptide probabilities using a decoy search strategy.</p><p><strong>IDEvaluatorGUI</strong> -- Computes a 'q-value vs. #PSM' plot to visualize the number identifications for a certain q-value.</p><p><strong>IDExtractor</strong> -- Extracts 'n' peptides randomly or best 'n' from idXML files.</p><p><strong>IDFileConverter</strong> -- Converts identification engine file formats.</p><p><strong>IDFilter</strong> -- Filters results from protein or peptide identification engines based on different criteria.</p><p><strong>IDMapper</strong> -- Assigns protein/peptide identifications to features or consensus features.</p><p><strong>IDMassAccuracy</strong> -- Calculates a distribution of the mass error from given mass spectra and IDs.</p><p><strong>IDMerger</strong> -- Merges several protein/peptide identification files into one file.</p><p><strong>IDPosteriorErrorProbability</strong> -- Estimates probabilities for incorrectly assigned peptide sequences and a set of search engine scores using a mixture model.</p><p><strong>IDRTCalibration</strong> -- Can be used to calibrate RTs of peptide hits linearly to standards.</p><p><strong>IDRipper</strong> -- Split protein/peptide identification file into several files according annotated file origin.</p><p><strong>IDSplitter</strong> -- Splits protein/peptide identifications off of annotated data files.</p><p><strong>INIFileEditor</strong> -- An editor for OpenMS configuration files.</p><p><strong>INIUpdater</strong> -- Update INI and TOPPAS files to new OpenMS version.</p><p><strong>ITRAQAnalyzer</strong> -- Calculates iTRAQ quantitative values for peptides.</p><p><strong>ImageCreator</strong> -- Transforms an LC-MS map into an image.</p><p><strong>InclusionExclusionListCreator</strong> -- Creates inclusion and/or exclusion lists.</p><p><strong>InspectAdapter</strong> -- Annotates MS/MS spectra using Inspect.</p><p><strong>InternalCalibration</strong> -- Applies an internal calibration.</p><p><strong>IsobaricAnalyzer</strong> -- Calculates isobaric quantitative values for peptides.</p><p><strong>LabeledEval</strong> -- Evaluation tool for isotope-labeled quantitation experiments.</p><p><strong>MRMMapper</strong> -- MRMMapper maps measured chromatograms (mzML) and the transitions used (TraML).</p><p><strong>MRMPairFinder</strong> -- Util which can be used to evaluate labeled pair ratios on MRM features.</p><p><strong>MSSimulator</strong> -- A highly configurable simulator for mass spectrometry experiments.</p><p><strong>MapAlignerIdentification</strong> -- Corrects retention time distortions between maps based on common peptide identifications.</p><p><strong>MapAlignerPoseClustering</strong> -- Corrects retention time distortions between maps using a pose clustering approach.</p><p><strong>MapAlignerSpectrum</strong> -- Corrects retention time distortions between maps by spectrum alignment.</p><p><strong>MapAlignmentEvaluation</strong> -- Evaluates alignment results against a ground truth.</p><p><strong>MapNormalizer</strong> -- Normalizes peak intensities in an MS run.</p><p><strong>MapRTTransformer</strong> -- Applies retention time transformations to maps.</p><p><strong>MapStatistics</strong> -- Extract extended statistics on the features of a map for quality control.</p><p><strong>MascotAdapter</strong> -- Annotates MS/MS spectra using Mascot.</p><p><strong>MascotAdapterOnline</strong> -- Annotates MS/MS spectra using Mascot.</p><p><strong>MassCalculator</strong> -- Calculates masses and mass-to-charge ratios of peptide sequences.</p><p><strong>MassTraceExtractor</strong> -- Detects mass traces in centroided LC-MS data.</p><p><strong>MyriMatchAdapter</strong> -- Annotates MS/MS spectra using MyriMatch.</p><p><strong>MzTabExporter</strong> -- Exports various XML formats to an mzTab file.</p><p><strong>NoiseFilterGaussian</strong> -- Removes noise from profile spectra by using Gaussian filter (on uniform as well as non-uniform data).</p><p><strong>NoiseFilterSGolay</strong> -- Removes noise from profile spectra by using a Savitzky Golay filter. Requires uniform (equidistant) data.</p><p><strong>OMSSAAdapter</strong> -- Annotates MS/MS spectra using OMSSA.</p><p><strong>OpenSwathAnalyzer</strong> -- Picks peaks and finds features in an SRM experiment.</p><p><strong>OpenSwathChromatogramExtractor</strong> -- Extract chromatograms (XIC) from a MS2 map file.</p><p><strong>OpenSwathConfidenceScoring</strong> -- Compute confidence scores for OpenSwath results.</p><p><strong>OpenSwathDIAPreScoring</strong> -- Scoring spectra using the DIA scores.</p><p><strong>OpenSwathDecoyGenerator</strong> -- Generates decoys according to different models for a specific TraML.</p><p><strong>OpenSwathFeatureXMLToTSV</strong> -- Converts a featureXML to a mProphet tsv.</p><p><strong>OpenSwathMzMLFileCacher</strong> -- This tool caches the spectra and chromatogram data of an mzML to disk.</p><p><strong>OpenSwathRTNormalizer</strong> -- This tool will take a description of RT peptides and their normalized retention time to write out a transformation file on how to transoform the RT space into the normalized space.</p><p><strong>OpenSwathRewriteToFeatureXML</strong> -- Combines featureXML and mProphet tsv to FDR filtered featureXML.</p><p><strong>PILISIdentification</strong> -- performs a peptide/protein identification with the PILIS engine.</p><p><strong>PILISModelCV</strong> -- Perform a cross validation of the PILIS model parameters.</p><p><strong>PILISModelTrainer</strong> -- Train the PILIS model with a given set of spectra and identifications.</p><p><strong>PILISSpectraGenerator</strong> -- Generate spectra given a list of peptides and a PILIS model.</p><p><strong>PTModel</strong> -- Trains a model for the prediction of proteotypic peptides from a training set.</p><p><strong>PTPredict</strong> -- predicts the likelihood of peptides to be proteotypic via svm_model which is trained by PTModel.</p><p><strong>PeakPickerHiRes</strong> -- Finds mass spectrometric peaks in profile mass spectra.</p><p><strong>PeakPickerWavelet</strong> -- Finds mass spectrometric peaks in profile mass spectra.</p><p><strong>PepNovoAdapter</strong> -- Adapter to PepNovo supporting all PepNovo command line parameters. The results are converted from the PepNovo text outfile format into the idXML format.</p><p><strong>PeptideIndexer</strong> -- Refreshes the protein references for all peptide hits.</p><p><strong>PhosphoScoring</strong> -- Scores potential phosphorylation sites and therby tries to localize the most probable sites.</p><p><strong>PrecursorIonSelector</strong> -- PrecursorIonSelector.</p><p><strong>PrecursorMassCorrector</strong> -- Corrects the precursor entries of MS/MS spectra, by using MS1 information.</p><p><strong>ProteinInference</strong> -- Protein inference based on the number of identified peptides.</p><p><strong>ProteinQuantifier</strong> -- Compute peptide and protein abundances.</p><p><strong>ProteinResolver</strong> -- protein inference.</p><p><strong>QCCalculator</strong> -- Calculates basic quality parameters from MS experiments and subsequent analysis data as identification or feature detection.</p><p><strong>QCEmbedder</strong> -- Attaches a table or an image to a given qc parameter.</p><p><strong>QCExporter</strong> -- Will extract several qp from several run/sets in a tabular format.</p><p><strong>QCExtractor</strong> -- Extracts a table attachment to a given qc parameter.</p><p><strong>QCImporter</strong> -- Imports tables with quality control parameters into qcml files.</p><p><strong>QCMerger</strong> -- Merges two qcml files together.</p><p><strong>QCShrinker</strong> -- This application is used to remove the verbose table attachments from a qcml file that are not needed anymore, e.g. for a final report.</p><p><strong>RNPxl</strong> -- Tool for RNP cross linking experiment analysis.</p><p><strong>RNPxlXICFilter</strong> -- Remove MS2 spectra from treatment based on the fold change between control and treatment.</p><p><strong>RTEvaluation</strong> -- Application that evaluates TPs (true positives), TNs, FPs, and FNs for an idXML file with predicted RTs.</p><p><strong>RTModel</strong> -- Trains a model for the retention time prediction of peptides from a training set.</p><p><strong>RTPredict</strong> -- Predicts retention times for peptides using a model trained by RTModel.</p><p><strong>Resampler</strong> -- Transforms an LC/MS map into a resampled map or a PNG image.</p><p><strong>SILACAnalyzer</strong> -- Determination of peak ratios in LC-MS data.</p><p><strong>SeedListGenerator</strong> -- Generates seed lists for feature detection.</p><p><strong>SemanticValidator</strong> -- SemanticValidator for analysisXML and mzML files.</p><p><strong>SequenceCoverageCalculator</strong> -- Prints information about idXML files.</p><p><strong>SpecLibCreator</strong> -- Creates an MSP formatted spectral library.</p><p><strong>SpecLibSearcher</strong> -- Identifies peptide MS/MS spectra by spectral matching with a searchable spectral library.</p><p><strong>SpectraFilterBernNorm</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraFilterMarkerMower</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraFilterNLargest</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraFilterNormalizer</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraFilterParentPeakMower</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraFilterScaler</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraFilterSqrtMower</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraFilterThresholdMower</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraFilterWindowMower</strong> -- Applies thresholdfilter to peak spectra.</p><p><strong>SpectraMerger</strong> -- Merges spectra (each MS level separately), increasing S/N ratios.</p><p><strong>SvmTheoreticalSpectrumGeneratorTrainer</strong> -- Trainer for SVM models as input for SvmTheoreticalSpectrumGenerator.</p><p><strong>TMTAnalyzer</strong> -- Calculates TMT quantitative values for peptides.</p><p><strong>TOFCalibration</strong> -- Applies time of flight calibration.</p><p><strong>TOPPAS</strong> -- An assistant for GUI-driven TOPP workflow design.</p><p><strong>TOPPView</strong> -- A viewer for mass spectrometry data.</p><p><strong>TextExporter</strong> -- Exports various XML formats to a text file.</p><p><strong>TransformationEvaluation</strong> -- Applies a transformation to a range of values.</p><p><strong>XMLValidator</strong> -- Validates XML files against an XSD schema.</p><p><strong>XTandemAdapter</strong> -- Annotates MS/MS spectra using XTandem.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BIBLIOGRAPHICAL REFERENCE TO BE CITED</h2>
        <div class="sectioncontent">
<p>Marc Sturm, Andreas Bertsch, Clemens Gröpl, Andreas Hildebrandt, Rene Hussong, Eva Lange, Nico Pfeifer, Ole Schulz-Trieglaff, Alexandra Zerck, Knut Reinert, and Oliver Kohlbacher (2008)  OpenMS – an Open-Source Software Framework for Mass Spectrometry. <em> BMC</em> Bioinformatics, 9:163. doi:10.1186/1471-2105-9-163.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO RTEvaluation&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man3/libopenms.3.html"><strong>libopenms</strong>(3)</a>, <a href="../man7/openms-common.7.html"><strong>openms-common</strong>(7)</a>, <a href="../man7/openms.7.html"><strong>openms</strong>(7)</a>, <a href="../man7/openms-doc.7.html"><strong>openms-doc</strong>(7)</a>.</p><p>The program is documented fully in the <em>Tutorials</em>, that are packaged in openms-doc.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>This manual page was written by Filippo Rusconi &lt;lopippo@debian.org&gt;. Permission is granted to copy, distribute and/or modify this document under the terms of the GNU General Public License, Version 3, published by the Free Software Foundation.</p><p> On a Debian system the complete text of the GNU General Public License version 3 can be found in the file `<em>/usr/share/common-licenses/GPL-3</em>'.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="RNPxlXICFilter.1.html"><span aria-hidden="true">&larr;</span> RNPxlXICFilter.1: Library for lc/ms data management and analysis - tools</a></li>
   <li class="next"><a href="RTModel.1.html">RTModel.1: Library for lc/ms data management and analysis - tools <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
