<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>score_conservation: Score protein sequence conservation</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Score protein sequence conservation">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="score_conservation (1) manual">
  <meta name="twitter:description" content="Score protein sequence conservation">
  <meta name="twitter:image" content="https://www.carta.tech/images/conservation-code-score_conservation-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/score_conservation.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="score_conservation (1) manual" />
  <meta property="og:description" content="Score protein sequence conservation" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/conservation-code-score_conservation-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">score_conservation<small> (1)</small></h1>
        <p class="lead">Score protein sequence conservation</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/score_conservation.1.html">
      <span itemprop="name">score_conservation: Score protein sequence conservation</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/conservation-code/">
      <span itemprop="name">conservation-code</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/score_conservation.1.html">
      <span itemprop="name">score_conservation: Score protein sequence conservation</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p>score_conservation [options] \s-1ALIGNFILE\s0</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>Score protein sequence conservation in <strong>\s-1ALIGNFILE\s0</strong>.  <strong>\s-1ALIGNFILE\s0</strong> must be in \s-1FASTA\s0, \s-1CLUSTAL\s0 or Stockholm format.</p><p>The following conservation scoring methods are implemented:  * sum of pairs  * weighted sum of pairs  * Shannon entropy  * Shannon entropy with property groupings (Mirny and Shakhnovich 1995,</p>
<pre>
   Valdar and Thornton 2001)
</pre>
<p> * relative entropy with property groupings (Williamson 1995)  * von Neumann entropy (Caffrey et al 2004)  * relative entropy (Samudrala and Wang 2006)  * Jensen-Shannon divergence (Capra and Singh 2007)</p><p>A window-based extension that incorporates the estimated conservation of sequentially adjacent residues into the score for each column is also given. This window approach can be applied to any of the conservation scoring methods.</p><p>With default parameters <em>score_conservation</em>\|(1) computes the conservation scores for the alignment using the Jensen-Shannon divergence and a window <strong>-w</strong> of <em>3</em>.</p><p>The sequence-specific output can be used as the conservation input for <em>concavity</em>\|(1).</p><p>Conservation is highly predictive in identifying catalytic sites and residues near bound ligands.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REFERENCES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    Capra \s-1JA\s0 and Singh M. Predicting functionally important residues from sequence conservation. Bioinformatics, 23(15):1875-82, 2007.
  </dt>
  <dd>
    
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    -a [\s-1NAME\s0]
  </dt>
  <dd>
    <p>Reference sequence. Print scores in reference to the named sequence (ignoring gaps). Default prints the entire column.</p>
  </dd>
  <dt>
    -b [0-1]
  </dt>
  <dd>
    <p>Lambda for window heuristic linear combination. Default=<em>.5</em>. Equation: \*(C`score = (1 - lambda) * average_score_over_window_around_middle + lambda * score_of_middle\*(C'</p>
  </dd>
  <dt>
    -d [\s-1FILE\s0]
  </dt>
  <dd>
    <p>Background distribution file, e.g. <em>distributions/swissprot.distribution</em>. Default=built-in \s-1BLOSUM62\s0.</p>
  </dd>
  <dt>
    -g [0-1)]
  </dt>
  <dd>
    <p>Gap cutoff. Do not score columns that contain more than gap cutoff fraction gaps. Default=<em>.3</em>.</p>
  </dd>
  <dt>
    -h
  </dt>
  <dd>
    <p>Print help.</p>
  </dd>
  <dt>
    -l [true|false]
  </dt>
  <dd>
    <p>Use sequence weighting. Default=<em>true</em>.</p>
  </dd>
  <dt>
    -m [\s-1FILE\s0]
  </dt>
  <dd>
    <p>Similarity matrix file, e.g. <em>matrix/blosum62.bla</em> or .qij. Default=<em>matrix/blosum62.bla</em>. Some methods, e.g. <em>js_divergence</em>, do not use this.</p>
  </dd>
  <dt>
    -n [true|false]
  </dt>
  <dd>
    <p>Normalize scores. Print the z-score (over the alignment) of each column raw score. Default=<em>false</em>.</p>
  </dd>
  <dt>
    -o \s-1FILE\s0
  </dt>
  <dd>
    <p>Output file. Default: standard output stream.</p>
  </dd>
  <dt>
    -p [true|false]
  </dt>
  <dd>
    <p>Use gap penalty. Lower the score of columns that contain gaps, proportionally to the sum weight of the gapped sequences. Default=<em>true</em>.</p>
  </dd>
  <dt>
    -s [\s-1METHOD\s0]
  </dt>
  <dd>
    <p>Conservation estimation method, one of <em>shannon_entropy property_entropy property_relative_entropy vn_entropy relative_entropy js_divergence sum_of_pairs</em>. Default=<em>js_divergence</em>.</p>
  </dd>
  <dt>
    -w [0-INT]
  </dt>
  <dd>
    <p>Window size. Number of residues on either side included in the window. Default=<em>3</em>.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">
<p>Note: you may have to copy and uncompress the example data files before running the following examples.</p>
<dl class='dl-vertical'>
  <dt>
    Compute conservation scores for the alignment using the Jensen-Shannon divergence with default settings and print out the scores:
  </dt>
  <dd>
    
<pre>
 score_conservation /usr/share/doc/conservation-code/examples/2plc_\|_hssp-filtered.aln
</pre>

  </dd>
  <dt>
    Score an alignment using Jensen-Shannon divergence, a window of size 3 (on either side of the residue), and the swissprot background distribution:
  </dt>
  <dd>
    <p> score_conservation -s js_divergence -w 3 -d &#92;   /usr/share/conservation-code/distributions/swissprot.distribution &#92;   /usr/share/doc/conservation-code/examples/2plc_\|_hssp-filtered.aln</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FILES</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    Distributions
  </dt>
  <dd>
    <p><em>/usr/share/conservation-code/distributions</em></p>
  </dd>
  <dt>
    Matrices
  </dt>
  <dd>
    <p><em>/usr/share/conservation-code/matrix</em></p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO score_conservation&hellip;</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    Homepage &lt;http://compbio.cs.princeton.edu/conservation/&gt;
  </dt>
  <dd>
    
  </dd>
  <dt>
    Publication &lt;http://bioinformatics.oxfordjournals.org/cgi/content/full/23/15/1875&gt;
  </dt>
  <dd>
    
  </dd>
  <dt>
    <em>concavity</em>\|(1)
  </dt>
  <dd>
    
  </dd>

</dl>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="scor2prt.1.html"><span aria-hidden="true">&larr;</span> scor2prt.1: Source-parts separation program for pmx, a musixtex preprocessor</a></li>
   <li class="next"><a href="scotch_esmumps.1.html">scotch_esmumps.1: Scotch command <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
