<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>seqnr: Removes redundancy from dhf files.</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Removes redundancy from dhf files.">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="seqnr (1e) manual">
  <meta name="twitter:description" content="Removes redundancy from dhf files.">
  <meta name="twitter:image" content="https://www.carta.tech/images/embassy-domsearch-seqnr-1e.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1e/seqnr.1e.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="seqnr (1e) manual" />
  <meta property="og:description" content="Removes redundancy from dhf files." />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/embassy-domsearch-seqnr-1e.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">seqnr<small> (1e)</small></h1>
        <p class="lead">Removes redundancy from dhf files.</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1e/">
      <span itemprop="name">MISSING SECTION</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1e/seqnr.1e.html">
      <span itemprop="name">seqnr: Removes redundancy from dhf files.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/embassy-domsearch/">
      <span itemprop="name">embassy-domsearch</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1e/seqnr.1e.html">
      <span itemprop="name">seqnr: Removes redundancy from dhf files.</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>seqnr</strong> <strong>-dhfinpath&nbsp;</strong><strong></strong><em>dirlist</em> <strong>-dosing&nbsp;</strong><strong></strong><em>toggle</em> <strong>-singletsdir&nbsp;</strong><strong></strong><em>directory</em> <strong>-dosets&nbsp;</strong><strong></strong><em>toggle</em> <strong>-insetsdir&nbsp;</strong><strong></strong><em>directory</em> [<strong>-matrix&nbsp;</strong><strong></strong><em>matrixf</em>] <strong>-mode&nbsp;</strong><strong></strong><em>list</em> <strong>-thresh&nbsp;</strong><strong></strong><em>float</em> <strong>-threshlow&nbsp;</strong><strong></strong><em>float</em> <strong>-threshup&nbsp;</strong><strong></strong><em>float</em> [<strong>-gapopen&nbsp;</strong><strong></strong><em>float</em>] [<strong>-gapextend&nbsp;</strong><strong></strong><em>float</em>] <strong>-dhfoutdir&nbsp;</strong><strong></strong><em>outdir</em> <strong>-dored&nbsp;</strong><strong></strong><em>toggle</em> <strong>-redoutdir&nbsp;</strong><strong></strong><em>outdir</em> <strong>-logfile&nbsp;</strong><strong></strong><em>outfile</em> <strong>seqnr</strong> <strong>-help</strong></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>seqnr</strong> is a command line program from EMBOSS (&ldquo;the European Molecular Biology Open Software Suite&rdquo;). It is part of the "Utils:Database creation" command group(s).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<h3>Input section</h3>
<p><strong>-dhfinpath</strong> <em>dirlist</em></p><p>This option specifies the location of DHF files (domain hits files) (input). A &apos;domain hits file&apos; contains database hits (sequences) with domain classification information, in the DHF format (FASTA or EMBL-like). The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH. Default value: ./</p><p><strong>-dosing</strong> <em>toggle</em></p><p>This option specifies whether to use singlet sequences (e.g. DHF files) to filter input. Optionally, up to two further directories of sequences may be read: these are considered in the redundancy calculation but never appear in the output files. Default value: Y</p><p><strong>-singletsdir</strong> <em>directory</em></p><p>This option specifies the location of singlet filter sequences (e.g. DHF files) (input). A &apos;domain hits file&apos; contains database hits (sequences) with domain classification information, in the DHF format (FASTA or EMBL-like). The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH. Default value: ./</p><p><strong>-dosets</strong> <em>toggle</em></p><p>This option specifies whether to use sets of sequences (e.g. DHF files) to filter input. Optionally, up to two further directories of sequences may be read: these are considered in the redundancy calculation but never appear in the output files. Default value: Y</p><p><strong>-insetsdir</strong> <em>directory</em></p><p>This option specifies location of sets of filter sequences (e.g. DAF files) (input). A &apos;domain alignment file&apos; contains a sequence alignment of domains belonging to the same SCOP or CATH family. The file is in clustal format annotated with domain family classification information. The files generated by using SCOPALIGN will contain a structure-based sequence alignment of domains of known structure only. Such alignments can be extended with sequence relatives (of unknown structure) by using SEQALIGN. Default value: ./</p><p><strong>-matrix</strong> <em>matrixf</em></p><p>This option specifies the residue substitution matrix that is used for sequence comparison. Default value: EBLOSUM62</p>
<h3>Required section</h3>
<p><strong>-mode</strong> <em>list</em></p><p>This option specifies whether to remove redundancy at a single threshold % sequence similarity or remove redundancy outside a range of acceptable threshold % similarity. All permutations of pair-wise sequence alignments are calculated for each set of input sequences in turn using the EMBOSS implementation of the Needleman and Wunsch global alignment algorithm. Redundant sequences are removed in one of two modes as follows: (i) If a pair of proteins achieve greater than a threshold percentage sequence similarity (specified by the user) the shortest sequence is discarded. (ii) If a pair of proteins have a percentage sequence similarity that lies outside an acceptable range (specified by the user) the shortest sequence is discarded. Default value: 1</p><p><strong>-thresh</strong> <em>float</em></p><p>This option specifies the % sequence identity redundancy threshold. The % sequence identity redundancy threshold determines the redundancy calculation. If a pair of proteins achieve greater than this threshold the shortest sequence is discarded. Default value: 95.0</p><p><strong>-threshlow</strong> <em>float</em></p><p>This option specifies the % sequence identity redundancy threshold (lower limit). The % sequence identity redundancy threshold determines the redundancy calculation. If a pair of proteins have a percentage sequence similarity that lies outside an acceptable range the shortest sequence is discarded. Default value: 30.0</p><p><strong>-threshup</strong> <em>float</em></p><p>This option specifies the % sequence identity redundancy threshold (upper limit). The % sequence identity redundancy threshold determines the redundancy calculation. If a pair of proteins have a percentage sequence similarity that lies outside an acceptable range the shortest sequence is discarded. Default value: 90.0</p>
<h3>Additional section</h3>
<p><strong>-gapopen</strong> <em>float</em></p><p>This option specifies the gap insertion penalty. The gap insertion penalty is the score taken away when a gap is created. The best value depends on the choice of comparison matrix. The default value assumes you are using the EBLOSUM62 matrix for protein sequences, and the EDNAFULL matrix for nucleotide sequences. Default value: 10</p><p><strong>-gapextend</strong> <em>float</em></p><p>This option specifies the gap extension penalty. The gap extension, penalty is added to the standard gap penalty for each base or residue in the gap. This is how long gaps are penalized. Usually you will expect a few long gaps rather than many short gaps, so the gap extension penalty should be lower than the gap penalty. Default value: 0.5</p>
<h3>Output section</h3>
<p><strong>-dhfoutdir</strong> <em>outdir</em></p><p>This option specifies the location of DHF files (domain hits files) of non-redundant sequences (output). A &apos;domain hits file&apos; contains database hits (sequences) with domain classification information, in the DHF format (FASTA or EMBL-like). The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH. Default value: ./</p><p><strong>-dored</strong> <em>toggle</em></p><p>This option specifies whether to retain redundant sequences. If this option is set a DHF file (domain hits file) of redundant sequences is written. Default value: N</p><p><strong>-redoutdir</strong> <em>outdir</em></p><p>This option specifies the location of DHF files (domain hits files) of redundant sequences (output). A &apos;domain hits file&apos; contains database hits (sequences) with domain classification information, in the DHF format (FASTA or EMBL-like). The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH. Default value: ./</p><p><strong>-logfile</strong> <em>outfile</em></p><p>This option specifies the name of SEQNR log file (output). The log file contains messages about any errors arising while SEQNR ran. Default value: seqnr.log</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>Bugs can be reported to the Debian Bug Tracking system (http://bugs.debian.org/emboss), or directly to the EMBOSS developers (http://sourceforge.net/tracker/?group_id=93650&atid=605031).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO seqnr&hellip;</h2>
        <div class="sectioncontent">
<p>seqnr is fully documented via the <strong>tfm</strong>(1) system.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p><strong>Debian Med Packaging Team</strong> &lt;debian-med-packaging@lists.alioth.debian.org&gt;</p><p>Wrote the script used to autogenerate this manual page.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">
<p>This manual page was autogenerated from an Ajax Control Definition of the EMBOSS package. It can be redistributed under the same terms as EMBOSS itself.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="seqmatchall.1e.html"><span aria-hidden="true">&larr;</span> seqmatchall.1e: All-against-all word comparison of a sequence set</a></li>
   <li class="next"><a href="seqret.1e.html">seqret.1e: Reads and writes (returns) sequences <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
