<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>shuffle: </title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="shuffle (1) manual">
  <meta name="twitter:description" content="">
  <meta name="twitter:image" content="https://www.carta.tech/images/biosquid-shuffle-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/shuffle.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="shuffle (1) manual" />
  <meta property="og:description" content="" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/biosquid-shuffle-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">shuffle<small> (1)</small></h1>
        <p class="lead"></p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/shuffle.1.html">
      <span itemprop="name">shuffle: </span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/biosquid/">
      <span itemprop="name">biosquid</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/shuffle.1.html">
      <span itemprop="name">shuffle: </span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>shuffle</strong> <em>[options]</em> <em>seqfile</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>shuffle</strong> reads a sequence file <em>seqfile,</em> randomizes each sequence, and prints the randomized sequences in FASTA format on standard output. The sequence names are unchanged; this allows you to track down the source of each randomized sequence if necessary.</p><p>The default is to simply shuffle each input sequence, preserving monosymbol composition exactly. To shuffle each sequence while preserving both its monosymbol and disymbol composition exactly, use the <em>-d</em> option.</p><p>The <em>-0</em> and <em>-1</em> options allow you to generate sequences with the same Markov properties as each input sequence. With <em>-0,</em> for each input sequence, 0th order Markov statistics are collected (e.g. symbol composition), and a new sequence is generated with the same composition. With <em>-1,</em> the generated sequence has the same 1st order Markov properties as the input sequence (e.g. the same disymbol frequencies).</p><p>Note that the default and <em>-0,</em> or <em>-d</em> and <em>-1,</em> are similar; the shuffling algorithms preserve composition exactly, while the Markov algorithms only expect to generate a sequence of similar composition on average.</p><p>Other shuffling algorithms are also available, as documented below in the options.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-0</strong></p>
  </dt>
  <dd>
    <p>Calculate 0th order Markov frequencies of each input sequence (e.g. residue composition); generate output sequence using the same 0th order Markov frequencies.</p>
  </dd>
  <dt>
    <p><strong>-1</strong></p>
  </dt>
  <dd>
    <p>Calculate 1st order Markov frequencies for each input sequence (e.g. diresidue composition); generate output sequence using the same 1st order Markov frequencies. The first residue of the output sequence is always the same as the first residue of the input sequence.</p>
  </dd>
  <dt>
    <p><strong>-d</strong></p>
  </dt>
  <dd>
    <p>Shuffle the input sequence while preserving both monosymbol and disymbol composition exactly. Uses an algorithm published by  S.F. Altschul and B.W. Erickson, Mol. Biol. Evol. 2:526-538, 1985.</p>
  </dd>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Print brief help; includes version number and summary of all options, including expert options.</p>
  </dd>
  <dt>
    <p><strong>-l</strong></p>
  </dt>
  <dd>
    <p>Look only at the length of each input sequence; generate an i.i.d. output protein sequence of that length, using monoresidue frequencies typical of proteins (taken from Swissprot 35).</p>
  </dd>
  <dt>
    <p><strong>-n</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Make <em>&lt;n&gt;</em> different randomizations of each input sequence in <em>seqfile,</em> rather than the default of one.</p>
  </dd>
  <dt>
    <p><strong>-r</strong></p>
  </dt>
  <dd>
    <p>Generate the output sequence by reversing the input sequence. (Therefore only one "randomization" per input sequence is possible, so it's not worth using <em>-n</em> if you use reversal.)</p>
  </dd>
  <dt>
    <p><strong>-t</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Truncate each input sequence to a fixed length of exactly <em>&lt;n&gt;</em> residues. If the input sequence is shorter than <em>&lt;n&gt;</em> it is discarded (therefore the output file may contain fewer sequences than the input file). If the input sequence is longer than <em>&lt;n&gt;</em> a contiguous subsequence is randomly chosen.</p>
  </dd>
  <dt>
    <p><strong>-w</strong><em> &lt;n&gt;</em></p>
  </dt>
  <dd>
    <p>Regionally shuffle each input sequence in window sizes of <em>&lt;n&gt;,</em> preserving local residue composition in each window. Probably a better shuffling algorithm for biosequences with nonstationary residue composition (e.g. composition that is varying along the sequence, such as between different isochores in human genome sequence).</p>
  </dd>
  <dt>
    <p><strong>-B</strong></p>
  </dt>
  <dd>
    <p>(Babelfish). Autodetect and read a sequence file format other than the default (FASTA). Almost any common sequence file format is recognized (including Genbank, EMBL, SWISS-PROT, PIR, and GCG unaligned sequence formats, and Stockholm, GCG MSF, and Clustal alignment formats). See the printed documentation for a complete list of supported formats.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXPERT OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--informat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Specify that the sequence file is in format <em>&lt;s&gt;,</em> rather than the default FASTA format. Common examples include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF, or PHYLIP; see the printed documentation for a complete list of accepted format names. This option overrides the default expected format (FASTA) and the <em>-B</em> Babelfish autodetection option.</p>
  </dd>
  <dt>
    <p><strong>--nodesc</strong></p>
  </dt>
  <dd>
    <p>Do not output any sequence description in the output file, only the sequence names.</p>
  </dd>
  <dt>
    <p><strong>--seed</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Set the random number seed to <em>&lt;s&gt;.</em> If you want reproducible results, use the same seed each time. By default, <strong>shuffle</strong> uses a different seed each time, so does not generate the same output in subsequent runs with the same input.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO shuffle&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man1/afetch.1.html"><strong>afetch</strong>(1)</a>, <a href="../man1/alistat.1.html"><strong>alistat</strong>(1)</a>, <a href="../man1/compalign.1.html"><strong>compalign</strong>(1)</a>, <a href="../man1/compstruct.1.html"><strong>compstruct</strong>(1)</a>, <a href="../man1/revcomp.1.html"><strong>revcomp</strong>(1)</a>, <a href="../man1/seqsplit.1.html"><strong>seqsplit</strong>(1)</a>, <a href="../man1/seqstat.1.html"><strong>seqstat</strong>(1)</a>, <a href="../man1/sfetch.1.html"><strong>sfetch</strong>(1)</a>, <a href="../man1/sindex.1.html"><strong>sindex</strong>(1)</a>, <a href="../man1/sreformat.1.html"><strong>sreformat</strong>(1)</a>, <a href="../man1/stranslate.1.html"><strong>stranslate</strong>(1)</a>, <a href="../man1/weight.1.html"><strong>weight</strong>(1)</a>.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Biosquid and its documentation are Copyright (C) 1992-2003 HHMI/Washington University School of Medicine Freely distributed under the GNU General Public License (GPL) See COPYING in the source code distribution for more details, or contact me.</p>
<pre>
Sean Eddy
HHMI/Department of Genetics
Washington University School of Medicine
4444 Forest Park Blvd., Box 8510
St Louis, MO 63108 USA
Phone: 1-314-362-7666
FAX  : 1-314-362-2157
Email: eddy@genetics.wustl.edu
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="shuf.1.html"><span aria-hidden="true">&larr;</span> shuf.1: Generate random permutations</a></li>
   <li class="next"><a href="shunit2.1.html">shunit2.1: A unit test framework for shell scripts <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
