<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>sigma: Simple greedy multiple alignment of non-coding dna sequences</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Simple greedy multiple alignment of non-coding dna sequences">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="sigma (1) manual">
  <meta name="twitter:description" content="Simple greedy multiple alignment of non-coding dna sequences">
  <meta name="twitter:image" content="https://www.carta.tech/images/sigma-align-sigma-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/sigma.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="sigma (1) manual" />
  <meta property="og:description" content="Simple greedy multiple alignment of non-coding dna sequences" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/sigma-align-sigma-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">sigma<small> (1)</small></h1>
        <p class="lead">Simple greedy multiple alignment of non-coding dna sequences</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/sigma.1.html">
      <span itemprop="name">sigma: Simple greedy multiple alignment of non-coding dna sequences</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/sigma-align/">
      <span itemprop="name">sigma-align</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/sigma.1.html">
      <span itemprop="name">sigma: Simple greedy multiple alignment of non-coding dna sequences</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>sigma</strong> [options] [<em>inputfile.fasta</em>] [<em>inputfile2.fasta&nbsp;...</em>]</p><p>Each fasta file may contain a single sequence or multiple sequences; all sequences will be aligned together.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>Sigma</strong> ("Simple greedy multiple alignment") is an alignment program with a new algorithm and scoring scheme designed specifically for non-coding DNA sequence. It uses a strategy of seeking the best possible gapless local alignments, at each step making the best possible alignment consistent with existing alignments, and scores the significance of the alignment based on the lengths of the aligned fragments and a background model which may be supplied or estimated from an auxiliary file of intergenic DNA. With real data, while "correctness" can't be directly quantified for the alignment, running the PhyloGibbs motif finder on pre-aligned sequence suggests that Sigma's alignments are superior.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p><strong>-A</strong> <strong>--aligned_output</strong></p><p>Aligned, pretty-printed output (compare with <strong>-F</strong> option) (default: only this). See also <strong>-C</strong>.</p><p><strong>-b</strong> <strong>--bgprobfile</strong> <em>filename</em></p><p>Auxiliary file (in fasta format) from which to read background sequences (overridden by <strong>-B</strong>). Typically this is a file containing large quantities of similar non-coding sequence, from which background probabilities of single- and di-nucleotides may be estimated.</p><p><strong>-B</strong> <strong>--bgseqfile</strong> <em>filename</em></p><p>File containing background probabilities. The format is described further below.</p><p><strong>-C</strong> <strong>--caps_only</strong></p><p>Use only upper-case letters in output sequence, for compatibility with output of some other programs like ClustalW and MLagan. By default, output is mixed-case (as in Dialign), and lower-case bases are treated as not aligned.</p><p><strong>-F</strong> <strong>--fasta_output</strong></p><p>Multi-fasta output (can use both <strong>-A</strong> and <strong>-F</strong> in either order). See also <strong>-C</strong>.</p><p><strong>-n</strong> <strong>--ncorrel</strong> <em>number</em></p><p>Background correlation (default <em>2</em>=dinucleotide; <em>1</em>=single-site basecounts, <em>0</em>=0.25 per base).</p><p><strong>-x</strong>, <strong>--significance</strong> <em>number</em></p><p>Set limit for how probable the match is by chance (default 0.002, smaller=more stringent).</p><p><strong>-h</strong>, <strong>--help</strong></p><p>Displays this list of options.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">MORE HELP</h2>
        <div class="sectioncontent">
<p>The "significance" parameter (<strong>-x</strong>) determines whether local alignments are accepted or rejected. The default at present is 0.002. Experiments on synthetic data (described in the paper) suggest that 0.002 is about the threshold where sigma fails to align phylogenetically-unrelated data that has moderate (yeast-like) dinucleotide correlation.</p><p>Using a &ldquo;background model&rdquo; appropriate to the sequences being aligned greatly reduces spurious alignments on synthetic data (and, one hopes, on real data too). The simplest way to ensure this is to supply, via the <strong>-b</strong> parameter, a FASTA-format file containing large quantities of similar sequence data (eg, if one is aligning yeast sequences, supply a file containing all intergenic yeast sequence).</p><p>Instead of this, if the single-site and dinucleotide frequencies are known already, they may be supplied in a file via the <strong>-B</strong> option. The file format should be: one entry per line, with the mononucleotide or dinucleotide (case-insensitive) followed by the frequency. (eg, "A 0.3", "AT 0.16", etc on successive lines.) A sample file is in the "Background" subdirectory of the source distribution (on Debian systems, this file can be found in the <em>/usr/share/doc/sigma-align/Background</em> directory). A file like "yeast.nc.3.freq" in the "tests" subdirectory of the MEME source distribution works fine (trinucleotide counts are ignored).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">REFERENCE</h2>
        <div class="sectioncontent">
<p>Please cite Sigma: Rahul Siddharthan (2006) Multiple alignment of weakly-conserved non-coding DNA sequence BMC Bioinformatics 2006, 7:143 doi:10.1186/1471-2105-7-143 Published 16 March 2006, available online at http://www.biomedcentral.com/1471-2105/7/143/</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p><strong>Rahul Siddharthan</strong> &lt;rsidd@imsc.res.in&gt;</p><ul>
<li><p>Wrote sigma. If you're using Sigma for actual research, please let the author know so that he can alert you of bugfixes or new releases.</p></li>
</ul><p><strong>Charles Plessy</strong> &lt;charles-debian-nospam@plessy.org&gt;</p><ul>
<li><p>Wrote the manpage in DocBook XML for the Debian distribution.</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">COPYRIGHT</h2>
        <div class="sectioncontent">
<p>Copyright &copy; 2006-2007 Rahul Siddharthan</p><p>Copyright &copy; 2006-2007 Charles Plessy</p><p>Sigma is free software. You can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation.</p><p>On Debian systems, the complete text of the GNU General Public License can be found in /usr/share/common-licenses/GPL.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="siggen.1.html"><span aria-hidden="true">&larr;</span> siggen.1: An ncurses based signal generator program</a></li>
   <li class="next"><a href="sigmon.1.html">sigmon.1: Vodafone 3g/gprs datacard signal strength monitor <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
