<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>sim4: Align an expressed dna sequence with a genomic sequence</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Align an expressed dna sequence with a genomic sequence">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="sim4 (1) manual">
  <meta name="twitter:description" content="Align an expressed dna sequence with a genomic sequence">
  <meta name="twitter:image" content="https://www.carta.tech/images/sim4-sim4-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/sim4.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="sim4 (1) manual" />
  <meta property="og:description" content="Align an expressed dna sequence with a genomic sequence" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/sim4-sim4-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">sim4<small> (1)</small></h1>
        <p class="lead">Align an expressed dna sequence with a genomic sequence</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/sim4.1.html">
      <span itemprop="name">sim4: Align an expressed dna sequence with a genomic sequence</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/sim4/">
      <span itemprop="name">sim4</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/sim4.1.html">
      <span itemprop="name">sim4: Align an expressed dna sequence with a genomic sequence</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>sim4</strong> <em>seqfile1</em> <em>seqfile2</em> {[WXKCRDAPNB]=<em>value</em>}</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>sim4</strong> is a similarity-based tool for aligning an expressed DNA sequence (EST, cDNA, mRNA) with a genomic sequence for the gene. It also detects end matches when the two input sequences overlap at one end (i.e., the start of one sequence overlaps the end of the other). If <em>seqfile2</em> is a database of sequences, the sequence in <em>seqfile1</em> will be aligned with each of the sequences in <em>seqfile2</em>.</p><p><strong>sim4</strong> employs a blast-based technique to first determine the basic matching blocks representing the "exon cores". In this first stage, it detects all possible exact matches of W-mers (i.e., DNA words of size W) between the two sequences and extends them to maximal scoring gap-free segments. In the second stage, the exon cores are extended into the adjacent as-yet-unmatched fragments using greedy alignment algorithms, and heuristics are used to favor configurations that conform to the splice-site recognition signals (GT-AG, CT-AC). If necessary, the process is repeated with less stringent parameters on the unmatched fragments.</p><p>By default, <strong>sim4</strong> searches both strands and reports the best match, measured by the number of matching nucleotides found in the alignment. The R command line option can be used to restrict the search to one orientation (strand) only.</p><p>Currently, five major alignment display options are supported, controlled by the A option. By default (A=0), only the endpoints, overall similarity, and orientation of the introns are reported. An arrow sign (`-&gt;' or `&lt;-') indicates the orientation of the intron (`+' or `-' strand), when the signals flanking the intron have three or more position matches with either the GT-AG or the CT-AC splice recognition signals. When the same number of matches is found for both orientations, the intron is reported as ambiguous, and represented by `--'. The sign `==' marks the absence from the alignment of a cDNA fragment starting at that position. Alternative formats (lav-block format, text, PipMaker-type `exons file', or certain combinations of these options) can be requested by specifying a different value for A.</p><p>If the P option is specified with a non-zero value, <strong>sim4</strong> will remove any 3'-end poly-A tails that it detects in the alignment.</p><p>Occasionally, <strong>sim4</strong> may miss an internal exon when surrounded by very large introns, typically longer than 100 Kb. When this is suspected, the H option can be used to reset the exons' weight to compensate for the intron gap penalty.</p><p>Ambiguity codes are by default allowed in sequence data, but <strong>sim4</strong> treats them non-differentially. If desired, the B command option can restrict the set of acceptable characters to A,C,G,T,N and X only.</p><p><strong>sim4</strong> compares the lengths of the input sequences to distinguish between the cDNA (`short') and the genomic (`long') components in the comparison. When <em>seqfile2</em> contains a collection of sequences, the first entry in the file will be used to determine the type of this and all subsequent comparisons.</p><p>In the description below, the term MSP denotes a <em>M</em>aximal <em>S</em>egment <em>P</em>air, that is, a pair of highly similar fragments in the two sequences, obtained during the blast-like procedure by extending a W-mer hit by matches and perhaps a few mismatches.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p>The algorithm parameters (included in the first two sections below) have already been tuned and do not normally require adjustment by the user.</p><p>Parameters internal to the blast-like procedure:</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>W</strong></p>
  </dt>
  <dd>
    <p>Sets the word size for blast hits in the first stage of the algorithm. The default value is 12, but it can be increased for a more stringent search or decreased to find weaker matches.</p>
  </dd>
  <dt>
    <p><strong>X</strong></p>
  </dt>
  <dd>
    <p>Controls the limits for terminating word extensions in the blast-like stage of the algorithm. The default value is 12.</p>
  </dd>
  <dt>
    <p><strong>K</strong></p>
  </dt>
  <dd>
    <p>Sets the threshold for the MSP scores when determining the basic `exon cores', during the first stage of the algorithm. (If this option is not specified, the threshold is computed from the lengths of the sequences, using statistical criteria.) For example, a good value for genomic sequences in the range of a few hundred Kb is 16. To avoid spurious matches, however, a larger value may be needed for longer sequences.</p>
  </dd>
  <dt>
    <p><strong>C</strong></p>
  </dt>
  <dd>
    <p>Sets the threshold for the MSP scores when aligning the as-yet-unmatched fragments, during the second stage of the algorithm. By default, the smaller of the constant 12 and a statistics-based threshold is chosen.</p>
  </dd>

</dl>
<p>Additional algorithm parameters:</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>D</strong></p>
  </dt>
  <dd>
    <p>Sets the bound for the "diagonal" distance within consecutive MSPs in an exon. The default value is 10.</p>
  </dd>

</dl>
<p>Context parameters:</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>R</strong></p>
  </dt>
  <dd>
    <p>Specifies the direction of the search. If R=0, only the "+" (direct) strand is searched. If R=1, only the "-" (reverse complement) matches are sought. By default (R=2), sim4 searches both strands and reports the best match, measured by the number of matching pairs in the alignment.</p>
  </dd>
  <dt>
    <p><strong>A</strong></p>
  </dt>
  <dd>
    <p>Specifies the format of the output: exon endpoints only (A=0), exon endpoints and boundaries of the coding region (CDS) in the genomic sequence, when specified for the input mRNA (A=5), alignment text (A=1), alignment in lav-block format (A=2), or both exon endpoints and alignment text (A=3 or A=4). If a reverse complement match is found, A=0,1,2,3,5 will give its position in the "+" strand of the longer sequence and the "-" strand of the shorter sequence. A=4 will give its position in the "+" strand of the first sequence (seqfile1) and the "-" strand of the second sequence (seqfile2), regardless of which sequence is longer. The A=5 option can be used with the S command line option to specify the endpoints of the CDS in the mRNA, and produces output in the `exons file' format required by PipMaker.</p>
  </dd>
  <dt>
    <p><strong>P</strong></p>
  </dt>
  <dd>
    <p>Specifies whether or not the program should report the fragment of the alignment containing the poly-A tail (if found). By default (P=0) the alignment is displayed as computed, but specifying a non-zero value will request sim4 to remove the poly-A tails. When this feature is enabled, all display options produce additional lav alignment headers.</p>
  </dd>
  <dt>
    <p><strong>H</strong></p>
  </dt>
  <dd>
    <p>Resets the MSPs' weight to compensate for very large introns. The default value is H=500, but some introns larger than 100 Kb may require higher values, typically between 1000 and 2500. This option should be used cautiously, generally in cases where an unmatched internal portion of the cDNA may disguise a missed exon within a very large intron. It is not recommended for ESTs, where they may produce spurious exons.</p>
  </dd>
  <dt>
    <p><strong>N</strong></p>
  </dt>
  <dd>
    <p>Requests an additional search for small marginal exons (N=1) guided by the splice-site recognition signals. This option can be used when a high accuracy match is expected. The default value is N=0, specifying no additional search.</p>
  </dd>
  <dt>
    <p><strong>B</strong></p>
  </dt>
  <dd>
    <p>Controls the set of characters allowed in the input sequences. By default (B=1), ambiguity characters (ABCDGHKMNRSTVWXY) are allowed. By specifying B=0, the set of acceptable characters is restricted to A,C,G,T,N and X only.</p>
  </dd>
  <dt>
    <p><strong>S</strong></p>
  </dt>
  <dd>
    <p>Allows the user to specify the endpoints of the CDS in the input mRNA, with the syntax: S=n1..n2. This option is only available with the A=5 flag, which produces output in the format required by PipMaker. Alternatively, the CDS coordinates could appear in a construct CDS=n1..n2 in the FastA header of the mRNA sequence. When the second file is an mRNA database, the command line specification for the CDS will apply to the first sequence in the file only.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">
<p>sim4 est genomic</p><p>sim4 genomic estdb</p><p>sim4 est genomic A=1 P=1</p><p>sim4 est1 est2 R=1</p><p>sim4 mRNA genomic A=5 S=123..1020</p><p>sim4 mouse_cDNA human_genomic K=15 C=11 A=3 W=10</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p>sim4 was written by Liliana Florea &lt;florea@gwu.edu&gt; and Scott Schwartz.</p><p>This manual page was written by Nelson A. de Oliveira &lt;naoliv@gmail.com&gt;, based on the online documentation at http://globin.cse.psu.edu/html/docs/sim4.html, for the Debian project (but may be used by others).</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="sim2trad.1.html"><span aria-hidden="true">&larr;</span> sim2trad.1: Clipboard data conversion for traditional chinese and simplified chinese</a></li>
   <li class="next"><a href="sim_c.1.html">sim_c.1: Find similarities in c, java, pascal, modula-2, lisp, miranda, or text files <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
