<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>spidey: Align mrna sequences to a genome</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Align mrna sequences to a genome">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="spidey (1) manual">
  <meta name="twitter:description" content="Align mrna sequences to a genome">
  <meta name="twitter:image" content="https://www.carta.tech/images/ncbi-tools-bin-spidey-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/spidey.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="spidey (1) manual" />
  <meta property="og:description" content="Align mrna sequences to a genome" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/ncbi-tools-bin-spidey-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">spidey<small> (1)</small></h1>
        <p class="lead">Align mrna sequences to a genome</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/spidey.1.html">
      <span itemprop="name">spidey: Align mrna sequences to a genome</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/ncbi-tools-bin/">
      <span itemprop="name">ncbi-tools-bin</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/spidey.1.html">
      <span itemprop="name">spidey: Align mrna sequences to a genome</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>spidey</strong> [\|<strong>-</strong>\|] [\|<strong>-F</strong>&nbsp;<em>N</em>\|] [\|<strong>-G</strong>\|] [\|<strong>-L</strong>&nbsp;<em>N</em>\|] [\|<strong>-M</strong>&nbsp;<em>filename</em>\|] [\|<strong>-N</strong>&nbsp;<em>filename</em>\|] [\|<strong>-R</strong>&nbsp;<em>filename</em>\|] [\|<strong>-S</strong>&nbsp;<em>p/m</em>\|] [\|<strong>-T</strong>&nbsp;<em>N</em>\|] [\|<strong>-X</strong>\|] [\|<strong>-a</strong>&nbsp;<em>filename</em>\|] [\|<strong>-c</strong>&nbsp;<em>N</em>\|] [\|<strong>-d</strong>\|] [\|<strong>-e</strong>&nbsp;<em>X</em>\|] [\|<strong>-f</strong>&nbsp;<em>X</em>\|] [\|<strong>-g</strong>&nbsp;<em>X</em>\|] <strong>-i</strong>&nbsp;<em>filename</em> [\|<strong>-j</strong>\|] [\|<strong>-k</strong>&nbsp;<em>filename</em>\|] [\|<strong>-l</strong>&nbsp;<em>N</em>\|] <strong>-m</strong>&nbsp;<em>filename</em> [\|<strong>-n</strong>&nbsp;<em>N</em>\|] [\|<strong>-o</strong>&nbsp;<em>str</em>\|] [\|<strong>-p</strong>&nbsp;<em>N</em>\|] [\|<strong>-r</strong>&nbsp;<em>c/d/m/p/v</em>\|] [\|<strong>-s</strong>\|] [\|<strong>-t</strong>&nbsp;<em>filename</em>\|] [\|<strong>-u</strong>\|] [\|<strong>-w</strong>\|]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>spidey</strong> is a tool for aligning one or more mRNA sequences to a given genomic sequence.  <strong>spidey</strong> was written with two main goals in mind: find good alignments regardless of intron size; and avoid getting confused by nearby pseudogenes and paralogs.  Towards the first goal, <strong>spidey</strong> uses BLAST and Dot View (another local alignment tool) to find its alignments; since these are both local alignment tools, <strong>spidey</strong> does not intrinsically favor shorter or longer introns and has no maximum intron size.  To avoid mistakenly including exons from paralogs and pseudogenes, <strong>spidey</strong> first defines windows on the genomic sequence and then performs the mRNA-to-genomic alignment separately within each window.  Because of the way the windows are constructed, neighboring paralogs or pseudogenes should be in separate windows and should not be included in the final spliced alignment.</p><h3>Initial alignments and construction of genomic windows</h3>
<p><strong>spidey</strong> takes as input a single genomic sequence and a set of mRNA accessions or FASTA sequences.  All processing is done one mRNA sequence at a time.  The first step for each mRNA sequence is a high-stringency BLAST against the genomic sequence.  The resulting hits are analyzed to find the genomic windows.</p><p>The BLAST alignments are sorted by score and then assigned into windows by a recursive function which takes the first alignment and then goes down the alignment list to find all alignments that are consistent with the first (same strand of mRNA, both the mRNA and genomic coordinates are nonoverlapping and linearly consistent).  On subsequent passes, the remaining alignments are examined and are put into their own nonoverlapping, consistent windows, until no alignments are left.  Depending on how many gene models are desired, the top <em>n</em> windows are chosen to go on to the next step and the others are deleted.</p>
<h3>Aligning in each window</h3>
<p>Once the genomic windows are constructed, the initial BLAST alignments are freed and another BLAST search is performed, this time with the entire mRNA against the genomic region defined by the window, and at a lower stringency than the initial search.  <strong>spidey</strong> then uses a greedy algorithm to generate a high-scoring, nonoverlapping subset of the alignments from the second BLAST search.  This consistent set is analyzed carefully to make sure that the entire mRNA sequence is covered by the alignments.  When gaps are found between the alignments, the appropriate region of genomic sequence is searched against the missing mRNA, first using a very low-stringency BLAST and, if the BLAST fails to find a hit, using DotView functions to locate the alignment.  When gaps are found at the ends of the alignments, the BLAST and DotView searches are actually allowed to extend past the boundaries of the window.  If the 3' end of the mRNA does not align completely, it is first examined for the presence of a poly(A) tail. No attempt is made to align the portion of the mRNA that seems to be a poly(A) tail; sometimes there is a poly(A) tail that does align to the genomic sequence, and these are noted because they indicate the possibility of a pseudogene.</p><p>Now that the mRNA is completely covered by the set of alignments, the boundaries of the alignments (there should be one alignment per exon now) are adjusted so that the alignments abut each other precisely and so that they are adjacent to good splice donor and acceptor sites. Most commonly, two adjacent exons' alignments overlap by as much as 20 or 30 base pairs on the mRNA sequence.  The true exon boundary may lie anywhere within this overlap, or (as we have seen empirically) even a few base pairs outside the overlap.  To position the exon boundaries, the overlap plus a few base pairs on each side is examined for splice donor sites, using functions that have different splice matrices depending on the organism chosen.  The top few splice donor sites (by score) are then evaluated as to how much they affect the original alignment boundaries.  The site that affects the boundaries the least is chosen, and is evaluated as to the presence of an acceptor site. The alignments are truncated or extended as necessary so that they terminate at the splice donor site and so that they do not overlap.</p>
<h3>Final result</h3>
<p>The windows are examined carefully to get the percent identity per exon, the number of gaps per exon, the overall percent identity, the percent coverage of the mRNA, presence of an aligning or non-aligning poly(A) tail, number of splice donor sites and the presence or absence of splice donor and acceptor sites for each exon, and the occurrence of an mRNA that has a 5' or 3' end (or both) that does not align to the genomic sequence.  If the overall percent identity and percent length coverage are above the user-defined cutoffs, a summary report is printed, and, if requested, a text alignment showing identities and mismatches is also printed.</p>
<h3>Interspecies alignments</h3>
<p><strong>spidey</strong> is capable of performing interspecies alignments.  The major difference in interspecies alignments is that the mRNA-genomic identity will not be close to 100% as it is in intraspecies alignments; also, the alignments have numerous and lengthy gaps.  If <strong>spidey</strong> is used in its normal mode to do interspecies alignments, it produces gene models with many, many short exons.  When the interspecies flag is set, <strong>spidey</strong> uses different BLAST parameters to encourage longer and more gaps and to not penalize as heavily for mismatches.  This way, the alignments for the exons are much longer and more closely approximate the actual gene structure.</p>
<h3>Extracting CDS alignments</h3>
<p>When <strong>spidey</strong> is run in network-aware mode or when ASN.1 files are used for the mRNA records, it is capable of extracting a CDS alignment from an mRNA alignment and printing the CDS information also.  Since the CDS alignment is just a subset of the mRNA alignment, it is relatively straightforward to truncate the exon alignments as necessary and to generate a CDS alignment.  Furthermore, the untranslated regions are now defined, so the percent identity for the 5' and 3' untranslated regions is also calculated.</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p>A summary of options is included below.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-</strong></p>
  </dt>
  <dd>
    <p>Print usage message.</p>
  </dd>
  <dt>
    <p><strong>-F</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>Start of genomic interval desired (from; 0-based).</p>
  </dd>
  <dt>
    <p><strong>-G</strong></p>
  </dt>
  <dd>
    <p>Input file is a GI list.</p>
  </dd>
  <dt>
    <p><strong>-L</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>The extra-large intron size to use (default = 220000).</p>
  </dd>
  <dt>
    <p><strong>-M</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>File with donor splice matrix.</p>
  </dd>
  <dt>
    <p><strong>-N</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>File with acceptor splice matrix.</p>
  </dd>
  <dt>
    <p><strong>-R</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>File (including path) to repeat blast database for filtering.</p>
  </dd>
  <dt>
    <p><strong>-S</strong>&nbsp;<em>p/m</em></p>
  </dt>
  <dd>
    <p>Restrict to plus (p) or minus (m) strand of genomic sequence.</p>
  </dd>
  <dt>
    <p><strong>-T</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>Stop of genomic interval desired (to; 0-based).</p>
  </dd>
  <dt>
    <p><strong>-X</strong></p>
  </dt>
  <dd>
    <p>Use extra-large intron sizes (increases the limit for initial and terminal introns from 100kb to 240kb and for all others from 35kb to 120kb); may result in significantly longer compute times.</p>
  </dd>
  <dt>
    <p><strong>-a</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Output file for alignments when directed to a separate file with <strong>-p&nbsp;3</strong> (default = spidey.aln).</p>
  </dd>
  <dt>
    <p><strong>-c</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>Identity cutoff, in percent, for quality control purposes.</p>
  </dd>
  <dt>
    <p><strong>-d</strong></p>
  </dt>
  <dd>
    <p>Also try to align coding sequences corresponding to the given mRNA records (may require network access).</p>
  </dd>
  <dt>
    <p><strong>-e</strong>&nbsp;<em>X</em></p>
  </dt>
  <dd>
    <p>First-pass e-value (default = 1.0e-10).  Higher values increase speed at the cost of sensitivity.</p>
  </dd>
  <dt>
    <p><strong>-f</strong>&nbsp;<em>X</em></p>
  </dt>
  <dd>
    <p>Second-pass e-value (default = 0.001).</p>
  </dd>
  <dt>
    <p><strong>-g</strong>&nbsp;<em>X</em></p>
  </dt>
  <dd>
    <p>Third-pass e-value (default = 10).</p>
  </dd>
  <dt>
    <p><strong>-i</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Input file containing the genomic sequence in ASN.1 or FASTA format. If your computer is running on a network that can access GenBank, you can substitute the desired accession number for the filename.</p>
  </dd>
  <dt>
    <p><strong>-j</strong></p>
  </dt>
  <dd>
    <p>Print ASN.1 alignment?</p>
  </dd>
  <dt>
    <p><strong>-k</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>File for ASN.1 output with <strong>-k</strong> (default = spidey.asn).</p>
  </dd>
  <dt>
    <p><strong>-l</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>Length coverage cutoff, in percent.</p>
  </dd>
  <dt>
    <p><strong>-m</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Input file containing the mRNA sequence(s) in ASN.1 or FASTA format, or a list of their accessions (with <strong>-G</strong>).  If your computer is running on a network that can access GenBank, you can substitute a single accession number for the filename.</p>
  </dd>
  <dt>
    <p><strong>-n</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>Number of gene models to return per input mRNA (default = 1).</p>
  </dd>
  <dt>
    <p><strong>-o</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Main output file (default = stdout; contents controlled by <strong>-p</strong>).</p>
  </dd>
  <dt>
    <p><strong>-p</strong>&nbsp;<em>N</em></p>
  </dt>
  <dd>
    <p>Print alignment?</p>
<dl class='dl-vertical'>
  <dt>
    <strong>0</strong>
  </dt>
  <dd>
    <p>summary and alignments together (default)</p>
  </dd>
  <dt>
    <strong>1</strong>
  </dt>
  <dd>
    <p>just the summary</p>
  </dd>
  <dt>
    <strong>2</strong>
  </dt>
  <dd>
    <p>just the alignments</p>
  </dd>
  <dt>
    <strong>3</strong>
  </dt>
  <dd>
    <p>summary and alignments in different files</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-r</strong>&nbsp;<em>c/d/m/p/v</em></p>
  </dt>
  <dd>
    <p>Organism of genomic sequence, used to determine splice matrices.</p>
<dl class='dl-vertical'>
  <dt>
    <strong>c</strong>
  </dt>
  <dd>
    <p>C. elegans</p>
  </dd>
  <dt>
    <strong>d</strong>
  </dt>
  <dd>
    <p>Drosophila</p>
  </dd>
  <dt>
    <strong>m</strong>
  </dt>
  <dd>
    <p>Dictyostelium discoideum</p>
  </dd>
  <dt>
    <strong>p</strong>
  </dt>
  <dd>
    <p>plant</p>
  </dd>
  <dt>
    <strong>v</strong>
  </dt>
  <dd>
    <p>vertebrate (default)</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-s</strong></p>
  </dt>
  <dd>
    <p>Tune for interspecies alignments.</p>
  </dd>
  <dt>
    <p><strong>-t</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>File with feature table, in 4 tab-delimited columns:</p>
<dl class='dl-vertical'>
  <dt>
    <em>seqid</em>
  </dt>
  <dd>
    <p>(e.g., <strong>NM_04377.1</strong>)</p>
  </dd>
  <dt>
    <em>name</em>
  </dt>
  <dd>
    <p>(only <strong>repetitive_region</strong> is currently supported)</p>
  </dd>
  <dt>
    <em>start</em>
  </dt>
  <dd>
    <p>(0-based)</p>
  </dd>
  <dt>
    <em>stop</em>
  </dt>
  <dd>
    <p>(0-based)</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Make a multiple alignment of all input mRNAs (which must overlap on the genomic sequence).</p>
  </dd>
  <dt>
    <p><strong>-w</strong></p>
  </dt>
  <dd>
    <p>Consider lowercase characters in input FASTA sequences to be masked.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Sarah Wheelan and others at the National Center for Biotechnology Information; Steffen Moeller contributed to this documentation.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO spidey&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man1/blast.1.html"><strong>blast</strong>(1)</a>, &lt;http://www.ncbi.nlm.nih.gov/spidey&gt;</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="spicy-stats.1.html"><span aria-hidden="true">&larr;</span> spicy-stats.1: Spice client used for testing and measurements</a></li>
   <li class="next"><a href="spikeproxy.1.html">spikeproxy.1: Web application auditing tool <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
