<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>splazers: Splazers ========  synopsis splazers [options] genome file reads file splazers [options] genome file reads file 1 reads file 2  description splazers uses a prefix-suffix mapping strategy to split-map read sequences.if a sam file of mapped reads is given as input, all unmapped but anchoredreads are split-mapped onto anchoring target regions (specify option -an),if a fasta/q file of reads is given, reads are split-mapped onto the wholereference sequence. (c) copyright 2010 by anne-katrin emde. -h, --help displays this help message. --version display version information main options:: -o, --output file change output filename. default: reads file.result. -f, --forward only compute forward matches -r, --reverse only compute reverse complement matches -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num set the percent recognition rate in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -mcl, --min-clipped-len num min. read length for read clipping in range [1..inf]. default: 0. -qih, --quality-in-header quality string in fasta header -ou, --outputunmapped file output filename for unmapped reads -v, --verbose verbose mode -vv, --vverbose very verbose mode output format options:: -a, --alignment dump the alignment for each match -pa, --purge-ambiguous purge reads with more than max-hits best matches -dr, --distance-range num only consider matches with at most num more errors compared to the best (default output all) -of, --output-format num set output format. 0 = razers, 1 = enhanced fasta, 2 = eland, 3 = gff, 4 = sam. in range [0..4]. -gn, --genome-naming num select how genomes are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -rn, --read-naming num select how reads are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -so, --sort-order num select how matches are sorted. 0 = read number, 1 = genome position. in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). 0 = gap space, 1 = position space. in range [0..1]. default: 0. split mapping options:: -sm, --split-mapping num min. match length for prefix/suffix mapping (to disable split mapping, set to 0) default: 18. -maxg, --max-gap num max. length of middle gap default: 10000. -ming, --min-gap num min. length of middle gap (for edit distance mapping about 10% of read length is recommended) default: 0. -ep, --errors-prefix num max. number of errors in prefix match default: 1. -es, --errors-suffix num max. number of errors in suffix match default: 1. -gl, --genome-len num genome length in mb, for computation of expected number of random matches in range [-inf..10000]. default: 3000. -an, --anchored anchored split mapping, only unmapped reads with mapped mates will be considered, requires the reads to be given in sam format -pc, --penalty-c num percent of read length, used as penalty for split-gap default: 2. filtration options:: -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file.  version splazers version: 1.1 last update apr 2011</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Splazers ========  synopsis splazers [options] genome file reads file splazers [options] genome file reads file 1 reads file 2  description splazers uses a prefix-suffix mapping strategy to split-map read sequences.if a sam file of mapped reads is given as input, all unmapped but anchoredreads are split-mapped onto anchoring target regions (specify option -an),if a fasta/q file of reads is given, reads are split-mapped onto the wholereference sequence. (c) copyright 2010 by anne-katrin emde. -h, --help displays this help message. --version display version information main options:: -o, --output file change output filename. default: reads file.result. -f, --forward only compute forward matches -r, --reverse only compute reverse complement matches -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num set the percent recognition rate in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -mcl, --min-clipped-len num min. read length for read clipping in range [1..inf]. default: 0. -qih, --quality-in-header quality string in fasta header -ou, --outputunmapped file output filename for unmapped reads -v, --verbose verbose mode -vv, --vverbose very verbose mode output format options:: -a, --alignment dump the alignment for each match -pa, --purge-ambiguous purge reads with more than max-hits best matches -dr, --distance-range num only consider matches with at most num more errors compared to the best (default output all) -of, --output-format num set output format. 0 = razers, 1 = enhanced fasta, 2 = eland, 3 = gff, 4 = sam. in range [0..4]. -gn, --genome-naming num select how genomes are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -rn, --read-naming num select how reads are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -so, --sort-order num select how matches are sorted. 0 = read number, 1 = genome position. in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). 0 = gap space, 1 = position space. in range [0..1]. default: 0. split mapping options:: -sm, --split-mapping num min. match length for prefix/suffix mapping (to disable split mapping, set to 0) default: 18. -maxg, --max-gap num max. length of middle gap default: 10000. -ming, --min-gap num min. length of middle gap (for edit distance mapping about 10% of read length is recommended) default: 0. -ep, --errors-prefix num max. number of errors in prefix match default: 1. -es, --errors-suffix num max. number of errors in suffix match default: 1. -gl, --genome-len num genome length in mb, for computation of expected number of random matches in range [-inf..10000]. default: 3000. -an, --anchored anchored split mapping, only unmapped reads with mapped mates will be considered, requires the reads to be given in sam format -pc, --penalty-c num percent of read length, used as penalty for split-gap default: 2. filtration options:: -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file.  version splazers version: 1.1 last update apr 2011">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="splazers (1) manual">
  <meta name="twitter:description" content="Splazers ========  synopsis splazers [options] genome file reads file splazers [options] genome file reads file 1 reads file 2  description splazers uses a prefix-suffix mapping strategy to split-map read sequences.if a sam file of mapped reads is given as input, all unmapped but anchoredreads are split-mapped onto anchoring target regions (specify option -an),if a fasta/q file of reads is given, reads are split-mapped onto the wholereference sequence. (c) copyright 2010 by anne-katrin emde. -h, --help displays this help message. --version display version information main options:: -o, --output file change output filename. default: reads file.result. -f, --forward only compute forward matches -r, --reverse only compute reverse complement matches -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num set the percent recognition rate in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -mcl, --min-clipped-len num min. read length for read clipping in range [1..inf]. default: 0. -qih, --quality-in-header quality string in fasta header -ou, --outputunmapped file output filename for unmapped reads -v, --verbose verbose mode -vv, --vverbose very verbose mode output format options:: -a, --alignment dump the alignment for each match -pa, --purge-ambiguous purge reads with more than max-hits best matches -dr, --distance-range num only consider matches with at most num more errors compared to the best (default output all) -of, --output-format num set output format. 0 = razers, 1 = enhanced fasta, 2 = eland, 3 = gff, 4 = sam. in range [0..4]. -gn, --genome-naming num select how genomes are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -rn, --read-naming num select how reads are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -so, --sort-order num select how matches are sorted. 0 = read number, 1 = genome position. in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). 0 = gap space, 1 = position space. in range [0..1]. default: 0. split mapping options:: -sm, --split-mapping num min. match length for prefix/suffix mapping (to disable split mapping, set to 0) default: 18. -maxg, --max-gap num max. length of middle gap default: 10000. -ming, --min-gap num min. length of middle gap (for edit distance mapping about 10% of read length is recommended) default: 0. -ep, --errors-prefix num max. number of errors in prefix match default: 1. -es, --errors-suffix num max. number of errors in suffix match default: 1. -gl, --genome-len num genome length in mb, for computation of expected number of random matches in range [-inf..10000]. default: 3000. -an, --anchored anchored split mapping, only unmapped reads with mapped mates will be considered, requires the reads to be given in sam format -pc, --penalty-c num percent of read length, used as penalty for split-gap default: 2. filtration options:: -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file.  version splazers version: 1.1 last update apr 2011">
  <meta name="twitter:image" content="https://www.carta.tech/images/seqan-apps-splazers-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/splazers.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="splazers (1) manual" />
  <meta property="og:description" content="Splazers ========  synopsis splazers [options] genome file reads file splazers [options] genome file reads file 1 reads file 2  description splazers uses a prefix-suffix mapping strategy to split-map read sequences.if a sam file of mapped reads is given as input, all unmapped but anchoredreads are split-mapped onto anchoring target regions (specify option -an),if a fasta/q file of reads is given, reads are split-mapped onto the wholereference sequence. (c) copyright 2010 by anne-katrin emde. -h, --help displays this help message. --version display version information main options:: -o, --output file change output filename. default: reads file.result. -f, --forward only compute forward matches -r, --reverse only compute reverse complement matches -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num set the percent recognition rate in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -mcl, --min-clipped-len num min. read length for read clipping in range [1..inf]. default: 0. -qih, --quality-in-header quality string in fasta header -ou, --outputunmapped file output filename for unmapped reads -v, --verbose verbose mode -vv, --vverbose very verbose mode output format options:: -a, --alignment dump the alignment for each match -pa, --purge-ambiguous purge reads with more than max-hits best matches -dr, --distance-range num only consider matches with at most num more errors compared to the best (default output all) -of, --output-format num set output format. 0 = razers, 1 = enhanced fasta, 2 = eland, 3 = gff, 4 = sam. in range [0..4]. -gn, --genome-naming num select how genomes are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -rn, --read-naming num select how reads are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -so, --sort-order num select how matches are sorted. 0 = read number, 1 = genome position. in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). 0 = gap space, 1 = position space. in range [0..1]. default: 0. split mapping options:: -sm, --split-mapping num min. match length for prefix/suffix mapping (to disable split mapping, set to 0) default: 18. -maxg, --max-gap num max. length of middle gap default: 10000. -ming, --min-gap num min. length of middle gap (for edit distance mapping about 10% of read length is recommended) default: 0. -ep, --errors-prefix num max. number of errors in prefix match default: 1. -es, --errors-suffix num max. number of errors in suffix match default: 1. -gl, --genome-len num genome length in mb, for computation of expected number of random matches in range [-inf..10000]. default: 3000. -an, --anchored anchored split mapping, only unmapped reads with mapped mates will be considered, requires the reads to be given in sam format -pc, --penalty-c num percent of read length, used as penalty for split-gap default: 2. filtration options:: -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file.  version splazers version: 1.1 last update apr 2011" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/seqan-apps-splazers-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">splazers<small> (1)</small></h1>
        <p class="lead">Splazers ========  synopsis splazers [options] genome file reads file splazers [options] genome file reads file 1 reads file 2  description splazers uses a prefix-suffix mapping strategy to split-map read sequences.if a sam file of mapped reads is given as input, all unmapped but anchoredreads are split-mapped onto anchoring target regions (specify option -an),if a fasta/q file of reads is given, reads are split-mapped onto the wholereference sequence. (c) copyright 2010 by anne-katrin emde. -h, --help displays this help message. --version display version information main options:: -o, --output file change output filename. default: reads file.result. -f, --forward only compute forward matches -r, --reverse only compute reverse complement matches -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num set the percent recognition rate in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -mcl, --min-clipped-len num min. read length for read clipping in range [1..inf]. default: 0. -qih, --quality-in-header quality string in fasta header -ou, --outputunmapped file output filename for unmapped reads -v, --verbose verbose mode -vv, --vverbose very verbose mode output format options:: -a, --alignment dump the alignment for each match -pa, --purge-ambiguous purge reads with more than max-hits best matches -dr, --distance-range num only consider matches with at most num more errors compared to the best (default output all) -of, --output-format num set output format. 0 = razers, 1 = enhanced fasta, 2 = eland, 3 = gff, 4 = sam. in range [0..4]. -gn, --genome-naming num select how genomes are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -rn, --read-naming num select how reads are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -so, --sort-order num select how matches are sorted. 0 = read number, 1 = genome position. in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). 0 = gap space, 1 = position space. in range [0..1]. default: 0. split mapping options:: -sm, --split-mapping num min. match length for prefix/suffix mapping (to disable split mapping, set to 0) default: 18. -maxg, --max-gap num max. length of middle gap default: 10000. -ming, --min-gap num min. length of middle gap (for edit distance mapping about 10% of read length is recommended) default: 0. -ep, --errors-prefix num max. number of errors in prefix match default: 1. -es, --errors-suffix num max. number of errors in suffix match default: 1. -gl, --genome-len num genome length in mb, for computation of expected number of random matches in range [-inf..10000]. default: 3000. -an, --anchored anchored split mapping, only unmapped reads with mapped mates will be considered, requires the reads to be given in sam format -pc, --penalty-c num percent of read length, used as penalty for split-gap default: 2. filtration options:: -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file.  version splazers version: 1.1 last update apr 2011</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/splazers.1.html">
      <span itemprop="name">splazers: Splazers ========  synopsis splazers [options] genome file reads file splazers [options] genome file reads file 1 reads file 2  description splazers uses a prefix-suffix mapping strategy to split-map read sequences.if a sam file of mapped reads is given as input, all unmapped but anchoredreads are split-mapped onto anchoring target regions (specify option -an),if a fasta/q file of reads is given, reads are split-mapped onto the wholereference sequence. (c) copyright 2010 by anne-katrin emde. -h, --help displays this help message. --version display version information main options:: -o, --output file change output filename. default: reads file.result. -f, --forward only compute forward matches -r, --reverse only compute reverse complement matches -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num set the percent recognition rate in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -mcl, --min-clipped-len num min. read length for read clipping in range [1..inf]. default: 0. -qih, --quality-in-header quality string in fasta header -ou, --outputunmapped file output filename for unmapped reads -v, --verbose verbose mode -vv, --vverbose very verbose mode output format options:: -a, --alignment dump the alignment for each match -pa, --purge-ambiguous purge reads with more than max-hits best matches -dr, --distance-range num only consider matches with at most num more errors compared to the best (default output all) -of, --output-format num set output format. 0 = razers, 1 = enhanced fasta, 2 = eland, 3 = gff, 4 = sam. in range [0..4]. -gn, --genome-naming num select how genomes are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -rn, --read-naming num select how reads are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -so, --sort-order num select how matches are sorted. 0 = read number, 1 = genome position. in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). 0 = gap space, 1 = position space. in range [0..1]. default: 0. split mapping options:: -sm, --split-mapping num min. match length for prefix/suffix mapping (to disable split mapping, set to 0) default: 18. -maxg, --max-gap num max. length of middle gap default: 10000. -ming, --min-gap num min. length of middle gap (for edit distance mapping about 10% of read length is recommended) default: 0. -ep, --errors-prefix num max. number of errors in prefix match default: 1. -es, --errors-suffix num max. number of errors in suffix match default: 1. -gl, --genome-len num genome length in mb, for computation of expected number of random matches in range [-inf..10000]. default: 3000. -an, --anchored anchored split mapping, only unmapped reads with mapped mates will be considered, requires the reads to be given in sam format -pc, --penalty-c num percent of read length, used as penalty for split-gap default: 2. filtration options:: -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file.  version splazers version: 1.1 last update apr 2011</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/seqan-apps/">
      <span itemprop="name">seqan-apps</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/splazers.1.html">
      <span itemprop="name">splazers: Splazers ========  synopsis splazers [options] genome file reads file splazers [options] genome file reads file 1 reads file 2  description splazers uses a prefix-suffix mapping strategy to split-map read sequences.if a sam file of mapped reads is given as input, all unmapped but anchoredreads are split-mapped onto anchoring target regions (specify option -an),if a fasta/q file of reads is given, reads are split-mapped onto the wholereference sequence. (c) copyright 2010 by anne-katrin emde. -h, --help displays this help message. --version display version information main options:: -o, --output file change output filename. default: reads file.result. -f, --forward only compute forward matches -r, --reverse only compute reverse complement matches -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num set the percent recognition rate in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -mcl, --min-clipped-len num min. read length for read clipping in range [1..inf]. default: 0. -qih, --quality-in-header quality string in fasta header -ou, --outputunmapped file output filename for unmapped reads -v, --verbose verbose mode -vv, --vverbose very verbose mode output format options:: -a, --alignment dump the alignment for each match -pa, --purge-ambiguous purge reads with more than max-hits best matches -dr, --distance-range num only consider matches with at most num more errors compared to the best (default output all) -of, --output-format num set output format. 0 = razers, 1 = enhanced fasta, 2 = eland, 3 = gff, 4 = sam. in range [0..4]. -gn, --genome-naming num select how genomes are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -rn, --read-naming num select how reads are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -so, --sort-order num select how matches are sorted. 0 = read number, 1 = genome position. in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). 0 = gap space, 1 = position space. in range [0..1]. default: 0. split mapping options:: -sm, --split-mapping num min. match length for prefix/suffix mapping (to disable split mapping, set to 0) default: 18. -maxg, --max-gap num max. length of middle gap default: 10000. -ming, --min-gap num min. length of middle gap (for edit distance mapping about 10% of read length is recommended) default: 0. -ep, --errors-prefix num max. number of errors in prefix match default: 1. -es, --errors-suffix num max. number of errors in suffix match default: 1. -gl, --genome-len num genome length in mb, for computation of expected number of random matches in range [-inf..10000]. default: 3000. -an, --anchored anchored split mapping, only unmapped reads with mapped mates will be considered, requires the reads to be given in sam format -pc, --penalty-c num percent of read length, used as penalty for split-gap default: 2. filtration options:: -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file.  version splazers version: 1.1 last update apr 2011</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    <nav>
  <ul class="pager">
   <li class="previous"><a href="splay.1.html"><span aria-hidden="true">&larr;</span> splay.1: Mpeg-1,2 audio layer 1,2,3 file player</a></li>
   <li class="next"><a href="spline.1.html">spline.1: Interpolate datasets using splines under tension <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
