<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>spline: Interpolate datasets using splines under tension</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Interpolate datasets using splines under tension">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="spline (1) manual">
  <meta name="twitter:description" content="Interpolate datasets using splines under tension">
  <meta name="twitter:image" content="https://www.carta.tech/images/plotutils-spline-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/spline.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="spline (1) manual" />
  <meta property="og:description" content="Interpolate datasets using splines under tension" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/plotutils-spline-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">spline<small> (1)</small></h1>
        <p class="lead">Interpolate datasets using splines under tension</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/spline.1.html">
      <span itemprop="name">spline: Interpolate datasets using splines under tension</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/plotutils/">
      <span itemprop="name">plotutils</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/spline.1.html">
      <span itemprop="name">spline: Interpolate datasets using splines under tension</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>spline</strong> [ <em>options</em> ] [ <em>files</em> ]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>spline</strong> reads datasets from standard input or from one or more files, and fits a smooth curve (a "spline") through each dataset. An interpolated version of each dataset, consisting of points from the smooth curve, is written to standard output.</p><p>Unless the <strong>-a</strong> or <strong>-A</strong> options are used (see below), each dataset should be a sequence of values for a vector-valued function of a single scalar variable. That is, each dataset should be a sequence of data points, given as alternating <em>t</em>\^ and <em>y</em>\^ values. <em>t</em>\^ is a scalar independent variable, and <em>y</em>\^ is a vector-valued dependent variable. The dimensionality of <em>y</em>\^ is specified with the <strong>-d</strong> option (the default dimensionality is 1). Between each data point and the next, <em>t</em>\^ should increase.</p><p>An input file may contain more than a single dataset. If an input file is in <small>ASCII</small> format (the default), its datasets should be separated by blank lines. The <em>t</em>\^ and <em>y</em>\^ values of the data points in each dataset may be arranged arbitrarily, so long as they are separated by white space. Besides datasets, an input file may contain any number of comment lines, which should begin with the comment character `#'. Comment lines are ignored. They are not treated as blank, i.e., they do not interrupt a dataset in progress.</p><p>Options and file names may be interspersed on the command line, but the options are processed before the file names are read. If <strong>--</strong> is seen, it is interpreted as the end of the options. If no file names are specified, or the file name <strong>-</strong> is encountered, the standard input is read.</p><p>The type of interpolation, and the format of the input and output files, may be selected by command-line options.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<h3>Interpolation-Related Options</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>-f</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--filter</strong></p>
  </dt>
  <dd>
    <p>Use a local interpolation algorithm (the cubic Bessel algorithm), so that <strong>spline</strong> can be used as a real-time filter. The slope of the interpolating curve at each point in a dataset will be chosen by fitting a quadratic function through that point and the two adjacent points in the dataset. If <strong>-f</strong> is specified then the <strong>-t</strong> option, otherwise optional, must be used as well. Also, if <strong>-f</strong> is specified then the <strong>-k</strong>, <strong>-p</strong>, and <strong>-T</strong> options may not be used.</p><ul>
<li><p>If <strong>-f</strong> is <em>not</em>\^ specified, then the default (global) interpolation algorithm will be used.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-k</strong><em> k</em></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--boundary-condition</strong><em> k</em></p>
  </dt>
  <dd>
    <p>Set the boundary condition parameter for each constructed spline to be <em>k</em>. (The default value is 1.0.) In each of its components, the spline will satisfy the two boundary conditions y"[0]=ky"[1] and y"[n]=ky"[n-1]. Here y[0] and y[1] signify the values of a specified component of the vector-valued dependent variable <em>y</em>\^ at the first two points of a dataset, and y[n-1] and y[n] the values at the last two points. Setting <em>k</em>\^ to zero will yield a "natural" spline, i.e., one that has zero curvature at the two ends of the dataset. The <strong>-k</strong> option may not be used if <strong>-f</strong> or <strong>-p</strong> is specified.</p>
  </dd>
  <dt>
    <p><strong>-n</strong><em> n</em></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--no-of-intervals</strong><em> n</em></p>
  </dt>
  <dd>
    <p>Subdivide the interval over which interpolation occurs into <em>n</em>\^ subintervals. The number of data points computed, and written to the output, will be <em>n+1</em>. The default value for <em>n</em>\^ is 100.</p>
  </dd>
  <dt>
    <p><strong>-p</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--periodic</strong></p>
  </dt>
  <dd>
    <p>Construct a periodic spline. If this option is specified, the <em>y</em>\^ values for the first and last points in each dataset must be equal. The <strong>-f</strong> and <strong>-k</strong> options may not be used if <strong>-p</strong> is specified.</p>
  </dd>
  <dt>
    <p><strong>-T</strong><em> tension</em></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--tension</strong><em> tension</em></p>
  </dt>
  <dd>
    <p>Each interpolating curve will be a spline under tension. This option sets the tension value (the default is 0.0).</p><ul>
<li><p>If <em>tension</em>\^ equals zero, the curve will be a piecewise cubic spline. Increasing the tension above zero makes the curve "tighter", and reduces the likelihood of spurious inflection points. That is because between each pair of successive points in a dataset, the curve will satisfy the fourth-order differential equation y""=sgn(<em>tension</em>\^)*(<em>tension</em>\^^2)y" in each of its components. As <em>tension</em>\^ increases to positive infinity, it will converge to a polygonal line. The <strong>-T</strong> option may not be used if <strong>-f</strong> is specified.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-t </strong><em>tmin tmax [tspacing]</em><strong></strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--t-spacing </strong><em>tmin tmax [tspacing]</em><strong></strong></p>
  </dt>
  <dd>
    <p>For each dataset, set the interval over which interpolation occurs to be the interval between <em>tmin</em>\^ and <em>tmax</em>. If <em>tspacing</em>\^ is not specified, the interval will be divided into the number of subintervals specified by the <strong>-n</strong> option.</p><ul>
<li><p>If the <strong>-t</strong> option is not used, the interval over which interpolation occurs will be the entire range of the independent variable in the dataset. The <strong>-t</strong> option must always be used if the <strong>-f</strong> option is used to request filter-like behavior (see above).</p></li>
</ul>
  </dd>

</dl>

<h3>Format-Related Options</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>-d</strong><em> dimension</em></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--y-dimension</strong><em> dimension</em></p>
  </dt>
  <dd>
    <p>Set the dimensionality of the dependent variable <em>y</em> in the input and output files to be <em>dimension</em>. The default dimension is 1.</p>
  </dd>
  <dt>
    <p><strong>-I</strong><em> data-format</em></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--input-format</strong><em> data-format</em></p>
  </dt>
  <dd>
    <p>Set the data format for the input file(s) to be <em>data-format</em>, which may be one of the following.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>a</strong></p>
  </dt>
  <dd>
    <p><small>ASCII</small> format (the default). Each file is a sequence of floating point numbers, interpreted as the <em>t</em>\^ and <em>y</em>\^ coordinates of the successive data points in a dataset. If <em>y</em>\^ is <em>d</em>\^-dimensional, there will be <em>d+1</em>\^ numbers for each point. The <em>t</em>\^ and <em>y</em>\^ coordinates of a point need not appear on the same line, and points need not appear on different lines. But if a blank line occurs (i.e., two newlines in succession are seen), it is interpreted as the end of a dataset, and the beginning of the next.</p>
  </dd>
  <dt>
    <p><strong>f</strong></p>
  </dt>
  <dd>
    <p>Single precision binary format. Each file is a sequence of floating point numbers, interpreted as the <em>t</em>\^ and <em>y</em>\^ coordinates of the successive data points in a dataset. If <em>y</em>\^ is <em>d</em>\^-dimensional, there will be <em>d+1</em>\^ numbers for each point. Successive datasets are separated by a single occurrence of the quantity <small>FLT_MAX,</small> which is the largest possible single precision floating point number. On most machines this is approximately 3.4x10^38.</p>
  </dd>
  <dt>
    <p><strong>d</strong></p>
  </dt>
  <dd>
    <p>Double precision binary format. Each file is a sequence of double precision floating point numbers, interpreted as the <em>t</em>\^ and <em>y</em>\^ coordinates of the successive data points in a dataset. If <em>y</em>\^ is <em>d</em>\^-dimensional, there will be <em>d+1</em>\^ numbers for each point. Successive datasets are separated by a single occurrence of the quantity <small>DBL_MAX,</small> which is the largest possible double precision floating point number. On most machines this is approximately 1.8x10^308.</p>
  </dd>
  <dt>
    <p><strong>i</strong></p>
  </dt>
  <dd>
    <p>Integer binary format. Each file is a sequence of integers, interpreted as the <em>t</em>\^ and <em>y</em>\^ coordinates of the successive data points in a dataset. If <em>y</em>\^ is <em>d</em>\^-dimensional, there will be <em>d+1</em>\^ numbers for each point. Successive datasets are separated by a single occurrence of the quantity <small>INT_MAX,</small> which is the largest possible integer. On most machines this is 2^31-1.</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-a </strong><em>[step_size [lower_limit]]</em><strong></strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--auto-abscissa </strong><em>[step_size [lower_limit]]</em><strong></strong></p>
  </dt>
  <dd>
    <p>Automatically generate values for <em>t</em>, the independent variable (the default values of <em>step_size</em>\^ and <em>lower_limit</em>\^ are 1.0 and 0.0, respectively).</p><ul>
<li><p>Irrespective of data format (`a', `f', `d', or `i'), this option specifies that the values of <em>t</em>\^ are missing from the input file: the dataset(s) to be read contain only values of <em>y</em>, the dependent variable. So if <em>y</em>\^ is <em>d</em>\^-dimensional, there will be only <em>d</em>\^ numbers for each point. The increment from each <em>t</em>\^ value to the next will be <em>step_size</em>, and the first <em>t</em>\^ value will be <em>lower_limit</em>. This option is useful, e.g., when interpolating curves rather than functions.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-A</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--auto-dist-abscissa</strong></p>
  </dt>
  <dd>
    <p>Automatically generate values for <em>t</em>, the independent variable. This is a variant form of the <strong>-a</strong> option. The increment from each <em>t</em>\^ value to the next will be the distance in <em>d</em>\^-dimensional space between the corresponding <em>y</em>\^ values, and the first <em>t</em>\^ value will be 0.0. That is, <em>t</em>\^ will be "polygonal arclength". This option is useful when interpolating curves rather than functions.</p>
  </dd>
  <dt>
    <p><strong>-O</strong><em> data-format</em></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--output-format</strong><em> data-format</em></p>
  </dt>
  <dd>
    <p>Set the data format for the output file to be <em>data-format</em>. The interpretation of <em>data-format</em>\^ is the same as for the <strong>-I</strong> option. The default is `a', i.e., <small>ASCII</small> format.</p>
  </dd>
  <dt>
    <p><strong>-P</strong><em> significant-digits</em></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--precision</strong><em> significant-digits</em></p>
  </dt>
  <dd>
    <p>Set the numerical precision for the <em>t</em>\^ and <em>y</em>\^ values in the output file to be <em>significant-digits</em>. This takes effect only if the output file is written in `a' format, i.e., in <small>ASCII.</small> <em>significant-digits</em>\^ must be a positive integer (the default is 6).</p>
  </dd>
  <dt>
    <p><strong>-s</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--suppress-abscissa</strong></p>
  </dt>
  <dd>
    <p>Omit the independent variable <em>t</em>\^ from the output file; for each point, supply only the dependent variable <em>y</em>. If <em>y</em>\^ is <em>d</em>\^-dimensional, there will be only <em>d</em>\^ numbers for each point, not <em>d+1</em>. This option is useful when interpolating curves rather than functions.</p>
  </dd>

</dl>

<h3>Informational Options</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--help</strong></p>
  </dt>
  <dd>
    <p>Print a list of command-line options, and exit.</p>
  </dd>
  <dt>
    <p><strong>--version</strong></p>
  </dt>
  <dd>
    <p>Print the version number of <strong>spline</strong> and the plotting utilities package, and exit.</p>
  </dd>

</dl>


        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXAMPLES</h2>
        <div class="sectioncontent">
<p>Typing</p><p><strong>echo 0 0 1 1 2 0 \||\| spline</strong></p><p>will produce on standard output an interpolated dataset consisting of 101 data points. If graphed, this interpolated dataset will yield a parabola.</p><p>It is sometimes useful to interpolate between a sequence of arbitrarily placed points in <em>d</em>\^-dimensional space, i.e., to "spline a curve" rather than a function. The <strong>-a</strong> and <strong>-s</strong> options are used for this. For example,</p><p><strong>echo 0 0 1 0 1 1 0 1 \||\| spline -d 2 -a -s</strong></p><p>will produce on standard output a 101-point dataset that interpolates between the four points (0,0), (1,0), (1,1), and (0,1). The <strong>-d 2</strong> option specifies that the dependent variable <em>y</em>\^ is two-dimensional. The <strong>-a</strong> option specifies that the <em>t</em>\^ values are missing from the input and should be automatically generated. The <strong>-s</strong> option specifies that the <em>t</em>\^ values should be stripped from the output.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHORS</h2>
        <div class="sectioncontent">
<p><strong>spline</strong> was written by Robert S. Maier (<strong>rsm@math.arizona.edu</strong>), starting with an earlier version by Rich Murphey (<strong>rich@freebsd.org</strong>). The algorithms for constructing splines under tension are similar to those used in the FITPACK subroutine library, and are ultimately due to Alan K. Cline (<strong>cline@cs.utexas.edu</strong>).</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO spline&hellip;</h2>
        <div class="sectioncontent">
<p>"The GNU Plotting Utilities Manual".</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>Email bug reports to <strong>bug-gnu-utils@gnu.org</strong>.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="splazers.1.html"><span aria-hidden="true">&larr;</span> splazers.1: Splazers ========  synopsis splazers [options] genome file reads file splazers [options] genome file reads file 1 reads file 2  description splazers uses a prefix-suffix mapping strategy to split-map read sequences.if a sam file of mapped reads is given as input, all unmapped but anchoredreads are split-mapped onto anchoring target regions (specify option -an),if a fasta/q file of reads is given, reads are split-mapped onto the wholereference sequence. (c) copyright 2010 by anne-katrin emde. -h, --help displays this help message. --version display version information main options:: -o, --output file change output filename. default: reads file.result. -f, --forward only compute forward matches -r, --reverse only compute reverse complement matches -i, --percent-identity num percent identity threshold. in range [50..100]. default: 92. -rr, --recognition-rate num set the percent recognition rate in range [80..100]. default: 99. -pd, --param-dir dir read user-computed parameter files in the directory dir. -id, --indels allow indels. default: mismatches only. -ll, --library-length num paired-end library length. in range [1..inf]. default: 220. -le, --library-error num paired-end library length tolerance. in range [0..inf]. default: 50. -m, --max-hits num output only num of the best hits. in range [1..inf]. default: 100. --unique output only unique best matches (-m 1 -dr 0 -pa). -tr, --trim-reads num trim reads to given length. default: off. in range [14..inf]. -mcl, --min-clipped-len num min. read length for read clipping in range [1..inf]. default: 0. -qih, --quality-in-header quality string in fasta header -ou, --outputunmapped file output filename for unmapped reads -v, --verbose verbose mode -vv, --vverbose very verbose mode output format options:: -a, --alignment dump the alignment for each match -pa, --purge-ambiguous purge reads with more than max-hits best matches -dr, --distance-range num only consider matches with at most num more errors compared to the best (default output all) -of, --output-format num set output format. 0 = razers, 1 = enhanced fasta, 2 = eland, 3 = gff, 4 = sam. in range [0..4]. -gn, --genome-naming num select how genomes are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -rn, --read-naming num select how reads are named. 0 = use fasta id, 1 = enumerate beginning with 1. in range [0..1]. default: 0. -so, --sort-order num select how matches are sorted. 0 = read number, 1 = genome position. in range [0..1]. default: 0. -pf, --position-format num select begin/end position numbering (see coordinate section below). 0 = gap space, 1 = position space. in range [0..1]. default: 0. split mapping options:: -sm, --split-mapping num min. match length for prefix/suffix mapping (to disable split mapping, set to 0) default: 18. -maxg, --max-gap num max. length of middle gap default: 10000. -ming, --min-gap num min. length of middle gap (for edit distance mapping about 10% of read length is recommended) default: 0. -ep, --errors-prefix num max. number of errors in prefix match default: 1. -es, --errors-suffix num max. number of errors in suffix match default: 1. -gl, --genome-len num genome length in mb, for computation of expected number of random matches in range [-inf..10000]. default: 3000. -an, --anchored anchored split mapping, only unmapped reads with mapped mates will be considered, requires the reads to be given in sam format -pc, --penalty-c num percent of read length, used as penalty for split-gap default: 2. filtration options:: -oc, --overabundance-cut num set k-mer overabundance cut ratio. in range [0..1]. -rl, --repeat-length num skip simple-repeats of length num. in range [1..inf]. default: 1000. -tl, --taboo-length num set taboo length. in range [1..inf]. default: 1. -lm, --low-memory decrease memory usage at the expense of runtime verification options: -mn, --match-n n matches all other characters. default: n matches nothing. -ed, --error-distr file write error distribution to file.  version splazers version: 1.1 last update apr 2011</a></li>
   <li class="next"><a href="splint.1.html">splint.1: A tool for statically checking c programs <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
