<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>sreformat: </title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="sreformat (1) manual">
  <meta name="twitter:description" content="">
  <meta name="twitter:image" content="https://www.carta.tech/images/biosquid-sreformat-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/sreformat.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="sreformat (1) manual" />
  <meta property="og:description" content="" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/biosquid-sreformat-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">sreformat<small> (1)</small></h1>
        <p class="lead"></p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/sreformat.1.html">
      <span itemprop="name">sreformat: </span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/biosquid/">
      <span itemprop="name">biosquid</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/sreformat.1.html">
      <span itemprop="name">sreformat: </span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>sreformat</strong> <em>[options]</em> <em>format</em> <em>seqfile</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>sreformat</strong> reads the sequence file <em>seqfile</em> in any supported format, reformats it into a new format specified by <em>format,</em> then prints the reformatted text.</p><p>Supported input formats include (but are not limited to) the unaligned formats FASTA, Genbank, EMBL, SWISS-PROT, PIR, and GCG, and the aligned formats Stockholm, Clustal, GCG MSF, and Phylip.</p><p>Available unaligned output file format codes include <em>fasta</em> (FASTA format); <em>embl</em> (EMBL/SWISSPROT format); <em>genbank</em> (Genbank format); <em>gcg</em> (GCG single sequence format); <em>gcgdata</em> (GCG flatfile database format); <em>pir</em> (PIR/CODATA flatfile format); <em>raw</em> (raw sequence, no other information).</p><p>The available aligned output file format codes include <em>stockholm</em> (PFAM/Stockholm format); <em>msf</em> (GCG MSF format); <em>a2m</em> (an aligned FASTA format); <em>PHYLIP</em> (Felsenstein's PHYLIP format); and <em>clustal</em> (Clustal V/W/X format); and <em>selex</em> (the old SELEX/HMMER/Pfam annotated alignment format);</p><p>All thee codes are interpreted case-insensitively (e.g. MSF, Msf, or msf all work).</p><p>Unaligned format files cannot be reformatted to aligned formats. However, aligned formats can be reformatted to unaligned formats -- gap characters are simply stripped out.</p><p>This program was originally named <strong>reformat,</strong> but that name clashes with a GCG program of the same name.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>-d</strong></p>
  </dt>
  <dd>
    <p>DNA; convert U's to T's, to make sure a nucleic acid sequence is shown as DNA not RNA. See <strong>-r.</strong></p>
  </dd>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Print brief help; includes version number and summary of all options, including expert options.</p>
  </dd>
  <dt>
    <p><strong>-l</strong></p>
  </dt>
  <dd>
    <p>Lowercase; convert all sequence residues to lower case. See <strong>-u.</strong></p>
  </dd>
  <dt>
    <p><strong>-n</strong></p>
  </dt>
  <dd>
    <p>For DNA/RNA sequences, converts any character that's not unambiguous RNA/DNA (e.g. ACGTU/acgtu) to an N. Used to convert IUPAC ambiguity codes to N's, for software that can't handle all IUPAC codes (some public RNA folding codes, for example). If the file is an alignment, gap characters are also left unchanged. If sequences are not nucleic acid sequences, this option will corrupt the data in a predictable fashion.</p>
  </dd>
  <dt>
    <p><strong>-r</strong></p>
  </dt>
  <dd>
    <p>RNA; convert T's to U's, to make sure a nucleic acid sequence is shown as RNA not DNA. See <strong>-d.</strong></p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Uppercase; convert all sequence residues to upper case. See <strong>-l.</strong></p>
  </dd>
  <dt>
    <p><strong>-x</strong></p>
  </dt>
  <dd>
    <p>For DNA sequences, convert non-IUPAC characters (such as X's) to N's. This is for compatibility with benighted people who insist on using X instead of the IUPAC ambiguity character N. (X is for ambiguity in an amino acid residue).</p><ul>
<li><p>Warning: like the <strong>-n</strong> option, the code doesn't check that you are actually giving it DNA. It simply literally just converts non-IUPAC DNA symbols to N. So if you accidentally give it protein sequence, it will happily convert most every amino acid residue to an N.</p></li>
</ul>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">EXPERT OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <p><strong>--gapsym</strong><em> &lt;c&gt;</em></p>
  </dt>
  <dd>
    <p>Convert all gap characters to <em>&lt;c&gt;.</em> Used to prepare alignment files for programs with strict requirements for gap symbols. Only makes sense if the input <em>seqfile</em> is an alignment.</p>
  </dd>
  <dt>
    <p><strong>--informat</strong><em> &lt;s&gt;</em></p>
  </dt>
  <dd>
    <p>Specify that the sequence file is in format <em>&lt;s&gt;,</em> rather than allowing the program to autodetect the file format. Common examples include Genbank, EMBL, GCG, PIR, Stockholm, Clustal, MSF, or PHYLIP; see the printed documentation for a complete list of accepted format names.</p>
  </dd>
  <dt>
    <p><strong>--mingap</strong></p>
  </dt>
  <dd>
    <p>If <em>seqfile</em> is an alignment, remove any columns that contain 100% gap characters, minimizing the overall length of the alignment. (Often useful if you've extracted a subset of aligned sequences from a larger alignment.)</p>
  </dd>
  <dt>
    <p><strong>--nogap</strong></p>
  </dt>
  <dd>
    <p>Remove any aligned columns that contain any gap symbols at all. Useful as a prelude to phylogenetic analyses, where you only want to analyze columns containing 100% residues, so you want to strip out any columns with gaps in them. Only makes sense if the file is an alignment file.</p>
  </dd>
  <dt>
    <p><strong>--pfam</strong></p>
  </dt>
  <dd>
    <p>For SELEX alignment output format only, put the entire alignment in one block (don't wrap into multiple blocks). This is close to the format used internally by Pfam in Stockholm and Cambridge.</p>
  </dd>
  <dt>
    <p><strong>--sam</strong></p>
  </dt>
  <dd>
    <p>Try to convert gap characters to UC Santa Cruz SAM style, where a . means a gap in an insert column, and a - means a deletion in a consensus/match column. This only works for converting aligned file formats, and only if the alignment already adheres to the SAM convention of upper case for residues in consensus/match columns, and lower case for residues in insert columns. This is true, for instance, of all alignments produced by old versions of HMMER. (HMMER2 produces alignments that adhere to SAM's conventions even in gap character choice.) This option was added to allow Pfam alignments to be reformatted into something more suitable for profile HMM construction using the UCSC SAM software.</p>
  </dd>
  <dt>
    <p><strong>--samfrac</strong><em> &lt;x&gt;</em></p>
  </dt>
  <dd>
    <p>Try to convert the alignment gap characters and residue cases to UC Santa Cruz SAM style, where a . means a gap in an insert column and a - means a deletion in a consensus/match column, and upper case means match/consensus residues and lower case means inserted resiudes. This will only work for converting aligned file formats, but unlike the <strong>--sam</strong> option, it will work regardless of whether the file adheres to the upper/lower case residue convention. Instead, any column containing more than a fraction <em>&lt;x&gt;</em> of gap characters is interpreted as an insert column, and all other columns are interpreted as match columns. This option was added to allow Pfam alignments to be reformatted into something more suitable for profile HMM construction using the UCSC SAM software.</p>
  </dd>
  <dt>
    <p><strong>--wussify</strong></p>
  </dt>
  <dd>
    <p>Convert RNA secondary structure annotation strings (both consensus and individual) from old "KHS" format, &gt;&lt;, to the new WUSS notation, &lt;&gt;. If the notation is already in WUSS format, this option will screw it up, without warning. Only SELEX and Stockholm format files have secondary structure markup at present.</p>
  </dd>
  <dt>
    <p><strong>--dewuss</strong></p>
  </dt>
  <dd>
    <p>Convert RNA secondary structure annotation strings from the new WUSS notation, &lt;&gt;, back to the old KHS format, &gt;&lt;. If the annotation is already in KHS, this option will corrupt it, without warning. Only SELEX and Stockholm format files have secondary structure markup.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO sreformat&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man1/afetch.1.html"><strong>afetch</strong>(1)</a>, <a href="../man1/alistat.1.html"><strong>alistat</strong>(1)</a>, <a href="../man1/compalign.1.html"><strong>compalign</strong>(1)</a>, <a href="../man1/compstruct.1.html"><strong>compstruct</strong>(1)</a>, <a href="../man1/revcomp.1.html"><strong>revcomp</strong>(1)</a>, <a href="../man1/seqsplit.1.html"><strong>seqsplit</strong>(1)</a>, <a href="../man1/seqstat.1.html"><strong>seqstat</strong>(1)</a>, <a href="../man1/sfetch.1.html"><strong>sfetch</strong>(1)</a>, <a href="../man1/shuffle.1.html"><strong>shuffle</strong>(1)</a>, <a href="../man1/sindex.1.html"><strong>sindex</strong>(1)</a>, <a href="../man1/stranslate.1.html"><strong>stranslate</strong>(1)</a>, <a href="../man1/weight.1.html"><strong>weight</strong>(1)</a>.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Biosquid and its documentation are Copyright (C) 1992-2003 HHMI/Washington University School of Medicine Freely distributed under the GNU General Public License (GPL) See COPYING in the source code distribution for more details, or contact me.</p>
<pre>
Sean Eddy
HHMI/Department of Genetics
Washington University School of Medicine
4444 Forest Park Blvd., Box 8510
St Louis, MO 63108 USA
Phone: 1-314-362-7666
FAX  : 1-314-362-2157
Email: eddy@genetics.wustl.edu
</pre>

        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="srec_input.1.html"><span aria-hidden="true">&larr;</span> srec_input.1: Input file specifications</a></li>
   <li class="next"><a href="sreport.1.html">sreport.1: Generate reports from the slurm accounting data. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
