<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>tbl2asn: Prepare a genbank submission using an ascii feature table</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Prepare a genbank submission using an ascii feature table">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="tbl2asn (1) manual">
  <meta name="twitter:description" content="Prepare a genbank submission using an ascii feature table">
  <meta name="twitter:image" content="https://www.carta.tech/images/ncbi-tools-bin-tbl2asn-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/tbl2asn.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="tbl2asn (1) manual" />
  <meta property="og:description" content="Prepare a genbank submission using an ascii feature table" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/ncbi-tools-bin-tbl2asn-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">tbl2asn<small> (1)</small></h1>
        <p class="lead">Prepare a genbank submission using an ascii feature table</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/tbl2asn.1.html">
      <span itemprop="name">tbl2asn: Prepare a genbank submission using an ascii feature table</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/ncbi-tools-bin/">
      <span itemprop="name">ncbi-tools-bin</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/tbl2asn.1.html">
      <span itemprop="name">tbl2asn: Prepare a genbank submission using an ascii feature table</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>tbl2asn</strong> [\|<strong>-</strong>\|] [\|<strong>-A</strong>&nbsp;<em>str</em>\|] [\|<strong>-C</strong>&nbsp;<em>str</em>\|] [\|<strong>-D</strong>&nbsp;<em>filename</em>\|] [\|<strong>-E</strong>\|] [\|<strong>-F</strong>&nbsp;<em>str</em>\|] [\|<strong>-G</strong>&nbsp;<em>str</em>\|] [\|<strong>-H</strong>&nbsp;<em>str</em>\|] [\|<strong>-J</strong>\|] [\|<strong>-K</strong>\|] [\|<strong>-L</strong>\|] [\|<strong>-M</strong>&nbsp;<em>str</em>\|] [\|<strong>-N</strong>&nbsp;<em>n</em>\|] [\|<strong>-O</strong>\|] [\|<strong>-P</strong>\|] [\|<strong>-Q</strong>&nbsp;<em>str</em>\|] [\|<strong>-R</strong>\|] [\|<strong>-S</strong>\|] [\|<strong>-T</strong>\|] [\|<strong>-U</strong>\|] [\|<strong>-V</strong>&nbsp;<em>str</em>\|] [\|<strong>-W</strong>\|] [\|<strong>-X</strong>&nbsp;<em>str</em>\|] [\|<strong>-Y</strong>&nbsp;<em>filename</em>\|] [\|<strong>-Z</strong>&nbsp;<em>filename</em>\|] [\|<strong>-a</strong>&nbsp;<em>str</em>\|] [\|<strong>-b</strong>\|] [\|<strong>-c</strong>&nbsp;<em>str</em>\|] [\|<strong>-f</strong>&nbsp;<em>filename</em>\|] [\|<strong>-g</strong>\|] [\|<strong>-h</strong>\|] [\|<strong>-i</strong>&nbsp;<em>filename</em>\|] [\|<strong>-j</strong>&nbsp;<em>str</em>\|] [\|<strong>-k</strong>&nbsp;<em>str</em>\|] [\|<strong>-n</strong>&nbsp;<em>str</em>\|] [\|<strong>-o</strong>&nbsp;<em>filename</em>\|] [\|<strong>-p</strong>&nbsp;<em>str</em>\|] [\|<strong>-q</strong>\|] [\|<strong>-r</strong>&nbsp;<em>str</em>\|] [\|<strong>-s</strong>\|] [\|<strong>-t</strong>&nbsp;<em>filename</em>\|] [\|<strong>-u</strong>\|] [\|<strong>-v</strong>\|] [\|<strong>-w</strong>&nbsp;<em>filename</em>\|] [\|<strong>-x</strong>&nbsp;<em>str</em>\|] [\|<strong>-y</strong>&nbsp;<em>str</em>\|] [\|<strong>-z</strong>\|]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>tbl2asn</strong> reads a template along with sequence and table files, and outputs ASN.1 for submission to GenBank. Thus, the submitter does not need to read each set of table and sequence files into Sequin. Furthermore, the template file can contain the organism and submitter information common to all records, obviating the need to input these data for each sequence/table pair.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<p>A summary of options is included below.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>-</strong></p>
  </dt>
  <dd>
    <p>Print usage message</p>
  </dd>
  <dt>
    <p><strong>-a</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Accession</p>
  </dd>
  <dt>
    <p><strong>-C</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Genome Center tag</p>
  </dd>
  <dt>
    <p><strong>-D</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Descriptors file</p>
  </dd>
  <dt>
    <p><strong>-E</strong></p>
  </dt>
  <dd>
    <p>Recurse</p>
  </dd>
  <dt>
    <p><strong>-F</strong></p>
  </dt>
  <dd>
    <p>Feature ID links (<strong>o</strong> by Overlap, <strong>p</strong> by Product)</p>
  </dd>
  <dt>
    <p><strong>-G</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Alignment Gap Flags (comma separated fields, e.g., <strong>p,-,-,-,?,.</strong> ) <strong>n</strong> Nucleotide or <strong>p</strong> Protein, Begin, Middle, End Gap Characters, Missing Characters, Match Characters Alignment middle Gap characters</p>
  </dd>
  <dt>
    <p><strong>-H</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Hold until publication:</p>
<dl class='dl-vertical'>
  <dt>
    y
  </dt>
  <dd>
    <p>For one year</p>
  </dd>
  <dt>
    <em>mm/dd/yyyy</em>
  </dt>
  <dd>
    <p>Until the specified date</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-J</strong></p>
  </dt>
  <dd>
    <p>Delayed genomic product set</p>
  </dd>
  <dt>
    <p><strong>-K</strong></p>
  </dt>
  <dd>
    <p>Safe Bioseq-set</p>
  </dd>
  <dt>
    <p><strong>-L</strong></p>
  </dt>
  <dd>
    <p>Force Local protein_id/transcript_id</p>
  </dd>
  <dt>
    <p><strong>-M</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Master genome flags:</p>
<dl class='dl-vertical'>
  <dt>
    n
  </dt>
  <dd>
    <p>Normal</p>
  </dd>
  <dt>
    b
  </dt>
  <dd>
    <p>Big sequence</p>
  </dd>
  <dt>
    p
  </dt>
  <dd>
    <p>Power option</p>
  </dd>
  <dt>
    t
  </dt>
  <dd>
    <p>TSA</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-N</strong>&nbsp;<em>n</em></p>
  </dt>
  <dd>
    <p>Project version number</p>
  </dd>
  <dt>
    <p><strong>-O</strong></p>
  </dt>
  <dd>
    <p>Allow run-on ORFs</p>
  </dd>
  <dt>
    <p><strong>-P</strong></p>
  </dt>
  <dd>
    <p>Remote publication lookup</p>
  </dd>
  <dt>
    <p><strong>-Q</strong></p>
  </dt>
  <dd>
    <p>mRNA title policy</p>
<dl class='dl-vertical'>
  <dt>
    s
  </dt>
  <dd>
    <p>Special mRNA titles</p>
  </dd>
  <dt>
    r
  </dt>
  <dd>
    <p>RefSeq mRNA titles</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-R</strong></p>
  </dt>
  <dd>
    <p>Remote sequence record fetching from ID</p>
  </dd>
  <dt>
    <p><strong>-S</strong></p>
  </dt>
  <dd>
    <p>Smart feature annotation</p>
  </dd>
  <dt>
    <p><strong>-T</strong></p>
  </dt>
  <dd>
    <p>Remote Taxonomy lookup</p>
  </dd>
  <dt>
    <p><strong>-U</strong></p>
  </dt>
  <dd>
    <p>Remove Unnecessary gene xref</p>
  </dd>
  <dt>
    <p><strong>-V</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Verification (combine any of the following letters)</p>
<dl class='dl-vertical'>
  <dt>
    v
  </dt>
  <dd>
    <p>Validate with Normal Stringency</p>
  </dd>
  <dt>
    r
  </dt>
  <dd>
    <p>Validate without Country Check</p>
  </dd>
  <dt>
    c
  </dt>
  <dd>
    <p>BarCode validation</p>
  </dd>
  <dt>
    b
  </dt>
  <dd>
    <p>Generate GenBank Flatfile</p>
  </dd>
  <dt>
    g
  </dt>
  <dd>
    <p>Generate Gene Report</p>
  </dd>
  <dt>
    t
  </dt>
  <dd>
    <p>Validate with TSA check</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-W</strong></p>
  </dt>
  <dd>
    <p>Log progress</p>
  </dd>
  <dt>
    <p><strong>-X</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Extra flags (combine any of the following letters)</p>
<dl class='dl-vertical'>
  <dt>
    A
  </dt>
  <dd>
    <p>Automatic definition line generation</p>
  </dd>
  <dt>
    C
  </dt>
  <dd>
    <p>Apply Comments in <strong>.cmt</strong> files to all sequences</p>
  </dd>
  <dt>
    E
  </dt>
  <dd>
    <p>Treat like Eukarypota in the discrepancy report</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-Y</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Read a comment string from <em>filename</em></p>
  </dd>
  <dt>
    <p><strong>-Z</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Write a discrepancy report to <em>filename</em></p>
  </dd>
  <dt>
    <p><strong>-a</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>File type:</p>
<dl class='dl-vertical'>
  <dt>
    a
  </dt>
  <dd>
    <p>Any (default)</p>
  </dd>
  <dt>
    r20u
  </dt>
  <dd>
    <p>Runs of 20+ Ns are gaps, 100 Ns are unknown length</p>
  </dd>
  <dt>
    r20k
  </dt>
  <dd>
    <p>Runs of 20+ Ns are gaps, 100 Ns are known length</p>
  </dd>
  <dt>
    r10u
  </dt>
  <dd>
    <p>Runs of 10+ Ns are gaps, 100 Ns are unknown length</p>
  </dd>
  <dt>
    r10k
  </dt>
  <dd>
    <p>Runs of 10+ Ns are gaps, 100 Ns are known length</p>
  </dd>
  <dt>
    s
  </dt>
  <dd>
    <p>FASTA Set (<strong>s</strong> Batch, <strong>s1</strong> Pop, <strong>s2</strong> Phy, <strong>s3</strong> Mut, <strong>s4</strong> Eco, <strong>s9</strong> Small-genome)</p>
  </dd>
  <dt>
    d
  </dt>
  <dd>
    <p>FASTA Delta</p>
  </dd>
  <dt>
    di
  </dt>
  <dd>
    <p>FASTA Delta with Implicit Gaps</p>
  </dd>
  <dt>
    l
  </dt>
  <dd>
    <p>FASTA+Gap Alignment (<strong>l</strong> Batch, <strong>l1</strong> Pop, <strong>l2</strong> Phy, <strong>l3</strong> Mut, <strong>l4</strong> Eco, <strong>l9</strong> Small-genome)</p>
  </dd>
  <dt>
    z
  </dt>
  <dd>
    <p>FASTA with Gap Lines</p>
  </dd>
  <dt>
    e
  </dt>
  <dd>
    <p>PHRAP/ACE</p>
  </dd>
  <dt>
    b
  </dt>
  <dd>
    <p>ASN.1 (in conjunction with \bF-M)</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-b</strong></p>
  </dt>
  <dd>
    <p>Generate GenBank file (deprecated in favor of <strong>-V b</strong>)</p>
  </dd>
  <dt>
    <p><strong>-c</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Cleanup (combine any of the following letters)</p>
<dl class='dl-vertical'>
  <dt>
    d
  </dt>
  <dd>
    <p>Correct Collection Dates (assume month first)</p>
  </dd>
  <dt>
    D
  </dt>
  <dd>
    <p>Correct Collection Dates (assume day first)</p>
  </dd>
  <dt>
    b
  </dt>
  <dd>
    <p>Append note to coding regions that overlap other coding regions with similar product names and do not contain 'ABC'</p>
  </dd>
  <dt>
    x
  </dt>
  <dd>
    <p>Extend partial ends of features by one or two nucleotides to abut gaps or sequence ends</p>
  </dd>
  <dt>
    s
  </dt>
  <dd>
    <p>Add exception to short introns</p>
  </dd>
  <dt>
    f
  </dt>
  <dd>
    <p>Fix product names</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-f</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Single table file</p>
  </dd>
  <dt>
    <p><strong>-g</strong></p>
  </dt>
  <dd>
    <p>Input is a genomic product set</p>
  </dd>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Convert general ID to note</p>
  </dd>
  <dt>
    <p><strong>-i</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Single input file</p>
  </dd>
  <dt>
    <p><strong>-j</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Source qualifiers</p>
  </dd>
  <dt>
    <p><strong>-k</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>CDS flags (combine any of the following letters)</p>
<dl class='dl-vertical'>
  <dt>
    c
  </dt>
  <dd>
    <p>Annotate Longest ORF</p>
  </dd>
  <dt>
    r
  </dt>
  <dd>
    <p>Allow Runon ORFs</p>
  </dd>
  <dt>
    m
  </dt>
  <dd>
    <p>Allow Alternative Starts</p>
  </dd>
  <dt>
    k
  </dt>
  <dd>
    <p>Set Conflict on Mismatch</p>
  </dd>

</dl>

  </dd>
  <dt>
    <p><strong>-n</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Organism name</p>
  </dd>
  <dt>
    <p><strong>-o</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Single output file</p>
  </dd>
  <dt>
    <p><strong>-p</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Path to files</p>
  </dd>
  <dt>
    <p><strong>-q</strong></p>
  </dt>
  <dd>
    <p>Set sequence ID from input file name</p>
  </dd>
  <dt>
    <p><strong>-r</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Path for results</p>
  </dd>
  <dt>
    <p><strong>-s</strong></p>
  </dt>
  <dd>
    <p>Read FASTAs as Set</p>
  </dd>
  <dt>
    <p><strong>-t</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Read template from <em>filename</em></p>
  </dd>
  <dt>
    <p><strong>-u</strong></p>
  </dt>
  <dd>
    <p>Convert GenProdSet to NucProtSet</p>
  </dd>
  <dt>
    <p><strong>-v</strong></p>
  </dt>
  <dd>
    <p>Validate (deprecated in favor of <strong>-V v</strong>)</p>
  </dd>
  <dt>
    <p><strong>-w</strong>&nbsp;<em>filename</em></p>
  </dt>
  <dd>
    <p>Single structured comment file</p>
  </dd>
  <dt>
    <p><strong>-x</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    <p>Suffix (default = <strong>.fsa</strong>)</p>
  </dd>
  <dt>
    <p><strong>-y</strong>&nbsp;<em>str</em></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>-z</strong></p>
  </dt>
  <dd>
    <p>Clean up log file Comment</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>The National Center for Biotechnology Information.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO tbl2asn&hellip;</h2>
        <div class="sectioncontent">
<p><a href="../man1/Psequin.1.html"><strong>Psequin</strong>(1)</a>, <a href="../man1/sbtedit.1.html"><strong>sbtedit</strong>(1)</a>, /usr/share/doc/ncbi-tools-bin/tbl2asn.txt.gz, &lt;http://www.ncbi.nlm.nih.gov/Sequin/table.html&gt;.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="tbl-dctrl.1.html"><span aria-hidden="true">&larr;</span> tbl-dctrl.1: Generate tabular representations of data in dctrl format</a></li>
   <li class="next"><a href="tblgen-3.4.1.html">tblgen-3.4.1: Target description to c++ code generator <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
