<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>theseus: Maximum likelihood, multiple simultaneous superpositions with statistical analysis</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Maximum likelihood, multiple simultaneous superpositions with statistical analysis">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="theseus (1) manual">
  <meta name="twitter:description" content="Maximum likelihood, multiple simultaneous superpositions with statistical analysis">
  <meta name="twitter:image" content="https://www.carta.tech/images/theseus-theseus-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/theseus.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="theseus (1) manual" />
  <meta property="og:description" content="Maximum likelihood, multiple simultaneous superpositions with statistical analysis" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/theseus-theseus-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">theseus<small> (1)</small></h1>
        <p class="lead">Maximum likelihood, multiple simultaneous superpositions with statistical analysis</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/theseus.1.html">
      <span itemprop="name">theseus: Maximum likelihood, multiple simultaneous superpositions with statistical analysis</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/theseus/">
      <span itemprop="name">theseus</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/theseus.1.html">
      <span itemprop="name">theseus: Maximum likelihood, multiple simultaneous superpositions with statistical analysis</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>theseus</strong> [-aAbBcCdDeEfFgGhHiIjklLmMnNoOpPqQrRsStTuvVwWxXyYZ] <em>pdbfile1 [pdbfile2 ...]</em></p><p>and</p><p><strong>theseus_align</strong> [-aAbBcCdDeEfFgGhHiIjklLmMnNoOpPqQrRsStTuvVwWxXyYZ] -f <em>pdbfile1 [pdbfile2 ...]</em></p><p>Default usage is equivalent to:</p><p><strong>theseus</strong> -a0 -e2 -g1 -i200 -k-1 -p1e-7 -r theseus -v -P0 <em>your.pdb</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>Theseus</strong> superposes a set of macromolecular structures simultaneously using the method of maximum likelihood (ML), rather than the conventional least-squares criterion. <strong>Theseus</strong> assumes that the structures are distributed according to a matrix Gaussian distribution and that the eigenvalues of the atomic covariance matrix are hierarchically distributed according to an inverse gamma distribution. This ML superpositioning model produces much more accurate results by essentially downweighting variable regions of the structures and by correcting for correlations among atoms.</p><p><strong>Theseus</strong> operates in two main modes, a mode for superimposing structures with identical sequences and a mode for structures with different sequences but similar structures:</p><ul>
<li><p>(1) A mode for superpositioning macromolecules with identical sequences and numbers of residues, for instance, multiple models in an NMR family or multiple structures from different crystal forms of the same protein. In this mode, <strong>Theseus</strong> will read every model in every file on the command line and superposition them.</p></li><li><p>Example:</p></li><li><p><strong>theseus</strong> <em>1s40.pdb</em></p></li><li><p>In the above example, <em>1s40.pdb</em> is a pdb file of 10 NMR models.</p></li><li><p>(2) An "alignment" mode for superpositioning structures with different sequences, for example, multiple structures of the cytochrome c protein from different species or multiple mutated structures of hen egg white lysozyme. This mode requires the user to supply a sequence alignment file of the structures being superpositioned (see option <strong>-A</strong> and "FILE FORMATS" below). Additionally, it may be necessary to supply a mapfile that tells <strong>theseus</strong> which PDB structure files correspond to which sequences in the alignment (see option <strong>-M</strong> and "FILE FORMATS" below). When superpositioning based on a seqeunce alignment, <strong>theseus</strong> uses a novel maximum likelihood algorithm for superpositioning multiple structures that include arbitrary gaps and insertions relative to each other. Unlike other algorithms for simultaneous superpositioning of multiple structures, our Expectation-Maximization algorithm uses all available data by including all residues aligned with gaps in the calculations. In this mode, if there are multiple structural models in a PDB file, <strong>theseus</strong> only reads the first model in each file on the command line. In other words, <strong>theseus</strong> treats the files on the command line as if there were only one structure per file.</p></li><li><p>Example 1:</p></li><li><p><strong>theseus</strong> -A cytc.aln -M cytc.filemap d1cih__.pdb d1csu__.pdb d1kyow_.pdb</p></li><li><p>In the above example, d1cih__.pdb, d1csu__.pdb, and d1kyow_.pdb are pdb files of cytochrome c domains from the SCOP database.</p></li><li><p>Example 2:</p></li><li><p><strong>theseus_align</strong> -f d1cih__.pdb d1csu__.pdb d1kyow_.pdb</p></li><li><p>In this example, the <strong>theseus_align</strong> script is called to do the hard work for you. It will calculate a sequence alignment and then superimpose based on that alignment. The script <strong>theseus_align</strong> takes the same options as the <strong>theseus</strong> program. Note, the first few lines of this script must be modified for your system, since it calls an external multiple sequence alignment program to do the alignment. See the <strong>examples/</strong> directory for more details, including example files.</p></li>
</ul>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">
<h3>Algorithmic options, defaults in {brackets}:</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--amber</strong></p>
  </dt>
  <dd>
    <p>Do special processing for AMBER8 formatted PDB files</p><ul>
<li><p>Most people will never need to use this long option, unless you are processing MD traces from AMBER. AMBER puts the atom names in the wrong column in the PDB file.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-a [</strong><em>selection</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Atoms to include in the superposition. This option takes two types of arguments, either (1) a number specifying a preselected set of atom types, or (2) an explict PDB-style, colon-delimited list of the atoms to include.</p><ul>
<li><p>For the preselected atom type subsets, the following integer options are available:</p></li><li><ul>
<li><p>0, alpha carbons for proteins, C1\' atoms for nucleic acids</p></li><li><p>1, backbone</p></li><li><p>2, all</p></li><li><p>3, alpha and beta carbons</p></li><li><p>4, all heavy atoms (no hydrogens)</p></li>
</ul></li><li><p>Note, only the <strong>-a0</strong> option is available when superpositioning structures with different sequences.</p></li><li><p>To custom select an explicit set of atom types, the atom types must be specified exactly as given in the PDB file field, including spaces, and the atom-types must encapsulated in quotation marks. Multiple atom types must be delimited by a colon. For example,</p></li><li><p><strong>-a' N  : CA : C  : O  '</strong></p></li><li><p>would specify the atom types in the peptide backbone.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-c</strong></p>
  </dt>
  <dd>
    <p>Use ML atomic covariance weighting (fit correlations, much slower)</p><ul>
<li><p>Unless you have many different structures with few residues, fitting the correlation matrix is likely unwarranted statistically due to a plethora of parameters and a paucity of data.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-e [</strong><em>n</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Embedding algorithm for initializing the average structure</p><ul>
<li><p>0 = none; use randomly chosen model</p></li><li><p>{2} = {ML embedded structure}</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-f</strong></p>
  </dt>
  <dd>
    <p>Only read the first model of a multi-model PDB file</p>
  </dd>
  <dt>
    <p><strong>-g [</strong><em>n</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Hierarchical model for variances</p><ul>
<li><p>0 = none (may not converge)</p></li><li><p>{1} = inverse gamma distribution</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help/usage</p>
  </dd>
  <dt>
    <p><strong>-i [</strong><em>nnn</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Maximum iterations, {200}</p>
  </dd>
  <dt>
    <p><strong>-k [</strong><em>n</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>constant minimum variance {-1} {if set to negative value, the minimum variance is determined empirically}</p>
  </dd>
  <dt>
    <p><strong>-p [</strong><em>precision</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Requested relative precision for convergence, {1e-7}</p>
  </dd>
  <dt>
    <p><strong>-r [</strong><em>root name</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Root name to be used in naming the output files, {theseus}</p>
  </dd>
  <dt>
    <p><strong>-s [</strong><em>n-n:...</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Residue selection (e.g. -s15-45:50-55), {all}</p>
  </dd>
  <dt>
    <p><strong>-S [</strong><em>n-n:...</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Residues to exclude (e.g. -S15-\45:50-55) {none}</p><ul>
<li><p>The previous two options have the same format. Residue (or alignment column) ranges are indicated by beginning and end separated by a dash. Multiple ranges, in any arbitrary order, are separated by a colon. Chains may also be selected by giving the chain ID immediately preceding the residue range. For example, <strong>-sA1-20:A40-71</strong> will only include residues 1 through 20 and 40 through 70 in chain A. Chains cannot be specified when superpositioning structures with different sequences.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-v</strong></p>
  </dt>
  <dd>
    <p>use ML variance weighting (no correlations) {default}</p>
  </dd>

</dl>

<h3>Input/output options:</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>-A [</strong><em>sequence alignment file</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Sequence alignment file to use as a guide (CLUSTAL or A2M format)</p><ul>
<li><p>For use when superpositioning structures with different sequences. See "FILE FORMATS" below.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-E</strong></p>
  </dt>
  <dd>
    <p>Print expert options</p>
  </dd>
  <dt>
    <p><strong>-F</strong></p>
  </dt>
  <dd>
    <p>Print FASTA files of the sequences in PDB files and quit</p><ul>
<li><p>A useful option when superpositioning structures with different sequences. The files output with this option can be aligned with a multiple sequence alignment program such as CLUSTAL or MUSCLE, and the resulting output alignment file used as <strong>theseus</strong> input with the <strong>-A</strong> option.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-h</strong></p>
  </dt>
  <dd>
    <p>Help/usage</p>
  </dd>
  <dt>
    <p><strong>-I</strong></p>
  </dt>
  <dd>
    <p>Just calculate statistics for input file; don't superposition</p>
  </dd>
  <dt>
    <p><strong>-M [</strong><em>mapfile</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>File that maps PDB files to sequences in the alignment.</p><ul>
<li><p>A simple two-column formatted file; see "FILE FORMATS" below. Used with mode 2.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-n</strong></p>
  </dt>
  <dd>
    <p>Don't write transformed pdb file</p>
  </dd>
  <dt>
    <p><strong>-o [</strong><em>reference structure</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Reference file to superposition on, all rotations are relative to the first model in this file</p><ul>
<li><p>For example, 'theseus -o cytc1.pdb cytc1.pdb cytc2.pdb cytc3.pdb' will superposition the structures and rotate the entire final superposition so that the structure from cytc1.pdb is in the same orientation as the structure in the original cytc1.pdb PDB file.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-O</strong></p>
  </dt>
  <dd>
    <p>Olve's segID file</p><ul>
<li><p>Useful output when superpositioning structures with different sequences (mode 2). In 'theseus_sup.pdb', the main output superposition PDB file, the segID field now holds the number of the sequence alignment column that it belongs to. This number, divided by 100, is also echoed in the B-factor field. When using <strong>O</strong> (or any other capable molecular visualization program), one can then color by B-factor ranges and immediately see in the superposition which regions of the structure are aligned in the sequence alignment file. An additional file is also output, called 'theseus_olve.pdb' which only contains the very atoms that were included in the ML superposition calculation. That is, it will only contain alpha carbons or phosphorous atoms, and it will only contain atoms from the columns selected with the <strong>-s</strong> or <strong></strong> "-S" options. Requested by Olve Peersen of Colorado State University.</p></li>
</ul>
  </dd>
  <dt>
    <p><strong>-V</strong></p>
  </dt>
  <dd>
    <p>Version</p>
  </dd>

</dl>

<h3>Principal components analysis:</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>-C</strong></p>
  </dt>
  <dd>
    <p>Use covariance matrix for PCA (correlation matrix is default)</p>
  </dd>
  <dt>
    <p><strong>-P [</strong><em>nnn</em><strong>]</strong></p>
  </dt>
  <dd>
    <p>Number of principal components to calculate {0}</p><ul>
<li><p>In both of the above, the corresponding principal component is written in the B-factor field of the output PDB file. Usually only the first few PCs are of any interest (maybe up to six).</p><p> EXAMPLES <strong>theseus</strong> <em>2sdf.pdb</em></p></li>
</ul>
  </dd>

</dl>
<p><strong>theseus</strong> -l -r new2sdf <em>2sdf.pdb</em></p><p><strong>theseus</strong> -s15-45 -P3 <em>2sdf.pdb</em></p><p><strong>theseus</strong> -A <em>cytc.aln</em> -M <em>cytc.mapfile</em> -o <em>cytc1.pdb</em> -s1-40 <em>cytc1.pdb cytc2.pdb cytc3.pdb cytc4.pdb</em></p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">ENVIRONMENT</h2>
        <div class="sectioncontent">
<p>You can set the environment variable 'PDBDIR' to your PDB file directory and <strong>theseus</strong> will look there after the present working directory. For example, in the C shell (tcsh or csh), you can put something akin to this in your .cshrc file:</p><p>setenv PDBDIR '/usr/share/pdbs/'</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">FILE FORMATS</h2>
        <div class="sectioncontent">
<p><strong>Theseus</strong> will read standard PDB formatted files (see &lt;http://www.rcsb.org/pdb/&gt;). Every effort has been made for the program to accept nonstandard CNS and X-PLOR file formats also.</p><p>Two other files deserve mention, a sequence alignment file and a mapfile.</p><h3>Sequence alignment file</h3>
<p>When superpositioning structures with different residue identities (where the lengths of each the macromolecules in terms of residues are not necessarily equal), a sequence alignment file must be included for <strong>theseus</strong> to use as a guide (specified by the <strong>-A</strong> option). <strong>Theseus</strong> accepts both CLUSTAL and A2M (FASTA) formatted multiple sequence alignment files.</p><p>NOTE 1: The residue sequence in the alignment must match exactly the residue sequence given in the coordinates of the PDB file. That is, there can be no missing or extra residues that do not correspond to the sequence in the PDB file. An easy way to ensure that your sequences exactly match the PDB files is to generate the sequences using <strong>theseus'</strong> <strong>-F</strong> option, which writes out a FASTA formatted sequence file of the chain(s) in the PDB files. The files output with this option can then be aligned with a multiple sequence alignment program such as CLUSTAL or MUSCLE, and the resulting output alignment file used as <strong>theseus</strong> input with the <strong>-A</strong> option.</p><p>NOTE 2: Every PDB file must have a corresponding sequence in the alignment. However, not every sequence in the alignment needs to have a corresponding PDB file. That is, there can be extra sequences in the alignment that are not used for guiding the superposition.</p>
<h3>PDB -&gt; Sequence mapfile</h3>
<p>If the names of the PDB files and the names of the corresponding sequences in the alignemnt are identical, the mapfile may be omitted.  Otherwise, <strong>Theseus</strong> needs to know which sequences in the alignment file correspond to which PDB structure files. This information is included in a mapfile with a very simple format (specified with the <strong>-M</strong> option). There are only two columns separated by whitespace: the first column lists the names of the PDB structure files, while the second column lists the corresponding sequence names exactly as given in the multiple sequence alignment file.</p><p>An example of the mapfile:</p><p>cytc1.pdb    seq1</p><p>cytc2.pdb    seq2</p><p>cytc3.pdb    seq3</p>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">SCREEN OUTPUT</h2>
        <div class="sectioncontent">
<p>Theseus provides output describing both the progress of the superpositioning and several statistics for the final result:</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>Least-squares &lt;sigma&gt;:</strong></p>
  </dt>
  <dd>
    <p>The standard deviation for the superposition, based on the conventional assumption of no correlation and equal variances. Basically equal to the RMSD from the average structure.</p>
  </dd>
  <dt>
    <p><strong>Classical LS pairwise &lt;RMSD&gt;:</strong></p>
  </dt>
  <dd>
    <p>The conventional RMSD for the superposition, the average RMSD for all pairwise combinations of structures in the ensemble.</p>
  </dd>
  <dt>
    <p><strong>Maximum Likelihood &lt;sigma&gt;:</strong></p>
  </dt>
  <dd>
    <p>The ML analog of the standard deviation for the superposition. When assuming that the correlations are zero (a diagonal covariance matrix), this is equal to the square root of the harmonic average of the variances for each atom. In contrast, the 'Least-squares &lt;sigma&gt;' given above reports the square root of the arithmetic average of the variances.  The harmonic average is always less than the arithmetic average, and the harmonic average downweights large values proportional to their magnitude. This makes sense statistically, because when combining values one should weight them by the reciprocal of their variance (which is in fact what the ML superpositioning method does).</p>
  </dd>
  <dt>
    <p><strong>Log Likelihood:</strong></p>
  </dt>
  <dd>
    <p>The final log likelihood of the superposition, assuming the matrix Gaussian distribution of the structures and the hierarchical inverse gamma distribution of the eigenvalues of the covariance matrix.</p>
  </dd>
  <dt>
    <p><strong>AIC:</strong></p>
  </dt>
  <dd>
    <p>The Akaike Information Criterion for the final superposition. This is an important statistic in likelihood analysis and model selection theory. It allows an objective comparison of multiple theoretical models with different numbers of parameters. In this case, the higher the number the better. There is a tradeoff between fit to the data and the number of parameters being fit. Increasing the number of parameters in a model will always give a better fit to the data, but it also increases the uncertainty of the estimated values. The AIC criterion finds the best combination by (1) maximizing the fit to the data while (2) minimizing the uncertainty due to the number of parameters. In the superposition case, one can compare the least squares superposition to the maximum likelihood superposition. The method (or model) with the higher AIC is preferred. A difference in the AIC of 2 or more is considered strong statistical evidence for the better model.</p>
  </dd>
  <dt>
    <p><strong>BIC:</strong></p>
  </dt>
  <dd>
    <p>The Bayesian Information Criterion. Similar to the AIC, but with a Bayesian emphasis.</p>
  </dd>
  <dt>
    <p><strong>Rotational, translational, covar chi^2:</strong></p>
  </dt>
  <dd>
    <p>The reduced chi-squared statistic for the fit of the structures to the model. With a good fit it should be close to 1.0, which indicates a perfect fit of the data to the statistical model.  In the case of least-squares, the assumed model is a matrix Gaussian distribution of the structures with equal variances and no correlations.  For the ML fits, the assumed models can either be (1) unequal variances and no correlations, as calculated with the <strong>-v</strong> option [default] or (2) unequal variances and correlations, as calculated with the <strong>-c</strong> option.  This statistic is for the superposition only, and does not include the fit of the covariance matrix eigenvalues to an inverse gamma distribution.  See 'Omnibus chi^2' below.</p>
  </dd>
  <dt>
    <p><strong>Hierarchical minimum var:</strong></p>
  </dt>
  <dd>
    <p>The hierarchical fit of the inverse gamma distribution constrains the variances of the atoms by making large ones smaller and small ones larger. This statistic reports the minimum possible variance given the inferred inverse gamma parameters.</p>
  </dd>
  <dt>
    <p><strong>Hierarchical var (alpha, gamma) chi^2:</strong></p>
  </dt>
  <dd>
    <p>The reduced chi-squared for the inverse gamma fit of the covariance matrix eigenvalues. As before, it should ideally be close to 1.0.  The two values in the parentheses are the ML estimates of the scale and shape parameters, respectively, for the inverse gamma distribtuion.</p>
  </dd>
  <dt>
    <p><strong>Omnibus chi^2:</strong></p>
  </dt>
  <dd>
    <p>The overall reduced chi-squared statistic for the entire fit, including the rotations, translations, covariances, and the inverse gamma parameters. This is probably the most important statistic for the superposition. In some cases, the inverse gamma fit may be poor, yet the overall fit is still very good. Again, it should ideally be close to 1.0, which would indicate a perfect fit. However, if you think it is too large, make sure to compare it to the chi^2 for the least-squares fit; it's probably not that bad after all. A large chi^2 often indicates a violation of the assumptions of the model. The most common violation is when superpositioning two or more independent domains that can rotate relative to each other. If this is the case, then there will likely be not just one Gaussian distribution, but several mixed Gaussians, one for each domain.  Then, it would be better to superposition each domain independently.</p>
  </dd>
  <dt>
    <p><strong>skewness, skewness Z-value, kurtosis & kurtosis Z-value:</strong></p>
  </dt>
  <dd>
    <p>The skewness and kurtosis of the residuals. Both should be 0.0 if the residuals fit a Gaussian distribution perfectly.  They are followed by the P-value for the statistics. This is a very stringent test; residuals can be very non-Gaussian and yet the estimated rotations, translations, and covariance matrix may still be rather accurate.</p>
  </dd>
  <dt>
    <p><strong>FP error in transformed coordinates:</strong></p>
  </dt>
  <dd>
    <p>The empirically determined floating point error in the coordinates after rotation and translation.</p>
  </dd>
  <dt>
    <p><strong>Minimum RMSD error per atom:</strong></p>
  </dt>
  <dd>
    <p>The empirically determined minimum RMSD error per atom, based on the floating point error of the computer.</p>
  </dd>
  <dt>
    <p><strong>Data pts, Free params, D/P:</strong></p>
  </dt>
  <dd>
    <p>The total number of data points given all observed structures, the number of parameters being fit in the model, and the data-to-parameter ratio.</p>
  </dd>
  <dt>
    <p><strong>Median structure:</strong></p>
  </dt>
  <dd>
    <p>The structure that is overall most similar to the average structure. This can be considered to be the most "typical" structure in the ensemble.</p>
  </dd>
  <dt>
    <p><strong>Total rounds:</strong></p>
  </dt>
  <dd>
    <p>The number of iterations that the algorithm took to converge.</p>
  </dd>
  <dt>
    <p><strong>Fractional precision:</strong></p>
  </dt>
  <dd>
    <p>The actual precision that the algorithm converged to.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OUTPUT FILES</h2>
        <div class="sectioncontent">
<p>Theseus writes out the following files:</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>theseus_sup.pdb</strong></p>
  </dt>
  <dd>
    <p>The final superposition, rotated to the principle axes of the mean structure.</p>
  </dd>
  <dt>
    <p><strong>theseus_ave.pdb</strong></p>
  </dt>
  <dd>
    <p>The estimate of the mean structure.</p>
  </dd>
  <dt>
    <p><strong>theseus_cor.mat, theseus_cov.mat</strong></p>
  </dt>
  <dd>
    <p>The atomic correlation matrix and covariance matrices, based on the final superposition. The format is suitable for input to GNU's <strong>octave.</strong> These are the matrices used in the Principal Components Analysis.</p>
  </dd>
  <dt>
    <p><strong>theseus_embed_ave.pdb</strong></p>
  </dt>
  <dd>
    <p>The average structure as calculated by S. Lele's EDMA embedding algorithm, used as the starting point for the maximum likelihood iterations.</p>
  </dd>
  <dt>
    <p><strong>theseus_residuals.txt</strong></p>
  </dt>
  <dd>
    <p>The normalized residuals of the superposition. These can be analyzed for deviations from normality (whether they fit a standard Gaussian distribution). E.g., the chi^2, skewness, and kurtosis statistics are based on these values.</p>
  </dd>
  <dt>
    <p><strong>theseus_transf.txt</strong></p>
  </dt>
  <dd>
    <p>The final transformation rotation matrices and translation vectors.</p>
  </dd>
  <dt>
    <p><strong>theseus_variances.txt</strong></p>
  </dt>
  <dd>
    <p>The vector of estimated variances for each atom.</p>
  </dd>

</dl>
<p>When Principal Components are calculated (with the <strong>-P</strong> option), the following files are also produced:</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>theseus_pcvecs.txt</strong></p>
  </dt>
  <dd>
    <p>The principal component vectors.</p>
  </dd>
  <dt>
    <p><strong>theseus_pcstats.txt</strong></p>
  </dt>
  <dd>
    <p>Simple statistics for each principle component (loadings, variance explained, etc.).</p>
  </dd>
  <dt>
    <p><strong>theseus_pcN_ave.pdb</strong></p>
  </dt>
  <dd>
    <p>The average structure with the Nth principal component written in the temperature factor field.</p>
  </dd>
  <dt>
    <p><strong>theseus_pcN.pdb</strong></p>
  </dt>
  <dd>
    <p>The final superposition with the Nth principal component written in the temperature factor field. This file is omitted when superpositioning molecules with different residue sequences (mode 2).</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">BUGS</h2>
        <div class="sectioncontent">
<p>Please send me (DLT) reports of all problems.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RESTRICTIONS</h2>
        <div class="sectioncontent">
<p><strong>Theseus</strong> is <em>not</em> a structural alignment program. The structure-based alignment problem is completely different from the structural superposition problem. In order to do a structural superposition, there must be a 1-to-1 mapping that associates the atoms in one structure with the atoms in the other structures. In the simplest case, this means that structures must have equivalent numbers of atoms, such as the models in an NMR PDB file. For structures with different numbers of residues/atoms, superpositioning is only possible when the sequences have been aligned previously. Finding the best sequence alignment based on only structural information is a difficult problem, and one for which there is currently no maximum likelihood approach. Extending <strong>theseus</strong> to address the structural alignment problem is an ongoing research project.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>Douglas L. Theobald</p><p>dtheobald@brandeis.edu</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">CITATION</h2>
        <div class="sectioncontent">
<p>When using <strong>theseus</strong> in publications please cite:</p><p>Douglas L. Theobaldand Phillip A. Steindel (2012)</p><p>"Optimal simultaneous superpositioning of multiple structures with missing data."</p><p>Bioinformatics 28(15):1972-1979</p><p>The following papers also report <strong>theseus</strong> developments:</p><p>Douglas L. Theobald and Deborah S. Wuttke (2006)</p><p>"Empirical Bayes models for regularizing maximum likelihood estimation in the matrix Gaussian Procrustes problem."</p><p>PNAS 103(49):18521-18527</p><p>Douglas L. Theobald and Deborah S. Wuttke (2006)</p><p>"THESEUS: Maximum likelihood superpositioning and analysis of macromolecular structures."</p><p>Bioinformatics 22(17):2171-2172</p><p>Douglas L. Theobald and Deborah S. Wuttke (2008)</p><p>"Accurate structural correlations from maximum likelihood superpositions."</p><p>PLoS Computational Biology 4(2):e43</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">HISTORY</h2>
        <div class="sectioncontent">
<p>Long, tedious, and sordid.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="therion-viewer.1.html"><span aria-hidden="true">&larr;</span> therion-viewer.1: Loch  therion model viewer</a></li>
   <li class="next"><a href="theseus_align.1.html">theseus_align.1: Quick-and-dirty way to superimpose proteins <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
