<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>tigr-glimmer3: Tigr-glimmer  find/score potential genes in genome-file using the probability model in icm-file</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Tigr-glimmer  find/score potential genes in genome-file using the probability model in icm-file">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="tigr-glimmer3 (1) manual">
  <meta name="twitter:description" content="Tigr-glimmer  find/score potential genes in genome-file using the probability model in icm-file">
  <meta name="twitter:image" content="https://www.carta.tech/images/tigr-glimmer-tigr-glimmer3-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/tigr-glimmer3.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="tigr-glimmer3 (1) manual" />
  <meta property="og:description" content="Tigr-glimmer  find/score potential genes in genome-file using the probability model in icm-file" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/tigr-glimmer-tigr-glimmer3-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">tigr-glimmer3<small> (1)</small></h1>
        <p class="lead">Tigr-glimmer  find/score potential genes in genome-file using the probability model in icm-file</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/tigr-glimmer3.1.html">
      <span itemprop="name">tigr-glimmer3: Tigr-glimmer  find/score potential genes in genome-file using the probability model in icm-file</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/tigr-glimmer/">
      <span itemprop="name">tigr-glimmer</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/tigr-glimmer3.1.html">
      <span itemprop="name">tigr-glimmer3: Tigr-glimmer  find/score potential genes in genome-file using the probability model in icm-file</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>tigr-glimmer3</strong> [<strong></strong><em>genome-file</em><em>]  [</em><strong></strong><em>icm-file</em><em>]  [</em><strong></strong><em>[options]</em><em>]</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p><strong>tigr-glimmer</strong> is a system for finding genes in microbial DNA, especially the genomes of bacteria and archaea. <strong>tigr-glimmer</strong> (Gene Locator and Interpolated Markov Modeler) uses interpolated Markov models (IMMs) to identify the coding regions and distinguish them from noncoding DNA. The IMM approach, described in our Nucleic Acids Research paper on <strong>tigr-glimmer</strong> 1.0 and in our subsequent paper on <strong>tigr-glimmer</strong> 2.0, uses a combination of Markov models from 1st through 8th-order, weighting each model according to its predictive power. <strong>tigr-glimmer</strong> 1.0 and 2.0 use 3-periodic nonhomogenous Markov models in their IMMs.</p><p><strong>tigr-glimmer</strong> is the primary microbial gene finder at TIGR, and has been used to annotate the complete genomes of B. burgdorferi (Fraser et al., Nature, Dec. 1997), T. pallidum (Fraser et al., Science, July 1998), T. maritima, D. radiodurans, M. tuberculosis, and non-TIGR projects including C. trachomatis, C. pneumoniae, and others. Its analyses of some of these genomes and others is available at the TIGR microbial database site.</p><p>A special version of <strong>tigr-glimmer</strong> designed for small eukaryotes, GlimmerM, was used to find the genes in chromosome 2 of the malaria parasite, P. falciparum.. GlimmerM is described in S.L. Salzberg, M. Pertea, A.L. Delcher, M.J. Gardner, and H. Tettelin, "Interpolated Markov models for eukaryotic gene finding," Genomics 59 (1999), 24-31.  Click here (http://www.tigr.org/software/glimmerm/) to visit the GlimmerM site, which includes information on how to download the GlimmerM system.</p><p>The <strong>tigr-glimmer</strong> system consists of two main programs. The first of these is the training program, build-imm. This program takes an input set of sequences and builds and outputs the IMM for them. These sequences can be complete genes or just partial orfs. For a new genome, this training data can consist of those genes with strong database hits as well as very long open reading frames that are statistically almost certain to be genes. The second program is glimmer, which uses this IMM to identify putative genes in an entire genome. <strong>tigr-glimmer</strong> automatically resolves conflicts between most overlapping genes by choosing one of them. It also identifies genes that are suspected to truly overlap, and flags these for closer inspection by the user. These ``suspect'' gene candidates have been a very small percentage of the total for all the genomes analyzed thus far. <strong>tigr-glimmer</strong> is a program that...</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <strong>-C </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Use n as GC percentage of independent model</p>
  </dd>
  <dt>
    
  </dt>
  <dd>
    <p>Note:  n should be a percentage, e.g., -C 45.2</p>
  </dd>
  <dt>
    -f
  </dt>
  <dd>
    <p>Use ribosome-binding energy to choose start codon</p>
  </dd>
  <dt>
    <strong>+f</strong>
  </dt>
  <dd>
    <p>Use first codon in orf as start codon</p>
  </dd>
  <dt>
    <strong>-g </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Set minimum gene length to n</p>
  </dd>
  <dt>
    <strong>-i </strong><em>filename</em><em></em>
  </dt>
  <dd>
    <p>Use <strong></strong><em>filename</em><em> 	to select regions of bases that are off</em> limits, so that no bases within that area will be examined</p>
  </dd>
  <dt>
    <strong>-l</strong>
  </dt>
  <dd>
    <p>Assume linear rather than circular genome, i.e., no wraparound</p>
  </dd>
  <dt>
    <strong>-L </strong><em>filename</em><em></em>
  </dt>
  <dd>
    <p>Use filename to specify a list of orfs that should be scored separately, with no overlap rules</p>
  </dd>
  <dt>
    <strong>-M</strong>
  </dt>
  <dd>
    <p>Input is a multifasta file of separate genes to be scored separately, with no overlap rules</p>
  </dd>
  <dt>
    <strong>-o </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Set minimum overlap length to n.  Overlaps shorter than this are ignored.</p>
  </dd>
  <dt>
    <strong>-p </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Set minimum overlap percentage to n%.  Overlaps shorter than this percentage of *both* strings are ignored.</p>
  </dd>
  <dt>
    <strong>-q </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Set the maximum length orf that can be rejected because of the independent probability score column to (n - 1)</p>
  </dd>
  <dt>
    <strong>-r</strong>
  </dt>
  <dd>
    <p>Don't use independent probability score column</p>
  </dd>
  <dt>
    <strong>+r</strong>
  </dt>
  <dd>
    <p>Use independent probability score column</p>
  </dd>
  <dt>
    <strong>-r</strong>
  </dt>
  <dd>
    <p>Don't use independent probability score column</p>
  </dd>
  <dt>
    <strong>-s </strong><em>s</em><em></em>
  </dt>
  <dd>
    <p>Use string s as the ribosome binding pattern to find start codons.</p>
  </dd>
  <dt>
    <strong>+S</strong>
  </dt>
  <dd>
    <p>Do use stricter independent intergenic model that doesn't give probabilities to in-frame stop codons.  (Option is obsolete since this is now the only behaviour</p>
  </dd>
  <dt>
    <strong>-t </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Set threshold score for calling as gene to n.  If the in-frame score &gt;= n, then the region is given a number and considered a potential gene.</p>
  </dd>
  <dt>
    <strong>-w </strong><em>n</em> <em></em>
  </dt>
  <dd>
    <p>Use "weak" scores on tentative genes n or longer.  Weak scores ignore the independent probability score.</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO tigr-glimmer3&hellip;</h2>
        <div class="sectioncontent">
<p>tigr-adjust (1), tigr-anomaly	(1), tigr-build-icm (1), tigr-check (1), tigr-codon-usage (1), tigr-compare-lists (1), tigr-extract (1), tigr-generate (1), tigr-get-len (1), tigr-get-putative (1), tigr-glimmer3 (1), tigr-long-orfs (1)</p><p>http://www.tigr.org/software/glimmer/</p><p>Please see the readme in /usr/share/doc/glimmer for a description on how to use Glimmer.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>This manual page was quickly copied from the glimmer web site by Steffen Moeller moeller@debian.org for the <strong>Debian</strong> system.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="tigr-glimmer.1.html"><span aria-hidden="true">&larr;</span> tigr-glimmer.1: Runs various programs of the tigr glimmer suite</a></li>
   <li class="next"><a href="tigr-long-orfs.1.html">tigr-long-orfs.1: Long-orfs  find/score potential genes in genome-file using the probability model in icm-file <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
