<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>tigr-long-orfs: Long-orfs  find/score potential genes in genome-file using the probability model in icm-file</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Long-orfs  find/score potential genes in genome-file using the probability model in icm-file">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="tigr-long-orfs (1) manual">
  <meta name="twitter:description" content="Long-orfs  find/score potential genes in genome-file using the probability model in icm-file">
  <meta name="twitter:image" content="https://www.carta.tech/images/tigr-glimmer-tigr-long-orfs-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/tigr-long-orfs.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="tigr-long-orfs (1) manual" />
  <meta property="og:description" content="Long-orfs  find/score potential genes in genome-file using the probability model in icm-file" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/tigr-glimmer-tigr-long-orfs-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">tigr-long-orfs<small> (1)</small></h1>
        <p class="lead">Long-orfs  find/score potential genes in genome-file using the probability model in icm-file</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/tigr-long-orfs.1.html">
      <span itemprop="name">tigr-long-orfs: Long-orfs  find/score potential genes in genome-file using the probability model in icm-file</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/tigr-glimmer/">
      <span itemprop="name">tigr-glimmer</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/tigr-long-orfs.1.html">
      <span itemprop="name">tigr-long-orfs: Long-orfs  find/score potential genes in genome-file using the probability model in icm-file</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>tigr-long-orgs</strong> [genome-file <strong></strong><em>options</em><em>]</em></p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>Program long-orfs takes a sequence file (in FASTA format) and outputs a list of all long "potential genes" in it that do not overlap by too much.  By "potential gene" I mean the portion of an orf from the first start codon to the stop codon at the end.</p><p>The first few lines of output specify the settings of various parameters in the program:</p><p>Minimum gene length is the length of the smallest fragment considered to be a gene.  The length is measured from the first base of the start codon to the last base *before* the stop codon. This value can be specified when running the program with the  -g  option. By default, the program now (April 2003) will compute an optimal length for this parameter, where "optimal" is the value that produces the greatest number of long ORFs, thereby increasing the amount of data used for training.</p><p>Minimum overlap length is a lower bound on the number of bases overlap between 2 genes that is considered a problem.  Overlaps shorter than this are ignored.</p><p>Minimum overlap percent is another lower bound on the number of bases overlap that is considered a problem.  Overlaps shorter than this percentage of *both* genes are ignored.</p><p>The next portion of the output is a list of potential genes:</p><p>Column 1 is an ID number for reference purposes.  It is assigned sequentially starting with  1  to all long potential genes.  If overlapping genes are eliminated, gaps in the numbers will occur. The ID prefix is specified in the constant  ID_PREFIX .</p><p>Column 2 is the position of the first base of the first start codon in the orf.  Currently I use atg, and gtg as start codons.  This is easily changed in the function  Is_Start () .</p><p>Column 3 is the position of the last base *before* the stop codon.  Stop codons are taa, tag, and tga.  Note that for orfs in the reverse reading frames have their start position higher than the end position. The order in which orfs are listed is in increasing order by Max {OrfStart, End}, i.e., the highest numbered position in the orf, except for orfs that "wrap around" the end of the sequence.</p><p>When two genes with ID numbers overlap by at least a sufficient amount (as determined by Min_Olap and Min_Olap_Percent ), they are eliminated and do not appear in the output.</p><p>The final output of the program (sent to the standard error file so it does not show up when output is redirected to a file) is the length of the longest orf found.</p><p>Specifying Different Start and Stop Codons:</p><p>To specify different sets of start and stop codons, modify the file gene.h .  Specifically, the functions:</p><p>Is_Forward_Start     Is_Reverse_Start     Is_Start Is_Forward_Stop      Is_Reverse_Stop      Is_Stop</p><p>are used to determine what is used for start and stop codons.</p><p>Is_Start  and  Is_Stop  do simple string comparisons to specify which patterns are used.  To add a new pattern, just add the comparison for it.  To remove a pattern, comment out or delete the comparison for it.</p><p>The other four functions use a bit comparison to determine start and stop patterns.  They represent a codon as a 12-bit pattern, with 4 bits for each base, one bit for each possible value of the bases, T, G, C or A.  Thus the bit pattern  0010 0101 1100  represents the base pattern  [C] [A or G] [G or T].  By doing bit operations (& | ~) and comparisons, more complicated patterns involving ambiguous reads can be tested efficiently.  Simple patterns can be tested as in the current code.</p><p>For example, to insert an additional start codon of CAT requires 3 changes: 1. The line || (Codon & 0x218) == Codon should be inserted into  Is_Forward_Start , since 0x218 = 0010 0001 1000 represents CAT. 2. The line || (Codon & 0x184) == Codon should be inserted into  Is_Reverse_Start , since 0x184 = 0001 1000 0100 represents ATG, which is the reverse-complement of CAT.  Alternately, the #define constant  ATG_MASK  could be used. 3. The line || strncmp (S, "cat", 3) == 0 should be inserted into  Is_Start .</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">OPTIONS</h2>
        <div class="sectioncontent">

<dl class='dl-vertical'>
  <dt>
    <strong>-g </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Set minimum gene length to n.  Default is to compute an optimal value automatically.  Don't change this unless you know what you're doing.</p>
  </dd>
  <dt>
    <strong>-l</strong>
  </dt>
  <dd>
    <p>Regard the genome as linear (not circular), i.e., do not allow genes to "wrap around" the end of the genome. This option works on both  glimmer and long-orfs . The default behavior is to regard the genome as circular.</p>
  </dd>
  <dt>
    <strong>-o </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Set maximum overlap length to n.  Overlaps shorter than this are permitted.  (Default is 0 bp.)</p>
  </dd>
  <dt>
    <strong>-p </strong><em>n</em><em></em>
  </dt>
  <dd>
    <p>Set maximum overlap percentage to n%.  Overlaps shorter than this percentage of *both* strings are ignored.  (Default is 10%.)</p>
  </dd>

</dl>

        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">RELATED TO tigr-long-orfs&hellip;</h2>
        <div class="sectioncontent">
<p>tigr-glimmer3 (1), tigr-adjust (1), tigr-anomaly	(1), tigr-build-icm (1), tigr-check (1), tigr-codon-usage (1), tigr-compare-lists (1), tigr-extract (1), tigr-generate (1), tigr-get-len (1), tigr-get-putative (1),</p><p>http://www.tigr.org/software/glimmer/</p><p>Please see the readme in /usr/share/doc/tigr-glimmer for a description on how to use Glimmer3.</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">AUTHOR</h2>
        <div class="sectioncontent">
<p>This manual page was quickly copied from the glimmer web site by Steffen Moeller moeller@debian.org for the <strong>Debian</strong> system.</p>
        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="tigr-glimmer3.1.html"><span aria-hidden="true">&larr;</span> tigr-glimmer3.1: Tigr-glimmer  find/score potential genes in genome-file using the probability model in icm-file</a></li>
   <li class="next"><a href="tigr-run-glimmer3.1.html">tigr-run-glimmer3.1: Tigr-glimmer  apply the suite of programs within glimmer3 to a a prokaryotic or archean genome. <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
