<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>vcftools: Analyse vcf files</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Analyse vcf files">
  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:site" content="@CartaTech">
  <meta name="twitter:creator" content="@CartaTech">
  <meta name="twitter:title" content="vcftools (1) manual">
  <meta name="twitter:description" content="Analyse vcf files">
  <meta name="twitter:image" content="https://www.carta.tech/images/vcftools-vcftools-1.png">
  <meta property="og:url" content="https://www.carta.tech/man-pages/man1/vcftools.1.html" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="vcftools (1) manual" />
  <meta property="og:description" content="Analyse vcf files" />
  <meta property="fb:app_id" content="1241677679199500" />
  <meta property="og:image" content="https://www.carta.tech/images/vcftools-vcftools-1.png" />
  <meta property="og:image:width" content="600" />
  <meta property="og:image:height" content="315" />
</head>
<body>
  <div class="container final">
          <div class="page-header">
        <h1 class="font-effect-destruction">vcftools<small> (1)</small></h1>
        <p class="lead">Analyse vcf files</p>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/">
      <span itemprop="name">Man Pages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/">
      <span itemprop="name">Executable programs or shell commands</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/vcftools.1.html">
      <span itemprop="name">vcftools: Analyse vcf files</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
<ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/vcftools/">
      <span itemprop="name">vcftools</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/man-pages/man1/vcftools.1.html">
      <span itemprop="name">vcftools: Analyse vcf files</span>
    </a>
    <meta itemprop="position" content="4" />
  </li>
</ol>
    
      <section>
        <h2 class="font-effect-shadow-multiple">SYNOPSIS</h2>
        <div class="sectioncontent">
<p><strong>vcftools </strong>[<em>OPTIONS</em>]</p>
        </div>
      </section>

      <section>
        <h2 class="font-effect-shadow-multiple">DESCRIPTION</h2>
        <div class="sectioncontent">
<p>The vcftools program is run from the command line. The interface is inspired by PLINK, and so should be largely familiar to users of that package. Commands take the following form:</p>
<pre>
  vcftools --vcf file1.vcf --chr 20 --freq
</pre>
<p>The above command tells vcftools to read in the file file1.vcf, extract sites on chromosome 20, and calculate the allele frequency at each site. The resulting allele frequency estimates are stored in the output file, out.freq. As in the above example, output from vcftools is mainly sent to output files, as opposed to being shown on the screen.</p><p>Note that some commands may only be available in the latest version of vcftools. To obtain the latest version, you should use SVN to checkout the latest code, as described on the home page.</p><p>Also note that polyploid genotypes are not currently supported.</p><h3>Basic Options</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--vcf</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>This option defines the VCF file to be processed. The files need to be decompressed prior to use with vcftools. vcftools expects files in VCF format v4.0, a specification of which can be found here.</p>
  </dd>
  <dt>
    <p><strong>--gzvcf</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>This option can be used in place of the --vcf option to read compressed (gzipped) VCF files directly. Note that this option can be quite slow when used with large files.</p>
  </dd>
  <dt>
    <p><strong>--out</strong> &lt;prefix&gt;</p>
  </dt>
  <dd>
    <p>This option defines the output filename prefix for all files generated by vcftools. For example, if &lt;prefix&gt; is set to output_filename, then all output files will be of the form output_filename.*** . If this option is omitted, all output files will have the prefix 'out.'.</p>
  </dd>

</dl>

<h3>Site Filter Options</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--chr</strong> &lt;chromosom&gt;</p>
  </dt>
  <dd>
    <p>Only process sites with a chromosome identifier matching &lt;chromosome&gt;</p>
  </dd>
  <dt>
    <p><strong>--from-bp</strong> &lt;integer&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--to-bp</strong> &lt;integer&gt;</p>
  </dt>
  <dd>
    <p>These options define the physical range of sites will be processed. Sites outside of this range will be excluded. These options can only be used in conjunction with --chr.</p>
  </dd>
  <dt>
    <p><strong>--snp</strong> &lt;string&gt;</p>
  </dt>
  <dd>
    <p>Include SNP(s) with matching ID. This command can be used multiple times in order to include more than one SNP.</p>
  </dd>
  <dt>
    <p><strong>--snps</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Include a list of SNPs given in a file. The file should contain a list of SNP IDs, with one ID per line.</p>
  </dd>
  <dt>
    <p><strong>--exclude</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Exclude a list of SNPs given in a file. The file should contain a list of SNP IDs, with one ID per line.</p>
  </dd>
  <dt>
    <p><strong>--positions</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Include a set of sites on the basis of a list of positions. Each line of the input file should contain a (tab-separated) chromosome and position. The file should have a header line. Sites not included in the list are excluded.</p>
  </dd>
  <dt>
    <p><strong>--bed</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--exclude-bed</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Include or exclude a set of sites on the basis of a BED file. Only the first three columns (chrom, chromStart and chromEnd) are required. The BED file should have a header line.</p>
  </dd>
  <dt>
    <p><strong>--remove-filtered-all</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--remove-filtered</strong> &lt;sting&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--keep-filtered</strong> &lt;sting&gt;</p>
  </dt>
  <dd>
    <p>These options are used to filter sites on the basis of their FILTER flag. The first option removes all sites with a FILTER flag. The second option can be used to exclude sites with a specific filter flag. The third option can be used to select sites on the basis of specific filter flags. The second and third options can be used multiple times to specify multiple FILTERs. The --keep-filtered option is applied before the --remove-filtered option.</p>
  </dd>
  <dt>
    <p><strong>--minQ</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Include only sites with Quality above this threshold.</p>
  </dd>
  <dt>
    <p><strong>--min-meanDP</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--max-meanDP</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Include sites with mean Depth within the thresholds defined by these options.</p>
  </dd>
  <dt>
    <p><strong>--maf</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--max-maf</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Include only sites with Minor Allele Frequency within the specified range.</p>
  </dd>
  <dt>
    <p><strong>--non-ref-af</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--max-non-ref-af</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Include only sites with Non-Reference Allele Frequency within the specified range.</p>
  </dd>
  <dt>
    <p><strong>--hue</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Assesses sites for Hardy-Weinberg Equilibrium using an exact test, as defined by Wigginton, Cutler and Abecasis (2005). Sites with a p-value below the threshold defined by this option are taken to be out of HWE, and therefore excluded.</p>
  </dd>
  <dt>
    <p><strong>--geno</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Exclude sites on the basis of the proportion of missing data (defined to be between 0 and 1).</p>
  </dd>
  <dt>
    <p><strong>--min-alleles</strong> &lt;int&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--max-alleles</strong> &lt;int&gt;</p>
  </dt>
  <dd>
    <p>Include only sites with a number of alleles within the specified range. For example, to include only bi-allelic sites, one could use:</p>
<pre>
      vcftools --vcf file1.vcf --min-alleles 2 --max-alleles 2
</pre>

  </dd>
  <dt>
    <p><strong>--mask</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--invert-mask</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--mask-min</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Include sites on the basis of a FASTA-like file. The provided file contains a sequence of integer digits (between 0 and 9) for each position on a chromosome that specify if a site at that position should be filtered or not. An example mask file would look like:</p>
<pre>
      &gt;1
      0000011111222...
</pre>
<p>In this example, sites in the VCF file located within the first 5 bases of the start of chromosome 1 would be kept, whereas sites at position 6 onwards would be filtered out. The threshold integer that determines if sites are filtered or not is set using the --mask-min option, which defaults to 0. The chromosomes contained in the mask file must be sorted in the same order as the VCF file. The --mask option is used to specify the mask file to be used, whereas the --invert-mask option can be used to specify a mask file that will be inverted before being applied.</p>
  </dd>

</dl>

<h3>Individual Filters</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--indv</strong> &lt;string&gt;</p>
  </dt>
  <dd>
    <p>Specify an individual to be kept in the analysis. This option can be used multiple times to specify multiple individuals.</p>
  </dd>
  <dt>
    <p><strong>--keep</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Provide a file containing a list of individuals to include in subsequent a nalysis. Each individual ID (as defined in the VCF headerline) should be included on a separate line.</p>
  </dd>
  <dt>
    <p><strong>--remove-indv</strong> &lt;string&gt;</p>
  </dt>
  <dd>
    <p>Specify an individual to be removed from the analysis. This option can be used multiple times to specify multiple individuals. If the --indv option is also specified, then the --indv option is executed before the --remove-indv option.</p>
  </dd>
  <dt>
    <p><strong>--remove</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Provide a file containing a list of individuals to exclude in subsequent analysis. Each individual ID (as defined in the VCF headerline) should be included on a separate line. If both the --keep and the --remove options are used, then the --keep option is execute before the --remove option.</p>
  </dd>
  <dt>
    <p><strong>--mon-indv-meanDP</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--max-indv-meanDP</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Calculate the mean coverage on a per-individual basis. Only individuals with coverage within the range specified by these options are included in subsequent analyses.</p>
  </dd>
  <dt>
    <p><strong>--mind</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Specify the minimum call rate threshold for each individual.</p>
  </dd>
  <dt>
    <p><strong>--phased</strong></p>
  </dt>
  <dd>
    <p>First excludes all individuals having all genotypes unphased, and subsequently excludes all sites with unphased genotypes. The remaining data therefore consists of phased data only.</p>
  </dd>

</dl>

<h3>Genotype Filters</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--remove-filtered-geno-all</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--remove-filtered-geno</strong> &lt;string&gt;</p>
  </dt>
  <dd>
    <p>The first option removes all genotypes with a FILTER flag. The second option can be used to exclude genotypes with a specific filter flag.</p>
  </dd>
  <dt>
    <p><strong>--minGQ</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Exclude all genotypes with a quality below the threshold specified by this option (GQ).</p>
  </dd>
  <dt>
    <p><strong>--minDP</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>Exclude all genotypes with a sequencing depth below that specified by this option (DP)</p>
  </dd>

</dl>

<h3>Output Statistics</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--freq</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--counts</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--freq2</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--counts2</strong></p>
  </dt>
  <dd>
    <p>Output per-site frequency information. The --freq outputs the allele frequency in a file with the suffix '.frq'. The --counts option outputs a similar file with the suffix '.frq.count', that contains the raw allele counts at each site. The --freq2 and --count2 options are used to suppress allele information in the output file. In this case, the order of the freqs/counts depends on the numbering in the VCF file.</p>
  </dd>
  <dt>
    <p><strong>--depth</strong></p>
  </dt>
  <dd>
    <p>Generates a file containing the mean depth per individual. This file has the suffix '.idepth'.</p>
  </dd>
  <dt>
    <p><strong>--site-depth</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--site-mean-depth</strong></p>
  </dt>
  <dd>
    <p>Generates a file containing the depth per site. The --site-depth option outputs the depth for each site summed across individuals. This file has the suffix '.ldepth'. Likewise, the --site-mean-depth outputs the mean depth for each site, and the output file has the suffix '.ldepth.mean'.</p>
  </dd>
  <dt>
    <p><strong>--geno-depth</strong></p>
  </dt>
  <dd>
    <p>Generates a (possibly very large) file containing the depth for each genotype in the VCF file. Missing entries are given the value -1. The file has the suffix '.gdepth'.</p>
  </dd>
  <dt>
    <p><strong>--site-quality</strong></p>
  </dt>
  <dd>
    <p>Generates a file containing the per-site SNP quality, as found in the QUAL column of the VCF file. This file has the suffix '.lqual'.</p>
  </dd>
  <dt>
    <p><strong>--het</strong></p>
  </dt>
  <dd>
    <p>Calculates a measure of heterozygosity on a per-individual basis. Specfically, the inbreeding coefficient, F, is estimated for each individual using a method of moments. The resulting file has the suffix '.het'.</p>
  </dd>
  <dt>
    <p><strong>--hardy</strong></p>
  </dt>
  <dd>
    <p>Reports a p-value for each site from a Hardy-Weinberg Equilibrium test (as defined by Wigginton, Cutler and Abecasis (2005)). The resulting file (with suffix '.hwe') also contains the Observed numbers of Homozygotes and Heterozygotes and the corresponding Expected numbers under HWE.</p>
  </dd>
  <dt>
    <p><strong>--missing</strong></p>
  </dt>
  <dd>
    <p>Generates two files reporting the missingness on a per-individual and per-site basis. The two files have suffixes '.imiss' and '.lmiss' respectively.</p>
  </dd>
  <dt>
    <p><strong>--hap-r2</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--geno-r2</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--ld-window</strong> &lt;int&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--ld-window-bp</strong> &lt;int&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--min-r2</strong> &lt;float&gt;</p>
  </dt>
  <dd>
    <p>These options are used to report Linkage Disequilibrium (LD) statistics as summarised by the r2 statistic. The --hap-r2 option informs vcftools to output a file reporting the r2 statistic using phased haplotypes. This is the traditional measure of LD often reported in the population genetics literature. If phased haplotypes are unavailable then the --geno-r2 option may be used, which calculates the squared correlation coefficient between genotypes encoded as 0, 1 and 2 to represent the number of non-reference alleles in each individual. This is the same as the LD measure reported by PLINK. The haplotype version outputs a file with the suffix '.hap.ld', whereas the genotype version outputs a file with the suffix '.geno.ld'. The haplotype version implies the option --phased.</p><p>The --ld-window option defines the maximum SNP separation for the calculation of LD. Likewise, the --ld-window-bp option can be used to define the maximum physical separation of SNPs included in the LD calculation. Finally, the --min-r2 sets a minimum value for r2 below which the LD statistic is not reported.</p>
  </dd>
  <dt>
    <p><strong>--SNPdnsity</strong> &lt;int&gt;</p>
  </dt>
  <dd>
    <p>Calculates the number and density of SNPs in bins of size defined by this option. The resulting output file has the suffix '.snpden'.</p>
  </dd>
  <dt>
    <p><strong>--TsTv</strong> &lt;int&gt;</p>
  </dt>
  <dd>
    <p>Calculates the Transition / Transversion ratio in bins of size defined by this option. The resulting output file has the suffix '.TsTv'. A summary is also supplied in a file with the suffix '.TsTv.summary'.</p>
  </dd>
  <dt>
    <p><strong>--FILTER-summary</strong></p>
  </dt>
  <dd>
    <p>Generates a summary of the number of SNPs and Ts/Tv ratio for each FILTER category. The output file has the suffix '.FILTER.summary.</p>
  </dd>
  <dt>
    <p><strong>--filtered-sites</strong></p>
  </dt>
  <dd>
    <p>Creates two files listing sites that have been kept or removed after filtering. The first file, with suffix '.kept.sites', lists sites kept by vcftools after filters have been applied. The second file, with the suffix '.removed.sites', list sites removed by the applied filters.</p>
  </dd>
  <dt>
    <p><strong>--singletons</strong></p>
  </dt>
  <dd>
    <p>This option will generate a file detailing the location of singletons, and the individual they occur in. The file reports both true singletons, and private doubletons (i.e. SNPs where the minor allele only occurs in a single individual and that individual is homozygotic for that allele). The output file has the suffix '.singletons'.</p>
  </dd>
  <dt>
    <p><strong>--site-pi</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--window-pi</strong> &lt;int&gt;</p>
  </dt>
  <dd>
    <p>These options are used to estimate levels of nucleotide diversity. The first option does this on a per-site basis, and the output file has the suffix '.sites.pi'. The second option calculates the nucleotide diversity in windows, with the window size defined in the option argument. Output for this option has the suffix '.windowed.pi'. The windowed version requires phased data, and hence use of this option implies the --phased option.</p>
  </dd>

</dl>

<h3>Output in Other Formats</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--O12</strong></p>
  </dt>
  <dd>
    <p>This option outputs the genotypes as a large matrix. Three files are produced. The first, with suffix '.012', contains the genotypes of each individual on a separate line. Genotypes are represented as 0, 1 and 2, where the number represent that number of non-reference alleles. Missing genotypes are represented by -1. The second file, with suffix '.012.indv' details the individuals included in the main file. The third file, with suffix '.012.pos' details the site locations included in the main file.</p>
  </dd>
  <dt>
    <p><strong>--IMPUTE</strong></p>
  </dt>
  <dd>
    <p>This option outputs phased haplotypes in IMPUTE reference-panel format. As IMPUTE requires phased data, using this option also implies --phased. Unphased individuals and genotypes are therefore excluded. Only bi-allelic sites are included in the output. Using this option generates three files. The IMPUTE haplotype file has the suffix '.impute.hap', and the IMPUTE legend file has the suffix '.impute.hap.legend'. The third file, with suffix '.impute.hap.indv', details the individuals included in the haplotype file, although this file is not needed by IMPUTE.</p>
  </dd>
  <dt>
    <p><strong>--ldhat</strong></p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--ldhat-geno</strong></p>
  </dt>
  <dd>
    <p>These options output data in LDhat format. Use of these options  also require the --chr option to by used. The --ldhat option outputs phased data only, and therefore also implies --phased, leading to unphased individuals and genotypes being excluded. Alternatively, the --ldhat-geno option treats all of the data as unphased, and therefore outputs LDhat files in genotype/unphased format. In either case, two files are generated with the suffixes '.ldhat.sites' and '.ldhat.locs', which correspond to the LDhat 'sites' and 'locs' input files respectively.</p>
  </dd>
  <dt>
    <p><strong>--BEAGLE-GL</strong></p>
  </dt>
  <dd>
    <p>This option outputs genotype likelihood information for input into the BEAGLE program. This option requires the VCF file to contain the FORMAT GL tag, which can generally be output by SNP callers such as the GATK. Use of this option requires a chromosome to be specified via the --chr option. The resulting output file (with the suffix '.BEAGLE.GL') contains genotype likelihoods for biallelic sites, and is suitable for input into BEAGLE via the 'like=' argument.</p>
  </dd>
  <dt>
    <p><strong>--plink</strong></p>
  </dt>
  <dd>
    <p>This option outputs the genotype data in PLINK PED format. Two files are generated, with suffixes '.ped' and '.map'. Note that only bi-allelic loci will be output. Further details of these files can be found in the PLINK documentation.</p><p>Note: This option can be very slow on large datasets. Using the --chr option to divide up the dataset is advised.</p>
  </dd>
  <dt>
    <p><strong>--plink-tped</strong></p>
  </dt>
  <dd>
    <p>The --plink option above can be extremely slow on large datasets. An alternative that might be considerably quicker is to output in the PLINK transposed format. This can be achieved using the --plink-tped option, which produces two files with suffixes '.tped' and '.tfam'.</p>
  </dd>
  <dt>
    <p><strong>--recode</strong></p>
  </dt>
  <dd>
    <p>The --recode option is used to generate a VCF file from the input VCF file having applied the options specified by the user. The output file has the suffix '.recode.vcf'.</p><p>By default, the INFO fields are removed from the output file, as the INFO values may be invalidated by the recoding (e.g. the total depth may need to be recalculated if individuals are removed). This default functionality can be overridden by using the --keep-INFO &lt;string&gt; option, where &lt;string&gt; defines the INFO key to keep in the output file. The --keep-INFO flag can be used multiple times. Alternatively, the option --keep-INFO-all can be used to retain all INFO fields.</p>
  </dd>

</dl>

<h3>Miscellaneous</h3>

<dl class='dl-vertical'>
  <dt>
    <p><strong>--extract-FORMAT-info</strong> &lt;string&gt;</p>
  </dt>
  <dd>
    <p>Extract information from the genotype fields in the VCF file relating to a specfied FORMAT identifier. For example, using the option '--extract-FORMAT-info GT' would extract the all of the GT (i.e. Genotype) entries. The resulting output file has the suffix '.&lt;FORMAT_ID&gt;.FORMAT'.</p>
  </dd>
  <dt>
    <p><strong>--get-INFO</strong> &lt;string&gt;</p>
  </dt>
  <dd>
    <p>This option is used to extract information from the INFO field in the VCF file. The &lt;string&gt; argument specifies the INFO tag to be extracted, and the option can be used multiple times in order to extract multiple INFO entries. The resulting file, with suffix '.INFO', contains the required INFO information in a tab-separated table. For example, to extract the NS and DB flags, one would use the command:</p>
<pre>
      vcftools --vcf file1.vcf --get-INFO NS --get-INFO DB
</pre>

  </dd>

</dl>

<h3>VCF File Comparison Options</h3>
<p>The file comparison options are currently in a state of flux and likely buggy. If you find a bug, please report it. Note that genotype-level filters are not supported in these options.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--diff</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--gzdiff</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Select a VCF file for comparison with the file specified by the --vcf option. Outputs two files describing the sites and individuals common / unique to each file. These files have the suffixes '.diff.sites_in_files' and '.diff.indv_in_files' respectively. The --gzdiff version can be used to read compressed VCF files.</p>
  </dd>
  <dt>
    <p><strong>--diff-site-discordance</strong></p>
  </dt>
  <dd>
    <p>Used in conjunction with the --diff option to calculate discordance on a site by site basis. The resulting output file has the suffix '.diff.sites'.</p>
  </dd>
  <dt>
    <p><strong>--diff-indv-discordance</strong></p>
  </dt>
  <dd>
    <p>Used in conjunction with the --diff option to calculate discordance on a per-individual basis. The resulting output file has the suffix '.diff.indv'.</p>
  </dd>
  <dt>
    <p><strong>--diff-discordance-matrix</strong></p>
  </dt>
  <dd>
    <p>Used in conjunction with the --diff option to calculate a discordance matrix. This option only works with bi-allelic loci with matching alleles that are present in both files. The resulting output file has the suffix '.diff.discordance.matrix'.</p>
  </dd>
  <dt>
    <p><strong>--diff-switch-error</strong></p>
  </dt>
  <dd>
    <p>Used in conjunction with the --diff option to calculate phasing errors (specifically 'switch errors'). This option generates two output files describing switch errors found between sites, and the average switch error per individual. These two files have the suffixes '.diff.switch' and '.diff.indv.switch' respectively.</p>
  </dd>

</dl>

<h3>Options still in development</h3>
<p>The following options are yet to be finalised, are likely to contain bugs, and are likely to change in the future.</p>
<dl class='dl-vertical'>
  <dt>
    <p><strong>--fst</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    
  </dd>
  <dt>
    <p><strong>--gzfst</strong> &lt;filename&gt;</p>
  </dt>
  <dd>
    <p>Calculate FST for a pair of VCF files, with the second file being specified by this option. FST is currently calculated using the formula described in the supplementary material of the Phase I HapMap paper. Currently, only pairwise FST calculations are supported, although this will likely change in the future. The --gzfst option can be used to read compressed VCF files.</p>
  </dd>
  <dt>
    <p><strong>--LROH</strong></p>
  </dt>
  <dd>
    <p>Identify Long Runs of Homozygosity.</p>
  </dd>
  <dt>
    <p><strong>--relatedness</strong></p>
  </dt>
  <dd>
    <p>Output Individual Relatedness Statistics.</p>
  </dd>

</dl>


        </div>
      </section>
<nav>
  <ul class="pager">
   <li class="previous"><a href="vcf2birthday.1.html"><span aria-hidden="true">&larr;</span> vcf2birthday.1: Convert vcard files to birthday database</a></li>
   <li class="next"><a href="vcsh.1.html">vcsh.1: Version control system for $home - multiple git repositories in $home <span aria-hidden="true">&rarr;</span></a></li>
  </ul>
</nav>

  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
