<!DOCTYPE html>
<html lang="en">
<head>
  <link href="https://fonts.googleapis.com/css?family=Fira+Mono:400,700&effect=destruction%7Cshadow-multiple" rel="stylesheet" type="text/css">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Man Pages in genometools</title>
  <link href="/css/bootstrap.min.css" rel="stylesheet">
  <link href="/css/manpage.css" rel="stylesheet">
  <link rel="apple-touch-icon" sizes="57x57" href="/icons/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/icons/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/icons/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/icons/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/icons/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/icons/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/icons/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/icons/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/icons/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/icons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/icons/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/icons/favicon-16x16.png">
  <link rel="manifest" href="/icons/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/icons/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">
  <meta name="description" content="Man Pages in genometools">
</head>
<body>
  <div class="container final">
          <div class="page-header">
      <h1 class="font-effect-destruction"><small>Man Pages in</small> genometools</h1>
      </div>

    <ol class="breadcrumb" itemscope itemtype="http://schema.org/BreadcrumbList">
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/">
      <span itemprop="name">Carta.tech</span>
    </a>
    <meta itemprop="position" content="1" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/">
      <span itemprop="name">Packages</span>
    </a>
    <meta itemprop="position" content="2" />
  </li>
  <li itemprop="itemListElement" itemscope itemtype="http://schema.org/ListItem">
    <a itemscope itemtype="http://schema.org/Thing" itemprop="item" href="/packages/genometools/">
      <span itemprop="name">genometools</span>
    </a>
    <meta itemprop="position" content="3" />
  </li>
</ol>
    <ul class="list-group">
<li class="list-group-item">
<span class="badge">91</span>
<h4>Section 1: <a href="https://www.carta.tech/man-pages/man1/">Executable programs or shell commands</a></h4>

<dl class="dl-vertical">
 <dt><a href="/man-pages/man1/gt.1.html"><strong>gt.1</strong></a></dt>

 <dd>The genometools genome analysis system.</dd>

 <dt><a href="/man-pages/man1/gt-bed_to_gff3.1.html"><strong>gt-bed_to_gff3.1</strong></a></dt>

 <dd>Parse bed file and convert it to gff3.</dd>

 <dt><a href="/man-pages/man1/gt-cds.1.html"><strong>gt-cds.1</strong></a></dt>

 <dd>Add cds (coding sequence) features to exon features given in gff3 file.</dd>

 <dt><a href="/man-pages/man1/gt-chain2dim.1.html"><strong>gt-chain2dim.1</strong></a></dt>

 <dd>Chain pairwise matches.</dd>

 <dt><a href="/man-pages/man1/gt-chseqids.1.html"><strong>gt-chseqids.1</strong></a></dt>

 <dd>Change sequence ids by the mapping given in a mapping file.</dd>

 <dt><a href="/man-pages/man1/gt-clean.1.html"><strong>gt-clean.1</strong></a></dt>

 <dd>Remove all files in the current directory which are automatically created by gt.</dd>

 <dt><a href="/man-pages/man1/gt-compreads.1.html"><strong>gt-compreads.1</strong></a></dt>

 <dd>Call a fastq file compression tool.</dd>

 <dt><a href="/man-pages/man1/gt-compreads-compress.1.html"><strong>gt-compreads-compress.1</strong></a></dt>

 <dd>Generates compact encoding for fastq data.</dd>

 <dt><a href="/man-pages/man1/gt-compreads-decompress.1.html"><strong>gt-compreads-decompress.1</strong></a></dt>

 <dd>Decodes a file of compressed reads.</dd>

 <dt><a href="/man-pages/man1/gt-compreads-refcompress.1.html"><strong>gt-compreads-refcompress.1</strong></a></dt>

 <dd>Generates compact encoding for fastq data using reference compressed reads (rcr).</dd>

 <dt><a href="/man-pages/man1/gt-compreads-refdecompress.1.html"><strong>gt-compreads-refdecompress.1</strong></a></dt>

 <dd>Decodes a given rcr (reference compressed reads).</dd>

 <dt><a href="/man-pages/man1/gt-condenser.1.html"><strong>gt-condenser.1</strong></a></dt>

 <dd>Call one of the condenser tools to prepare or manipulate redundancie compressed genomic data.</dd>

 <dt><a href="/man-pages/man1/gt-congruence.1.html"><strong>gt-congruence.1</strong></a></dt>

 <dd>Call a congruence subtool and pass argument(s) to it.</dd>

 <dt><a href="/man-pages/man1/gt-congruence-spacedseed.1.html"><strong>gt-congruence-spacedseed.1</strong></a></dt>

 <dd>Match spaced seeds.</dd>

 <dt><a href="/man-pages/man1/gt-convertseq.1.html"><strong>gt-convertseq.1</strong></a></dt>

 <dd>Parse and convert sequence file formats.</dd>

 <dt><a href="/man-pages/man1/gt-csa.1.html"><strong>gt-csa.1</strong></a></dt>

 <dd>Transform spliced alignments from gff3 file into consensus spliced alignments.</dd>

 <dt><a href="/man-pages/man1/gt-dot.1.html"><strong>gt-dot.1</strong></a></dt>

 <dd>Prints feature graphs in dotfile format.</dd>

 <dt><a href="/man-pages/man1/gt-dupfeat.1.html"><strong>gt-dupfeat.1</strong></a></dt>

 <dd>Duplicate internal feature nodes in given gff3 files.</dd>

 <dt><a href="/man-pages/man1/gt-encseq.1.html"><strong>gt-encseq.1</strong></a></dt>

 <dd>Call an encoded sequence manipulation tool and pass argument(s) to it.</dd>

 <dt><a href="/man-pages/man1/gt-encseq-bench.1.html"><strong>gt-encseq-bench.1</strong></a></dt>

 <dd>Perform benchmark on extractions from encseq.</dd>

 <dt><a href="/man-pages/man1/gt-encseq-bitextract.1.html"><strong>gt-encseq-bitextract.1</strong></a></dt>

 <dd>Extracts internal data from encoded sequences.</dd>

 <dt><a href="/man-pages/man1/gt-encseq-check.1.html"><strong>gt-encseq-check.1</strong></a></dt>

 <dd>Check the consistency of an encoded sequence file.</dd>

 <dt><a href="/man-pages/man1/gt-encseq-decode.1.html"><strong>gt-encseq-decode.1</strong></a></dt>

 <dd>Decode/extract encoded sequences.</dd>

 <dt><a href="/man-pages/man1/gt-encseq-encode.1.html"><strong>gt-encseq-encode.1</strong></a></dt>

 <dd>Encode sequence files efficiently.</dd>

 <dt><a href="/man-pages/man1/gt-encseq-info.1.html"><strong>gt-encseq-info.1</strong></a></dt>

 <dd>Display meta-information about an encoded sequence.</dd>

 <dt><a href="/man-pages/man1/gt-encseq-md5.1.html"><strong>gt-encseq-md5.1</strong></a></dt>

 <dd>Display md5 sums for an encoded sequence.</dd>

 <dt><a href="/man-pages/man1/gt-encseq-sample.1.html"><strong>gt-encseq-sample.1</strong></a></dt>

 <dd>Decode/extract encoded sequences by random choice.</dd>

 <dt><a href="/man-pages/man1/gt-encseq2spm.1.html"><strong>gt-encseq2spm.1</strong></a></dt>

 <dd>Compute suffix prefix matches from encoded sequence.</dd>

 <dt><a href="/man-pages/man1/gt-eval.1.html"><strong>gt-eval.1</strong></a></dt>

 <dd>Compare annotation files and show accuracy measures (prediction vs. reference).</dd>

 <dt><a href="/man-pages/man1/gt-extractfeat.1.html"><strong>gt-extractfeat.1</strong></a></dt>

 <dd>Extract features given in gff3 file from sequence file.</dd>

 <dt><a href="/man-pages/man1/gt-extractseq.1.html"><strong>gt-extractseq.1</strong></a></dt>

 <dd>Extract sequences from given sequence file(s) or fastaindex.</dd>

 <dt><a href="/man-pages/man1/gt-fastq_sample.1.html"><strong>gt-fastq_sample.1</strong></a></dt>

 <dd>Print samples by random choice from given fastq files using at least n sequence-chars. output is fastq/fasta format depending on whether qualities are available.</dd>

 <dt><a href="/man-pages/man1/gt-featureindex.1.html"><strong>gt-featureindex.1</strong></a></dt>

 <dd>Retrieve annotations from a persistent feature index as gff3 output.</dd>

 <dt><a href="/man-pages/man1/gt-fingerprint.1.html"><strong>gt-fingerprint.1</strong></a></dt>

 <dd>Compute md5 fingerprints for each sequence given in a set of sequence files.</dd>

 <dt><a href="/man-pages/man1/gt-genomediff.1.html"><strong>gt-genomediff.1</strong></a></dt>

 <dd>Calculates kr: pairwise distances between genomes.</dd>

 <dt><a href="/man-pages/man1/gt-gff3.1.html"><strong>gt-gff3.1</strong></a></dt>

 <dd>Parse, possibly transform, and output gff3 files.</dd>

 <dt><a href="/man-pages/man1/gt-gff3_to_gtf.1.html"><strong>gt-gff3_to_gtf.1</strong></a></dt>

 <dd>Parse gff3 file(s) and show them as gtf2.2.</dd>

 <dt><a href="/man-pages/man1/gt-gff3validator.1.html"><strong>gt-gff3validator.1</strong></a></dt>

 <dd>Strictly validate given gff3 files.</dd>

 <dt><a href="/man-pages/man1/gt-gtf_to_gff3.1.html"><strong>gt-gtf_to_gff3.1</strong></a></dt>

 <dd>Parse gtf2.2 file and convert it to gff3.</dd>

 <dt><a href="/man-pages/man1/gt-hop.1.html"><strong>gt-hop.1</strong></a></dt>

 <dd>Cognate sequence-based homopolymer error correction.</dd>

 <dt><a href="/man-pages/man1/gt-id_to_md5.1.html"><strong>gt-id_to_md5.1</strong></a></dt>

 <dd>Change sequence ids in given gff3 files to md5 fingerprints of the corresponding sequences.</dd>

 <dt><a href="/man-pages/man1/gt-interfeat.1.html"><strong>gt-interfeat.1</strong></a></dt>

 <dd>Add intermediary features between outside features in given gff3 file(s).</dd>

 <dt><a href="/man-pages/man1/gt-loccheck.1.html"><strong>gt-loccheck.1</strong></a></dt>

 <dd>Checks parent-child containment in gff3 input.</dd>

 <dt><a href="/man-pages/man1/gt-ltrclustering.1.html"><strong>gt-ltrclustering.1</strong></a></dt>

 <dd>Cluster features of ltrs.</dd>

 <dt><a href="/man-pages/man1/gt-ltrdigest.1.html"><strong>gt-ltrdigest.1</strong></a></dt>

 <dd>Identifies and annotates sequence features in ltr retrotransposon candidates.</dd>

 <dt><a href="/man-pages/man1/gt-ltrharvest.1.html"><strong>gt-ltrharvest.1</strong></a></dt>

 <dd>Predict ltr retrotransposons.</dd>

 <dt><a href="/man-pages/man1/gt-matchtool.1.html"><strong>gt-matchtool.1</strong></a></dt>

 <dd>Parse match formats and/or invoke matching tools.</dd>

 <dt><a href="/man-pages/man1/gt-matstat.1.html"><strong>gt-matstat.1</strong></a></dt>

 <dd>Compute matching statistics.</dd>

 <dt><a href="/man-pages/man1/gt-md5_to_id.1.html"><strong>gt-md5_to_id.1</strong></a></dt>

 <dd>Change md5 fingerprints used as sequence ids in given gff3 files to ``regular ones.</dd>

 <dt><a href="/man-pages/man1/gt-merge.1.html"><strong>gt-merge.1</strong></a></dt>

 <dd>Merge sorted gff3 files in sorted fashion.</dd>

 <dt><a href="/man-pages/man1/gt-mergefeat.1.html"><strong>gt-mergefeat.1</strong></a></dt>

 <dd>Merge adjacent features without children of the same type in given gff3 file(s).</dd>

 <dt><a href="/man-pages/man1/gt-mkfeatureindex.1.html"><strong>gt-mkfeatureindex.1</strong></a></dt>

 <dd>Creates a new featureindex from annotation data.</dd>

 <dt><a href="/man-pages/man1/gt-mmapandread.1.html"><strong>gt-mmapandread.1</strong></a></dt>

 <dd>Map the supplied files into memory and read them once.</dd>

 <dt><a href="/man-pages/man1/gt-orffinder.1.html"><strong>gt-orffinder.1</strong></a></dt>

 <dd>Identifies orfs (open reading frames) in sequences.</dd>

 <dt><a href="/man-pages/man1/gt-packedindex.1.html"><strong>gt-packedindex.1</strong></a></dt>

 <dd>Call apacked index subtool and pass argument(s) to it.</dd>

 <dt><a href="/man-pages/man1/gt-prebwt.1.html"><strong>gt-prebwt.1</strong></a></dt>

 <dd>Precompute bwt-bounds for some prefix length.</dd>

 <dt><a href="/man-pages/man1/gt-readjoiner.1.html"><strong>gt-readjoiner.1</strong></a></dt>

 <dd>Readjoiner: a string graph-based sequence assembler.</dd>

 <dt><a href="/man-pages/man1/gt-readjoiner-assembly.1.html"><strong>gt-readjoiner-assembly.1</strong></a></dt>

 <dd>Construct string graph and output contigs.</dd>

 <dt><a href="/man-pages/man1/gt-readjoiner-overlap.1.html"><strong>gt-readjoiner-overlap.1</strong></a></dt>

 <dd>Compute suffix prefix matches from encoded sequence.</dd>

 <dt><a href="/man-pages/man1/gt-readjoiner-prefilter.1.html"><strong>gt-readjoiner-prefilter.1</strong></a></dt>

 <dd>Remove contained and low-quality reads and encode read set in gtencseq format.</dd>

 <dt><a href="/man-pages/man1/gt-repfind.1.html"><strong>gt-repfind.1</strong></a></dt>

 <dd>Compute maximal repeats.</dd>

 <dt><a href="/man-pages/man1/gt-scriptfilter.1.html"><strong>gt-scriptfilter.1</strong></a></dt>

 <dd>Get info about and validate lua script filters.</dd>

 <dt><a href="/man-pages/man1/gt-select.1.html"><strong>gt-select.1</strong></a></dt>

 <dd>Select certain features (specified by the used options) from given gff3 file(s).</dd>

 <dt><a href="/man-pages/man1/gt-seq.1.html"><strong>gt-seq.1</strong></a></dt>

 <dd>Parse the given sequence file(s) and construct the corresponding index files.</dd>

 <dt><a href="/man-pages/man1/gt-seqfilter.1.html"><strong>gt-seqfilter.1</strong></a></dt>

 <dd>Filter the given sequence file(s) and show the results on stdout.</dd>

 <dt><a href="/man-pages/man1/gt-seqids.1.html"><strong>gt-seqids.1</strong></a></dt>

 <dd>Show sequence ids from annotation file.</dd>

 <dt><a href="/man-pages/man1/gt-seqmutate.1.html"><strong>gt-seqmutate.1</strong></a></dt>

 <dd>Mutate the sequences of the given sequence file(s).</dd>

 <dt><a href="/man-pages/man1/gt-seqorder.1.html"><strong>gt-seqorder.1</strong></a></dt>

 <dd>Output sequences as multifasta in specified order.</dd>

 <dt><a href="/man-pages/man1/gt-seqstat.1.html"><strong>gt-seqstat.1</strong></a></dt>

 <dd>Calculate statistics for fasta file(s).</dd>

 <dt><a href="/man-pages/man1/gt-seqtransform.1.html"><strong>gt-seqtransform.1</strong></a></dt>

 <dd>Perform simple transformations on the given sequence file(s).</dd>

 <dt><a href="/man-pages/man1/gt-seqtranslate.1.html"><strong>gt-seqtranslate.1</strong></a></dt>

 <dd>Translates a nucleotide sequence into a protein sequence.</dd>

 <dt><a href="/man-pages/man1/gt-sequniq.1.html"><strong>gt-sequniq.1</strong></a></dt>

 <dd>Filter out repeated sequences in given sequence files.</dd>

 <dt><a href="/man-pages/man1/gt-shredder.1.html"><strong>gt-shredder.1</strong></a></dt>

 <dd>Shredder sequence file(s) into consecutive pieces of random length.</dd>

 <dt><a href="/man-pages/man1/gt-shulengthdist.1.html"><strong>gt-shulengthdist.1</strong></a></dt>

 <dd>Compute distribution of pairwise shustring lengths.</dd>

 <dt><a href="/man-pages/man1/gt-simreads.1.html"><strong>gt-simreads.1</strong></a></dt>

 <dd>Simulate sequencing reads from random positions in the input sequence(s).</dd>

 <dt><a href="/man-pages/man1/gt-sketch.1.html"><strong>gt-sketch.1</strong></a></dt>

 <dd>Create graphical representation of gff3 annotation files.</dd>

 <dt><a href="/man-pages/man1/gt-sketch_page.1.html"><strong>gt-sketch_page.1</strong></a></dt>

 <dd>Draw a multi-page pdf/ps representation of an annotation file.</dd>

 <dt><a href="/man-pages/man1/gt-snpper.1.html"><strong>gt-snpper.1</strong></a></dt>

 <dd>Annotates snps according to their effect on the genome as given by a genomic annotation.</dd>

 <dt><a href="/man-pages/man1/gt-speck.1.html"><strong>gt-speck.1</strong></a></dt>

 <dd>Checks spec definition compliance in gff3 input.</dd>

 <dt><a href="/man-pages/man1/gt-splicesiteinfo.1.html"><strong>gt-splicesiteinfo.1</strong></a></dt>

 <dd>Show information about splice sites given in gff3 files.</dd>

 <dt><a href="/man-pages/man1/gt-splitfasta.1.html"><strong>gt-splitfasta.1</strong></a></dt>

 <dd>Split the supplied fasta file.</dd>

 <dt><a href="/man-pages/man1/gt-stat.1.html"><strong>gt-stat.1</strong></a></dt>

 <dd>Show statistics about features contained in gff3 files.</dd>

 <dt><a href="/man-pages/man1/gt-tagerator.1.html"><strong>gt-tagerator.1</strong></a></dt>

 <dd>Map short sequence tags in given index.</dd>

 <dt><a href="/man-pages/man1/gt-tallymer.1.html"><strong>gt-tallymer.1</strong></a></dt>

 <dd>Call a tallymer subtool and pass argument(s) to it.</dd>

 <dt><a href="/man-pages/man1/gt-tallymer-mkindex.1.html"><strong>gt-tallymer-mkindex.1</strong></a></dt>

 <dd>Count and index k-mers in the given enhanced suffix array for a fixed value of k.</dd>

 <dt><a href="/man-pages/man1/gt-tallymer-occratio.1.html"><strong>gt-tallymer-occratio.1</strong></a></dt>

 <dd>Compute occurrence ratio for a set of sequences represented by an enhanced suffix array.</dd>

 <dt><a href="/man-pages/man1/gt-tallymer-search.1.html"><strong>gt-tallymer-search.1</strong></a></dt>

 <dd>Search a set of k-mers in an index constructed by ``gt tallymer mkindex.</dd>

 <dt><a href="/man-pages/man1/gt-tirvish.1.html"><strong>gt-tirvish.1</strong></a></dt>

 <dd>Identify terminal inverted repeat (tir) elements,such as dna transposons.</dd>

 <dt><a href="/man-pages/man1/gt-uniq.1.html"><strong>gt-uniq.1</strong></a></dt>

 <dd>Filter out repeated feature node graphs in a sorted gff3 file.</dd>

 <dt><a href="/man-pages/man1/gt-uniquesub.1.html"><strong>gt-uniquesub.1</strong></a></dt>

 <dd>Compute length of minimum unique prefixes.</dd>

 <dt><a href="/man-pages/man1/gt-wtree.1.html"><strong>gt-wtree.1</strong></a></dt>

 <dd>Call an wtree manipulation tool and pass argument(s) to it.</dd>

</dl>

</li>

</ul>
  </div>
  <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
  <!-- Include all compiled plugins (below), or include individual files as needed -->
  <script src="/js/bootstrap.min.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-60781387-1', 'auto');
    ga('send', 'pageview');

  </script>
</body>
</html>
